<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Object Oriented Tools &mdash; CanlabCore 1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/nature.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <link rel="top" title="CanlabCore 1.0 documentation" href="index.html" />
    <link rel="prev" title="CanlabCore documentation home" href="index.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="mat-modindex.html" title="MATLAB Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="index.html" title="CanlabCore documentation home"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">CanlabCore 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="toctree-wrapper compound">
<ul class="simple">
</ul>
</div>
<div class="section" id="object-oriented-tools">
<h1>Object Oriented Tools<a class="headerlink" href="#object-oriented-tools" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-&#64;canlab_dataset">
<span id="canlab-dataset"></span><h2>canlab_dataset<a class="headerlink" href="#module-@canlab_dataset" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="&#64;canlab_dataset.bars">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">bars</code><span class="sig-paren">(</span><em>obj</em>, <em>varnames</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.bars" title="Permalink to this definition">¶</a></dt>
<dd><p>Bar plot for canlab_dataset object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[dat, descrip, colors, h1, s1] = bars(obj, varnames, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">canlab_dataset object</p>
</dd>
<dt><strong>varnames:</strong></dt>
<dd><p class="first last">Cell string of variable names to plot</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><blockquote class="first">
<div><dl class="docutils">
<dt><strong>colors:</strong></dt>
<dd><p class="first last">defined colors (default are set by scn_standard_colors.m)</p>
</dd>
</dl>
</div></blockquote>
<dl class="docutils">
<dt><strong>nofig:</strong></dt>
<dd><p class="first last">do not generate figure</p>
</dd>
</dl>
<p>Takes any optional inputs to barplot_colored.m</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">data matrix for each variable</p>
</dd>
<dt><strong>descrip:</strong></dt>
<dd><p class="first last">the description for this variable</p>
</dd>
<dt><strong>colors:</strong></dt>
<dd><p class="first last">selected colors (default are set by scn_standard_colors.m)</p>
</dd>
<dt><strong>h1:</strong></dt>
<dd><p class="first last">figure handle</p>
</dd>
<dt><strong>s1:</strong></dt>
<dd><p class="first last">axis handle</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>create_figure(&#39;NPS values - All subjects&#39;);

varnames = {&#39;15&#39; &#39;13&#39; &#39;11&#39; &#39; 9&#39; &#39;16&#39; &#39;14&#39; &#39;12&#39; &#39;10&#39;};
xvals = [1 2 4 5 8 9 11 12];
colors = {[1 0 0] [0 1 0] [1 0 0] [0 1 0] [1 0 0] [0 1 0] [1 0 0] [0 1 0]};
bars(LevoNPS, varnames, &#39;x&#39;, xvals, &#39;colors&#39;, colors, &#39;XTickLabels&#39;, varnames, &#39;within&#39;, &#39;nofig&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.concatenate">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">concatenate</code><span class="sig-paren">(</span><em>D</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.concatenate" title="Permalink to this definition">¶</a></dt>
<dd><p>Concatenates Subject-level and Event-level data across all subjects</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[names ids dat] = concatenate(D, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">canlab_dataset object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>a logical array</strong></dt>
<dd><p class="first last">a vector of 1/0 values to use as wh_keep</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>names:</strong></dt>
<dd><p class="first last">cell array of variable names</p>
</dd>
<dt><strong>ids:</strong></dt>
<dd><p class="first last">subject IDs matching data rows in dat</p>
</dd>
<dt><strong>dat:</strong></dt>
<dd><dl class="first last docutils">
<dt>subjects*events x variables matrix</dt>
<dd><ul class="first last simple">
<li>subject number, event number are included</li>
<li>all subject-level and event-level data are included</li>
<li>this format appropriate for, e.g., SAS/HLM</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>descrips:</strong></dt>
<dd><p class="first last">cell array of variable descriptions</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[names, ids, flatdat] = concatenate(D);
id_numbers = flatdat(:, 1);

wh_subjs = true(size(D.Subj_Level.id));
wh_subjs([13 18 19]) = false;
[names, ids, dat] = concatenate(D, wh_subjs);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.get_var">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">get_var</code><span class="sig-paren">(</span><em>D</em>, <em>varname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.get_var" title="Permalink to this definition">¶</a></dt>
<dd><p>Get Subject-level or Event-level variable from dataset D and return in
rect matrix and cell array. Multiple variables can be requested, but 
all data requested must be either numeric or text, and not a combination of the two.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[dat, datcell, wh_level, descrip] = get_var(D, varname, [opt inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
<dt><strong>varname:</strong></dt>
<dd><dl class="first last docutils">
<dt>the name of a variable to get from dataset</dt>
<dd><ul class="first last simple">
<li>Looks for var name at either level, returns error if exists at both levels</li>
<li>can be a cell array of multiple var names
in this case, dat is a n x m matrix, where n=subjs and m=variables requested</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first"><strong>a logical array vector of 1/0 values to use as wh_keep</strong></p>
<dl class="docutils">
<dt><strong>conditional:</strong></dt>
<dd><p class="first last">to be followed by a cell array; the first cell is the name
of the variable to be conditionally selected upon, the second cell
contains the condition which must be met.
Example: get_var(D, &#8216;DeltaDon&#8217;, &#8216;conditional&#8217;, {&#8216;trained&#8217; 1})
will get DeltaDon whenever trained==1.  Currently only implemented for
event-level data.  Could be expanded to include multiple conditions.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dat:</strong></dt>
<dd><blockquote class="first">
<div><p>rect matrix of subjects X events (X variables)</p>
</div></blockquote>
<ul class="last simple">
<li>good for plotting individuals, means/std. errors across subjects</li>
<li>is actually a cell matrix if textual data is requested.</li>
</ul>
</dd>
<dt><strong>datcell:</strong></dt>
<dd><blockquote class="first">
<div><p>1 x subjects cell array, each cell containing event data for one subject</p>
</div></blockquote>
<ul class="last simple">
<li>good for input into some stats functions, e.g., glmfit_multilevel
and igls.m</li>
</ul>
</dd>
<dt><strong>wh_level:</strong></dt>
<dd><p class="first last">1 = &#8216;Subject&#8217;; 2 = &#8216;Event&#8217;;</p>
</dd>
<dt><strong>descrip:</strong></dt>
<dd><p class="first last">the description for this variable</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.glm">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">glm</code><span class="sig-paren">(</span><em>D</em>, <em>Yvarname</em>, <em>Xvarnames</em>, <em>wh_keep</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.glm" title="Permalink to this definition">¶</a></dt>
<dd><p>predict Y from X using GLM</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">out</span> <span class="o">=</span> <span class="n">glm</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="n">Yvarname</span><span class="p">,</span> <span class="n">Xvarnames</span><span class="p">,</span> <span class="n">wh_keep</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
<dt><strong>Yvarname:</strong></dt>
<dd><p class="first last">the name of a variable to predict. must be subject level</p>
</dd>
<dt><strong>Xvarnames:</strong></dt>
<dd><p class="first last">the name(s) of predictor variables. if multiple, put in
cell array. must be subject_level</p>
</dd>
<dt><strong>wh_keep:</strong></dt>
<dd><p class="first last">a logical vector of 1/0 values</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>out:</strong></dt>
<dd><p class="first last">structure containing same output as for glmfit()
out.b: a vector of coefficient estimates
out.dev: the deviance of the fit
out.stat: see glmfit documentation for stat structure fields</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">out</span> <span class="o">=</span> <span class="n">glm</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="s">&#39;DeltaDon_avg&#39;</span><span class="p">,</span> <span class="n">prednames</span><span class="p">,</span> <span class="n">wh_keep</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.glm_multilevel">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">glm_multilevel</code><span class="sig-paren">(</span><em>D</em>, <em>Yvarname</em>, <em>Xvarnames</em>, <em>wh_keep</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.glm_multilevel" title="Permalink to this definition">¶</a></dt>
<dd><p>Predict Y from X using GLM</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">b</span><span class="p">,</span> <span class="n">dev</span><span class="p">,</span> <span class="n">stat</span><span class="p">]</span> <span class="o">=</span> <span class="n">glm_multilevel</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="n">Yvarname</span><span class="p">,</span> <span class="n">Xvarnames</span><span class="p">,</span> <span class="n">wh_keep</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
<dt><strong>Yvarname:</strong></dt>
<dd><p class="first last">the name of a variable to predict. must be event level</p>
</dd>
<dt><strong>Xvarnames:</strong></dt>
<dd><p class="first last">the name(s) of predictor variables. if multiple, put in
cell array. must be event level</p>
</dd>
<dt><strong>wh_keep:</strong></dt>
<dd><p class="first last">a logical vector of 1/0 values</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>b:</strong></dt>
<dd><p class="first last">a vector of coefficient estimates (same as for glmfit())</p>
</dd>
<dt><strong>dev:</strong></dt>
<dd><p class="first last">the deviance of the fit (same as for glmfit())</p>
</dd>
<dt><strong>stat:</strong></dt>
<dd><p class="first last">structure containing stats fields (see glmfit() documentation)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.histogram">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">histogram</code><span class="sig-paren">(</span><em>D</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.histogram" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Histogram of one variable in dataset</dt>
<dd><ul class="first last simple">
<li>can be either event-level or subject-level</li>
<li>event-level data is plotted as concatenated events across subject-level</li>
<li>both variables must be valid names (case-sensitive)</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>fig_han = histogram(D, [optional inputs]);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">canlab_dataset</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first"><a href="#id1"><span class="problematic" id="id2">**</span></a>nofig&#8217;: suppress creation of new figure</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>fig_han:</strong></dt>
<dd><p class="first last">figure handle</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.mediation">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">mediation</code><span class="sig-paren">(</span><em>D</em>, <em>xvarname</em>, <em>yvarname</em>, <em>mvarname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.mediation" title="Permalink to this definition">¶</a></dt>
<dd><p>Run single or multilevel mediation analysis on a canlab_dataset object.
Calls mediation.m (see mediation.m) in mediation toolbox</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[paths, stats] = mediation(D, xvarname, yvarname, mvarname, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">is a canlab_dataset object</p>
</dd>
<dt><strong>xvarname:</strong></dt>
<dd><p class="first last">X, the initial variable (valid variable name in the dataset)</p>
</dd>
<dt><strong>yvarname:</strong></dt>
<dd><p class="first last">Y, the outcome variable (valid variable name in the dataset)</p>
</dd>
<dt><strong>mvarname:</strong></dt>
<dd><p class="first last">M, the potential mediator ((valid variable name in the dataset)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">Takes any optional inputs to mediation.m
e.g., &#8216;noverbose&#8217;, &#8216;dosave&#8217;, &#8216;names&#8217;, &#8216;M&#8217;, &#8216;L2M&#8217;, &#8216;covs&#8217;, others</p>
<dl class="docutils">
<dt><strong>wh_keep:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by 1/0 vector of subjects to keep.</dt>
<dd><ul class="first last simple">
<li>must be same length as subjects</li>
<li>subjects with value 0 will be excluded</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>rankdata:</strong></dt>
<dd><p class="first last">ranks all data before mediation; &#8220;Nonparametric&#8221;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>paths:</strong></dt>
<dd><p class="first last">see mediation.m from mediation toolbox</p>
</dd>
<dt><strong>stats:</strong></dt>
<dd><p class="first last">see mediation.m from mediation toolbox</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">paths</span><span class="p">,</span> <span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">mediation</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="s">&#39;Group&#39;</span><span class="p">,</span> <span class="s">&#39;DeltaDon&#39;</span><span class="p">,</span> <span class="s">&#39;DeltaDist&#39;</span><span class="p">,</span> <span class="s">&#39;M2&#39;</span><span class="p">,</span> <span class="s">&#39;DeltaTend&#39;</span><span class="p">,</span> <span class="s">&#39;wh_keep&#39;</span><span class="p">,</span> <span class="n">wh_keep</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.plot_var">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">plot_var</code><span class="sig-paren">(</span><em>D</em>, <em>varname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.plot_var" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot the mean and standard error of a variable across events.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[meandat, stedat] = plot_var(D, varname, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
<dt><strong>varname:</strong></dt>
<dd><dl class="first last docutils">
<dt>the name of a valid variable to get from dataset</dt>
<dd><p class="first last">-Looks for var name at either level, returns Event level if exists at both levels</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>subjtype:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by name of grouping variable</dt>
<dd><ul class="first last simple">
<li>must be categorical subject-level variable</li>
<li>if entered, plot lines or bars based on these categories</li>
<li>&#8216;eventmeans&#8217; will plot bars; without, it will plot line
plots across events with standard error shading</li>
<li>the grouping variable&#8217;s description, if it exists, will
be split along commas, and those values will be used as
column lables</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>eventmeans:</strong></dt>
<dd><dl class="first last docutils">
<dt>calculate and plot subject means across event-level variables</dt>
<dd><ul class="first last simple">
<li>if entered, will plot bar plots of means by condition</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>wh_keep:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by 1/0 vector of subjects to keep.</dt>
<dd><ul class="first last simple">
<li>must be same length as subjects</li>
<li>subjects with value 0 will be excluded</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>color:</strong></dt>
<dd><p class="first last">followed by one color for all bars, or cell array with names of colors cell for each line/bar</p>
</dd>
<dt><strong>nofig:</strong></dt>
<dd><p class="first last">don&#8217;t make a new figure</p>
</dd>
<dt><strong>other:</strong></dt>
<dd><p class="first last">other varargin are passed directly to barplot_columns.  So
for example, &#8216;95CI&#8217; will make 95% confidence interals, instead
of SE bars.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>meandat:</strong></dt>
<dd><p class="first last">mean values</p>
</dd>
<dt><strong>stedat:</strong></dt>
<dd><p class="first last">standard error values</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">plot_var</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="s">&#39;Frustration&#39;</span><span class="p">)</span>
<span class="n">plot_var</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="s">&#39;RT&#39;</span><span class="p">)</span>
<span class="n">plot_var</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="s">&#39;RT&#39;</span><span class="p">,</span> <span class="s">&#39;eventmeans&#39;</span><span class="p">);</span>
<span class="n">plot_var</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="s">&#39;RT&#39;</span><span class="p">,</span> <span class="s">&#39;subjtype&#39;</span><span class="p">,</span> <span class="s">&#39;Placebo&#39;</span><span class="p">);</span>
<span class="n">plot_var</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="s">&#39;RT&#39;</span><span class="p">,</span> <span class="s">&#39;eventmeans&#39;</span><span class="p">,</span> <span class="s">&#39;subjtype&#39;</span><span class="p">,</span> <span class="s">&#39;Placebo&#39;</span><span class="p">);</span>
<span class="n">plot_var</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="s">&#39;RT&#39;</span><span class="p">,</span> <span class="s">&#39;eventmeans&#39;</span><span class="p">,</span> <span class="s">&#39;subjtype&#39;</span><span class="p">,</span> <span class="s">&#39;Placebo&#39;</span><span class="p">,</span> <span class="s">&#39;color&#39;</span><span class="p">,</span> <span class="p">{</span><span class="s">&#39;r&#39;</span> <span class="s">&#39;b&#39;</span><span class="p">});</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.print_summary">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">print_summary</code><span class="sig-paren">(</span><em>D</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.print_summary" title="Permalink to this definition">¶</a></dt>
<dd><p>Prints summaries for every variable, or specified variables</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>print_summary(D, [optional inputs]) 
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>subj:</strong></dt>
<dd><p class="first last">followed by a cell array of subject level var names, to only see those vars</p>
</dd>
<dt><strong>event:</strong></dt>
<dd><p class="first last">followed by a cell array of event level var names, to only see those vars</p>
</dd>
</dl>
<p class="last">if either varargin is unspecified, all variables will be printed</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.read_from_excel">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">read_from_excel</code><span class="sig-paren">(</span><em>dat</em>, <em>ExperimentFileName</em>, <em>SubjectFileList</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.read_from_excel" title="Permalink to this definition">¶</a></dt>
<dd><p>Read from datafile into canlab_dataset format - currently requires file
extensions .xls or .xlsx, but in the future will use importdata to take
.csv or .txt extensions as well.</p>
<dl class="docutils">
<dt>Datafiles require column headers</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>Design file requires: id, names, units, descrip</dt>
<dd><ul class="first last">
<li><p class="first">other columns can be added</p>
</li>
<li><dl class="first docutils">
<dt>ONLY between subject columns identified in the &#8216;names&#8217;</dt>
<dd><p class="first last">column are added.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>See Sample_canlab_dataset_experiment_level.xlsx for an</dt>
<dd><p class="first last">example design file</p>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Subject files require no specific column headers, but all column</dt>
<dd><p class="first last">headers must be identical across all subjects.
- Enter NaN for data field in file if no value for that column within a specific Event
- ALL columns of subject files are written to canlab_dataset</p>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>dat = read_from_excel(dat, ExperimentFileName, SubjectFileList, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
<dt><strong>ExperimentFileName:</strong></dt>
<dd><p class="first last">the absolute path of the experiment data file</p>
</dd>
<dt><strong>SubjectFileList:</strong></dt>
<dd><dl class="first last docutils">
<dt>list of absolute paths for individual subject files</dt>
<dd><ul class="first last simple">
<li>plays well with filenames()</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>fmri:</strong></dt>
<dd><p class="first last">Indicates construction of canlab_dataset object using &#8216;fmri&#8217;
code. Suppresses overwrite warnings specific to &#8216;fmri&#8217; inputs.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">canlab_dataset object with uploaded values</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% To output a file into a raw fmri dataset

DesignFile = fullfile(pwd,&#39;Sample_canlab_dataset_experiment_level.xlsx&#39;);
SubjectFiles = filenames(fullfile(pwd,&#39;Sample_canlab_dataset_subject*.xlsx&#39;));
dat = canlab_dataset(&#39;fmri&#39;);
dat = read_from_excel(dat,DesignFile,SubjectFiles,&#39;fmri&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.scattermatrix">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">scattermatrix</code><span class="sig-paren">(</span><em>D</em>, <em>wh_level</em>, <em>wh_vars</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.scattermatrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Scatterplot matrix of pairwise event-level variables</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fig_han</span> <span class="o">=</span> <span class="n">scattermatrix</span><span class="p">(</span><span class="n">D</span><span class="p">,</span> <span class="n">wh_level</span><span class="p">,</span> <span class="n">wh_vars</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
<dt><strong>wh_level:</strong></dt>
<dd><p class="first last">1 (Subject) or 2 (Event)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>fig_han:</strong></dt>
<dd><p class="first last">figure handle</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>fig_han = scattermatrix(D);

wh = [5:9];
fig_han = scattermatrix(D, 2, wh);

f = scattermatrix(D, 2, {&#39;Choice&#39; &#39;RT&#39; &#39;Pain&#39; &#39;SwitchNext&#39; &#39;Frustration&#39; &#39;Anxiety&#39; &#39;Control&#39;});
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.scatterplot">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">scatterplot</code><span class="sig-paren">(</span><em>D</em>, <em>v1</em>, <em>v2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.scatterplot" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Scatterplot of two variables in dataset</dt>
<dd><ul class="first last simple">
<li>can be either event-level or subject-level</li>
<li>event-level data is plotted as multi-line plot, one line per subject</li>
<li>both variables must be valid names (case-sensitive)</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>fig_han = scatterplot(D, varname1, varname2, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
<dt><strong>v1:</strong></dt>
<dd><p class="first last">x variable</p>
</dd>
<dt><strong>v2:</strong></dt>
<dd><p class="first last">y variable</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>nofig:</strong></dt>
<dd><p class="first last">suppress creation of new figure</p>
</dd>
<dt><strong>subjtype:</strong></dt>
<dd><p class="first last">group by the following variable name</p>
</dd>
<dt><strong>wh_keep:</strong></dt>
<dd><p class="first last">followed by logical</p>
</dd>
<dt><strong>colors:</strong></dt>
<dd><p class="first last">followed by colors.</p>
</dd>
<dt><strong>dorobust:</strong></dt>
<dd><p class="first last">do robust corr.  if enabled, colors will not work and subjtype grouping will not work well until
the function plot_correlation_samefig is updated, at some point in the future.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>fig_han:</strong></dt>
<dd><p class="first last">figure handle</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>scatterplot(D, &#39;Anxiety&#39;, &#39;Frustration&#39;);
fig_han = scatterplot(D, D.Subj_Level.names{1}, D.Subj_Level.names{2});
scatterplot(D, D.Event_Level.names{1}, D.Event_Level.names{2});
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.spm2canlab_dataset">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">spm2canlab_dataset</code><span class="sig-paren">(</span><em>obj</em>, <em>subject</em>, <em>spm</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.spm2canlab_dataset" title="Permalink to this definition">¶</a></dt>
<dd><p>Extract Event_Level data from subjects&#8217; SPM.mat files to add data to
canlab_dataset object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">spm2canlab_dataset</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">subject</span><span class="p">,</span> <span class="n">spm</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">Canlab_dataset object (see canlab_dataset)</p>
</dd>
<dt><strong>subject:</strong></dt>
<dd><p class="first last">Subject list (it could be one subject [in a string 
format], or it could be multiple subjects in cell array)</p>
</dd>
<dt><strong>spm:</strong></dt>
<dd><p class="first last">This could be loaded SPM (struct), or one path for one 
subject&#8217;s SPM.mat file (string), or multiple loaded SPM or
paths in cell array</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">Canlab_dataset object with new data</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>subj = {&#39;dpsp002&#39;,&#39;dpsp003&#39;};
spm = {&#39;dpsp002_SPM.mat&#39;, &#39;dpsp003_SPM.mat&#39;};

D = canlab_dataset; % if D doesn&#39;t exist yet
D = spm2canlab_dataset(D, subj, spm);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body">canlab_dataset
spm_mat2batchinput</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.ttest2">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">ttest2</code><span class="sig-paren">(</span><em>D</em>, <em>varname</em>, <em>wh_keep1</em>, <em>wh_keep2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.ttest2" title="Permalink to this definition">¶</a></dt>
<dd><p>Two sample ttest for two samples of one subject-level variable</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>ttest2(D, varname, wh_keep1, wh_keep2, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
<dt><strong>varname:</strong></dt>
<dd><p class="first last">the name of a valid variable to get from dataset</p>
</dd>
<dt><strong>wh_keep1:</strong></dt>
<dd><p class="first last">subjects forming first sample</p>
</dd>
<dt><strong>wh_keep2:</strong></dt>
<dd><p class="first last">subjects forming second sample</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>noverbose:</strong></dt>
<dd><p class="first last">will suppress print out of results and bargraph</p>
</dd>
<dt><strong>varargin:</strong></dt>
<dd><p class="first last">other variables passed directly to MATLAB&#8217;s ttest2</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first last">same as MATLAB&#8217;s ttest2 output</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;canlab_dataset.write_text">
<code class="descclassname">&#64;canlab_dataset.</code><code class="descname">write_text</code><span class="sig-paren">(</span><em>D</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@canlab_dataset.write_text" title="Permalink to this definition">¶</a></dt>
<dd><p>&#8220;Flatten&#8221; dataset and write text files with header and data
For all Event-level and Subject-level data.  Files are created in the
current working directory.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [headername, dataname, fid] = write_text(D, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">a canlab_dataset object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">the first varargin parameter is the delimiter. Comma-delimited by default.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>headername:</strong></dt>
<dd><p class="first last">filename of header output file</p>
</dd>
<dt><strong>dataname:</strong></dt>
<dd><p class="first last">filename of data output file</p>
</dd>
<dt><strong>fid:</strong></dt>
<dd><p class="first last">file ID (currently does not look to be used)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-&#64;fmri_data">
<span id="fmri-data"></span><h2>fmri_data<a class="headerlink" href="#module-@fmri_data" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="&#64;fmri_data.canlab_connectivity_preproc">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">canlab_connectivity_preproc</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.canlab_connectivity_preproc" title="Permalink to this definition">¶</a></dt>
<dd><p>This function prepares data for connectivity analysis by removing nuisance
variables and temporal filtering (high, low, or bandpass filter). This also
can extract values from given masks and return averaged activity or pattern
expression values.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">preprocessed_dat</span><span class="p">,</span> <span class="n">roi_val</span><span class="p">]</span> <span class="o">=</span> <span class="n">canlab_connectivity_preproc</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Features:</th><td class="field-body"><ul class="first last simple">
<li>can regress out nuisance variables with any additional nuisance matrix</li>
<li>can remove signal from ventricle and white matter (calls
canlab_extract_ventricle_wm_timeseries.m and canlab_create_wm_ventricle_masks.m)</li>
<li>can do temporal filtering, including high-pass, low-pass, or bandpass
filtering (it uses conn_filter.m from conn toolbox; see subfunction below)</li>
<li>can extract data from given ROIs, and return averaged value or pattern
expression value (dot-product).</li>
</ul>
</td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt><em>Steps in order [with defaults]:</em></dt>
<dd><ol class="first last arabic simple">
<li>Remove nuisance covariates (and linear trend if requested)</li>
<li>Remove ventricle and white matter - needs structural images</li>
<li>Windsorize based on distribution of full data matrix</li>
<li>High/low/bandpass filter</li>
<li>Extract region-by-region average ROI or pattern expression data</li>
</ol>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">fmri_data object with data</p>
</dd>
<dt><strong>dat.covariate:</strong></dt>
<dd><p class="first last">basic nuisance matrix</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>additional_nuisance</strong></dt>
<dd><p class="first last">When you have additional nuisance variables that you want
regress out from the data, you can use this option. This
option should be followed by a nuisance matrix (or values).
The matrix should have the same number of rows with the
number of images.</p>
</dd>
<dt><strong>vw</strong></dt>
<dd><p class="first">When you want to regress out signals from ventricle and
white matter, you can use this option. To use this option,
You should provide the directory where the subjects&#8217; data
are saved using the &#8216;datdir&#8217; (for example, see below).
Requires specific subdirectory structure (CANlab) - see code.</p>
<p class="last">You can also choose what to use to remove ventricle and
white matter signal between raw data or top 5 PCA
components (default). You can just put &#8216;raw&#8217; if you want to
use raw signal than PCA compoenents.
also see: canlab_extract_ventricle_wm_timeseries.m
canlab_create_wm_ventricle_masks.m)
- <em>Example:</em> &#8216;vw&#8217;, &#8216;datdir&#8217;, subject_dir, &#8216;raw&#8217;</p>
</dd>
<dt><strong>windsorize:</strong></dt>
<dd><p class="first last">Windsorizing entire data matrix to k x STD.
- <em>Example:</em> &#8216;windsorize&#8217;, 5 (windsorize to 5 STD)</p>
</dd>
<dt><strong>linear_trend:</strong></dt>
<dd><p class="first last">This option will include the linear trend to nuisance variables.</p>
</dd>
<dt><strong>hpf&#8217;, &#8216;lpf&#8217;, or &#8216;bpf:</strong></dt>
<dd><p class="first">This option will do temporal filtering.
- &#8216;hpf&#8217;: high pass filter. This option should be followed by</p>
<blockquote>
<div><p>the lower bound of the frequency (e.g., .01 Hz [= 100 sec]).</p>
</div></blockquote>
<ul class="last">
<li><dl class="first docutils">
<dt>&#8216;lpf&#8217;: low pass filter. This option should be followed by</dt>
<dd><p class="first last">the upper bound of the frequency (e.g., .25 Hz [= 4 sec]).</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>&#8216;bpf&#8217;: bandpass filter. This should be followed by lower</dt>
<dd><p class="first last">and upper bounds of the frequency (e.g., [.01 .25]).
After the frequency value, you need to provide TR.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt><em>Example:</em> &#8216;hpf&#8217;, .01, TR</dt>
<dd><p class="first last">&#8216;bpf&#8217;, [.01 .25], TR</p>
</dd>
</dl>
</li>
</ul>
</dd>
<dt><strong>extract_roi:</strong></dt>
<dd><p class="first">This option will extract data from ROIs specified. This
option should be followed by one or more masks.
For one mask (potentially multiple ROIs, enter a char array with the mask name.
For multiple masks (1 or more), enter in a cell array of mask names.
You can specify methods with &#8216;roi_methods&#8217; option.
- &#8216;average_over&#8217; (default): calculate averaged value across the ROIs.
- &#8216;pattern_expression&#8217;: calculate dot-products between</p>
<blockquote>
<div><p>pattern mask and data</p>
</div></blockquote>
<ul class="last">
<li><dl class="first docutils">
<dt>&#8216;unique_mask_values&#8217; (default): will divide a mask into</dt>
<dd><p class="first last">multiple regions that have different discrete values.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>&#8216;contiguous_regions&#8217;: will divide a mask into multiple</dt>
<dd><p class="first last">contiguous regions.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>&#8216;whole&#8217;: will do average_over or pattern_expression across</dt>
<dd><p class="first last">all the voxels within the mask.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt><em>Example:</em> &#8216;extract_roi&#8217;, mask, &#8216;contiguous_regions&#8217;</dt>
<dd><p class="first last">&#8216;extract_roi&#8217;, mask, &#8216;pattern_expression&#8217;</p>
</dd>
</dl>
</li>
</ul>
</dd>
<dt><strong>no_preproc:</strong></dt>
<dd><p class="first last">If you want to skip the preprocessing part, and want to
extract ROI values only, you can use this option.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>preprocessed_dat:</strong></dt>
<dd><p class="first last">fmri_data object after removing nuisance variables and
filtering temporal confounds.</p>
</dd>
<dt><strong>roi_val:</strong></dt>
<dd><p class="first last">returns values extracted from ROIs in cell arrays (if there are many different ROIs).
Each cell will have roi_val.dat, roi_val.mask_name, and roi_val.methods.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>roi_masks = which(&#39;weights_NSF_grouppred_cvpcr.img&#39;);
[preprocessed_dat, roi_val] = canlab_connectivity_preproc(dat, &#39;vw&#39;, &#39;datdir&#39;,
       subject_dir, &#39;bpf&#39;, [.008 .25], TR, &#39;extract_roi&#39;, roi_masks, &#39;pattern_expression&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.create">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">create</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.create" title="Permalink to this definition">¶</a></dt>
<dd><p>Create an object from an empty obj structure, assigning fieldname/value
pairs as optional arguments.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[obj = create(obj, varargin)
</pre></div>
</div>
<p>Used in fmri_data.m class constructor.
if &#8216;noverbose&#8217; is entered, suppress output</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.extract_roi_averages">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">extract_roi_averages</code><span class="sig-paren">(</span><em>obj</em>, <em>mask_image</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.extract_roi_averages" title="Permalink to this definition">¶</a></dt>
<dd><p>This fmri_data method a extracts and averages data stored in an fmri_data object 
from a set of ROIs defined in a mask.</p>
<p>If no mask_image is entered, it uses the mask defined with the fmri_data object as a default.</p>
<dl class="docutils">
<dt>If mask_image is a new image file name, this method:</dt>
<dd><ol class="first last arabic simple">
<li>Defines an fmri_mask_image object using create_fmri_mask</li>
<li>Maps to the space in fmri_data object using resample_to_image_space</li>
</ol>
</dd>
</dl>
<p>Regions to average over can be either regions of contiguous voxels
bounded by voxels with values of 0 or NaN, which are considered non-data
values, or regions defined by unique integer codes in the mask image
(i.e., for atlas images with unique codes for each defined region.)</p>
<p>Mask/Atlas image does NOT have to be in the same space as the images to
extract from.  It will be remapped/resliced.
NOTE: Mask is <em>reloaded</em> from original data if space is remapped, and you
cannot use manual thresholding of the mask. This is a feature of the
map_to_image_space method and scn_map_image</p>
<p>Extracted data is returned in single data format.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[[cl, clroimean, clpattern] = extract_roi_averages(fmri_data obj, [mask_image], [average_over])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><ol class="first arabic simple">
<li>char array of strings containing 4D image file names (data extracted from these)</li>
<li>mask_image to extract from.</li>
</ol>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>how to average</strong></dt>
<dd><p class="first last">Default = &#8216;unique_mask_values&#8217; to average over unique integer codes in the mask image
bounded by voxels of 0 or NaN (non-data values)
(i.e., for atlas images with unique codes for each defined region)
Alt. option = &#8216;contiguous_regions&#8217; to average over contiguous voxels</p>
</dd>
<dt><strong>pattern_expression</strong></dt>
<dd><p class="first">Use values in mask images to get weighted average within each
region, rather than simple average.  See also apply_mask with
&#8216;pattern_expression&#8217; option.</p>
<p class="last">Optional outputs (varargout): 
[cl, cl_roimean, cl_roipattern] = ...
roimean: pattern expression is average over ROI (unit vector)
roipattern: pattern expression is dot product of activity and mean-centered pattern weights</p>
</dd>
<dt><strong>nonorm</strong></dt>
<dd><p class="first last">Turn off L1 norm in pattern expression.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">imgs_to_extract_from</span> <span class="o">=</span> <span class="n">filenames</span><span class="p">(</span><span class="s">&#39;w*.nii&#39;</span><span class="p">,</span><span class="s">&#39;char&#39;</span><span class="p">);</span>
<span class="n">mask_image</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="s">&#39;anat_lbpa_thal.img&#39;</span><span class="p">);</span>
<span class="p">[</span><span class="n">cl</span><span class="p">,</span> <span class="n">clroimean</span><span class="p">,</span> <span class="n">clpattern</span><span class="p">]</span> <span class="o">=</span> <span class="n">extract_image_data</span><span class="p">(</span><span class="n">imgs_to_extract_from</span><span class="p">,</span> <span class="n">mask_image</span><span class="p">);</span>

<span class="n">region_obj</span> <span class="o">=</span> <span class="n">extract_roi_averages</span><span class="p">(</span><span class="n">data_obj</span><span class="p">,</span> <span class="n">mask_char_name</span><span class="p">,</span> <span class="s">&#39;pattern_expression&#39;</span><span class="p">,</span> <span class="s">&#39;contiguous_regions&#39;</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Related functions:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body">For an non-object-oriented alternative, see extract_image_data.m</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.horzcat">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">horzcat</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.horzcat" title="Permalink to this definition">¶</a></dt>
<dd><p>Implements the horzcat ([a b]) operator on image_vector objects across voxels.
Requires that each object has an equal number of columns and voxels</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function s = horzcat(varargin)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>c = [dat1 dat2];
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.hrf_fit">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">hrf_fit</code><span class="sig-paren">(</span><em>obj</em>, <em>TR</em>, <em>Runc</em>, <em>T</em>, <em>method</em>, <em>mode</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.hrf_fit" title="Permalink to this definition">¶</a></dt>
<dd><p>HRF estimation on fmri_data class object</p>
<p>HRF estimation function for a single voxel;</p>
<p>Implemented methods include: IL-model (Deterministic/Stochastic), FIR
(Regular/Smooth), and HRF (Canonical/+ temporal/+ temporal &amp; dispersion)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj</strong></dt>
<dd><p class="first last">fMRI object</p>
</dd>
<dt><strong>TR</strong></dt>
<dd><p class="first last">time resolution</p>
</dd>
<dt><strong>Runs</strong></dt>
<dd><p class="first last">expermental design</p>
</dd>
<dt><strong>T</strong></dt>
<dd><p class="first last">length of estimated HRF ij seconds</p>
</dd>
<dt><strong>type</strong></dt>
<dd><p class="first last">Model type: &#8216;FIR&#8217;, &#8216;IL&#8217;, or &#8216;CHRF&#8217;</p>
</dd>
<dt><strong>mode</strong></dt>
<dd><p class="first last">Mode</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Model Types:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<ol class="upperalpha">
<li><dl class="first docutils">
<dt><strong>Fit HRF using IL-function</strong></dt>
<dd><dl class="first docutils">
<dt>Choose mode (deterministic/stochastic)</dt>
<dd><ul class="first last simple">
<li>0 - deterministic aproach</li>
<li>1 - simulated annealing approach</li>
</ul>
</dd>
</dl>
<p class="last">Please note that when using simulated annealing approach you
may need to perform some tuning before use.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt><strong>Fit HRF using FIR-model</strong></dt>
<dd><dl class="first last docutils">
<dt>Choose mode (FIR/sFIR)</dt>
<dd><ul class="first last simple">
<li>0 - FIR</li>
<li>1 - smooth FIR</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt><strong>Fit HRF using FIR-model</strong></dt>
<dd><dl class="first last docutils">
<dt>Choose mode (FIR/sFIR)</dt>
<dd><ul class="first last simple">
<li>0 - FIR</li>
<li>1 - smooth FIR</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</li>
</ol>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>SIMULATE DATA AND RUN</p>
<div class="highlight-python"><div class="highlight"><pre>%params for sim and fitting
TR = 2;   % repetition time (sec)
n = 200;  % time points measured (for simulation) must be multiple of 10
T = 30;   % duration of HRF to estimate (seconds)
nconds = 2; % num conditions
nevents = 8; % events per condition

% Create fake data
h = spm_hrf(TR);
y = zeros(n, 1);

% onsets - indicator
Condition = {};
for i = 1:nconds
    Condition{i} = zeros(n,1);
    wh = randperm(n);
    Condition{i}(wh(1:nevents)) = 1;

    ytmp{i} =  conv(Condition{i}, h);
    ytmp{i} = ytmp{i}(1:n);
end

y = sum(cat(2, ytmp{:}), 2);

dat = fmri_data(&#39;VMPFC_mask_neurosynth.img&#39;);  % AVAILABLE ON WIKI IN MASK GALLERY
dat = threshold(dat, [5 Inf], &#39;raw-between&#39;);

v = size(dat.dat, 1); % voxels in mask
dat.dat = repmat(y&#39;,v, 1) + .1 * randn(v, n);

% Fit data - estimate HRFs across the brain mask
[params_obj hrf_obj] = hrf_fit(dat,TR, Condition, T,&#39;FIR&#39;, 1);

hrf = fmri_data(&#39;HRF_timecourse_cond0001.img&#39;);
hrf = remove_empty(hrf);
create_figure(&#39;hrfs&#39;, 1, 2); 
plot(hrf.dat&#39;);
title(&#39;Condition 1&#39;)
hrf = fmri_data(&#39;HRF_timecourse_cond0002.img&#39;);
hrf = remove_empty(hrf);
subplot(1, 2, 2);
plot(hrf.dat&#39;);
title(&#39;Condition 2&#39;)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.plot">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">plot</code><span class="sig-paren">(</span><em>fmridat</em>, <em>plotmethod</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.plot" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot means by condition
plot(fmri_data_object, &#8216;means_for_unique_Y&#8217;)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt>Plot methods:</dt>
<dd><ul class="first last simple">
<li>plot data matrix</li>
<li>plot(fmri_data_object)</li>
</ul>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">plot</span><span class="p">(</span><span class="n">fmridat</span><span class="p">,</span> <span class="p">[</span><span class="n">plotmethod</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>5 plots and an SPM orthviews presentation of the data.  In the below 
and elsewhere, &#8220;image&#8221; connotes a 3D brain volume captured every TR.</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>subplot 1:</strong></dt>
<dd>the fMRI data itself. Color is intensity of signal.</dd>
<dt><strong>subplot 2:</strong></dt>
<dd>presented as a histogram of values for every voxel collected.
The low values are typically out-of-brain voxels, as there is
no signal there.</dd>
<dt><strong>subplot 3:</strong></dt>
<dd>each point is an image.  The point&#8217;s X value is the mean
intensity of every voxel in that image, and the Y value is the
stdev of intensities for all voxels in that image.</dd>
<dt><strong>subplot 4:</strong></dt>
<dd>covariance between images</dd>
<dt><strong>subplot 5:</strong></dt>
<dd>each point is an image (case = image).  X value is image
number in the run, Y is image mean intensity, and the size of
the circular marker represents stdev for that image</dd>
<dt><strong>Orthviews:</strong></dt>
<dd>mean and STD for a given voxel averaged over time.  Note that
the values for mean and STD here are higher than in the plots
above. That is because mean and STD are calculated here by
voxel, but in the plots above they are calculated by image.
Images also include out-of-brain areas.</dd>
</dl>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.predict">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">predict</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.predict" title="Permalink to this definition">¶</a></dt>
<dd><p>Predict outcome (Y) from brain data and test cross-validated error rate for an fmri_data object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cverr</span><span class="p">,</span> <span class="n">stats</span><span class="p">,</span> <span class="n">optional_outputs</span><span class="p">]</span> <span class="o">=</span> <span class="n">predict</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Features:</th><td class="field-body"><ul class="first simple">
<li>flexible specification of algorithm by function name</li>
<li>k-fold cross-validation, default = 5-fold, can enter custom fold membership</li>
<li>folds are stratified on outcome</li>
<li>choice of multiple error metrics (class loss, mse, etc.)</li>
<li>by default, chooses error metric based on outcome type (classes vs. continuous-valued)</li>
<li>returns all outputs for each fold returned by the algorithm in optout cell array variable</li>
<li>bootstrapping of weights built in [optional keyword]</li>
<li>select variable number of components (for pcr-based techniques)</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Inputs:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>obj is mandatory, rest are optional</dt>
<dd><dl class="first last docutils">
<dt><strong>obj:</strong></dt>
<dd>fmri_data or image_vector object, with fields .dat (data used to predict) and .Y (outcome)</dd>
</dl>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">(with their default values)</p>
<dl class="docutils">
<dt><strong>nfolds</strong> = 5</dt>
<dd><p class="first last">number of folds</p>
</dd>
<dt><strong>nfolds</strong> = [vector of integers]</dt>
<dd><p class="first last">can also input vector of integers for holdout set IDs</p>
</dd>
<dt><strong>error_type</strong> = mcr</dt>
<dd><p class="first last">mcr, mse: misclassification rate or mean sq. error</p>
</dd>
<dt><strong>algorithm_name</strong> = &#8216;cv_regress&#8217;</dt>
<dd><p class="first last">name of m-file defining training/test function</p>
</dd>
<dt><strong>useparallel</strong> = 1</dt>
<dd><p class="first last">Use parallel processing, if available; follow by 1 for yes, 0 for no</p>
</dd>
<dt><strong>bootweights</strong> = 0</dt>
<dd><p class="first last">bootstrap voxel weights; enter bootweights do bootstrapping of weight maps (based on all observations)</p>
</dd>
<dt><strong>savebootweights</strong></dt>
<dd><p class="first last">save bootstraped weights (useful for combining across multiple iterations of predict())</p>
</dd>
<dt><strong>bootsamples</strong> = 100</dt>
<dd><p class="first last">number of bootstrap samples to use</p>
</dd>
<dt><strong>numcomponents</strong> = xxx:</dt>
<dd><p class="first last">save first xxx components (for pca-based methods)</p>
</dd>
<dt><strong>nopcr</strong></dt>
<dd><p class="first last">for cv_lassopcr and cv_lassopcrmatlab: do not do pcr, use original variables</p>
</dd>
<dt><strong>lasso_num</strong> = xxx</dt>
<dd><p class="first last">followed by number of components/vars to retain after shrinkage</p>
</dd>
<dt><strong>hvblock</strong> = [h,v]</dt>
<dd><p class="first last">use hvblock cross-validation with a block size of &#8216;h&#8217; (0 reduces to v-fold xval) and
number of test observations &#8216;v&#8217; (0 reduces to h-block xval)</p>
</dd>
<dt><strong>rolling</strong> = [h,v,g]</dt>
<dd><p class="first last">use rolling cross-validation with a block size of &#8216;h&#8217; (0 reduces to v-fold xval) and
number of test observations &#8216;v&#8217; (0 reduces to h-block xval), and a training size
of g * 2 surrounding hv</p>
</dd>
<dt><strong>verbose</strong> = 1</dt>
<dd><p class="first last">Set to 0 to suppress output to command window</p>
</dd>
<dt><strong>platt_scaling</strong></dt>
<dd><p class="first last">calculate cross-validated platt scaling if using SVM.
Softmax parameters [A,B] are in other_output{3}</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Algorithm choices:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">You can input the name (as a string array) of any algorithm with the
appropriate inputs and outputs. i.e., this can either be one of the
built-in choices below, or the name of another m-file.
The format for algorithm functions is :
[yfit, other_outputs] = predfun(xtrain, ytrain, xtest, optional_inputs)
Each algorithm can take/interpret its own optional inputs.
For bootstrapping of weights, algorithms MUST RETURN 3 OUTPUTS
(programming &#8216;feature&#8217;)</p>
<p>To choose an algorithm, enter &#8216;algorithm_name&#8217; followed by a text string
with a built-in algorithm name, or a function handle for a custom algorithm
Built-in algorithm choices include:</p>
<dl class="docutils">
<dt><strong>cv_multregress:</strong></dt>
<dd><p class="first last">[default] multiple regression</p>
</dd>
<dt><strong>cv_univregress:</strong></dt>
<dd><p class="first last">Average predictions from separate univariate regression of outcome on each feature</p>
</dd>
<dt><strong>cv_svr:</strong></dt>
<dd><p class="first last">Support vector regression with Spider package; requires spider</p>
</dd>
<dt><strong>cv_pcr:</strong></dt>
<dd><p class="first last">Cross-validated principal components regression</p>
</dd>
<dt><strong>cv_lassopcr:</strong></dt>
<dd><p class="first last">Cross-val LASSO-PCR; can enter &#8216;lasso_num&#8217; followed by components to retain by shrinkage
NOTE: can enter &#8216;EstimateParams&#8217; to use shrankage
lasso method based on the estimated optimal lambda
that minimizes the mean squared error (MSE) of nested
cross-validation models. Output of nested cv model is
saved in stats.other_output_cv{:,3}. Output includes
&#8216;Lambda&#8217; parameter and min MSE value.</p>
</dd>
<dt><strong>cv_lassopcrmatlab:</strong></dt>
<dd><p class="first last">Cross-val LASSO-PCR; can enter &#8216;lasso_num&#8217; followed by components to retain by shrinkage
NOTE: this uses the matlab implementation of LASSO,
but can also run ridge or elastic net. Reduces to PCR
when no lasso_num is entered by default.  Use MSE for
predicting continuous data and MCR for classifying
binary data.
NOTE: You can input any optional inputs that lassoglm
takes.
Enter &#8216;Alpha&#8217;, (0,1] as optional inputs to
run ridge (Alpha approaches 0, but excluding 0), lasso (Alpha = 1), or elastic
net (Alpha between 0 and 1)
NOTE: Requires Matlab R2012a and higher.
NOTE: Optional input: &#8216;EstimateParams&#8217; - this will
use grid search and nested cross validation to
estimate Lambda and Alpha.  Output is saved in
stats.other_output_cv{:,3}.  Output includes &#8216;Alpha&#8217;
parameter which is the elastic net mixture value
between l1 and l2 regularization, &#8216;Lambda&#8217; parameter,
which is amount of LASSO regularization/shrinkage, and
&#8216;errorMatrix&#8217;, which is the amount of error for each
parameter combination.  Use
imagesc(obj.stats_other_output_cv{:,3}.errorMatrix)
to view matrix.  Min of this matrix is the best
fitting parameters.</p>
</dd>
<dt><strong>cv_svm:</strong></dt>
<dd><p class="first last">Cross-val support vector machine using Spider package
NOTE: This is sensitive to scale of outputs! Use -1 , 1
NOTE: Optional inputs: Slack var parameter: &#8216;C&#8217;, 1 [default], &#8216;C&#8217;, 3 etc.
Distance from hyperplane saved in
stats.other_output_cv{:,2}.  Recommend using the reordered
cross-validated distance from hyperplane saved in stats.other_output{3}
stats.dist_from_hyperplane_xval =  cross-validated distance from hyperplane
stats.weight_obj = voxel (variable) weight object
e.g., orthviews(stats.weight_obj)
Intercept for calculating dist from hy is in stats.other_output_cv{:,3}
e.g., dist_hy = stats.weight_obj.dat&#8217; * obj.dat, where obj is a new set of test images
NOTE: To run nonlinear SVM using radial basis
function.  Add &#8216;rbf&#8217; followed by size of sigma (e.g., 2).
NOTE: To estimate some of the parameters using
nested cross validation add &#8216;EstimateParams&#8217; as optional input.
NOTE: To run multiclass SVM (i.e., one vs rest) add
&#8216;MultiClass&#8217; as optional input.  Important - Obj.Y must be a matrix (data x
class) with a column of 1 and -1 indicating each
class.  For example, if using 3 classes, then obj.Y
must have 3 columns.
NOTE: To run a balanced SVM where the number of cases for each class are unequal (i.e., one vs rest) add
&#8216;Balanced&#8217; as optional input, followed by a numerical value indicating the ridge amount (e.g., 0.01).</p>
</dd>
<dt><strong>cv_multilevel_glm:</strong></dt>
<dd><p class="first last">Runs glmfit_multilevel. Must pass in &#8216;&#8217;subjIDs&#8217;&#8217; followed by an array specifying which subject each trial belongs to
Subjects&#8217; trials must all be &#8220;adjacent&#8221;, i.e., don&#8217;t
put some of subject 1&#8217;s trials at the beginning and
other trials at the end &#8211; subjIDs does not handle
this case correctly. Also, 2ND LEVEL PREDICTORS NOT
CURRENTLY SUPPORTED.  code can be expanded to support this.
mean-centering X and/or Y will NOT impact the
predictor betas.  Note that it WILL impact the intercept
esimate as well as how much variance is explained
(pred_outcome_r).  Stratified CV partition not
supported either, pass in custom holdout set.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">Copy of outcome data to be predicted</p>
</dd>
<dt><strong>algorithm_name:</strong></dt>
<dd><p class="first last">Name of algorithm; see options above</p>
</dd>
<dt><strong>function_call:</strong></dt>
<dd><p class="first last">String of the command evaluated to call the prediction function</p>
</dd>
<dt><strong>function_handle:</strong></dt>
<dd><p class="first last">Handle for the command evaluated to call the prediction function</p>
</dd>
<dt><strong>yfit:</strong></dt>
<dd><p class="first last">Predicted outcome data (cross-validated)</p>
</dd>
<dt><strong>err:</strong></dt>
<dd><p class="first last">Residuals/misclassification vector (cross-validated)</p>
</dd>
<dt><strong>error_type:</strong></dt>
<dd><p class="first last">Name of error metric used for cverr</p>
</dd>
<dt><strong>cverr:</strong></dt>
<dd><p class="first last">Cross-validated error</p>
</dd>
<dt><strong>nfolds:</strong></dt>
<dd><p class="first">Number of folds in stratified cross-validation, or
vector of integers for membership in custom holdout set of each fold</p>
<blockquote class="last">
<div><ul class="simple">
<li>if k = 1, will estimate weights for full data object
and not crossvalidate (useful for bootstrapping)</li>
</ul>
</div></blockquote>
</dd>
<dt><strong>cvpartition:</strong></dt>
<dd><p class="first last">Cross-val partition object or structure with fold info</p>
</dd>
<dt><strong>teIdx:</strong></dt>
<dd><p class="first last">Cell array of logical vectors with test samples in each fold</p>
</dd>
<dt><strong>trIdx:</strong></dt>
<dd><p class="first last">Cell array of logical vectors with training samples in each fold</p>
</dd>
<dt><strong>other_output:</strong></dt>
<dd><p class="first last">Other outputs returned by the algorithm; number and nature depend on algo choice; e.g., beta weights, svr weights, etc.
For many algorithms, other_output{1} is a vector of
weights on variables (e.g., voxels)</p>
</dd>
<dt><strong>other_output_descrip:</strong></dt>
<dd><p class="first last">String description of other outputs</p>
</dd>
<dt><strong>other_output_cv:</strong></dt>
<dd><p class="first last">Other outputs for each cross-validation fold</p>
</dd>
<dt><strong>other_output_cv_descrip:</strong></dt>
<dd><p class="first last">Other output from algorithm - for each CV fold</p>
</dd>
<dt><strong>mse:</strong></dt>
<dd><p class="first last">For regression only; mean squared error</p>
</dd>
<dt><strong>rmse:</strong></dt>
<dd><p class="first last">For regression only; root mean squared error</p>
</dd>
<dt><strong>meanabserr:</strong></dt>
<dd><p class="first last">For regression only; mean absolute error</p>
</dd>
<dt><strong>pred_outcome_r:</strong></dt>
<dd><p class="first last">For regression only; prediction-outcome correlation</p>
</dd>
<dt><strong>WTS:</strong></dt>
<dd><p class="first last">bootstrapped weights on voxels</p>
</dd>
<dt><strong>weight_obj:</strong></dt>
<dd><p class="first last">for some algorithms, an fmri_data object with the predictive weights (from full sample)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>obj = fmri_data;
obj.dat = randn(30, 50); %    30 voxels, 50 images (observations)
obj.Y = obj.dat&#39; * rand(30, 1) + randn(50, 1); %    toy Y, linear combo of X plus noise
[cverr, stats, regression_outputs] = predict(obj);

Simulated example with 100 observations, 1000 voxels, with bootstrapping
dat = fmri_data;
dat.Y = rand(100, 1);
dat.dat = repmat(dat.Y&#39;, 1000, 1) + 10*rand(1000, 100);
[err,stats] = predict(dat, &#39;bootweights&#39;, &#39;algorithm_name&#39;, &#39;cv_lassopcr&#39;);

[cverr, stats, regression_outputs] = predict(obj, &#39;nfolds&#39;, 3, &#39;error_type&#39;, &#39;meanabserr&#39;);
[cverr, stats, regression_outputs] = predict(obj, &#39;algorithm_name&#39;, &#39;cv_univregress&#39;, &#39;error_type&#39;, &#39;meanabserr&#39;);
[cverr, stats, optout] = predict(obj, &#39;algorithm_name&#39;, &#39;cv_lassopcr&#39;, &#39;lasso_num&#39;, 5, &#39;nfolds&#39;, 5, &#39;error_type&#39;, &#39;mse&#39;, &#39;bootweights&#39;);
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_svm&#39;, &#39;nfolds&#39;, 5, &#39;error_type&#39;, &#39;mse&#39;);
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_svm&#39;, &#39;rbf&#39;, 2, &#39;nfolds&#39;, 5, &#39;error_type&#39;, &#39;mse&#39;); %SVM w/ radial basis function
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_svm&#39;, &#39;rbf&#39;, 2, &#39;EstimateParams&#39;, &#39;nfolds&#39;, 5, &#39;error_type&#39;, &#39;mse&#39;); %SVM w/ radial basis function w/ parameters estimated using nested cross-valdiation
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_svm&#39;, &#39;nfolds&#39;, 5, &#39;MultiClass&#39;, &#39;error_type&#39;, &#39;mse&#39;);

Elastic net with first 10 components:
[cverr, stats, optout] = predict(dat_masked, &#39;algorithm_name&#39;, &#39;cv_lassopcrmatlab&#39;, &#39;nfolds&#39;, 5, &#39;error_type&#39;, &#39;mse&#39;, &#39;numcomponents&#39;, 10, &#39;Alpha&#39;, .5); stats.pred_outcome_r

Ridge with first 10 components:
[cverr, stats, optout] = predict(dat_masked, &#39;algorithm_name&#39;, &#39;cv_lassopcrmatlab&#39;, &#39;nfolds&#39;, 5, &#39;error_type&#39;, &#39;mse&#39;, &#39;numcomponents&#39;, 10, &#39;Alpha&#39;, 0.00001); stats.pred_outcome_r

Lasso with all components, but shrink to retain 2 components only:
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_lassopcrmatlab&#39;, &#39;nfolds&#39;, whfolds, &#39;nopcr&#39;, &#39;lasso_num&#39;, 2, &#39;Alpha&#39;, 1);
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_lassopcr&#39;, &#39;nfolds&#39;, whfolds, &#39;lasso_num&#39;, 2);

Lasso with the shrinkage methods based on the estimated optimal lambda that minimizes MSE of nested cross-validation models.
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_lassopcr&#39;, &#39;nfolds&#39;, whfolds, &#39;estimateparam&#39;);
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_lassopcr&#39;, &#39;nfolds&#39;, 5, &#39;estimateparam&#39;);

Lasso without doing PCR:
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_lassopcrmatlab&#39;, &#39;nfolds&#39;, whfolds, &#39;nopcr&#39;, &#39;lasso_num&#39;, 2, &#39;Alpha&#39;, 1);
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_lassopcr&#39;, &#39;nfolds&#39;, whfolds, &#39;lasso_num&#39;, 2, &#39;nopcr&#39;);
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_lassopcr&#39;, &#39;nfolds&#39;, 5, &#39;estimateparam&#39;, &#39;nopcr&#39;);

Lasso pcr using hvblock cross-validation on time-series, h = 3, v = 5;
[cverr, stats, optout] = predict(dat, &#39;algorithm_name&#39;, &#39;cv_lassopcr&#39;, &#39;hvblock&#39;,[3,5]);

Output display:
orthviews(stats.weight_obj)
line_plot_multisubject(stats.yfit, stats.Y, &#39;subjid&#39;, id_numbers);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"><p class="first">predict_test_suite method for fmri_data, which runs predict with multiple
options and summarizes output.</p>
<p class="last">xval_regression_multisubject, xval_lasso_brain</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.predict_test_suite">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">predict_test_suite</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.predict_test_suite" title="Permalink to this definition">¶</a></dt>
<dd><p>Run a set of cross-validated prediction algorithms on an fmri_data object
and plot the outcome.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[allcverr, allyhat] = predict_test_suite(dat, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Functionality:</th><td class="field-body"><ul class="first simple">
<li>Requires matlab 2012a or later for full functionality</li>
<li>Handles categorical or continuous outcomes automatically</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">an fMRI data object. 
dat.Y must be assigned, and must have continuous or binary outcomes assigned.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Optional:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt><strong>quick:</strong></dt>
<dd>Skip extended output</dd>
<dt><strong>nfolds:</strong></dt>
<dd>Followed by number of folds or custom holdout vector (default = 5-fold balanced)</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">predict_test_suite</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;nfolds&#39;</span><span class="p">,</span> <span class="n">subjid</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.regress">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">regress</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.regress" title="Permalink to this definition">¶</a></dt>
<dd><p>Regression method for fmri_data object</p>
<p>Regress dat.X on dat.dat at each voxel, and return voxel-wise statistic
images. Each column of dat.X is a predictor in a multiple regression,
and the intercept is the last column. Intercept will automatically be
added if not detected unless &#8216;nointercept&#8217; is specified.</p>
<p>This function can also create a map of brain regions that predict the dat.Y
vector using the &#8216;brainony&#8217; option.  This is essentially a univariate
version of the &#8216;predict&#8217; command.  Warning: this is very slow as it loops
through all voxels.</p>
<p>Regression is OLS by default, but can be robust using &#8216;robust&#8217; flag.</p>
<p>Creates thresholded plot by default</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">out</span> <span class="o">=</span> <span class="n">regress</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">should be an fmri_data object with X field defined.
dat.X can be a design_matrix() object.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>[threshold, &#8216;unc&#8217;]:</strong></dt>
<dd><p class="first last">p-value threshold string indicating threshold type
(see help statistic_image.threshold for options)</p>
</dd>
<dt><strong>nointercept:</strong></dt>
<dd><p class="first last">Do not add intercept to model</p>
</dd>
<dt><strong>nodisplay:</strong></dt>
<dd><p class="first last">Do not plot thresholded results using orthviews</p>
</dd>
<dt><strong>brainony:</strong></dt>
<dd><p class="first last">univariate approach to predict obj.Y from brain data</p>
</dd>
<dt><strong>residual:</strong></dt>
<dd><p class="first last">Output residual as fmri_data() object</p>
</dd>
<dt><strong>noverbose:</strong></dt>
<dd><p class="first last">Suppress verbose outputs</p>
</dd>
<dt><strong>robust:</strong></dt>
<dd><p class="first last">Run a robust regression (default is OLS).  Robust is considerably
slower than OLS</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>out:</strong></dt>
<dd><p class="first last">A structure containing stats_img and fmri_data objects.</p>
</dd>
<dt><strong>out.b:</strong></dt>
<dd><p class="first last">stats_img object of beta values estimated from regression</p>
</dd>
<dt><strong>out.t:</strong></dt>
<dd><p class="first last">stats_img object of t-values with input threshold</p>
</dd>
<dt><strong>out.df:</strong></dt>
<dd><p class="first last">fmri_data object of degrees of freedom</p>
</dd>
<dt><strong>out.sigma:</strong></dt>
<dd><p class="first last">fmri_data object of variance of residual</p>
</dd>
<dt><strong>out.residual:</strong></dt>
<dd><p class="first last">fmri_data object of residual data after model has been regressed out (optional).</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Run regression with liberal threshold
out = regress(dat, .05, &#39;unc&#39;);

% Run regression with conservative threshold and save residual
out = regress(dat, .001, &#39;unc&#39;, &#39;residual);

% Run robust regression with fdr threshold
out = regress(dat, .05, &#39;fdr&#39;,&#39;robust&#39;);

% Run a regression predicting behavior from brain at liberal threshold
out  = regress(data_comb, .05, &#39;unc&#39;, &#39;brainony&#39;)

% Re-threshold at different values
out.t = threshold(out.t, .05, &#39;fdr&#39;);
out.t = threshold(out.t, .001, &#39;unc&#39;);

% Re-display results of thresholding
orthviews(out.t);

% Write out beta image to current directory
out.b.fullpath = fullfile(pwd,&#39;beta.nii&#39;);
write(out)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.rescale">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">rescale</code><span class="sig-paren">(</span><em>fmridat</em>, <em>meth</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.rescale" title="Permalink to this definition">¶</a></dt>
<dd><p>Rescales data in an fmri_data object
Data is observations x images, so operating on the columns operates on
images, and operating on the rows operates on voxels (or variables more
generally) across images.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fmridat</span> <span class="o">=</span> <span class="n">rescale</span><span class="p">(</span><span class="n">fmridat</span><span class="p">,</span> <span class="n">meth</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Methods:</strong></dt>
<dd><ul class="first last simple">
<li>centervoxels</li>
<li>zscorevoxels</li>
<li>centerimages</li>
<li>zscoreimages</li>
<li>rankvoxels</li>
<li>windsorizevoxels</li>
<li>percentchange</li>
<li>tanh</li>
</ul>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>Appropriate for multi-session (time series) only:</dt>
<dd><ul class="first last simple">
<li>session_global_percent_change</li>
<li>session_global_z</li>
<li>session_multiplicative</li>
</ul>
</dd>
</dl>
<p>See also fmri_data.preprocess</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.saveplots">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">saveplots</code><span class="sig-paren">(</span><em>fmri_dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.saveplots" title="Permalink to this definition">¶</a></dt>
<dd><p>Output dir</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.signtest">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">signtest</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.signtest" title="Permalink to this definition">¶</a></dt>
<dd><p>Sign test for each voxel of an fmri_data object
returns voxel-wise statistic images.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[out, statimg] = signtest(dat, [p-val threshold], [thresh_type])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">Should be an fmri_data object with .dat field containing voxels x observations matrix</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">in [  ] above are
<strong>p-value threshold:</strong></p>
<blockquote>
<div><p>string indicating threshold type (see help statistic_image.threshold for options)</p>
</div></blockquote>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>out:</strong></dt>
<dd><p class="first last">is a structure of information about the sign test</p>
</dd>
<dt><strong>statimg:</strong></dt>
<dd><blockquote class="first">
<div><p>is a statistic_image object that can be thresholded and
plotted/imaged.  statimg.dat contains signed direction values,</p>
</div></blockquote>
<p class="last">.p contains p-values</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>c Tor Wager, 2011
..</p>
<p>See also: fmri_data.regress</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.ttest">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">ttest</code><span class="sig-paren">(</span><em>fmridat</em>, <em>pvalthreshold</em>, <em>thresh_type</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.ttest" title="Permalink to this definition">¶</a></dt>
<dd><p>T-test on fmri_data class object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">statsimg</span> <span class="o">=</span> <span class="n">ttest</span><span class="p">(</span><span class="n">fmridat</span><span class="p">,</span> <span class="n">pvalthreshold</span><span class="p">,</span> <span class="n">thresh_type</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>p-value threshold:</strong></dt>
<dd><p class="first last">p-value, e.g., .05 or .001 or [.001 .01 .05]</p>
</dd>
<dt><strong>thresh_type:</strong></dt>
<dd><p class="first last">&#8216;uncorrected&#8217;, &#8216;fwe&#8217;, or &#8216;fdr&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>%T-test, Construct a stats_image object, threshold and display:
statsimg = ttest(fmridat, .001, &#39;unc&#39;);
orthviews(statsimg);

%Re-threshold and display:
statsimg = threshold(statsimg, .000001, &#39;unc&#39;);
orthviews(statsimg);

statsimg = threshold(statsimg, .01, &#39;fdr&#39;);
orthviews(statsimg);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body">for two-sample T-test, use fmri_data.regress</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_data.windsorize">
<code class="descclassname">&#64;fmri_data.</code><code class="descname">windsorize</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_data.windsorize" title="Permalink to this definition">¶</a></dt>
<dd><p>Windsorize an fMRI data object to madlimit Median Absolute Deviations.
Default = 5 MADs.
Works across rows and columns.
Registers this step in history.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">windsorize</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="p">[</span><span class="n">madlimit</span><span class="p">])</span>
</pre></div>
</div>
</dd></dl>

</div>
<div class="section" id="module-&#64;fmri_mask_image">
<span id="fmri-mask-image"></span><h2>fmri_mask_image<a class="headerlink" href="#module-@fmri_mask_image" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="&#64;fmri_mask_image.resample_to_image_space">
<code class="descclassname">&#64;fmri_mask_image.</code><code class="descname">resample_to_image_space</code><span class="sig-paren">(</span><em>obj</em>, <em>sampleto</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_mask_image.resample_to_image_space" title="Permalink to this definition">¶</a></dt>
<dd><p>Resamples data in an fmri_mask_image object (obj) to the space of another
image (e.g., a functional image, for data extraction)
The volInfo field will be the same as the sampleto volume info.
The mask will have zeros in obj.dat for out-of-mask voxels.
THIS FUNCTION USES SCN_MAP_IMAGE AND REQUIRES THAT THE ORIGINAL IMAGE BE
AVAILABLE ON DISK.  Multiple resamplings will break the function because
the new space will be different from the original one on disk.  Use the
more general resample_space.</p>
<p>NOTE: Mask is <em>reloaded</em> from original data if space is remapped, and you
cannot use manual thresholding of the mask. This is a feature of the
map_to_image_space method and scn_map_image</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>obj = resample_to_image_space(obj, sampleto &lt;img name or image_vector object&gt;)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">must be an fmri_mask_image object</p>
</dd>
<dt><strong>sampleto:</strong></dt>
<dd><dl class="first last docutils">
<dt>can be either:</dt>
<dd><ol class="first last arabic simple">
<li>An image name to sample to</li>
<li>Another fmri_mask_image object (but image must exist on path!)</li>
</ol>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>mask:</strong></dt>
<dd><p class="first last">Apply sampleto as mask so that only voxels in the sampleto mask
are retained in obj.dat.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>THIS FUNCTION WORKS, BUT IS DEPRECATED BECAUSE RESAMPLE_SPACE IS MORE
GENERAL.  resample_space does not require the resampling of the original
image from disk, which this does.  resample_space is slower, though.</p>
<p>See Also: resample_space, for a method that does not require images to
exist on disk on the path.</p>
</dd></dl>

</div>
<div class="section" id="module-&#64;fmri_model">
<span id="fmri-model"></span><h2>fmri_model<a class="headerlink" href="#module-@fmri_model" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="&#64;fmri_model.build">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">build</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.build" title="Permalink to this definition">¶</a></dt>
<dd><p>Build the design matrix (xx) for an fmri_model object</p>
<p>We assume that the same conditions are modeled for each session.
We assume that you have one basis set per condition (this is different
from SPM, which only allows a single basis set across all conditions)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">build</span><span class="p">(</span><span class="n">fmri_model_obj</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_model.build_single_trial">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">build_single_trial</code><span class="sig-paren">(</span><em>obj</em>, <em>inputhrf</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.build_single_trial" title="Permalink to this definition">¶</a></dt>
<dd><p>Build a single-trial design matrix (xx) for an fmri_model object</p>
<p>We assume that the same conditions are modeled for each session
We assume that you have one basis set per condition (this is different
from SPM, which only allows a single basis set across all conditions)</p>
<p>This is used in single_trial_estimates, which assumes that you have
estimated an initial model and saved image data.</p>
<p>The idea behind this is somewhat different from other canlab single-trial
analyses, in that it takes in a single, custom HRF for each condition,
rather than using a basis set.  In single_trial_estimates, custom HRFs
are created for each voxel by using the condition- and voxel-specific hrf
estimates stored during model fitting.</p>
<dl class="docutils">
<dt>The sequence would be:</dt>
<dd><ol class="first last arabic simple">
<li>robustfit(my_model), to fit average model and get HRF est for each
voxel</li>
<li>single_trial_estimates(my_model), to use this function to build
single-trial design matrices and fit them.</li>
</ol>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">build_single_trial</span><span class="p">(</span><span class="n">fmri_model_obj</span><span class="p">,</span> <span class="n">inputhrf</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>inputhrf:</strong></dt>
<dd><p class="first last">should be a cell array of length nconds (number of conditions).</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_model.get_condition_assignments">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">get_condition_assignments</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.get_condition_assignments" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Condition assignments</dt>
<dd><ul class="first last simple">
<li>Indicator matrix coding for which columns in X belong to the same
modeled condition, and are part of the same HRF fit</li>
<li>There is one set of columns for each condition modeled, and one set of
columns for each parametric modulator of each condition</li>
<li>Because parametric modulators may not exist for all conditions, we need
to build this dynamically for modulators.</li>
</ul>
</dd>
</dl>
<p>Design matrix build (which calls method get_session_X) builds columns in
this order:</p>
<p>All within Session:
Regressors of interest, basis functions within conditions
Parametric modulators, basis functions within conditions
Covariates of no interest</p>
<p>Then:
Baselines (session/run intercepts)</p>
<p>This method is called automatically in the build method.</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_model.get_session_X">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">get_session_X</code><span class="sig-paren">(</span><em>obj</em>, <em>s</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.get_session_X" title="Permalink to this definition">¶</a></dt>
<dd><p>Get design matrix (predictors) for one session of fmri_model object, using
basis functions defined in the object and onsets for one session (s).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[Xs, delta, C, B, names] = get_session_X(obj, session number)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_model.plot">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">plot</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.plot" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot an fmri_model object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">plot</span><span class="p">(</span><span class="n">obj</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_model.replace_basis_set">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">replace_basis_set</code><span class="sig-paren">(</span><em>obj</em>, <em>condition_num</em>, <em>xBF_hires</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.replace_basis_set" title="Permalink to this definition">¶</a></dt>
<dd><p>Replace a basis set in an fmri_model object with another one of your
choosing.</p>
<p>This allows one to use a custom basis set, and also to use different
basis sets for different trial types.</p>
<p>Each condition across all sessions must be modeled with the same basis
set. That is, there can be only one basis set per condition, e.g., one
for anticipation (used in each session) and one for pain.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">replace_basis_set</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">condition_num</span><span class="p">,</span> <span class="n">xBF_hires</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% generate a custom spline basis set and use that for Condition 1,
% and the standard one for Condition 2:

[xBF_hires, xBF] = fmri_spline_basis(2, &#39;length&#39;, 12, &#39;nbasis&#39;, 3, &#39;order&#39;, 3, &#39;plot&#39;);

%save this to get info that is not typically in basis set until after
%model is built.
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_model.robustfit">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">robustfit</code><span class="sig-paren">(</span><em>fmri_model_obj</em>, <em>fmri_data_obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.robustfit" title="Permalink to this definition">¶</a></dt>
<dd><p>Robust fit for a model object to data object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>robustfit(fmri_model_obj, fmri_data_obj, [optional args])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Features:</th><td class="field-body"><p class="first">spatial smoothing of weights at 12 mm FWHM
ridge regression <strong>*not yet*</strong></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Preproc scaling:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><ol class="first arabic simple">
<li>Remove covariates using ridge reg; ridge trace for full model</li>
<li>scale to % signal change across time (cols) OR rank time points (for
w/i ss predictions??) AND/OR rank or center rows (images; for &#8216;shape&#8217;
analysis</li>
</ol>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"><p class="first">%sig across time, rank across rows: relative % sig change</p>
<p>Different models of noise lead to different ideas about optimal preproc
If large diffs in nuisance scaling in BOLD across individuals, ranking cols may
be good idea. but then individual diffs in overall activity will be removed...</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Options:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>tune:</strong></dt>
<dd><p class="first last">tuning const for robust reg</p>
</dd>
<dt><strong>iter:</strong></dt>
<dd><p class="first last">&#8216;maxiterations&#8217;, robust reg /WLS iterations. 1 = OLS only!</p>
</dd>
<dt><strong>smooth:</strong></dt>
<dd><p class="first last">&#8216;spatial_smooth_fwhm&#8217;, 0 or smoothing kernel for weights</p>
</dd>
<dt><strong>nosmooth:</strong></dt>
<dd><p class="first last">spatial_smooth_fwhm = 0;</p>
</dd>
<dt><strong>stats:</strong></dt>
<dd><p class="first last">&#8216;calculate_stats&#8217;, calculate_stats = 1; IN DEVELOPMENT</p>
</dd>
<dt><strong>noresiduals:</strong></dt>
<dd><p class="first last">write_residuals = 0;</p>
</dd>
<dt><strong>noplots:</strong></dt>
<dd><p class="first last">save_plots = 0;</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_model.rotate_to_pca">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">rotate_to_pca</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.rotate_to_pca" title="Permalink to this definition">¶</a></dt>
<dd><p>Rotate design matrix columns within all conditions to principal component projection.</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmri_model.single_trial_estimates">
<code class="descclassname">&#64;fmri_model.</code><code class="descname">single_trial_estimates</code><span class="sig-paren">(</span><em>obj</em>, <em>fmri_data_obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmri_model.single_trial_estimates" title="Permalink to this definition">¶</a></dt>
<dd><p>Write single trial estimates associated with an estimated fmri_model object.
must have estimated the model (robustfit(obj); see fmri_model.robustfit)
and saved hrf*.img images for each condition.</p>
<p>Also input an fmri_data object with time series data.</p>
<p>This function writes images, one 4-D image for each condition, with the
number of frames equalling the number of trials (onsets) for that
condition.</p>
<p>It does this by constructing a separate design matrix for each voxel,
which is based on the HRF estimates for that voxel for each condition.
Fits for all conditions are added to the same model, so that their
colinearity influences the single-trial parameter estimates.</p>
</dd></dl>

</div>
<div class="section" id="module-&#64;fmridisplay">
<span id="fmridisplay"></span><h2>fmridisplay<a class="headerlink" href="#module-@fmridisplay" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="&#64;fmridisplay.addblobs">
<code class="descclassname">&#64;fmridisplay.</code><code class="descname">addblobs</code><span class="sig-paren">(</span><em>obj</em>, <em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmridisplay.addblobs" title="Permalink to this definition">¶</a></dt>
<dd><p>This is a method for fmridisplay objects that adds blobs to one or more montages and other surface plot(s).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">addblobs</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">cl</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>See addthreshblobs and multi_threshold methods for a multiple thresholds version
render_blobs does most of the hard work.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">an fmridisplay object</p>
</dd>
<dt><strong>cl:</strong></dt>
<dd><p class="first last">a region object. If you&#8217;re using an fmri_data object pass in region(fmri_data_obj)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>There are many optional inputs that control display features of blobs.
These are determined by render_blobs</p>
<blockquote>
<div><p><strong>COLOR:</strong></p>
<dl class="docutils">
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd>followed by color vector, e.g., [0 1 1]</dd>
<dt><strong>&#8216;maxcolor&#8217;:</strong></dt>
<dd>followed by color vector for max color range, e.g., [0 1 1]</dd>
<dt><strong>&#8216;mincolor&#8217;:</strong></dt>
<dd>followed by color vector for min color range, e.g., [0 0 1]</dd>
<dt><strong>&#8216;onecolor&#8217;:</strong></dt>
<dd>force solid-color blobs</dd>
<dt><strong>&#8216;splitcolor&#8217;:</strong></dt>
<dd>Positive and negative values are mapped to different
colormaps. Default is +=hot, -=cool colors. Followed
optionally by cell array with 
vectors of 4 colors defining max/min for +/- range,
e.g., {[0 0 1] [.3 0 .8] [.8 .3 0] [1 1 0]}</dd>
</dl>
<p><strong>&#8216;OUTLINING:</strong></p>
<p><a href="#id3"><span class="problematic" id="id4">**</span></a>&#8216;&#8217;outline&#8217;</p>
<dl class="docutils">
<dt><strong>&#8216;linewidth&#8217;:</strong></dt>
<dd><p class="first">followed by width value, e.g., 1</p>
<p class="last">Note: add &#8216;no_surface&#8217; to stop updating the existing surface blobs</p>
</dd>
</dl>
<p><strong>&#8216;COLOR RANGE:</strong></p>
<dl class="docutils">
<dt><strong>&#8216;&#8217;cmaprange&#8217;:</strong></dt>
<dd>followed by range of values, e.g., [0 40], [-3 3]. Used in
color and transparency setting under some circumstances.</dd>
<dt><strong>&#8216;TRANSPARENCY:</strong></dt>
<dd>{&#8216;trans&#8217;, &#8216;transparent&#8217;,&#8217;scaledtransparency&#8217;, &#8216;constanttrans&#8217;, [val], &#8216;transvalue&#8217;, [val]}</dd>
<dt><strong>&#8216;trans&#8217;:</strong></dt>
<dd>Transparent blobs; with no other input, transparency = 0.75 (1 is opaque, 0 is transparent/invisible)</dd>
<dt><strong>&#8216;scaledtransparency&#8217;:</strong></dt>
<dd>Transparency is a function of voxel value, lower values are more transparent</dd>
<dt><strong>&#8216;transvalue&#8217;:</strong></dt>
<dd>Followed by width value, e.g., 1. also &#8216;constanttrans&#8217;</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Other Options:</th><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;smooth&#8217;:</strong></dt>
<dd><p class="first last">Smooth blobs</p>
</dd>
</dl>
<p><strong>&#8216;contour&#8217;:</strong></p>
<dl class="last docutils">
<dt><strong>&#8216;no_surface&#8217;:</strong></dt>
<dd><p class="first last">Do not add blobs to surface handles, if they exist</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>CONTROL OF WHICH MONTAGE</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;wh_montages&#8217;:</strong></dt>
<dd>followed by vector of montage numbers as they appear in
the list of registered montages in the fmridisplay object</dd>
</dl>
</div></blockquote>
<p>CONTROL OF WHICH SURFACE</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;wh_surfaces&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by vector of surface numbers as they appear in</dt>
<dd>the list of registered surfaces in the fmridisplay object</dd>
</dl>
</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>obj = addblobs(obj, cl, &#39;color&#39;, [0 1 1]);
obj = addblobs(obj, cl, &#39;color&#39;, [0 0 1], &#39;outline&#39;);
obj = addblobs(obj, cl, &#39;color&#39;, [0 1 0], &#39;outline&#39;, &#39;linewidth&#39;, 1, &#39;smooth&#39;);
obj = addblobs(obj, cl, &#39;color&#39;, [1 0 0], &#39;smooth&#39;, &#39;cmaprange&#39;, [0 40]);
obj = addblobs(obj, cl, ... &#39;wh_montages&#39;, 1);
obj = addblobs(obj, cl, &#39;splitcolor&#39;, ... &#39;cmaprange&#39;, ... &#39;trans&#39;);
</pre></div>
</div>
<p>Add only to montage 2 in vector of montages in obj.montage</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">addblobs</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">cl</span><span class="p">,</span> <span class="s">&#39;which_montages&#39;</span><span class="p">,</span> <span class="mi">2</span><span class="p">);</span> 
</pre></div>
</div>
<p>Map values to the colormap red-&gt;yellow.</p>
<p>This uses the default percentile-based mapping so that 20% of voxels
will have the low color and 20% will have the high color, and the rest will be in between:</p>
<div class="highlight-python"><div class="highlight"><pre>obj = addblobs(obj, cl, &#39;maxcolor&#39;, [1 1 0], &#39;mincolor&#39;, [1 0 0]);
</pre></div>
</div>
<p>Same, but now Map a specific range of values in image ([0 to .05])</p>
<div class="highlight-python"><div class="highlight"><pre>obj = addblobs(obj, cl, &#39;maxcolor&#39;, [1 1 0], &#39;mincolor&#39;, [1 0 0], &#39;cmaprange&#39;, [0 .05]);
</pre></div>
</div>
<p>Separate positive and negative activations and map to a split colormap;</p>
<p>See render_blobs</p>
<div class="highlight-python"><div class="highlight"><pre>o2 = addblobs(o2, cl, &#39;splitcolor&#39;, {[0 0 1] [.3 0 .8] [.8 .3 0] [1 1 0]}, &#39;wh_montages&#39;, 1);
</pre></div>
</div>
<p>It is possible to transparency-map values in a statistic image so you
can show &#8216;unthresholded&#8217; statistic values.  e.g.:</p>
<div class="highlight-python"><div class="highlight"><pre>o2 = addblobs(o2, cl, &#39;splitcolor&#39;, {[0 0 1] [0 1 1] [1 .5 0] [1 1 0]}, &#39;cmaprange&#39;, [-2 2], &#39;trans&#39;, &#39;scaledtransparency&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmridisplay.addpoints">
<code class="descclassname">&#64;fmridisplay.</code><code class="descname">addpoints</code><span class="sig-paren">(</span><em>obj</em>, <em>xyz</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmridisplay.addpoints" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots points on fmridisplay objects (e.g., montages of slices)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">newax</span> <span class="o">=</span> <span class="n">addpoints</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">xyz</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Registers handles with the object (referred to as obj)</p>
<ul class="simple">
<li>enter xyz as n x 3 list of coordinates in mm to plot (world space)</li>
<li>Points or text labels or both</li>
<li>Flexible slice spacing, colors, marker sizes/styles, axis layout (one row/standard square)</li>
<li>axial, sagittal, or coronal orientation handled automatically</li>
<li>Multiple different sets of points can be plotted in different colors/text labels</li>
</ul>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Takes all inputs of plot_points_on_slice.  See help for additional
documentation of options.</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>{&#8216;text&#8217;, &#8216;textcodes&#8217;}:</strong></dt>
<dd>cell array of text values corresponding to points</dd>
<dt><strong>{&#8216;condf&#8217; &#8216;colorcond&#8217;}:</strong></dt>
<dd>vector of integers to define color conditions</dd>
<dt><strong>&#8216;close_enough&#8217;:</strong></dt>
<dd>mm within which to plot; defined automatically based on slice distance if not entered</dd>
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd>string, &#8216;b&#8217;, or vector, [1 0 0], to define colors; cell if condf is used, e.g., {&#8216;b&#8217; &#8216;g&#8217;}</dd>
<dt><strong>{&#8216;marker&#8217;, &#8216;MarkerStyle&#8217;}:</strong></dt>
<dd>e.g., &#8216;o&#8217;, &#8216;v&#8217;, &#8216;s&#8217;</dd>
</dl>
<p><strong>{&#8216;MarkerSize&#8217;, &#8216;markersize&#8217;}:</strong></p>
<dl class="docutils">
<dt><strong>{&#8216;MarkerFaceColor&#8217;, &#8216;markerfacecolor&#8217;}:</strong></dt>
<dd>see color above</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Plot points (i.e., coordinate locations) for xyz coords:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">o2</span> <span class="o">=</span> <span class="n">addpoints</span><span class="p">(</span><span class="n">o2</span><span class="p">,</span> <span class="n">DB</span><span class="o">.</span><span class="n">xyz</span><span class="p">,</span> <span class="s">&#39;MarkerFaceColor&#39;</span><span class="p">,</span> <span class="s">&#39;b&#39;</span><span class="p">,</span> <span class="s">&#39;Marker&#39;</span><span class="p">,</span> <span class="s">&#39;o&#39;</span><span class="p">,</span> <span class="s">&#39;MarkerSize&#39;</span><span class="p">,</span> <span class="mi">4</span><span class="p">);</span>
<span class="n">o2</span> <span class="o">=</span> <span class="n">addpoints</span><span class="p">(</span><span class="n">o2</span><span class="p">,</span> <span class="n">DB</span><span class="o">.</span><span class="n">xyz</span><span class="p">,</span> <span class="s">&#39;text&#39;</span><span class="p">,</span> <span class="n">DB</span><span class="o">.</span><span class="n">textcodes</span><span class="p">,</span> <span class="s">&#39;condf&#39;</span><span class="p">,</span> <span class="n">DB</span><span class="o">.</span><span class="n">condf</span><span class="p">,</span> <span class="s">&#39;color&#39;</span><span class="p">,</span> <span class="p">{</span><span class="s">&#39;b&#39;</span> <span class="s">&#39;g&#39;</span><span class="p">});</span>
<span class="n">o2</span> <span class="o">=</span> <span class="n">removepoints</span><span class="p">(</span><span class="n">o2</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmridisplay.addthreshblobs">
<code class="descclassname">&#64;fmridisplay.</code><code class="descname">addthreshblobs</code><span class="sig-paren">(</span><em>obj</em>, <em>statimg</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmridisplay.addthreshblobs" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">addthreshblobs</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">statimg</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Add blobs from a statistical image object at multiple thresholds to
montage and other surface plots.</p>
<p>It passes options specified in varargin onto addblobs, see help addblobs
for options. In addition it has the parameters &#8216;thresh&#8217; and
&#8216;pruneclusters&#8217; are used and have default values.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">fmridisplay object, e.g. from a montage</p>
</dd>
<dt><strong>statimg:</strong></dt>
<dd><p class="first last">statistics_image object</p>
</dd>
<dt><strong>&#8216;thresh&#8217;,{&#8216;fdr&#8217;,0.01}:</strong></dt>
<dd><p class="first">cell array of p-value thresholds in ascending order. 
can be &#8216;fdr&#8217;, or uncorrecetd p-value. defaults to:</p>
<p class="last">thresh = {&#8216;fdr&#8217;,&#8216;0.001&#8217;,&#8216;0.01&#8217;}</p>
</dd>
<dt><strong>&#8216;pruneclusters&#8217;, [0 or 1]:</strong></dt>
<dd><p class="first last">Prune clusters that do not have at least one voxel
surving at the most stringent threshold. Defaults to 1.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Add blobs from statimg at 3 significance levels [FDR, 0.001 unc., 0.01 unc.].
Prune clusters.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">addthreshblobs</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">statimg</span><span class="p">);</span> 
</pre></div>
</div>
<p>Add blobs from statimg at 3 significance levels [FDR, 0.001 unc., 0.01 unc.]. 
Prune clusters.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">addthreshblobs</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">statimg</span><span class="p">,</span> <span class="s">&#39;thresh&#39;</span><span class="p">,</span> <span class="p">{</span><span class="s">&#39;fdr&#39;</span><span class="p">,</span> <span class="mf">0.001</span><span class="p">,</span> <span class="mf">0.01</span><span class="p">});</span> 
</pre></div>
</div>
<p>Add blobs from statimg at 2 significance levels [FDR, 0.001 unc.]. 
Do not prune clusters</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">addthreshblobs</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">statimg</span><span class="p">,</span> <span class="s">&#39;thresh&#39;</span><span class="p">,</span> <span class="p">{</span><span class="s">&#39;fdr&#39;</span><span class="p">,</span> <span class="mf">0.001</span><span class="p">},</span><span class="s">&#39;pruneclusters&#39;</span><span class="p">,</span><span class="mi">0</span><span class="p">);</span>
</pre></div>
</div>
<p>set defaults for thresholds and cluster pruning</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmridisplay.legend">
<code class="descclassname">&#64;fmridisplay.</code><code class="descname">legend</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmridisplay.legend" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates legend for fmridisplay object
Adds legend axis handles to obj.activation_maps{:}</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>obj = legend(obj, varargin)

obj = legend(obj, &#39;figure&#39;) % new figure
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;fmridisplay.montage">
<code class="descclassname">&#64;fmridisplay.</code><code class="descname">montage</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmridisplay.montage" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Creates montage of slices</dt>
<dd><ul class="first last simple">
<li>Solid brain slices or contour outlines</li>
<li>Points or text labels or both</li>
<li>Flexible slice spacing, colors, marker sizes/styles, axis layout (one row/standard square)</li>
<li>axial or sagittal orientation</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">montage</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Takes all inputs of plot_points_on_slice.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>{&#8216;noslice&#8217;, &#8216;nodraw&#8217;}:</strong></dt>
<dd><p class="first last">drawslice = 0;</p>
</dd>
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd><p class="first last">color = varargin{i+1}; varargin{i+1} = [];</p>
</dd>
<dt><strong>&#8216;marker&#8217;:</strong></dt>
<dd><p class="first last">marker = varargin{i+1}; varargin{i + 1} = [];</p>
</dd>
<dt><strong>{&#8216;wh_slice&#8217;}:</strong></dt>
<dd><p class="first last">wh_slice = varargin{i+1};</p>
</dd>
<dt><strong>{&#8216;close&#8217;, &#8216;closeenough&#8217;, &#8216;close_enough&#8217;}:</strong></dt>
<dd><p class="first last">close_enough =    varargin{i+1};</p>
</dd>
<dt><strong>{&#8216;sagg&#8217;,&#8217;sagittal&#8217;,&#8217;sagittal&#8217;}:</strong></dt>
<dd><p class="first last">orientation = &#8216;sagittal&#8217;;</p>
</dd>
<dt><strong>{&#8216;MarkerSize&#8217;, &#8216;markersize&#8217;}:</strong></dt>
<dd><p class="first last">markersize = varargin{i+1};</p>
</dd>
<dt><strong>{&#8216;MarkerFaceColor&#8217;, &#8216;markerfacecolor&#8217;}:</strong></dt>
<dd><p class="first last">facecolor = varargin{i+1};</p>
</dd>
<dt><strong>&#8216;solid&#8217;:</strong></dt>
<dd><p class="first last">disptype = &#8216;solid&#8217;;</p>
</dd>
<dt><strong>&#8216;overlay&#8217;:</strong></dt>
<dd><p class="first last">ovl = varargin{i + 1};
NOTE! DO NOT ENTER THIS HERE; ENTER WHEN YOU INITIALIZE FMRIDISPLAY OBJECT</p>
</dd>
<dt><strong>{&#8216;text&#8217;, &#8216;textcodes&#8217;}:</strong></dt>
<dd><p class="first last">textcodes = varargin{i + 1};</p>
</dd>
<dt><strong>{&#8216;condf&#8217; &#8216;colorcond&#8217;}:</strong></dt>
<dd><p class="first last">condf = varargin{i + 1};</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>In addition:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;onerow&#8217;:</strong></dt>
<dd>arrange axes in one row</dd>
<dt><strong>&#8216;slice_range&#8217;:</strong></dt>
<dd>[min max] values in mm for slices to plot</dd>
<dt><strong>&#8216;wh_slice&#8217; or &#8216;custom_coords&#8217;:</strong></dt>
<dd><blockquote class="first">
<div>followed my mm values for slices desired</div></blockquote>
<p class="last">e.g., for cluster/region centers, xyz = cat(1, cl.mm_center)
o2 = montage(o2, &#8216;axial&#8217;, &#8216;wh_slice&#8217;, xyz, &#8216;onerow&#8217;);
o2 = montage(o2, &#8216;sagittal&#8217;, &#8216;wh_slice&#8217;, xyz, &#8216;onerow&#8217;);</p>
</dd>
<dt><strong>&#8216;spacing&#8217;:</strong></dt>
<dd>followed by inter-slice spacing in mm</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">an fmridisplay object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Properties:</th><td class="field-body"><ul class="first last simple">
<li>overlay: [1x105 char]</li>
<li>SPACE: [1x1 struct]</li>
<li>activation_maps: {[1x1 struct]}</li>
<li>montage: {[1x1 struct]}</li>
<li>surface: {}</li>
<li>orthviews: {}</li>
<li>history: {}</li>
<li>history_descrip: []</li>
<li>additional_info: &#8216;&#8217;</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>Examples:</p>
<div class="highlight-python"><div class="highlight"><pre>o2 = fmridisplay; % create starting fmridisplay container object
</pre></div>
</div>
<p>Define new axes in existing figure, and use those for montage:</p>
<div class="highlight-python"><div class="highlight"><pre>axh = axes(&#39;Position&#39;, [0.05 0.4 .1 .5]);
o2 = montage(o2, &#39;sagittal&#39;, &#39;wh_slice&#39;, xyz(1,:), &#39;existing_axes&#39;, axh);

o2 = montage(o2, &#39;sagittal&#39;, &#39;slice_range&#39;, [-10 10], &#39;onerow&#39;);
o2 = montage(o2, &#39;axial&#39;, &#39;slice_range&#39;, [-40 50], &#39;onerow&#39;, &#39;spacing&#39;, 4);
o2 = montage(o2, &#39;axial&#39;, &#39;slice_range&#39;, [-20 30], &#39;onerow&#39;, &#39;spacing&#39;, 8);
o2 = montage(o2, &#39;axial&#39;, &#39;wh_slice&#39;, xyz, &#39;onerow&#39;);

% Parasagittal only:
o2 = montage(o2, &#39;sagittal&#39;, &#39;slice_range&#39;, [-4 4], &#39;onerow&#39;, &#39;spacing&#39;, 8);
</pre></div>
</div>
<p>Add/remove blobs and points with fmridisplay.addblobs,
fmridisplay.addpoints, fmridisplay.removeblobs, fmridisplay.removepoints</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>fmridisplay, cluster_orthviews, montage_clusters and variants</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmridisplay.surface">
<code class="descclassname">&#64;fmridisplay.</code><code class="descname">surface</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmridisplay.surface" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds surfaces of brain to figure</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">surface</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">fmridisplay</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">an fmridisplay object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Properties:</th><td class="field-body"><ul class="first simple">
<li>overlay: &#8216;&#8217;</li>
<li>SPACE: &#8216;&#8217;</li>
<li>activation_maps: {}</li>
<li>montage: {}</li>
<li>surface: {[1x1 struct]}</li>
<li>orthviews: {}</li>
<li>history: {}</li>
<li>history_descrip: []</li>
<li>additional_info: &#8216;&#8217;</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"><p class="first last">o2 = surface(o2, axes, [0.15 0.28 .15 1], &#8216;direction&#8217;, &#8216;hires right&#8217;, &#8216;orientation&#8217;, &#8216;lateral&#8217;);</p>
</td>
</tr>
</tbody>
</table>
<p>See help fmridisplay</p>
</dd></dl>

<dl class="function">
<dt id="&#64;fmridisplay.transparency_change">
<code class="descclassname">&#64;fmridisplay.</code><code class="descname">transparency_change</code><span class="sig-paren">(</span><em>o2</em>, <em>multval</em><span class="sig-paren">)</span><a class="headerlink" href="#@fmridisplay.transparency_change" title="Permalink to this definition">¶</a></dt>
<dd><p>Change the transparency of blobs in an fmridisplay object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>multval:</strong></dt>
<dd><p class="first last">multiply transparency values by this.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>values &lt; 1 makes blobs more transparent, &gt; 1 makes blobs more opaque</p>
</dd></dl>

</div>
<div class="section" id="fmridisplay-helper-functions">
<h2>fmridisplay_helper_functions<a class="headerlink" href="#fmridisplay-helper-functions" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-fmridisplay_helper_functions"></span><dl class="function">
<dt id="fmridisplay_helper_functions.clusters2mask2011">
<code class="descclassname">fmridisplay_helper_functions.</code><code class="descname">clusters2mask2011</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#fmridisplay_helper_functions.clusters2mask2011" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns 3-D mask of voxels included in a cl structure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">mask</span><span class="p">,</span> <span class="n">mask2</span><span class="p">]</span> <span class="o">=</span> <span class="n">clusters2mask2011</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span> <span class="p">[</span><span class="n">dim</span><span class="p">])</span>
</pre></div>
</div>
<p>Mask values are coded with the cluster index. That is, the non-zero entries
in mask are integers that reflect the index of the unique contiguous cluster to
which each voxel belongs.</p>
<p>Any non-zero value indicates membership in a cluster
Uses VOXEL values from cl, so define cl in the space you wish to have for
mask first!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Optional:</th><td class="field-body"><p class="first">A 2nd argument will be treated as &#8216;dim&#8217;, dimensions in voxels of the
new mask image.  If empty, uses max value in cluster to determine
automatically, but then the mask may not match image dimensions desired
for .img/.nii reading/writing purposes.</p>
<p class="last">If a second output is requested, the second output (maskz) is a mask
like the first, but the values in the mask reflect the numeric value
stored in the cl.Z field (whether Z-scores or other values, depending on
how cl is constructed.)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="fmridisplay_helper_functions.define_sampling_space">
<code class="descclassname">fmridisplay_helper_functions.</code><code class="descname">define_sampling_space</code><span class="sig-paren">(</span><em>V</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#fmridisplay_helper_functions.define_sampling_space" title="Permalink to this definition">¶</a></dt>
<dd><p>Define the sampling space of an image, with an upsampled space to 0.5 mm
resolution</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">SPACE</span> <span class="o">=</span> <span class="n">define_sampling_space</span><span class="p">(</span><span class="n">V</span><span class="p">,</span> <span class="p">[</span><span class="n">upsamplefactor</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>V:</strong></dt>
<dd><p class="first last">spm-style .mat structure, e.g., from spm_vol</p>
</dd>
<dt><strong>V.mat:</strong></dt>
<dd><p class="first last">4 x 4 matrix of voxel sizes and mm coords for the bottom
back left vox</p>
</dd>
<dt><strong>V.dim:</strong></dt>
<dd><p class="first last">dimensions of image</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Xo, Yo:</strong></dt>
<dd><p class="first last">Meshgrid for original voxel space</p>
</dd>
<dt><strong>X, Y:</strong></dt>
<dd><p class="first last">Meshgrid for upsampled voxel space at 0.5 mm resolution</p>
</dd>
<dt><strong>Xmm, Ymm:</strong></dt>
<dd><p class="first last">Meshgrid for upsampled space in mm</p>
</dd>
<dt><strong>xcoords, ycoords:</strong></dt>
<dd><p class="first last">mm coordinates for rows and cols for slice locations</p>
</dd>
<dt><strong>new_voxSize:</strong></dt>
<dd><p class="first last">new voxel size in mm for upsampled space</p>
</dd>
<dt><strong>usfactor:</strong></dt>
<dd><p class="first last">Upsampling factor for new sampleing space</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>overlay = which(&#39;SPM8_colin27T1_seg.img&#39;);  % spm8 seg cleaned up
V = spm_vol(overlay);
SPACE = define_sampling_space(V)

% Define mm sampling space in original voxel coord resolution
SPACE = define_sampling_space(V, 1)

original (o) and new (X, Y) grid space
xcoords, ycoords: mm coords centered on origin
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="fmridisplay_helper_functions.display_slice">
<code class="descclassname">fmridisplay_helper_functions.</code><code class="descname">display_slice</code><span class="sig-paren">(</span><em>dat</em>, <em>wh_slice</em>, <em>SPACE</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#fmridisplay_helper_functions.display_slice" title="Permalink to this definition">¶</a></dt>
<dd><p>Resample slice data in dat to SPACE and display</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">Z</span> <span class="o">=</span> <span class="n">display_slice</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">wh_slice</span><span class="p">,</span> <span class="n">SPACE</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="fmridisplay_helper_functions.map_to_world_space">
<code class="descclassname">fmridisplay_helper_functions.</code><code class="descname">map_to_world_space</code><span class="sig-paren">(</span><em>V</em><span class="sig-paren">)</span><a class="headerlink" href="#fmridisplay_helper_functions.map_to_world_space" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">SPACE</span> <span class="o">=</span> <span class="n">map_to_world_space</span><span class="p">(</span><span class="n">V</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>V:</strong></dt>
<dd><p class="first last">spm-style .mat structure, e.g., from spm_vol</p>
</dd>
<dt><strong>V.mat:</strong></dt>
<dd><p class="first last">4 x 4 matrix of voxel sizes and mm coords for the bottom
back left vox</p>
</dd>
<dt><strong>V.dim:</strong></dt>
<dd><p class="first last">dimensions of image</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">SPACE structure, with fields:</p>
<dl class="last docutils">
<dt><strong>Xmm, Ymm, Zmm:</strong></dt>
<dd><p class="first last">Meshgrid for voxel volume in mm space</p>
</dd>
<dt><strong>xcoords, ycoords, zcoords:</strong></dt>
<dd><p class="first last">mm coordinates for rows, cols, slices</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="fmridisplay_helper_functions.render_blobs">
<code class="descclassname">fmridisplay_helper_functions.</code><code class="descname">render_blobs</code><span class="sig-paren">(</span><em>currentmap</em>, <em>mymontage</em>, <em>SPACE</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#fmridisplay_helper_functions.render_blobs" title="Permalink to this definition">¶</a></dt>
<dd><p>This is a helper function for fmridisplay objects, called by the addblobs method</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">blobhan</span><span class="p">,</span> <span class="n">cmaprange</span><span class="p">,</span> <span class="n">mincolor</span><span class="p">,</span> <span class="n">maxcolor</span><span class="p">]</span> <span class="o">=</span> <span class="n">render_blobs</span><span class="p">(</span><span class="n">currentmap</span><span class="p">,</span> <span class="n">mymontage</span><span class="p">,</span> <span class="n">SPACE</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>See fmridisplay.m and addblobs.m method in fmridisplay for more details and options.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>currentmap:</strong></dt>
<dd><p class="first last">see addblobs method. Montage within fmridisplay object.</p>
</dd>
<dt><strong>mymontage:</strong></dt>
<dd><p class="first last">ditto</p>
</dd>
<dt><strong>SPACE:</strong></dt>
<dd><p class="first last">space of map to sample to (object display SPACE in fmridisplay object)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">There are many optional inputs that control display features of blobs.</p>
<p><strong>COLOR:</strong></p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd><p class="first last">followed by color vector, e.g., [0 1 1]</p>
</dd>
<dt><strong>maxcolor&#8217;:</strong></dt>
<dd><p class="first last">followed by color vector for max color range, e.g., [0 1 1]</p>
</dd>
<dt><strong>mincolor&#8217;:</strong></dt>
<dd><p class="first last">followed by color vector for min color range, e.g., [0 0 1]</p>
</dd>
<dt><strong>onecolor&#8217;:</strong></dt>
<dd><p class="first last">force solid-color blobs</p>
</dd>
<dt><strong>splitcolor&#8217;:</strong></dt>
<dd><p class="first last">Positive and negative values are mapped to different
colormaps. Default is +=hot, -=cool colors.  Followed
optionally by cell array with vectors of 4 colors defining
max/min for +/- range, e.g., {[0 0 1] [.3 0 .8] [.8 .3 0] [1 1 0]}</p>
</dd>
</dl>
</div></blockquote>
<p><strong>OUTLINING:</strong></p>
<blockquote>
<div><p><strong>&#8216;outline&#8217;</strong></p>
<dl class="docutils">
<dt><strong>&#8216;linewidth&#8217;:</strong></dt>
<dd><p class="first last">followed by width value, e.g., 1</p>
</dd>
</dl>
</div></blockquote>
<p><strong>COLOR RANGE:</strong></p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;cmaprange&#8217;:</strong></dt>
<dd><p class="first last">followed by range of values, e.g., [0 40], [-3 3]. Used in
color and transparency setting under some circumstances.</p>
</dd>
</dl>
</div></blockquote>
<p><strong>TRANSPARENCY:</strong></p>
<p>{&#8216;trans&#8217;, &#8216;transparent&#8217;,&#8217;scaledtransparency&#8217;, &#8216;constanttrans&#8217;, [val], &#8216;transvalue&#8217;, [val]}</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;trans&#8217;:</strong></dt>
<dd><p class="first last">Transparent blobs; with no other input, transparency = 0.75 (1 is opaque, 0 is transparent/invisible)</p>
</dd>
<dt><strong>&#8216;scaledtransparency&#8217;:</strong></dt>
<dd><p class="first last">Transparency is a function of voxel value, lower values are more transparent</p>
</dd>
<dt><strong>&#8216;transvalue&#8217;:</strong></dt>
<dd><p class="first last">Followed by width value, e.g., 1. also &#8216;constanttrans&#8217;</p>
</dd>
</dl>
</div></blockquote>
<p><strong>OTHER OPTIONS:</strong></p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;smooth&#8217;:</strong></dt>
<dd><p class="first last">Smooth blobs</p>
</dd>
</dl>
<p><strong>&#8216;contour&#8217;:</strong></p>
</div></blockquote>
<dl class="docutils">
<dt><strong>Orientation:</strong></dt>
<dd><p class="first last">&#8216;sagittal&#8217;, &#8216;coronal&#8217;, &#8216;axial&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first last">[blobhan, cmaprange, mincolor, maxcolor]</p>
</td>
</tr>
</tbody>
</table>
<p>All used in addblobs.m</p>
<p>Use addblobs; do not run this function directly unless you are
programming with it.</p>
<p>See also:
fmridisplay/addblobs, fmridisplay, fmridisplay/multi_threshold</p>
</dd></dl>

<dl class="function">
<dt id="fmridisplay_helper_functions.resample_space">
<code class="descclassname">fmridisplay_helper_functions.</code><code class="descname">resample_space</code><span class="sig-paren">(</span><em>dat</em>, <em>V</em>, <em>targetsp</em><span class="sig-paren">)</span><a class="headerlink" href="#fmridisplay_helper_functions.resample_space" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[resampled_dat, SPACEto] = resample_space(dat, V, [target V or target SPACE])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>imdat:</strong></dt>
<dd><p class="first last">3-D volume data</p>
</dd>
<dt><strong>V:</strong></dt>
<dd><p class="first last">spm-style .mat structure for dat, e.g., from spm_vol</p>
</dd>
<dt><strong>V.mat:</strong></dt>
<dd><dl class="first last docutils">
<dt>4 x 4 matrix of voxel sizes and mm coords for the bottom</dt>
<dd><p class="first last">back left vox</p>
</dd>
</dl>
</dd>
<dt><strong>V.dim:</strong></dt>
<dd><p class="first last">dimensions of image</p>
</dd>
</dl>
<p><strong>targetsp:</strong></p>
<blockquote>
<div><dl class="docutils">
<dt><strong>target V:</strong></dt>
<dd><p class="first last">spm-style .mat structure defining space to transform to</p>
</dd>
</dl>
<p class="attribution">&mdash;OR &#8211;</p>
</div></blockquote>
<blockquote>
<div><dl class="docutils">
<dt><strong>target SPACE:</strong></dt>
<dd><p class="first last">target SPACE, with Xmm, Ymm, Zmm; see map_to_world_space.m</p>
</dd>
</dl>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>resampled_dat:</strong></dt>
<dd><p class="first last">data sampled in new space</p>
</dd>
</dl>
<p><strong>SPACE structure, with fields:</strong></p>
<blockquote class="last">
<div><dl class="docutils">
<dt><strong>Xmm, Ymm, Zmm:</strong></dt>
<dd><p class="first last">Meshgrid for voxel volume in mm space</p>
</dd>
<dt><strong>xcoords, ycoords, zcoords:</strong></dt>
<dd><p class="first last">mm coordinates for rows, cols, slices</p>
</dd>
<dt><strong>V:</strong></dt>
<dd><p class="first last">Vol info structure for image in new space</p>
</dd>
</dl>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-&#64;image_vector">
<span id="image-vector"></span><h2>image_vector<a class="headerlink" href="#module-@image_vector" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="&#64;image_vector.apply_mask">
<code class="descclassname">&#64;image_vector.</code><code class="descname">apply_mask</code><span class="sig-paren">(</span><em>dat</em>, <em>mask</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.apply_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply a mask image (image filename or fmri_mask_image object) to an image_vector object
stored in dat.</p>
<dl class="docutils">
<dt>This can be used to:</dt>
<dd><ul class="first last simple">
<li>Mask an image_vector or fmri_data object with a mask</li>
<li>Obtain &#8220;pattern expression&#8221; for a weight map (entered as the
mask, here) in a series of images stored in dat.</li>
</ul>
</dd>
</dl>
<p>The mask or weight map does not have to be in the same space as the dat;
it will be resampled to the space of the data in dat.</p>
<p>To extract pattern expression values for each ROI within a mask use extract_roi_averages()</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>pattern_expression:</strong></dt>
<dd><p class="first last">calculate and return the cross-product of each
image in dat and the values in the mask.  This is useful if comparing
expression values that are comprised of different datasets or differing
number of voxels.</p>
</dd>
<dt><strong>correlation:</strong></dt>
<dd><p class="first last">calculate the pearson correlation coefficient of each
image in dat and the values in the mask.</p>
</dd>
<dt><strong>norm_mask:</strong></dt>
<dd><p class="first last">normalize the mask weights by L2 norm, for patt expression
only.</p>
</dd>
<dt><strong>ignore_missing:</strong></dt>
<dd><p class="first last">use with pattern expression only. Ignore weights on voxels
with zero values in test image. If this is not entered, the function will
check for these values and give a warning.</p>
</dd>
<dt><strong>invert:</strong></dt>
<dd><p class="first last">Invert the mask so that out-of-mask voxels are now in (using
the mask as an &#8216;exclude mask&#8217; rather than an include-mask. If pattern
expression is requested, the behavior is different, and it inverts the
sign of in-mask pattern weights.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[dat, mask] = apply_mask(dat, mask)
[dat, mask] = apply_mask(dat, mask image name)
[dat, mask] = apply_mask(dat, mask image vector object)
[pattern_exp_values] = apply_mask(dat, weight map image, &#39;pattern_expression&#39;, &#39;ignore_missing&#39;)
[pattern_exp_values] = apply_mask(dat, weight map image, &#39;pattern_expression&#39;, &#39;ignore_missing&#39;,&#39;correlation&#39;)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>extract_roi_averages, to get individual region averages / local pattern expression
apply_nps, which does whole-pattern and local regional expression</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.check_image_filenames">
<code class="descclassname">&#64;image_vector.</code><code class="descname">check_image_filenames</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.check_image_filenames" title="Permalink to this definition">¶</a></dt>
<dd><p>Check whether images listed in obj.fullpath actually exist</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">check_image_filenames</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;noverbose&#39;</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Behavior:</th><td class="field-body"><ul class="first last simple">
<li>If there are no file names, do nothing.</li>
<li>If file names are entered and full path is not, attempt to find full
path.</li>
<li>If full path info is entered, check to see if files exist.
Return output in obj.files_exist, and print a warning if only some exist.</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>Image names should be stored in .fullpath
abbreviated image names may be stored in image_names.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>fullpath should have full path to each volume in a string matrixm, with
trailing ,volume# for 4-D images as per SPM style expanded list.</p>
<p>image_names should have image name only for each volume</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.compare_space">
<code class="descclassname">&#64;image_vector.</code><code class="descname">compare_space</code><span class="sig-paren">(</span><em>obj</em>, <em>obj2</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.compare_space" title="Permalink to this definition">¶</a></dt>
<dd><p>Compare spaces of two image_vector objects</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function isdiff = compare_space(obj, obj2)
</pre></div>
</div>
<p>Returns 0 if same, 1 if different spaces, 2 if no volInfo info for one or
more objects. 3 if same space, but different in-mask voxels in .dat or
volInfo.image_indx</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.extract_gray_white_csf">
<code class="descclassname">&#64;image_vector.</code><code class="descname">extract_gray_white_csf</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.extract_gray_white_csf" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts mean values (values) and top 5 component scores (components)
from each of gray, white, and CSF masks.
Images must be in standard MNI space for this to apply.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">values</span><span class="p">,</span> <span class="n">components</span><span class="p">]</span> <span class="o">=</span> <span class="n">extract_gray_white_csf</span><span class="p">(</span><span class="n">obj</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">an image_vector (e.g., fmri_data) object</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.extract_roi_averages">
<code class="descclassname">&#64;image_vector.</code><code class="descname">extract_roi_averages</code><span class="sig-paren">(</span><em>obj</em>, <em>mask</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.extract_roi_averages" title="Permalink to this definition">¶</a></dt>
<dd><p>This image_vector method a extracts and averages data stored in an fmri_data object 
from a set of ROIs defined in a mask.
It is <em>slightly</em> different from the fmri_data method, as fmri_data has
more fields.</p>
<p>This version requires the mask_image to be in the same space as the obj.</p>
<p>Regions to average over can be either regions of contiguous voxels
bounded by voxels with values of 0 or NaN, which are considered non-data
values, or regions defined by unique integer codes in the mask image
(i.e., for atlas images with unique codes for each defined region.)</p>
<p>Mask/Atlas image does NOT have to be in the same space as the images to
extract from.  It will be remapped/resliced.</p>
<p>Extracted data is returned in single data format.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cl = extract_roi_averages(image_vector obj, mask, [average_over])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><ul class="first simple">
<li>char array of strings containing 4D image file names (data extracted from these)</li>
<li>mask_image to extract from.</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>average_over:</strong></dt>
<dd><ul class="first last simple">
<li>Default: &#8216;contiguous_regions&#8217; to average over contiguous voxels
bounded by voxels of 0 or NaN (non-data values)</li>
<li>Alt. option = &#8216;unique_mask_values&#8217; to average over unique integer codes in the mask image
(i.e., for atlas images with unique codes for each defined region)</li>
</ul>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">imgs_to_extract_from</span> <span class="o">=</span> <span class="n">filenames</span><span class="p">(</span><span class="s">&#39;w*.nii&#39;</span><span class="p">,</span><span class="s">&#39;char&#39;</span><span class="p">);</span>
<span class="n">mask_image</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="s">&#39;anat_lbpa_thal.img&#39;</span><span class="p">);</span>
<span class="p">[</span><span class="n">cl</span><span class="p">,</span> <span class="n">imgdat</span><span class="p">]</span> <span class="o">=</span> <span class="n">extract_image_data</span><span class="p">(</span><span class="n">imgs_to_extract_from</span><span class="p">,</span> <span class="n">mask_image</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>For an non-object-oriented alternative, see extract_image_data.m</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.fastmontage">
<code class="descclassname">&#64;image_vector.</code><code class="descname">fastmontage</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.fastmontage" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates 3 separate montage views - ax, cor, sagg in a special figure window</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fastmontage</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="p">[</span><span class="n">myview</span><span class="p">],</span> <span class="p">[</span><span class="s">&#39;spacing&#39;</span><span class="p">,</span> <span class="n">slicespacing</span><span class="p">],</span> <span class="p">[</span><span class="s">&#39;vertical&#39;</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fastmontage</span><span class="p">(</span><span class="n">dat</span><span class="p">);</span>
<span class="n">fastmontage</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;coronal&#39;</span><span class="p">);</span>
<span class="n">fastmontage</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;sagittal&#39;</span><span class="p">,</span> <span class="s">&#39;spacing&#39;</span><span class="p">,</span> <span class="mi">10</span><span class="p">);</span>
<span class="n">fastmontage</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;sagittal&#39;</span><span class="p">,</span> <span class="s">&#39;spacing&#39;</span><span class="p">,</span> <span class="mi">10</span><span class="p">,</span> <span class="s">&#39;vertical&#39;</span><span class="p">);</span>
<span class="n">fastmontage</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;sagittal&#39;</span><span class="p">,</span> <span class="s">&#39;slices_per_row&#39;</span><span class="p">,</span> <span class="mi">12</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.flip">
<code class="descclassname">&#64;image_vector.</code><code class="descname">flip</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.flip" title="Permalink to this definition">¶</a></dt>
<dd><p>Flips an image_vector object left to right</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body">Input &#8216;mirror&#8217; to make a symmetrical image, averaging the left
and right hemispheres</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">dat</span> <span class="o">=</span> <span class="n">flip</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;mirror&#39;</span><span class="p">])</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.get_wh_image">
<code class="descclassname">&#64;image_vector.</code><code class="descname">get_wh_image</code><span class="sig-paren">(</span><em>dat</em>, <em>wh</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.get_wh_image" title="Permalink to this definition">¶</a></dt>
<dd><p>For an image_vector with multiple images (cases, contrasts, etc.), select a subset.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function obj_out = get_wh_image(obj1, wh)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj1:</strong></dt>
<dd><p class="first last">An image_vector object</p>
</dd>
<dt><strong>wh:</strong></dt>
<dd><p class="first last">An array indicating which images</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>my_image_vector = get_wh_image(dat, 3) %to get 3rd image
my_image_vector = get_wh_image(dat, [1 3]) %to get 1st and 3rd image
</pre></div>
</div>
<p>check that wh is in range</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.histogram">
<code class="descclassname">&#64;image_vector.</code><code class="descname">histogram</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.histogram" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="&#64;image_vector.history">
<code class="descclassname">&#64;image_vector.</code><code class="descname">history</code><span class="sig-paren">(</span><em>dat</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.history" title="Permalink to this definition">¶</a></dt>
<dd><p>Display history for image_vector object</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.horzcat">
<code class="descclassname">&#64;image_vector.</code><code class="descname">horzcat</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.horzcat" title="Permalink to this definition">¶</a></dt>
<dd><p>Implements the horzcat ([a b]) operator on image_vector objects across voxels.
Requires that each object has an equal number of columns and voxels</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function s = horzcat(varargin)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>c = [dat1 dat2];
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.ica">
<code class="descclassname">&#64;image_vector.</code><code class="descname">ica</code><span class="sig-paren">(</span><em>fmridat_obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.ica" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Spatial ICA of an fmri_data object</dt>
<dd><ul class="first last simple">
<li>icadat = ica(fmridat_obj, [number of ICs to save])</li>
<li>icadat is also an fmri_data object, with .dat field voxels x components</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Notes:</th><td class="field-body"><ul class="first last simple">
<li>icasig = W * mixedsig</li>
<li>icasig = icadat.dat&#8217; = W * fmridat_obj.dat&#8217;</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>A is scaled version of fmridat_obj.dat&#8217; * icadat.dat</p>
<p>A and W are stored in additional_info field of icadat</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.image_math">
<code class="descclassname">&#64;image_vector.</code><code class="descname">image_math</code><span class="sig-paren">(</span><em>obj1</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.image_math" title="Permalink to this definition">¶</a></dt>
<dd><p>Perform simple mathematical and boolean operations on image objects</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>obj_out = image_math(obj1, [optional inputs, e.g., a 2nd object, keywords])
</pre></div>
</div>
<dl class="docutils">
<dt>For objects: Type methods(object_name) for a list of special commands</dt>
<dd>Type help object_name.method_name for help on specific
methods.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj1:</strong></dt>
<dd><p class="first last">An image_vector object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>obj2:</strong></dt>
<dd><p class="first last">An additional image_vector object</p>
</dd>
<dt><strong>{&#8216;add&#8217;, &#8216;plus&#8217;}:</strong></dt>
<dd><dl class="first last docutils">
<dt>Keyword to perform image-wise addition of images in obj1</dt>
<dd><p class="first last">and obj2.  Assumes these are paired/matched objects.</p>
</dd>
</dl>
</dd>
<dt><strong>{&#8216;subtract&#8217;, &#8216;minus&#8217;}:</strong></dt>
<dd><dl class="first last docutils">
<dt>Keyword to perform image-wise subtraction of images</dt>
<dd><p class="first last">in obj1 and obj2</p>
</dd>
</dl>
</dd>
<dt><strong>{&#8216;cat&#8217;, &#8216;concatenate&#8217;}:</strong></dt>
<dd><dl class="first last docutils">
<dt>Concatenate obj1 and obj2 image-wise.  Requires same</dt>
<dd><p class="first last">number of voxels in both image sets.  Returns effects
codes of 1, -1 in obj_out.Y.</p>
</dd>
</dl>
</dd>
<dt><strong>{&#8216;power&#8217;}:</strong></dt>
<dd><dl class="first last docutils">
<dt>Keyword to raise data to power element-wise; obj.dat = obj.dat.^b;</dt>
<dd><p class="first last">Followed by exponent to apply (b)</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>obj_out:</strong></dt>
<dd><p class="first last">The result - an image_vector object</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>DEFAULTS AND INPUTS
..</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.image_similarity_plot">
<code class="descclassname">&#64;image_vector.</code><code class="descname">image_similarity_plot</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.image_similarity_plot" title="Permalink to this definition">¶</a></dt>
<dd><p>Point-biserial correlations between images in fmri_data obj and set of
&#8216;spatial basis function&#8217; images (e.g., &#8216;signatures&#8217; or pre-defined maps)</p>
<p>Usage:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stats</span> <span class="o">=</span> <span class="n">image_similarity_plot</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="s">&#39;average&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>This is a method for an image_vector object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">An image object with one or more images loaded</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>average:</strong></dt>
<dd><p class="first last">Calculate average over images in obj with standard errors
Useful if obj contains one image per subject and you want
to test similarity with maps statistically.
Default behavior is to plot each individual image.</p>
</dd>
<dt><strong>bucknerlab</strong></dt>
<dd><p class="first last">Use 7 network parcellation from Yeo et al. as basis for
comparisons</p>
</dd>
<dt><strong>kragelemotion</strong></dt>
<dd><p class="first">Use 7 emotion-predictive models from Kragel &amp; LaBar 2015 for
basis of comparisons</p>
<p class="last"><strong>compareGroups</strong>
Perform multiple one-way ANOVAs with group as a factor (one for
each spatial basis); requires group as subsequent input</p>
</dd>
<dt><strong>group</strong></dt>
<dd><p class="first last">Indicates group membership for each image</p>
</dd>
<dt><strong>noplot</strong></dt>
<dd><p class="first last">Omits plot (print stats only)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>stats:</strong></dt>
<dd><dl class="first last docutils">
<dt>Structure including:</dt>
<dd><ul class="first last simple">
<li>.r, Correlations in [7 networks x images in obj] matrix</li>
<li>.t, T-test (if &#8216;average&#8217; is specified)</li>
<li>.line_handles Handles to polar plot lines so you can
customize</li>
<li>.fill_handles Handles to polar plot fills so you can
customize</li>
<li>.table_spatial, ANOVA table with subject as row factor and
spatial basis as column factor (one way repeated measures
ANOVA, requires &#8216;average&#8217; to be specified)</li>
<li>.multcomp_spatial, multiple comparisons of means across
different spatial bases, critical value determined
by Tukey-Kramer method (see multcompare)</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>table_group</strong></dt>
<dd><p class="first last">multiple one-way ANOVA tables (one for each
spatial basis) with group as column factor (requires
&#8216;average&#8217; to be specified)</p>
</dd>
<dt><strong>multcomp_group</strong></dt>
<dd><p class="first last">mutiple comparisons of means across groups, one output
cell for each spatial basis, critical value determined
by Tukey-Kramer method (see multcompare)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% corrdat is an fmri_data object with 18 images from searchlight
% correlation in it.  Then:
stats = image_similarity_plot_bucknermaps(corrdat, &#39;average&#39;);

% t_diff is a thresholded statistic_image object
stats = image_similarity_plot_bucknermaps(t_diff);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>tor_polar_plot</p>
<p>List dates and changes here, and author of changes
11/30/2015 (Phil Kragel)</p>
<blockquote>
<div><ul class="simple">
<li>added anova (rm) comparing means across spatial bases</li>
<li>added anova (1-way) comparing means across groups for each spatial
basis (e.g., for each buckner network)</li>
</ul>
</div></blockquote>
<dl class="docutils">
<dt>12/15/2015 (Phil Kragel)</dt>
<dd><ul class="first last simple">
<li>added option to omit plotting</li>
</ul>
</dd>
</dl>
<p>DEFAULTS AND INPUTS
..</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.image_similarity_plot_bucknermaps">
<code class="descclassname">&#64;image_vector.</code><code class="descname">image_similarity_plot_bucknermaps</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.image_similarity_plot_bucknermaps" title="Permalink to this definition">¶</a></dt>
<dd><p>Point-biserial correlations between images in fmri_data obj and Bucker
Lab 7-network maps, with polar plot</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stats</span> <span class="o">=</span> <span class="n">image_similarity_plot_bucknermaps</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="s">&#39;average&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>This is a method for an image_vector object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">An image object with one or more images loaded</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>average:</strong></dt>
<dd><p class="first last">Calculate average over images in obj with standard errors
Useful if obj contains one image per subject and you want
to test similarity with maps statistically.
Default behavior is to plot each individual image.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>stats:</strong></dt>
<dd><dl class="first last docutils">
<dt>Structure including:</dt>
<dd><ul class="first last simple">
<li>.r, Correlations in [7 networks x images in obj] matrix</li>
<li>.t, T-test (if &#8216;average&#8217; is specified)</li>
<li>.line_handles Handles to polar plot lines so you can
customize</li>
<li>.fill_handles Handles to polar plot fills so you can
customize</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% corrdat is an fmri_data object with 18 images from searchlight
% correlation in it.  Then:
stats = image_similarity_plot_bucknermaps(corrdat, &#39;average&#39;);

% t_diff is a thresholded statistic_image object
stats = image_similarity_plot_bucknermaps(t_diff);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>tor_polar_plot</p>
<p>DEFAULTS AND INPUTS
..</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.interpolate">
<code class="descclassname">&#64;image_vector.</code><code class="descname">interpolate</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.interpolate" title="Permalink to this definition">¶</a></dt>
<dd><p>Interpolate over missing values in image_vector object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">dat</span> <span class="o">=</span> <span class="n">interpolate</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body">image_vector object (dat; e.g., an fmri_data object)</td>
</tr>
</tbody>
</table>
<p>Use when there are some missing values in the mask image
Performs 3-D linear interpolation to fill in all values in the original
mask.</p>
<p>e.g., For a standard brain image space that is 91 x 109 x 91, you may
have 300,000 in-mask values. Only 150,000 of these may be defined in the
image, however, and the rest are missing (0 or NaN).
This function will return a dat image with non-missing values for all
300,000 voxels (the &#8220;in-mask&#8221; space). 
It will not return values for all voxels in the 91 x 109 x 91 space,
however.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>This function does not upsample the data now, but could be extended
to do so fairly easily.</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.mean">
<code class="descclassname">&#64;image_vector.</code><code class="descname">mean</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.mean" title="Permalink to this definition">¶</a></dt>
<dd><p>Create an image_vector object with mean values for each voxel (cols)
across images (rows) of an fmri_data object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function m = mean(obj, [optional args])
</pre></div>
</div>
<p>m is an image_vector object whose data contains the mean values.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><ul class="first simple">
<li>&#8216;write&#8217;, followed by file name</li>
<li>&#8216;path&#8217;, followed by location for file (default = current directory)</li>
<li>&#8216;orthviews&#8217; -&gt; show orthviews for this image, same as orthviews(m)</li>
<li>&#8216;histogram&#8217; -&gt; show histogram for this image, same as histogram(m)</li>
<li>&#8216;plot&#8217; -&gt; do both</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% If sdat is an fmri_data object with multiple images,
m = mean(sdat, &#39;plot&#39;, &#39;write&#39;, anatmeanname, &#39;path&#39;, maskdir);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.minus">
<code class="descclassname">&#64;image_vector.</code><code class="descname">minus</code><span class="sig-paren">(</span><em>obj1</em>, <em>obj2</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.minus" title="Permalink to this definition">¶</a></dt>
<dd><p>Implements the minus (-) operator on image_vector objects across voxels.
Requires that each object has an equal number of columns and voxels</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.montage">
<code class="descclassname">&#64;image_vector.</code><code class="descname">montage</code><span class="sig-paren">(</span><em>image_obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.montage" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a montage of an image_vector (or statistic_image or fmri_data) object</p>
<p><a href="#id5"><span class="problematic" id="id6">*</span></a>Usage:</p>
<div class="highlight-python"><div class="highlight"><pre>[fig_handle or o2 fmridisp object] = montage(image_obj, [optional arguments])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>fmridisplay:</strong></dt>
<dd><p class="first last">for fmridisplay object style montage [default]</p>
</dd>
<dt><strong>scnmontage:</strong></dt>
<dd><p class="first last">for circa 2008-style SCN lab montage for each image vector</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">o2</span> <span class="o">=</span> <span class="n">montage</span><span class="p">(</span><span class="n">mask</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.orthviews">
<code class="descclassname">&#64;image_vector.</code><code class="descname">orthviews</code><span class="sig-paren">(</span><em>image_obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.orthviews" title="Permalink to this definition">¶</a></dt>
<dd><p>Orthviews display (SPM) for CANlab image_vector (or fmri_data, statistic_image) object</p>
<p><a href="#id7"><span class="problematic" id="id8">*</span></a>Usage:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">orthviews</span><span class="p">(</span><span class="n">image_obj</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>posneg:</strong></dt>
<dd><p class="first last">input generates orthviews using solid colors.</p>
</dd>
<dt><strong>largest_region:</strong></dt>
<dd><p class="first last">to center the orthviews on the largest region in the image</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.plot_current_orthviews_coord">
<code class="descclassname">&#64;image_vector.</code><code class="descname">plot_current_orthviews_coord</code><span class="sig-paren">(</span><em>dat</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.plot_current_orthviews_coord" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieves and plots the image data series at the current crosshairs in spm_orthviews</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.plus">
<code class="descclassname">&#64;image_vector.</code><code class="descname">plus</code><span class="sig-paren">(</span><em>obj1</em>, <em>obj2</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.plus" title="Permalink to this definition">¶</a></dt>
<dd><p>Implements the plus (+) operator on image_vector objects across voxels.
Requires that each object has an equal number of columns and voxels</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">c</span> <span class="o">=</span> <span class="n">dat1</span> <span class="o">+</span> <span class="n">dat2</span><span class="p">;</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.power">
<code class="descclassname">&#64;image_vector.</code><code class="descname">power</code><span class="sig-paren">(</span><em>obj</em>, <em>b</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.power" title="Permalink to this definition">¶</a></dt>
<dd><p>Implements the power (^) operator on image_vector objects across voxels.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">c</span> <span class="o">=</span> <span class="n">dat1</span><span class="o">^</span><span class="mi">2</span><span class="p">;</span>
</pre></div>
</div>
<p>Programmer Notes:
Created 3/14/14 by Luke Chang
..</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.preprocess">
<code class="descclassname">&#64;image_vector.</code><code class="descname">preprocess</code><span class="sig-paren">(</span><em>obj</em>, <em>meth</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.preprocess" title="Permalink to this definition">¶</a></dt>
<dd><p>Preprocesses data in an fmri_data object</p>
<p>Data is observations (i.e., voxels, subjects) x images, so operating on the columns operates on
images, and operating on the rows operates on voxels (or variables more
generally) across images.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first"><strong>meth:</strong> Options</p>
<dl class="docutils">
<dt><strong>resid:</strong></dt>
<dd><p class="first last">Residualize voxels with respect to covariates
Uses obj.covariates, obj.dat.
Adds intercept automatically. You can tell it to add the mean response per voxel back in:
obj = preprocess(obj, &#8216;resid&#8217;, [add mean back in flag])</p>
</dd>
<dt><strong>hpfilter:</strong></dt>
<dd><p class="first last">High-pass filter and remove run intercepts and first two
images per run. Uses obj.dat, obj.images_per_session
obj = preprocess(obj, &#8216;hpfilter&#8217;, HPlen in s, TR)</p>
</dd>
<dt><strong>windsorize:</strong></dt>
<dd><p class="first last">Windsorize entire data matrix to 3 STD</p>
</dd>
<dt><strong>windsorizevoxels:</strong></dt>
<dd><p class="first last">Windsorize each time series in data matrix to 3 STD</p>
</dd>
<dt><strong>session_outliers:</strong></dt>
<dd><p class="first last">Identify session-wise (run-wise) outliers with significant
based on mahalanobis distance with FDR-corrected P-values in chi-square test.
Impute session grand mean outliers.</p>
</dd>
<dt><strong>outliers:</strong></dt>
<dd><p class="first last">Identify outlier time points for each session based on
mahalanobis distance (see above) across global mean for slices and
spatial STD for slices, as in scn_session_spike_id.
Outliers at 3 SD based on timeseries added to obj.covariates.</p>
</dd>
<dt><strong>outliers_rmssd:</strong></dt>
<dd><p class="first last">Identify outlier time points for each session based on
root-mean-square successive differences between images (across voxels.)
this is the std (across voxels) of the successive diffs across images.
Outliers at 3.5 SD based on timeseries added to obj.covariates.</p>
</dd>
<dt><strong>smooth:</strong></dt>
<dd><dl class="first docutils">
<dt>Smoothed images with Gaussian filter</dt>
<dd><ul class="first last simple">
<li>obj = preprocess(obj, &#8216;smooth&#8217;, FWHM in mm)</li>
</ul>
</dd>
</dl>
<p class="last"><em>NOTE</em> SMOOTHING KERNEL MAY BE IN VOX, AS VOL INFO IS NOT PASSED IN</p>
</dd>
<dt><strong>interp_images:</strong></dt>
<dd><p class="first">Interpolate all voxels in a series of images specified
by logical vector whout.</p>
<blockquote class="last">
<div><ul class="simple">
<li>obj = preprocess(obj, &#8216;interp_images&#8217;, whout);</li>
</ul>
</div></blockquote>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% two complementary ways to get and plot outliers:
dat = preprocess(dat, &#39;outliers&#39;, &#39;plot&#39;);
subplot(5, 1, 5); % go to new panel...
dat = preprocess(dat, &#39;outliers_rmssd&#39;, &#39;plot&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.read_from_file">
<code class="descclassname">&#64;image_vector.</code><code class="descname">read_from_file</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.read_from_file" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads data from image filenames into obj.dat</p>
<p>Try obj = check_image_filenames(obj) first.</p>
<p>This is automatically called if you create a new image_vector object with
names but do not directly enter data. e.g., the commands below will load data:</p>
<blockquote>
<div><ul class="simple">
<li>name = &#8216;salientmap.nii&#8217;;</li>
<li>img = image_vector(&#8216;image_names&#8217;, name);</li>
</ul>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.rebuild_volinfo_from_dat">
<code class="descclassname">&#64;image_vector.</code><code class="descname">rebuild_volinfo_from_dat</code><span class="sig-paren">(</span><em>dat</em>, <em>newdat</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.rebuild_volinfo_from_dat" title="Permalink to this definition">¶</a></dt>
<dd><p>Will rebuild volInfo (the image space, or sometimes &#8220;mask&#8221;) from a vectorized image. 
In other words, will rebuild dat.volInfo from newdat.</p>
<p>Also resets all voxels to be significant, if a statistic image</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">an image_vector</p>
</dd>
<dt><strong>newdat:</strong></dt>
<dd><p class="first last">a vector that MUST be size of ENTIRE image (dat.volInfo.nvox)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">dat.dat contains the non-zero values of newdat, and dat.volInfo is
correctly defining the image space</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.reconstruct_image">
<code class="descclassname">&#64;image_vector.</code><code class="descname">reconstruct_image</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.reconstruct_image" title="Permalink to this definition">¶</a></dt>
<dd><p>Reconstruct a 3-D or 4-D image from image_vector object obj</p>
<p>voldata is and X x Y x Z x Images matrix
vectorized_voldata is the same, with all voxels vectorized</p>
<p>This output has one element for every voxel in THE ENTIRE IMAGE, and so
can be very memory-intensive.  But it&#8217;s useful for lining up voxels
across images with different masks/in-mask voxels.</p>
<p>This function returns output in memory;
see image_vector.write for writing .img files to disk.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>voldata:</strong></dt>
<dd><p class="first last">3-D recon volume</p>
</dd>
<dt><strong>vectorized_voldata:</strong></dt>
<dd><p class="first last">volume in column vetor, iimg_xxx function format</p>
</dd>
<dt><strong>xyz_coord_struct:</strong></dt>
<dd><dl class="first last docutils">
<dt>has fields with coordinate information in mm (world) space</dt>
<dd><ul class="first last simple">
<li>x, y, z : vectors of coordinates in mm for each of the 3
dimensions of the image</li>
<li>X, Y, Z : output matrices from meshgrid with mm coordinates,
for volume visualization.
These can be passed to surf or isocaps functions for volume
visualization in world space (mm).</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.remove_empty">
<code class="descclassname">&#64;image_vector.</code><code class="descname">remove_empty</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.remove_empty" title="Permalink to this definition">¶</a></dt>
<dd><p>remove vox: logical vector of custom voxels to remove, VOX x 1</p>
<p>remove im: logical vector of custom images to remove, 1 x IMAGES</p>
<p>indices of removed data will be stored in removed_voxels and
removed_images fields, to preserve ability to later reconstruct into 3D images</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>dat = remove_empty(dat, [logical vector of custom voxels to remove], [logical vector of imgs to remove])
</pre></div>
</div>
<p>Indicator vectors stored in:
removed_images
removed_voxels</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body">replace_empty</td>
</tr>
</tbody>
</table>
<p>force logical</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.reparse_contiguous">
<code class="descclassname">&#64;image_vector.</code><code class="descname">reparse_contiguous</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.reparse_contiguous" title="Permalink to this definition">¶</a></dt>
<dd><p>Re-construct list of contiguous voxels in an image based on in-image
voxel coordinates.  Coordinates are taken from obj.volInfo.xyzlist.</p>
<p>Results are saved in obj.volInfo.cluster.</p>
<p>xyzlist can be generated from iimg_read_img, and is done automatically by
object-oriented fMRI image classes (fmri_image, image_vector,
statistic_image)</p>
<p>If &#8216;nonempty&#8217; is entered as an optional argument, will use only voxels
that are non-zero, non-nan in all columns of obj.dat.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">reparse_contiguous</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;nonempty&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>.cluster and .xyzlist should both always be length v in-mask voxels
if &#8216;nonempty&#8217; is entered, then .dat should be length v in-mask voxels too</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.replace_empty">
<code class="descclassname">&#64;image_vector.</code><code class="descname">replace_empty</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.replace_empty" title="Permalink to this definition">¶</a></dt>
<dd><p>Replace empty/missing values in an image data object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>obj = replace_empty(obj, [optional keywords])
</pre></div>
</div>
<p>Replace missing values in obj.dat stored in obj.removed_voxels and
obj.removed_images with zeros.  This returns obj.dat in a format that can
be reconstructed into a 3-D or 4-D image matrix for brain visualization.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional keywords:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;voxels&#8217; or &#8216;images&#8217;:</strong></dt>
<dd><p class="first last">replace only missing voxels/images</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body">remove_empty, zeroinsert, nanremove, naninsert</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.resample_space">
<code class="descclassname">&#64;image_vector.</code><code class="descname">resample_space</code><span class="sig-paren">(</span><em>obj</em>, <em>sampleto</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.resample_space" title="Permalink to this definition">¶</a></dt>
<dd><p>Resample the images in an fmri_data object (obj) to the space of another
image (sampleto; e.g., a mask image). Works for all image_vector objects.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>obj = resample_space(obj, sampleto, [sampling method])
</pre></div>
</div>
<dl class="docutils">
<dt>Sampleto may be one of these:</dt>
<dd><ol class="first last arabic simple">
<li>a volInfo structure (the image does not have to exist on the path)</li>
<li>an image_vector, fmri_data, fmri_mask_image object</li>
<li>a string with the name of an image</li>
</ol>
</dd>
</dl>
<p>Can enter resampling method as optional input. Takes any input to
interp3:</p>
<blockquote>
<div><p>&#8216;nearest&#8217; - nearest neighbor interpolation
&#8216;linear&#8217;  - linear interpolation (default)
&#8216;spline&#8217;  - spline interpolation
&#8216;cubic&#8217;   - cubic interpolation as long as the data is uniformly</p>
<blockquote>
<div>spaced, otherwise the same as &#8216;spline&#8217;</div></blockquote>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>label_mask = fmri_data(which(&#39;atlas_labels_combined.img&#39;));
label_mask = resample_space(label_mask, ivec, &#39;nearest&#39;) % resamples and masks label image
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.resample_time">
<code class="descclassname">&#64;image_vector.</code><code class="descname">resample_time</code><span class="sig-paren">(</span><em>obj</em>, <em>source_TR</em>, <em>target_TR</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.resample_time" title="Permalink to this definition">¶</a></dt>
<dd><p>Resample the time-series images (source_time_interval) in an fmri_data object (obj) 
to the different time series (target_time_interval). Works for all image_vector objects.</p>
<blockquote>
<div><ul class="simple">
<li>obj = resample_time(obj, source_time_interval, target_time_interval, varargin)</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>meth (Interpolation methods):</strong></dt>
<dd><dl class="first last docutils">
<dt>You can enter resampling method as optional input. Takes any input to</dt>
<dd><ul class="first last">
<li><p class="first">&#8216;nearest&#8217; - nearest neighbor interpolation</p>
</li>
<li><p class="first">&#8216;linear&#8217;  - linear interpolation (default)</p>
</li>
<li><p class="first">&#8216;spline&#8217;  - spline interpolation</p>
</li>
<li><dl class="first docutils">
<dt>&#8216;cubic&#8217;   - cubic interpolation as long as the data is uniformly</dt>
<dd><p class="first last">spaced, otherwise the same as &#8216;spline&#8217;</p>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>slice:</strong></dt>
<dd><p class="first last">A fraction of the slice timing correction.
The default is 0.5, meaning if your TR is 2s, the time point of your TR image
will be considered as the middle point of the TR bins. You can use this option
to use different time points. If you are upsampling your data (i.e.,
your target TR is shorter than your source TR), you need to discard the
first column of your data. This function will return the first time point data as NaN.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>dat = fmri_data(&#39;/Volumes/RAID1/labdata/current/BMRK3/Imaging/spatiotemp_biomarker/STmarker1.img&#39;);
dat = resample_time(dat, 2, 1.3) 

% with options:
dat = resample_time(dat, 2, 1.3, &#39;meth&#39;, &#39;linear&#39;, &#39;slice&#39;, .3)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.sagg_slice_movie">
<code class="descclassname">&#64;image_vector.</code><code class="descname">sagg_slice_movie</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.sagg_slice_movie" title="Permalink to this definition">¶</a></dt>
<dd><p>Movie of successive differences (sagittal slice)
Enter an image_vector or fmri_data object (usually with time series)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">sagg_slice_movie</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="p">[</span><span class="n">full_path_of_movie_output_file</span><span class="p">,</span><span class="n">image_skip_interval</span><span class="p">])</span>
</pre></div>
</div>
<p><a href="#id9"><span class="problematic" id="id10">*</span></a>Optional Inputs:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>movie_output_file:</strong></dt>
<dd><dl class="first last docutils">
<dt>a char array detailing the full path to save the</dt>
<dd>movie file</dd>
</dl>
</dd>
<dt><strong>image_skip_interval:</strong></dt>
<dd><blockquote class="first">
<div>An integer value describing the interval 
between images in each subsequent frame of the movie</div></blockquote>
<p class="last">(default = 1)</p>
</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre> sagg_slice_movie(fmri_dat, ...
            &#39;/Volumes/engram/labdata/fmri_data/Study1/Subj1/qc_images&#39;, 5)

This would save an movie based on the images in fmri_dat to the
above directory, with an interval of 5 images between each
frame (so, the movie would show image 1, 6, 11, 16, etc)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.searchlight">
<code class="descclassname">&#64;image_vector.</code><code class="descname">searchlight</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.searchlight" title="Permalink to this definition">¶</a></dt>
<dd><p>Run searchlight multivariate prediction/classification on an image_vector
or fmri_data object OR two objects, for cross-prediction.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[list outputs here] = function_name(list inputs here, [optional inputs])
[results_obj, stats, indx] = searchlight(dat, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Features:</th><td class="field-body"><ul class="first last simple">
<li>Runs searchlight with standard, pre-defined algorithms</li>
<li>Custom-entry definition of holdout sets</li>
<li>Can re-use searchlight spheres after initial definition</li>
<li>Custom-entry definition of any spheres/regions of interest</li>
<li>Uses Matlab&#8217;s parallel processing toolbox (parfor)</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>Type help image_vector.searchlight to display this help information</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">image_vector or fmri_data object with data</p>
</dd>
<dt><strong>dat.Y:</strong></dt>
<dd><p class="first last">required: true outcomes for each observation (image) in dat</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>:Optional Inputs:* Keyword followed by input variable:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>r:</strong></dt>
<dd>searchlight radius, voxels</dd>
<dt><strong>dat2:</strong></dt>
<dd>second dataset, for cross-prediction</dd>
<dt><strong>indx:</strong></dt>
<dd>sparse logical matrix. each COLUMN is index of inclusion sets for each region/sphere in searchlight
This takes a long time to calculate, but can be saved and
re-used for a given mask</dd>
<dt><strong>holdout_set:</strong></dt>
<dd>Followed by integer vector of which observations belong to which
holdout set, for cross-validation. This is passed into fmri_data.predict.m.  Default is
empty.</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>results_obj:</strong></dt>
<dd><p class="first last">fmri_data object with results maps</p>
</dd>
<dt><strong>stats:</strong></dt>
<dd><p class="first last">selected statistics for each sphere in searchlight</p>
</dd>
<dt><strong>indx:</strong></dt>
<dd><p class="first last">sparse logical matrix. each COLUMN is index of inclusion sets for each region/sphere in searchlight
* this can be re-used for all data with the same mask/structure. *</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Define a sensible gray-matter mask:
dat = fmri_data(which(&#39;scalped_avg152T1_graymatter.img&#39;));
dat = threshold(dat, [.8 Inf], &#39;raw-between&#39;);
dat = trim_mask(dat);

% Create fake data and holdout indicator index vector
dat.dat = randn(dat.volInfo.n_inmask, 30);
dat.Y = dat.dat(111111, :)&#39; + .3 * randn(30, 1);
holdout_set = ones(6, 1); for i = 2:5, holdout_set = [holdout_set; i*ones(6, 1)]; end

% Run, and run again with existing indx
pool = parpool(12);  % initialize parallel processing (12 cores)
[results_obj, stats, indx] = searchlight(dat, &#39;holdout_set&#39;, holdout_set);
results_obj = searchlight(dat, &#39;holdout_set&#39;, holdout_set, &#39;indx&#39;, indx);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>region.m, fmri_data.predict.m</p>
<p>DEFAULTS AND INPUTS
..</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.slices">
<code class="descclassname">&#64;image_vector.</code><code class="descname">slices</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.slices" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a montage of single-slice results for every image in an
image_vector object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">o</span> <span class="o">=</span> <span class="n">slices</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="s">&#39;orientation&#39;</span><span class="p">,</span> <span class="p">[</span><span class="n">orientation</span><span class="p">],</span> <span class="s">&#39;slice&#39;</span><span class="p">,</span> <span class="p">[</span><span class="n">slice_mm</span><span class="p">],</span> <span class="s">&#39;nimages&#39;</span><span class="p">,</span> <span class="p">[</span><span class="n">nimgs</span><span class="p">])</span>
</pre></div>
</div>
<p>obj is an image_vector, fmri_data, or statistic_image object with
multiple images (only the first 64 will display), which are stored as
columns in its .dat field.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>orientation:</strong></dt>
<dd><p class="first last">can be followed by &#8216;sagittal&#8217;, &#8216;axial&#8217;, or &#8216;coronal&#8217;</p>
</dd>
<dt><strong>slice_mm:</strong></dt>
<dd><p class="first last">is followed by the mm coord of the slice to display; default = 0</p>
</dd>
<dt><strong>nimgs:</strong></dt>
<dd><p class="first last">can be followed by the number of images to display, 1:nimgs</p>
</dd>
<dt><strong>names:</strong></dt>
<dd><p class="first last">is followed by a cell array of names for the images.</p>
</dd>
<dt><strong>color:</strong></dt>
<dd><p class="first last">is followed by color vector or string specification. default is
color-mapped with split colors (hot/cool) for pos and neg effects.</p>
</dd>
<dt><strong>outline:</strong></dt>
<dd><p class="first last">is followed by a color vector for outline around blobs.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>The output, o, is an fmridisplay object.</p>
<p>This function uses fmridisplay objects, and may be memory-intensive for
older computers.</p>
<p><em>Common Errors:</em></p>
<p>This function uses the volInfo.cluster field. If you create a mask in an
ad hoc way, this field may not be updated.  use this to fix:</p>
<blockquote>
<div><ul class="simple">
<li>mask = reparse_contiguous(mask);</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>slices(dat);
slices(dat, &#39;orientation&#39;, &#39;axial&#39;);
slices(dat, &#39;slice&#39;, -5);                 % display sagg at x = -5
o = slices(dat, &#39;names&#39;, terms); % use &#39;terms&#39; var as names

o2 = slices(all_chi2_images, &#39;orientation&#39;, &#39;sagittal&#39;, &#39;slice&#39;, 0);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.surface">
<code class="descclassname">&#64;image_vector.</code><code class="descname">surface</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.surface" title="Permalink to this definition">¶</a></dt>
<dd><p>[all_surf_handles, pcl, ncl] = surface(obj)</p>
<dl class="docutils">
<dt><em>Usage:</em></dt>
<dd><ul class="first last simple">
<li>[all_surf_handles, pcl, ncl] = surface(r, [&#8216;cutaways&#8217;, any optional inputs to surface_cutaway])</li>
</ul>
</dd>
</dl>
<p>This function uses region.surface to create surface figures.
See help region.surface for options.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Create an initial surface plot from an fmri_data object:
han = surface(regionmasks{2});  

% Now add a second region in green:
cluster_surf(region(regionmasks{2}), {[0 1 0]}, han, 5);

% Use optional arguments taken by surface_cutaway:
poscm = colormap_tor([1 .3 0], [1 1 0]); % orange to yellow
[all_surf_handles, pcl, ncl] = surface(t, &#39;cutaway&#39;, &#39;ycut_mm&#39;, -30, &#39;pos_colormap&#39;, poscm, &#39;existingfig&#39;);
[all_surf_handles2, pcl, ncl] = surface(t, &#39;foursurfaces&#39;, &#39;pos_colormap&#39;, poscm, &#39;neg_colormap&#39;, negcm);
[all_surf_handles2, pcl, ncl] = surface(t, &#39;foursurfaces&#39;, &#39;existingfig&#39;, &#39;color_upperboundpercentile&#39;, 95, &#39;color_lowerboundpercentile&#39;, 5, &#39;neg_colormap&#39;, colormap_tor([0 0 1], [.3 0 .5]));

% Use mediation_brain_surface_figs and re-make colors
all_surf_handles = mediation_brain_surface_figs([]);
surface(t2, &#39;cutaway&#39;, &#39;surface_handles&#39;, all_surf_handles, &#39;color_upperboundpercentile&#39;, 95, &#39;color_lowerboundpercentile&#39;, 5, &#39;neg_colormap&#39;, colormap_tor([0 0 1], [.2 0 .5]));
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.threshold">
<code class="descclassname">&#64;image_vector.</code><code class="descname">threshold</code><span class="sig-paren">(</span><em>obj</em>, <em>input_threshold</em>, <em>thresh_type</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.threshold" title="Permalink to this definition">¶</a></dt>
<dd><p>Threshold image_vector (or fmri_data or fmri_obj_image) object based on
raw threshold values. For statistical thresholding, convert to a
statistic_image object and see the threshold method for that object.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>obj = threshold(obj, input_threshold, thresh_type, [optional arguments])
</pre></div>
</div>
<p>This is a method for an image_vector object</p>
<p>Thresholding is not reversible. For statistic_image objects it is.</p>
<dl class="docutils">
<dt>For objects: Type methods(object_name) for a list of special commands</dt>
<dd>Type help object_name.method_name for help on specific
methods.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">image_vector object</p>
</dd>
<dt><strong>input_threshold:</strong></dt>
<dd><p class="first last">Vector of 2 values defining data value bounds at which
to threshold, e.g., [0 Inf] or [-3 3]</p>
</dd>
<dt><strong>thresh_type:</strong></dt>
<dd><p class="first last">String: &#8216;raw-between&#8217; or &#8216;raw-outside&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">Argument or argument followed by value:</p>
<dl class="docutils">
<dt><strong>k:</strong></dt>
<dd><p class="first last">Followed by extent threshold cluster size, default = 1</p>
</dd>
<dt><strong>trim_mask:</strong></dt>
<dd><p class="first last">Reduce the mask in obj.voInfo based on thresholding</p>
</dd>
<dt><strong>noverbose:</strong></dt>
<dd><p class="first last">Suppress verbose output</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">thresholded image_vector object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Retain positive values, cluster extent &gt; 100 voxels
obj = threshold(obj, [0 Inf], &#39;raw-between&#39;, &#39;k&#39;, 100)

% Retain voxels with absolute value &gt; 3
obj = threshold(obj, [-3 3], &#39;raw-outside&#39;)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>statistic_image.threshold, statistic_image.multi_threshold</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.trim_mask">
<code class="descclassname">&#64;image_vector.</code><code class="descname">trim_mask</code><span class="sig-paren">(</span><em>obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.trim_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Exclude empty voxels from mask information in obj.volInfo structure, and re-make obj.volInfo</p>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.union">
<code class="descclassname">&#64;image_vector.</code><code class="descname">union</code><span class="sig-paren">(</span><em>dat1</em>, <em>dat2</em>, <em>outputname</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.union" title="Permalink to this definition">¶</a></dt>
<dd><p>Union and intersection masks for two image_vector objects</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[dat_union, dat_intersection] = union(dat1, dat2, outputname)

 dat = union(dat1, dat2, outputname)
 outputname = character array name for union image
               INCLUDE .img at the end.
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;image_vector.write">
<code class="descclassname">&#64;image_vector.</code><code class="descname">write</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@image_vector.write" title="Permalink to this definition">¶</a></dt>
<dd><p>Write an image_vector object to an Analyze image.
Option to write thresholded image, for statistic_image objects.</p>
<p>obj.dat should contain data, with one COLUMN for each 3-D frame in the
4-D image to be written.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>write(obj)  -&gt; writes to the image(s) specified in obj.fullpath
write(obj, &#39;thresh&#39;) -&gt; for statistic_image objects, writes thresholded
write(obj, &#39;fname&#39;, &#39;~/Documents/test.nii&#39;)  -&gt; writes the image(s) to specific path
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>mni:</strong></dt>
<dd><p class="first last">resample image to standard MNI FOV (91x109x91)
uses mri_data.resample_space</p>
</dd>
<dt><strong>keepdt:</strong></dt>
<dd><p class="first last">output image will be keep original data type (default = float32)</p>
</dd>
<dt><strong>fname:</strong></dt>
<dd><p class="first last">writes out image to specific file name.  &#8216;fname&#8217; must be
followed by image name with path</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% If m is an image_vector object m.X(m.X &lt; .12) = 0; % apply an 
% arbitrary but reasonable custom threshold
orthviews(m);

% write the thresholded image to disk:
anatmeanname = &#39;mean_gray_matter_mask.img&#39;;
m.filename = anatmeanname;
m.fullpath = fullfile(maskdir, anatmeanname);
write(m)
</pre></div>
</div>
</dd></dl>

</div>
<div class="section" id="module-&#64;region">
<span id="region"></span><h2>region<a class="headerlink" href="#module-@region" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="&#64;region.check_extracted_data">
<code class="descclassname">&#64;region.</code><code class="descname">check_extracted_data</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.check_extracted_data" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Checks the data, just in case of space/programming issues, </dt>
<dd>by re-extracting the region average data from 5 random regions 
using spm_get_data.m, and compares it to the already-saved values</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><blockquote class="first">
<div><dl class="docutils">
<dt><strong>cl:</strong></dt>
<dd><p class="first last">must be a valid region object (see region.m)
and cl(1).source_images must still be on the path.</p>
</dd>
</dl>
</div></blockquote>
<p class="last">You should not need to run this regularly &#8211; but you should if you
suspect things have gone awry.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;region.extract_data">
<code class="descclassname">&#64;region.</code><code class="descname">extract_data</code><span class="sig-paren">(</span><em>r</em>, <em>data_obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.extract_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Extract data from image_vector object (data_obj) for voxels specified
by a region object (r). Returns extracted data and averages.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">region_obj</span> <span class="o">=</span> <span class="n">extract_data</span><span class="p">(</span><span class="n">region_obj</span><span class="p">,</span> <span class="n">data_obj</span><span class="p">)</span>
</pre></div>
</div>
<p>Type methods(region) for a list of special commands for region object
Type help object_name.method_name for help on specific methods.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Features:</th><td class="field-body">data_obj does not have to be in the same space, uses mm coordinates</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>r:</strong></dt>
<dd><p class="first last">a region object</p>
</dd>
<dt><strong>data_obj:</strong></dt>
<dd><p class="first last">an image_vector or fmri_data object to extract data from
does not have to be in the same space, uses mm coordinates</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>r:</strong></dt>
<dd><p class="first last">a region object, with data attached</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;region.merge">
<code class="descclassname">&#64;region.</code><code class="descname">merge</code><span class="sig-paren">(</span><em>cl</em>, <em>wh_merge</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.merge" title="Permalink to this definition">¶</a></dt>
<dd><p>Merge two or more regions together in a region object.
Combines fields from all clusters in the named series with the first one
in the series.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>wh_merge = [3 4];
cl = merge(cl, wh_merge)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;region.posneg_separate">
<code class="descclassname">&#64;region.</code><code class="descname">posneg_separate</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.posneg_separate" title="Permalink to this definition">¶</a></dt>
<dd><p>Separate a region object (cl) into clusters with positive and negative
peak values, based on max (peak) value in .val or .Z field (default =
val)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">pcl</span><span class="p">,</span> <span class="n">ncl</span><span class="p">]</span> <span class="o">=</span> <span class="n">posneg_separate</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;Z&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>Returns pcl and ncl, region structures with positive- and negative-valued
peaks, respectively, copied from the original cl input.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Optional Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Z:</strong></dt>
<dd><p class="first last">To use .Z field</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Note:</th><td class="field-body"><p class="first last">You may have to use reparse_continguous to get this to work right.</p>
</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">r</span> <span class="o">=</span> <span class="n">reparse_continguous</span><span class="p">(</span><span class="n">r</span><span class="p">);</span>
<span class="p">[</span><span class="n">pcl</span><span class="p">,</span> <span class="n">ncl</span><span class="p">]</span> <span class="o">=</span> <span class="n">posneg_separate</span><span class="p">(</span><span class="n">r</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;region.region2imagevec">
<code class="descclassname">&#64;region.</code><code class="descname">region2imagevec</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.region2imagevec" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert a region object to an image_vector object, replacing the voxels
and reconstructing as much info as possible.</p>
<p>The .dat field of the new &#8220;ivecobj&#8221; is made from the cl.all_data field.
if this is empty, uses cl.val field, then cl.Z as a backup.
Mask information is available in ivecobj.volInfo.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ivecobj</span> <span class="o">=</span> <span class="n">region2imagevec</span><span class="p">(</span><span class="n">cl</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;region.region2imagevec2tmp">
<code class="descclassname">&#64;region.</code><code class="descname">region2imagevec2tmp</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.region2imagevec2tmp" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert a region object to an image_vector object, replacing the voxels
and reconstructing as much info as possible.</p>
<p>The .dat field of the new &#8220;ivecobj&#8221; is made from the cl.all_data field.
if this is empty, uses cl.val field, then cl.Z as a backup.
Mask information is available in ivecobj.volInfo.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ivecobj</span> <span class="o">=</span> <span class="n">region2imagevec</span><span class="p">(</span><span class="n">cl</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;region.region2struct">
<code class="descclassname">&#64;region.</code><code class="descname">region2struct</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.region2struct" title="Permalink to this definition">¶</a></dt>
<dd><p>Convert a region object to a simple structure, primarily for
compatibility with other, older CANlab tools.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body">cluster2region, for the reverse transformation</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;region.reparse_continguous">
<code class="descclassname">&#64;region.</code><code class="descname">reparse_continguous</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.reparse_continguous" title="Permalink to this definition">¶</a></dt>
<dd><p>Re-define regions in region object based on contiguous blobs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">clout</span> <span class="o">=</span> <span class="n">reparse_continguous</span><span class="p">(</span><span class="n">cl</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;region.subdivide_by_atlas">
<code class="descclassname">&#64;region.</code><code class="descname">subdivide_by_atlas</code><span class="sig-paren">(</span><em>r</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.subdivide_by_atlas" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>r = subdivide_by_atlas(r, [atlas name])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>r:</strong></dt>
<dd><p class="first last">a region object, defined using region(mask)</p>
</dd>
<dt><strong>atlas name:</strong></dt>
<dd><p class="first last">Optional mask image with integer codes defining in-mask
regions.  Default is &#8216;atlas_labels_combined.img&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><p class="first">A region object with separate clusters for each contiguous blob,
subdivided by regions labeled in atlas.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>r = subdivide_by_atlas(r);
r(cat(1, r.numVox) &lt; 20) = []; % get rid of small regions
cluster_orthviews(r, &#39;unique&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;region.subdivide_by_local_max">
<code class="descclassname">&#64;region.</code><code class="descname">subdivide_by_local_max</code><span class="sig-paren">(</span><em>r</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.subdivide_by_local_max" title="Permalink to this definition">¶</a></dt>
<dd><p>Subdivide regions into sub-regions based on local peak Z-scores/maxima</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">subregions</span> <span class="o">=</span> <span class="n">subdivide_by_local_max</span><span class="p">(</span><span class="n">r</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;mm_distance&#39;</span><span class="p">,</span> <span class="n">value</span><span class="p">],</span> <span class="p">[</span><span class="s">&#39;noorthviews&#39;</span><span class="p">])</span>
</pre></div>
</div>
<dl class="docutils">
<dt>For objects: Type methods(object_name) for a list of special commands</dt>
<dd>Type help object_name.method_name for help on specific
methods.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>mm_distance:</strong></dt>
<dd><p class="first last">Followed by mm distance minimum for dividing subclusters</p>
</dd>
<dt><strong>noorthviews:</strong></dt>
<dd><p class="first last">Suppress display of orthviews</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>subregions:</strong></dt>
<dd><p class="first last">subdivided region object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>region.subdivide_by_atlas, subclusters_from_local_max, cluster_local_maxima</p>
</dd></dl>

<dl class="function">
<dt id="&#64;region.surface">
<code class="descclassname">&#64;region.</code><code class="descname">surface</code><span class="sig-paren">(</span><em>r</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.surface" title="Permalink to this definition">¶</a></dt>
<dd><p>Surface method for region object - renders blobs on multiple types of 3-D surface</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[all_surf_handles, pcl, ncl] = surface(r, [&#39;cutaways&#39;, any optional inputs to surface_cutaway])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>r:</strong></dt>
<dd><p class="first last">A region object</p>
</dd>
<dt><strong>cutaway:</strong></dt>
<dd><dl class="first last docutils">
<dt>String command for rendering cutaways instead of the default</dt>
<dd><ul class="first last simple">
<li>default is call to mediation_brain_surface_figs</li>
<li>cutaways calls surface_cutaway</li>
<li>all optional arguments are passed to surface_cutaway</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>rightsurface:</strong></dt>
<dd><dl class="first last docutils">
<dt>String command for rendering a right frontal cortical</dt>
<dd><p class="first last">view complementary to &#8216;cutaways&#8217;</p>
</dd>
</dl>
</dd>
<dt><strong>foursurfaces:</strong></dt>
<dd><p class="first last">Compact plots of four surfaces</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Other optional inputs to surface_cutaway
e.g., &#8216;pos_colormap&#8217;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>all_surf_handles:</strong></dt>
<dd><p class="first last">surface patch handles</p>
</dd>
<dt><strong>pcl:</strong></dt>
<dd><p class="first last">region object with positive-only clusters</p>
</dd>
<dt><strong>ncl:</strong></dt>
<dd><p class="first last">region object with negative-only clusters</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Use surface(r), with optional arguments taken by surface_cutaway:
poscm = colormap_tor([1 .3 0], [1 1 0]); % orange to yellow
[all_surf_handles, pcl, ncl] = surface(r, &#39;cutaway&#39;, &#39;ycut_mm&#39;, -30, &#39;pos_colormap&#39;, poscm, &#39;existingfig&#39;);
[all_surf_handles2, pcl, ncl] = surface(r, &#39;foursurfaces&#39;, &#39;pos_colormap&#39;, poscm, &#39;neg_colormap&#39;, negcm);
[all_surf_handles2, pcl, ncl] = surface(r, &#39;foursurfaces&#39;, &#39;existingfig&#39;, &#39;color_upperboundpercentile&#39;, 95, &#39;color_lowerboundpercentile&#39;, 5, &#39;neg_colormap&#39;, colormap_tor([0 0 1], [.4 0 .7]));

% use mediation_brain_surface_figs and re-make colors
all_surf_handles = mediation_brain_surface_figs([]);
surface(r, &#39;cutaway&#39;, &#39;surface_handles&#39;, all_surf_handles, &#39;color_upperboundpercentile&#39;, 95, &#39;color_lowerboundpercentile&#39;, 5, &#39;neg_colormap&#39;, colormap_tor([0 0 1], [.2 0 .5]));
</pre></div>
</div>
<p>:See also:*</p>
<p>surface_cutaway, cluster_surf, mediation_brain_surface_figs
..</p>
<blockquote>
<div>DEFAULTS AND INPUTS</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="&#64;region.table">
<code class="descclassname">&#64;region.</code><code class="descname">table</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@region.table" title="Permalink to this definition">¶</a></dt>
<dd><p>Print a table of all regions in a region object (cl)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[poscl, negcl] = table(cl, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>k:</strong></dt>
<dd><p class="first last">Print only regions with k or more contiguous voxels</p>
</dd>
<dt><strong>nosep:</strong></dt>
<dd><p class="first last">do not separate cl with pos and neg effects based on peak in .val</p>
</dd>
<dt><strong>names:</strong></dt>
<dd><p class="first last">name clusters before printing to table and output; saves in .shorttitle field</p>
</dd>
<dt><strong>forcenames:</strong></dt>
<dd><p class="first last">force naming of cl by removing existing names in .shorttitle field</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first last">Returns region objects for cl with pos and neg effects, limited by size if entered
and named if entered as optional input</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-&#64;statistic_image">
<span id="statistic-image"></span><h2>statistic_image<a class="headerlink" href="#module-@statistic_image" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="&#64;statistic_image.conjunction">
<code class="descclassname">&#64;statistic_image.</code><code class="descname">conjunction</code><span class="sig-paren">(</span><em>si1</em>, <em>si2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@statistic_image.conjunction" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the conjunction of two statistic_images.  considers positive and
negative activations separately.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body">Two thresholded statistic images.  Optional 3rd argument:  -1 to</td>
</tr>
</tbody>
</table>
<p>get only negative conjunction, or 1 to get only positive conjunction</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body">A statistic_image with all voxels suprathreshold (in the same direction) in both input</td>
</tr>
</tbody>
</table>
<p>images.  Voxel values are set to 1 and -1, to indicate direction.</p>
</dd></dl>

<dl class="function">
<dt id="&#64;statistic_image.convert2mask">
<code class="descclassname">&#64;statistic_image.</code><code class="descname">convert2mask</code><span class="sig-paren">(</span><em>stats_image_obj</em><span class="sig-paren">)</span><a class="headerlink" href="#@statistic_image.convert2mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts each image in a statistic_image object into a mask object, based
on significant voxels in the .sig field.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">region</span><span class="p">(</span><span class="n">convert2mask</span><span class="p">(</span><span class="n">timg</span><span class="p">),</span> <span class="n">group</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;statistic_image.multi_threshold">
<code class="descclassname">&#64;statistic_image.</code><code class="descname">multi_threshold</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@statistic_image.multi_threshold" title="Permalink to this definition">¶</a></dt>
<dd><p>Multiple threshold function for statistic_image object for visualization</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[o2, sig, pcl, ncl] = multi_threshold(dat, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">a statistic_image object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>poscolors:</strong></dt>
<dd><p class="first last">followed by cell array of colors for positive values, one per thresh</p>
</dd>
<dt><strong>negcolors:</strong></dt>
<dd><p class="first last">followed by cell array of colors for negative values, one per thresh</p>
</dd>
<dt><strong>thresh:</strong></dt>
<dd><p class="first last">followed vector of p-value thresholds, one per thresh</p>
</dd>
<dt><strong>sizethresh:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by vector of cluster sizes, one per thresh</dt>
<dd><ul class="first last simple">
<li>this &#8216;prunes&#8217; by default, so sizes after first can be 1
voxel</li>
<li>Default thresholds: thresh = [.001 .005 .05],
10 voxels at .001, &#8220;pruned&#8221;</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>nodisplay:</strong></dt>
<dd><p class="first last">suppress fmridisplay</p>
</dd>
<dt><strong>o2:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by an existing fmridisplay object</dt>
<dd><ul class="first last simple">
<li>will remove blobs and re-use montages</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>o2:</strong></dt>
<dd><p class="first last">handle to fmridisplay object created by default</p>
</dd>
<dt><strong>sig:</strong></dt>
<dd><dl class="first last docutils">
<dt>vector of significant voxels at each thresh, for each region</dt>
<dd><ul class="first simple">
<li>cell array of images in object with matrix of values</li>
</ul>
<p class="last">for each threshold</p>
</dd>
</dl>
</dd>
<dt><strong>pcl:</strong></dt>
<dd><dl class="first last docutils">
<dt>positive valued clusters cell, one cell per threshold</dt>
<dd><ul class="first last simple">
<li>FIRST image in object only</li>
<li>pass into mediation_brain_surface_figs.m</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>ncl:</strong></dt>
<dd><dl class="first last docutils">
<dt>positive valued clusters cell, one cell per threshold</dt>
<dd><ul class="first last simple">
<li>FIRST image in object only</li>
<li>pass into mediation_brain_surface_figs.m</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[o2, sig, poscl, negcl] = multi_threshold(hr_intercept, &#39;nodisplay&#39;);
mediation_brain_surface_figs(poscl, negcl);

% Create empty montage set and (re)use it:
o2 = canlab_results_fmridisplay([], &#39;compact2&#39;, &#39;noverbose&#39;);
o2 = multi_threshold(out.t, &#39;o2&#39;, o2);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>mediation_brain_surface_figs, iimg_multi_threshold, mediation_brain_results</p>
</dd></dl>

<dl class="function">
<dt id="&#64;statistic_image.orthviews">
<code class="descclassname">&#64;statistic_image.</code><code class="descname">orthviews</code><span class="sig-paren">(</span><em>image_obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@statistic_image.orthviews" title="Permalink to this definition">¶</a></dt>
<dd><p>Orthviews display (SPM) for CANlab object</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cl = orthviews(image_object)

% OR

cl = orthviews(image_object, handle_number of existing orthviews)
</pre></div>
</div>
<p>Output is clusters structure (see also region.m)</p>
<p>Pass in &#8216;largest_region&#8217; to center the orthviews on the largest region in the  image</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% T-test, Construct a stats_image object, threshold and display:
statsimg = ttest(fmridat, .001, &#39;unc&#39;);

% Re-threshold and display:
statsimg = threshold(statsimg, .000001, &#39;unc&#39;);
orthviews(statsimg);

statsimg = threshold(statsimg, .01, &#39;fdr&#39;);
orthviews(statsimg);

% Create an orthviews and view at multiple thresholds in different panes:
overlay = which(&#39;SPM8_colin27T1_seg.img&#39;);
spm_check_registration(repmat(overlay, n, 1));
statsimg = ttest(fmridat);
statsimg = threshold(statsimg, .001, &#39;unc&#39;);
orthviews(statsimg, &#39;handle&#39;, 1);

statsimg = threshold(statsimg, .000001, &#39;unc&#39;);
orthviews(statsimg, &#39;handle&#39;, 2);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body">statistic_image.multi_threshold</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="&#64;statistic_image.reparse_contiguous">
<code class="descclassname">&#64;statistic_image.</code><code class="descname">reparse_contiguous</code><span class="sig-paren">(</span><em>obj</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@statistic_image.reparse_contiguous" title="Permalink to this definition">¶</a></dt>
<dd><p>Re-construct list of contiguous voxels in an image based on in-image
voxel coordinates.  Coordinates are taken from obj.volInfo.xyzlist.
Results are saved in obj.volInfo.cluster.
xyzlist can be generated from iimg_read_img, and is done automatically by
object-oriented fMRI image classes (fmri_image, image_vector,
statistic_image)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">obj</span> <span class="o">=</span> <span class="n">reparse_contiguous</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;nonempty&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>If &#8216;nonempty&#8217; is entered as an optional argument, will use only voxels
that are non-zero, non-nan in the first column of obj.dat.</p>
<p>The statistic_image object version of reparse_contiguous uses 
the significance of the first image in the object (obj.sig(:, 1)) as a
filter as well, so clustering will be based on the latest threshold applied.
it is not usually necessary to enter &#8216;nonempty&#8217;.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Given timg, a statistic_image object:
test = reparse_contiguous(timg, &#39;nonempty&#39;);
cl = region(test, &#39;contiguous_regions&#39;);
cluster_orthviews(cl, &#39;unique&#39;)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="&#64;statistic_image.select_one_image">
<code class="descclassname">&#64;statistic_image.</code><code class="descname">select_one_image</code><span class="sig-paren">(</span><em>obj</em>, <em>wh</em><span class="sig-paren">)</span><a class="headerlink" href="#@statistic_image.select_one_image" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="&#64;statistic_image.threshold">
<code class="descclassname">&#64;statistic_image.</code><code class="descname">threshold</code><span class="sig-paren">(</span><em>stats_image_obj</em>, <em>input_threshold</em>, <em>thresh_type</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#@statistic_image.threshold" title="Permalink to this definition">¶</a></dt>
<dd><p>Threshold statistic_image object based on statistical threshold values.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>stats_image_obj = threshold(stats_image_obj, pvalthreshold or other thresh, thresh_type, [&#39;k&#39;, extent_thresh])
</pre></div>
</div>
<p>This is a method for an statistic_image object.
Thresholding is reversible.</p>
<dl class="docutils">
<dt>For objects: Type methods(object_name) for a list of special commands</dt>
<dd>Type help object_name.method_name for help on specific
methods.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>stats_image_obj:</strong></dt>
<dd><p class="first last">statistic_image object</p>
</dd>
<dt><strong>input_threshold:</strong></dt>
<dd><p class="first last">[pvalthreshold or other thresh]
A numeric value corresponding to the threshold desired.
Either a p-value or a range of raw values, depending on the threshold
type.</p>
</dd>
<dt><strong>thresh_type:</strong></dt>
<dd><dl class="first last docutils">
<dt>Threshold type which can be one of:</dt>
<dd><ul class="first last simple">
<li>&#8216;fdr&#8217; : FDR-correct based on p-values already stored in image .p field</li>
<li>&#8216;fwe&#8217; : FWE-correct; not implemented</li>
<li>&#8216;bfr&#8217; : Bonferroni correction (FWE).</li>
<li>&#8216;unc&#8217; : Uncorrected p-value threshold: p-value, e.g., .05 or .001</li>
<li>&#8216;raw-between&#8217; : threshold raw image values; save those &gt; input_threshold(1) and &lt; input_threshold(2)</li>
<li>&#8216;raw-outside&#8217; : threshold raw image values; save those &lt; input_threshold(1) or &gt; input_threshold(2)</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>k:</strong></dt>
<dd><p class="first last">Followed by cluster extent in voxels: extent-based thresholding of any of the above</p>
</dd>
<dt><strong>noverbose:</strong></dt>
<dd><p class="first last">Suppress verbose output</p>
</dd>
<dt><strong>mask:</strong></dt>
<dd><dl class="first last docutils">
<dt>Followed by name of mask or fmri_mask_image object</dt>
<dd><ul class="first last simple">
<li>this will affect corrected significance levels</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>stats_image_obj:</strong></dt>
<dd><p class="first last">thresholded statistic_image object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Retain sig pos or neg results at p &lt; .001 uncorrected, cluster extent &gt;= 100 voxels
obj = threshold(obj, .001, &#39;unc&#39;, &#39;k&#39;, 100)

% Retain sig pos or neg results at q &lt; .05 FDR, cluster extent &gt;= 10 voxels
obj = threshold(obj, .05, &#39;fdr&#39;, &#39;k&#39;, 10)

% Retain voxels with absolute statistic/data value &gt; 3
obj = threshold(obj, [-3 3], &#39;raw-outside&#39;)

dat = threshold(dat, 0.001, &#39;unc&#39;, &#39;k&#39;, 35, &#39;mask&#39;, which(&#39;scalped_avg152T1_graymatter_smoothed.img&#39;));
dat = threshold(dat, 0.001, &#39;unc&#39;, &#39;k&#39;, 35, &#39;mask&#39;, maskobj);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>image_vector.threshold, statistic_image.multi_threshold</p>
</dd></dl>

</div>
</div>
<div class="section" id="toolboxes">
<h1>Toolboxes<a class="headerlink" href="#toolboxes" title="Permalink to this headline">¶</a></h1>
<div class="section" id="cluster-contig-region-tools">
<h2>Cluster_contig_region_tools<a class="headerlink" href="#cluster-contig-region-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Cluster_contig_region_tools"></span><dl class="function">
<dt id="Cluster_contig_region_tools.anat_subclusters">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">anat_subclusters</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.anat_subclusters" title="Permalink to this definition">¶</a></dt>
<dd><p>Clusters voxels within &#8216;clusters&#8217; structure based on anatomical locations
in space.  Outputs subgroups of smaller clusters.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>clout = anat_subclusters(cl,[resume at],[output cl to resume])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster2region">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster2region</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster2region" title="Permalink to this definition">¶</a></dt>
<dd><p>Transform a CANlab/SCANlab &#8220;clusters&#8221; structure into a region object, the
standard in 2011 toolbox functions and beyond.</p>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster2subclusters">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster2subclusters</code><span class="sig-paren">(</span><em>cl_in</em>, <em>class</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster2subclusters" title="Permalink to this definition">¶</a></dt>
<dd><p>Take a single cluster cl_in and separate into subclusters based on
vector of integers class</p>
<p>Class must code unique subclusters subcluster order is only preserved
if class contains all integers from 1 to nclasses:</p>
<p>i.e., class 3 will only be in subcluster 3 if there are no missing class
numbers in class</p>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_close_enough">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_close_enough</code><span class="sig-paren">(</span><em>cl_match_to</em>, <em>cl_match</em>, <em>mind</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_close_enough" title="Permalink to this definition">¶</a></dt>
<dd><p>Finds whether each cluster center in cl_match is within mind mm of a cluster
center in cl_match_to.</p>
<p>Useful for selecting a list of clusters that are not close to another
list to, e.g., make a table of.  or this could be used to find clusters
in a set of correlated clusters that are close to centers in activated
clsuters.</p>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_export_pngs">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_export_pngs</code><span class="sig-paren">(</span><em>cl</em>, <em>useexisting</em>, <em>overlay</em>, <em>xhairson</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_export_pngs" title="Permalink to this definition">¶</a></dt>
<dd><p>Save png images of SPM orthviews windows for each cluster in a set (cl
structure)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">cluster_export_pngs</span><span class="p">(</span><span class="n">cl</span><span class="p">,[</span><span class="n">useexisting</span><span class="p">],[</span><span class="n">overlayimagename</span><span class="p">],[</span><span class="n">xhairson</span><span class="p">])</span>
</pre></div>
</div>
<p>names from cl(x).shorttitle are used
useexisting is optional: 1 uses existing orthviews display (default), 0 creates a
new one with the clusters</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% use existing
cluster_export_pngs(cl, 1, EXPT.overlay);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_find_index">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_find_index</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_find_index" title="Permalink to this definition">¶</a></dt>
<dd><p>Ever see an interesting blob when visualizing a clusters structure, but
don&#8217;t know which index number in the clusters structure vector it
corresponds to?</p>
<p>With this function, find the index number of the closest cluster to one you specify
graphically by clicking on.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [wh_cluster, min_distance] = cluster_find_index(cl, [keep display flag, 1/0])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_interp">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_interp</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_interp" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Interpolates voxels and mm in a clusters structure (cl) to match the</dt>
<dd><p class="first">image dimensions and voxel sizes of a target mask image.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function cl = cluster_interp(cl,maskimg,[keep sep clusters flag])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="last highlight-python"><div class="highlight"><pre>cl = cluster_interp(cl,maskimg,1);

%default mask 2 x 2 x 2, SPM2 default:
cl = cluster_interp(cl,[],1);
</pre></div>
</div>
</dd>
</dl>
<p>maskimg = which(&#8216;scalped_avg152T1_graymatter_smoothed.img&#8217;);</p>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_intersection">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_intersection</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_intersection" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the intersection of the clusters passed in.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>intersect_cl = cluster_intersection(cl1, cl2, cl3, ...)

% simple intersection
cl = cluster_intersection(robust0001_poscl(2), robust0002_poscl(4), robust0003_poscl(17));

% intersection between sets of clusters
% alternatively, see cluster_set_intersection.m
cl = cluster_intersection(clusters2CLU(robust0001_poscl), clusters2CLU(robust0002_poscl));
</pre></div>
</div>
<dl class="docutils">
<dt>Note:</dt>
<dd>Only works with single clusters. To compute the intersection between sets of clusters,
use cluster_set_intersection()</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_local_maxima">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_local_maxima</code><span class="sig-paren">(</span><em>cl</em>, <em>dthresh</em>, <em>verbose</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_local_maxima" title="Permalink to this definition">¶</a></dt>
<dd><p>Clusters are chosen so that they must be at least dthresh mm apart
default is 10 mm</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[xyz, XYZmm, Z, class] = cluster_local_maxima(cl, [dthresh], [verbose])
</pre></div>
</div>
<p>verbose output: 1/0, default is 0</p>
<p>additional optional outputs (slower):</p>
<p>class: vector of integers for which subcluster this cluster belongs to</p>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_set_intersection">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_set_intersection</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_set_intersection" title="Permalink to this definition">¶</a></dt>
<dd><p>Computes the intersection of the sets of clusters passed in.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">intersect_cl</span> <span class="o">=</span> <span class="n">cluster_set_intersection</span><span class="p">(</span><span class="n">cls1</span><span class="p">,</span> <span class="n">cls2</span><span class="p">,</span> <span class="n">cls3</span><span class="p">,</span> <span class="o">...</span><span class="p">)</span>
<span class="n">cl</span> <span class="o">=</span> <span class="n">cluster_intersection</span><span class="p">(</span><span class="n">robust0001_poscls</span><span class="p">,</span> <span class="n">robust0002_poscls</span><span class="p">,</span> <span class="n">robust0003_poscls</span><span class="p">);</span>
</pre></div>
</div>
<dl class="docutils">
<dt>Note:</dt>
<dd>Designed for sets of clusters. To compute the intersection between individual clusters,
use cluster_intersection(). cluster_set_intersection will work, but is not needed.</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_table">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_table</code><span class="sig-paren">(</span><em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_table" title="Permalink to this definition">¶</a></dt>
<dd><p>Print output of clusters in table</p>
<p>Option to print text labels from Carmack atlas.</p>
<p>Database loading is done from talairach_info.mat which should be in the 
path.</p>
<p>To speed up performance, load talairach_info.mat in the base workspace or
calling function and include xyz, L3 and L5 as inputs to cluster_table.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% create subclusters on the fly, prompt for labels
cluster_table(cl);

% no subclusters, no labels
cluster_table(cl, 0, 0);

% do subclusters, no labels
cluster_table(cl, 1, 0);

create subclusters on the fly, do labels
cluster_table(cl, 1, 1);

% 3 input variables following &#39;tal_info&#39; are interpreted as xyz, L3,
% and L5 from talairach_info.mat.
cluster_table(..., &#39;tal_info&#39;, xyz, L3, L5);

% loads labels from taldata.mat (Talairach database) instead of
% talairach_info.mat. Note that you should use the &#39;tal_info&#39; call
% above if xyz, L3, and L5 have already been loaded to theworkspace
% from taldata.mat. Also, if the talairach database is being used,
% your cl.XYZmm values MUST correspond to the TALAIRACH, NOT MNI,
% database, or the labels will be innaccurate.
cluster_table(..., &#39;talairach&#39;);

% print table to ASCII file, &#39;filename&#39;, instead of to the matlab
% command window.
cluster_table(..., &#39;writefile&#39;,&#39;filename&#39;);

% any set of inputs from above, also print clusters.myfield in output
cluster_table(..., &#39;myfield&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.cluster_table_successive_threshold">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">cluster_table_successive_threshold</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.cluster_table_successive_threshold" title="Permalink to this definition">¶</a></dt>
<dd><p>Cluster table of a cell array of clusters cl{1} cl{2} etc.
Prints table of cl{1} and then any additional regions in cl{2:n} that are
not within 10 mm of a previously printed cluster</p>
<p>Also: merges clusters in set within 10 mm</p>
<p>Table titles are hard-coded to be consistent with meta-analysis toolbox right now</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">cluster_table_successive_threshold</span><span class="p">(</span><span class="n">cl</span><span class="p">,[</span><span class="n">sizethr</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Print a series of tables with custom fields:
cl = cluster_table_successive_threshold(cl,5,&#39;myfield1&#39;,&#39;myfield2&#39;)
</pre></div>
</div>
<p>Use <em>merge_nearby_clusters</em> and <em>subclusters_from_local_max</em>
or some other way to get clusters appropriately separated and distanced
before running.
see Meta_cluster_tools for code to run this for meta-analysis.</p>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.clusters2CLU">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">clusters2CLU</code><span class="sig-paren">(</span><em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.clusters2CLU" title="Permalink to this definition">¶</a></dt>
<dd><p>Inputting an M matrix will transform the coordinates
by that M, to convert between voxel sizes, etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function CLU = clusters2CLU(clusters,[opt] M)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.clusters2mask">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">clusters2mask</code><span class="sig-paren">(</span><em>cl</em>, <em>V</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.clusters2mask" title="Permalink to this definition">¶</a></dt>
<dd><p>This function has 2 modes!  If V is a structure:</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[m,V,cl] = clusters2mask(cl,V,[opt: write Z-scores])
</pre></div>
</div>
<p>Converts clusters structure to a mask image, given V structure with V.mat
field.  V.mat is an SPM mat file. V.dim is dims of image uses cl.XYZmm
m is mask img data, V is mask vol info.</p>
<p>Also replaces cl.XYZ (voxels)</p>
<p>If V is a vector of mask dimensions:
converts clusters to mask image using existing XYZ and dims of mask</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body">voxels2mask, for a faster function that uses XYZ voxel coords</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Save an image file with just one cluster from a set (#7 in this ex.)
cl = mask2clusters(&#39;roi_group1.img&#39;);
V = spm_vol(&#39;roi_group1.img&#39;);  % we need .mat and .dim from this, or

% just dim
[m,V,cl] = clusters2mask(cl(7),struct(&#39;mat&#39;,cl(1).M,&#39;dim&#39;,V.dim));
%or
[m,V,cl] = clusters2mask(cl(7),V);

clusters2mask(cl,struct(&#39;mat&#39;,V.mat,&#39;dim&#39;,V.dim),0,&#39;spm2_hy.img&#39;);

%for SPM5:
clusters2mask(cl,struct(&#39;mat&#39;,V.mat,&#39;dim&#39;,V.dim, &#39;dt&#39;, V.dt),0,&#39;spm2_hy.img&#39;);
clusters2mask(cl,
struct(&#39;mat&#39;,MC_Setup.volInfo.mat,&#39;dim&#39;,MC_Setup.volInfo.dim, &#39;dt&#39;, MC_Setup.volInfo.dt),0,&#39;acc_roi_mask.img&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.image2clusters">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">image2clusters</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.image2clusters" title="Permalink to this definition">¶</a></dt>
<dd><p>Menu-driven function for getting clusters from an image file (e.g., a
t-image)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cl = image2clusters([overlay image name])
</pre></div>
</div>
<p>Can also return clusters active in two contrasts, sorted by increases in
both, decreases in both, inc in first, dec in first
useful for testing whether something is both activated and correlated!
e.g., see active_plus_corr_scatterplot_plugin</p>
<p>% :Example:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">pospos</span><span class="p">,</span><span class="n">negneg</span><span class="p">,</span><span class="n">posneg</span><span class="p">,</span><span class="n">negpos</span><span class="p">]</span> <span class="o">=</span> <span class="n">image2clusters</span><span class="p">(</span><span class="n">overlay</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.mask2clusters">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">mask2clusters</code><span class="sig-paren">(</span><em>P</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.mask2clusters" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts clusters and con img data from mask</p>
<p>Use with <em>mask_intersection.m</em></p>
<p>To get clusters but not extract data, enter only one argument.</p>
<p>To get clusters and choose extraction imgs with the GUI, enter an empty [] 2nd argument.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[clusters,CLU,subclusters] = mask2clusters(img mask file with voxels,[imgs to extract data from],[df])
</pre></div>
</div>
<p>DOES <em>NOT</em> CONVERT BETWEEN DIFFERENT VOXEL SIZES AND POSITIONS BETWEEN IMNAMES AND SPM/VOL STRUCTS</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body">roi_probe</td>
</tr>
</tbody>
</table>
<p>If no imgs are entered, Z-scores are values from mask</p>
<p>If df is entered, values in mask img are converted to Z-scores with spm_t2z.m</p>
<p>If extract img names are empty and df is entered, assume we&#8217;re using
values from mask as t-values and convert to Z-scores</p>
<p>WARNING: for spm2 compatibility, ABSOLUTE VALUES of voxel sizes are
returned; e.g., ignores analyze flipping in SPM2.</p>
<p>% Matlab 6.5/OSX bug gives seg fault or something if mask is too big.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cl = mask2clusters(&#39;myimage.img&#39;,[img string mtx],[]); % no z-score
conversion, extracts data from [img string mtx]

cl = mask2clusters(&#39;rob_tmap_0002_filt_t_3-05_k10_neg.img&#39;)

% This one works with already-loaded image data and a mat matrix:
V = spm_vol(&#39;rob_tmap_0002_filt_t_3-05_k10_neg.img&#39;); dat = spm_read_vols(V);
cl = mask2clusters(dat,V.mat);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.mask2struct">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">mask2struct</code><span class="sig-paren">(</span><em>maskname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.mask2struct" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function V = mask2struct(maskname,crit_t,cl_size)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>maskname:</strong></dt>
<dd><p class="first last">name of spmT, con, or filtered image without .img extension,
in single quotes</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>crit_t, cl_size::</strong></dt>
<dd><p class="first last">critical t and cluster size at which to mask</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">structure compatible with SPM viewing and with cluster definition
algorithm tor_extract_rois</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% to extract clusters:
[clusters] = tor_extract_rois(maskname,V,V);

% to display:
spm_image   (and choose anatomical)
spm_orthviews(&#39;AddBlobs&#39;,1,V.XYZ,V.Z,V.mat)
spm_orthviews(&#39;AddColouredBlobs&#39;,1,V.XYZ,V.Z,V.mat,[0 0 1])

% to overlay on Talairach atlas
fixed_TSU(clusters)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.merge_clusters">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">merge_clusters</code><span class="sig-paren">(</span><em>c2m</em>, <em>subcl</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.merge_clusters" title="Permalink to this definition">¶</a></dt>
<dd><p>Function for synchronizing the field list of cluster structures
and merging them</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">subclusters</span> <span class="o">=</span> <span class="n">merge_clusters</span><span class="p">(</span><span class="n">clusters_to_match</span><span class="p">,</span> <span class="n">subclusters_to_change</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.merge_nearby_clusters">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">merge_nearby_clusters</code><span class="sig-paren">(</span><em>cl</em>, <em>thr</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.merge_nearby_clusters" title="Permalink to this definition">¶</a></dt>
<dd><p>Merge sets of clusters whose centers are all within thr mm of each other
uses parcel_complete_sets.m</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">newcl</span> <span class="o">=</span> <span class="n">merge_nearby_clusters</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span> <span class="n">thr</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% The command below runs the function recursively until all clusters are
% greater than thr mm apart
newcl = merge_nearby_clusters(cl, thr, &#39;recursive&#39;)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.subclusters_from_local_max">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">subclusters_from_local_max</code><span class="sig-paren">(</span><em>cl</em>, <em>dist_thresh</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.subclusters_from_local_max" title="Permalink to this definition">¶</a></dt>
<dd><p>Breaks apart a cluster into smaller clusters</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">subcl</span> <span class="o">=</span> <span class="n">subclusters_from_local_max</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span> <span class="n">dist_thresh</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Cluster_contig_region_tools.xyz2clusters">
<code class="descclassname">Cluster_contig_region_tools.</code><code class="descname">xyz2clusters</code><span class="sig-paren">(</span><em>xyz</em>, <em>P</em><span class="sig-paren">)</span><a class="headerlink" href="#Cluster_contig_region_tools.xyz2clusters" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts a 3-column x, y, z list of mm coordinates to a clusters
structure given P, the filename of an analyze .img file to provide
dimensions and voxel sizes.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function cl = xyz2clusters(xyz,P)
</pre></div>
</div>
<dl class="docutils">
<dt>Uses this info from the image:</dt>
<dd><ul class="first last simple">
<li>VOL.M     - spm-style mat matrix</li>
<li>VOL.VOX   - voxel sizes</li>
<li>SPM.Z     - now 1s; could stores values in the original image in clusters.Z</li>
</ul>
</dd>
<dt>The following is created internally:</dt>
<dd><ul class="first last simple">
<li>SPM.XYZmm - mm coords, you input these</li>
<li>SPM.XYZ   - voxel coords</li>
</ul>
</dd>
</dl>
</dd></dl>

</div>
<div class="section" id="module-diagnostics">
<span id="diagnostics"></span><h2>diagnostics<a class="headerlink" href="#module-diagnostics" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="diagnostics.BiasPowerloss">
<code class="descclassname">diagnostics.</code><code class="descname">BiasPowerloss</code><span class="sig-paren">(</span><em>tc</em>, <em>X</em>, <em>c</em>, <em>beta</em>, <em>df</em>, <em>z</em>, <em>pval</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.BiasPowerloss" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the approximate bias and power loss due to mis-modeling
This works with the Mismodeling Toolbox described by Loh et al. 2008</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[b bias pl Pc Pe] = BiasPowerloss(tc, X, c, beta, df, z, pval)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>tc:</strong></dt>
<dd><p class="first last">fMRI time course</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">design matrix for multiple regression</p>
</dd>
<dt><strong>c:</strong></dt>
<dd><p class="first last">contrast of interest</p>
</dd>
<dt><strong>beta:</strong></dt>
<dd><p class="first last">(mismodeled) beta value</p>
</dd>
<dt><strong>df:</strong></dt>
<dd><p class="first last">degrees of freedom</p>
</dd>
<dt><strong>z:</strong></dt>
<dd><p class="first last">p-value calculated from ResidScan</p>
</dd>
<dt><strong>pval:</strong></dt>
<dd><p class="first last">cut-off p-value</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>b:</strong></dt>
<dd><p class="first last">updated (correct) beta value</p>
</dd>
<dt><strong>bias:</strong></dt>
<dd><p class="first last">bias</p>
</dd>
<dt><strong>pl:</strong></dt>
<dd><p class="first last">power loss</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">References:</th><td class="field-body"><p class="first last">Loh, J. M., Lindquist, M. A., Wager, T. D. (2008). Residual Analysis for Detecting Mis-modeling in fMRI. Statistica Sinica, 18, 1421-1448.</p>
</td>
</tr>
</tbody>
</table>
<p>Update design matrix using correct model</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.ResidScan">
<code class="descclassname">diagnostics.</code><code class="descname">ResidScan</code><span class="sig-paren">(</span><em>res</em>, <em>FWHM</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.ResidScan" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates P(M&gt;=t) where M is the max value of the smoothed residuals.
In this implementation the residuals are smoothed using a Gaussian
kernel.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [z sres] = ResidScan(res, FWHM)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>res:</strong></dt>
<dd><p class="first last">residual time course</p>
</dd>
<dt><strong>FWHM:</strong></dt>
<dd><p class="first last">Full Width Half Maximum (in time units)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>z:</strong></dt>
<dd><p class="first last">pvalues</p>
</dd>
<dt><strong>sres:</strong></dt>
<dd><p class="first last">smoothed residuals</p>
</dd>
<dt><strong>sres_ns:</strong></dt>
<dd><p class="first last">smoothed residuals (non standardized)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="diagnostics.add_nuisance_to_SPMcfg">
<code class="descclassname">diagnostics.</code><code class="descname">add_nuisance_to_SPMcfg</code><span class="sig-paren">(</span><em>Xn</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.add_nuisance_to_SPMcfg" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Adds a matrix Xn to the end of covariates of interest in</dt>
<dd><p class="first">xX structure in SPMcfg.mat</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function add_nuisance_to_SPMcfg(Xn)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>oXn:</strong></dt>
<dd><p class="first last">should contain ALL nuisance covariates and intercepts
as in output of tor_get_physio.m</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p class="last">This function is automatically run by tor_get_physio.m</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="diagnostics.batch_efficiency">
<code class="descclassname">diagnostics.</code><code class="descname">batch_efficiency</code><span class="sig-paren">(</span><em>dwcard</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.batch_efficiency" title="Permalink to this definition">¶</a></dt>
<dd><p>Start in directory above individual model/results directories</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function batch_efficiency(dwcard)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dwcard:</strong></dt>
<dd><p class="first last">is a wildcard for directories to probe, e.g., &#8216;subject*&#8217;</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="diagnostics.batch_t_histograms">
<code class="descclassname">diagnostics.</code><code class="descname">batch_t_histograms</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.batch_t_histograms" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates page(s) of t stat histograms for each subject level contrast in
set of subject level analyses using image_intensity_histograms.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">batch_t_histograms</span><span class="p">([</span><span class="n">options</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>{analysis_dirs}:</strong></dt>
<dd><blockquote class="first">
<div><p>run on all contrasts in directories of cell array {analysis_dirs}</p>
</div></blockquote>
<p class="last">(DEFAULT: use all directories in working directory containing spmT_*.img files)</p>
</dd>
<dt>&#8216;o&#8217;, &#8216;output_directory&#8217;:**</dt>
<dd><p class="first last">specify output directory to contain saved .png files</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="diagnostics.canlab_qc_metrics1">
<code class="descclassname">diagnostics.</code><code class="descname">canlab_qc_metrics1</code><span class="sig-paren">(</span><em>epi_names</em>, <em>mask_name</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.canlab_qc_metrics1" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate quality control metrics on a 4-D EPI Analyze or Nifti file</p>
<p>Standard CANlab usage is to enter a single 4-D ravol* for one run, and
the brain mask implicit_mask.img created in canlab_preproc.m</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>epi_names:</strong></dt>
<dd><p class="first last">Names of (usually one) 4-D EPI file, in cell or string, full path</p>
</dd>
<dt><strong>mask_name:</strong></dt>
<dd><dl class="first last docutils">
<dt>Name of brain mask image, string, full path</dt>
<dd><p class="first last">IF EMPTY: Uses implict masking (better) and calculates
ghost/signal outside mask</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>noplot:</strong></dt>
<dd><p class="first last">skip plots</p>
</dd>
<dt><strong>noverbose:</strong></dt>
<dd><p class="first last">skip output printout to screen</p>
</dd>
<dt><strong>printfile:</strong></dt>
<dd><p class="first last">followed by name of file to print output to, full path</p>
</dd>
<dt><strong>noheader:</strong></dt>
<dd><p class="first last">suppress printing of header (var names) in output</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Missing values and basic info:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>num_images:</strong></dt>
<dd><p class="first last">number of images</p>
</dd>
<dt><strong>missing_vox:</strong></dt>
<dd><p class="first last">Voxels in mask with NaN values or zero values at every time point</p>
</dd>
<dt><strong>missing_images:</strong></dt>
<dd><p class="first last">Images with NaN values or zero values at every voxel</p>
</dd>
<dt><strong>missing_values:</strong></dt>
<dd><p class="first last">NaN or zero values in valid images / voxels. Zeros could
be interpreted as values of zero in analysis, causing artifacts in results
if these are actually invalid values.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Missing voxels will often be ignored in analyses in most software, but
Missing images/values could cause problems</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Basic signal to noise:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>perc-mean-ghost:</strong></dt>
<dd><p class="first last">mean signal outside the mask / mean total signal</p>
</dd>
<dt><strong>mean_snr:</strong></dt>
<dd><blockquote class="first">
<div><p>mean Cohen&#8217;s d (signal/noise, SNR) across time (temporal SNR) within the mask.</p>
</div></blockquote>
<p class="last">Mean divided by standard deviation across time at each voxel, averaged.  Higher is better.</p>
</dd>
<dt><strong>snr_inhomogeneity:</strong></dt>
<dd><p class="first last">standard deviation of SNR within the mask. Lower is better.</p>
</dd>
<dt><strong>snr_inhomogeneity95:</strong></dt>
<dd><p class="first last">95% confidence range for SNR within the mask. Lower is better.</p>
</dd>
<dt><strong>rms_successive_diffs:</strong></dt>
<dd><p class="first last">Essentially a high-pass filtered version of SNR,
expressed as a fraction of the overall mean and averaged across voxels. Lower is better.</p>
</dd>
<dt><strong>rms_successive_diffs_inhomogeneity:</strong></dt>
<dd><p class="first last">standard deviation of the above across voxels. Lower is better.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Left-right asymmetry:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>signal_rms_asymmetry:</strong></dt>
<dd><p class="first last">Voxel-wise left/right root mean square asymmetry in mean signal across time, expressed as
a fraction of the mean SNR.  Reflects both gross inhomogeneity and noise.  Lower is better.</p>
</dd>
<dt><strong>signal_hemispheric_asymmetry:</strong></dt>
<dd><p class="first last">Root mean square difference between left and
right hemispheres, expressed as a fraction of the grand mean signal across time.  Reflects
gross inhomogeneity.  Lower is better.</p>
</dd>
<dt><strong>snr_rms_asymmetry:</strong></dt>
<dd><p class="first last">Voxel-wise left/right root mean square asymmetry in SNR, expressed as
a fraction of the mean SNR.  Reflects both gross inhomogeneity and noise.  Lower is better.</p>
</dd>
<dt><strong>snr_hemispheric_asymmetry:</strong></dt>
<dd><p class="first last">Root mean square difference between left and
right hemispheres, expressed as a fraction of the mean SNR.  Reflects
gross inhomogeneity.  Lower is better.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>%SETUP
mydir{1} = &#39;/Users/tor/Documents/Tor_Documents/Coursework_and_Teaching/psyc7215/Data/UM_Face_House/060518mw/Functional/Preprocessed/run_01&#39;;
wcard = &#39;rarun*img&#39;;
epi_names = filenames(fullfile(mydir{1}, wcard), &#39;absolute&#39;);

maskdir = &#39;/Users/tor/Documents/Tor_Documents/Coursework_and_Teaching/psyc7215/Data/UM_Face_House/060518mw&#39;;
mask = &#39;implicit_mask.img&#39;;
mask_name = fullfile(maskdir, maskname);

%RUN
QC = canlab_qc_metrics1(epi_names, mask_name);

QC = canlab_qc_metrics1(epi_names, mask_name, &#39;noplot&#39;, &#39;printfile&#39;, &#39;test_qc.txt&#39;);
QC = canlab_qc_metrics1(epi_names, mask_name, &#39;noplot&#39;, &#39;printfile&#39;, &#39;test_qc.txt&#39;, &#39;noheader&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.check_cluster_data">
<code class="descclassname">diagnostics.</code><code class="descname">check_cluster_data</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.check_cluster_data" title="Permalink to this definition">¶</a></dt>
<dd><p>loads the first 5 images from the first voxel</p>
<blockquote>
<div><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">check_cluster_data</span><span class="p">(</span><span class="n">cl</span><span class="p">)</span>
</pre></div>
</div>
</div></blockquote>
<p>cl(1).imnames(1:5,:)</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.compare_subjects">
<code class="descclassname">diagnostics.</code><code class="descname">compare_subjects</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.compare_subjects" title="Permalink to this definition">¶</a></dt>
<dd><p>This function compares a set of images to one another and does some diagnostics on the similarity among images.
- It returns multivariate distances and dissimilarities among images
- It works on the GLOBAL signal after standardizing each image (case 1) or the REGIONAL values in each cluster (case 2) 
- You can also enter a reference image, in which case each image will be correlated with the ref.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [ds, g, mystd, d, d2, c, c2, mi, b, eigv, eigval] = compare_subjects([img files or clusters], [mask], ...
                                   [plot flag], [title on figure], [standardize flag], [text labels], [ref image])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first">a list of image names to compare</p>
<p>OR</p>
<p class="last">a clusters structure, with data to compare
in timeseries field</p>
</td>
</tr>
</tbody>
</table>
<p>If a mask is entered, only voxels in the mask (e.g., with value of 1) will be used.
You can use this option to specify brain-only or gray-matter only voxels</p>
<p>textlab: optional text labels for each image, can be empty []</p>
<p>If a ref image is entered, each image will be correlated with the ref,
and values will be saved for the correlation (plot 2 will show these values)
Useful for comparing anatomical imgs with template, etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">from correls with ref image are in variable &#8220;c&#8221;</p>
<dl class="docutils">
<dt><strong>ds:</strong></dt>
<dd><p class="first last">multivariate distance (sim. to Mahalanobis) for each image
ds is a matrix of squared distances, case numbers, and
expected chi2 values (in columns in this order) rows are cases</p>
</dd>
<dt><strong>g:</strong></dt>
<dd><p class="first last">global value for each image</p>
</dd>
<dt><strong>d:</strong></dt>
<dd><p class="first last">global distance from mean image
distance, or dissimilarity, is the average absolute deviation between images</p>
</dd>
<dt><strong>d2:</strong></dt>
<dd><p class="first last">matrix of distances among all images</p>
</dd>
<dt><strong>c:</strong></dt>
<dd><p class="first last">correlation between real valued voxels and mean image</p>
</dd>
<dt><strong>c2:</strong></dt>
<dd><p class="first last">correlations among all images (treating voxels as cases)</p>
</dd>
<dt><strong>mi:</strong></dt>
<dd><p class="first last">mutual information between images, with hist2.m</p>
</dd>
<dt><strong>b:</strong></dt>
<dd><p class="first last">principal component scores on correlation matrix for eigenvalues &gt; 1</p>
</dd>
<dt><strong>eigv:</strong></dt>
<dd><p class="first last">eigenvectors</p>
</dd>
<dt><strong>eigval:</strong></dt>
<dd><p class="first last">eigenvalues</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Compare normalized anatomcals with standard brain
P = get_filename2([&#39;sub*\Anatomy\nscalped_ft1.img&#39;]);
[ds, g, mystd, d, d2, c, c2, mi] = compare_subjects(P, which(&#39;brain_avg152T1.img&#39;), 1, &#39;intext_countloc&#39;, 1, [], which(&#39;avg152T1.img&#39;));
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.compare_subjects256">
<code class="descclassname">diagnostics.</code><code class="descname">compare_subjects256</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.compare_subjects256" title="Permalink to this definition">¶</a></dt>
<dd><p>This function compares a set of images to one another and does some diagnostics on the similarity among images.
- It returns multivariate distances and dissimilarities among images
- It works on the GLOBAL signal after standardizing each image (case 1) or the REGIONAL values in each cluster (case 2) 
- You can also enter a reference image, in which case each image will be correlated with the ref.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [ds,g,mystd,d,d2,c,c2,mi,b,eigv,eigval] = compare_subjects256([img files or clusters],[mask], ...
                               [plot flag],[title on figure],[standardize flag],[text labels],[ref image])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first">a list of image names to compare</p>
<p>OR</p>
<p class="last">a clusters structure, with data to compare
in timeseries field</p>
</td>
</tr>
</tbody>
</table>
<p>If a mask is entered, only voxels in the mask (e.g., with value of 1) will be used.
You can use this option to specify brain-only or gray-matter only voxels</p>
<p>textlab: optional text labels for each image, can be empty []</p>
<p>If a ref image is entered, each image will be correlated with the ref,
and values will be saved for the correlation (plot 2 will show these values)
Useful for comparing anatomical imgs with template, etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">from correls with ref image are in variable &#8220;c&#8221;</p>
<dl class="docutils">
<dt><strong>ds:</strong></dt>
<dd><p class="first last">multivariate distance (sim. to Mahalanobis) for each image
ds is a matrix of squared distances, case numbers, and
expected chi2 values (in columns in this order) rows are cases</p>
</dd>
<dt><strong>g:</strong></dt>
<dd><p class="first last">global value for each image</p>
</dd>
<dt><strong>d:</strong></dt>
<dd><p class="first last">global distance from mean image
distance, or dissimilarity, is the average absolute deviation between images</p>
</dd>
<dt><strong>d2:</strong></dt>
<dd><p class="first last">matrix of distances among all images</p>
</dd>
<dt><strong>c:</strong></dt>
<dd><p class="first last">correlation between real valued voxels and mean image</p>
</dd>
<dt><strong>c2:</strong></dt>
<dd><p class="first last">correlations among all images (treating voxels as cases)</p>
</dd>
<dt><strong>mi:</strong></dt>
<dd><p class="first last">mutual information between images, with hist2.m</p>
</dd>
<dt><strong>b:</strong></dt>
<dd><p class="first last">principal component scores on correlation matrix for eigenvalues &gt; 1</p>
</dd>
<dt><strong>eigv:</strong></dt>
<dd><p class="first last">eigenvectors</p>
</dd>
<dt><strong>eigval:</strong></dt>
<dd><p class="first last">eigenvalues</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Compare normalized anatomcals with standard brain
P = get_filename2([&#39;sub*\Anatomy\nscalped_ft1.img&#39;]);
[ds,g,mystd,d,d2,c,c2,mi] = compare_subjects256(P,which(&#39;brain_avg152T1.img&#39;),1,&#39;intext_countloc&#39;,1,[],which(&#39;avg152T1.img&#39;));
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.displayme">
<code class="descclassname">diagnostics.</code><code class="descname">displayme</code><span class="sig-paren">(</span><em>mm</em>, <em>txtlab</em>, <em>tlab2</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.displayme" title="Permalink to this definition">¶</a></dt>
<dd><p>Used in img_hist2 - included as internal function there.
this function is for indepenent re-display after img_hist2 is finished.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">subjM</span><span class="p">,</span><span class="n">Mtotalv</span><span class="p">]</span> <span class="o">=</span> <span class="n">displayme</span><span class="p">(</span><span class="n">mm</span><span class="p">,</span><span class="n">txtlab</span><span class="p">,</span><span class="n">tlab2</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% TO run:
[O.subjM,O.Mtotalv] = displayme(O.m,txtlab,&#39;MEANS&#39;);
[O.subjS,O.Stotalv] = displayme(O.s,txtlab,&#39;STD&#39;);
[O.subjW,O.Wtotalv] = displayme(O.w,txtlab,&#39;SKEWNESS&#39;);
[O.subjK,O.Ktotalv] = displayme(O.k,txtlab,&#39;KURTOSIS&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.ellipse">
<code class="descclassname">diagnostics.</code><code class="descname">ellipse</code><span class="sig-paren">(</span><em>x</em>, <em>v1</em>, <em>v2</em>, <em>c</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.ellipse" title="Permalink to this definition">¶</a></dt>
<dd><p>Gives x and y coordinates for an ellipse, given x coordinates,
at a distance of c</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[x,y] = ellipse(x,v1,v2,c,[sorting method])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first">Based on the formula for an ellipse, x^2/v1^2 + y^2/v2^2 = c</p>
<dl class="docutils">
<dt><strong>c:</strong></dt>
<dd><p class="first last">is the distance from the origin</p>
</dd>
<dt><strong>v1:</strong></dt>
<dd><p class="first last">is the x half-length</p>
</dd>
<dt><strong>v2:</strong></dt>
<dd><p class="first last">is the y half-length</p>
</dd>
<dt><strong>x:</strong></dt>
<dd><p class="first last">is a vector of points covering the x coordinates in the ellipse</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Sorting methods:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><ul class="first simple">
<li>sort by x, produces elliptical line in plot</li>
<li>sort by y, produces horizontal lines in plot</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[x,y]=ellipse((randn(1000,1)),1.5,2.5,1); figure; hh = plot(x(2:end-1),y(2:end-1),&#39;r-&#39;)
rotate(hh,[0 90],45)  % rotate around z-axis by 45 degrees
x2 = get(hh,&#39;XData&#39;); y2 = get(hh,&#39;YData&#39;); hold on; plot(x2,y2,&#39;bx&#39;);
rotate(hh,[0 90],-45)  % rotate original ellipse back

% fill
fill(x,y,&#39;r&#39;,&#39;FaceAlpha&#39;,.2)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.fft_calc">
<code class="descclassname">diagnostics.</code><code class="descname">fft_calc</code><span class="sig-paren">(</span><em>dat</em>, <em>TR</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.fft_calc" title="Permalink to this definition">¶</a></dt>
<dd><p>Simple function to calculate the FFT power of a 
data vector (dat) as a function of frequency,
given a sample-to-sample repetition time (TR)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">myfft</span><span class="p">,</span> <span class="n">freq</span><span class="p">]</span> <span class="o">=</span> <span class="n">fft_calc</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">TR</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.fmri_mask_thresh_canlab">
<code class="descclassname">diagnostics.</code><code class="descname">fmri_mask_thresh_canlab</code><span class="sig-paren">(</span><em>fmri_file</em>, <em>outputname</em>, <em>implicit_masking_method</em>, <em>plotfigs</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.fmri_mask_thresh_canlab" title="Permalink to this definition">¶</a></dt>
<dd><p>Implicit determination of which voxels are in-brain, based on the intensities of
functional images.  Assumes much (most) of the image has near-zero
background noise values, and the in-brain values are substantially
higher.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">mask_thresh</span><span class="p">,</span> <span class="n">cl</span><span class="p">,</span> <span class="n">inmaskvox</span><span class="p">,</span> <span class="n">in_mask_logical_vector</span><span class="p">,</span> <span class="n">maskfilename</span><span class="p">]</span> <span class="o">=</span> <span class="n">fmri_mask_thresh_canlab</span><span class="p">(</span><span class="n">fmri_file</span><span class="p">,</span> <span class="n">outputname</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>fmri_file:</strong></dt>
<dd><p class="first last">is either a list of file names or an fmri_data object</p>
</dd>
<dt><strong>File names:</strong></dt>
<dd><p class="first last">a (preferably) 4-D file of imaging data, Analyze .img or .nii</p>
</dd>
<dt><strong>fmri_data object:</strong></dt>
<dd><p class="first last">With multiple images loaded with <em>no</em> mask</p>
</dd>
<dt><strong>outputname:</strong></dt>
<dd><p class="first last">is a mask file output name, e.g., &#8216;mask.img&#8217;, with .img
extension. Empty [] means do not write output image.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Implicit_masking_method:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>mean:</strong></dt>
<dd><blockquote class="first">
<div><p>take the top 95% of voxels above the mean value.  used by</p>
</div></blockquote>
<p class="last">default if no value is entered</p>
</dd>
<dt><strong>dip:</strong></dt>
<dd><blockquote class="first">
<div><p>smooth the histogram and take the top 95% of values above the</p>
</div></blockquote>
<p class="last">first positive gradient</p>
</dd>
<dt><a href="#id11"><span class="problematic" id="id12">**</span></a>plotfigs</dt>
<dd><p class="first last">[1/0]: enable or suppress mask display and orthviews</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>mask_thresh:</strong></dt>
<dd><p class="first last">signal-value above which voxels are considered in brain</p>
</dd>
<dt><strong>c1:</strong></dt>
<dd><p class="first last">clusters, from iimg_indx2clusters</p>
</dd>
<dt><strong>inmaskvox:</strong></dt>
<dd><p class="first last">number of inmask voxels</p>
</dd>
<dt><strong>dat:</strong></dt>
<dd><p class="first last">binary matrix of voxels that are in (1) or out (0) of mask</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Note: we want to be more inclusive than not at this stage.</p>
<p>last edited Oct 2011 - add support for fmri_data/image_vector objects
added figure suppression, SG 12/14/15
defaults</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.get_filename">
<code class="descclassname">diagnostics.</code><code class="descname">get_filename</code><span class="sig-paren">(</span><em>dwcard</em>, <em>wcard</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.get_filename" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function P = get_filename(dir_wcard,file_wcard,[verbose])
</pre></div>
</div>
<p>Start in directory above individual subject directories</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dwcard:</strong></dt>
<dd><p class="first last">Enter dwcard for wildcard of directories to look in; e.g. &#8216;02*&#8217;</p>
</dd>
<dt><strong>wcard:</strong></dt>
<dd><p class="first last">Enter wcard for image or file to get - e.g., &#8216;beta_0001.img&#8217;
This can also include subdirectories (no <a href="#id13"><span class="problematic" id="id14">*</span></a>&#8216;s) &#8216;anatomy/beta*img&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><p class="first last">Returns list of all files in individual directories  in string matrix</p>
</td>
</tr>
</tbody>
</table>
<p>Missing files, or entries in directory that do not contain files, are 
removed from the list.</p>
<p>NOT entering a * in dwcard seems to produce an error.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">P</span> <span class="o">=</span> <span class="n">get_filename</span><span class="p">(</span><span class="s">&#39;02*&#39;</span><span class="p">,</span><span class="s">&#39;beta_0001*&#39;</span><span class="p">)</span>
<span class="n">P</span> <span class="o">=</span> <span class="n">get_filename</span><span class="p">(</span><span class="s">&#39;02*&#39;</span><span class="p">,</span><span class="s">&#39;beta_0001.img&#39;</span><span class="p">)</span>
<span class="n">P</span> <span class="o">=</span> <span class="n">get_filename</span><span class="p">(</span><span class="s">&#39;02*&#39;</span><span class="p">,</span><span class="s">&#39;anatomy/nnhe*_seg1.img&#39;</span><span class="p">)</span>
<span class="n">P</span> <span class="o">=</span> <span class="n">get_filename</span><span class="p">(</span><span class="s">&#39;020515sp*&#39;</span><span class="p">,</span><span class="s">&#39;anatomy/nnhe*_seg1.img&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>one * is allowed in first string, multiple <a href="#id15"><span class="problematic" id="id16">*</span></a>&#8216;s in second,
as long as they are in the filename, not directory names!</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.get_filename2">
<code class="descclassname">diagnostics.</code><code class="descname">get_filename2</code><span class="sig-paren">(</span><em>dwcard</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.get_filename2" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [P,P2,d] = get_filename2(search string (as with ls command),[verbose])
</pre></div>
</div>
<p>Start in directory above individual subject directories</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dwcard:</strong></dt>
<dd><p class="first last">Enter dwcard for wildcard of directories to look in; e.g. &#8216;02*&#8217;</p>
</dd>
<dt><strong>wcard:</strong></dt>
<dd><p class="first last">Enter wcard for image or file to get - e.g., &#8216;beta_0001.img&#8217;
This can also include subdirectories (no <a href="#id17"><span class="problematic" id="id18">*</span></a>&#8216;s) &#8216;anatomy/beta*img&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><blockquote class="first">
<div><p>Returns list of all files in individual directories  in string matrix</p>
</div></blockquote>
<dl class="last docutils">
<dt><strong>P:</strong></dt>
<dd><p class="first last">file names with full paths</p>
</dd>
<dt><strong>P2:</strong></dt>
<dd><p class="first last">file names only</p>
</dd>
<dt><strong>d:</strong></dt>
<dd><p class="first last">list of directories searched for files</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Missing files, or entries in directory that do not contain files, are 
removed from the list.
NOT entering a * in dwcard seems to produce an error.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">P</span> <span class="o">=</span> <span class="n">get_filename2</span><span class="p">(</span><span class="s">&#39;02*/beta_0001*&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>one * is allowed in the directory structure right now, 
multiple <a href="#id19"><span class="problematic" id="id20">*</span></a>&#8216;s in the filename.</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.hist2">
<code class="descclassname">diagnostics.</code><code class="descname">hist2</code><span class="sig-paren">(</span><em>A</em>, <em>B</em>, <em>res</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.hist2" title="Permalink to this definition">¶</a></dt>
<dd><p>2-D histogram with res bins</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">H</span><span class="p">,</span><span class="n">mi</span><span class="p">,</span><span class="n">H2</span><span class="p">]</span> <span class="o">=</span> <span class="n">hist2</span><span class="p">(</span><span class="n">A</span><span class="p">,</span><span class="n">B</span><span class="p">,</span><span class="n">res</span><span class="p">,[</span><span class="n">plot</span><span class="p">])</span>
</pre></div>
</div>
<p>A and B can be 3D, as in image volumes
mi is mutual information, a la spm_mireg.m</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.image_intensity_histograms">
<code class="descclassname">diagnostics.</code><code class="descname">image_intensity_histograms</code><span class="sig-paren">(</span><em>fout</em>, <em>imgs</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.image_intensity_histograms" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Makes a sheet (fout.png) of intensity distribution histograms of imgs.</dt>
<dd>Will put an even number of rows of subplots on each page saved.
If more than one page is needed, will title outputs fout_1.png, etc.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">image_intensity_histograms</span><span class="p">(</span><span class="n">fout</span><span class="p">,</span> <span class="n">imgs</span><span class="p">,</span> <span class="p">[</span><span class="n">options</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>fout:</strong></dt>
<dd><p class="first last">imfilename to be saved (&#8216;.png&#8217; will be appended)</p>
</dd>
<dt><strong>imgs:</strong></dt>
<dd><p class="first last">ima cell array of filenames of images to make histograms of</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">im</p>
</dd>
<dt><strong>&#8216;titles&#8217;, cellarray:</strong></dt>
<dd><p class="first last">use strings in cellarray as plot titles (DEFAULT: use file names from imgs)</p>
</dd>
<dt><strong>&#8216;bins&#8217;, n:</strong></dt>
<dd><p class="first last">use n bins in histograms (DEFAULT: 100)</p>
</dd>
<dt><strong>&#8216;ymax&#8217;, y:</strong></dt>
<dd><p class="first last">use y-axis from 0 to y (DEFAULT: 10,000)</p>
</dd>
<dt><strong>&#8216;xmax&#8217;, x:</strong></dt>
<dd><p class="first last">use x-axis from -x to x (DEFAULT: 10)</p>
</dd>
<dt><strong>&#8216;cols&#8217;, c:</strong></dt>
<dd><p class="first last">use c columns of subplots (DEFAULT: 5)</p>
</dd>
<dt><strong>&#8216;maxrows&#8217;, r:</strong></dt>
<dd><p class="first last">use no more than r columns of subplots per page (DEFAULT: 7)</p>
</dd>
<dt><strong>&#8216;includezeros&#8217;:</strong></dt>
<dd><p class="first last">include zero intensities in histograms (DEFAULT: exclude zeros)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Tor Wager
..
add path if necessary</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.img_hist">
<code class="descclassname">diagnostics.</code><code class="descname">img_hist</code><span class="sig-paren">(</span><em>imgname</em>, <em>subdir</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.img_hist" title="Permalink to this definition">¶</a></dt>
<dd><p>A general function for plotting histograms of any image
For each subject, comparing across subjects</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>imgname:</strong></dt>
<dd><p class="first last">name of image file to make intensity histograms from</p>
</dd>
<dt><strong>subdir:</strong></dt>
<dd><p class="first last">cell array of text strings containing names of individual subject
directories (wherein are contained the file specified in imgname 
or each subject)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Performs the histogram plot twice, once for CSF space
and once for gray matter</p>
<p>Locations of gray and CSF masks for each subject must
be defined in the defaults section of the script.
(hard-coded)</p>
<p>Start in directory above individual subject results</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>img_hist(&#39;beta_0010.img&#39;,subdir)
img_hist(&#39;con_0002.img&#39;,{&#39;020827mk&#39; &#39;020829jh&#39; &#39;020903lb&#39;}

% for batch
d = dir(&#39;020726ag/beta*img&#39;); d = str2mat(d.name);
for i = 1:10:size(d,1)
    img_hist(deblank(d(i,:)),EXPT.subjects)
end

for i = 2:19, 
   if i &lt; 10, myz = &#39;000&#39;;, else, myz = &#39;00&#39;;, end, 
   img_hist([&#39;con_&#39; myz num2str(i) &#39;.img&#39;],EXPT.subjects);, 
end
</pre></div>
</div>
<p>Tor Wager
..
..</p>
<blockquote>
<div>defaults</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="diagnostics.img_hist2">
<code class="descclassname">diagnostics.</code><code class="descname">img_hist2</code><span class="sig-paren">(</span><em>subdir</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.img_hist2" title="Permalink to this definition">¶</a></dt>
<dd><p>A general function for plotting histograms of any image
For each subject, comparing across subjects</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>imgname:</strong></dt>
<dd><p class="first last">name of image file to make intensity histograms from</p>
</dd>
<dt><strong>subdir:</strong></dt>
<dd><p class="first last">cell array of text strings containing names of individual subject
directories (wherein are contained the file specified in imgname 
or each subject)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Performs the histogram plot a number of times, without plotting
and reports the variance in pdf moments as a function of subject, 
run, and condition (beta img within run).</p>
<p>Start in directory above individual subject results</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">img_hist2</span><span class="p">(</span><span class="n">EXPT</span><span class="o">.</span><span class="n">subjects</span><span class="p">)</span>
<span class="n">img_hist2</span><span class="p">({</span><span class="s">&#39;020827mk&#39;</span> <span class="s">&#39;020829jh&#39;</span> <span class="s">&#39;020903lb&#39;</span><span class="p">})</span>
</pre></div>
</div>
<p>Tor Wager
..
..</p>
<blockquote>
<div>defaults</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="diagnostics.joint_hist">
<code class="descclassname">diagnostics.</code><code class="descname">joint_hist</code><span class="sig-paren">(</span><em>x</em>, <em>y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.joint_hist" title="Permalink to this definition">¶</a></dt>
<dd><p>Create 2-D joint histogram from vectors x and y</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">z</span><span class="p">,</span> <span class="n">xbins</span><span class="p">,</span> <span class="n">ybins</span><span class="p">]</span> <span class="o">=</span> <span class="n">joint_hist</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,[</span><span class="n">nbins</span><span class="p">],[</span><span class="n">noplot</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt>x and y:**</dt>
<dd><p class="first last">are vectors of paired observations on two variables</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>z:</strong></dt>
<dd><p class="first last">is the matrix representing the joint histogram
cols of z are bins of x, rows are bins of y
in plot, X axis is y, Y axis is x</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Optional: number of bins, suppress plotting</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>z = joint_hist(nnmfscores{i}{j}(:, 1),nnmfscores{i}{j}(:, 2), 50, &#39;noplot&#39;);
h = plot_joint_hist_contour(z, [0 0 1]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.make_conv_mtx">
<code class="descclassname">diagnostics.</code><code class="descname">make_conv_mtx</code><span class="sig-paren">(</span><em>sz</em>, <em>sampres</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.make_conv_mtx" title="Permalink to this definition">¶</a></dt>
<dd><p>Constructs the matrix (H) for a linear convolution
With the canonical SPM hrf
such that Hx = conv(x,hrf)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function H = make_conv_mtx(sz,sampres)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>sz:</strong></dt>
<dd><p class="first last">size of output matrix (elements)</p>
</dd>
<dt><strong>sampres:</strong></dt>
<dd><p class="first">spm_hrf sampling resolution (~ TR), OR</p>
<p class="last">if a vector, a custom HRF
sampled at the appropriate frequency.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Tor Wager
..</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.multivar_dist">
<code class="descclassname">diagnostics.</code><code class="descname">multivar_dist</code><span class="sig-paren">(</span><em>X</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.multivar_dist" title="Permalink to this definition">¶</a></dt>
<dd><p>multivariate normality checking and diagnostic plots</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">ds</span><span class="p">,</span> <span class="n">S</span><span class="p">,</span> <span class="n">p</span><span class="p">]</span> <span class="o">=</span> <span class="n">multivar_dist</span><span class="p">(</span><span class="n">X</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><p class="first">given matrix X with cases = rows, cols = variables</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>ds:</strong></dt>
<dd><p class="first">is matrix of squared distances, case numbers, and
expected chi2 values (in columns in this order)
rows are cases</p>
<p class="last">NOTE: Sorted in order of ascending distance!</p>
</dd>
<dt><strong>S:</strong></dt>
<dd><p class="first last">estimated covariance matrix</p>
</dd>
<dt><strong>mv_distance:</strong></dt>
<dd><p class="first last">squared distances in original order of rows</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">p-values in original order of rows</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>center</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.orthogonalize">
<code class="descclassname">diagnostics.</code><code class="descname">orthogonalize</code><span class="sig-paren">(</span><em>mX</em>, <em>X</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.orthogonalize" title="Permalink to this definition">¶</a></dt>
<dd><p>orthogonalizes X with respect to mX, optionally scaling predictors of X
For each nuisance covariate (column of X)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function X = orthogonalize(mX,X,[scale])
</pre></div>
</div>
<p>Regresses out model fits and saves residuals in X</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.power_from_variance">
<code class="descclassname">diagnostics.</code><code class="descname">power_from_variance</code><span class="sig-paren">(</span><em>con</em>, <em>N</em>, <em>sig2b</em>, <em>sig2wi</em>, <em>pthresh</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.power_from_variance" title="Permalink to this definition">¶</a></dt>
<dd><p>Power and effect size measures, given contrast, N, and variance component
estimates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>con:</strong></dt>
<dd><p class="first last">contrast/effect magnitude estimate; &#8220;mean difference&#8221;</p>
</dd>
<dt><strong>N:</strong></dt>
<dd><p class="first last">sample size</p>
</dd>
<dt><strong>sig2b:</strong></dt>
<dd><p class="first last">between-subjects variance estimate</p>
</dd>
<dt><strong>sig2wi:</strong></dt>
<dd><dl class="first last docutils">
<dt>within-subjects variance estimate</dt>
<dd><p class="first last"><em>note</em> this is not the &#8220;raw&#8221; within-subjects variance; it is
the contribution to the group (2nd-level) variance, which is
sig2within / number of images within-subjects</p>
</dd>
</dl>
</dd>
<dt><strong>pthresh:</strong></dt>
<dd><dl class="first last docutils">
<dt>alpha (Type I error) rate; p-value threshold for power</dt>
<dd><p class="first last">calculation</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>con, sig2b, and sig2wi can all be vectors, so you can run this function
voxel-wise for a whole map at once</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>power:</strong></dt>
<dd><p class="first last">Power from 0 to 1</p>
</dd>
<dt><strong>t:</strong></dt>
<dd><p class="first last">effect size : expected t-value</p>
</dd>
<dt><strong>d:</strong></dt>
<dd><p class="first last">effect size : Cohen&#8217;s d</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>see effect_size_map.m for a whole-brain, image-based power mapping
function</p>
<p>t-value threshold for significance at alpha level pthresh</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.power_loss">
<code class="descclassname">diagnostics.</code><code class="descname">power_loss</code><span class="sig-paren">(</span><em>y</em>, <em>ons</em>, <em>X</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.power_loss" title="Permalink to this definition">¶</a></dt>
<dd><p>&#8216;true&#8217; model fit
assume &#8216;true&#8217; is FIR estimate</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.publish_scn_session_spike_id">
<code class="descclassname">diagnostics.</code><code class="descname">publish_scn_session_spike_id</code><span class="sig-paren">(</span><em>inputimgs</em>, <em>SUBJDATA</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.publish_scn_session_spike_id" title="Permalink to this definition">¶</a></dt>
<dd><p>This function is a wrapper function to call scn_session_spike_id in
&#8216;multi-session&#8217; mode, using input data across the runs for a single
subject.  It runs the program, and generates both a yaml-format text file
for uploading into the CANlab database, and an html file with all the
results and images for that subject embedded.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>inputimgs:</strong></dt>
<dd><p class="first last">is a cell array of images (4-D) for each run in a separate cell.</p>
</dd>
<dt><strong>SUBJDATA:</strong></dt>
<dd><p class="first last">Input fields of SUBJDATA define the experiment name, subject name,
and directories for saving both QC images + yaml and HTML</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>SUBJDATA.study = &#39;NSF&#39;;
SUBJDATA.subject = subjects{i};
SUBJDATA.html_save_dir = fullfile(output_basedir, &#39;html_output&#39;);
SUBJDATA.subject_dir = fullfile(output_basedir, &#39;SubjectData&#39;, &#39;denoised_canlab&#39;, SUBJDATA.subject);
</pre></div>
</div>
<p>Initialize yaml file for database integration</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.qchist">
<code class="descclassname">diagnostics.</code><code class="descname">qchist</code><span class="sig-paren">(</span><em>images</em>, <em>Nbins</em>, <em>sparse</em>, <em>XLim</em>, <em>titles</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.qchist" title="Permalink to this definition">¶</a></dt>
<dd><p>This function generates a histogram of activations from a set of
statistic images.  Generally, you want the images to have a normal
distribution.  Highly skewed distributions may be indicative of bad
data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function: h = qchist(dat,Nbins,sparse,XLim)

  This function may generate multiple figuresa with 30 histograms
  each
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>images:</strong></dt>
<dd><p class="first last">List of image file names OR fmri_data object.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>Nbins:</strong></dt>
<dd><p class="first last">Number of bins in each histogram (default = 100)</p>
</dd>
<dt><strong>sparse:</strong></dt>
<dd><p class="first last">flag for generating ONLY histograms (default = 0)</p>
</dd>
<dt><strong>XLim:</strong></dt>
<dd><p class="first last">Xlim (default = [-1 1])</p>
</dd>
<dt><strong>titles:</strong></dt>
<dd><p class="first last">a cell array of subplot titles.  If omitted, titles
are inferred from assuming images come from a
directory structure that looks like the following:
/.../subjname/contrastimage.nii</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="diagnostics.reset_SPMcfg">
<code class="descclassname">diagnostics.</code><code class="descname">reset_SPMcfg</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.reset_SPMcfg" title="Permalink to this definition">¶</a></dt>
<dd><p>resets columns in SPMcfg by removing all non-intercept nuisance covariates.
runs on the SPMcfg.mat file in the current directory</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scale_imgs_by_csf">
<code class="descclassname">diagnostics.</code><code class="descname">scale_imgs_by_csf</code><span class="sig-paren">(</span><em>hP</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scale_imgs_by_csf" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes a string matrix of image file names
finds the mean and std of the CSF space
specified in a mask (hard-coded)
and standardizes images by these values</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">Pout</span> <span class="o">=</span> <span class="n">scale_imgs_by_csf</span><span class="p">(</span><span class="n">hP</span><span class="p">)</span>
</pre></div>
</div>
<p>Writes SC* images (SCaled)</p>
<p>assumes images are spatially normalized.
uses a canonical CSF mask!</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scn_component_rsquare">
<code class="descclassname">diagnostics.</code><code class="descname">scn_component_rsquare</code><span class="sig-paren">(</span><em>V</em>, <em>nuisanceX</em>, <em>designX</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scn_component_rsquare" title="Permalink to this definition">¶</a></dt>
<dd><p>Print a table of r-square values (variance explained) for each of V data
vectors by nuisance (mvmt, physio) and task-related predictors</p>
<p>Designed to work with components</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Typical operation
scn_component_rsquare(compscore, movement_params(1:157, :), X(1:157, :));

% No design
scn_component_rsquare(compscore, movement_params(1:157, :));

% Neither design nor nuisance, uses linear drift
scn_component_rsquare(compscore, []);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scn_session_spike_id">
<code class="descclassname">diagnostics.</code><code class="descname">scn_session_spike_id</code><span class="sig-paren">(</span><em>imgs</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scn_session_spike_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets global image values for a session, and uses trimts.m to find
outliers. The optional input MADs allows one to lower or raise the
threshold for identifying scans as spikes (default = 10).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[g, spikes, gtrim, nuisance_covs, spikesperimg, snr] = scn_session_spike_id(imgs,&#39;mask&#39;,[mask name],&#39;MADs&#39;,[MADs],&#39;doplot&#39;,[0/1])
</pre></div>
</div>
<p>Multi-session mode returns much more output and more images, and
takes in a cell array with images (preferably 4-D) for each session
(run).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;mask&#8217;,[pathtomaskfile]:</strong></dt>
<dd><p class="first last">mask images using the mask in pathtomaskfile, default: implicit mask</p>
</dd>
<dt><strong>&#8216;MADs&#8217;,[scalar]:</strong></dt>
<dd><p class="first last">change Mahalanobis distance, default: 10</p>
</dd>
<dt><strong>&#8216;doplot&#8217;,[0 / 1]:</strong></dt>
<dd><p class="first last">plot result figures, default: true</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Returns:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>g:</strong></dt>
<dd>global values</dd>
<dt><strong>spikes:</strong></dt>
<dd>identified spikes</dd>
<dt><strong>gtrim:</strong></dt>
<dd>trimmed/adjusted global values, can be used as covariate in GLM</dd>
<dt><strong>nuisance_covs:</strong></dt>
<dd>a matrix of 1)gtrim and 2) dummy regressors that can be used to minimize
spike influence in GLM</dd>
</dl>
</div></blockquote>
<p>We may want to save norms on the number of outliers found.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Get image names
for i = 1:6, sess_images{i} = filenames(sprintf(&#39;run%02d/vol0*img&#39;, i), &#39;char&#39;, &#39;absolute&#39;); end

% Run
[g, spikes, gtrim, nuisance_covs, snr] = scn_session_spike_id(sess_images);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scn_spm_choose_hpfilter">
<code class="descclassname">diagnostics.</code><code class="descname">scn_spm_choose_hpfilter</code><span class="sig-paren">(</span><em>spm_results_dir</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scn_spm_choose_hpfilter" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots and choice of optimal high-pass filter from an SPM first-level
model directory (with statistics and contrasts estimated.)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">scn_spm_choose_hpfilter</span><span class="p">(</span><span class="n">spm_results_dir</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;events_only&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>SPM5 compatible and SPM8.</p>
<p>Called by: scn_spm_design_check.m
For all regressors or events only: see scn_spm_choose_hpfilter.m</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scn_spm_design_check">
<code class="descclassname">diagnostics.</code><code class="descname">scn_spm_design_check</code><span class="sig-paren">(</span><em>spm_results_dir</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scn_spm_design_check" title="Permalink to this definition">¶</a></dt>
<dd><p>Run in a single-subject (first-level) SPM directory to check 
design matrix variance inflation factors and high-pass filtering.
Prints out table of regressors and their above-threshold VIFs (see options).
Saves .png images of the key figures.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">scn_spm_design_check</span><span class="p">(</span><span class="n">spm_results_dir</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;events_only&#8217;:</strong></dt>
<dd><p class="first last">Show plots and diagnostics for ONLY events, not nuisance covariates or
other user-specified regressors.  Useful when you have many nuisance
covs.</p>
</dd>
<dt><strong>&#8216;vif_thresh&#8217;, t&#8217;:</strong></dt>
<dd><p class="first last">Only regressors with a VIF &gt; t will be printed in VIF table.</p>
</dd>
<dt><strong>&#8216;sort_by_vif&#8217;&#8216;:</strong></dt>
<dd><p class="first last">Sort regressors in VIF table by VIF (DEFAULT: order regressors as in model).</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Calls: scn_spm_choose_hpfilter.m, scn_spm_get_events_of_interest.m</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">scn_spm_design_check</span><span class="p">(</span><span class="n">pwd</span><span class="p">,</span> <span class="s">&#39;events_only&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scn_spm_get_events_of_interest">
<code class="descclassname">diagnostics.</code><code class="descname">scn_spm_get_events_of_interest</code><span class="sig-paren">(</span><em>SPM</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scn_spm_get_events_of_interest" title="Permalink to this definition">¶</a></dt>
<dd><p>Gets events of interest.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">wh_cols</span> <span class="o">=</span> <span class="n">scn_spm_get_events_of_interest</span><span class="p">(</span><span class="n">SPM</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>All regressors, or events only if &#8216;events_only&#8217; is input as keyword
&#8216;from_multireg&#8217;:  followed by an integer, to include first n columns from
the multireg R matrix as &#8220;of interest&#8221;.  only works with &#8216;events_only&#8217;
flag, of course.</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scnlab_norm_check">
<code class="descclassname">diagnostics.</code><code class="descname">scnlab_norm_check</code><span class="sig-paren">(</span><em>template</em>, <em>wanat_files</em>, <em>mean_func_files</em>, <em>subjects</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scnlab_norm_check" title="Permalink to this definition">¶</a></dt>
<dd><p>Compares the similarity of one or two sets of images (wanat_files,
mean_func_files) to a template image and to one another (via Malanobis
distance) to determine whether some images are potential outliers.
This is used to check the quality of spatial warping/normalization for a
group of subjects, though it could be used for other purposes as well.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">NORM_CHECK</span> <span class="o">=</span> <span class="n">scnlab_norm_check</span><span class="p">(</span><span class="n">template</span><span class="p">,</span> <span class="n">wanat_files</span><span class="p">,</span> <span class="n">mean_func_files</span><span class="p">,</span> <span class="n">subjs</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>template:</strong></dt>
<dd><p class="first last">Char array with name of image of normalization template</p>
</dd>
<dt><strong>wanat_files:</strong></dt>
<dd><p class="first last">Warped (to template) anatomical file names</p>
</dd>
<dt><strong>mean_func_files:</strong></dt>
<dd><p class="first last">Names of mean functional images
These images should all be in the same space/in register.</p>
</dd>
<dt><strong>Subjs:</strong></dt>
<dd><p class="first last">Optional cell array of names for each subject, for display
purposes</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">A structure with metrics (NORM_CHECK)</p>
<dl class="last docutils">
<dt><strong>NORM_CHECK.global_t1:</strong></dt>
<dd><p class="first last">global values of first image series (wanat_files)</p>
</dd>
<dt><strong>NORM_CHECK.std_t1:</strong></dt>
<dd><p class="first last">spatial standard deviation of first image series (wanat_files)</p>
</dd>
<dt><strong>NORM_CHECK.names_t1:</strong></dt>
<dd><p class="first last">Names for columns of NORM_CHECK.norm_vs_template</p>
</dd>
<dt><strong>NORM_CHECK.subjects:</strong></dt>
<dd><p class="first last">Cell array of names for each subject</p>
</dd>
<dt><strong>NORM_CHECK.norm_vs_template:</strong></dt>
<dd><dl class="first last docutils">
<dt>Similarity data for subjects (rows) x metrics (cols)</dt>
<dd><p class="first last">{&#8216;Dist. from group, actual chi2&#8217;, &#8216;Mutual info with template&#8217;, &#8216;Correlation with template&#8217;};</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>NB: Leave mean_func_files empty (e.g., []) to only check structural images</p>
<p>Computes metrics on the goodness of normalization based on multivariate distance,
mutual information, and correlation with template. Automatically saves a .mat file
of the results into the current directory.</p>
<p>the template file (i.e., avg152T1.nii) must be in the CURRENT working
directory and have read/write permissions</p>
<p>USES the subfunction compare_subjects, which may be useful as a
stand-alone function.</p>
<p>USED in canlab_preproc_norm_check.m</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scnlab_norm_check3">
<code class="descclassname">diagnostics.</code><code class="descname">scnlab_norm_check3</code><span class="sig-paren">(</span><em>wt1</em>, <em>subjlabels</em>, <em>template</em>, <em>mask</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scnlab_norm_check3" title="Permalink to this definition">¶</a></dt>
<dd><p>WARNING:  scnlab_norm_check3 is deprecated! All improvements are being placed in scnlab_norm_check.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>EXPT = scnlab_norm_check3(wt1,subjlabels,template,mask,[print out MI])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>wt1:</strong></dt>
<dd><p class="first last">char array of wT1.img files, one per line</p>
</dd>
<dt><strong>subjlabels:</strong></dt>
<dd><p class="first last">cell array of subject labels</p>
</dd>
<dt><strong>template:</strong></dt>
<dd><p class="first last">template img that everything has been normalized to (usually the avg152T1.img file)</p>
</dd>
<dt><strong>mask:</strong></dt>
<dd><p class="first last">image to mask with</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Optional Input:</th><td class="field-body"><p class="first">print out mutual information table - flag for whether or not to print out the MI table; defaults to 0</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cd(studyroot); % wherever your study root is
wt1s = filenames(&#39;hr*/structural/wT1.img&#39;, &#39;char&#39;, &#39;absolute&#39;); % assuming that hr is your study code
subjlabels = filenames(&#39;hr*&#39;);
template = which(&#39;avg152T1.nii&#39;);
mask = filenames(&#39;scalped_avg152T1_graymatter.img&#39;, &#39;char&#39;, &#39;absolute&#39;); % set to wherever your mask is...

EXPT = scnlab_norm_check3(wt1, subjlabels, template, mask);
</pre></div>
</div>
<p>THIS FUNCTION IS DEPRECATED; SCNLAB_NORM_CHECK IS PREFERRED</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.scnlab_pca_check1">
<code class="descclassname">diagnostics.</code><code class="descname">scnlab_pca_check1</code><span class="sig-paren">(</span><em>imgs</em>, <em>realign_files</em>, <em>X</em>, <em>spersess</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.scnlab_pca_check1" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function scnlab_pca_check1(imgs, realign_files or params (t x 6) across all runs, X, spersess)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>imgs:</strong></dt>
<dd><p class="first last">list of all image names in order</p>
</dd>
<dt><strong>realign_files:</strong></dt>
<dd><p class="first">movement param file for each session, names in a cell array, OR</p>
<p class="last">a t x 6 matrix of realignment parameters across all sessions</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">design matrix; no intercept is needed</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% setup code for auditory oddball data
cd(&#39;/Users/tor/Documents/Tor_Documents/Coursework_and_Teaching/Mind_Res_Net_fMRI_Course_2008/data/auditory_oddball/2subjects-processed/s01/&#39;)

imgs = filenames(&#39;*/sw*img&#39;,&#39;absolute&#39;,&#39;char&#39;)
realign_files = filenames(&#39;*/rp*txt&#39;)

% LOAD TASK ONSETS and CREATE DESIGN MATRIX
onsets{1} = load(&#39;novel_stimuli_run1.asc&#39;);
onsets{2} = load(&#39;target_stimuli_run1.asc&#39;);
onsets{3} = load(&#39;standard_stimuli_run1.asc&#39;);
onsets{4} = load(&#39;novel_stimuli_run2.asc&#39;);
onsets{5} = load(&#39;target_stimuli_run2.asc&#39;);
onsets{6} = load(&#39;standard_stimuli_run2.asc&#39;);

regs_per_sess = 3;
nsess = 2;
for i = 1:length(onsets), onsets{i} = onsets{i}&#39;; end
X = cell(1, nsess);
X{1} = onsets2delta(onsets(1:3), 1, 249);
X{1} = X{1}(:, 1:end-1);
X{2} = onsets2delta(onsets(4:6), 1, 249);
X{2} = X{2}(:, 1:end-1);
X = blkdiag(X{:});
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.spm_general_hist">
<code class="descclassname">diagnostics.</code><code class="descname">spm_general_hist</code><span class="sig-paren">(</span><em>hP</em>, <em>mP</em>, <em>textlab</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.spm_general_hist" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function M = spm_general_hist(hP,mP,textlab,[suppress plot - enter anything])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>hP:</strong></dt>
<dd><p class="first last">list of file names to compute histograms from</p>
</dd>
<dt><strong>mP:</strong></dt>
<dd><p class="first last">list of file names to compute masks from</p>
</dd>
<dt><strong>textlab:</strong></dt>
<dd><p class="first last">text string, e.g. &#8216;ventricles&#8217; to label output tiffs</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><p class="first last">histograms for all input images (usually contrast images from individual
subjects) plotted against a normal curve.</p>
</td>
</tr>
</tbody>
</table>
<p>The expected output is that each image will have roughly mean 0, with
bumps or tails in the distribution of there are real activations in some
parts of the brain.</p>
<p>Looking at these histograms may be helpful for detecting outliers or 
subjects with strange contrast values.  These may be caused by
bad scaling, multicolinearity in the design matrix, acquisition artifacts,
task-correlated head movement, or ???</p>
<p>Histograms (blue) are overlaid on a Gaussian distribution (red)
with a mean of 0 and a standard deviation equal to that of the observed data.</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.spm_rfx_hist">
<code class="descclassname">diagnostics.</code><code class="descname">spm_rfx_hist</code><span class="sig-paren">(</span><em>cwd</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.spm_rfx_hist" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function spm_rfx_hist(cwd)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body">a directory name where an spm or SnPM random effects analysis lives</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body">histograms for all input images (usually contrast images from individual
subjects) plotted against a normal curve.</td>
</tr>
</tbody>
</table>
<p>The expected output is that each image will have roughly mean 0, with
bumps or tails in the distribution of there are real activations in some
parts of the brain.</p>
<p>Looking at these histograms may be helpful for detecting outliers or 
subjects with strange contrast values.  These may be caused by
bad scaling, multicolinearity in the design matrix, acquisition artifacts,
task-correlated head movement, or ???</p>
<p>Histograms (blue) are overlaid on a Gaussian distribution (red)
with a mean of 0 and a standard deviation equal to that of the observed data.</p>
</dd></dl>

<dl class="function">
<dt id="diagnostics.struct2yaml">
<code class="descclassname">diagnostics.</code><code class="descname">struct2yaml</code><span class="sig-paren">(</span><em>yamlfilename</em>, <em>DB</em>, <em>yamlfilemethod</em>, <em>dbmethod</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.struct2yaml" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">struct2yaml</span><span class="p">(</span><span class="n">yamlfilename</span><span class="p">,</span> <span class="n">DB</span><span class="p">,</span> <span class="n">yamlfilemethod</span><span class="p">,</span> <span class="n">dbmethod</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>yamlfilemethod:</strong></dt>
<dd><p class="first last">&#8216;new&#8217; or &#8216;add&#8217; (append)</p>
</dd>
<dt><strong>dbmethod:</strong></dt>
<dd><p class="first last">how the canlab database will handle the record.
&#8216;add&#8217;, &#8216;replace&#8217;, or &#8216;keep_existing&#8217;</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>translate structure into YAML format text file
this will be interpretable by the canlab database</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>yamlfilename = &#39;YAML_tmp.yaml&#39;;

DB.study = &#39;NSF&#39;;     % string; study code letters
DB.subject = &#39;001&#39;;   % string; subject ID number
DB.occasion = &#39;21&#39;;   % string; occasion ID; unique to subj*session
DB.unique_id = [DB.study &#39;_&#39; DB.subject &#39;_&#39; DB.occasion];
DB.mean_spikes_per_image = mean(cat(2, spikesperimg{:}));

struct2yaml(yamlfilename, DB, &#39;add&#39;, &#39;replace&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="diagnostics.tor_get_physio">
<code class="descclassname">diagnostics.</code><code class="descname">tor_get_physio</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#diagnostics.tor_get_physio" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">X</span><span class="p">,</span><span class="n">mP</span><span class="p">,</span><span class="n">spmP</span><span class="p">]</span> <span class="o">=</span> <span class="n">tor_get_physio</span><span class="p">([</span><span class="n">mP</span><span class="p">],[</span><span class="n">spmP</span><span class="p">],[</span><span class="n">nvoxels</span><span class="p">],[</span><span class="n">doortho</span><span class="p">])</span>
</pre></div>
</div>
<p>arguments are optional, but you must enter them in this order.</p>
<p>Tor Wager 10/21/02</p>
<p>Get nuisance covariates likely to be related to physiological noise and head motion
The algorithm:</p>
<p>The program extracts raw/preprocessed image data from the ventricles (CSF space), as
defined by a mask denoting which voxels are CSF for that subject.  
Either all voxels or a randomly selected subset [nvoxels] is subjected to
principal components analysis, to determine regular patters of drift over time
and across voxels.  Those patterns are expected to be related to global signal drift,
head movement, and physiological noise, and are assumed to be UNrelated to the task
of interest, by virtue of the fact that they occur in the ventricles.</p>
<p>PCA is done twice on the timeseries&#8217; of CSF voxels.  The first time, PCA is done
on the sums of squared values (not the correlations) of voxel timeserieses across
the entire experiment, mean-centered based on the whole experiment.  Most of the
coherent variation in this case is expected to be due to head movement and changes
in shims/gradients/etc. from run to run.  The SS values are used because we want to
weight the voxels with the highest variation most heavily, as they are presumably
picking up most of this signal.  The first 3 eigenvariates (canonical timeseries)
are saved.</p>
<p>Following, a separate, second PCA is done on the correlation matrix of data
within each session.  Session data for each voxel are mean-centered and scaled
relative to the session (variance of each voxel = 1).  We do this because 
physiological noise-related signals may produce periodic signals of different
magnitudes in different voxels, and we want to extract the most coherent signals
we can within each session.  So these eigenvariates are expected to reflect
primarily noise related to physiology (heart rate, respiration).  Up to 5 eigenvariates
for each session are saved (nothing with eigenvalue &lt; 1 is saved).</p>
<p>Next, the CSF-related nuisance covariates (eigenvariates from PCA) are combined
with existing nuisance covariates and intercept columns from the existing 
design matrix (SPMcfg xX).  The proportion of variance in each predictor of interest
explained by this nuisance basis set is calculated using regression, and the
nuisance covariates are orthogonalized with respect to each predictor of interest.
There are good and bad results of this step.  The bad is that any signal that 
tracks the predictors is attributed to the task, not to noise, even if it&#8217;s actually
caused by physiological artifact.  So the orthogonalized basis set does not
protect you from physiology or movement-related false positives.  However,
the nuisance covariates are also unlikely to reduce power in estimating you effects
of interest.  More importantly, it avoids false positives created when one 
predictor (A) is more highly correlated with the nuisance covariates than another
(B).  In practice, betas for A will tend to be smaller than B, given the same
actual response to both, and a random effects analysis on A-B will produce
false positive activations.  Orthogonalization of the nuisance set precludes this.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>mP:</strong></dt>
<dd><p class="first last">CSF mask image file.  <a href="#id21"><span class="problematic" id="id22">*</span></a>_seg3.img output from SPM is appropriate
should be in same space and have same dims as functionals
but automatic reslicing is done if necessary.</p>
</dd>
<dt><strong>spmP:</strong></dt>
<dd><p class="first last">name (full path name preferred) of SPMcfg.mat file to use
This contains the design matrix and raw/preproc image file names to use.</p>
</dd>
<dt><strong>nvoxels:</strong></dt>
<dd><p class="first last">Number of CSF voxels to use in PCA analysis
More than 100 can be very slow and memory intensive.
Fewer than 100 voxels loads a different way, and may be slower.
Best is probably between 100 - 1000.  800 runs pretty fast.</p>
</dd>
<dt><strong>doortho:</strong></dt>
<dd><p class="first last">Orthogonalize nuisance covariates with respect to regs of interest
This assumes that any signal that covaries with the task is, in fact,
due to the task, so it gives you some bias towards finding positive results.
However, the alternative is that nuisance covariates may soak up variance
related to the task, and you&#8217;ll miss activations.
In addition, if some regressors are more colinear with the nuisance set,
you can create false &#8220;activations&#8221; when comparing these regressors to other
ones.  This problem exists whether or not we choose to model nuisance 
covariates.  One solution is to use the ortho when doing random effects analyses,
as the sign and magnitude of nuisance-related activations would not be expected to be
the same across subjects unless the variance was really task-related.
Default is 1, or &#8220;yes, do orthogonalization.&#8221;</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>for functions called, see this .m file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% get filenames for SPMcfg files and CSF mask for each subject
cd C:\Tor_Documents\CurrentExperiments\intext2\RESULTS\model1
spmP = get_filename(&#39;sub*&#39;,&#39;SPMcfg.mat&#39;);
cd C:\Tor_Documents\CurrentExperiments\intext2\
mP = get_filename(&#39;sub*&#39;,&#39;anatomy/nscalped_f*seg3.img&#39;);
% Now run:
for i = 1:size(mP,1)  
    tor_get_physio(mP(i,:),spmP(i,:),300);  % 300 voxels
    pause(10); close all
end
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Functions called:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><ul class="first last simple">
<li>spm functions: spm_get, etc.</li>
<li>timeseries2.m      (for &lt; 100 voxels)</li>
<li>read_hdr.m (big-little endian dependent; validate for your data)</li>
<li>timeseries3.m      (for &gt; 100 voxels; uses SPM&#8217;s image reading)</li>
<li>reslice_imgs.m</li>
<li>mask2voxel.m       (only if ind2sub.m from Matlab is not found)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-GLM_Batch_tools">
<span id="glm-batch-tools"></span><h2>GLM_Batch_tools<a class="headerlink" href="#module-GLM_Batch_tools" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="GLM_Batch_tools.canlab_glm_getinfo">
<code class="descclassname">GLM_Batch_tools.</code><code class="descname">canlab_glm_getinfo</code><span class="sig-paren">(</span><em>modeldir</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#GLM_Batch_tools.canlab_glm_getinfo" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">SUBJECT LEVEL input:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">INFO</span> <span class="o">=</span> <span class="n">canlab_glm_getinfo</span><span class="p">(</span><span class="n">spm_subject_level_directory</span><span class="p">,</span> <span class="n">option</span><span class="p">,</span> <span class="p">[</span><span class="n">n</span><span class="p">])</span>
</pre></div>
</div>
<p>Get information out of an spm subject level analysis.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><p class="first">(each option can be called by the listed letter or word + a number, when noted)</p>
<dl class="docutils">
<dt><strong>i&#8217; &#8216;input&#8217;:</strong></dt>
<dd><p class="first last">number of volumes and (first) volume name for each run</p>
</dd>
<dt><strong>&#8216;b&#8217; &#8216;betas&#8217; [n]:</strong></dt>
<dd><p class="first last">beta names (for nth session)</p>
</dd>
<dt><strong>&#8216;B&#8217; &#8216;taskbetas&#8217; [n]:</strong></dt>
<dd><p class="first last">beta names (that didn&#8217;t come from multiple regressors)  (for nth session)</p>
</dd>
<dt><strong>&#8216;c&#8217; &#8216;cons&#8217; [n]:</strong></dt>
<dd><p class="first last">contrast names (for nth contrast)</p>
</dd>
<dt><strong>&#8216;C&#8217; &#8216;conw&#8217; [n]:</strong></dt>
<dd><p class="first last">beta names and weights for contrasts (for nth con)</p>
</dd>
<dt><strong>&#8216;v&#8217; &#8216;image&#8217; [n]:</strong></dt>
<dd><dl class="first last docutils">
<dt>create figure of design matrix (for nth session)</dt>
<dd><p class="first last">(design matrix is multiplied by 100 for visibility)
(works well for multiple runs)</p>
</dd>
</dl>
</dd>
<dt><strong>&#8216;V&#8217; &#8216;taskimage&#8217; [n]:</strong></dt>
<dd><p class="first last">same as &#8216;image&#8217;, but only for task betas</p>
</dd>
<dt><strong>&#8216;imagesc&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>same as &#8216;image&#8217;, but uses imagesc</dt>
<dd><p class="first last">(works well for single runs)</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">GROUP LEVEL input:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">INFO</span> <span class="o">=</span> <span class="n">canlab_glm_getinfo</span><span class="p">(</span><span class="n">robfit_group_level_directory</span><span class="p">,</span> <span class="n">option</span><span class="p">,</span> <span class="p">[</span><span class="n">n</span><span class="p">])</span>
</pre></div>
</div>
<p>Get information out of a robfit group level analysis.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><p class="first">(each option can be called by the listed word or letter + a number, when noted)</p>
<p>Any of the subject level options can be used on a group level robfit
analysis by prefixing &#8216;1i&#8217; (output is generated based on the first input
to the first robust analysis).</p>
<p>Ex:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">canlab_glm_getinfo</span><span class="p">(</span><span class="s">&#39;second_level/model3&#39;</span><span class="p">,</span><span class="s">&#39;1iconw&#39;</span><span class="p">)</span>
</pre></div>
</div>
<dl class="docutils">
<dt><strong>&#8216;i&#8217; &#8216;input&#8217; [n]:</strong></dt>
<dd><p class="first last">input contrasts by number and name (for nth analysis)</p>
</dd>
<dt><strong>&#8216;I&#8217; &#8216;allinput&#8217; [n]:</strong></dt>
<dd><p class="first last">input images (for nth analysis)</p>
</dd>
<dt><strong>&#8216;m&#8217; &#8216;model&#8217;:</strong></dt>
<dd><p class="first last">weights by subject (i.e., directory containing input contrast images)</p>
</dd>
<dt><strong>&#8216;M&#8217; &#8216;allmodels&#8217; [n]:</strong></dt>
<dd><p class="first last">weights and input images (for nth analysis)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Assumptions:</th><td class="field-body"><p class="first">In some options, the first contrasts and group level analysis
directories are assumed to represent the rest, which may not be the
case.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body"><p class="first last">group level options do not yet return a usable INFO struct.</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="GLM_Batch_tools.canlab_glm_group_levels">
<code class="descclassname">GLM_Batch_tools.</code><code class="descname">canlab_glm_group_levels</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#GLM_Batch_tools.canlab_glm_group_levels" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Performs group level robust GLM analysis with robfit</dt>
<dd><ol class="first last arabic simple">
<li>sets up analysis</li>
<li>runs robfit</li>
<li>(optionally) make inverse p maps (for FSL viewing)</li>
<li>(optionally) estimate significant cluster sizes</li>
<li>publishes analysis with robfit_results_batch</li>
</ol>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">canlab_glm_group_levels</span><span class="p">([</span><span class="n">options</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;s&#8217;, subjects:</strong></dt>
<dd><p class="first last">cell array of filenames of subject-level analysis directories IN
modeldir
(note: modeldir won&#8217;t be prepended to absolute paths)</p>
</dd>
<dt><strong>&#8216;m&#8217;, modeldir:</strong></dt>
<dd><p class="first last">filename of directory containing subject level analyses</p>
</dd>
<dt><strong>&#8216;o&#8217;, grpmodeldir:</strong></dt>
<dd><p class="first last">output directory name</p>
</dd>
<dt><strong>&#8216;c&#8217;, cov:</strong></dt>
<dd><p class="first">a matrix describing group level model
(do not include intercept, it is automatically included as first regressor)</p>
<p>see help robfit</p>
<p>note: requires specifying an output directory name</p>
<p class="last">note: ordering of inputs (rows) must match subjects ordering</p>
</dd>
<dt><strong>&#8216;n&#8217;, covname:</strong></dt>
<dd><p class="first last">a cell array of names for the covariates in cov</p>
</dd>
<dt><strong>&#8216;f&#8217;, covfile:</strong></dt>
<dd><p class="first last">a csv file will specify the group level model:
first column, header &#8216;subject&#8217;, contains names of subject
directories (to be found in modeldir)
subsequent columns have covariates, headers are names of covariates
name of covfile will be name of group analysis directory (placed in
grpmodeldir)</p>
</dd>
<dt><strong>&#8216;mask&#8217;, maskimage:</strong></dt>
<dd><p class="first last">filename of mask image</p>
</dd>
<dt><strong>DSGN:</strong></dt>
<dd><p class="first">will use the following fields of the DSGN structure:
modeldir    = DSGN.modeldir</p>
<p>subjects    = DSGN.subjects</p>
<p class="last">maskimage   = DSGN.mask</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Note:</th><td class="field-body"><p class="first">A covfile will cause other specifications of subject, cov, and 
covnames to be ignored.</p>
<p>If parameters are defined more than once (e.g., modeldir or subjects), 
only the last entered option will count.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Defaults:</th><td class="field-body"><dl class="first docutils">
<dt><strong>subjects:</strong></dt>
<dd><p class="first last">all SPM.mat-containing directories in modeldir</p>
</dd>
<dt><strong>modeldir:</strong></dt>
<dd><p class="first last">pwd</p>
</dd>
<dt><strong>grpmodeldir:</strong></dt>
<dd><p class="first last">modeldir/one_sample_t_test</p>
</dd>
<dt><strong>cov:</strong></dt>
<dd><p class="first last">{} (run 1 sample t-test, see help robfit)</p>
</dd>
<dt><strong>covname:</strong></dt>
<dd><p class="first last">&#8216;groupmean&#8217;</p>
</dd>
<dt><strong>mask:</strong></dt>
<dd><p class="first last">&#8216;brainmask.nii&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Options:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;README&#8217;:</strong></dt>
<dd><p class="first last">prints canlab_glm_README, an overview of canlab_glm_{subject,group}_levels</p>
</dd>
<dt><strong>&#8216;overwrite&#8217;:</strong></dt>
<dd><p class="first last">overwrite existing output directories</p>
</dd>
<dt><strong>&#8216;noresults&#8217;:</strong></dt>
<dd><p class="first last">don&#8217;t run/publish robfit_results_batch</p>
</dd>
<dt><strong>&#8216;onlyresults&#8217;:</strong></dt>
<dd><p class="first last">just run/publish robfit_results_batch, don&#8217;t run robfit (assumes existing analyses)</p>
</dd>
<dt><a href="#id23"><span class="problematic" id="id24">**</span></a>&#8216;whichcons&#8217;, [which cons]</dt>
<dd><p class="first last">vector of contrasts to analyze (DEFAULT: aall subject level contrasts)
see [which cons] in help robfit</p>
</dd>
<dt><strong>&#8216;invp&#8217; [, target_space_image]:</strong></dt>
<dd><p class="first last">generate inverse p maps and resample to the voxel and image dimensions
of target_space_image
(viewable on dream with ~ruzicl/scripts/invpview)</p>
</dd>
<dt><strong>&#8216;nolinks&#8217;:</strong></dt>
<dd><p class="first last">do not make directory of named links to robust directories (using contrast names)</p>
</dd>
<dt><strong>&#8216;dream&#8217;:</strong></dt>
<dd><blockquote class="first">
<div><p>if you&#8217;re running on the dream cluster, this option will cause</p>
</div></blockquote>
<p class="last">all analyses (e.g., lower level contrasts) to be run in parallel 
(submitted with matlab DCS and the Sun Grid Engine)
Note: currently only works with MATLAB R2009a</p>
</dd>
<dt><strong>&#8216;email&#8217;, address:</strong></dt>
<dd><p class="first last">send notification email to address when done running</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="GLM_Batch_tools.canlab_glm_group_levels_run1input">
<code class="descclassname">GLM_Batch_tools.</code><code class="descname">canlab_glm_group_levels_run1input</code><span class="sig-paren">(</span><em>wd</em>, <em>c</em><span class="sig-paren">)</span><a class="headerlink" href="#GLM_Batch_tools.canlab_glm_group_levels_run1input" title="Permalink to this definition">¶</a></dt>
<dd><p>child process of canlab_glm_group_levels
(see canlab_glm_README.txt for an overview)</p>
</dd></dl>

<dl class="function">
<dt id="GLM_Batch_tools.canlab_glm_maskstats">
<code class="descclassname">GLM_Batch_tools.</code><code class="descname">canlab_glm_maskstats</code><span class="sig-paren">(</span><em>DIRS</em>, <em>MASK</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#GLM_Batch_tools.canlab_glm_maskstats" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Returns a MASKSTATS structure containing data from robfitdir&#8217;s input subject level</dt>
<dd>SPM analyses.</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">MASKSTATS</span> <span class="o">=</span> <span class="n">canlab_glm_maskstats</span><span class="p">(</span><span class="n">robfitdir</span><span class="p">,</span> <span class="n">mask</span><span class="p">,</span> <span class="p">[</span><span class="n">options</span><span class="p">])</span>
</pre></div>
</div>
<p>output structure:
MASKSTATS</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>COV:</strong></dt>
<dd>subject x covariate matrix (EXPT.cov from robfit design)</dd>
<dt><strong>COVNAME:</strong></dt>
<dd>names of covariates in COV</dd>
<dt><strong>MASK:</strong></dt>
<dd>array of structs (1 per mask)</dd>
<dt><strong>MASKFILE:</strong></dt>
<dd>the filename of the mask used</dd>
<dt><strong>SUB:</strong></dt>
<dd>struct containing data from subject level images</dd>
<dt><strong>CON:</strong></dt>
<dd>struct contains data from contrast images</dd>
<dt><strong>NAME:</strong></dt>
<dd>cell array (1 cell per contrast) of contrast names</dd>
<dt><strong>IMGFILES:</strong></dt>
<dd>cell array (1 cell per contrast) of character arrays of 
contrast image filenames</dd>
<dt><strong>(MEASURE):</strong></dt>
<dd>subject X contrast matrix of measures (see canlab_maskstats)</dd>
<dt><strong>COVxCON.(MEASURE):</strong></dt>
<dd>arrays of covariate matrix X contrast means correlation
results (RHO and P, see help corr())</dd>
<dt><strong>GRP:</strong></dt>
<dd>struct containing data from group level images</dd>
<dt><strong>BETA:</strong></dt>
<dd>struct array (1 struct per group level regressor) of data
from beta images</dd>
<dt><strong>NAME:</strong></dt>
<dd>name of group level regressor</dd>
<dt><strong>IMGFILES:</strong></dt>
<dd>cell array (1 cell per robust directory) of beta image files</dd>
<dt><strong>(MEASURE):</strong></dt>
<dd>row vector (1 value per robust directory) of measures
(see canlab_maskstats)</dd>
</dl>
</div></blockquote>
<dl class="docutils">
<dt>The following plots are saved in each mask&#8217;s directory in the plots directory:</dt>
<dd><ul class="first last simple">
<li>contrast means by contrast (means are lines across subjects on x axis)</li>
<li>contrast means by subject (means are dots, lined up along the x axis by contrast)</li>
<li>group level betas (bar plot with group level regressors grouped by
subject level contrasts)</li>
</ul>
</dd>
<dt>If there&#8217;s more than one regressor in the group level model, for each regressor:</dt>
<dd><ul class="first last simple">
<li>scatter plot of subject level contrast means against group level regressor</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Arguments:</th><td class="field-body"><dl class="first docutils">
<dt><strong>robfitdir:</strong></dt>
<dd><p class="first last">a directory in which robfit was run
contains robfit directories (e.g., robust0001)
preferably contains EXPT.mat or EXPTm.mat</p>
</dd>
<dt><strong>mask:</strong></dt>
<dd><p class="first last">a filename or cell array of filenames of masks to apply to data</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Options:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>MEASURE OPTIONS:</strong></dt>
<dd><p class="first last">(see canlab_maskstats) (DEFAULT: mean (within mask&#8217;s non-zero voxels))</p>
</dd>
<dt><strong>&#8216;cons&#8217;, connums:</strong></dt>
<dd><p class="first last">(vector of contrast numbers)
only include data from contrasts as specified by numbers</p>
</dd>
<dt><strong>&#8216;cons&#8217;, conname(s):</strong></dt>
<dd><p class="first last">(string or cell array of strings)
only include data from contrasts as specified by name</p>
</dd>
<dt><strong>&#8216;plots&#8217;:</strong></dt>
<dd><p class="first last">make plots</p>
</dd>
<dt><strong>&#8216;od&#8217;, dir:</strong></dt>
<dd><p class="first last">will save plots in dir (DEFAULT: robfitdir/stats_scatterplots)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="GLM_Batch_tools.canlab_glm_publish">
<code class="descclassname">GLM_Batch_tools.</code><code class="descname">canlab_glm_publish</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#GLM_Batch_tools.canlab_glm_publish" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">canlab_glm_publish</span><span class="p">(</span><span class="n">directory_specifications</span> <span class="p">[</span><span class="n">options</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Directory Specification:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;s&#8217;, dirs:</strong></dt>
<dd><p class="first">Generates HTML reports of the output from scn_spm_design_check for
directories in cell array dires (string allowable for single dir).
If a directory is a subject level analysis, the HTML will be generated 
for that subject in the analysis directory.
Ex:</p>
<p>canlab_glm_publish(&#8216;s&#8217;,{&#8216;model1/1011&#8217; &#8216;model1/1014&#8217;})</p>
<p>If a directory contains subject level analyses, an HTML will be
generated with output for each subject level analysis.
Ex:</p>
<p>canlab_glm_publish(&#8216;s&#8217;,&#8217;model1&#8217;)</p>
<p class="last">ASSUMPTION: lower level analyses contain an SPM.mat file.</p>
</dd>
<dt><strong>&#8216;g&#8217;, dirs:</strong></dt>
<dd><p class="first">For each &#8220;robfit directory&#8221; in cell array dirs, will run robust_results_batch 
on all contrast directories (e.g., robust0001/) (string allowable for single dir).
(&#8220;robfit directories&#8221; contain robfit contrast directories (like robust0001))
EITHER directories contain EXPT.mat files (see help robfit)
OR an EXPT struct is loaded in the workspace and a single directory is specified
OR will do best to figure out info normally contained in EXPT.mat
Ex:</p>
<p class="last">canlab_glm_publish(&#8216;g&#8217;, {&#8216;group_n35&#8217; &#8216;group_anxiety_n35&#8217; &#8216;group_sadness_n35&#8217;})</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Note:</th><td class="field-body"><p class="first">directory paths may be absolute or relative (to working directory)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;t&#8217;, {[pthresh clustersize] ...}:</strong></dt>
<dd><p class="first">Use the paired voxelwise_pthresh and minimum_cluster_size thresholds 
with which to produce robfit results maps.
This option must follow immediately after a &#8216;g&#8217; option (see above) and
will only apply to the analyses specified in that option.
ONLY applies to robfit directories (no bearing on lower level design checks)</p>
<p>DEFAULT: {[.001 5] [.005 1] [.05 1]}
Ex:</p>
<p class="last">canlab_glm_publish(&#8216;g&#8217;, pwd, &#8216;t&#8217;, {[.001 1] [.005 10] [.05 10] [.01 25]})</p>
</dd>
<dt><strong>&#8216;email&#8217;, address:</strong></dt>
<dd><p class="first">send notification email to address when done running
Ex:</p>
<p class="last">canlab_glm_publish(&#8216;g&#8217;, pwd, &#8216;email&#8217;, <a class="reference external" href="mailto:'ruzic&#37;&#52;&#48;colorado&#46;edu">'ruzic<span>&#64;</span>colorado<span>&#46;</span>edu</a>&#8216;)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="GLM_Batch_tools.canlab_glm_roistats">
<code class="descclassname">GLM_Batch_tools.</code><code class="descname">canlab_glm_roistats</code><span class="sig-paren">(</span><em>DIRS</em>, <em>ROI</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#GLM_Batch_tools.canlab_glm_roistats" title="Permalink to this definition">¶</a></dt>
<dd><p>Use canlab_glm_maskstats instead</p>
</dd></dl>

<dl class="function">
<dt id="GLM_Batch_tools.canlab_glm_subject_levels">
<code class="descclassname">GLM_Batch_tools.</code><code class="descname">canlab_glm_subject_levels</code><span class="sig-paren">(</span><em>dsgnarg</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#GLM_Batch_tools.canlab_glm_subject_levels" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Performs lower level GLM analysis with SPM:</dt>
<dd><ol class="first last arabic simple">
<li>specifies model</li>
<li>estimates model</li>
<li>generates contrast images for model</li>
<li>creates directory with named links to spmT and con maps</li>
<li>publishes analyses with scn_spm_design_check</li>
</ol>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">canlab_glm_subject_levels</span><span class="p">(</span><span class="n">DSGN</span> <span class="p">[</span><span class="n">options</span><span class="p">])</span>
</pre></div>
</div>
<p>DSGN struct - defines the model and analysis parameters</p>
<p>canlab_glm_subject_levels(&#8216;README&#8217;) to see description</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;README&#8217;:</strong></dt>
<dd><p class="first last">prints canlab_glm_README, an overview of canlab_glm_{subject,group}_levels</p>
</dd>
<dt><strong>&#8216;dsgninfo&#8217;:</strong></dt>
<dd><p class="first last">prints description of DSGN structure</p>
</dd>
<dt><strong>&#8216;subjects&#8217;, subject_list:</strong></dt>
<dd><p class="first last">ignore DSGN.subjects, use cell array subject_list</p>
</dd>
<dt><strong>&#8216;overwrite&#8217;:</strong></dt>
<dd><p class="first last">turn on overwriting of existing analyses (DEFAULT: skip existing)</p>
</dd>
<dt><strong>&#8216;onlycons&#8217;:</strong></dt>
<dd><p class="first last">only run contrast job (no model specification or estimation)
note: will overwrite existing contrasts
note: to not run contrasts, simply do not include a contrasts field in DSGN</p>
</dd>
<dt><strong>&#8216;addcons&#8217;:</strong></dt>
<dd><p class="first last">only run contrasts that aren&#8217;t already in SPM.mat
option to canlab_spm_contrast_job</p>
</dd>
<dt><strong>&#8216;nodelete&#8217;:</strong></dt>
<dd><p class="first last">do not delete existing contrasts (consider using addcons, above)
option to canlab_spm_contrast_job</p>
</dd>
<dt><strong>&#8216;nolinks&#8217;:</strong></dt>
<dd><p class="first last">will not make directory with named links to contrast images</p>
</dd>
<dt><strong>&#8216;noreview&#8217;:</strong></dt>
<dd><p class="first last">will not run scn_spm_design_check</p>
</dd>
<dt><strong>&#8216;dream&#8217;:</strong></dt>
<dd><p class="first last">if you&#8217;re running on the dream cluster, this option will cause
all subjects to be run in parallel (submitted with matlab DCS and
the Sun Grid Engine)
Note: currently only works with MATLAB R2009a</p>
</dd>
<dt><strong>&#8216;email&#8217;, address:</strong></dt>
<dd><p class="first last">send notification email to address when done running</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Model specification and estimation done by canlab_spm_fmri_model_job</p>
<p>Contrasts are specified by canlab_spm_contrast_job_luka
see that function for more info.</p>
</dd></dl>

<dl class="function">
<dt id="GLM_Batch_tools.canlab_glm_subject_levels_run1subject">
<code class="descclassname">GLM_Batch_tools.</code><code class="descname">canlab_glm_subject_levels_run1subject</code><span class="sig-paren">(</span><em>wd</em>, <em>s</em><span class="sig-paren">)</span><a class="headerlink" href="#GLM_Batch_tools.canlab_glm_subject_levels_run1subject" title="Permalink to this definition">¶</a></dt>
<dd><p>child process of canlab_glm_subject_levels
(see canlab_glm_README.txt for an overview)</p>
</dd></dl>

</div>
<div class="section" id="hrf-est-toolbox2">
<h2>HRF_Est_Toolbox2<a class="headerlink" href="#hrf-est-toolbox2" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-HRF_Est_Toolbox2"></span><dl class="function">
<dt id="HRF_Est_Toolbox2.Anneal_Logit">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">Anneal_Logit</code><span class="sig-paren">(</span><em>theta0</em>, <em>t</em>, <em>tc</em>, <em>Run</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.Anneal_Logit" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimate inverse logit (IL) HRF model using Simulated Annealing
Creates fitted curve - 3 logistic functions to be summed together - from parameter estimates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">theta</span><span class="p">,</span><span class="n">HH</span><span class="p">,</span><span class="n">C</span><span class="p">,</span><span class="n">P</span><span class="p">]</span> <span class="o">=</span> <span class="n">Anneal_Logit</span><span class="p">(</span><span class="n">theta0</span><span class="p">,</span><span class="n">t</span><span class="p">,</span><span class="n">tc</span><span class="p">,</span><span class="n">Run</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Run:</strong></dt>
<dd><p class="first last">stick function</p>
</dd>
<dt><strong>tc:</strong></dt>
<dd><p class="first last">time course</p>
</dd>
<dt><strong>t:</strong></dt>
<dd><p class="first last">vector of time points</p>
</dd>
<dt><strong>theta0:</strong></dt>
<dd><p class="first last">initial value for the parameter vector</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.Det_Logit">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">Det_Logit</code><span class="sig-paren">(</span><em>V0</em>, <em>t</em>, <em>tc</em>, <em>Run</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.Det_Logit" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimate inverse logit (IL) HRF model 
Creates fitted curve - 3 logistic functions to be summed together - from parameter estimates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">VM</span><span class="p">,</span> <span class="n">h</span><span class="p">,</span> <span class="n">fit</span><span class="p">,</span> <span class="n">e</span><span class="p">,</span> <span class="n">param</span><span class="p">]</span> <span class="o">=</span> <span class="n">Det_Logit_allstim</span><span class="p">(</span><span class="n">V0</span><span class="p">,</span><span class="n">t</span><span class="p">,</span><span class="n">tc</span><span class="p">,</span><span class="n">Run</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Run:</strong></dt>
<dd><p class="first last">stick function</p>
</dd>
<dt><strong>tc:</strong></dt>
<dd><p class="first last">time course</p>
</dd>
<dt><strong>t:</strong></dt>
<dd><p class="first last">vector of time points</p>
</dd>
<dt><strong>V0:</strong></dt>
<dd><p class="first last">initial value for the parameter vector</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.Fit_Canonical_HRF">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">Fit_Canonical_HRF</code><span class="sig-paren">(</span><em>tc</em>, <em>TR</em>, <em>Run</em>, <em>T</em>, <em>p</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.Fit_Canonical_HRF" title="Permalink to this definition">¶</a></dt>
<dd><p>Fits GLM using canonical hrf (with option of using time and dispersion derivatives)&#8217;;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">hrf</span><span class="p">,</span> <span class="n">fit</span><span class="p">,</span> <span class="n">e</span><span class="p">,</span> <span class="n">param</span><span class="p">,</span> <span class="n">info</span><span class="p">]</span> <span class="o">=</span> <span class="n">Fit_Canonical_HRF</span><span class="p">(</span><span class="n">tc</span><span class="p">,</span><span class="n">TR</span><span class="p">,</span><span class="n">Runs</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">p</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>tc:</strong></dt>
<dd><p class="first last">time course</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">time resolution</p>
</dd>
<dt><strong>Runs:</strong></dt>
<dd><p class="first last">expermental design</p>
</dd>
<dt><strong>T:</strong></dt>
<dd><p class="first last">length of estimated HRF</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">Model type</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Options:</th><td class="field-body"><ul class="first simple">
<li>p=1 - only canonical HRF</li>
<li>p=2 - canonical + temporal derivative</li>
<li>p=3 - canonical + time and dispersion derivative</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>hrf:</strong></dt>
<dd><p class="first last">estimated hemodynamic response function</p>
</dd>
<dt><strong>fit:</strong></dt>
<dd><p class="first last">estimated time course</p>
</dd>
<dt><strong>e:</strong></dt>
<dd><p class="first last">residual time course</p>
</dd>
<dt><strong>param:</strong></dt>
<dd><p class="first last">estimated amplitude, height and width</p>
</dd>
<dt><strong>info:</strong></dt>
<dd><p class="first last">struct containing design matrices, beta values etc</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.Fit_Logit2">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">Fit_Logit2</code><span class="sig-paren">(</span><em>tc</em>, <em>TR</em>, <em>Run</em>, <em>T</em>, <em>mode</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.Fit_Logit2" title="Permalink to this definition">¶</a></dt>
<dd><p>Fits FIR and smooth FIR model</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">hrf</span><span class="p">,</span> <span class="n">fit</span><span class="p">,</span> <span class="n">e</span><span class="p">,</span> <span class="n">param</span><span class="p">]</span> <span class="o">=</span> <span class="n">Fit_Logit</span><span class="p">(</span><span class="n">tc</span><span class="p">,</span><span class="n">Run</span><span class="p">,</span><span class="n">t</span><span class="p">,</span><span class="n">mode</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>tc:</strong></dt>
<dd><p class="first last">time course</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">time resolution</p>
</dd>
<dt><strong>Runs:</strong></dt>
<dd><p class="first last">expermental design</p>
</dd>
<dt><strong>T:</strong></dt>
<dd><p class="first last">length of estimated HRF</p>
</dd>
<dt><strong>mode:</strong></dt>
<dd><blockquote class="first">
<div><p>deterministic or stochastic</p>
<dl class="docutils">
<dt>Options:</dt>
<dd><p class="first">0 - deterministic aproach</p>
<p class="last">1 - simulated annealing approach</p>
</dd>
</dl>
</div></blockquote>
<p class="last">Please note that when using simulated annealing approach you
may need to perform some tuning before use.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>hrf:</strong></dt>
<dd><p class="first last">estimated hemodynamic response function</p>
</dd>
<dt><strong>fit:</strong></dt>
<dd><p class="first last">estimated time course</p>
</dd>
<dt><strong>e:</strong></dt>
<dd><p class="first last">residual time course</p>
</dd>
<dt><strong>param:</strong></dt>
<dd><p class="first last">estimated amplitude, height and width</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.Fit_sFIR">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">Fit_sFIR</code><span class="sig-paren">(</span><em>tc</em>, <em>TR</em>, <em>Run</em>, <em>T</em>, <em>mode</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.Fit_sFIR" title="Permalink to this definition">¶</a></dt>
<dd><p>Fits FIR and smooth FIR model</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">hrf</span><span class="p">,</span> <span class="n">fit</span><span class="p">,</span> <span class="n">e</span><span class="p">,</span> <span class="n">param</span><span class="p">]</span> <span class="o">=</span> <span class="n">Fit_sFIR</span><span class="p">(</span><span class="n">tc</span><span class="p">,</span><span class="n">TR</span><span class="p">,</span><span class="n">Runs</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">mode</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>tc:</strong></dt>
<dd><p class="first last">time course</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">time resolution</p>
</dd>
<dt><strong>Runs:</strong></dt>
<dd><p class="first last">expermental design</p>
</dd>
<dt><strong>T:</strong></dt>
<dd><p class="first last">length of estimated HRF</p>
</dd>
<dt><strong>mode:</strong></dt>
<dd><p class="first">FIR or smooth FIR</p>
<dl class="last docutils">
<dt>Options:</dt>
<dd><p class="first">0 - standard FIR</p>
<p class="last">1 - smooth FIR</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>hrf:</strong></dt>
<dd><p class="first last">estimated hemodynamic response function</p>
</dd>
<dt><strong>fit:</strong></dt>
<dd><p class="first last">estimated time course</p>
</dd>
<dt><strong>e:</strong></dt>
<dd><p class="first last">residual time course</p>
</dd>
<dt><strong>param:</strong></dt>
<dd><p class="first last">estimated amplitude, height and width</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.Get_Logit">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">Get_Logit</code><span class="sig-paren">(</span><em>V</em>, <em>t</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.Get_Logit" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate inverse logit (IL) HRF model
Creates fitted curve - 3 logistic functions to be summed together - from parameter estimates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">h</span><span class="p">]</span> <span class="o">=</span> <span class="n">get_logit</span><span class="p">(</span><span class="n">V</span><span class="p">,</span><span class="n">t</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>t:</strong></dt>
<dd><p class="first last">vector of time points</p>
</dd>
<dt><strong>V:</strong></dt>
<dd><p class="first last">parameters</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.HMHRFest">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">HMHRFest</code><span class="sig-paren">(</span><em>y</em>, <em>Runs</em>, <em>TR</em>, <em>nbasis</em>, <em>norder</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.HMHRFest" title="Permalink to this definition">¶</a></dt>
<dd><p>HRF estimation algorithm</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>y:</strong></dt>
<dd><p class="first last">Data matrix (#time points) by (#subjects) by (#voxels)</p>
</dd>
<dt><strong>Runs:</strong></dt>
<dd><p class="first last">Stick functions for each subject (#time points) by (#conditions) by (#subjects)</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">Time resolution</p>
</dd>
<dt><strong>nbasis:</strong></dt>
<dd><p class="first last">Number of b-spline basis</p>
</dd>
<dt><strong>norder:</strong></dt>
<dd><p class="first last">Order of b-spline basis</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.PowerLoss">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">PowerLoss</code><span class="sig-paren">(</span><em>modres</em>, <em>modfit</em>, <em>moddf</em>, <em>tc</em>, <em>TR</em>, <em>Run</em>, <em>alpha</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.PowerLoss" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimates Power-loss due to mis-modeling.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">PowLoss</span><span class="p">]</span> <span class="o">=</span> <span class="n">PowerLoss</span><span class="p">(</span><span class="n">modres</span><span class="p">,</span> <span class="n">modfit</span><span class="p">,</span> <span class="n">moddf</span><span class="p">,</span> <span class="n">tc</span><span class="p">,</span> <span class="n">TR</span><span class="p">,</span> <span class="n">Run</span><span class="p">,</span> <span class="n">alpha</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>modres:</strong></dt>
<dd><p class="first last">residuals</p>
</dd>
<dt><strong>modfit:</strong></dt>
<dd><p class="first last">model fit</p>
</dd>
<dt><strong>moddf:</strong></dt>
<dd><p class="first last">model degrees of freedom</p>
</dd>
<dt><strong>tc:</strong></dt>
<dd><p class="first last">time course</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">time resolution</p>
</dd>
<dt><strong>Runs:</strong></dt>
<dd><p class="first last">expermental design</p>
</dd>
<dt><strong>alpha:</strong></dt>
<dd><p class="first last">alpha value</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>PowLoss:</strong></dt>
<dd><p class="first last">Estimated power loss</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.ResidScan">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">ResidScan</code><span class="sig-paren">(</span><em>res</em>, <em>FWHM</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.ResidScan" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculates P(M&gt;=t) where M is the max value of the smoothed residuals.
In this implementation the residuals are smoothed using a Gaussian
kernel.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [p sres sres_ns] = ResidScan(res, FWHM)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>res:</strong></dt>
<dd><p class="first last">residual time course</p>
</dd>
<dt><strong>FWHM:</strong></dt>
<dd><p class="first last">Full Width Half Maximum (in time units)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>p:</strong></dt>
<dd><p class="first last">pvalues</p>
</dd>
<dt><strong>sres:</strong></dt>
<dd><p class="first last">smoothed residuals</p>
</dd>
<dt><strong>sres_ns:</strong></dt>
<dd><p class="first last">smoothed residuals (non standardized)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.get_parameters2">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">get_parameters2</code><span class="sig-paren">(</span><em>hdrf</em>, <em>t</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.get_parameters2" title="Permalink to this definition">¶</a></dt>
<dd><p>Find model parameters</p>
<p>Height - h</p>
<p>Time to peak - p (in time units of TR seconds)</p>
<p>Width (at half peak) - w</p>
<p>Calculate Heights and Time to peak:</p>
<p>delta = 1/(t(2)-t(1));</p>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.hrf_fit_one_voxel">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">hrf_fit_one_voxel</code><span class="sig-paren">(</span><em>tc</em>, <em>TR</em>, <em>Runc</em>, <em>T</em>, <em>method</em>, <em>mode</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.hrf_fit_one_voxel" title="Permalink to this definition">¶</a></dt>
<dd><p>HRF estimation function for a single voxel;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">h</span><span class="p">,</span> <span class="n">fit</span><span class="p">,</span> <span class="n">e</span><span class="p">,</span> <span class="n">param</span><span class="p">]</span> <span class="o">=</span>  <span class="n">hrf_fit_one_voxel</span><span class="p">(</span><span class="n">tc</span><span class="p">,</span><span class="n">TR</span><span class="p">,</span><span class="n">Runc</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="nb">type</span><span class="p">,</span><span class="n">mode</span><span class="p">)</span>
</pre></div>
</div>
<p>Implemented methods include: IL-model (Deterministic/Stochastic), FIR
(Regular/Smooth), and HRF (Canonical/+ temporal/+ temporal &amp; dispersion)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>tc:</strong></dt>
<dd><p class="first last">time course</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">time resolution</p>
</dd>
<dt><strong>Runs:</strong></dt>
<dd><p class="first last">expermental design</p>
</dd>
<dt><strong>T:</strong></dt>
<dd><p class="first last">length of estimated HRF</p>
</dd>
<dt><strong>type:</strong></dt>
<dd><p class="first last">Model type</p>
</dd>
<dt><strong>mode:</strong></dt>
<dd><p class="first last">Mode</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Options:</th><td class="field-body"><ul class="first simple">
<li>p=1 - only canonical HRF</li>
<li>p=2 - canonical + temporal derivative</li>
<li>p=3 - canonical + time and dispersion derivative</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>hrf:</strong></dt>
<dd><p class="first last">estimated hemodynamic response function</p>
</dd>
<dt><strong>fit:</strong></dt>
<dd><p class="first last">estimated time course</p>
</dd>
<dt><strong>e:</strong></dt>
<dd><p class="first last">residual time course</p>
</dd>
<dt><strong>param:</strong></dt>
<dd><p class="first last">estimated amplitude, height and width</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="HRF_Est_Toolbox2.ilogit">
<code class="descclassname">HRF_Est_Toolbox2.</code><code class="descname">ilogit</code><span class="sig-paren">(</span><em>t</em><span class="sig-paren">)</span><a class="headerlink" href="#HRF_Est_Toolbox2.ilogit" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the inverse logit function corresponding to the value t</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">L</span><span class="p">]</span> <span class="o">=</span> <span class="n">ilogit</span><span class="p">(</span><span class="n">t</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>L:</strong></dt>
<dd><p class="first last">exp(t)./(1+exp(t));</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="optimizedesign11">
<h2>OptimizeDesign11<a class="headerlink" href="#optimizedesign11" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-OptimizeDesign11"></span><dl class="function">
<dt id="OptimizeDesign11.optimizeGA">
<code class="descclassname">OptimizeDesign11.</code><code class="descname">optimizeGA</code><span class="sig-paren">(</span><em>GA</em><span class="sig-paren">)</span><a class="headerlink" href="#OptimizeDesign11.optimizeGA" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">M</span> <span class="o">=</span> <span class="n">optimizeGA</span><span class="p">(</span><span class="n">GA</span><span class="p">)</span>
</pre></div>
</div>
<p>outputs a pseudo-random list of condition codes that optimizes
multiple fitness measures for fMRI task designs</p>
<p>more help can be found in ga_example_script.m
and in Genetic_Algorithm_readme.rtf</p>
</dd></dl>

<dl class="function">
<dt id="OptimizeDesign11.optimizeGA_epochs">
<code class="descclassname">OptimizeDesign11.</code><code class="descname">optimizeGA_epochs</code><span class="sig-paren">(</span><em>GA</em><span class="sig-paren">)</span><a class="headerlink" href="#OptimizeDesign11.optimizeGA_epochs" title="Permalink to this definition">¶</a></dt>
<dd><p>outputs a random-ordered list of condition #s that optimizes 3 fMRI considerations</p>
<dl class="docutils">
<dt>Ways to avoid block designs</dt>
<dd>counterbalancing factor
power lower limit cutoff pushes power higher</dd>
<dt>Why using avg power is better than efficiency</dt>
<dd>efficiency doesn&#8217;t account for 1/f model (altho here we just use a cutoff, the same thing)
avoid transformation errors in using high pass filter
efficiency is based on the sample size, determined by TR, of the model - but so is fft power...</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="OptimizeDesign11.optimize_rand_search">
<code class="descclassname">OptimizeDesign11.</code><code class="descname">optimize_rand_search</code><span class="sig-paren">(</span><em>GA</em><span class="sig-paren">)</span><a class="headerlink" href="#OptimizeDesign11.optimize_rand_search" title="Permalink to this definition">¶</a></dt>
<dd><p>Just like the GA, but generates random designs each time!</p>
<p>outputs a random-ordered list of condition #s that optimizes 3 fMRI considerations%</p>
<dl class="docutils">
<dt>Ways to avoid block designs</dt>
<dd>counterbalancing factor
power lower limit cutoff pushes power higher</dd>
<dt>Why using avg power is better than efficiency</dt>
<dd>efficiency doesn&#8217;t account for 1/f model (altho here we just use a cutoff, the same thing)
avoid transformation errors in using high pass filter
efficiency is based on the sample size, determined by TR, of the model - but so is fft power...</dd>
</dl>
</dd></dl>

</div>
</div>
<div class="section" id="miscellaneous-tools">
<h1>Miscellaneous Tools<a class="headerlink" href="#miscellaneous-tools" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-Data_extraction">
<span id="data-extraction"></span><h2>Data_extraction<a class="headerlink" href="#module-Data_extraction" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="Data_extraction.canlab_maskstats">
<code class="descclassname">Data_extraction.</code><code class="descname">canlab_maskstats</code><span class="sig-paren">(</span><em>msks</em>, <em>imgs</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.canlab_maskstats" title="Permalink to this definition">¶</a></dt>
<dd><p>Produces comparison of pattern mask and images
e.g., look at NPS pattern expression in set of beta images</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">MASKSTATS</span> <span class="o">=</span> <span class="n">canlab_maskstats</span><span class="p">(</span><span class="n">maskfiles</span><span class="p">,</span><span class="n">imgfiles</span><span class="p">,[</span><span class="n">options</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>maskfiles:</strong></dt>
<dd><p class="first last">string or cellstring of mask filenames or fmri_data object</p>
</dd>
<dt><strong>imgfiles:</strong></dt>
<dd><p class="first last">string or cellstring of image filenames or fmri_data object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>ts:</strong></dt>
<dd><p class="first">timeseries treatment: each string in imgfiles is assumed to be
a 4D file. Data will be returned with one column per time series and
one volume per row. If not all timeseries are same length, all will
be NaN-padded to the length of the longest timeseries.</p>
<p class="last">Note: does not work with imgfiles input as fmri_data object</p>
</dd>
<dt><strong>dir2cell:</strong></dt>
<dd><p class="first">will sort stats into cells based on directory containing the imgfile
they belong to such that each cell contains one directory&#8217;s worth of
stats which is a vector with a value for each imgfile.
:Examples:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="c"># Input: a list of single trial betas for a set of subjects</span>
<span class="n">b</span> <span class="o">=</span> <span class="n">filenames</span><span class="p">(</span><span class="s">&#39;sub*/*heat_trials*.img&#39;</span><span class="p">);</span>
<span class="n">ms</span> <span class="o">=</span> <span class="n">canlab_maskstats</span><span class="p">(</span><span class="s">&#39;nps&#39;</span><span class="p">,</span><span class="n">b</span><span class="p">,</span><span class="s">&#39;dot_product&#39;</span><span class="p">,</span><span class="s">&#39;dir2cell&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p class="last">Output: includes the set of cells that are input to a mediation
analysis.</p>
</dd>
<dt><strong>keepzeros:</strong></dt>
<dd><p class="first last">don&#8217;t remove zeros from imgfiles before taking measurements</p>
</dd>
<dt><strong>keepzerosmask:</strong></dt>
<dd><p class="first last">don&#8217;t remove zeros from maskfiles before taking measurements</p>
</dd>
<dt><strong>single:</strong></dt>
<dd><p class="first last">leave data as single (DEFAULT: convert to double)</p>
</dd>
<dt><strong>trinarize:</strong></dt>
<dd><p class="first last">trinarize maskfile (set values larger than 0 to 1 and values less than
zero to -1)</p>
</dd>
<dt><strong>noreshape:</strong></dt>
<dd><p class="first last">don&#8217;t attempt to reshape results according to imgfiles array</p>
</dd>
<dt><strong>nobin:</strong></dt>
<dd><p class="first last">don&#8217;t binarize mask before extracting mean or std</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body"><p class="first last">ts, dir2cell, and noreshape are mutually exclusive options</p>
</td>
</tr>
</tbody>
</table>
<p><strong>Built-In Masks</strong>
The following strings can be given as the maskfile argument to
call up built-in mask files:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>nps:</strong></dt>
<dd>weights_NSF_grouppred_cvpcr.img</dd>
<dt><strong>nps_thresh:</strong></dt>
<dd>weights_NSF_grouppred_cvpcr_FDR05.img</dd>
<dt><strong>nps_thresh_smooth:</strong></dt>
<dd>weights_NSF_grouppred_cvpcr_FDR05_smoothed_fwhm05.img</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Measure Options:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>all:</strong></dt>
<dd><dl class="first last docutils">
<dt>add: mean, dot_product, centered_dot_product,</dt>
<dd><p class="first last">cosine_similarity, and correlation</p>
</dd>
</dl>
</dd>
<dt><strong>mean:</strong> (DEFAULT)</dt>
<dd><p class="first last">apply binarized mask to images and return means
mean(img .* abs(bin(mask)))</p>
</dd>
<dt><strong>std:</strong></dt>
<dd><p class="first last">apply binarized mask to images and return standard deviations
std(img .* abs(bin(mask)))</p>
</dd>
<dt><strong>dot_product:</strong></dt>
<dd><p class="first last">dot(mask, img)</p>
</dd>
<dt><strong>cosine_similarity:</strong></dt>
<dd><p class="first last">dot(mask, img) / (norm(mask) * norm(img))</p>
</dd>
<dt><strong>correlation:</strong></dt>
<dd><p class="first last">corr(mask, img)</p>
</dd>
<dt><strong>centered_dot_product:</strong></dt>
<dd><p class="first last">dot(mask-mean(mask), img-mean(img))</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Details:</th><td class="field-body"><ul class="first last simple">
<li>imgfiles are spatially resampled to maskfiles</li>
<li>voxels with zeros in maskfile are removed</li>
<li>in-mask voxels with zeros in imgfiles will generate warnings</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.cluster_tmask">
<code class="descclassname">Data_extraction.</code><code class="descname">cluster_tmask</code><span class="sig-paren">(</span><em>cl</em>, <em>tm</em>, <em>si</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.cluster_tmask" title="Permalink to this definition">¶</a></dt>
<dd><p>Given clusters and a string name of a t-image,  finds voxels that exceed a
specified t-threshold</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cl</span><span class="p">,</span> <span class="n">varargout</span><span class="p">]</span> <span class="o">=</span> <span class="n">cluster_tmask</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span> <span class="n">tm</span><span class="p">,</span> <span class="n">si</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>cl:</strong></dt>
<dd><p class="first last">clusters</p>
</dd>
<dt><strong>tm:</strong></dt>
<dd><p class="first last">t-image</p>
</dd>
<dt><strong>si:</strong></dt>
<dd><p class="first last">subject index integer</p>
</dd>
<dt><strong>[dat]:</strong></dt>
<dd><p class="first last">cluster_barplot data structure</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>creates new XYZ in the space of t-image using cl.XYZmm coordinates in mm.
Required fields of cl:  XYZmm</p>
<dl class="docutils">
<dt>Calculates and saves single-subject data avgd over voxels if:</dt>
<dd><ol class="first last upperalpha">
<li><dl class="first docutils">
<dt>cl.all_data field is present</dt>
<dd><p class="first last">THIS WORKS if all_data has individual subject contrast estimates
in it,  with rows as subjects and columns as voxels
indiv data saved in cl(region).timeseries(subject)</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>cl.raw_data is present</dt>
<dd><p class="first last">raw_data should be time x voxels x subjects,  a 3D matrix
see output of extract_raw_data.
indiv data saved in cl(region).indiv_timeseries(:, subject)</p>
</dd>
</dl>
</li>
</ol>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body">Retains upper 50% of voxels; highest t-values</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.extract_contrast_data">
<code class="descclassname">Data_extraction.</code><code class="descname">extract_contrast_data</code><span class="sig-paren">(</span><em>P</em>, <em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.extract_contrast_data" title="Permalink to this definition">¶</a></dt>
<dd><p>This function does not plot,  but separates subclusters using pca / cluster_princomp.m
based on pattern across all conditions,  covariance (not correlation),</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">clusters</span><span class="p">,</span> <span class="n">subcl</span><span class="p">]</span> <span class="o">=</span> <span class="n">extract_contrast_data</span><span class="p">(</span><span class="n">P</span><span class="p">,</span> <span class="n">clusters</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>P:</strong></dt>
<dd><p class="first last">cell array of strings containing image file names (data extracted from these)</p>
</dd>
</dl>
<p><strong>clusters:</strong></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>subclusters:</strong></dt>
<dd><p class="first last">to get sub-clustering based on pca and clustering of voxels
add the string &#8216;subclusters&#8217; as an input argument</p>
</dd>
</dl>
<p><strong>cell array:</strong> of strings with condition names</p>
<dl class="docutils">
<dt><strong>split:</strong></dt>
<dd><p class="first">value is 1: to split into 2 plots (first half and last half of P)</p>
<p class="last">value is 2: to plot individual subjects over bars</p>
</dd>
<dt><strong>center:</strong></dt>
<dd><p class="first last">center parmeter values in plot (subtract row means)
this gives closer to correct &#8220;within subjects&#8221; error bars
and may be used when the overall parameter values have no meaning</p>
</dd>
<dt><strong>covs:</strong></dt>
<dd><p class="first last">followed by between-subject covariates (e.g.,  behavioral regressors)
plots remove these before plotting means and std. errors</p>
</dd>
<dt><strong>max:</strong></dt>
<dd><p class="first last">to make plots based on max z-values within region for each dataset P
not compatible with &#8216;split&#8217; and &#8216;center&#8217; (ignores these commands)
right now,  special for inhib - see also inhib2_cluster_barplot (good function)</p>
</dd>
<dt><strong>indiv:</strong></dt>
<dd><p class="first">to threshold based on individual t-statistics or contrast values</p>
<p class="last">FOLLOW with cell array of t-images or con images &#8211; usually,  there will be one cell,  with images 
for each subject in rows,  to define voxels for each ss.
BUT Tnames can be the same length as
contrast images,  one t-img per subject per contrast,  if
desired.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first last">clusters struture,  with CONTRAST substructure added
substructure contains data extracted and image file names</p>
</td>
</tr>
</tbody>
</table>
<p>This program uses XYZmm millimeter coordinates in clusters to find voxels
So clusters and data files may have different dimensions.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cl = extract_contrast_data(EXPT.SNPM.P, cl, &#39;indiv&#39;, EXPT.FILES.Timgs{1});
cluster_barplot(EXPT.SNPM.P(7:12), clusters(2:3), {&#39;ObjE&#39; &#39;AttE&#39; &#39;InteractE&#39; &#39;ObjI&#39; &#39;AttI&#39; &#39;InteractI&#39;}, &#39;split&#39;)
[clusters, subclusters] = cluster_barplot(EXPT.SNPM.P(17:24), clusters, &#39;subclusters&#39;, &#39;split&#39;)
RS2_8vs2_placeboCP = cluster_barplot(EXPT.SNPM.P([8 10 12 14 16]), RS2meta, &#39;indiv&#39;, T);
</pre></div>
</div>
<p>also see mask2clusters.m,  a simpler version that just extracts clusters from a mask file.</p>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.extract_from_rois">
<code class="descclassname">Data_extraction.</code><code class="descname">extract_from_rois</code><span class="sig-paren">(</span><em>imgs_to_extract_from</em>, <em>mask_image</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.extract_from_rois" title="Permalink to this definition">¶</a></dt>
<dd><p>Generic function for extracting image data from a mask or atlas image,
and returning the data and averages within regions specified by the user.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>  [cl, imgdat] = extract_from_rois(imgs_to_extract_from, mask_image, varargin)

Regions to average over can be either regions of contiguous voxels
bounded by voxels with values of 0 or NaN, which are considered non-data
values, or regions defined by unique integer codes in the mask image
(i.e., for atlas images with unique codes for each defined region.)

Mask/Atlas image does NOT have to be in the same space as the images to
extract from.  It will be remapped/resliced.

extracted data is returned in single data format.
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first"><strong>char array</strong> of strings containing 4D image file names (data extracted from these)</p>
<p><strong>mask_image</strong> to extract from</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>average_over:</strong></dt>
<dd><p class="first">Default = &#8216;unique_mask_values&#8217; to average over unique integer codes in the mask image
(i.e., for atlas images with unique codes for each defined region)</p>
<p class="last">OPT = &#8216;contiguous_regions&#8217; to average over contiguous voxels
bounded by voxels of 0 or NaN (non-data values)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">imgs_to_extract_from</span> <span class="o">=</span> <span class="n">filenames</span><span class="p">(</span><span class="s">&#39;w*.nii&#39;</span><span class="p">,</span><span class="s">&#39;char&#39;</span><span class="p">);</span>
<span class="n">mask_image</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="s">&#39;anat_lbpa_thal.img&#39;</span><span class="p">);</span>
<span class="p">[</span><span class="n">cl</span><span class="p">,</span> <span class="n">imgdat</span><span class="p">]</span> <span class="o">=</span> <span class="n">extract_from_rois</span><span class="p">(</span><span class="n">imgs_to_extract_from</span><span class="p">,</span> <span class="n">mask_image</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.extract_image_data">
<code class="descclassname">Data_extraction.</code><code class="descname">extract_image_data</code><span class="sig-paren">(</span><em>imgs_to_extract_from</em>, <em>mask_image</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.extract_image_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Generic function for extracting image data from a mask or atlas image,
and returning the data and averages within regions specified by the user.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">imgdat</span><span class="p">,</span> <span class="n">volInfo</span><span class="p">,</span> <span class="n">cl</span><span class="p">]</span> <span class="o">=</span> <span class="n">extract_image_data</span><span class="p">(</span><span class="n">imgs_to_extract_from</span><span class="p">,</span> <span class="n">mask_image</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Regions to average over can be either regions of contiguous voxels
bounded by voxels with values of 0 or NaN, which are considered non-data
values, or regions defined by unique integer codes in the mask image
(i.e., for atlas images with unique codes for each defined region.)</p>
<p>Mask/Atlas image does NOT have to be in the same space as the images to
extract from.  It will be remapped/resliced.</p>
<p>extracted data is returned in single data format.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first"><strong>char array</strong> of strings containing 4D image file names (data extracted from these)</p>
<p><strong>mask_image</strong> to extract from</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>average_over:</strong></dt>
<dd><p class="first">Default = &#8216;contiguous_regions&#8217; to average over contiguous voxels
bounded by voxels of 0 or NaN (non-data values)</p>
<p class="last">Alt. option = &#8216;unique_mask_values&#8217; to average over unique integer codes in the mask image
(i.e., for atlas images with unique codes for each defined
region)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">imgs_to_extract_from</span> <span class="o">=</span> <span class="n">filenames</span><span class="p">(</span><span class="s">&#39;w*.nii&#39;</span><span class="p">,</span><span class="s">&#39;char&#39;</span><span class="p">);</span>
<span class="n">mask_image</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="s">&#39;anat_lbpa_thal.img&#39;</span><span class="p">);</span>
<span class="p">[</span><span class="n">imgdat</span><span class="p">,</span> <span class="n">volInfo</span><span class="p">,</span> <span class="n">cl</span><span class="p">]</span> <span class="o">=</span> <span class="n">extract_image_data</span><span class="p">(</span><span class="n">imgs_to_extract_from</span><span class="p">,</span> <span class="n">mask_image</span><span class="p">,</span> <span class="s">&#39;unique_mask_values&#39;</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Related functions:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>For an object-oriented alternative, see the fmri_data class and extract_roi_averages method</p>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.extract_indiv_peak_data">
<code class="descclassname">Data_extraction.</code><code class="descname">extract_indiv_peak_data</code><span class="sig-paren">(</span><em>cl</em>, <em>imgs</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.extract_indiv_peak_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Purpose: to find individually significant regions within each subject
and save average timecourses for each individual ROI for each subject</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">extract_indiv_peak_data</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">imgs</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>cl:</strong></dt>
<dd><p class="first last">is a clusters structure with one element per region.  Each element
(cluster) is a structure containing coordinates and data.</p>
</dd>
<dt><strong>imgs:</strong></dt>
<dd><p class="first last">is a string matrix with one row per subject, with names of images
used to define thresholds.  These may be contrast or t-images from
individual subjects</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>The method of extraction is defined in cluster_tmask, which is currently
to use 50% of voxels with the highest values in imgs(subject) for each
subject, or 100% if 50% returns less than 5 voxels.  
It&#8217;s easy in cluster_tmask to use absolute t-thresholds instead.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Required fields of cl:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>XYZmm:</strong></dt>
<dd><p class="first last">3 x k list of k coordinates in mm space</p>
</dd>
<dt><strong>raw_data:</strong></dt>
<dd><p class="first last">time x voxels x subjects matrix of data for this region</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Outputs appended to cl structure:
indiv_timeseries, time x subjects averaged individual ROI timecourses</p>
<dl class="docutils">
<dt>:.INDIV:, a structure with the following information:</dt>
<dd><ul class="first last simple">
<li>tname, the name of the t-mask (or contrast mask) entered</li>
<li>XYZ, voxel coords for significant voxels for each subject</li>
<li>XYZmm, mm coords for sig voxels for each subject</li>
<li>sigt, logical matrix subjects x voxels for sig voxels</li>
<li>maxt, max map value for each subject</li>
<li>center, average coordinate for sig voxels for each subject</li>
<li>mm_center, the same in mm</li>
</ul>
</dd>
</dl>
<p>The spatial information may be used to correlate peak voxel location with
behavior, for example.</p>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.read_hdr">
<code class="descclassname">Data_extraction.</code><code class="descname">read_hdr</code><span class="sig-paren">(</span><em>name</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.read_hdr" title="Permalink to this definition">¶</a></dt>
<dd><p>Loads the analyze format header file from a file &#8216;name&#8217;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function hdr = read_hdr(name,[opt] datatype)
</pre></div>
</div>
<p>The function returns a structure defined as</p>
<dl class="docutils">
<dt>hdr = struct(...</dt>
<dd><p class="first">&#8216;sizeof_hdr&#8217;, fread(pFile, 1,&#8217;int32&#8217;),...</p>
<p>&#8216;pad1&#8217;, setstr(fread(pFile, 28, &#8216;char&#8217;)),...</p>
<p>&#8216;extents&#8217;, fread(pFile, 1,&#8217;int32&#8217;),...</p>
<p>&#8216;pad2&#8217;, setstr(fread(pFile, 2, &#8216;char&#8217;)),...</p>
<p>&#8216;regular&#8217;,setstr(fread(pFile, 1,&#8217;char&#8217;)), ...</p>
<p>&#8216;pad3&#8217;, setstr(fread(pFile,1, &#8216;char&#8217;)),...</p>
<p>&#8216;dims&#8217;, fread(pFile, 1,&#8217;int16&#8217;),...</p>
<p>&#8216;xdim&#8217;, fread(pFile, 1,&#8217;int16&#8217;),...</p>
<p>&#8216;ydim&#8217;, fread(pFile, 1,&#8217;int16&#8217;),...</p>
<p>&#8216;zdim&#8217;, fread(pFile, 1,&#8217;int16&#8217;),...</p>
<p>&#8216;tdim&#8217;, fread(pFile, 1,&#8217;int16&#8217;),...</p>
<p>&#8216;pad4&#8217;, setstr(fread(pFile,20, &#8216;char&#8217;)),...</p>
<p>&#8216;datatype&#8217;, fread(pFile, 1,&#8217;int16&#8217;),...</p>
<p>&#8216;bits&#8217;, fread(pFile, 1,&#8217;int16&#8217;),...</p>
<p>&#8216;pad5&#8217;, setstr(fread(pFile, 6, &#8216;char&#8217;)),...</p>
<p>&#8216;xsize&#8217;, fread(pFile, 1,&#8217;float&#8217;),...</p>
<p>&#8216;ysize&#8217;, fread(pFile, 1,&#8217;float&#8217;),...</p>
<p>&#8216;zsize&#8217;, fread(pFile, 1,&#8217;float&#8217;),...</p>
<p>&#8216;pad6&#8217;, setstr(fread(pFile, 48, &#8216;char&#8217;))...</p>
<p>&#8216;glmax&#8217;, fread(pFile, 1,&#8217;int32&#8217;),...</p>
<p>&#8216;glmin&#8217;, fread(pFile, 1,&#8217;int32&#8217;),... 
&#8216;descrip&#8217;, setstr(fread(pFile, 80,&#8217;char&#8217;)),...</p>
<p>&#8216;aux_file&#8217;        , setstr(fread(pFile,24,&#8217;char&#8217;))&#8217;,...</p>
<p>&#8216;orient&#8217;          , fread(pFile,1,&#8217;char&#8217;),...</p>
<p>&#8216;origin&#8217;          , fread(pFile,5,&#8217;int16&#8217;),...</p>
<p>&#8216;generated&#8217;       , setstr(fread(pFile,10,&#8217;char&#8217;))&#8217;,...</p>
<p>&#8216;scannum&#8217;         , setstr(fread(pFile,10,&#8217;char&#8217;))&#8217;,...</p>
<p>&#8216;patient_id&#8217;      , setstr(fread(pFile,10,&#8217;char&#8217;))&#8217;,...</p>
<p>&#8216;exp_date&#8217;        , setstr(fread(pFile,10,&#8217;char&#8217;))&#8217;,...</p>
<p>&#8216;exp_time&#8217;        , setstr(fread(pFile,10,&#8217;char&#8217;))&#8217;,...</p>
<p>&#8216;hist_un0&#8217;        , setstr(fread(pFile,3,&#8217;char&#8217;))&#8217;,...</p>
<p>&#8216;views&#8217;           , fread(pFile,1,&#8217;int32&#8217;),...</p>
<p>&#8216;vols_added&#8217;      , fread(pFile,1,&#8217;int32&#8217;),...</p>
<p>&#8216;start_field&#8217;     , fread(pFile,1,&#8217;int32&#8217;),...</p>
<p>&#8216;field_skip&#8217;      , fread(pFile,1,&#8217;int32&#8217;),...</p>
<p>&#8216;omax&#8217;            , fread(pFile,1,&#8217;int32&#8217;),...</p>
<p>&#8216;omin&#8217;            , fread(pFile,1,&#8217;int32&#8217;),...</p>
<p>&#8216;smax&#8217;            , fread(pFile,1,&#8217;int32&#8217;),...</p>
<p class="last">&#8216;smin&#8217;            , fread(pFile,1,&#8217;int32&#8217;) );</p>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.readim2">
<code class="descclassname">Data_extraction.</code><code class="descname">readim2</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.readim2" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">array</span><span class="p">,</span><span class="n">hdr</span><span class="p">,</span><span class="n">h</span><span class="p">,</span><span class="n">whichslices</span><span class="p">,</span><span class="n">rows</span><span class="p">,</span><span class="n">cols</span><span class="p">,</span><span class="n">figh</span><span class="p">]</span> <span class="o">=</span> <span class="n">readim2</span><span class="p">(</span><span class="n">basename</span> <span class="ow">or</span> <span class="n">array</span> <span class="p">[</span><span class="n">opt</span><span class="p">],</span><span class="s">&#39;p&#39;</span> <span class="p">[</span><span class="n">opt</span><span class="p">],</span> <span class="s">&#39;sagg&#39;</span> <span class="ow">or</span> <span class="s">&#39;cor&#39;</span> <span class="p">[</span><span class="n">opt</span><span class="p">],</span><span class="n">flipy</span><span class="p">[</span><span class="n">opy</span><span class="p">],</span><span class="nb">range</span> <span class="p">[</span><span class="n">opt</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><ul class="first simple">
<li>basename of file, without image extension ,OR</li>
<li>3-D array in the workspace to plot, OR</li>
<li>nothing, to browse for file</li>
</ul>
<dl class="docutils">
<dt><strong>p:</strong></dt>
<dd><p class="first last">to plot montage of slices to the screen</p>
</dd>
<dt><strong>sagg:</strong></dt>
<dd><p class="first last">to rotate to sagittal view</p>
</dd>
<dt><strong>cor:</strong></dt>
<dd><p class="first last">to rotate to coronal view</p>
</dd>
<dt><strong>t:</strong></dt>
<dd><p class="first last">to save array as double instead of int16 - to save negative t values.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><ul class="first simple">
<li>3d array of image</li>
<li>hdr of image</li>
<li>handles for axes of montage if plotting</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Special Features:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>range:</strong></dt>
<dd><dl class="first last docutils">
<dt>specified in mm, must be LAST and FIFTH input argument. </dt>
<dd><ul class="first last simple">
<li>OR range can specify slices, e.g. 1:4:28</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>clim:</strong></dt>
<dd><p class="first last">color limits for axis plot, must be 1st or 2nd argument.  form: [-1 1]</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.timeseries_extract_slice">
<code class="descclassname">Data_extraction.</code><code class="descname">timeseries_extract_slice</code><span class="sig-paren">(</span><em>V</em>, <em>sliceno</em>, <em>orientation</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.timeseries_extract_slice" title="Permalink to this definition">¶</a></dt>
<dd><p>For a given set of image names or memory mapped volumes (V)
extracts data from slice # sliceno and returns an X x Y x time
matrix of data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function sl = timeseries_extract_slice(V,sliceno)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_extraction.tor_extract_rois">
<code class="descclassname">Data_extraction.</code><code class="descname">tor_extract_rois</code><span class="sig-paren">(</span><em>imnames</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_extraction.tor_extract_rois" title="Permalink to this definition">¶</a></dt>
<dd><p>This function gets timeseries data from all clusters in an SPM results output.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [clusters, SPM, xX, xCon] = tor_extract_rois(imnames [can be empty],[opt] SPM, [opt] VOL, [opt] xX)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>imnames:</strong></dt>
<dd><p class="first last">a matrix of image names, in spm_list_files output format
if empty, no timeseries data will be extracted.</p>
</dd>
<dt><strong>clusters:</strong></dt>
<dd><p class="first last">if only 2 arguments, clusters structure is 2nd arg, and we
extract data using existing clusters structure</p>
</dd>
</dl>
<p>If 3 arguments, enter SPM and VOL to extract data from VOXEL
coordinates in these structures
- SPM:        SPM variable from loaded results
- VOL:        VOL variable from loaded results</p>
<p class="last">Optional 4th argument fits a design matrix and returns betas
- xX:         xX design matrix to fit to timeseries
OR 4th argument can be 0 (or any non-structure arg), suppressing verbose output.</p>
</td>
</tr>
</tbody>
</table>
<p>[Last 2 arguments are optional.  Use if results are already loaded into workspace]</p>
<p>Automatic fitting of model to cluster timeseries average using analyze_cluster_rois
with High-Pass filter length of your choice.
This only works if you input only the file names or input all optional arguments, including xX</p>
<p>NOTE (WARNING): WORKS ON XYZ VOXEL COORDINATES - TRANSFORMATION TO
DIFFERENT SPACES ONLY IF ENTERING 2 ARGS, 1st one names, 2nd one clusters</p>
<p>see transform_coordinates.m for transformation, or check_spm_mat, or cluster_interp.</p>
</dd></dl>

</div>
<div class="section" id="data-processing-tools">
<h2>Data_processing_tools<a class="headerlink" href="#data-processing-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Data_processing_tools"></span><dl class="function">
<dt id="Data_processing_tools.center_of_mass">
<code class="descclassname">Data_processing_tools.</code><code class="descname">center_of_mass</code><span class="sig-paren">(</span><em>XYZ</em>, <em>Z</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.center_of_mass" title="Permalink to this definition">¶</a></dt>
<dd><p>This function returns the center of mass of a cluster of voxels or mm coordinates
defined as the nearest in-list coordinate to the average of the 
coordinate values weighted by the Z-score</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">com</span> <span class="o">=</span> <span class="n">center_of_mass</span><span class="p">(</span><span class="n">XYZ</span><span class="p">,</span><span class="n">Z</span><span class="p">)</span>
</pre></div>
</div>
<p>assigns a rank to each coordinate based on Z scores
and includes</p>
<p>enter a 3 x n list of XYZ coordinates
returns 1 x 3 center of mass</p>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.detransition">
<code class="descclassname">Data_processing_tools.</code><code class="descname">detransition</code><span class="sig-paren">(</span><em>y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.detransition" title="Permalink to this definition">¶</a></dt>
<dd><p>For fMRI timeseries that contains large &#8216;jump&#8217; artifacts due to motion correction
or other problems.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">y</span> <span class="o">=</span> <span class="n">detransition</span><span class="p">(</span><span class="n">y</span><span class="p">,[</span><span class="n">doplot</span><span class="p">])</span>
</pre></div>
</div>
<p>Removes these large spikes.
Updated version built into spikecorrect in trimts</p>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.downsample_scnlab">
<code class="descclassname">Data_processing_tools.</code><code class="descname">downsample_scnlab</code><span class="sig-paren">(</span><em>y</em>, <em>orig_samprate</em>, <em>new_samprate</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.downsample_scnlab" title="Permalink to this definition">¶</a></dt>
<dd><p>Uses linear interpolation to resample a vector from one sampling
rate to another</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">yi</span><span class="p">,</span> <span class="n">xi</span><span class="p">]</span> <span class="o">=</span> <span class="n">downsample_scnlab</span><span class="p">(</span><span class="n">y</span><span class="p">,</span> <span class="n">orig_samprate</span><span class="p">,</span> <span class="n">new_samprate</span><span class="p">,</span> <span class="p">[</span><span class="n">doplot</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>orig_samprate:</strong></dt>
<dd><p class="first last">sampling rate in Hz</p>
</dd>
<dt><strong>new_samprate:</strong></dt>
<dd><p class="first last">desired sampling rate in Hz</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Downsample a 100 Hz signal to a scanning TR of 2 sec
% signal at 100 Hz, sample to low-freq TR of 0.5 hz (2 sec TR)
% every 100 / TR = 100/.5 = 200 samples

[yi, xi] = downsample_scnlab(y, 100, .5)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.fft_plot_scnlab">
<code class="descclassname">Data_processing_tools.</code><code class="descname">fft_plot_scnlab</code><span class="sig-paren">(</span><em>dat</em>, <em>TR</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.fft_plot_scnlab" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">myfft</span><span class="p">,</span> <span class="n">freq</span><span class="p">,</span> <span class="n">handle</span><span class="p">]</span> <span class="o">=</span> <span class="n">fft_plot_scnlab</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">TR</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">is a data vector (column)</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">is the sampling rate of the data you put in
in seconds / sample, or 1/Hz</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><ul class="first simple">
<li>&#8216;samefig&#8217;</li>
<li>&#8216;color, [&#8216;b&#8217;] or other color</li>
<li>&#8216;bar&#8217;</li>
<li>&#8216;linebar&#8217;: both line and bar</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% plot effects of filtering on a difference
% between two regressors
spm_hplength = SPM.xX.K.HParam;
d = SPM.xX.X * SPM.xCon(mycon).c(:, 1);
create_figure(&#39;Contrast&#39;); plot(d) % contrast we care about
px = pinv(SPM.xX.K.X0);           % pinv of the filtering matrix
y = d;
y = y - SPM.xX.K.X0 * px * y;     % residuals after filtering

[myfft, freq, handle] = fft_plot_scnlab(d, 2);
hold on;
[myfft2, freq2, handle] = fft_plot_scnlab(y, 2); set(handle,&#39;Color&#39;,&#39;r&#39;)
plot_vertical_line(1/spm_hplength)
set(ans, &#39;Color&#39;, &#39;b&#39;, &#39;LineWidth&#39;, 3)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.filterAdjust">
<code class="descclassname">Data_processing_tools.</code><code class="descname">filterAdjust</code><span class="sig-paren">(</span><em>O</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.filterAdjust" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">y</span><span class="p">,</span><span class="n">O</span><span class="p">,</span><span class="n">X</span><span class="p">,</span><span class="n">S</span><span class="p">]</span> <span class="o">=</span> <span class="n">filterAdjust</span><span class="p">(</span><span class="n">OPTIONS</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">O:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>O.y:</strong></dt>
<dd><p class="first last">signal</p>
</dd>
<dt><strong>O.HP:</strong></dt>
<dd><p class="first last">high pass freq. cutoff</p>
</dd>
<dt><strong>O.TR:</strong></dt>
<dd><p class="first last">sampling rate in s</p>
</dd>
<dt><strong>O.doHP:</strong></dt>
<dd><p class="first last">[0 or 1] - do HP filter (spm), default is 0</p>
</dd>
<dt><strong>O.doLP:</strong></dt>
<dd><dl class="first last docutils">
<dt>[0, 1, 2] - do LP filter (spm), default is 0</dt>
<dd><p class="first last">2 = Gaussian filter with TR*2 s length</p>
</dd>
</dl>
</dd>
<dt><strong>O.firstimg:</strong></dt>
<dd><p class="first last">sets values for first image in each run to mean of the
remaining values.  Good for removing first-image artifacts present in
some scanner sequences.  Default is 1.</p>
</dd>
<dt><strong>O.cyclecorrection:</strong></dt>
<dd><p class="first last">checks for unimodal (normally distributed) data
within each session, because some bimodal data that cycles between two
mean scanner values has been observed in some data.  if a high proportion
of outliers are found in non-normal data, subtracts mean of higher mode
to adjust data. Sorry-not clear.  Check the code. Default is 0</p>
</dd>
<dt><strong>O.cyclecorrection2:</strong></dt>
<dd><p class="first">removes large transitions from data, as in
detransition.m.  Default is 0.  Artifacts may be acquisition or
motion-correction/resampling related.</p>
<p class="last">We do this after filtering, but if there&#8217;s trouble, we re-do the
scanadjust and filtering, because &#8216;cycling&#8217; can affect these.</p>
</dd>
<dt><strong>O.scanadjust:</strong></dt>
<dd><dl class="first last docutils">
<dt>[0 or 1] - adjust to scan means, default is 0</dt>
<dd><ul class="first simple">
<li>If O.X is entered, assumes this is the session mean matrix,</li>
</ul>
<p class="last">instead of recomputing.</p>
</dd>
</dl>
</dd>
<dt><strong>O.percent:</strong></dt>
<dd><p class="first last">[0 or 1] - adjust to percent change from baseline, default is 0</p>
</dd>
<dt><strong>O.filtertype:</strong></dt>
<dd><dl class="first last docutils">
<dt>filter style, default is &#8216;none&#8217;</dt>
<dd><ul class="first last simple">
<li>&#8216;spm&#8217;, use spm&#8217;s filtering</li>
<li>if O.S is entered, uses this instead of recomputing</li>
<li>&#8216;fourier&#8217;, Doug&#8217;s fourier filter</li>
<li>&#8216;fouriernotch&#8217;, Omit frequencies between HP(1) and HP(2)</li>
<li>&#8216;cheby&#8217;, chebyshev</li>
<li>&#8216;Luis&#8217;, Luis&#8217; custom filter</li>
<li>&#8216;none&#8217;, no filtering (or leave field out).</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>O.HP:</strong></dt>
<dd><p class="first last">for SPM, the filter cutoff in s
for fourier, the HP value or the [HP LP] values
notches out everything slower than HP and faster than LP, in s
(1/s = Hz).</p>
</dd>
<dt><strong>O.nruns:</strong></dt>
<dd><p class="first last">number of runs (scanadjust), default is 1</p>
</dd>
<dt><strong>O.adjustmatrix:</strong></dt>
<dd><p class="first last">custom adjustment matrix to regress out (e.g., movement params)</p>
</dd>
<dt><strong>O.plot [0 or 1]:</strong></dt>
<dd><p class="first last">plots intermediate products, default is 0</p>
</dd>
<dt><strong>O.verbose [0 or 1]:</strong></dt>
<dd><p class="first last">verbose output</p>
</dd>
<dt><strong>O.trimts [0 or std]:</strong></dt>
<dd><p class="first last">trim overall timseries values to std, 0 for no trimming</p>
</dd>
<dt><strong>O.lindetrend:</strong></dt>
<dd><p class="first">specify linear detrending of timeseries.
occurs after adjustment and filtering and windsorizing</p>
<blockquote class="last">
<div><ul class="simple">
<li>detrending option -&gt; what to enter in this field:</li>
<li>no detrending  -&gt; empty, missing field, or 0</li>
<li>detrending every n elements -&gt; single number (n)</li>
<li>piecewise linear detrend -&gt; ROW vector of breakpoints
(do not specify 1 as the start of the 1st segment.)</li>
</ul>
</div></blockquote>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.fir2htw2">
<code class="descclassname">Data_processing_tools.</code><code class="descname">fir2htw2</code><span class="sig-paren">(</span><em>b</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.fir2htw2" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimates height, time to peak, and width of FIR response</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[h,t,w,w_times,halfh, auc] = fir2htw2(b,[hconstraint],[doplot],[colors cell])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>b:</strong></dt>
<dd><p class="first last">beta/estimate series for hemodynamic response curve</p>
</dd>
<dt>hconstraint:**</dt>
<dd><p class="first last">max time in samples that can be considered the peak
(default = last sample)</p>
</dd>
<dt>doplot:**</dt>
<dd><p class="first last">flag for plot, 1/0</p>
</dd>
<dt>colors:**</dt>
<dd><p class="first last">cell vector of colors for plot</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>minh = min height</p>
<p>This version uses turning points (zero gradient) to find the largest
&#8220;hump&#8221; in the data and the time it occurs.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>hrf = spm_hrf(.5); hrf = hrf ./ max(hrf); hrf = hrf + .1 * randn(length(hrf), 1);
create_figure(&#39;hrf&#39;); plot(hrf);
[h,t,w,w_times,halfh, auc] = fir2htw2(hrf, [], 1);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.get_snr">
<code class="descclassname">Data_processing_tools.</code><code class="descname">get_snr</code><span class="sig-paren">(</span><em>data</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.get_snr" title="Permalink to this definition">¶</a></dt>
<dd><p>Data is a matrix whos columns index voxels, and rows index subjects (or trials, etc.)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">snr</span> <span class="o">=</span> <span class="n">get_snr</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.htw_from_fit">
<code class="descclassname">Data_processing_tools.</code><code class="descname">htw_from_fit</code><span class="sig-paren">(</span><em>hrf</em>, <em>b</em>, <em>dt</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.htw_from_fit" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimates height, time to peak,  width, and area under the curve of a fitted response</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>h, t, w, auc, w_times, halfh] = htw_from_fit(hrf, b, dt, [optional arguments])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>hrf:</strong></dt>
<dd><dl class="first last docutils">
<dt>a matrix of columns that form a linear basis set for an</dt>
<dd><p class="first last">event type in an fMRI design, t time points x k basis
functions</p>
</dd>
</dl>
</dd>
<dt><strong>b:</strong></dt>
<dd><p class="first last">betas associated with the columns of hrf, k x 1</p>
</dd>
<dt><strong>dt:</strong></dt>
<dd><p class="first last">the sampling resolution of hrf (in seconds)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>plot:</strong></dt>
<dd><p class="first last">make plot</p>
</dd>
<dt><strong>verbose:</strong></dt>
<dd><p class="first last">verbose output</p>
</dd>
<dt><strong>startval:</strong></dt>
<dd><p class="first last">followed by the starting value in sec within which to calculate peak</p>
</dd>
<dt><strong>endval:</strong></dt>
<dd><p class="first last">followed by the ending value in sec within which to calculate peak</p>
</dd>
<dt><strong>colors:</strong></dt>
<dd><p class="first last">followed by a cell array, for example, {&#8216;r&#8217;} or {[1 0 0]}</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This function is essentially the same as fir2htw2.m, but the main
differences are:</p>
<blockquote>
<div><ol class="arabic simple">
<li>It imposes a time constraint on the peak amplitude automatically,
which is constrained to be between 4 seconds and 12 seconds
(endval, which was hconstraint) by default. YOU MAY WANT TO CHANGE hconstraint
depending on whether you&#8217;re expecting delayed hemodynamic responses.
This requires input of the sampling resolution (e.g., dt)</li>
<li>This function will automatically create fitted responses, given a
basis set and betas (parameters).  This is different from fir2htw2.m,
which takes the fitted response as input.</li>
<li>The method for getting width (w) has been changed to work better
for multi-modal (multi-peak) responses.</li>
</ol>
</div></blockquote>
<p>Otherwise, the algorithm is the same.
See Lindquist and Wager, 2007, for simulations that use a version of
this method to estimate HRFs using different kinds of models.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Notes on scaling:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>The scaling of the amplitude depends on the scaling of the hrf basis
set, which (in SPM) depends on the time resolution.  You should at
least use an hrf basis set with the same scaling for all subjects in
a group analysis.  The amplitude of the fitted response is
interpreted as the amplitude of the unit &#8220;impulse response,&#8221; assuming
that the hrf you enter here is the same as the impulse response
function you used to create the design matrix.  In SPM5, higher-res
impulse response functions are normalized by their positive sum, and
the higher the time resolution, the lower the amplitude of the unit
HRF.  (The scaling of regressors in the SPM design matrix isn&#8217;t
affected, because the hrf basis functions are convolved with a boxcar
that also depends on the time resolution.  The bottom line is that if
all your subjects have the same scaling, you should be fine.  And,
secondly, the amplitudes that come out of this function reflect the
scaling of the HRF you put in and are for an impulse response, NOT
for an &#8220;event,&#8221; and so the scaling here would not be expected to
match up with the amplitudes on a plot that you&#8217;d get from a
selective average time-course plot, unless you adjust by multiplying
by the number of elements in SPMs &#8220;hi-res&#8221; onset boxcar to adjust.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">For example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>With &#8220;zero-duration&#8221; events, an hrf input scaled to reflect &#8220;event response&#8221; amplitudes
might look something like this: (<a href="#id25"><span class="problematic" id="id26">**</span></a><em>may not be exactly right because
i think dt is in sec)
figure; plot(conv(SPM.xBF.bf(:, 1), my_ons))
my_ons = ones(1, TR ./ SPM.xBF.dt .</em> SPM.Sess.U(1).dur(1));</p>
<p>If you have epochs and want &#8220;epoch response&#8221; amplitude, you have to consider that as well.
If your durations are specified in TRs, and all durations are the
same:
TR = SPM.xY.RT;
my_ons = ones(1, TR ./ SPM.xBF.dt .* SPM.Sess.U(1).dur(1));</p>
<p>minh = min height</p>
<p>This version uses turning points (zero gradient) to find the largest
&#8220;hump&#8221; in the data and the time it occurs.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Load and SPM mat file and use the basis set stored in that, and use
% that as the hrf.  Generate some arbitrary combos to test different shapes:
% cd(&#39;my spm directory&#39;)
% load SPM
[h, t, w, auc] = htw_from_fit(SPM.xBF.bf, [1 .4 .4]&#39;, SPM.xBF.dt, &#39;plot&#39;, &#39;verbose&#39;);

for i = 1:20
    [h, t, w, auc] = htw_from_fit(SPM.xBF.bf, randn(3, 1), SPM.xBF.dt, &#39;plot&#39;); pause(1.5);
end

% Generate an SPM basis set at a lower resolution, and try that:
bf = spm_get_bf(struct(&#39;name&#39;, &#39;hrf (with time and dispersion derivatives)&#39;, &#39;length&#39;, 30, &#39;dt&#39;, 1));
for i = 1:20
   [h, t, w, auc] = htw_from_fit(bf.bf, randn(3, 1), bf.dt, &#39;plot&#39;); h, t, w, auc, pause(1.5)
end
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.luisFilter">
<code class="descclassname">Data_processing_tools.</code><code class="descname">luisFilter</code><span class="sig-paren">(</span><em>data</em>, <em>TR</em>, <em>cutoff</em>, <em>verbose</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.luisFilter" title="Permalink to this definition">¶</a></dt>
<dd><p>This is a low pass FIR filter using the Parks Mclellan design algorithm
The phase introduced by the filter is linear and is un-done by 
filtering the data again, backwards
if you want to see what it does to the data, use verbose=1
if just want to filter the data, don&#8217;t use the argument at all.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function outdata = luisFilter(data,  TR, cutoff [, verbose])
</pre></div>
</div>
<p>close all</p>
<blockquote>
<div>These are the important lines of the code   %%%%%%%%%%%%%</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.nuisance_cov_estimates">
<code class="descclassname">Data_processing_tools.</code><code class="descname">nuisance_cov_estimates</code><span class="sig-paren">(</span><em>X</em>, <em>images</em>, <em>SETUP</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.nuisance_cov_estimates" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimates F-map for model parameters of interest
Writes to disk:
&#8216;F_cols_of_interest.img&#8217; &#8216;p_cols_of_interest.img&#8217; (3-D images)
&#8216;resid_full_model.img&#8217; (a 4-D image)</p>
<p>Locates voxels whose activity is unrelated to the model</p>
<p>Extracts principal components from these voxels for use as covariates
in subsequent models</p>
<p>Note: invalidates statistical inference in subsequent models based on
these data, but may improve predictive accuracy and single-trial model.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">nuisance_cov_estimates</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">images</span><span class="p">,</span> <span class="n">SETUP</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Defining the SETUP structure with inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>SETUP.(fields)</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>.wh_of_interest:</strong></dt>
<dd><dl class="first last docutils">
<dt>vector of which columns of X matrix are of interest</dt>
<dd>% columns of interest in X matrix; tests var explained by these with F-test</dd>
</dl>
</dd>
<dt><strong>.mask:</strong></dt>
<dd>name of mask image</dd>
<dt><strong>.TR:</strong></dt>
<dd>repetition time of volume (image) acquisition</dd>
<dt><strong>.HPlength:</strong></dt>
<dd>high-pass filter length, in s</dd>
<dt><strong>.scans_per_session:</strong></dt>
<dd>vector of # volumes in each run, e.g., [128 128 128 128 128]</dd>
<dt><strong>.dummyscans:</strong></dt>
<dd>indices of images in each run that will be modeled
with separate dummy variables</dd>
<dt><strong>.startslice:</strong></dt>
<dd>start at slice #...</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">SETUP Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>nopreproc:</strong></dt>
<dd><p class="first last">to skip preprocessing (i.e., for trial-level inputs)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.resample_scnlab">
<code class="descclassname">Data_processing_tools.</code><code class="descname">resample_scnlab</code><span class="sig-paren">(</span><em>data</em>, <em>p</em>, <em>q</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.resample_scnlab" title="Permalink to this definition">¶</a></dt>
<dd><p>Resample : Uses matlab&#8217;s resample.m, but pads ends to avoid edge
artifacts</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[y, x] = resample_scnlab(data, p, q)
% OR
[y, x] = resample_scnlab(data, p, q, origHz, targetHz)
</pre></div>
</div>
<dl class="docutils">
<dt>Y = RESAMPLE(X,P,Q) resamples the sequence in vector X at P/Q times</dt>
<dd>the original sample rate using a polyphase implementation.  Y is P/Q 
times the length of X (or the ceiling of this if P/Q is not an integer).  
P and Q must be positive integers.</dd>
</dl>
<p>Other features:</p>
<p>Returns x values for resampled data in original index scale</p>
<p>IF two additional args are entered (origHz and targetHz), 
p and q are determined automatically, based on your desired sampling rate
(targetHz)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>create_figure(&#39;test&#39;); plot(y);
[y2, x] = resample_scnlab(y, 1, 5);
plot(x, y2, &#39;r&#39;);

%Use target Hz...take 100 Hz vector and resample at 20 Hz
create_figure(&#39;test&#39;);
plot(y); [y2, x] = resample_scnlab(y, [], [], 100, 20);
plot(x, y2, &#39;r&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.scale">
<code class="descclassname">Data_processing_tools.</code><code class="descname">scale</code><span class="sig-paren">(</span><em>x</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.scale" title="Permalink to this definition">¶</a></dt>
<dd><p>Centers and scales column vectors to mean 0 and st. deviation 1</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>x = scale(x,[just center])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.scnlab_filter_fmri_data">
<code class="descclassname">Data_processing_tools.</code><code class="descname">scnlab_filter_fmri_data</code><span class="sig-paren">(</span><em>imgs</em>, <em>mvmt</em>, <em>mask</em>, <em>tr</em>, <em>spersess</em>, <em>hp</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.scnlab_filter_fmri_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Outlier and artifact removal for one subject
Writes new output images for timeseries</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">names</span> <span class="o">=</span> <span class="n">scnlab_filter_fmri_data</span><span class="p">(</span><span class="n">imgs</span><span class="p">,</span> <span class="n">mvmt</span><span class="p">,</span> <span class="n">mask</span><span class="p">,</span> <span class="n">tr</span><span class="p">,</span> <span class="n">spersess</span><span class="p">,</span> <span class="n">hp</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">OUT</span> <span class="o">=</span> <span class="n">mean_image</span><span class="p">(</span><span class="n">imgs</span><span class="p">,</span> <span class="s">&#39;mean_ravol.img&#39;</span><span class="p">,</span><span class="n">ones</span><span class="p">(</span><span class="n">size</span><span class="p">(</span><span class="n">imgs</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span><span class="mi">1</span><span class="p">));</span>
<span class="n">spm_imcalc_ui</span><span class="p">(</span><span class="s">&#39;mean_ravol.img&#39;</span><span class="p">,</span> <span class="s">&#39;graymatter.img&#39;</span><span class="p">,</span> <span class="s">&#39;i1 &gt; 0&#39;</span><span class="p">);</span>
<span class="n">spm_image</span><span class="p">(</span><span class="s">&#39;init&#39;</span><span class="p">,</span> <span class="s">&#39;graymatter.img&#39;</span><span class="p">);</span>
<span class="n">names</span> <span class="o">=</span> <span class="n">scnlab_filter_fmri_data</span><span class="p">(</span><span class="n">imgs</span><span class="p">,</span> <span class="n">mvmt</span><span class="p">,</span> <span class="s">&#39;graymatter.img&#39;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="n">repmat</span><span class="p">(</span><span class="mi">184</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">6</span><span class="p">),</span> <span class="mi">80</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.scnlab_outlier_id">
<code class="descclassname">Data_processing_tools.</code><code class="descname">scnlab_outlier_id</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.scnlab_outlier_id" title="Permalink to this definition">¶</a></dt>
<dd><p>Methods (modes of operation)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Setup:</th><td class="field-body">Run this method first to generate an options structure OPT
that can be passed in along with any data vector for speedy
processing</td>
</tr>
<tr class="field-even field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>OPT = scnlab_outlier_id(&#39;setup&#39;, &#39;tr&#39;, 2, &#39;spersess&#39;,
                        [184 184 184 184 184 184], &#39;dummy&#39;, 1:3,
                        &#39;hp&#39;, 100, &#39;mad&#39;, 4, &#39;niter&#39;, 3, &#39;mvmt&#39;, mvmt);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Data:</th><td class="field-body">Run this method second with an already-created OPT
and a data vector from one time series</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">y2</span><span class="p">,</span> <span class="n">outliers</span><span class="p">,</span> <span class="n">num_outliers</span><span class="p">,</span> <span class="n">mvmt_rsquare</span><span class="p">]</span> <span class="o">=</span> <span class="n">scnlab_outlier_id</span><span class="p">(</span><span class="s">&#39;data&#39;</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="s">&#39;options&#39;</span><span class="p">,</span> <span class="n">OPT</span><span class="p">);</span>
</pre></div>
</div>
<p>all outputs:</p>
<div class="highlight-python"><div class="highlight"><pre>[y2, out, nout, mvmtrsq, mvmt_baseline_rsquare, yperc, rawvarp, rawvarF, percvarp, percvarF, ...
ybase] = scnlab_outlier_id(&#39;data&#39;, y, &#39;options&#39;, OPT);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[dat, volInfo] = iimg_get_data(&#39;graymask.img&#39;, imgs);
y = dat(:,1);
% SETUP:
OPT = scnlab_outlier_id(&#39;setup&#39;, &#39;tr&#39;, 2, &#39;spersess&#39;, [184 184 184 184 184 184], &#39;dummy&#39;, 1:3, &#39;hp&#39;, 100, &#39;mad&#39;, 4, &#39;niter&#39;, 3, &#39;mvmt&#39;, mvmt);
% RUN:
[y2, outliers, num_outliers] = scnlab_outlier_id(&#39;data&#39;, y, &#39;options&#39;, OPT);

% run on whole brain
[dat, volInfo] = iimg_get_data(&#39;graymask.img&#39;, imgs);
OPT = scnlab_outlier_id(&#39;setup&#39;, &#39;tr&#39;, 2, &#39;spersess&#39;, [184 184 184 184 184 184], &#39;dummy&#39;, 1:2, &#39;hp&#39;, 100, &#39;mad&#39;, 4, &#39;niter&#39;, 5, &#39;mvmt&#39;, mvmt);
OPT.doplot = 0;
OPT.verbose = 0;
fhandle = @(y) scnlab_outlier_id(&#39;data&#39;, y, &#39;options&#39;, OPT);
y2 = matrix_eval_function(dat, fhandle)&#39;;
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.selective_average">
<code class="descclassname">Data_processing_tools.</code><code class="descname">selective_average</code><span class="sig-paren">(</span><em>y</em>, <em>onsets</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.selective_average" title="Permalink to this definition">¶</a></dt>
<dd><p>Purpose: Get a selective average of values of data vector y, given
onsets specified in onsets.  Onsets can be fractional; in this case,
linear interpolation is used.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">averages</span><span class="p">,</span> <span class="n">stderrs</span><span class="p">,</span> <span class="n">data</span><span class="p">,</span> <span class="n">indices</span><span class="p">]</span> <span class="o">=</span> <span class="n">selective_average</span><span class="p">(</span><span class="n">y</span><span class="p">,</span> <span class="n">onsets</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>y:</strong></dt>
<dd><p class="first last">is a data vector to get selective averages from.
It should be a column vector.</p>
</dd>
<dt><strong>onsets:</strong></dt>
<dd><p class="first last">should be a cell array, with one cell per condition
each cell should contain a column vector of onset times in SAMPLES (same
resolution as y; e.g., in TRs, if y is an fMRI time series.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>t:</strong></dt>
<dd><p class="first last">followed by number of time points following onset to use;
default is 20</p>
</dd>
<dt><strong>plot:</strong></dt>
<dd><p class="first last">plot results.</p>
</dd>
<dt><strong>baseline:</strong></dt>
<dd><p class="first last">followed by vector of which time points are baseline
values; will subtract from each</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>data:</strong></dt>
<dd><p class="first last">indices, averages, stderrs:  Cell vectors, one cell per condition</p>
</dd>
<dt><strong>data, indices:</strong></dt>
<dd><p class="first last">time points (observations) x trials (onsets)</p>
</dd>
<dt><strong>indices:</strong></dt>
<dd><p class="first last">Cell vector, one cell per condition; time points (observations) x trials (onsets)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>onsets = {[1 10 30 80]&#39;  [20 60 90]&#39;}; y = (1:120)&#39;;
[averages, stderrs, data, indices] = selective_average(y, onsets, &#39;t&#39;, 20)

V = spm_vol(EXPT.FILES.im_files{1});
y = spm_get_data(V, [10 10 10 1]&#39;);
[averages, stderrs, data, indices] = selective_average(y, onsets2(1), &#39;t&#39;, 20, &#39;plot&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.selective_average_group">
<code class="descclassname">Data_processing_tools.</code><code class="descname">selective_average_group</code><span class="sig-paren">(</span><em>V</em>, <em>onsets</em>, <em>xyz_mm_pos</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.selective_average_group" title="Permalink to this definition">¶</a></dt>
<dd><p>uses selective_average.m
used in selective_average_interactive_view_init.m</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[group_avgs, group_stes, subject_avgs, subject_stes] = selective_average_group(V, onsets, vox, &#39;basepts&#39;, 1:2, &#39;plotstes&#39;, 0);

% Format onsets from onsets2 (NSF study format) into correct format for this function
N = length(imgs); n_conditions = size(eventdesign{1}, 2);
onsets = cell(1, N);
for i = 1:N
   for j = 1:n_conditions
       onsets{i}{j} = onsets2{i}(find(eventdesign{i}(:, j)));
   end
end
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.smooth_timeseries">
<code class="descclassname">Data_processing_tools.</code><code class="descname">smooth_timeseries</code><span class="sig-paren">(</span><em>x</em>, <em>perc</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.smooth_timeseries" title="Permalink to this definition">¶</a></dt>
<dd><p>Make exponential smoothing function with perc proportion of data points
(0 &lt; perc &lt; 1)
OR specify length directly, as % of points to 0 weight</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">x</span><span class="p">,</span><span class="n">V</span><span class="p">]</span> <span class="o">=</span> <span class="n">smooth_timeseries</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">perc</span><span class="p">)</span>
</pre></div>
</div>
<p>apply smoothing filter V to data (x), V * x
(works just as well for a matrix of column vectors)
You could also apply it to a model matrix X</p>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.splineDetrend">
<code class="descclassname">Data_processing_tools.</code><code class="descname">splineDetrend</code><span class="sig-paren">(</span><em>v</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.splineDetrend" title="Permalink to this definition">¶</a></dt>
<dd><p>What it does:
A spline detrend with knot points every 2 s (hard coded number)
I made this up too - it&#8217;s not the FDA approved method.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">fv</span><span class="p">,</span><span class="n">bp</span><span class="p">,</span><span class="n">yy</span><span class="p">,</span><span class="n">myfft</span><span class="p">]</span> <span class="o">=</span> <span class="n">splineDetrend</span><span class="p">(</span><span class="n">v</span><span class="p">,</span><span class="s">&#39;p&#39;</span> <span class="p">[</span><span class="n">opt</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>v:</strong></dt>
<dd><p class="first last">a vector to be detrended</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>p:</strong></dt>
<dd><p class="first last">means plot</p>
</dd>
</dl>
<p>any other opt argument sets the knots and is treated as an integer, with detrending every n
elements.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>fv:</strong></dt>
<dd><p class="first last">the detrended vector, bp, the knot points, and myfft, the
abs(fft) of the detrended vector.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>tmp = cl(1).raw_data(:,1,6); tmp2 = trimts(tmp,3,[],1); [fv,bp,yy]=splineDetrend(tmp2);
[fv,bp] = splineDetrend(tmp2,&#39;p&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.splinetrim">
<code class="descclassname">Data_processing_tools.</code><code class="descname">splinetrim</code><span class="sig-paren">(</span><em>y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.splinetrim" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">y</span><span class="p">,</span><span class="n">ntrimmed</span><span class="p">,</span><span class="n">spikes</span><span class="p">,</span> <span class="n">yfit</span><span class="p">]</span> <span class="o">=</span> <span class="n">splinetrim</span><span class="p">(</span><span class="n">y</span><span class="p">,[</span><span class="n">iqrmult</span><span class="p">],[</span><span class="n">knotrate</span><span class="p">],[</span><span class="n">X</span><span class="p">],[</span><span class="s">&#39;p&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>Uses a robust measure of deviations in a timeseries gradient
to find high-velocity &#8216;spikes&#8217;, presumed to be artifacts</p>
<p>Uses spline interpolation to replace spikes with reasonable values.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>y:</strong></dt>
<dd><p class="first last">a timeseries</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>iqrmult:</strong></dt>
<dd><dl class="first last docutils">
<dt>how many times the interquartile range above which velocities are</dt>
<dd><p class="first last">outliers, default is 1.5</p>
</dd>
</dl>
</dd>
<dt><strong>knotrate:</strong></dt>
<dd><p class="first last">sets knot points every k observations, default is 3</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">matrix of session means or other linear regressors to remove</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">plot the results</p>
</dd>
</dl>
<p>X and &#8216;p&#8217; can be entered in any order, but after iqrmult and knotrate</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">y2</span><span class="p">,</span><span class="n">nt</span><span class="p">]</span> <span class="o">=</span> <span class="n">splinetrim</span><span class="p">(</span><span class="n">trialdat</span><span class="p">,</span><span class="mi">3</span><span class="p">,</span><span class="mi">5</span><span class="p">,</span><span class="s">&#39;p&#39;</span><span class="p">);</span> <span class="n">nt</span>
</pre></div>
</div>
<ul class="simple">
<li>setup</li>
</ul>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.trimts">
<code class="descclassname">Data_processing_tools.</code><code class="descname">trimts</code><span class="sig-paren">(</span><em>y</em>, <em>sd</em>, <em>X</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.trimts" title="Permalink to this definition">¶</a></dt>
<dd><ol class="arabic">
<li><p class="first">Adjusts for scan effects (unless X is empty)</p>
</li>
<li><dl class="first docutils">
<dt>Windsorizes timeseries to sd standard deviations</dt>
<dd><ul class="first last simple">
<li>Recursive: 3 steps</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Adds scan effects back into timeseries</p>
</li>
</ol>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [y,ntrimmed,allw] = trimts(y,sd,X,[do spike correct],[trimming iterations],[MADs])
</pre></div>
</div>
<p>Spike correct: Some attempt at automatic adjustment for abrupt level
shifts in data; default is 0, enter 1 to do this
&#8220;Spikes&#8221; are IDd as values more than 10 MADs (by default) from moving average with 20 image FWHM
Replaces &#8220;spike&#8221; data values with moving average</p>
<p>iterations: number of cycles through trimming; default is 3</p>
<p>MADs: allows you to change the number of MADs above which values are IDd
as &#8220;spikes&#8221;</p>
<p>filter y using X matrix; yf is residuals</p>
</dd></dl>

<dl class="function">
<dt id="Data_processing_tools.use_spm_filter">
<code class="descclassname">Data_processing_tools.</code><code class="descname">use_spm_filter</code><span class="sig-paren">(</span><em>TR</em>, <em>dims</em>, <em>LChoice</em>, <em>HChoice</em>, <em>HParam</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Data_processing_tools.use_spm_filter" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [S,KL,KH] = use_spm_filter(TR,dim of filter,LChoice,HChoice,HP filter in s,[LP Gauss len in s])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>K{s}.LChoice:</strong></dt>
<dd><p class="first last">Low-pass  filtering {&#8216;hrf&#8217; &#8216;Gaussian&#8217; &#8216;none&#8217;}</p>
</dd>
<dt><strong>K{s}.LParam:</strong></dt>
<dd><p class="first last">Gaussian parameter in seconds</p>
</dd>
<dt><strong>K{s}.HChoice:</strong></dt>
<dd><p class="first last">High-pass filtering {&#8216;specify&#8217; &#8216;none&#8217;}</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="module-Filename_tools">
<span id="filename-tools"></span><h2>Filename_tools<a class="headerlink" href="#module-Filename_tools" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="Filename_tools.check_valid_imagename">
<code class="descclassname">Filename_tools.</code><code class="descname">check_valid_imagename</code><span class="sig-paren">(</span><em>P</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.check_valid_imagename" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>P = check_valid_imagename(P, [return error])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Optional Input:</th><td class="field-body"><p class="first">0 to return empty output, 1 to break with an error message,</p>
<p>2 to return list of bad/missing images, or</p>
<p class="last">nothing to select missing filenames graphically</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.copy_image_files">
<code class="descclassname">Filename_tools.</code><code class="descname">copy_image_files</code><span class="sig-paren">(</span><em>image_list</em>, <em>to_dir</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.copy_image_files" title="Permalink to this definition">¶</a></dt>
<dd><p>Copies a set of image files from one directory to another, creating the
directory if needed.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function output_names = copy_image_files(image_list, to_dir, [method=&#39;copy&#39; or &#39;move&#39;], varargin)
</pre></div>
</div>
<p>MAC OSX only!!</p>
<p>For .img files, will look for a paired .hdr and copy it automatically.
Works for non-image files (e.g., .txt) as well.</p>
<p>Other variable args:</p>
<p>Will take flags:</p>
<p>&#8216;append image number&#8217; -&gt; will append a number to each file corresponding
to the order listed, e.g., 1 for the first, 2 for the 2nd, etc.</p>
<p>&#8216;append string&#8217; -&gt; followed by string to append</p>
<p>with both flags and &#8216;run&#8217; for append string, a matrix of
[run1/vols.img; run2/vols.img] would become [vols_run0001.img vols_run0002.img]</p>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.delete_ana_imgs">
<code class="descclassname">Filename_tools.</code><code class="descname">delete_ana_imgs</code><span class="sig-paren">(</span><em>imgs</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.delete_ana_imgs" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Function to delete a list of .img files </dt>
<dd><ul class="first last simple">
<li>automatically removes associated .hdr files as well.</li>
<li>accepts cellstr or char inputs</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">delete_ana_imgs</span><span class="p">(</span><span class="n">imgs</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.dicom_tarzip">
<code class="descclassname">Filename_tools.</code><code class="descname">dicom_tarzip</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.dicom_tarzip" title="Permalink to this definition">¶</a></dt>
<dd><p>tars all .dcm files in subdirectories you specify, and deletes the
original .dcm files if successful.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body">none (default): all directories under the current one
directory wildcard, e.g., &#8216;run*&#8217; to search only those subdirectories.
absolute path names should also be ok.</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">dicom_tarzip</span>

<span class="n">dicom_tarzip</span><span class="p">(</span><span class="s">&#39;18*&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Or, let&#8217;s say you have runs within subjects, and dicom files within run dirs.
Subject dirs all start with NSF*.  run dirs all start with 18*
Starting from the directory above subjects, you could do this to zip ALL subjects:</p>
<div class="highlight-python"><div class="highlight"><pre>subjdirs = dir(&#39;NSF*&#39;);
for s = 1:length(subjdirs)
    dicom_tarzip([subjdirs(s).name filesep &#39;18*&#39;]);
end
</pre></div>
</div>
<p>Needless to say, use this with extreme caution, as it deletes your original files.
And test it thoroughly with a backup dataset first!!</p>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.escapeForShell">
<code class="descclassname">Filename_tools.</code><code class="descname">escapeForShell</code><span class="sig-paren">(</span><em>string</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.escapeForShell" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns converted string for use with the current OS&#8217;s shell. Hence,
it&#8217;s operation will be different, depending on where it&#8217;s executed.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">escapedString</span><span class="p">]</span> <span class="o">=</span> <span class="n">escapeForShell</span><span class="p">(</span><span class="n">string</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.expand_4d_filenames">
<code class="descclassname">Filename_tools.</code><code class="descname">expand_4d_filenames</code><span class="sig-paren">(</span><em>imgs</em>, <em>nvols</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.expand_4d_filenames" title="Permalink to this definition">¶</a></dt>
<dd><p>Expand a list of image names to 4-D name format as used by SPM2/5
Also compatible with SPM8.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">spm_imgs</span><span class="p">]</span> <span class="o">=</span> <span class="n">expand_4d_filenames</span><span class="p">(</span><span class="n">imgs</span><span class="p">,</span> <span class="p">[</span><span class="n">nvols</span><span class="p">])</span>
</pre></div>
</div>
<p>This function runs with either imgs, nvols, or both variables defined
If the images exist, spm_imgs is returned as a char matrix with each row
in the following format: &#8216;path/filename.ext,[N]&#8217; where [N] is the number
of the given image.  The space after [N] is padded with blanks to ensure
that all the image names fit properly.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">spm_imgs</span> <span class="o">=</span> <span class="n">expand_4d_filenames</span><span class="p">(</span><span class="s">&#39;run03.img&#39;</span><span class="p">,</span> <span class="mi">10</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.filename_get_new_root_dir">
<code class="descclassname">Filename_tools.</code><code class="descname">filename_get_new_root_dir</code><span class="sig-paren">(</span><em>P</em>, <em>newroot</em>, <em>nlevels</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.filename_get_new_root_dir" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">Pout</span> <span class="o">=</span> <span class="n">filename_get_new_root_dir</span><span class="p">(</span><span class="n">P</span><span class="p">,</span><span class="n">newroot</span><span class="p">,</span><span class="n">nlevels</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>P:</strong></dt>
<dd><p class="first last">is files (string matrix)</p>
</dd>
<dt><strong>newroot:</strong></dt>
<dd><p class="first last">Append a new root directory to filenames
if newroot is empty, prompt</p>
</dd>
<dt><strong>nlevels:</strong></dt>
<dd><dl class="first last docutils">
<dt>Keeps nlevels dirs deep from old directory:</dt>
<dd><ul class="first last simple">
<li>keeps only filename</li>
<li>keeps filename plus last subdir</li>
<li>keeps filename plus last 2 subdirs, etc.</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>Pout = filename_get_new_root_dir(EXPT.SNPM.P{3},pwd,2)

% Change root dir to pwd for all images in EXPT
for i=1:length(EXPT.SNPM.P), EXPT.SNPM.P{i} = filename_get_new_root_dir(EXPT.SNPM.P{i},pwd,2); end
</pre></div>
</div>
<p>Note: see regexprep.m for a simpler way!!! This function could be
improved by using it.</p>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.getfullpath">
<code class="descclassname">Filename_tools.</code><code class="descname">getfullpath</code><span class="sig-paren">(</span><em>Pspm</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.getfullpath" title="Permalink to this definition">¶</a></dt>
<dd><p>Searches for file in curr dir, then in specified dir</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">Pso</span> <span class="o">=</span> <span class="n">getfullpath</span><span class="p">(</span><span class="n">Pspm</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Pspm:</strong></dt>
<dd><p class="first last">is file name with no path or relative path</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><p class="first last">Returns absolute path name (full path)</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.nums_from_text">
<code class="descclassname">Filename_tools.</code><code class="descname">nums_from_text</code><span class="sig-paren">(</span><em>mytext</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.nums_from_text" title="Permalink to this definition">¶</a></dt>
<dd><p>returns numeric values of numbers embedded in text strings
given text string input</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">nums</span><span class="p">,</span><span class="n">whnums</span><span class="p">]</span> <span class="o">=</span> <span class="n">nums_from_text</span><span class="p">(</span><span class="n">mytext</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.read_excel">
<code class="descclassname">Filename_tools.</code><code class="descname">read_excel</code><span class="sig-paren">(</span><em>excelfilename</em>, <em>has_header_row</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.read_excel" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads each sheet of an Excel input file into a data structure DAT</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>   DAT = read_excel(excelfilename, has_header_row [1 or 0])

- reads multiple sheets
- adds fields named with variable names for easy access
- uses importdata &lt;matlab internal&gt; to do most of the work
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>excelfilename:</strong></dt>
<dd><p class="first last">Char array with full path of input filename</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>DAT:</strong></dt>
<dd><p class="first last">Structure with DAT.(sheetname).(varname) data fields
DAT.(sheetname).varnames is cell array of var names</p>
</dd>
<dt><strong>behdat:</strong></dt>
<dd><p class="first last">Data structure as read in by importdata</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">excelfilename</span> <span class="o">=</span> <span class="n">fullfile</span><span class="p">(</span><span class="n">basedir</span><span class="p">,</span> <span class="s">&#39;data&#39;</span><span class="p">,</span> <span class="s">&#39;SchulzNewcorn_regressor_demograph.xlsx&#39;</span><span class="p">);</span>
<span class="n">has_header_row</span> <span class="o">=</span> <span class="mi">1</span><span class="p">;</span>
<span class="p">[</span><span class="n">DAT</span><span class="p">,</span> <span class="n">behdat</span><span class="p">]</span> <span class="o">=</span> <span class="n">read_excel</span><span class="p">(</span><span class="n">excelfilename</span><span class="p">,</span> <span class="n">has_header_row</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>importdata, read_database, read_database2, read_edat_output_2008, read_physio_data</p>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.remove_disdaq_vols">
<code class="descclassname">Filename_tools.</code><code class="descname">remove_disdaq_vols</code><span class="sig-paren">(</span><em>img_files</em>, <em>num_vols_per_run</em>, <em>num_disdaq_vols</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.remove_disdaq_vols" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">new_files</span> <span class="o">=</span> <span class="n">remove_disdaq_vols</span><span class="p">(</span><span class="n">img_files</span><span class="p">,</span> <span class="n">num_vols_per_run</span><span class="p">,</span> <span class="n">num_disdaq_vols</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;overwrite&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="o">|</span><span class="mi">1</span><span class="p">],</span> <span class="p">[</span><span class="s">&#39;FSLOUTPUTTYPE&#39;</span><span class="p">,</span> <span class="n">fsl_output_type</span><span class="p">],</span> <span class="p">[</span><span class="s">&#39;strict&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="o">|</span><span class="mi">1</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>img_files:</strong></dt>
<dd><p class="first last">cellstr of files to remove disdaqs from - each cell represents a run</p>
</dd>
<dt><strong>num_vols_per_run:</strong></dt>
<dd><p class="first last">vector of volume counts per run, <em>not</em> including disdaq vols</p>
</dd>
<dt><strong>num_disdaq_vols:</strong></dt>
<dd><p class="first last">constant describing how many data points to remove from the beginning of each run</p>
</dd>
<dt><strong>&#8216;overwrite&#8217;:</strong></dt>
<dd><p class="first last">if set, will overwriting images in place - defaults to 0</p>
</dd>
<dt><strong>&#8216;FSLOUTPUTTYPE&#8217;:</strong></dt>
<dd><p class="first last">outputfile type - defaults to &#8216;NIFTI&#8217;</p>
</dd>
<dt><strong>&#8216;strict&#8217;:</strong></dt>
<dd><p class="first last">if set, will error out unless data given to it is
exactly proper length - defaults to 1 - turn off ONLY with good reason</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>new_files:</strong></dt>
<dd><p class="first last">input files, but with a &#8216;d&#8217; prepended, unless &#8216;overwriting&#8217; was specified</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>NB: Not set up for 3d files yet!!!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% for an experiment
num_vols_per_run = [124 140 109];  % NOT including disdaqs
num_disdaq_vols = 4;
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.rename_lowercase">
<code class="descclassname">Filename_tools.</code><code class="descname">rename_lowercase</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.rename_lowercase" title="Permalink to this definition">¶</a></dt>
<dd><p>renames all files in dir with lower case versions
if all characters are capitalized.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>d = dir(&#39;0*&#39;); for i=1:length(d),cd(d(i).name),rename_lowercase, cd .., end
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.rename_uppercase">
<code class="descclassname">Filename_tools.</code><code class="descname">rename_uppercase</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.rename_uppercase" title="Permalink to this definition">¶</a></dt>
<dd><p>renames all files in dir with upper case versions
if all characters are capitalized.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>d = dir(&#39;0*&#39;); for i=1:length(d),cd(d(i).name),rename_lowercase, cd .., end
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.scan_get_files">
<code class="descclassname">Filename_tools.</code><code class="descname">scan_get_files</code><span class="sig-paren">(</span><em>n</em>, <em>filt</em>, <em>mesg</em>, <em>wd</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.scan_get_files" title="Permalink to this definition">¶</a></dt>
<dd><p>Drop-in replacement for spm_get. 
Main advantage is that it can use spm_select for SPM5+ to mimic older spm_get behavior.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">files</span> <span class="o">=</span> <span class="n">scan_get_files</span><span class="p">(</span><span class="n">n</span><span class="p">,</span> <span class="n">filt</span><span class="p">,</span> <span class="n">mesg</span><span class="p">,</span> <span class="n">wd</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Filename_tools.sort_image_filenames">
<code class="descclassname">Filename_tools.</code><code class="descname">sort_image_filenames</code><span class="sig-paren">(</span><em>P</em><span class="sig-paren">)</span><a class="headerlink" href="#Filename_tools.sort_image_filenames" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">P</span><span class="p">,</span> <span class="n">indices</span><span class="p">]</span> <span class="o">=</span> <span class="n">sort_image_filenames</span><span class="p">(</span><span class="n">P</span><span class="p">)</span>
</pre></div>
</div>
<p>Not all image name listing functions return imgs in the correct numbered
order!</p>
<p>This function resorts a string matrix of image file names by image number, 
in ascending order
At most, filename can have one number in it, or error is returned</p>
<p>P can be a string matrix of a cell of string matrices</p>
</dd></dl>

</div>
<div class="section" id="module-hewma_utility">
<span id="hewma-utility"></span><h2>hewma_utility<a class="headerlink" href="#module-hewma_utility" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="hewma_utility.Gaussian_mix">
<code class="descclassname">hewma_utility.</code><code class="descname">Gaussian_mix</code><span class="sig-paren">(</span><em>x</em>, <em>niter</em>, <em>basepts</em>, <em>verbose</em>, <em>doplot</em>, <em>doplot2</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.Gaussian_mix" title="Permalink to this definition">¶</a></dt>
<dd><p>Two-Gaussian mixture model</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">ind</span><span class="p">,</span><span class="n">ind2</span><span class="p">,</span><span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">Gaussian_mix</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">niter</span><span class="p">,</span><span class="n">basepts</span><span class="p">,</span><span class="n">verbose</span><span class="p">,</span><span class="n">doplot</span><span class="p">,</span> <span class="n">doplot2</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>x:</strong></dt>
<dd><p class="first last">data</p>
</dd>
<dt><strong>iter:</strong></dt>
<dd><p class="first last">number of iterations</p>
</dd>
<dt><strong>basepts:</strong></dt>
<dd><p class="first last">number of baseline pts at start of run. The modal class in the
baseline period is defined as the 0-class</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>ind:</strong></dt>
<dd><p class="first last">indicator function of class belonging</p>
</dd>
<dt><strong>ind2:</strong></dt>
<dd><p class="first last">indicator function of class belonging where 3 consecutive points
are needed to switch states</p>
</dd>
<dt><strong>mu:</strong></dt>
<dd><p class="first last">mean vector</p>
</dd>
<dt><strong>sigma:</strong></dt>
<dd><p class="first last">standard deviation</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">probability that latent class random variable (delta) is equal
to [0,1]</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[ind,ind2,stats] = Gaussian_mix(linear_detrending(dat),20);
% dat is n subjects by t time points
% plotting is on

% Simulated data
r = normrnd(1, 1, 200, 1); r(1:50) = r(1:50) + normrnd(3, 1, 50, 1);
[ind, ind2, stats] = Gaussian_mix(r, 50, [], 0, 1);
</pre></div>
</div>
<p>We recommend 50 iterations
..</p>
<p>Set up inputs</p>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.change_point">
<code class="descclassname">hewma_utility.</code><code class="descname">change_point</code><span class="sig-paren">(</span><em>t</em>, <em>tt</em>, <em>Mode</em>, <em>tthresh</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.change_point" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[cp,tm,b,t,wh,ooc_vector] = change_point(t,tt,Mode,tthresh)

[cp,baseline mean,max t value,time of max t] = change_point(Zm,sterr,tt,tthresh)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>tt:</strong></dt>
<dd><p class="first last">is number of baseline timepoints</p>
</dd>
<dt><strong>t:</strong></dt>
<dd><p class="first last">is group t-value timeseries</p>
</dd>
<dt><strong>Mode: &#8216;thresh&#8217; or &#8216;time&#8217;:</strong></dt>
<dd><p class="first">if &#8216;thresh&#8217;: tthresh = threshold for significant t-values.</p>
<p>change_point finds the first significant supra-threshold t-value and
looks back in time to estimate the change point.</p>
<p>if Mode = &#8216;time&#8217;</p>
<p class="last">change_point finds the estimated change-point for process determined
to be out of control at time tthresh</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>wh:</strong></dt>
<dd><p class="first last">indices of out of control points</p>
</dd>
<dt><strong>ooc_vector:</strong></dt>
<dd><p class="first last">indicator vector for ooc points</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.cluster_kmeans_parcel">
<code class="descclassname">hewma_utility.</code><code class="descname">cluster_kmeans_parcel</code><span class="sig-paren">(</span><em>x</em>, <em>CLU</em>, <em>doplot</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.cluster_kmeans_parcel" title="Permalink to this definition">¶</a></dt>
<dd><p>K-means clustering of voxels</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[cl2,nclasses,colors] = function cluster_kmeans_parcel(x,CLU,doplot,[overlay img])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>x:</strong></dt>
<dd><p class="first last">a voxels x data matrix</p>
</dd>
<dt><strong>CLU:</strong></dt>
<dd><p class="first last">a structure containing a list of XYZ coordinates and z-scores for voxels
(see clusters2CLU.m)</p>
</dd>
</dl>
<p>a plot flag</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>cl2:</strong></dt>
<dd><p class="first last">a k-length cell array of clusters structures
each cell contains a clusters structure for one data class</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.cnt_runs">
<code class="descclassname">hewma_utility.</code><code class="descname">cnt_runs</code><span class="sig-paren">(</span><em>tc</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.cnt_runs" title="Permalink to this definition">¶</a></dt>
<dd><p>Counts number of contiguous 1s in a timeseries</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">cnt</span><span class="p">,</span> <span class="n">tot</span><span class="p">,</span> <span class="n">lenmat</span><span class="p">]</span> <span class="o">=</span> <span class="n">cnt_runs</span><span class="p">(</span><span class="n">tc</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.ewma5">
<code class="descclassname">hewma_utility.</code><code class="descname">ewma5</code><span class="sig-paren">(</span><em>X</em>, <em>lambda</em>, <em>L</em>, <em>type</em>, <em>tt</em>, <em>doplot</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.ewma5" title="Permalink to this definition">¶</a></dt>
<dd><p>EWMA - Exponentially Weighted Moving Average for fMRI slice</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>SMap, SMapTh, CPMap, CNTMap, TWidMap, W1Map, W2Map, W3Map, Tsq,stats] = ewma3(X, lambda , L, type);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first last">Data over one slice (N x N x T data matrix) or (N x T) data matrix</p>
</dd>
<dt><strong>lambda:</strong></dt>
<dd><p class="first last">smoothing factor</p>
</dd>
<dt><strong>L:</strong></dt>
<dd><p class="first last">width of control limits, in standard errors</p>
</dd>
<dt><strong>type:</strong></dt>
<dd><p class="first last">Type of noise model - White Noise: &#8216;WN&#8217;, AR(1): &#8216;AR&#8217;, AR+WN:
&#8216;ARMA&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>SMap:</strong></dt>
<dd><p class="first last">Significance Map</p>
</dd>
<dt><strong>SMap:</strong></dt>
<dd><p class="first last">Significance Map Thresholded at L standard errors</p>
</dd>
<dt><strong>CPMap:</strong></dt>
<dd><p class="first last">Changepoint Map</p>
</dd>
<dt><strong>CNTMap:</strong></dt>
<dd><p class="first last">Map that depicts the number of times the process is
out-of-control</p>
</dd>
<dt><strong>TWidMap:</strong></dt>
<dd><p class="first last">map of total number of points out-of-control (all runs)</p>
</dd>
<dt><strong>W1Map:</strong></dt>
<dd><p class="first last">map of number of points out-of-control in first run</p>
</dd>
<dt><strong>W2Map:</strong></dt>
<dd><p class="first last">map of number of points out-of-control in second run</p>
</dd>
<dt><strong>W3Map:</strong></dt>
<dd><p class="first last">map of number of points out-of-control in third run</p>
</dd>
<dt><strong>Tsq:</strong></dt>
<dd><p class="first last">Squared, normalized data set for multi-subject analysis</p>
</dd>
<dt><strong>stats.model:</strong></dt>
<dd><p class="first last">noise type</p>
</dd>
<dt><strong>stats.lam:</strong></dt>
<dd><p class="first last">value of lambda</p>
</dd>
<dt><strong>stats.Z:</strong></dt>
<dd><p class="first last">EWMA statistic (smoothed timeseries)</p>
</dd>
<dt><strong>stats.var:</strong></dt>
<dd><p class="first last">variance of EWMA statistic</p>
</dd>
<dt><strong>stats.doasym:</strong></dt>
<dd><p class="first last">1 for asymptotic variance 0 otherwise</p>
</dd>
<dt><strong>stats.dim:</strong></dt>
<dd><p class="first last">data dimension</p>
</dd>
<dt><strong>stats.Z3:</strong></dt>
<dd><p class="first last">EWMA statistic on the format NxVxT</p>
</dd>
<dt><strong>stats.UCL:</strong></dt>
<dd><p class="first last">upper control limit (at L standard errors, depends on lambda)</p>
</dd>
<dt><strong>stats.LCL:</strong></dt>
<dd><p class="first last">lower control limit</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% to plot voxel i,j from a slice
i=8;
j=17;
ts=squeeze(fsl(i,j,:));
tor_fig;
plot(ts);
hold on;
plot(squeeze(stats.Z(i,j,:)),&#39;r&#39;);
plot(squeeze(stats.UCL(i,j,:)),&#39;r:&#39;);
plot(squeeze(stats.LCL(i,j,:)),&#39;r:&#39;);
plot([chptmap(i,j)],stats.Z(i,j,chptmap(i,j)),&#39;yo&#39;,&#39;MarkerFaceColor&#39;,&#39;r&#39;)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.get_ax_slice">
<code class="descclassname">hewma_utility.</code><code class="descname">get_ax_slice</code><span class="sig-paren">(</span><em>imgs</em>, <em>slice_num</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.get_ax_slice" title="Permalink to this definition">¶</a></dt>
<dd><p>Get an axial slice</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function slice_data = get_ax_slice(imgs, slice_num)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>imgs:</strong></dt>
<dd><p class="first last">img filenames or spm_vols eg. &#8216;spmT_0004.img&#8217;;</p>
</dd>
<dt><strong>slice_num:</strong></dt>
<dd><p class="first last">slice number eg. 31</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>slice_data:</strong></dt>
<dd><p class="first last">unprocessed slice data</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.get_max_t">
<code class="descclassname">hewma_utility.</code><code class="descname">get_max_t</code><span class="sig-paren">(</span><em>Zpop</em>, <em>sterr</em>, <em>tt</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.get_max_t" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">t</span><span class="p">,</span><span class="n">tm</span><span class="p">,</span><span class="n">b</span><span class="p">]</span> <span class="o">=</span> <span class="n">get_max_t</span><span class="p">(</span><span class="n">Zpop</span><span class="p">,</span><span class="n">sterr</span><span class="p">,</span><span class="n">tt</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>t:</strong></dt>
<dd><p class="first last">t-value timeseries</p>
</dd>
<dt><strong>tm:</strong></dt>
<dd><p class="first last">max t-value (abs)</p>
</dd>
<dt><strong>b:</strong></dt>
<dd><p class="first last">time (index) of max t-value</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma2">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma2</code><span class="sig-paren">(</span><em>Zm</em>, <em>varZm</em>, <em>lambda</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma2" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [p,tm,Zcor,sterr,Zpop,tvals,sb,stats] = hewma2(Zm, varZm, lambda,[doplot],[dodetrend],[grpcontrast], [tt]);
</pre></div>
</div>
<p>Hierarchical (Group) Ewma - HEWMA</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Zm:</strong></dt>
<dd><p class="first last">data matrix (nsubjects x Time)</p>
</dd>
<dt><strong>varZm:</strong></dt>
<dd><p class="first last">variance for each time point and subject</p>
</dd>
<dt><strong>lambda:</strong></dt>
<dd><p class="first last">smoothing parameter</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Optional:</th><td class="field-body"><p class="first">Plot flag, 1 or 0</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>p:</strong></dt>
<dd><p class="first last">p-value</p>
</dd>
<dt><strong>tm:</strong></dt>
<dd><p class="first last">maximumum (over time) group t-statistic</p>
</dd>
<dt><strong>Zcor:</strong></dt>
<dd><p class="first last">temporal-search corrected Z-score (max t - nullmaxt) / std(nullmax t)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Note:</th><td class="field-body"><p class="first last">The speed of this function is largely determined by nrep, which is</p>
</td>
</tr>
</tbody>
</table>
<p>the number of repititions in the Monte carlo integration. A large value
of nrep gives greater accuracy, but is slower.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Generate random data where there should be a true effect and test
n = noisevector(100,[1 .9 .8 .6 .4 .1],1);
n = [zeros(1,60) n];
for i = 1:20, Zm(i,:) = n + randn(1,160);,end
varZm = ones(size(Zm));
[p,tm,Zcor,sterr,Zpop,tvals,sb,stats] = hewma2(Zm, varZm,.2,1);

% Now try with random, non-identical weights:
varZm = rand(size(Zm));

% Now try with random autocorrelated data
for i = 1:20, Zm(i,:) = noisevector(160,[1 .9 .8 .5 .2],1)&#39;;,end
[p,tm,Zcor,sterr,Zpop,tvals,sb,stats] = hewma2(Zm, varZm, .2,1);

% Add a group difference:
n = noisevector(100,[1 .9 .8 .6 .4 .1],1);
n = [zeros(1,60) n];
for i = 1:10, Zm(i,:) = n + randn(1,160);,end
n = noisevector(100,[1 .9 .8 .6 .4 .1],1);
n = [zeros(1,60) n];
for i = 11:20, Zm(i,:) = n + randn(1,160);,end
varZm = ones(size(Zm));
[p,tm,Zcor,sterr,Zpop,tvals,sb,stats] = hewma2(Zm, varZm,.2,1,0,[ones(10,1); -1*ones(10,1)]);

% Add linear drift, to test detrending:
n = noisevector(100,[1 .9 .8 .6 .4 .1],1);
n = [zeros(1,60) n];
for i = 1:10, Zm(i,:) = n + randn(1,160) + (rand(1)-.5) * (1:160);,end
n = noisevector(100,[1 .9 .8 .6 .4 .1],1);
n = [zeros(1,60) n];
for i = 11:20, Zm(i,:) = n + randn(1,160) + (rand(1)-.5) * (1:160);,end
varZm = ones(size(Zm));
[p,tm,Zcor,sterr,Zpop,tvals,sb,stats] = hewma2(Zm, varZm,.2,1,0,[ones(10,1); -1*ones(10,1)]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma2_plot">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma2_plot</code><span class="sig-paren">(</span><em>Zpop</em>, <em>mu</em>, <em>cp</em>, <em>cp_ind</em>, <em>ooc_indices</em>, <em>maxtime</em>, <em>tthresh</em>, <em>tt</em>, <em>tm</em>, <em>q</em>, <em>Wm</em>, <em>Zm</em>, <em>sterr</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma2_plot" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">hewma2_plot</span><span class="p">(</span><span class="n">Zpop</span><span class="p">,</span><span class="n">mu</span><span class="p">,</span><span class="n">cp</span><span class="p">,</span><span class="n">cp_ind</span><span class="p">,</span><span class="n">ooc_indices</span><span class="p">,</span><span class="n">maxtime</span><span class="p">,</span><span class="n">tthresh</span><span class="p">,</span><span class="n">tt</span><span class="p">,</span><span class="n">tm</span><span class="p">,</span><span class="n">q</span><span class="p">,</span><span class="n">Wm</span><span class="p">,</span><span class="n">Zm</span><span class="p">,</span><span class="n">sterr</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>see hewma2.m</p>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma_extract_voxel">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma_extract_voxel</code><span class="sig-paren">(</span><em>EXPT</em>, <em>coords</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma_extract_voxel" title="Permalink to this definition">¶</a></dt>
<dd><p>Extracts data for one voxel and runs hewma2 on it.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">dat</span><span class="p">,</span><span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">hewma_extract_voxel</span><span class="p">(</span><span class="n">EXPT</span><span class="p">,</span><span class="n">coords</span><span class="p">)</span>
</pre></div>
</div>
<p>Uses plot option in hewma2 to create plots.</p>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma_from_raw_timeseries">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma_from_raw_timeseries</code><span class="sig-paren">(</span><em>raw_data</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma_from_raw_timeseries" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">STATS</span> <span class="o">=</span> <span class="n">hewma_from_raw_timeseries</span><span class="p">(</span><span class="n">raw_data</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">imraw_data, a N subjects x T time points matrix of data</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">For all optional inputs, enter a keyword followed by an input value/argument
(this is a standard Matlab format for entering optional arguments)
e.g., &#8216;lam&#8217;, .3 to enter a lambda value of .3</p>
<p>Here are the optional inputs and their defaults:</p>
<dl class="docutils">
<dt><strong>lam = .2;</strong></dt>
<dd><p class="first last">ewma smoothing param</p>
</dd>
<dt><strong>L = 2;</strong></dt>
<dd><p class="first last">ewma control limit</p>
</dd>
<dt><strong>noisemodel = &#8216;AR(2)&#8217;;</strong></dt>
<dd><p class="first last">noise structure type</p>
</dd>
<dt><strong>base_timepts = 60;</strong></dt>
<dd><p class="first last">baseline time points to use</p>
</dd>
<dt><strong>doplot = 1;</strong></dt>
<dd><p class="first last">plot toggle (1/0)</p>
</dd>
<dt><strong>dodetrend = 1;</strong></dt>
<dd><p class="first last">hewma linear detrending toggle (1/0)</p>
</dd>
<dt><strong>grpcontrast = [];</strong></dt>
<dd><p class="first last">vector of 1 or -1 values for group assignment for
each subject</p>
</dd>
<dt><strong>samprate = .5;</strong></dt>
<dd><p class="first last">sampling rate in Hz, 1/TR;  TR = 2 by default</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body"><p class="first last">see ewma5.m and hewma2.m for more information on these inputs</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma_gui">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma_gui</code><span class="sig-paren">(</span><em>Action</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma_gui" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">varargout</span><span class="o">=</span><span class="n">hewma_gui</span><span class="p">(</span><span class="n">Action</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>To run, type &#8220;hewma&#8221; at the Matlab prompt</p>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma_plot_bivariate">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma_plot_bivariate</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma_plot_bivariate" title="Permalink to this definition">¶</a></dt>
<dd><p>Visualize a change_point map stored in hewma_cp.img
and a run-length map stored in hewma_runlen.img
(output from hewma2)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[cl,cl2] = hewma_plot_bivariate([Method],[optional overlay image])
</pre></div>
</div>
<p>and classify voxels into groupings based on locations in the bivariate
space</p>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma_plot_coord_btnupfcn">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma_plot_coord_btnupfcn</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma_plot_coord_btnupfcn" title="Permalink to this definition">¶</a></dt>
<dd><p>function for loading and plotting hewma data from a voxel.  must have
declared globals f,f2,VOL,EXPT</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">dat</span><span class="p">,</span><span class="n">stats</span><span class="p">,</span><span class="n">mycov</span><span class="p">]</span> <span class="o">=</span> <span class="n">hewma_plot_coord_btnupfcn</span>
</pre></div>
</div>
<p>see hewma_timeseries_plot, the shell function.</p>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma_plot_cpmap">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma_plot_cpmap</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma_plot_cpmap" title="Permalink to this definition">¶</a></dt>
<dd><p>Visualize a change_point map stored in hewma_cp.img
(output from hewma2)
and classify voxels into groupings based on CP</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cl2</span><span class="p">,</span><span class="n">classes</span><span class="p">]</span> <span class="o">=</span> <span class="n">hewma_plot_cpmap</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma_plot_runlen">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma_plot_runlen</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma_plot_runlen" title="Permalink to this definition">¶</a></dt>
<dd><p>Visualize a change_point map stored in hewma_cp.img
(output from hewma2)
and classify voxels into groupings based on CP</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cl2</span><span class="p">,</span><span class="n">classes</span><span class="p">]</span> <span class="o">=</span> <span class="n">hewma_plot_runlen</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.hewma_save_timeseries">
<code class="descclassname">hewma_utility.</code><code class="descname">hewma_save_timeseries</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.hewma_save_timeseries" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cl = hewma_save_timeseries([mask to extract from or cl],[k],[[set of:grpmean,grpste,xdim,ydim]])
</pre></div>
</div>
<p>load hewma_timeseries
run this to save data in clusters format</p>
<p>Function: extract timeseries data from a mask (and extent thr k)
Gets group avg data for clusters, not individual.  For indiv, use
hewma_extract_voxel.m</p>
<p>Last args contain the data for the group for each voxel.
If last args are not entered, attempts to load from hewma_timeseries.mat
defaults</p>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.linear_detrending">
<code class="descclassname">hewma_utility.</code><code class="descname">linear_detrending</code><span class="sig-paren">(</span><em>Zm</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.linear_detrending" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">Zdt</span> <span class="o">=</span> <span class="n">linear_detrending</span><span class="p">(</span><span class="n">Zm</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Zm:</strong></dt>
<dd><p class="first last">nsub x T matrix;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Zdt:</strong></dt>
<dd><p class="first last">linearly detrended version of Zm (baseline retained)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.timeseries_btwngroups_plot">
<code class="descclassname">hewma_utility.</code><code class="descname">timeseries_btwngroups_plot</code><span class="sig-paren">(</span><em>dat</em>, <em>cov</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.timeseries_btwngroups_plot" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>h = timeseries_btwngroups_plot(dat,cov,[baseperiod],[dedetrend],[x],[legend text])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>imdat:</strong></dt>
<dd><p class="first last">subjects x time matrix of estimates</p>
</dd>
<dt><strong>baseperiod:</strong></dt>
<dd><p class="first last">integer; removes mean of 1:baseperiod for each subject</p>
</dd>
<dt><strong>cov:</strong></dt>
<dd><p class="first last">subjects x 1 contrast vector (individual/group differences)</p>
</dd>
<dt><strong>dodetrend:</strong></dt>
<dd><p class="first last">linear detrending of each subject&#8217;s estimates</p>
</dd>
<dt><strong>legend text:</strong></dt>
<dd><p class="first last">{&#8216;name1&#8217; &#8216;name2&#8217; ...}</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>h:</strong></dt>
<dd><p class="first last">line handles</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.timeseries_mc_pvalue">
<code class="descclassname">hewma_utility.</code><code class="descname">timeseries_mc_pvalue</code><span class="sig-paren">(</span><em>varZm</em>, <em>lambda</em>, <em>Wm</em>, <em>sesq</em>, <em>df</em>, <em>tm</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.timeseries_mc_pvalue" title="Permalink to this definition">¶</a></dt>
<dd><p>Uses Monte Carlo simulation on a timeeseries to get the null hypothesis
max t-value over time and establish an appropriate statistical threshold</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[Zcor,p,tthresh,q,C] = timeseries_mc_pvalue(varZm,lambda,Wm,sesq,df,tm)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>varZm:</strong></dt>
<dd><p class="first last">within subjects variances, subj x time</p>
</dd>
<dt><strong>lambda:</strong></dt>
<dd><p class="first last">smoothing parameter in EWMA</p>
</dd>
<dt><strong>Wm:</strong></dt>
<dd><p class="first last">individual case (subject) weights</p>
</dd>
<dt><strong>sesq:</strong></dt>
<dd><p class="first last">squared standard error between subjects (variance of group error estimate)</p>
</dd>
<dt><strong>df:</strong></dt>
<dd><p class="first last">degrees of freedom between subjects (est.)</p>
</dd>
<dt><strong>tm:</strong></dt>
<dd><p class="first last">max t stat for group over time</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.wb_hewma_shell">
<code class="descclassname">hewma_utility.</code><code class="descname">wb_hewma_shell</code><span class="sig-paren">(</span><em>EXPT</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.wb_hewma_shell" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>EXPT = wb_hewma_shell(EXPT,[start slice or vector of slices],[dools],[mask])
</pre></div>
</div>
<p>Run EWMA first: wb_multisubject_ewma.</p>
<dl class="docutils">
<dt>Fields in EXPT needed:</dt>
<dd><ul class="first last simple">
<li>subdir</li>
<li>im_files</li>
<li>FILES.ewma_z</li>
<li>FILES.ewma_var</li>
<li>[cov]</li>
<li>[mask]</li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.weighted_reg">
<code class="descclassname">hewma_utility.</code><code class="descname">weighted_reg</code><span class="sig-paren">(</span><em>Y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.weighted_reg" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate weighted average using weighted linear least squares
See examples below for usage</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Model:</th><td class="field-body"><p class="first">Y_i = 1*Ypop + noise</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">data matrix (nsub x T)</p>
</dd>
<dt><strong>w:</strong></dt>
<dd><p class="first last">weights</p>
</dd>
<dt><strong>varY:</strong></dt>
<dd><p class="first last">variance of data at each time point (nsub x T) + var between</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Ymean:</strong></dt>
<dd><p class="first last">weighted mean of each column of Y</p>
</dd>
<dt><strong>dfe:</strong></dt>
<dd><p class="first last">error degrees of freedom, adjusted for inequality of variance
(Sattherwaite) and pooled across data columns</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Extended output in stats structure:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>stats.t:</strong></dt>
<dd>t-values for weighted t-test</dd>
<dt><strong>stats.p:</strong></dt>
<dd>2-tailed p-values for weighted t-test</dd>
<dt><strong>r:</strong></dt>
<dd>weighted correlation coeff across columns of Y</dd>
<dt><strong>xy:</strong></dt>
<dd>weighted covariance matrix</dd>
<dt><strong>v:</strong></dt>
<dd><blockquote class="first">
<div>weighted variance estimates for each column of Y</div></blockquote>
<ul class="last simple">
<li>sqrt(v) is the standard error of the mean (or grp difference)</li>
</ul>
</dd>
<dt><strong>stats.fits:</strong></dt>
<dd>fits  for each group (Ymean by group), low contrast weight group then high
Fastest if no stats are asked for.</dd>
</dl>
</div></blockquote>
<p>Computation time:
For FULL stats report</p>
<blockquote>
<div><ul class="simple">
<li>Triples from 500 -&gt; 1000 columns of Y, continues to increase</li>
</ul>
</div></blockquote>
<p>For mean/dfe only, fast for full dataset (many columns of Y)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Basic multivariate stats for 1000 columns of dat, no weighting
% Multivariate covariances are meaningful if cols of Y are organized, e.g., timeseries
[means,stats] = weighted_reg(dat(:,1:1000));

% The same, but return univariate stats only (good for large Y)
[means,stats] = weighted_reg(dat,&#39;uni&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.weighted_reg_old2">
<code class="descclassname">hewma_utility.</code><code class="descname">weighted_reg_old2</code><span class="sig-paren">(</span><em>y</em>, <em>w</em>, <em>varz</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.weighted_reg_old2" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate weighted average using weighted linear least squares</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Model:</th><td class="field-body"><p class="first">z_i = 1*zpop + noise</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">data matrix (nsub x T)</p>
</dd>
<dt><strong>w:</strong></dt>
<dd><p class="first last">weights</p>
</dd>
<dt><strong>varz:</strong></dt>
<dd><p class="first last">variance of data at each time point (nsub x T)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>r:</strong></dt>
<dd><p class="first last">weighted correlation coeff across columns of y</p>
</dd>
<dt><strong>xy:</strong></dt>
<dd><p class="first last">weighted covariance matrix</p>
</dd>
<dt><strong>v:</strong></dt>
<dd><p class="first last">weighted variance estimates for each column of y</p>
</dd>
<dt><strong>zp:</strong></dt>
<dd><p class="first last">weighted mean of each column of y</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.weighted_reg_oldglmfit_old">
<code class="descclassname">hewma_utility.</code><code class="descname">weighted_reg_oldglmfit_old</code><span class="sig-paren">(</span><em>Y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.weighted_reg_oldglmfit_old" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate weighted average using weighted linear least squares
See examples below for usage</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Model:</th><td class="field-body"><p class="first">Y_i = 1*Ypop + noise</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">data matrix (nsub x T)</p>
</dd>
<dt><strong>w:</strong></dt>
<dd><p class="first last">weights</p>
</dd>
<dt><strong>varY:</strong></dt>
<dd><p class="first last">variance of data at each time point (nsub x T) + var between</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Ymean:</strong></dt>
<dd><p class="first last">weighted mean of each column of Y</p>
</dd>
<dt><strong>dfe:</strong></dt>
<dd><p class="first last">error degrees of freedom, adjusted for inequality of variance
(Sattherwaite) and pooled across data columns</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Extended output in stats structure:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>stats.t:</strong></dt>
<dd>t-values for weighted t-test</dd>
<dt><strong>stats.p:</strong></dt>
<dd>2-tailed p-values for weighted t-test</dd>
<dt><strong>r:</strong></dt>
<dd>weighted correlation coeff across columns of Y</dd>
<dt><strong>xy:</strong></dt>
<dd>weighted covariance matrix</dd>
<dt><strong>v:</strong></dt>
<dd><blockquote class="first">
<div>weighted variance estimates for each column of Y</div></blockquote>
<ul class="last simple">
<li>sqrt(v) is the standard error of the mean (or grp difference)</li>
</ul>
</dd>
<dt><strong>stats.fits:</strong></dt>
<dd>fits  for each group (Ymean by group), low contrast weight group then high
Fastest if no stats are asked for.</dd>
</dl>
</div></blockquote>
<p>Computation time:
For FULL stats report</p>
<blockquote>
<div><ul class="simple">
<li>Triples from 500 -&gt; 1000 columns of Y, continues to increase</li>
</ul>
</div></blockquote>
<p>For mean/dfe only, fast for full dataset (many columns of Y)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Basic multivariate stats for 1000 columns of dat, no weighting
% Multivariate covariances are meaningful if cols of Y are organized, e.g., timeseries
[means,stats] = weighted_reg(dat(:,1:1000));

% The same, but return univariate stats only (good for large Y)
[means,stats] = weighted_reg(dat,&#39;uni&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.whole_brain_ewma">
<code class="descclassname">hewma_utility.</code><code class="descname">whole_brain_ewma</code><span class="sig-paren">(</span><em>P</em>, <em>DX</em>, <em>TR</em>, <em>HP</em>, <em>nruns</em>, <em>numframes</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.whole_brain_ewma" title="Permalink to this definition">¶</a></dt>
<dd><p>Single-subject timeseries extraction and model fitting
saves filtered images f* and full-model betas (model fits)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function Pw = whole_brain_ewma(P,DX,TR,HP,nruns,numframes,[doplot],[mask],[smoothlen],[startslice])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>P:</strong></dt>
<dd><blockquote class="first">
<div><p>list of image names, raw functionals or beta images (see &#8220;type&#8221;</p>
</div></blockquote>
<p class="last">below)</p>
</dd>
<dt><strong>DX:</strong></dt>
<dd><p class="first last">model matrix of effects to REMOVE before EWMA, see, e.g., for FIR tor_make_deconv_mtx3.m</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">repetition time of your study (sampling rate)</p>
</dd>
<dt><strong>HP:</strong></dt>
<dd><p class="first last">high-pass filter cutoff in seconds to apply (SPM style)</p>
</dd>
<dt><strong>nruns:</strong></dt>
<dd><dl class="first last docutils">
<dt>number of sessions (intercepts) OR num. of images in each sess,</dt>
<dd><p class="first last">e.g., [169 169 172]</p>
</dd>
</dl>
</dd>
<dt><strong>numframes:</strong></dt>
<dd><dl class="first last docutils">
<dt>number of beta images per event type, e.g., [20 20 20] for FIR</dt>
<dd><p class="first last">model</p>
</dd>
</dl>
</dd>
<dt><strong>[doplot]:</strong></dt>
<dd><p class="first last">optional graphics</p>
</dd>
<dt><strong>[mask]:</strong></dt>
<dd><p class="first last">optional name of mask image to limit processing to brain, or []</p>
</dd>
<dt><strong>[smoothlen]:</strong></dt>
<dd><dl class="first docutils">
<dt>exponential smoothing filter length for timeseries and</dt>
<dd><p class="first last">betas; influence is 0 after smoothlen seconds; default:**</p>
</dd>
</dl>
<p class="last">6</p>
</dd>
<dt><strong>[startslice]:</strong></dt>
<dd><p class="first last">slice to start at, default = 1</p>
</dd>
<dt><strong>[type]:</strong></dt>
<dd><p class="first">optional type of analysis to run.  Options:
&#8216;full&#8217;    Everything, from filtering to height/time/width, enter</p>
<blockquote>
<div><p>raw image names and model matrix DX</p>
</div></blockquote>
<dl class="last docutils">
<dt>&#8216;data&#8217;    Filter data only and save images in f*img</dt>
<dd><p class="first last">Enter raw image names and empty model matrix []</p>
</dd>
<dt>&#8216;htw&#8217;     Smooths beta series and computes height/time/width only</dt>
<dd><dl class="first docutils">
<dt>Enter beta images from FIR model within condition,</dt>
<dd><p class="first last">e.g., beta1_event1 beta2_event1 ... beta1_event2 b2_e2
...</p>
</dd>
</dl>
<p class="last">One possibility is to use images from SPM2 at this
stage.</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>For a shell to run this function, see whole_brain_fir.m</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">The analysis sequence:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Run whole_brain_fir, which mainly just calls whole_brain_filter
whole_brain_filter gives you:</p>
<blockquote>
<div><ol class="arabic">
<li><dl class="first docutils">
<dt>trimmed, filtered images, saved in f*img</dt>
<dd><p class="first last">these can be passed to SPM2, e.g., if desired</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>extraction of beta images for FIR model, saved in dx_beta*img</dt>
<dd><p class="first last">in individual subject folders; also smooths betas if desired</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>calculation of height, delay, and width for each FIR extracted</dt>
<dd><p class="first last">saved on images in individual subject folders.</p>
</dd>
</dl>
</li>
</ol>
</div></blockquote>
<p>After running this, you&#8217;ll need to collect image names for use in random
effects analyses, and most likely create contrasts across conditions for
differences in height, delay, and width.</p>
<p>to do this, first go to main model directory, with subject subfolders
to save a list of height/delay/width images:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">EXPT</span> <span class="o">=</span> <span class="n">get_htw_image_names</span><span class="p">(</span><span class="n">EXPT</span><span class="p">)</span>
</pre></div>
</div>
<p>to create contrasts across those images and save in EXPT.SNPM for rfx
analysis:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">EXPT</span> <span class="o">=</span> <span class="n">make_htw_contrast_images</span><span class="p">(</span><span class="n">EXPT</span><span class="p">);</span>
</pre></div>
</div>
<p>Now you can run robust regression across the images in EXPT.SNPM.
The robust reg uses image names in EXPT.SNPM.P, so you may have to save
images from another field (e.g., EXPT.SNPM.heightP) in the .P field.
you may also want to test individual contrasts against zero, in which
case you can use EXPT.NLCON.height images, for example (and others in
NLCON) instead.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">EXPT</span> <span class="o">=</span> <span class="n">robfit</span><span class="p">(</span><span class="n">EXPT</span><span class="p">);</span>
</pre></div>
</div>
<p>Explanations of some more variables</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>vb:</strong></dt>
<dd>[optional] verbose output level: 0 none, 1 some, 2 lots</dd>
<dt><strong>mask:</strong></dt>
<dd>[optional] mask 3-D volume to apply before extracting</dd>
<dt><strong>P:</strong></dt>
<dd>image file names (str matrix)</dd>
<dt><strong>S:</strong></dt>
<dd>filtering matrix (e.g., high-pass)</dd>
<dt><strong>DX:</strong></dt>
<dd>full model to fit, unfiltered</dd>
<dt><strong>vb:</strong></dt>
<dd>[optional] verbose output level: 0 none, 1 some, 2 lots</dd>
<dt><strong>mask:</strong></dt>
<dd>[optional] mask 3-D volume to apply before extracting</dd>
<dt><strong>nsess:</strong></dt>
<dd>number of sessions (intercepts): assumes last nsess columns
of DX are run intercepts to be removed before trimming</dd>
<dt><strong>dims:</strong></dt>
<dd>dimensions of image files in data</dd>
<dt><strong>cols:</strong></dt>
<dd>no. of columns of DX, also no. of beta images to write</dd>
<dt><strong>SDX:</strong></dt>
<dd>smoothed (filtered) full model to fit</dd>
<dt><strong>PSDX:</strong></dt>
<dd>pseudoinverse of filtered full model</dd>
<dt><strong>PSDXS:</strong></dt>
<dd>PSDX * S, ready to multiply with data y for pinv(SX) * Sy</dd>
<dt><strong>betas:</strong></dt>
<dd>4-D array of x,y,z,column</dd>
<dt><strong>ntrimmed:</strong></dt>
<dd>number of outliers trimmed from timeseries at each voxel</dd>
<dt><strong>Pw:</strong></dt>
<dd>string matrix of output file names</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>Pw = whole_brain_filter(d,c.model(1:166,:),2,120,5,c.numframes,1,10);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="hewma_utility.zero_crossing">
<code class="descclassname">hewma_utility.</code><code class="descname">zero_crossing</code><span class="sig-paren">(</span><em>M</em>, <em>T</em>, <em>XX</em>, <em>ZIU</em>, <em>ZIL</em>, <em>Z</em>, <em>mu</em><span class="sig-paren">)</span><a class="headerlink" href="#hewma_utility.zero_crossing" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimate out of control points and change-point using zero-crossing
method.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">CP2</span><span class="p">,</span><span class="n">CNTmat</span><span class="p">,</span><span class="n">TOTmat</span><span class="p">,</span><span class="n">LENmat</span><span class="p">]</span> <span class="o">=</span> <span class="n">zero_crossing</span><span class="p">(</span><span class="n">M</span><span class="p">,</span><span class="n">T</span><span class="p">,</span><span class="n">XX</span><span class="p">,</span><span class="n">ZIU</span><span class="p">,</span><span class="n">ZIL</span><span class="p">,</span><span class="n">Z</span><span class="p">,</span><span class="n">mu</span><span class="p">)</span>
</pre></div>
</div>
<p>Used in ewma5.m  See ewma5 for description of variables.</p>
</dd></dl>

</div>
<div class="section" id="image-computation-tools">
<h2>Image_computation_tools<a class="headerlink" href="#image-computation-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Image_computation_tools"></span><dl class="function">
<dt id="Image_computation_tools.apply_derivative_boost">
<code class="descclassname">Image_computation_tools.</code><code class="descname">apply_derivative_boost</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.apply_derivative_boost" title="Permalink to this definition">¶</a></dt>
<dd><p>Allows for recalculation of amplitude images from the fitted responses
for each trial type. Necessary for calculating group statistics when
using multiple basis functions</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function apply_derivative_boost(varargin)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">NOTES:</th><td class="field-body">DB estimation only works for 2 specific basis sets:
timeonly, with 2 parameters (canonical + time derivative)
timedispersion,&#8217; with 3 parameters (canonical hrf + temporal and spatial dispersion)</td>
</tr>
</tbody>
</table>
<p>This function loads the SPM.mat file in the current directory and uses
the basis set specified in the loaded SPM structure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;amplitudes&#8217;</strong></dt>
<dd><p class="first last">will only create amping images (combination of betas
across basis functions)</p>
</dd>
<dt><strong>&#8216;contrasts&#8217;</strong></dt>
<dd><p class="first last">assumes that amping images are already created; will only
create contrast images</p>
</dd>
<dt><strong>&#8216;all&#8217;</strong></dt>
<dd><p class="first">will run both the amplitudes and contrasts sections</p>
<dl class="last docutils">
<dt>In addition, &#8216;amplitudes&#8217; now has two separate parts:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>The first uses Vince Calhoun&#8217;s derivative boost (Calhoun, 2004) to</dt>
<dd><p class="first last">estimate amplitudes.  NOTE: <em>We have not worked out the scaling yet, so
I&#8217;m not sure this is working right</em>
To turn this OFF, enter &#8216;nodb&#8217; as an optional argument</p>
</dd>
</dl>
</li>
<li><p class="first">The second way uses our HTW code to estimate height, time to peak,
width, and area under the curve (see Lindquist &amp; Wager 2007 for
simulations using a version of this code).
It requires SCANlab specific functions, in SCN_Core_Support
(unlike the deriv. boost).
To turn this OFF, enter &#8216;nohtw&#8217; as an optional argument</p>
</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;startend&#8217;</strong></dt>
<dd><p class="first last">followed by starting and ending values in seconds for amplitude
estimation window (for HTW estimation only).
If you do not enter this, it will show you a plot and ask you to pick
these values.
If you enter them here as inputs, you can skip the interactive step and
loop over subjects.</p>
</dd>
<dt><strong>&#8216;condition_numbers&#8217;</strong></dt>
<dd><p class="first last">followed by which index numbers in your list should
be used to calculate h, t, w from.  You should use this if you
are entering regressors of no interest, besides the intercepts.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Important for Contrasts:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">disp(&#8216;Using contrasts entered as F-contrasts. Assuming the first contrast vector in each F-contrast &#8216;</p>
<p>disp(&#8216;is a contrast of interest across the CANONICAL basis function regressors.&#8217;)</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% RUN THIS IN COMMAND WINDOW TO BATCH
subj = dir(&#39;06*&#39;)
for i = 1:length(subj), cd(subj(i).name), apply_derivative_boost, cd(&#39;..&#39;); end

% ANOTHER BATCH EXAMPLE:
d = dir(&#39;remi*&#39;); d = d(cat(2, d.isdir)); [mydirs{1:length(d)}] = deal(d.name)
for i = 1:length(mydirs), cd(mydirs{i}), apply_derivative_boost(&#39;all&#39;, &#39;nodb&#39;, &#39;startend&#39;, [4 15]), cd(&#39;..&#39;); end

%An example for an event-related design, specifying condition numbers to get HTW from:
apply_derivative_boost(&#39;all&#39;, &#39;nodb&#39;, &#39;contrasts&#39;, &#39;condition_numbers&#39;, 1:14, &#39;startend&#39;, [4 10]);

% CALCULATE CONTRASTS ONLY ON ALREADY-ESTIMATED HTW IMAGES
apply_derivative_boost(&#39;contrasts&#39;,&#39;condition_numbers&#39;,1:14);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.canlab_create_wm_ventricle_masks">
<code class="descclassname">Image_computation_tools.</code><code class="descname">canlab_create_wm_ventricle_masks</code><span class="sig-paren">(</span><em>wm_mask</em>, <em>gm_mask</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.canlab_create_wm_ventricle_masks" title="Permalink to this definition">¶</a></dt>
<dd><p>This function saves white matter and ventricle masks.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function canlab_create_wm_ventricle_masks(wm_mask, gm_mask)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>wm_mask:</strong></dt>
<dd><p class="first">white matter structural image file</p>
<div class="last highlight-python"><div class="highlight"><pre><span class="n">wm_mask</span> <span class="o">=</span> <span class="n">filenames</span><span class="p">(</span><span class="s">&#39;Structural/SPGR/wc2*.nii&#39;</span><span class="p">,</span> <span class="s">&#39;char&#39;</span><span class="p">,</span> <span class="s">&#39;absolute&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd>
<dt><strong>gm_mask:</strong></dt>
<dd><p class="first">gray matter structural image file</p>
<div class="last highlight-python"><div class="highlight"><pre><span class="n">gm_mask</span> <span class="o">=</span> <span class="n">filenames</span><span class="p">(</span><span class="s">&#39;Structural/SPGR/wc1*.nii&#39;</span><span class="p">,</span> <span class="s">&#39;char&#39;</span><span class="p">,</span> <span class="s">&#39;absolute&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Optional:</th><td class="field-body"><p class="first">You can specify how liberal or conservative to be in estimating white
matter and ventricles. 1 is most conservative and will yield no
ventricles, and 0 is very liberal. The default of wm_thr is .9, the
default of vent_thr is .9. 
e.g)</p>
<blockquote>
<div><ul class="simple">
<li>&#8216;wm_thr&#8217;, .99</li>
<li>&#8216;vent_thr&#8217;, .95</li>
</ul>
</div></blockquote>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8220;white_matter.img&#8221; and &#8220;ventricles.img&#8221;:</strong></dt>
<dd><p class="first last">in the same folder of the input structural files</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.fisherp">
<code class="descclassname">Image_computation_tools.</code><code class="descname">fisherp</code><span class="sig-paren">(</span><em>p</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.fisherp" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">z</span><span class="p">,</span><span class="n">p</span><span class="p">,</span><span class="n">sig</span><span class="p">,</span><span class="n">pt</span><span class="p">]</span> <span class="o">=</span> <span class="n">fisherp</span><span class="p">(</span><span class="n">p</span><span class="p">,[</span><span class="n">alph</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>p:</strong></dt>
<dd><p class="first">values in 4-D array</p>
<p class="last">1st 3 dims are within images, dim4 = image</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Optional:</th><td class="field-body"><p class="first">alpha value for thresholding</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>z:</strong></dt>
<dd><p class="first last">Fisher&#8217;s combined test statistic, compare to normal</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">p-values for combined test</p>
</dd>
<dt><strong>sig:</strong></dt>
<dd><p class="first last">signficance 1 / 0 binary mask, p &lt; .05 (or alph) FDR-corr</p>
</dd>
<dt><strong>pt:</strong></dt>
<dd><p class="first last">p-value threshold for FDR corrected significance at alph</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Described in:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Lazar, N. A., Luna, B., Sweeney, J. A., &amp; Eddy, W. F. (2002). 
Combining brains: a survey of methods for statistical pooling 
of information. Neuroimage, 16(2), 538-550.</p>
<p>Stouffer, S. A., Suchman, E. A., DeVinney, L. C., Star, S. A., and
Williams, R. M. 1949. The American Soldier: Vol. I. Adjustment
During Army Life. Princeton University Press, Princeton.</p>
<p>Threshold is determined with False Discovery Rate (Benjamini &amp; Hochberg, 1995)</p>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.get_mask_vol">
<code class="descclassname">Image_computation_tools.</code><code class="descname">get_mask_vol</code><span class="sig-paren">(</span><em>Pf</em>, <em>Pm</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.get_mask_vol" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns 3-D volume of binary mask values,
in the space of the functional image Pf.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function mask = get_mask_vol(Pf,Pm)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Pf:</strong></dt>
<dd><p class="first last">filename of functional image</p>
</dd>
<dt><strong>Pm:</strong></dt>
<dd><p class="first last">filename of mask image (should be 1&#8217;s and 0&#8217;s)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.image_eval_function">
<code class="descclassname">Image_computation_tools.</code><code class="descname">image_eval_function</code><span class="sig-paren">(</span><em>imageNames</em>, <em>fhandle</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.image_eval_function" title="Permalink to this definition">¶</a></dt>
<dd><p>Evaluate any function, defined by the function handle fhandle,
on each in-mask voxel for a set of images.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">varargout</span> <span class="o">=</span> <span class="n">image_eval_function</span><span class="p">(</span><span class="n">imgnames</span><span class="p">,</span> <span class="n">fhandle</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;mask&#39;</span><span class="p">,</span> <span class="n">mask</span><span class="p">],</span> <span class="p">[</span><span class="s">&#39;preprochandle&#39;</span><span class="p">,</span> <span class="n">preprochandle</span><span class="p">],</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Other Optional args:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body">&#8216;outimagelabels&#8217; , &#8216;connames&#8217;</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">varargout</span> <span class="o">=</span> <span class="n">image_eval_function</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">fhandle</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>evaluate fhandle on paired columns of X and Y</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body">You must call image_eval_function with outputs, one output for</td>
</tr>
</tbody>
</table>
<p>each output you&#8217;re requesting from the voxel-level function.
Eg:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">t</span><span class="p">,</span> <span class="n">df</span><span class="p">,</span> <span class="n">betaorcontrast</span><span class="p">,</span> <span class="n">Phi</span><span class="p">,</span> <span class="n">sigma</span><span class="p">,</span> <span class="n">stebeta</span><span class="p">,</span> <span class="n">F</span><span class="p">,</span> <span class="n">pvals</span><span class="p">]</span> <span class="o">=</span> <span class="o">...</span>
<span class="n">image_eval_function</span><span class="p">(</span><span class="n">imgs</span><span class="p">,</span> <span class="n">fhandle</span><span class="p">,</span> <span class="s">&#39;mask&#39;</span><span class="p">,</span> <span class="n">maskimg</span><span class="p">,</span> <span class="s">&#39;outimagelabels&#39;</span> <span class="p">,</span> <span class="n">names</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>fhandle:</strong></dt>
<dd><p class="first">is a function handle:</p>
<div class="last highlight-python"><div class="highlight"><pre>fhandle = @(variable inputs) fit_gls(variable_input, fixed_inputs);

fhandle = @(y) fit_gls(y, X, c, p, PX);
</pre></div>
</div>
</dd>
<dt><strong>&#8216;outimagelabels&#8217;:</strong></dt>
<dd><p class="first last">should be followed by a set of image names, one name
per output of fhandle per element. e.g., outnames{j}{i} is the output
image for output j and element (input image) i.  elements may be images for each
subject, if a set of one image per subject is entered, or something
else depending on the nature of input imgs.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Note: do not include suffixes: no .img</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Generalized least squares fitting on 100 Y-variables, same X</p>
<div class="highlight-python"><div class="highlight"><pre>% Get image list
imgs = filenames(&#39;trial_height*img&#39;, &#39;char&#39;)
imgs = sort_image_filenames(imgs)

% Get pre-stored design matrix
X = eventdesign{3};

preprochandle = @(y) trimts(y, 3, []);
</pre></div>
</div>
<p>Generate an image with the number of in-analysis (valid)
subjects in each voxel</p>
<p>EXPT.SNPM.P{2} is a list of subject-level contrast images.</p>
<div class="highlight-python"><div class="highlight"><pre>fhan = @(y) sum(abs(y) &gt; 0 &amp; ~isnan(y));
y = image_eval_function(EXPT.SNPM.P{2}, fhan, &#39;mask&#39;, EXPT.mask, &#39;outimagelabels&#39;, {{&#39;sum_valid_antic.img&#39;}});

y = rand(100, 100); X = y + rand(100, 1); X(:,end+1) = 1; c = [1 0]&#39;; p = 2; PX = pinv(X);
fhandle = @(y) fit_gls(y, X, c, p, PX);
[t, df, beta, Phi, sigma, stebeta, F] = fhandle(y);
</pre></div>
</div>
<p>setup inputs</p>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.image_eval_function_multisubj">
<code class="descclassname">Image_computation_tools.</code><code class="descname">image_eval_function_multisubj</code><span class="sig-paren">(</span><em>imageNames</em>, <em>fhandle</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.image_eval_function_multisubj" title="Permalink to this definition">¶</a></dt>
<dd><p>Evaluate any function, defined by the function handle fhandle,
on each in-mask voxel for a set of images.
imageNames is a cell array of N cells, each containing images for one
replication (i.e., subject)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">image_eval_function_multisubj</span><span class="p">(</span><span class="n">imgnames</span><span class="p">,</span><span class="n">fhandle</span><span class="p">,[</span><span class="s">&#39;mask&#39;</span><span class="p">,</span><span class="n">mask</span><span class="p">],[</span><span class="s">&#39;preprochandle&#39;</span><span class="p">,</span><span class="n">preprochandle</span><span class="p">],[</span><span class="s">&#39;outnames&#39;</span><span class="p">,</span><span class="n">outimagelabels</span><span class="p">],</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>other optional args: &#8216;outimagelabels&#8217; , &#8216;connames&#8217;</p>
<p>At each voxel, a cell array is formed, one cell per subject.
This would correspond to a matrix is formed of t x N, where t is time and N is
replication (subject) but cells can deal with unequal data vector lengths for each subject.
The anonymous function in fhandle should operate on data from each cell (subject).</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">varargout</span> <span class="o">=</span> <span class="n">image_eval_function</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span><span class="n">fhandle</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>evaluate fhandle on paired columns of X and Y</p>
<p>fhandle is a function handle:</p>
<div class="highlight-python"><div class="highlight"><pre>fhandle = @(variable inputs) fit_gls(variable_input,fixed_inputs);
fhandle = @(y) fit_gls(y,X,c,p,PX);
</pre></div>
</div>
<p>specify the outputs by adding them as output image names.
The number of outputs returned is determined by the number of named
images in the list entered following the &#8216;outnames&#8217; keyword.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Note on output images:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body">You specify the names of the output images</td>
</tr>
</tbody>
</table>
<p>One image will be written per output of fhandle.
The images will have one volume per element of the output variable.
If you are returning an output with one value per subject, for
example, then a single image will be written with one volume in it
per subject.</p>
<p>preprochandle is a function handle.
it encapsulates the preprocessing function.
the function should work on each cell (subject) of a t x N cell array of time courses for each
subject, where each  cell contains a t x v matrix of data from a
slice. The preproc function should thus be able to handle a whole slice as
input.
the function can itself be a cell array with multiple handles in
different cells</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">Generalized least squares fitting on 100 Y-variables, same X</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Get image list
imgs = filenames(&#39;trial_height*img&#39;,&#39;char&#39;)
imgs = sort_image_filenames(imgs)

% Get pre-stored design matrix
X = eventdesign{3};

preprochandle = @(y) trimts(y,3,[]);

y = rand(100,100); X = y + rand(100,1); X(:,end+1) = 1; c = [1 0]&#39;; p = 2; PX = pinv(X);
fhandle = @(y) fit_gls(y,X,c,p,PX);
[t, df, beta, Phi, sigma,stebeta, F] = fhandle(y);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.image_histogram1d">
<code class="descclassname">Image_computation_tools.</code><code class="descname">image_histogram1d</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.image_histogram1d" title="Permalink to this definition">¶</a></dt>
<dd><p>Visualize a change_point map stored in hewma_cp.img
(output from hewma2)
and classify voxels into groupings based on CP</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[cl2,classes] = image_histogram1d([image name],[overlay])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.mask_create_from_image_set">
<code class="descclassname">Image_computation_tools.</code><code class="descname">mask_create_from_image_set</code><span class="sig-paren">(</span><em>imgs</em>, <em>outname</em>, <em>atleastN</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.mask_create_from_image_set" title="Permalink to this definition">¶</a></dt>
<dd><p>Take a set of images and create a mask of voxels in which at least N
subjects have valid (not exactly zero, non NaN) data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">mask</span> <span class="o">=</span> <span class="n">mask_create_from_image_set</span><span class="p">(</span><span class="n">imgs</span><span class="p">,</span> <span class="n">outname</span><span class="p">,</span> <span class="n">atleastN</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;sum&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>This makes a useful results mask for a set of images, i.e., in a
group analysis.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Optional:</th><td class="field-body">&#8216;sum&#8217; input writes the sum image instead of the mask image,</td>
</tr>
</tbody>
</table>
<p>so that the values in the image reflect the number of input images
with valid values.</p>
<p>compatible with SPM5 and above only!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>mask_create_from_image_set(EXPT.SNPM.P{1}, &#39;mask_all_valid.img&#39;);

imgs = filenames(&#39;vascular_mask_*img&#39;);
mask_create_from_image_set(imgs, &#39;vascular_group_sum.img&#39;, 6, &#39;sum&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.mask_create_results_mask">
<code class="descclassname">Image_computation_tools.</code><code class="descname">mask_create_results_mask</code><span class="sig-paren">(</span><em>mask</em>, <em>study_img</em>, <em>kern_size</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.mask_create_results_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Takes a mask image name (mask)
smooths it by smooth_mm (optional; enter 0 for kern_size to avoid this)
Resamples it to study image dimensions (study_img)
writes output image</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[cl_out, out_name] = mask_create_results_mask(mask, study_img, kern_size, [opt: expression to evaluate on mask])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Optional:</th><td class="field-body">Applies an expression to be evaluated to the image,</td>
</tr>
</tbody>
</table>
<p>in spm_imcalc format, e.g., &#8216;i1 &lt; .05&#8217;  (uses spm_imcalc_ui.m)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body">Returns mask_clusters (mask_cl) and mask name (out_name)</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>mask = which(&#39;spm2_amy.img&#39;)
study_img = &#39;/Users/tor/Documents/Tor_Documents/PublishedProjects/inpress_2007_Emotion_handbook_2006/an_metaFWE_rad10/exp_vs_percept/Activation_FWE_all.img&#39;;
kern_size = 3;
cl_out = mask_create_results_mask(mask, study_img, kern_size);
cluster_orthviews(cl_out, {[0 1 0]}, &#39;add&#39;, &#39;handle&#39;, 1);

[cl_out, out_name] = mask_create_results_mask(&#39;X-Y_total_pvals.img&#39;, &#39;X-Y_total_pvals.img&#39;, 0, &#39;i1 &lt; .05 &amp; i1 &gt; eps&#39;);
spm_image(&#39;init&#39;, &#39;X-Y_total_pvals.img&#39;);
cluster_orthviews(cl_out, {[1 0 0]}, &#39;trans&#39;, &#39;add&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.mask_fisher">
<code class="descclassname">Image_computation_tools.</code><code class="descname">mask_fisher</code><span class="sig-paren">(</span><em>clsize</em>, <em>outname</em>, <em>k</em>, <em>Pimg</em>, <em>Timg</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.mask_fisher" title="Permalink to this definition">¶</a></dt>
<dd><p>This function performs combination of p-values across images using the 
Fisher method (see refs below).  Voxels are thresholded with FDR, and can
be positive and negative at once if both positive and negative sig
effects exist in input images.  FDR correction is based on 2-tailed
p-values - so this function assumes 2-tailed p-values!!  (robfit does
this, glmfit and robustfit both return 2-tailed p-values as well.)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">clpos</span><span class="p">,</span><span class="n">clneg</span><span class="p">]</span> <span class="o">=</span> <span class="n">mask_fisher</span><span class="p">(</span><span class="n">clsize</span><span class="p">,</span><span class="n">outname</span><span class="p">,</span><span class="n">k</span><span class="p">,</span><span class="n">Pimg</span><span class="p">,</span><span class="n">Timg</span><span class="p">)</span>
</pre></div>
</div>
<p>Assumes input p-values are 1-tailed, and divides by two to get 2-tailed
p-value inputs!!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Pimg:</strong></dt>
<dd><p class="first">string mtx of p-value image names</p>
<div class="last highlight-python"><div class="highlight"><pre><span class="n">Pimg</span><span class="o">=</span><span class="n">get_filename2</span><span class="p">(</span><span class="s">&#39;rob*\*_p_*0002.img&#39;</span><span class="p">);</span> <span class="n">Timg</span> <span class="o">=</span> <span class="n">get_filename2</span><span class="p">(</span><span class="s">&#39;rob*\*_tmap_0002.img&#39;</span><span class="p">)</span>
</pre></div>
</div>
</dd>
<dt><strong>Timg:</strong></dt>
<dd><p class="first last">t-value img names, used to ensure signs are same across all tests</p>
</dd>
<dt><strong>k:</strong></dt>
<dd><p class="first last">num voxels/num comparisons <a href="#id27"><span class="problematic" id="id28">**</span></a>NOT USED NOW - FDR USED INSTEAD
if empty, uses # of non-zero, non-NaN values in images</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>empty i1 prompts for graphic selection of filenames
extra arguments are more file names for 3 - n-way intersection
empty outname prompts for entry of output img file name</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Described in:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Lazar, N. A., Luna, B., Sweeney, J. A., &amp; Eddy, W. F. (2002). 
Combining brains: a survey of methods for statistical pooling 
of information. Neuroimage, 16(2), 538-550.</p>
<p>Stouffer, S. A., Suchman, E. A., DeVinney, L. C., Star, S. A., and
Williams, R. M. 1949. The American Soldier: Vol. I. Adjustment
During Army Life. Princeton University Press, Princeton.</p>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.mask_image">
<code class="descclassname">Image_computation_tools.</code><code class="descname">mask_image</code><span class="sig-paren">(</span><em>img</em>, <em>mask</em>, <em>outname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.mask_image" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">masked_dat</span> <span class="o">=</span> <span class="n">mask_image</span><span class="p">(</span><span class="n">img</span><span class="p">,</span> <span class="n">mask</span><span class="p">,</span> <span class="n">outname</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;reverse&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>Mask an image file (img) with a mask (mask), and save in outname.
zero and NaN values are considered invalid values, and so voxels with
these values are considered &#8220;excluded&#8221;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;minmask&#8217;:</strong></dt>
<dd><p class="first last">mask values less than next argument are excluded</p>
</dd>
<dt><strong>&#8216;maxmask&#8217;:</strong></dt>
<dd><p class="first last">mask values greater than next argument are excluded</p>
</dd>
<dt><strong>&#8216;minimg&#8217;:</strong></dt>
<dd><p class="first last">img values less than next argument are excluded</p>
</dd>
<dt><strong>&#8216;maximg&#8217;:</strong></dt>
<dd><p class="first last">img values greater than next argument are excluded</p>
</dd>
<dt><strong>&#8216;abs&#8217;:</strong></dt>
<dd><p class="first last">impose min/max thresholds based on absolute values</p>
</dd>
<dt><strong>&#8216;reverse&#8217;:</strong></dt>
<dd><blockquote class="first">
<div><p>make &#8220;reverse mask,&#8221; including only previously excluded
areas (values of zero or NaN)</p>
<p>Note: applies to mask, not img values
So values with 0 in img will always be 0, whether</p>
</div></blockquote>
<p class="last">standard or &#8220;reverse&#8221; mask is used.</p>
</dd>
<dt><strong>&#8216;binary&#8217;:</strong></dt>
<dd><p class="first last">make the image values binary (i.e., create a new mask)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This function can handle images of different dimensions.  The output
image will use the dimensions of img.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Create an image with non-zero numbers only where p-values in an image are greater than .05
img = &#39;X-M_pvals.img&#39;
mask = &#39;X-M_pvals.img&#39;;
maxmask = .05
outname = &#39;notX_p05.img&#39;;
mask_image(img, mask, outname, &#39;reverse&#39;, &#39;maxmask&#39;, maxmask);
spm_image(&#39;init&#39;, outname);

mask_image(my_mean_image, &#39;functional_mask.img&#39;, ...
              &#39;functional_mask.img&#39;, &#39;minimg&#39;, cutoff, &#39;abs&#39;);


mask_image(&#39;n15_avgpet.img&#39;,EXPT.mask,&#39;n15_avgpet_brain.img&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.mask_intersection">
<code class="descclassname">Image_computation_tools.</code><code class="descname">mask_intersection</code><span class="sig-paren">(</span><em>clsize</em>, <em>outname</em>, <em>i1</em>, <em>i2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.mask_intersection" title="Permalink to this definition">¶</a></dt>
<dd><p>empty i1 prompts for graphic selection of filenames
extra arguments are more file names for 3 - n-way intersection
empty outname prompts for entry of output img file name</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">vol</span><span class="p">,</span><span class="n">V</span><span class="p">,</span><span class="n">XYZ</span><span class="p">,</span><span class="n">clusters</span><span class="p">,</span><span class="n">Q</span><span class="p">]</span> <span class="o">=</span> <span class="n">mask_intersection</span><span class="p">(</span><span class="n">clsize</span><span class="p">,</span><span class="n">outname</span><span class="p">,</span><span class="n">i1</span><span class="p">,</span><span class="n">i2</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.mask_intersection2">
<code class="descclassname">Image_computation_tools.</code><code class="descname">mask_intersection2</code><span class="sig-paren">(</span><em>clsize</em>, <em>outname</em>, <em>P</em>, <em>calcstr</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.mask_intersection2" title="Permalink to this definition">¶</a></dt>
<dd><p>empty P prompts for graphic selection of filenames
extra arguments are more file names for 3 - n-way intersection
empty outname prompts for entry of output img file name</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">clusters</span><span class="p">,</span><span class="n">vol</span><span class="p">,</span><span class="n">V</span><span class="p">,</span><span class="n">XYZ</span><span class="p">,</span><span class="n">Q</span><span class="p">]</span> <span class="o">=</span> <span class="n">mask_intersection2</span><span class="p">(</span><span class="n">clsize</span><span class="p">,</span><span class="n">outname</span><span class="p">,</span><span class="n">P</span><span class="p">,</span><span class="n">calcstr</span><span class="p">)</span>
</pre></div>
</div>
<p>this version allows contrasts
last argument, calcstr, is the string to evaluate in imcalc</p>
<p>e.g., intersection of 4 images:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="s">&#39;i1 &amp; i2 &amp; i3 &amp; i4 &amp; ~(isnan(i1) | isnan(i2) | isnan(i3) | isnan(i4))&#39;</span>
</pre></div>
</div>
<p>e.g.,</p>
<div class="highlight-python"><div class="highlight"><pre>&#39;i1 &amp; i2 &amp; ~i3 &amp; ~i4 &amp; ~(isnan(i1) | isnan(i2)) = intersection of i1 and i2 and not i3 and not i4
</pre></div>
</div>
<p>or</p>
<div class="highlight-python"><div class="highlight"><pre>  &#39;i1 &amp; i2 &amp; isnan(i3) &amp; isnan(i4) &amp; ~(isnan(i1) | isnan(i2))&#39;
... for nan masked images

   cl = mask_intersection2(5,&#39;intersect001.img&#39;,P,&#39;i1 &amp; i2 &amp; i3 &amp; i4 &amp; ~(isnan(i1) | isnan(i2) | isnan(i3) | isnan(i4))&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.mask_stouffer">
<code class="descclassname">Image_computation_tools.</code><code class="descname">mask_stouffer</code><span class="sig-paren">(</span><em>clsize</em>, <em>outname</em>, <em>k</em>, <em>Pimg</em>, <em>Timg</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.mask_stouffer" title="Permalink to this definition">¶</a></dt>
<dd><p>This function performs combination of p-values across images using the 
Stouffer method (see refs below).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">vol</span><span class="p">,</span><span class="n">V</span><span class="p">,</span><span class="n">XYZ</span><span class="p">,</span><span class="n">clusters</span><span class="p">,</span><span class="n">Q</span><span class="p">]</span> <span class="o">=</span> <span class="n">mask_stouffer</span><span class="p">(</span><span class="n">clsize</span><span class="p">,</span><span class="n">outname</span><span class="p">,</span><span class="n">k</span><span class="p">,</span><span class="n">Pimg</span><span class="p">,</span><span class="n">Timg</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Pimg:</strong></dt>
<dd><p class="first last">string mtx of p-value image names</p>
</dd>
<dt><strong>Timg:</strong></dt>
<dd><p class="first last">t-value img names, used to ensure signs are same across all tests</p>
</dd>
<dt><strong>k:</strong></dt>
<dd><p class="first last">num voxels/num comparisons 
if empty, uses # of non-zero, non-NaN values in images</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>empty i1 prompts for graphic selection of filenames
extra arguments are more file names for 3 - n-way intersection
empty outname prompts for entry of output img file name</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Described in:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Lazar, N. A., Luna, B., Sweeney, J. A., &amp; Eddy, W. F. (2002). 
Combining brains: a survey of methods for statistical pooling 
of information. Neuroimage, 16(2), 538-550.</p>
<p>Stouffer, S. A., Suchman, E. A., DeVinney, L. C., Star, S. A., and
Williams, R. M. 1949. The American Soldier: Vol. I. Adjustment
During Army Life. Princeton University Press, Princeton.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">Pimg</span><span class="o">=</span><span class="n">get_filename2</span><span class="p">(</span><span class="s">&#39;rob*\*_p_*0002.img&#39;</span><span class="p">);</span>
<span class="n">Timg</span> <span class="o">=</span> <span class="n">get_filename2</span><span class="p">(</span><span class="s">&#39;rob*\*_tmap_0002.img&#39;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.mask_union">
<code class="descclassname">Image_computation_tools.</code><code class="descname">mask_union</code><span class="sig-paren">(</span><em>clsize</em>, <em>outname</em>, <em>i1</em>, <em>i2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.mask_union" title="Permalink to this definition">¶</a></dt>
<dd><p>empty i1 prompts for graphic selection of filenames
extra arguments are more file names for 3 - n-way intersection
empty outname prompts for entry of output img file name</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">vol</span><span class="p">,</span><span class="n">V</span><span class="p">,</span><span class="n">XYZ</span><span class="p">,</span><span class="n">clusters</span><span class="p">,</span><span class="n">Q</span><span class="p">]</span> <span class="o">=</span> <span class="n">mask_union</span><span class="p">(</span><span class="n">clsize</span><span class="p">,</span><span class="n">outname</span><span class="p">,</span><span class="n">i1</span><span class="p">,</span><span class="n">i2</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.percent_sig_image">
<code class="descclassname">Image_computation_tools.</code><code class="descname">percent_sig_image</code><span class="sig-paren">(</span><em>imgs</em>, <em>baseimg</em>, <em>outname</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.percent_sig_image" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a percent signal change image saved in outname by dividing each
image by baseimg</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function Vo = percent_sig_image(imgs, baseimg, outname)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.reslice_imgs">
<code class="descclassname">Image_computation_tools.</code><code class="descname">reslice_imgs</code><span class="sig-paren">(</span><em>sampleTo</em>, <em>resliceThis</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.reslice_imgs" title="Permalink to this definition">¶</a></dt>
<dd><p>arguments are file names of .img files
if empty, select from GUI% :Usage:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">P</span><span class="p">,</span> <span class="n">reslicedImgs</span><span class="p">]</span> <span class="o">=</span> <span class="n">reslice_imgs</span><span class="p">(</span><span class="n">sampleTo</span><span class="p">,</span> <span class="n">resliceThis</span><span class="p">,</span> <span class="p">[</span><span class="n">domask</span><span class="p">],</span> <span class="p">[</span><span class="n">overwrite</span><span class="p">])</span>
</pre></div>
</div>
<p>if domask, recalculates a 1 or 0 mask for each image file in resliceThis
if overwrite, overwrite the original instead of prepending an &#8216;r&#8217;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Reslice a mask image into the space of some functional images, and move to the current directory

[tmp, maskname] = reslice_imgs(image_names(1, :), maskname, 1);
eval([&#39;!mv &#39; maskname &#39; ./&#39;])
eval([&#39;!mv &#39; maskname(1:end-4) &#39;.hdr ./&#39;])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.reverse_mask">
<code class="descclassname">Image_computation_tools.</code><code class="descname">reverse_mask</code><span class="sig-paren">(</span><em>inname</em>, <em>outname</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.reverse_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Changes 1 and greater&#8217;s to 0&#8217;s in an .img file, and vice versa
NaNs are still NaNs.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">reverse_mask</span><span class="p">(</span><span class="n">inname</span><span class="p">,</span><span class="n">outname</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.scn_write_plane">
<code class="descclassname">Image_computation_tools.</code><code class="descname">scn_write_plane</code><span class="sig-paren">(</span><em>filenames_or_V</em>, <em>dat</em>, <em>wh_slice</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.scn_write_plane" title="Permalink to this definition">¶</a></dt>
<dd><p>Images can be 3D or 4D (4D is possible with SPM2+ using ,xx indexing, or SPM5+).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">V</span> <span class="o">=</span> <span class="n">scn_write_plane</span><span class="p">(</span><span class="n">filenames_or_V</span><span class="p">,</span> <span class="n">dat</span><span class="p">,</span> <span class="n">wh_slice</span><span class="p">,</span> <span class="p">[</span><span class="n">exampleV</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>ofilenames_or_V:</strong></dt>
<dd><p class="first last">is string matrix or cell array of names, or structures created with spm_vol(names)</p>
</dd>
<dt><strong>dat:</strong></dt>
<dd><p class="first last">is 3-D array of data, vox x vox x slices, with data on 3rd dim being
written to separate image files</p>
</dd>
<dt><strong>wh_slice:</strong></dt>
<dd><p class="first last">slice number (voxel space)</p>
</dd>
<dt><strong>exampleV:</strong></dt>
<dd><p class="first last">is optional, but required if filenames are passed in.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Write a plane, given filenames or spm_vol structures and 4-D data with
the slice to write.</p>
<p>SPM2/5 compatible, and creates image if necessary from file name and
example V structure.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">P</span> <span class="o">=</span> <span class="n">char</span><span class="p">({</span><span class="s">&#39;one.img&#39;</span><span class="p">,</span> <span class="s">&#39;two.img&#39;</span><span class="p">,</span> <span class="s">&#39;three.img&#39;</span><span class="p">})</span>
<span class="n">dat</span> <span class="o">=</span> <span class="n">randn</span><span class="p">(</span><span class="mi">64</span><span class="p">,</span> <span class="mi">64</span><span class="p">,</span> <span class="mi">3</span><span class="p">);</span>
<span class="n">Vout</span> <span class="o">=</span> <span class="n">scn_write_plane</span><span class="p">(</span><span class="n">P</span><span class="p">,</span> <span class="n">dat</span><span class="p">,</span> <span class="n">wh_slice</span><span class="p">,</span> <span class="n">V</span><span class="p">)</span>
<span class="n">spm_image</span><span class="p">(</span><span class="s">&#39;init&#39;</span><span class="p">,</span> <span class="n">Vout</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">fname</span><span class="p">);</span>
</pre></div>
</div>
<p>Notes on SPM5 and why we need to do what we do the way we do it:
Write data...in SPM5, by accessing file_array object in
V.private directly (with spm_write_plane. spm_write_vol 
first creates NIFTI obj/file array and then uses spm_write_plane.)</p>
<p>The file_array object points to data in
the actual file, so when values are assigned to the array
object, they are written directly in the file.
These values depend on the offset and slope values (spm
scaling factors!) in V.private.dat as well, so care must be
taken to assign data to a file_array object with the correct
scaling factors.  This is why it is better to load the
structure one wants to write to with spm_vol first, so that
the name in V.private.dat.fname will be correct, and
V.private.dat.scl_slope and ...inter will be correct as well.
Vout = spm_vol(V(i));  % loads correct .private info from V.fname</p>
<p>in SPM5, this simply assigns data in Vout.private.dat
in SPM2, it does something different, but should be
compatible, since spm_vol was used above...</p>
<div class="highlight-python"><div class="highlight"><pre>spm_write_plane(Vout, fsl(:, :, i), slicei);
if isfield(V, &#39;dt&#39;), Vout.dt = V.dt; end      % SPM5 only
if isfield(V, &#39;n&#39;), Vout.n = V.n;  end        % SPM5 only
    Vout = spm_create_vol(Vout);
else
    Vout = spm_vol(V(i).fname);
end
spm_write_plane(Vout, fsl(:, :, i), slicei);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_computation_tools.tor_spm_mean_ui">
<code class="descclassname">Image_computation_tools.</code><code class="descname">tor_spm_mean_ui</code><span class="sig-paren">(</span><em>P</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_computation_tools.tor_spm_mean_ui" title="Permalink to this definition">¶</a></dt>
<dd><p>Promts for a series of images and averages them</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">tor_spm_mean_ui</span><span class="p">(</span><span class="n">Pinputnames</span><span class="p">,[</span><span class="n">outputname</span><span class="p">])</span>
</pre></div>
</div>
</dd></dl>

</div>
<div class="section" id="image-space-tools">
<h2>Image_space_tools<a class="headerlink" href="#image-space-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Image_space_tools"></span><dl class="function">
<dt id="Image_space_tools.check_spm_mat">
<code class="descclassname">Image_space_tools.</code><code class="descname">check_spm_mat</code><span class="sig-paren">(</span><em>mat1</em>, <em>mat2</em>, <em>clusters</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.check_spm_mat" title="Permalink to this definition">¶</a></dt>
<dd><p>mat1 is from clusters, mat2 is functional (imgs to extract)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">check_spm_mat</span><span class="p">(</span><span class="n">mat1</span><span class="p">,</span><span class="n">mat2</span><span class="p">,</span><span class="n">clusters</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.check_spm_matfiles">
<code class="descclassname">Image_space_tools.</code><code class="descname">check_spm_matfiles</code><span class="sig-paren">(</span><em>P</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.check_spm_matfiles" title="Permalink to this definition">¶</a></dt>
<dd><p>Check a list of SPM-style images to see if they all have the same
space, dims.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">anybad</span><span class="p">]</span> <span class="o">=</span> <span class="n">check_spm_matfiles</span><span class="p">(</span><span class="n">P</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.img2voxel">
<code class="descclassname">Image_space_tools.</code><code class="descname">img2voxel</code><span class="sig-paren">(</span><em>P</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.img2voxel" title="Permalink to this definition">¶</a></dt>
<dd><p>Given a mask or filtered image file name, 
returns XYZ coordinates in voxels and mm
of nonzero, non-NaN voxels</p>
<p>and img values at these coordinates in val</p>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.mask2voxel">
<code class="descclassname">Image_space_tools.</code><code class="descname">mask2voxel</code><span class="sig-paren">(</span><em>mask</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.mask2voxel" title="Permalink to this definition">¶</a></dt>
<dd><p>convert from 3-D mask to voxel list in canonical orientation</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function voxels = mask2voxel(mask)
</pre></div>
</div>
<p>[i j k] = row, column, slice</p>
<p>[x y z] in brain if brain is in analyze format</p>
<p>(x is rows, y is columns, z is slices)</p>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.mni2tal">
<code class="descclassname">Image_space_tools.</code><code class="descname">mni2tal</code><span class="sig-paren">(</span><em>inpoints</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.mni2tal" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts coordinates from MNI brain to best guess
for equivalent Talairach coordinates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">outpoints</span> <span class="o">=</span> <span class="n">mni2tal</span><span class="p">(</span><span class="n">inpoints</span><span class="p">)</span>
</pre></div>
</div>
<p>Where inpoints is N by 3 or 3 by N matrix of coordinates
(N being the number of points)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>outpoints:</strong></dt>
<dd><p class="first last">is the coordinate matrix with Talairach points</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.scn_map_image">
<code class="descclassname">Image_space_tools.</code><code class="descname">scn_map_image</code><span class="sig-paren">(</span><em>loadImg</em>, <em>sampleTo</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.scn_map_image" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes an image name in loadImg
and loads the data, resampling to the space defined
in the image sampleTo. The resampled image will retain
the data type of the input image.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">imgData</span><span class="p">,</span> <span class="n">volInfo_mapto</span><span class="p">]</span> <span class="o">=</span> <span class="n">scn_map_image</span><span class="p">(</span><span class="n">loadImg</span><span class="p">,</span> <span class="n">sampleTo</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;write&#8217;:</strong></dt>
<dd><p class="first last">followed by name of resampled image to write</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Compatible with SPM5/8.</p>
<dl class="docutils">
<dt>Input images can have the following formats:</dt>
<dd><ol class="first last arabic simple">
<li>String with name of image file (.img or .nii)</li>
<li>spm_vol-style V struct (see spm_vol)</li>
<li>volInfo struct (see iimg_read_img)</li>
<li>fmri_mask_image object (see fmri_mask_image)</li>
</ol>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>img = scn_map_image(EXPT.mask, EXPT.SNPM.P{1}(1,:), &#39;write&#39;, &#39;resliced_mask.img&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.scn_resample_voxel_size">
<code class="descclassname">Image_space_tools.</code><code class="descname">scn_resample_voxel_size</code><span class="sig-paren">(</span><em>loadImg</em>, <em>voxsize</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.scn_resample_voxel_size" title="Permalink to this definition">¶</a></dt>
<dd><p>This function takes an image name in loadImg
and loads the data, resampling to the space defined
in the image sampleTo.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">imgData</span><span class="p">,</span> <span class="n">volInfo_mapto</span><span class="p">]</span> <span class="o">=</span> <span class="n">scn_resample_voxel_size</span><span class="p">(</span><span class="n">loadImg</span><span class="p">,</span> <span class="n">voxsize</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>take volInfo and fmri_mask_image inputs as well
as image file names</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;write&#8217;:</strong></dt>
<dd><p class="first last">followed by name of resampled image to write</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Compatible with SPM5/8.</p>
<dl class="docutils">
<dt>Input images can have the following formats:</dt>
<dd><ol class="first last arabic simple">
<li>String with name of image file (.img or .nii)</li>
<li>spm_vol-style V struct (see spm_vol)</li>
<li>volInfo struct (see iimg_read_img)</li>
<li>fmri_mask_image object (see fmri_mask_image)</li>
</ol>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Reslice standard brain mask to 3 x 3 x 3 voxels.
img = which(&#39;brainmask.nii&#39;);
[dat, Vto] = scn_resample_voxel_size(img, [3 3 3], &#39;write&#39;, &#39;test.img&#39;);
spm_image(&#39;init&#39;, &#39;test.img&#39;);
spm_check_registration(char(img, &#39;test.img&#39;));
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.tal2mni">
<code class="descclassname">Image_space_tools.</code><code class="descname">tal2mni</code><span class="sig-paren">(</span><em>inpoints</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.tal2mni" title="Permalink to this definition">¶</a></dt>
<dd><p>Converts coordinates to MNI brain best guess
from Talairach coordinates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">outpoints</span> <span class="o">=</span> <span class="n">tal2mni</span><span class="p">(</span><span class="n">inpoints</span><span class="p">)</span>
</pre></div>
</div>
<p>Where inpoints is N by 3 or 3 by N matrix of coordinates
(N being the number of points)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>outpoints:</strong></dt>
<dd><p class="first last">is the coordinate matrix with MNI points</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.tal2vox">
<code class="descclassname">Image_space_tools.</code><code class="descname">tal2vox</code><span class="sig-paren">(</span><em>tal</em>, <em>VOL</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.tal2vox" title="Permalink to this definition">¶</a></dt>
<dd><p>converts from talairach coordinate to voxel coordinate
based on variables from SPM.M (passed here for 
faster operation)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>foo = tal2vox([-30 28 -30], VOL)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.transform_coordinates">
<code class="descclassname">Image_space_tools.</code><code class="descname">transform_coordinates</code><span class="sig-paren">(</span><em>CLU</em>, <em>mat</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.transform_coordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Transforms XYZ voxel coordinates in CLU (clusters or CLU)
to new voxel coordinates, given mm coordinates in CLU
and a mat file describing the transformation, as in SPM99</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">CLU</span> <span class="o">=</span> <span class="n">transform_coordinates</span><span class="p">(</span><span class="n">CLU</span><span class="p">,</span><span class="n">mat</span><span class="p">)</span>
</pre></div>
</div>
<p>This preserves the order of the voxels, but is slower
and gives UNIQUE XYZ voxels given XYZmm.
see mm2voxel.m</p>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.voxel2mask">
<code class="descclassname">Image_space_tools.</code><code class="descname">voxel2mask</code><span class="sig-paren">(</span><em>voxels</em>, <em>maskdims</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.voxel2mask" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function mask = voxel2mask(voxels, x y z mask dimensions)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Voxels:</th><td class="field-body"><ul class="first last">
<li><p class="first">3 column vectors</p>
</li>
<li><p class="first">[i j k] = row, column, slice</p>
</li>
<li><p class="first">[x y z] in brain if brain is in analyze format</p>
<p>(x is rows, y is columns, z is slices)</p>
</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Image_space_tools.voxel2mm">
<code class="descclassname">Image_space_tools.</code><code class="descname">voxel2mm</code><span class="sig-paren">(</span><em>XYZ</em>, <em>m</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_space_tools.voxel2mm" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function XYZmm = voxel2mm(XYZ,m)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>XYZ:</strong></dt>
<dd><p class="first last">is 3 vector point list (3 rows, n columns)</p>
</dd>
<dt><strong>m:</strong></dt>
<dd><p class="first last">is SPM mat - 4 x 4 affine transform
(what&#8217;s stored in the .mat file)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>XYZmm = voxel2mm([x y z]&#39;,V.mat);
</pre></div>
</div>
</dd></dl>

</div>
<div class="section" id="image-thresholding">
<h2>Image_thresholding<a class="headerlink" href="#image-thresholding" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Image_thresholding"></span><dl class="function">
<dt id="Image_thresholding.FDR">
<code class="descclassname">Image_thresholding.</code><code class="descname">FDR</code><span class="sig-paren">(</span><em>p</em>, <em>q</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_thresholding.FDR" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">pt</span> <span class="o">=</span> <span class="n">FDR</span><span class="p">(</span><span class="n">p</span><span class="p">,</span><span class="n">q</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>p:</strong></dt>
<dd><p class="first last">vector of p-values</p>
</dd>
<dt><strong>q:</strong></dt>
<dd><p class="first last">False Discovery Rate level</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>pID:</strong></dt>
<dd><p class="first last">p-value threshold based on independence or positive dependence</p>
</dd>
<dt><strong>pN:</strong></dt>
<dd><p class="first last">Nonparametric p-value threshold</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Image_thresholding.cl_ext_3dClustSim">
<code class="descclassname">Image_thresholding.</code><code class="descname">cl_ext_3dClustSim</code><span class="sig-paren">(</span><em>corrected_p</em>, <em>prim_p</em>, <em>residual_images</em>, <em>mask</em>, <em>voxelsize_mm</em>, <em>ClustSim_dir</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_thresholding.cl_ext_3dClustSim" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cl_ext_ClustSim</span><span class="p">,</span> <span class="n">fwhm</span><span class="p">]</span> <span class="o">=</span> <span class="n">cl_ext_3dClustSim</span><span class="p">(</span><span class="n">corrected_p</span><span class="p">,</span> <span class="n">prim_p</span><span class="p">,</span> <span class="n">residual_images</span><span class="p">,</span> <span class="n">mask</span><span class="p">,</span> <span class="n">voxelsize_mm</span><span class="p">,</span> <span class="n">ClustSim_dir</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>corrected_p:</strong></dt>
<dd><p class="first">cluster-extent corrected p value</p>
<p class="last">e.g.) if cluster-extent corrected p &lt; .05: corrected_p = .05</p>
</dd>
<dt><strong>prim_p:</strong></dt>
<dd><p class="first">primary threshold for height (i.e., cluster-defining threshold)</p>
<p class="last">e.g.) prim_p = [0.01 0.005 0.001];</p>
</dd>
<dt><strong>residual_images:</strong></dt>
<dd><p class="first">residual image names; if you used</p>
<p>cl_ext_make_resid.m, this should be &#8216;Res4d.nii&#8217;.</p>
<p>e.g.) residual_images = filenames(&#8216;Res4d.hdr&#8217;, &#8216;char&#8217;, &#8216;absolute&#8217;);</p>
<blockquote class="last">
<div><p>residual_images = filenames(&#8216;Res4d.nii&#8217;, &#8216;char&#8217;, &#8216;absolute&#8217;);</p>
</div></blockquote>
</dd>
<dt><strong>mask:</strong></dt>
<dd><p class="first">mask image name (should have header)</p>
<p>e.g.) mask = filenames(&#8216;mask.hdr&#8217;, &#8216;char&#8217;, &#8216;absolute&#8217;);</p>
<blockquote class="last">
<div><p>mask = filenames(&#8216;mask.nii&#8217;, &#8216;char&#8217;, &#8216;absolute&#8217;);</p>
</div></blockquote>
</dd>
<dt><strong>voxelsize_mm:</strong></dt>
<dd><p class="first last">voxel sizes in milimeter. e.g) voxelsize_mm = [2 2 2];</p>
</dd>
<dt><strong>3dClustSim_dir:</strong></dt>
<dd><p class="first">directory where alphasim is installed.</p>
<p>e.g.) 3dClustSim_dir = &#8216;/Users/clinpsywoo/abin/macosx_10.6_Intel_64&#8217;;</p>
<p class="last">If you don&#8217;t have 3dClustSim, see <a class="reference external" href="http://afni.nimh.nih.gov/pub/dist/HOWTO/howto/ht00_inst/html/index.shtml">http://afni.nimh.nih.gov/pub/dist/HOWTO/howto/ht00_inst/html/index.shtml</a></p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;iter&#8217;:</strong></dt>
<dd><p class="first last">you can set up the iteration number for Monte Carlo simulation.
default is doing 1000 iterations.</p>
</dd>
<dt><strong>&#8216;twotail&#8217;:</strong></dt>
<dd><p class="first last">default is one-tail - with this option, primary_p/2 will be used 
for all clsuter extent estimations.</p>
</dd>
<dt><strong>&#8216;fwhm&#8217;:</strong></dt>
<dd><p class="first last">you can add fwhm manually</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>cl_ext_ClustSim:</strong></dt>
<dd><p class="first last">cl_ext_ClustSim is the cluster size that makes a corrected p value under 
corrected_p (e.g., 0.05).</p>
</dd>
<dt><strong>fwhm (x, y, z in voxel):</strong></dt>
<dd><p class="first last">intrinsic smoothness level estimated by AFNI(3dFWHMx). 
If you want to convert this into mm, you need to multiply these
values by voxel sizes in mm.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Image_thresholding.cl_ext_make_resid">
<code class="descclassname">Image_thresholding.</code><code class="descname">cl_ext_make_resid</code><span class="sig-paren">(</span><em>con_files</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_thresholding.cl_ext_make_resid" title="Permalink to this definition">¶</a></dt>
<dd><p>This function will create residual images (4d) and mask image in 
a current or assigned directory in order to use them in estimating smoothness 
(relevant functions: spm_est_smoothness (SPM), 3dFWHMx (AFNI), smoothest (FSL).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function cl_ext_make_resid(conimgs, varargin)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>con_files:</strong></dt>
<dd><p class="first">contrast image file names; This could be a cell array or
strings. This could be 4d images.</p>
<p>Best: Input a cell string. e.g., for a string matrix:</p>
<p class="last">Use cl_ext_make_resid(cellstr(imgs)); % save residual images</p>
</dd>
</dl>
<ul class="simple">
<li>If you are not providing the absolute paths of the images, you need to</li>
</ul>
<p>be in the directory that has the image files.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Res4d.nii:</strong></dt>
<dd><p class="first last">residual images saved by SPM.</p>
</dd>
<dt><strong>mask.nii:</strong></dt>
<dd><p class="first last">the mask image that was used.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Options for varargin:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;mask&#8217;</strong></dt>
<dd><p class="first">This option can be used to estimate a cluster size for the correction for multiple
comparisons &#8220;within the mask&#8221;. You can put in a ROI mask or gray matter,
whatever. If you don&#8217;t specify a mask image, brainmask.nii (default) will be
used, but the image has to be in your path.</p>
<blockquote class="last">
<div><p>e.g.)</p>
<div class="highlight-python"><div class="highlight"><pre>mask = fullfile(basedir, &#39;ROI_image.img&#39;); 
mask = which(&#39;scalped_avg152T1_graymatter_smoothed.img&#39;); % limited to gray matter
</pre></div>
</div>
</div></blockquote>
</dd>
<dt><strong>&#8216;outputdir&#8217;</strong></dt>
<dd><p class="first last">With this option, this will save residual and mask images and in the 
outputdir directory. If you don&#8217;t give outputdir, the current directory
will be used (default).</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This function calls cl_ext_spm_spm.m, which is a modified spm_spm not to
delete residual images.</p>
</dd></dl>

<dl class="function">
<dt id="Image_thresholding.cl_ext_spm_grf">
<code class="descclassname">Image_thresholding.</code><code class="descname">cl_ext_spm_grf</code><span class="sig-paren">(</span><em>corrected_p</em>, <em>prim_p</em>, <em>residual_images</em>, <em>mask</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_thresholding.cl_ext_spm_grf" title="Permalink to this definition">¶</a></dt>
<dd><p>This function is designed to estimate a cluster extent size for 
the correction for multiple comparisons based on a Gaussian Random Field 
Theory using SPM toolboxes.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cl_ext</span><span class="p">,</span> <span class="n">fwhm</span><span class="p">]</span> <span class="o">=</span> <span class="n">cl_ext_spm_grf</span><span class="p">(</span><span class="n">corrected_p</span><span class="p">,</span> <span class="n">prim_p</span><span class="p">,</span> <span class="n">residual_images</span><span class="p">,</span> <span class="n">mask</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span> 
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>corrected_p:</strong></dt>
<dd><p class="first">corrected p value</p>
<p class="last">e.g.) cluster-extent corrected p &lt; .05: corrected_p = .05</p>
</dd>
<dt><strong>prim_p:</strong></dt>
<dd><p class="first">primary threshold for height (i.e., cluster-defining threshold)</p>
<p class="last">e.g.) prim_p = [0.01 0.005 0.001];</p>
</dd>
<dt><strong>residual_images:</strong></dt>
<dd><p class="first">residual image names; if you used</p>
<p class="last">cl_ext_make_resid.m, this should be &#8216;Res4d.nii&#8217;</p>
</dd>
<dt><strong>mask:</strong></dt>
<dd><p class="first last">mask image name</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first"><strong>&#8216;doplot&#8217;:</strong></p>
<dl class="last docutils">
<dt><strong>&#8216;twotail&#8217;:</strong></dt>
<dd><p class="first last">default is one-tail - with this option, primary_p/2 will be used 
for all clsuter extent estimations.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Output:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>cl_ext_spm:</strong></dt>
<dd><blockquote class="first">
<div>cl_ext_spm is the cluster size that makes a corrected p value under</div></blockquote>
<p class="last">corrected_p (e.g., 0.05).</p>
</dd>
<dt><strong>fwhm (x, y, z in voxels):</strong></dt>
<dd><blockquote class="first">
<div>intrinsic smoothness level estimated by SPM (spm_est_smoothness.m)</div></blockquote>
<p class="last">If you want to convert this into mm, you need to multiply these
values by voxel sizes in mm.</p>
</dd>
</dl>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Image_thresholding.cl_ext_spm_spm">
<code class="descclassname">Image_thresholding.</code><code class="descname">cl_ext_spm_spm</code><span class="sig-paren">(</span><em>SPM</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_thresholding.cl_ext_spm_spm" title="Permalink to this definition">¶</a></dt>
<dd><p>[Re]ML Estimation of a General Linear Model</p>
<blockquote>
<div><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">FORMAT</span> <span class="p">[</span><span class="n">SPM</span><span class="p">]</span> <span class="o">=</span> <span class="n">spm_spm</span><span class="p">(</span><span class="n">SPM</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Required fields of SPM:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>xY.VY - nScan x 1 struct array of image handles (see spm_vol)</dt>
<dd><blockquote class="first">
<div>Images must have the same orientation, voxel size and data type</div></blockquote>
<ul class="last simple">
<li>Any scaling should have already been applied via the image handle
scalefactors.</li>
</ul>
</dd>
<dt>xX    - Structure containing design matrix information</dt>
<dd><ul class="first last">
<li><p class="first">Required fields are:
xX.X      - Design matrix (raw, not temporally smoothed)
xX.name   - cellstr of parameter names corresponding to columns</p>
<blockquote>
<div><p>of design matrix</p>
</div></blockquote>
</li>
<li><p class="first">Optional fields are:
xX.K      - cell of session-specific structures (see spm_filter)</p>
<blockquote>
<div><ul class="simple">
<li>Design &amp; data are pre-multiplied by K
(K*Y = K*X*beta + K*e)</li>
<li>Note that K should not smooth across block boundaries</li>
<li>defaults to speye(size(xX.X,1))</li>
</ul>
</div></blockquote>
<dl class="docutils">
<dt>xX.W      - Optional whitening/weighting matrix used to give</dt>
<dd><p class="first last">weighted least squares estimates (WLS). If not specified
spm_spm will set this to whiten the data and render
the OLS estimates maximum likelihood
i.e. W*W&#8217; = inv(xVi.V).</p>
</dd>
</dl>
</li>
</ul>
</dd>
<dt>xVi   - Structure describing intrinsic temporal non-sphericity</dt>
<dd><ul class="first last">
<li><p class="first">Required fields are:
xVi.Vi    - array of non-sphericity components</p>
<blockquote>
<div><ul class="simple">
<li>defaults to {speye(size(xX.X,1))} - i.i.d.</li>
<li>specifying a cell array of constraints (Qi)
These constraints invoke spm_reml to estimate
hyperparameters assuming V is constant over voxels.
that provide a high precise estimate of xX.V</li>
</ul>
</div></blockquote>
</li>
<li><p class="first">Optional fields are:
xX.V      - Optional non-sphericity matrix.  Cov(e) = sigma^2*V</p>
<blockquote>
<div><p>If not specified spm_spm will compute this using
a 1st pass to identify significant voxels over which
to estimate V.  A 2nd pass is then used to re-estimate
the parameters with WLS and save the ML estimates
(unless xX.W is already specified).</p>
</div></blockquote>
</li>
</ul>
</dd>
<dt>xM    - Structure containing masking information, or a simple column vector</dt>
<dd><blockquote class="first">
<div>of thresholds corresponding to the images in VY [default: -Inf]</div></blockquote>
<ul class="last">
<li><p class="first">If a structure, the required fields are:
xM.TH - nVar x nScan matrix of analysis thresholds, one per image
xM.I  - Implicit masking (0=&gt;none, 1 =&gt; implicit zero/NaN mask)
xM.VM - struct array of explicit mask image handles</p>
</li>
<li><dl class="first docutils">
<dt>(empty if no explicit masks)</dt>
<dd><ul class="first last simple">
<li>Explicit mask images are &gt;0 for valid voxels to assess.</li>
<li>Mask images can have any orientation, voxel size or data
type. They are interpolated using nearest neighbour
interpolation to the voxel locations of the data Y.</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Note that voxels with constant data (i.e. the same value across
scans) are also automatically masked out.</p>
</li>
</ul>
</dd>
<dt>swd   - Directory where the output files will be saved [default: pwd]</dt>
<dd>If exists, it becomes the current working directory.</dd>
</dl>
<p>In addition, global SPM &#8220;defaults&#8221; variable is used (see spm_defaults):</p>
<dl class="docutils">
<dt>stats.&lt;modality&gt;.UFp - critical F-threshold for selecting voxels over </dt>
<dd>which the non-sphericity is estimated (if 
required) [default: 0.001]</dd>
<dt>stats.maxres         - maximum number of residual images for smoothness</dt>
<dd>estimation</dd>
</dl>
<p>stats.maxmem         - maximum amount of data processed at a time (in bytes)</p>
<p>modality             - SPM modality {&#8216;PET&#8217;,&#8217;FMRI&#8217;,&#8217;EEG&#8217;}</p>
</div></blockquote>
<blockquote>
<div><p>spm_spm is the heart of the SPM package. Given image files and a
General Linear Model, it estimates the model parameters, variance
hyperparameters, and smoothness of standardised residual fields, writing
these out to disk in the current working directory for later
interrogation in the results section. (NB: Existing analyses in the
current working directory are overwritten).  This directory
now becomes the working directory for this analysis and all saved
images are relative to this directory.</p>
<p>The model is expressed via the design matrix (xX.X). The basic model
at each voxel is of the form is Y = X*B + e, for data Y, design
matrix X, (unknown) parameters B and residual errors e. The errors
are assumed to have a normal distribution.</p>
<p>Sometimes confounds (e.g. drift terms in fMRI) are necessary. These
can be specified directly in the design matrix or implicitly, in terms
of a residual forming matrix K to give a generalised linear model
K*Y = K*X*B + K*e.  In fact K can be any matrix (e.g. a convolution
matrix).</p>
<p>In some instances i.i.d. assumptions about errors do not hold. For
example, with serially correlated (fMRI) data or correlations among the
levels of a factor in repeated measures designs. This non-sphericity
can be specified in terms of components (SPM.xVi.Vi{i}). If specified
these covariance components will then be estimated with ReML (restricted
maximum likelihood) hyperparameters. This estimation assumes the same
non-sphericity for voxels that exceed the global F-threshold. The ReML
estimates can then be used to whiten the data giving maximum likelihood
(ML) or Gauss-Markov estimators. This entails a second pass of the data
with an augmented model K*W*Y = K*W*X*B + K*W*e where W*W&#8217; = inv(xVi.V).
xVi.V is the non-sphericity based on the hyperparameter estimates.
W is stored in xX.W and cov(K*W*e) in xX.V. The covariance of the
parameter estimates is then xX.Bcov = pinv(K*W*X)*xX.V*pinv(K*W*X)&#8217;.</p>
<p>If you do not want ML estimates but want to use ordinary least squares
(OLS) then simply set SPM.xX.W to the identity matrix. Any non-sphericity
V will still be estimated but will be used to adjust the degrees of freedom
of the ensuing statistics using the Satterthwaite approximation (c.f.
the Greenhouse-Geisser corrections).</p>
<p>If [non-spherical] variance components Vi are not specified xVi.Vi and
xVi.V default to the identity matrix (i.e. i.i.d). The parameters are
then estimated by OLS.  In this instance the OLS and ML estimates are
the same.</p>
<p>Note that only a single voxel-specific hyperparameter (i.e. variance
component) is estimated, even if V is not i.i.d.  This means spm_spm
always implements a fixed-effects model.
Random effects models can be emulated using a multi-stage procedure:
This entails summarising the data with contrasts such that the fixed
effects in a second model on the summary data are those effects of
interest (i.e. the population effects). This means contrasts are
re-entered into spm_spm to make an inference (SPM) at the next
level. At this higher hierarchical level the residual variance for the
model contains the appropriate variance components from lower levels.
See spm_RandFX.man for further details and below.</p>
<p>Under the additional assumption that the standardised error fields
are non-stationary standard Gaussian random fields, results from
Random field theory can be applied to estimate the significance
statistic images (SPM&#8217;s) adjusting p values for the multiple tests
at all voxels in the search volume. The parameters required for
this random field correction are the volume, and Lambda, the covariance
matrix of partial derivatives of the standardised error fields, estimated
by spm_est_smoothness.</p>
<blockquote>
<div></div></blockquote>
<p>The volume analysed is the intersection of the threshold masks,
explicit masks and implicit masks. See spm_spm_ui for further details
on masking options.</p>
</div></blockquote>
<blockquote>
<div><p>The output of spm_spm takes the form of an SPM.mat file of the analysis
parameters, and &#8216;float&#8217; flat-file images of the parameter and variance
[hyperparameter] estimates. An 8bit zero-one mask image indicating the
voxels assessed is also written out, with zero indicating voxels outside
tha analysed volume.</p>
<blockquote>
<div></div></blockquote>
<p>The following SPM.fields are set by spm_spm (unless specified)</p>
<blockquote>
<div><p>xVi.V      - estimated non-sphericity trace(V) = rank(V)
xVi.h      - hyperparameters  xVi.V = xVi.h(1)*xVi.Vi{1} + ...
xVi.Cy     - spatially whitened &lt;Y*Y&#8217;&gt; (used by ReML to estimate h)
xVi.CY     - &lt;(Y - &lt;Y&gt;)*(Y - &lt;Y&gt;)&#8217;&gt;    (used by spm_spm_Bayes)</p>
<blockquote>
<div></div></blockquote>
<p>Vbeta     - struct array of beta image handles (relative)
VResMS    - file struct of ResMS image handle  (relative)
VM        - file struct of Mask  image handle  (relative)</p>
<blockquote>
<div></div></blockquote>
<p>xX.W      - if not specified W*W&#8217; = inv(x.Vi.V)
xX.V      - V matrix (K*W*Vi*W&#8217;<a href="#id29"><span class="problematic" id="id30">*</span></a>K&#8217;) = correlations after K*W is applied
xX.xKXs   - space structure for K*W*X, the &#8216;filtered and whitened&#8217;</p>
<blockquote>
<div><blockquote>
<div>design matrix</div></blockquote>
<ul class="simple">
<li>given as spm_sp(&#8216;Set&#8217;,xX.K*xX.W*xX.X) - see spm_sp</li>
</ul>
</div></blockquote>
<p>xX.pKX    - pseudoinverse of K*W*X, computed by spm_sp
xX.Bcov   - xX.pKX*xX.V*xX.pKX - variance-covariance matrix of</p>
<blockquote>
<div>parameter estimates
(when multiplied by the voxel-specific hyperparameter ResMS
of the parameter estimates (ResSS/xX.trRV = ResMS) )</div></blockquote>
<p>xX.trRV   - trace of R*V
xX.trRVRV - trace of RVRV
xX.erdf   - effective residual degrees of freedom (trRV^2/trRVRV)
xX.nKX    - design matrix (xX.xKXs.X) scaled for display</p>
<blockquote>
<div><p>(see spm_DesMtx(&#8216;sca&#8217;,... for details)</p>
<blockquote>
<div></div></blockquote>
</div></blockquote>
<p>xVol.M    - 4x4 voxel-&gt;mm transformation matrix
xVol.iM   - 4x4 mm-&gt;voxel transformation matrix
xVol.DIM  - image dimensions - column vector (in voxels)
xVol.XYZ  - 3 x S vector of in-mask voxel coordinates
xVol.S    - Lebesgue measure or volume       (in voxels)
xVol.R    - vector of resel counts           (in resels)
xVol.FWHM - Smoothness of components - FWHM, (in voxels)</p>
<blockquote>
<div></div></blockquote>
<p>xCon      - Contrast structure (created by spm_FcUtil.m)
xCon.name - Name of contrast
xCon.STAT - &#8216;F&#8217;, &#8216;T&#8217; or &#8216;P&#8217; - for F/T-contrast (&#8216;P&#8217; for PPMs)
xCon.c    - (F) Contrast weights
xCon.X0   - Reduced design matrix (spans design space under Ho)</p>
<blockquote>
<div>It is in the form of a matrix (spm99b) or the
coordinates of this matrix in the orthogonal basis
of xX.X defined in spm_sp.</div></blockquote>
<dl class="docutils">
<dt>xCon.iX0  - Indicates how contrast was specified:</dt>
<dd>If by columns for reduced design matrix then iX0 contains
the column indices. Otherwise, it&#8217;s a string containing
the spm_FcUtil &#8216;Set&#8217; action: Usually one of {&#8216;c&#8217;,&#8217;c+&#8217;,&#8217;X0&#8217;}
(Usually this is the input argument F_iX0.)</dd>
<dt>xCon.X1o  - Remaining design space (orthogonal to X0).</dt>
<dd>It is in the form of a matrix (spm99b) or the
coordinates of this matrix in the orthogonal basis
of xX.X defined in spm_sp.</dd>
</dl>
<p>xCon.eidf - Effective interest degrees of freedom (numerator df)
xCon.Vcon - ...for handle of contrast/ESS image (empty at this stage)
xCon.Vspm - ...for handle of SPM image (empty at this stage)</p>
<blockquote>
<div></div></blockquote>
</div></blockquote>
<p>The following images are written to file</p>
<p>mask.{img,hdr}                                   - analysis mask image
8-bit (uint8) image of zero-s &amp; one&#8217;s indicating which voxels were
included in the analysis. This mask image is the intersection of the
explicit, implicit and threshold masks specified in the xM argument.
The XYZ matrix contains the voxel coordinates of all voxels in the
analysis mask. The mask image is included for reference, but is not
explicitly used by the results section.</p>
<blockquote>
<div></div></blockquote>
<p><a href="#id63"><span class="problematic" id="id64">beta_</span></a>????.{img,hdr}                                 - parameter images
These are 32-bit (float32) images of the parameter estimates. The image
files are numbered according to the corresponding column of the
design matrix. Voxels outside the analysis mask (mask.img) are given
value NaN.</p>
<blockquote>
<div></div></blockquote>
<p>ResMS.{img,hdr}                    - estimated residual variance image
This is a 64-bit (float64) image of the residual variance estimate.
Voxels outside the analysis mask are given value NaN.</p>
<blockquote>
<div></div></blockquote>
<p>RPV.{img,hdr}                      - estimated resels per voxel image
This is a 64-bit (float64) image of the RESELs per voxel estimate.
Voxels outside the analysis mask are given value 0.  These images
reflect the nonstationary aspects the spatial autocorrelations.</p>
<blockquote>
<div></div></blockquote>
<p><a href="#id65"><span class="problematic" id="id66">ResI_</span></a>????.{img,hdr}        - standardised residual (temporary) images
These are 64-bit (float64) images of standardised residuals. At most
maxres images will be saved and used by spm_est_smoothness, after which
they will be deleted.</p>
</div></blockquote>
<blockquote>
<div><p>References:</p>
<dl class="docutils">
<dt>Christensen R (1996) Plane Answers to Complex Questions</dt>
<dd>Springer Verlag</dd>
</dl>
<p>Friston KJ, Holmes AP, Worsley KJ, Poline JB, Frith CD, Frackowiak RSJ (1995)
<a href="#id31"><span class="problematic" id="id32">``</span></a>Statistical Parametric Maps in Functional Imaging:</p>
<blockquote>
<div><dl class="docutils">
<dt>A General Linear Approach&#8217;&#8216;</dt>
<dd>Human Brain Mapping 2:189-210</dd>
</dl>
</div></blockquote>
<p>Worsley KJ, Friston KJ (1995)
<a href="#id33"><span class="problematic" id="id34">``</span></a>Analysis of fMRI Time-Series Revisited - Again&#8217;&#8216;</p>
<blockquote>
<div>NeuroImage 2:173-181</div></blockquote>
</div></blockquote>
<blockquote>
<div>Copyright (C) 2008 Wellcome Trust Centre for Neuroimaging</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Image_thresholding.clusterSizeMask">
<code class="descclassname">Image_thresholding.</code><code class="descname">clusterSizeMask</code><span class="sig-paren">(</span><em>sizeThresh</em>, <em>height_mask</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_thresholding.clusterSizeMask" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">mask</span><span class="p">,</span><span class="n">numClusters</span><span class="p">,</span><span class="n">XYZ</span><span class="p">]</span> <span class="o">=</span> <span class="n">clusterSizeMask</span><span class="p">(</span><span class="n">sizeThresh</span><span class="p">,</span><span class="n">height_mask</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Image_thresholding.threshold_imgs">
<code class="descclassname">Image_thresholding.</code><code class="descname">threshold_imgs</code><span class="sig-paren">(</span><em>dd</em>, <em>u</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Image_thresholding.threshold_imgs" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">P2</span><span class="p">,</span><span class="n">P</span><span class="p">,</span><span class="n">sigmat</span><span class="p">,</span><span class="n">sigmatneg</span><span class="p">]</span> <span class="o">=</span> <span class="n">threshold_imgs</span><span class="p">(</span><span class="n">dd</span><span class="p">,</span><span class="n">u</span><span class="p">,[</span><span class="n">k</span><span class="p">],[</span><span class="s">&#39;pos&#39;</span> <span class="s">&#39;neg&#39;</span> <span class="s">&#39;both&#39;</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dd:</strong></dt>
<dd><p class="first last">list of filenames (str matrix)</p>
</dd>
<dt><strong>u:</strong></dt>
<dd><p class="first last">height threshold for images</p>
</dd>
<dt><strong>k:</strong></dt>
<dd><p class="first last">extent threshold for contiguous voxels</p>
</dd>
<dt><strong>[str]:</strong></dt>
<dd><dl class="first last docutils">
<dt>&#8216;pos&#8217; &#8216;neg&#8217; or &#8216;both&#8217;, to values above, below,</dt>
<dd><ul class="first last simple">
<li>and + threshold</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Output - threshold t - generic function</p>
<p>also do: plot rob vs ols benefit by tissue class and ols-irls average</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[P2,P,s,sn] = threshold_imgs(p([5 8],:),tinv(1-.001,10),0,&#39;both&#39;);
compare_filtered_t([],P2(1,:),P2(2,:))
[p2,p1] = threshold_imgs(&#39;irls-ols_z_0001.img&#39;,norminv(.9),0,&#39;both&#39;);

% compare_filtered_t([],P2(1,:),P2(2,:),p2)
h = image_scatterplot(str2mat(P,p1),&#39;avgvs3&#39;);
xlabel(&#39;Average OLS and Robust t-value&#39;), ylabel(&#39;Z-score of Robust - OLS difference&#39;)

s = str2mat(&#39;rob_tmap_0002.img&#39;,&#39;rob_tmap_0003.img&#39;);
P = threshold_imgs(s,tinv(1-.05,36),[],&#39;pos&#39;);P = threshold_imgs(s,tinv(1-.05,36),[],&#39;neg&#39;);
</pre></div>
</div>
</dd></dl>

</div>
<div class="section" id="index-image-manip-tools">
<h2>Index_image_manip_tools<a class="headerlink" href="#index-image-manip-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Index_image_manip_tools"></span><dl class="function">
<dt id="Index_image_manip_tools.flip_endianness">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">flip_endianness</code><span class="sig-paren">(</span><em>imgs</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.flip_endianness" title="Permalink to this definition">¶</a></dt>
<dd><p>Flips fMRI images (swap R/L)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">Vimgs</span><span class="p">]</span> <span class="o">=</span> <span class="n">flip_endianness</span><span class="p">(</span><span class="n">imgs</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>imgs:</strong></dt>
<dd><p class="first last">Give this a list of fMRI images (paths, char str)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Vimgs:</strong></dt>
<dd><p class="first last">Your images, flipped</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>::</dt>
<dd>load EXPT  %study specific spm design file
imgs = EXPT.SNPM.P{1};
[Vimgs] = flip_endianness(imgs)</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">References:</th><td class="field-body"><p class="first">N/A</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">See also:</th><td class="field-body"><ul class="first last simple">
<li>spm_vol</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_check_volinfo">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_check_volinfo</code><span class="sig-paren">(</span><em>maskInfo</em>, <em>imgInfo</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_check_volinfo" title="Permalink to this definition">¶</a></dt>
<dd><p>Checks a series of image .mat files and dims against a reference
(maskInfo)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">anybad</span> <span class="o">=</span> <span class="n">iimg_check_volinfo</span><span class="p">(</span><span class="n">maskInfo</span><span class="p">,</span><span class="n">imgInfo</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>maskInfo</strong> and <strong>volInfo :</strong></dt>
<dd><p class="first last">are spm-style volume info structures</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>see spm_vol.m</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_cluster_extent">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_cluster_extent</code><span class="sig-paren">(</span><em>dat</em>, <em>volInfo</em>, <em>k</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_cluster_extent" title="Permalink to this definition">¶</a></dt>
<dd><p>Apply a cluster size threshold to a series of index image data vectors
(each img is one column)
given volInfo (must be created with iimg_read_img, with extended output)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">dat</span><span class="p">,</span><span class="n">nvox</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_cluster_extent</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="n">volInfo</span><span class="p">,</span><span class="n">k</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">may be an indexed image of all image values or only those in mask
defined by volInfo</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_cluster_index">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_cluster_index</code><span class="sig-paren">(</span><em>dat</em>, <em>xyz</em>, <em>k</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_cluster_index" title="Permalink to this definition">¶</a></dt>
<dd><p>Get voxel cluster indices and sizes
:Usage:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">clindx</span><span class="p">,</span><span class="n">nvox</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_cluster_index</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="n">xyz</span><span class="p">,[</span><span class="n">k</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>xyz:</strong></dt>
<dd><p class="first last">is 3 x n list of voxel coords, volInfo.xyzlist&#8217;</p>
</dd>
<dt><strong>dat:</strong></dt>
<dd><p class="first last">is index vector of image values</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Returns: cluster index and cluster sizes for non-zero, non-nan voxels</p>
<p>initialize</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_cluster_intersect">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_cluster_intersect</code><span class="sig-paren">(</span><em>dat1</em>, <em>dat2</em>, <em>volInfo</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_cluster_intersect" title="Permalink to this definition">¶</a></dt>
<dd><p>Prunes two image data vectors (dat1 and dat2) assumed to contain suprathreshold
contiguous clusters (&#8220;blobs&#8221;) by saving only those blobs that have one or
more significant (nonzero) elements in both images</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cl1</span><span class="p">,</span><span class="n">cl2</span><span class="p">,</span><span class="n">dat1</span><span class="p">,</span><span class="n">dat2</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_cluster_intersect</span><span class="p">(</span><span class="n">dat1</span><span class="p">,</span><span class="n">dat2</span><span class="p">,</span><span class="n">volInfo</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>volInfo.xyzlist:</strong></dt>
<dd><p class="first last">must contain xyz coordinates corresponding to dat1 and dat2</p>
</dd>
<dt><strong>dat1 and dat2:</strong></dt>
<dd><p class="first last">can be either image-length or in-mask length vectorized images</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Try iimg_threshold to create dat vectors from Analyze images (e.g.,
statistic images)</p>
<p>The outputs cl1 and cl2 are the overlap (intersection) clusters, in the
space of dat1 (cl1) and dat2 (cl2).  The outputs dat1 and dat2 are the
&#8216;pruned&#8217; intersection data vectors.</p>
<p>prune dat1 with sig values in dat2</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_cluster_prune">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_cluster_prune</code><span class="sig-paren">(</span><em>dat</em>, <em>datsig</em>, <em>volInfo</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_cluster_prune" title="Permalink to this definition">¶</a></dt>
<dd><p>Prunes an image data vector (dat) assumed to contain suprathreshold
contiguous clusters (&#8220;blobs&#8221;) by saving only those blobs that have one or
more significant (nonzero) elements in another image, datsig</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">dat_out</span><span class="p">,</span><span class="n">clindx</span><span class="p">,</span><span class="n">keepit</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_cluster_prune</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="n">datsig</span><span class="p">,</span><span class="n">volInfo</span><span class="p">)</span>
</pre></div>
</div>
<p>One intended use is to define an FWE-corrected significance map in
datsig, and report blobs at some lower threshold (in dat) that have at
least one corrected voxel.</p>
<p>Try iimg_threshold to create dat vectors from Analyze images (e.g.,
statistic images)</p>
<p>check data type and reduce to in-mask only if necessary</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_clusters2indx">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_clusters2indx</code><span class="sig-paren">(</span><em>cl</em>, <em>volInfo</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_clusters2indx" title="Permalink to this definition">¶</a></dt>
<dd><p>Take a clusters structure and turn it into an indexed image of dims
volInfo.dim</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">imgvec</span><span class="p">,</span><span class="n">maskvec</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_clusters2indx</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">volInfo</span><span class="p">)</span>
<span class="p">[</span><span class="n">imgvec</span><span class="p">,</span><span class="n">maskvec</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_clusters2indx</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">my_image_name</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>imgvec:</strong></dt>
<dd><p class="first last">vector of all image voxels</p>
</dd>
<dt><strong>maskvec:</strong></dt>
<dd><p class="first last">vector of in-mask voxels</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Uses:  volInfo.nvox, .wh_inmask, .dim
convert image to volinfo struct, if necessary</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_indx2contiguousxyz">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_indx2contiguousxyz</code><span class="sig-paren">(</span><em>dat</em>, <em>volInfo</em>, <em>remove_mean_flag</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_indx2contiguousxyz" title="Permalink to this definition">¶</a></dt>
<dd><p>Take in index image data vector (in-mask values only) and a volume info structure with xyzlist
and return a cl structure whose XYZ values list contiguous sets of voxels
(&#8220;blobs&#8221;)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">iimg_indx2contiguousxyz</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="n">volInfo</span><span class="p">,[</span><span class="n">remove_mean_flag</span><span class="p">])</span>
</pre></div>
</div>
<p>If a 3rd arg is entered, means of each blob are subtracted
This is to facilitate randomizing blob centers in
meta_stochastic_activation_blobs.m</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_intersection">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_intersection</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_intersection" title="Permalink to this definition">¶</a></dt>
<dd><p>Make a mask of the intersection of n image files (pos or neg values)
If &#8216;name&#8217; followed by an image file name is entered, writes output to
that image</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[int_dat,mask_vol,outname] = iimg_intersection(name1, name2, etc.)
[int_dat,mask_vol,outname] = iimg_intersection(dat1, dat2, etc.)
</pre></div>
</div>
<p>If special string &#8216;posneg&#8217; is entered, separates first two images into
combinations of pos/neg values in each image.
The order returned in columns of int_dat is pospos, posneg, negpos, and
negneg</p>
<p>fastest if one output requested.</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_make_sure_indx">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_make_sure_indx</code><span class="sig-paren">(</span><em>inputarg</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_make_sure_indx" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">dat</span> <span class="o">=</span> <span class="n">iimg_make_sure_indx</span><span class="p">(</span><span class="n">inputarg</span><span class="p">)</span>
</pre></div>
</div>
<p>Make sure input is an index list, and convert if not</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_mask">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_mask</code><span class="sig-paren">(</span><em>maskindx</em>, <em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Masks index image vector or filenames in dat
With img vector or filenames in maskindx
* Checks dimensions to make sure origins and vox. sizes are same</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">masked_indx</span><span class="p">,</span> <span class="n">volInfo</span><span class="p">,</span> <span class="n">masked_vol</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_mask</span><span class="p">(</span><span class="n">maskindx</span><span class="p">,</span><span class="n">dat</span><span class="p">,[</span><span class="n">volInfo</span><span class="p">],</span> <span class="p">[</span><span class="n">outname</span><span class="p">])</span>
</pre></div>
</div>
<p>Fastest way to mask an image:</p>
<div class="highlight-python"><div class="highlight"><pre>[masked_indx_data, volInfo] = iimg_mask(maskindx,dat)

% or

[masked_indx, xyz, masked_vol] = iimg_mask(mask filename,image filenames)
</pre></div>
</div>
<p>even slower: reconstruct a 3-D volume and return in masked_vol</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_multi_threshold">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_multi_threshold</code><span class="sig-paren">(</span><em>inname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_multi_threshold" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cl</span><span class="p">,</span> <span class="n">dat</span><span class="p">,</span> <span class="n">cl_extent</span><span class="p">,</span> <span class="n">dat_extent</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_multi_threshold</span><span class="p">(</span><span class="n">inname</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Multi-threshold application and orthview display of blobs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>inname:</strong></dt>
<dd><p class="first">Either the name of an image file or a data vector</p>
<p class="last">if data, enter volInfo structure separately as below.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Command strings:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;prune&#8217;:</strong></dt>
<dd><p class="first last">consider only contiguous blobs for which at least 1 voxel meets
the most stringent threshold</p>
</dd>
<dt><strong>&#8216;pruneseed&#8217;:</strong></dt>
<dd><p class="first">followed by a vectorized thresholded activation image</p>
<p>Only those blobs overlapping with at least one &#8216;seed&#8217; data voxel
are saved</p>
<p class="last">Also: the first color (highest threshold) in the output images is assigned to the seed</p>
</dd>
<dt><strong>&#8216;add&#8217;:</strong></dt>
<dd><p class="first last">add new blobs to existing orthviews</p>
</dd>
<dt><strong>&#8216;p&#8217;:</strong></dt>
<dd><p class="first last">if image is a p-value image (and thresholds are p-values)</p>
</dd>
<dt><strong>&#8216;thresh&#8217;:</strong></dt>
<dd><p class="first last">followed by a vector of n thresholds, most to least stringent</p>
</dd>
<dt><strong>&#8216;size&#8217;:</strong></dt>
<dd><p class="first last">followed by a vector of size thresholds</p>
</dd>
</dl>
<p class="last"><a href="#id35"><span class="problematic" id="id36">**</span></a>&#8216;volInfo&#8217;: followed by volInfo struct; needed if inname is data rather</p>
</td>
</tr>
</tbody>
</table>
<p>than filenames</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;overlay&#8217;:</strong></dt>
<dd>followed by overlay image (anatomical) filename</dd>
<dt><strong>&#8216;transseed&#8217;:</strong></dt>
<dd>transparent seed</dd>
<dt><strong>&#8216;hideseed&#8217;:</strong></dt>
<dd>do not show seed regions on plot</dd>
<dt><strong>&#8216;pos&#8217;:</strong></dt>
<dd>positive results only</dd>
<dt><strong>&#8216;neg&#8217;:</strong></dt>
<dd>negative results only</dd>
<dt><strong>&#8216;add&#8217;:</strong></dt>
<dd>add to existing multi-threshold plot</dd>
</dl>
</div></blockquote>
<p>Needs (development): Legend, nice handling of colors, input
colors, color maps specified with command words (like &#8216;red&#8217;)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>inname = &#39;Activation_proportion.img&#39;;
[cl, dat] = iimg_multi_threshold(inname, &#39;prune&#39;, &#39;thresh&#39;, [.1 .5 .3], &#39;size&#39;, [1 5 10]);

cl2 = iimg_multi_threshold(Pimg(1, :), &#39;thresh&#39;, [.001 .01 .05], &#39;size&#39;, [3 5 10], &#39;p&#39;);
cl2 = iimg_multi_threshold(Pimg(1, :), &#39;thresh&#39;, [.001 .01 .05], &#39;size&#39;, [3 3 3], &#39;p&#39;, &#39;prune&#39;);
</pre></div>
</div>
<p>from act + corr results (see robust_results_act_plus_corr)
First prepare &#8216;seed&#8217; regions that overlap with correlated regions, then
use multi_threshold to see full extent of clusters</p>
<div class="highlight-python"><div class="highlight"><pre>[dattype, seeddat] = iimg_check_indx(res.act.overlapdat, res.volInfo, &#39;full&#39;);
[cl, dat] = iimg_multi_threshold(&#39;rob_p_0001.img&#39;, &#39;p&#39;, &#39;thresh&#39;, [.001 .01 .05], &#39;size&#39;, [1 1 1], &#39;pruneseed&#39;, seeddat)
</pre></div>
</div>
<p>Display an F-map from robust regression on a customized mean anatomical,
with pruning.</p>
<div class="highlight-python"><div class="highlight"><pre>cl = iimg_multi_threshold(&#39;rob_pmap_full.img&#39;, &#39;thresh&#39;, [.001 .01 .05], &#39;size&#39;, [1 1 1], &#39;p&#39;, &#39;prune&#39;, &#39;overlay&#39;, EXPT.overlay);
</pre></div>
</div>
<p>Display regions at 3 thresholds with an input data &#8216;seed&#8217; vector</p>
<div class="highlight-python"><div class="highlight"><pre>[cl, dat] = iimg_multi_threshold(pvals, &#39;p&#39;, &#39;thresh&#39;, [.001 .01 .05], &#39;size&#39;, [1 1 1], &#39;pruneseed&#39;, p_sig&#39;, &#39;volInfo&#39;, R.volInfo);
</pre></div>
</div>
<p>As above, but ONLY POSITIVE RESULTS</p>
<div class="highlight-python"><div class="highlight"><pre>[cl, datout] = iimg_multi_threshold(pimg1, &#39;thresh&#39;, [.001 .01 .05], &#39;size&#39;, [1 1 1], &#39;p&#39;, &#39;pruneseed&#39;, dat, &#39;overlay&#39;, EXPT.overlay, &#39;colors&#39;, colors, &#39;transseed&#39;, &#39;pos&#39;, &#39;rob_tmap_0001.img&#39;);
</pre></div>
</div>
<p>Complete example for showing positive and negative blobs:</p>
<div class="highlight-python"><div class="highlight"><pre>[clpos] = iimg_multi_threshold(&#39;slope_p.img&#39;, &#39;p&#39;, &#39;thresh&#39;, [.005 .01 .05], &#39;size&#39;, [1 1 1], &#39;prune&#39;, &#39;overlay&#39;, anat, &#39;pos&#39;, &#39;slope_t.img&#39;);
[clneg] = iimg_multi_threshold(&#39;slope_p.img&#39;, &#39;p&#39;, &#39;thresh&#39;, [.005 .01 .05], &#39;size&#39;, [1 1 1], &#39;prune&#39;, &#39;overlay&#39;, anat, &#39;neg&#39;, &#39;slope_t.img&#39;, &#39;add&#39;, &#39;colors&#39;, {[0 0 1] [0 .5 1] [.4 .5 1]});
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_princomp">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_princomp</code><span class="sig-paren">(</span><em>maskname</em>, <em>image_names</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_princomp" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">volInfo</span><span class="p">,</span><span class="n">clusters</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_princomp</span><span class="p">(</span><span class="n">maskname</span><span class="p">,</span><span class="n">image_names</span><span class="p">)</span> 
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_princomp_display">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_princomp_display</code><span class="sig-paren">(</span><em>volInfo</em>, <em>k</em>, <em>overlay</em>, <em>dofigs</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_princomp_display" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">iimg_princomp_display</span><span class="p">(</span><span class="n">volInfo</span><span class="p">,</span><span class="n">k</span><span class="p">,</span><span class="n">overlay</span><span class="p">,</span><span class="n">dofigs</span><span class="p">)</span>
</pre></div>
</div>
<p>Show output from iimg_princomp</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl2</span> <span class="o">=</span> <span class="n">iimg_princomp_display</span><span class="p">(</span><span class="n">volInfo2</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="n">EXPT</span><span class="o">.</span><span class="n">overlay</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_read_img">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_read_img</code><span class="sig-paren">(</span><em>inputimgs</em>, <em>extended_output_flag</em>, <em>reading_data</em>, <em>first_vol</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_read_img" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[volInfo, dat] = iimg_read_img(inputimgs, [extended_output_flag (1|2)], [reading_data (0|1)], [first vol only (0|1)]);
</pre></div>
</div>
<ul class="simple">
<li>reads 3-D or 4-D img/nii files, with extended volInfo output</li>
<li>flag to read first volume only</li>
<li>can read iimg_data vectorized form as well as file names (just passes dat to output)</li>
<li>uses SPM</li>
</ul>
<dl class="docutils">
<dt>volInfo fields:</dt>
<dd><blockquote class="first">
<div><ul class="simple">
<li>all fields returned by spm_vol, plus:</li>
<li>.nvox - number of voxels in 3d image</li>
</ul>
</div></blockquote>
<ul class="last simple">
<li>.image_indx - logical index (e.g., [0 1 1 0 1 ...]) of all nonzero voxels</li>
<li>.wh_inmask (extended_output_flag &gt; 0) - linear index (e.g., [2 3 5 ...]) - same as image_indx, but as a result of find</li>
<li>.n_inmask (extended_output_flag &gt; 0) - length of .wh_inmask</li>
<li>.xyzlist (extended_output_flag &gt; 0) - voxel coords of .wh_inmask voxels</li>
<li>.cluster (extended_output_flag &gt; 1) - cluster structure of in-mask voxels</li>
</ul>
</dd>
<dt>Extended output flag values:</dt>
<dd>1: Add xyz coord list and linear index of which voxels nonzero &amp; non-nan in mask
2: Add clusters from spm_clusters</dd>
</dl>
<p>Enforces that dat is a vector of data values, regardless of input.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">imname</span> <span class="o">=</span> <span class="s">&#39;rob_tmap_0001_filt_t_3-05_k5_pos.img&#39;</span><span class="p">;</span>
<span class="n">volInfo</span> <span class="o">=</span> <span class="n">iimg_read_img</span><span class="p">(</span><span class="n">imname</span><span class="p">);</span>
</pre></div>
</div>
<p>WARNING: 
The behavior of this function is SPM version dependent.
volInfo only contains the spm_vol structure of the FIRST image!!!
Data in SPM5/8 should be 4-D, but in SPM2 should be 3-D</p>
<blockquote>
<div><ol class="arabic">
<li><dl class="first docutils">
<dt>Get volume info from first volume of a 3-D or 4-D image</dt>
<dd><p class="first">volInfo structure has necessary info for converting to/from
&#8220;vectorized&#8221; format
dat returns 4-D data from entire image (all volumes)</p>
<div class="last highlight-python"><div class="highlight"><pre><span class="n">iimg_name</span> <span class="o">=</span> <span class="s">&#39;test_run1_pca.img&#39;</span><span class="p">;</span>
</pre></div>
</div>
</dd>
</dl>
</li>
</ol>
</div></blockquote>
<p>[maskInfo, dat] = iimg_read_img(img_name, 2);</p>
<blockquote>
<div><ol class="arabic" start="2">
<li><p class="first">Get data in &#8220;vectorized&#8221; image format for each volume in the
image. Works for a list of images too. data is in-mask voxels x volumes
this can be useful when you want to return whole-brain data for many
images, but in a search volume only (i.e., no extra-brain voxels)</p>
<div class="highlight-python"><div class="highlight"><pre>idata = iimg_get_data(maskInfo, img_name);
data = data&#39;;  % make sure columns are volumes
</pre></div>
</div>
</li>
</ol>
</div></blockquote>
<dl class="docutils">
<dt>% 3) Write out a 4-D image with the same data, called test_run1_pca2.img</dt>
<dd><div class="first last highlight-python"><div class="highlight"><pre><span class="n">ivoldat3D</span> <span class="o">=</span> <span class="n">iimg_reconstruct_vols</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">maskInfo</span><span class="p">,</span> <span class="s">&#39;outname&#39;</span><span class="p">,</span>
                                      <span class="s">&#39;test_run1_pca2.img&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_read_vols">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_read_vols</code><span class="sig-paren">(</span><em>V</em>, <em>mask</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_read_vols" title="Permalink to this definition">¶</a></dt>
<dd><p>Drop-in replacement for spm_read_vols
Primary difference is that if the images have the same voxel size but are NOT resliced
(e.g., have differing affine matrices), this will handle reading each individually and
putting them together.</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_reconstruct_3dvol">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_reconstruct_3dvol</code><span class="sig-paren">(</span><em>dat</em>, <em>volInfo</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_reconstruct_3dvol" title="Permalink to this definition">¶</a></dt>
<dd><p>Reconstruct a 3-D volume from a dat index list</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>voldata = iimg_reconstruct_3dvol(dat, volInfo, [optional args])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">if entered, will write .img file to disk</p>
<dl class="last docutils">
<dt><strong>&#8216;outname&#8217;:</strong></dt>
<dd><p class="first last">followed by output name</p>
</dd>
<dt><strong>&#8216;descrip&#8217;:</strong></dt>
<dd><p class="first last">followed by description for img file</p>
</dd>
<dt><strong>&#8216;slice&#8217;:</strong></dt>
<dd><p class="first last">followed by slice number of single-slice data in image</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>THIS FUNCTION IS DEPRECATED. USE IIMG_RECONSTRUCT_VOLS.M&#8217;
WHICH CAN DEAL WITH 4-D VOLUMES AS WELL</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_reconstruct_vols">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_reconstruct_vols</code><span class="sig-paren">(</span><em>dat</em>, <em>volInfo</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_reconstruct_vols" title="Permalink to this definition">¶</a></dt>
<dd><p>Reconstruct a 3-D or 4-D volume from a &#8220;dat&#8221; matrix of vectorized images</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>voldata = iimg_reconstruct_vols(dat, volInfo, [optional args])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">If entered, will write .img file to disk</p>
<dl class="docutils">
<dt><strong>&#8216;outname&#8217;:</strong></dt>
<dd><p class="first last">followed by output name</p>
</dd>
<dt><strong>&#8216;descrip&#8217;:</strong></dt>
<dd><p class="first last">followed by description for img file</p>
</dd>
<dt><strong>&#8216;slice&#8217;:</strong></dt>
<dd><p class="first last">followed by slice number of single-slice data in image</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Example of image reading:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><ol class="first last arabic">
<li><dl class="first docutils">
<dt>Get volume info from first volume of a 3-D or 4-D image</dt>
<dd><p class="first">volInfo structure has necessary info for converting to/from
&#8220;vectorized&#8221; format
dat returns 4-D data from entire image (all volumes)</p>
<div class="last highlight-python"><div class="highlight"><pre><span class="n">img_name</span> <span class="o">=</span> <span class="s">&#39;test_run1_pca.img&#39;</span><span class="p">;</span>
<span class="p">[</span><span class="n">maskInfo</span><span class="p">,</span> <span class="n">dat</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_read_img</span><span class="p">(</span><span class="n">img_name</span><span class="p">,</span> <span class="mi">2</span><span class="p">);</span>
</pre></div>
</div>
</dd>
</dl>
</li>
<li><p class="first">Get data in &#8220;vectorized&#8221; image format for each volume in the
image. Works for a list of images too. data is in-mask voxels x volumes
this can be useful when you want to return whole-brain data for many
images, but in a search volume only (i.e., no extra-brain voxels)</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">data</span> <span class="o">=</span> <span class="n">iimg_get_data</span><span class="p">(</span><span class="n">maskInfo</span><span class="p">,</span> <span class="n">img_name</span><span class="p">);</span>
</pre></div>
</div>
</li>
</ol>
</td>
</tr>
</tbody>
</table>
<p>data = data&#8217;;  % make sure columns are volumes</p>
<blockquote>
<div><ol class="arabic" start="3">
<li><p class="first">Write out a 4-D image with the same data, called test_run1_pca2.img</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">voldat3D</span> <span class="o">=</span> <span class="n">iimg_reconstruct_vols</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">maskInfo</span><span class="p">,</span> <span class="s">&#39;outname&#39;</span><span class="p">,</span> <span class="s">&#39;test_run1_pca2.img&#39;</span><span class="p">);</span>
</pre></div>
</div>
</li>
</ol>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_reslice">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_reslice</code><span class="sig-paren">(</span><em>matchto</em>, <em>reslicethis</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_reslice" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">out</span> <span class="o">=</span> <span class="n">iimg_reslice</span><span class="p">(</span><span class="n">matchto</span><span class="p">,</span> <span class="n">reslicethis</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>

<span class="n">out</span> <span class="o">=</span> <span class="n">iimg_reslice</span><span class="p">(</span><span class="n">matchto</span><span class="p">,</span> <span class="n">reslicethis</span><span class="p">,</span> <span class="s">&#39;write&#39;</span><span class="p">,</span> <span class="s">&#39;outname&#39;</span><span class="p">,</span> <span class="s">&#39;myimg.img&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>default flags
interp = 2 is trilinear</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_smooth_3d">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_smooth_3d</code><span class="sig-paren">(</span><em>w</em>, <em>volInfo</em>, <em>sfwhm</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_smooth_3d" title="Permalink to this definition">¶</a></dt>
<dd><p>Smooth 3-D images stored in columns with FWHM in mm of sfwhm</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function w = smooth_3d(w, volInfo, sfwhm, [badvox])
</pre></div>
</div>
<p>NOTE: SMOOTHING KERNEL MAY BE IN VOX, AS VOL INFO IS NOT PASSED IN</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>w:</strong></dt>
<dd><p class="first last">Take v x n matrix w, and smooth columns (images), returning
v x n matrix again</p>
</dd>
<dt><strong>volInfo:</strong></dt>
<dd><p class="first last">is volume info structure from iimg_read_img.m</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Optional:</th><td class="field-body"><p class="first last">If v is smaller than the original image because some voxels</p>
</td>
</tr>
</tbody>
</table>
<p>were removed, enter logical vector badvox, and the missing voxels
will be filled with zeros.</p>
<p>NOTE: 4-D version is horribly slow and memory intensive:</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>wvol = iimg_reconstruct_vols(w&#39;, fmri_data_obj.mask.volInfo);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_sphere_timeseries">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_sphere_timeseries</code><span class="sig-paren">(</span><em>images</em>, <em>XYZmmCenter</em>, <em>radius</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_sphere_timeseries" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">data</span><span class="p">,</span> <span class="n">XYZvoxSphere</span><span class="p">,</span> <span class="n">XYZmmSphere</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_sphere_timeseries</span><span class="p">(</span><span class="n">images</span><span class="p">,</span> <span class="n">XYZmm</span><span class="p">,</span> <span class="n">radius</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>images:</strong></dt>
<dd><p class="first last">list of image files</p>
</dd>
<dt><strong>XYZmm:</strong></dt>
<dd><p class="first last">[3 x n] array of mm coords</p>
</dd>
<dt><strong>radiu:</strong></dt>
<dd><p class="first last">radius in mm of sphere to generate</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>data:</strong></dt>
<dd><p class="first last">voxel data</p>
</dd>
<dt><strong>XYZvoxSphere:</strong></dt>
<dd><p class="first last">voxel</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_stouffer">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_stouffer</code><span class="sig-paren">(</span><em>Pimg</em>, <em>maskname</em>, <em>thr</em>, <em>clsize</em>, <em>outname</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_stouffer" title="Permalink to this definition">¶</a></dt>
<dd><p>This function performs combination of p-values across images using the 
Stouffer method (see refs below).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function cl = iimg_stouffer(Pimg,maskname,thr,clsize,outname)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Pimg:</strong></dt>
<dd><p class="first last">string mtx of p-value image names</p>
</dd>
<dt><strong>k:</strong></dt>
<dd><p class="first last">num voxels/num comparisons 
if empty, uses # of non-zero, non-NaN values in images</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>empty Pimg prompts for graphic selection of filenames
empty outname prompts for entry of output img file name</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>cl:</strong></dt>
<dd><p class="first last">clusters, cl{1} is stouffer, cl{2} is image 1, cl{3} is image 2</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Described in:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Lazar, N. A., Luna, B., Sweeney, J. A., &amp; Eddy, W. F. (2002). 
Combining brains: a survey of methods for statistical pooling 
of information. Neuroimage, 16(2), 538-550.</p>
<p>Stouffer, S. A., Suchman, E. A., DeVinney, L. C., Star, S. A., and
Williams, R. M. 1949. The American Soldier: Vol. I. Adjustment
During Army Life. Princeton University Press, Princeton.</p>
<p>...</p>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_threshold">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_threshold</code><span class="sig-paren">(</span><em>image_names</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_threshold" title="Permalink to this definition">¶</a></dt>
<dd><p>Thresholds images to be at least thresh(1) and at most thresh(2)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">function</span> <span class="p">[</span><span class="n">dat</span><span class="p">,</span> <span class="n">volInfo</span><span class="p">,</span> <span class="n">cl</span><span class="p">]</span> <span class="o">=</span> <span class="n">iimg_threshold</span><span class="p">(</span><span class="n">image_names</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Input types for image names:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><ol class="first arabic simple">
<li>String matrix of filenames</li>
<li>4-D array of 3-D data volumes</li>
<li>voxels x images 2-D index array</li>
<li>image_vector object</li>
</ol>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>imdat:</strong></dt>
<dd><p class="first last">index vector of thresholded data</p>
</dd>
<dt><strong>volInfo:</strong></dt>
<dd><p class="first last">structure of info about volume</p>
</dd>
<dt><strong>cl:</strong></dt>
<dd><p class="first last">optional (slower) clusters structure from dat</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Command strings:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;imgtype&#8217;:</strong></dt>
<dd><blockquote class="first">
<div><p>followed by &#8216;t&#8217;, &#8216;p&#8217;, or &#8216;data&#8217; (default)</p>
</div></blockquote>
<p class="last">specify type of values in image</p>
</dd>
<dt><strong>&#8216;threshtype&#8217;:</strong></dt>
<dd><p class="first last">followed by &#8216;t&#8217; or &#8216;p&#8217;</p>
</dd>
<dt><strong>&#8216;df&#8217;:</strong></dt>
<dd><p class="first last">followed by degrees of freedom, for p- and t-image threshold calc</p>
</dd>
<dt><strong>&#8216;k&#8217;:</strong></dt>
<dd><p class="first last">followed by extent threshold in voxels (slower)</p>
</dd>
<dt><strong>&#8216;abs&#8217;:</strong></dt>
<dd><p class="first last">absolute value must be &gt; threshold.  Use to get both pos. and neg.
results in two-tailed test.</p>
</dd>
<dt><strong>&#8216;intersect:</strong></dt>
<dd><p class="first last">Create intersection of images; uses abs, so + and - values
count as yesses for intersection</p>
</dd>
<dt><strong>&#8216;contrast&#8217;:</strong></dt>
<dd><p class="first last">followed by contrasts across images</p>
</dd>
<dt><strong>&#8216;volInfo:</strong></dt>
<dd><p class="first last">followed by volInfo structure.  Necessary for extended output</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Threshold an image set (p) to be at least zero
image_names =
/Users/tor/Documents/Tor_Documents/CurrentExperiments/Lab/Pre-appraisal/Results/rscalped_avg152T1_graymatter_smoothed.img
/Users/tor/Documents/Tor_Documents/CurrentExperiments/Lab/Pre-appraisal/Results/t_intercept.img

[dat, volInfo] = iimg_threshold(image_names, &#39;thr&#39;, 0);

% Do the same, but take the intersection and write an output image
[dat, volInfo] = iimg_threshold(image_names, &#39;thr&#39;, 3, &#39;outnames&#39;, &#39;intersct&#39;, &#39;masked_t.img&#39;);


% Threshold a t-image based on a p-value and a df
image_names = &#39;/Users/tor/Documents/Tor_Documents/CurrentExperiments/Lab/Pre-appraisal/Results/t_intercept.img&#39;
[dat, volInfo] = iimg_threshold(image_names, &#39;thr&#39;, .005, &#39;imgtype&#39;, &#39;t&#39;, &#39;threshtype&#39;, &#39;p&#39;, &#39;df&#39;, 28, &#39;outnames&#39;, &#39;thresh_t.img&#39;);

cl = mask2clusters(&#39;masked_t.img&#39;);
cluster_orthviews(cl);

% The same, but threshold based on absolute value (+ and - values)
[dat, volInfo] = iimg_threshold(image_names, &#39;thr&#39;, .005, &#39;abs&#39;, &#39;imgtype&#39;, &#39;t&#39;, &#39;threshtype&#39;, &#39;p&#39;, &#39;df&#39;, 28, &#39;outnames&#39;, &#39;thresh_t.img&#39;);

% Threshold a p-value image directly
[dat, volInfo] = iimg_threshold(&#39;X-M-Y_pvals.img&#39;, &#39;thr&#39;, [0 .05], &#39;outnames&#39;, &#39;X-M-Y_pvals_thresh.img&#39;);
[dat, volInfo, cl] = iimg_threshold(inname, &#39;thr&#39;, [0 .05], &#39;outnames&#39;, outname);

% Threshold a p-value image, with cluster sizes
[dat, volInfo, cl] = iimg_threshold(inname, &#39;thr&#39;, [0 .05], &#39;k&#39;, 10);

% Threshold and display a t-image using FDR, getting both positive and negative results
[dat, volInfo, cl] = iimg_threshold(&#39;contrast_t.img&#39;, &#39;imgtype&#39;, &#39;t&#39;, &#39;df&#39;, 37, &#39;thr&#39;, .2, &#39;threshtype&#39;, &#39;fdr&#39;, &#39;k&#39;, 3, &#39;abs&#39;);
cluster_orthviews(cl);
spm_orthviews_hotcool_colormap(cat(2,cl.Z), 1.52);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_weighted_ttest">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_weighted_ttest</code><span class="sig-paren">(</span><em>image_names</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_weighted_ttest" title="Permalink to this definition">¶</a></dt>
<dd><p>fast weighted test</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">iimg_weighted_ttest</span><span class="p">(</span><span class="n">image_names</span><span class="p">,</span><span class="s">&#39;mask&#39;</span><span class="p">,</span><span class="n">maskname</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_write_images">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_write_images</code><span class="sig-paren">(</span><em>dat</em>, <em>volInfo</em>, <em>outnames</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_write_images" title="Permalink to this definition">¶</a></dt>
<dd><p>Write a series of Analyze images given inputs.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">iimg_write_images</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="n">volInfo</span><span class="p">,</span><span class="n">outnames</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">voxels x images matrix of image data in index format</p>
</dd>
<dt><strong>volInfo:</strong></dt>
<dd><p class="first last">spm-style info structure; see iimg_read_img</p>
</dd>
<dt><strong>outnames:</strong></dt>
<dd><p class="first last">a string matrix (or cell array) of output filenames, or the name of a 4-D file to create</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_xyz2indx">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_xyz2indx</code><span class="sig-paren">(</span><em>xyz</em>, <em>xyztype</em>, <em>V</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_xyz2indx" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[indx, wh] = iimg_xyz2indx(xyz, input type:[&#39;mm&#39; or &#39;vox&#39;], [V: needed if &#39;mm&#39;])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Index_image_manip_tools.iimg_xyz2spheres">
<code class="descclassname">Index_image_manip_tools.</code><code class="descname">iimg_xyz2spheres</code><span class="sig-paren">(</span><em>xyz</em>, <em>mask_xyzlist</em>, <em>r</em><span class="sig-paren">)</span><a class="headerlink" href="#Index_image_manip_tools.iimg_xyz2spheres" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">indx</span> <span class="o">=</span> <span class="n">iimg_xyz2spheres</span><span class="p">(</span><span class="n">xyz</span><span class="p">,</span> <span class="n">mask_xyzlist</span><span class="p">,</span> <span class="n">r</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>mask_xyzlist:</strong></dt>
<dd><p class="first last">is a list of voxel coords of all in-mask voxels</p>
</dd>
<dt><strong>xyz:</strong></dt>
<dd><p class="first last">is a list of voxels to be convolved with spheres</p>
</dd>
<dt><strong>r:</strong></dt>
<dd><p class="first last">is sphere radius (voxels)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>more generally, finds mask_xyzlist entries that are within r units of xyz
i.e., to find database points within r mm of a cluster xyz list:
indx = iimg_xyz2spheres(clusterxyz,databasexyz,r)</p>
<p>indx is length mask_xyzlist and contains all in-mask, in-sphere voxels
for all xyz coordinates.</p>
</dd></dl>

</div>
<div class="section" id="module-Misc_utilities">
<span id="misc-utilities"></span><h2>Misc_utilities<a class="headerlink" href="#module-Misc_utilities" title="Permalink to this headline">¶</a></h2>
<dl class="function">
<dt id="Misc_utilities.CERTreader">
<code class="descclassname">Misc_utilities.</code><code class="descname">CERTreader</code><span class="sig-paren">(</span><em>fname</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.CERTreader" title="Permalink to this definition">¶</a></dt>
<dd><p>Given an output file from CERT, returns a struct with the data</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>fname:</strong></dt>
<dd><p class="first last">filename</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><p class="first last">struct with the data</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.append">
<code class="descclassname">Misc_utilities.</code><code class="descname">append</code><span class="sig-paren">(</span><em>a</em>, <em>b</em>, <em>field</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Appends second parameter to the end of the first. If field is set,
writes into the field instead of directly into the variable.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">a</span> <span class="o">=</span> <span class="n">append</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">data</span><span class="p">,</span> <span class="p">[</span><span class="n">field</span><span class="p">])</span>
</pre></div>
</div>
<p>This function solely exists because Matlab has no ability to create
empty objects (e.g. with one dimension being 0) and thus, trying to append
to an empty array results in type cast errors, since empty matrices are
double by default.</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.blank_struct">
<code class="descclassname">Misc_utilities.</code><code class="descname">blank_struct</code><span class="sig-paren">(</span><em>A</em>, <em>blank_val</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.blank_struct" title="Permalink to this definition">¶</a></dt>
<dd><p>Utility function to create a blank structure, using an existing one
as a template.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>A:</strong></dt>
<dd><p class="first last">template structure</p>
</dd>
<dt><strong>blank_val:</strong></dt>
<dd><p class="first last">all fields will be set to this value, e.g. [] or &#8216;&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>C:</strong></dt>
<dd><p class="first last">structure containing fields of A , with values set to empty array</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Only works for single-element non-nested structures, for now</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.checkMatlabVersion">
<code class="descclassname">Misc_utilities.</code><code class="descname">checkMatlabVersion</code><span class="sig-paren">(</span><em>datestr</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.checkMatlabVersion" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ok</span> <span class="o">=</span> <span class="n">checkMatlabVersion</span><span class="p">(</span><span class="n">datestr</span><span class="p">)</span>
</pre></div>
</div>
<p>check Matlab version against release date of version needed (or
recommended)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>datestr:</strong></dt>
<dd><p class="first last">format is one of the forms taken by datenum.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"><p class="first last">check for Matlab release date on or after Aug. 3, 2006</p>
</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ok</span> <span class="o">=</span> <span class="n">checkMatlabVersion</span><span class="p">(</span><span class="s">&#39;8/3/2006&#39;</span><span class="p">)</span> 
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.circle">
<code class="descclassname">Misc_utilities.</code><code class="descname">circle</code><span class="sig-paren">(</span><em>center</em>, <em>radius</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.circle" title="Permalink to this definition">¶</a></dt>
<dd><p>draws a circle</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[h, fillh] = circle(center, radius, [&#39;fill&#39;], [color string or numbers])
</pre></div>
</div>
<p>&#8216;fill&#8217; requires the geom2d toolbox, by David Legland</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.combine_structs">
<code class="descclassname">Misc_utilities.</code><code class="descname">combine_structs</code><span class="sig-paren">(</span><em>A</em>, <em>B</em>, <em>prefix</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.combine_structs" title="Permalink to this definition">¶</a></dt>
<dd><p>Utility function to combine the fields of two structures.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>A, B:</strong></dt>
<dd><p class="first last">structs to be combined</p>
</dd>
<dt><strong>prefix:</strong></dt>
<dd><p class="first last">string applied to fields of structure B</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>C:</strong></dt>
<dd><p class="first last">structure containing fields of both A and B</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Only works for single-element non-nested structures, for now</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.condf2indic">
<code class="descclassname">Misc_utilities.</code><code class="descname">condf2indic</code><span class="sig-paren">(</span><em>X</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.condf2indic" title="Permalink to this definition">¶</a></dt>
<dd><p>Create N x k indicator matrix of ones/zeros from N x 1 list of numeric
values (X)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">indic</span><span class="p">,</span> <span class="n">xlevels</span><span class="p">]</span> <span class="o">=</span> <span class="n">condf2indic</span><span class="p">(</span><span class="n">X</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>indic:</strong></dt>
<dd><p class="first last">is returned as single precision type, because it can then be
used as a design matrix in a GLM</p>
</dd>
<dt><strong>xlevels:</strong></dt>
<dd><p class="first last">are the values of X corresponding to columns of indic</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.depfun_aggregate">
<code class="descclassname">Misc_utilities.</code><code class="descname">depfun_aggregate</code><span class="sig-paren">(</span><em>funname</em>, <em>excludes</em>, <em>save_dir</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.depfun_aggregate" title="Permalink to this definition">¶</a></dt>
<dd><p>depfun_aggregate is for packaging all the files needed for a function in one spot</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">dependencies</span> <span class="o">=</span> <span class="n">depfun_aggregate</span><span class="p">(</span><span class="n">funname</span><span class="p">,</span> <span class="n">excludes</span><span class="p">,</span> <span class="n">save_dir</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>funname:</strong></dt>
<dd><p class="first last">function to analyze</p>
</dd>
<dt><strong>excludes:</strong></dt>
<dd><p class="first last">cellstrs to exclude from the matches</p>
</dd>
<dt><strong>save_dir:</strong></dt>
<dd><p class="first last">directory to copy files to - defaults to &#8216;tmp_files&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">dependencies</span> <span class="o">=</span> <span class="n">depfun_aggregate</span><span class="p">(</span><span class="s">&#39;whole_brain_fir&#39;</span><span class="p">,</span> <span class="p">{</span><span class="s">&#39;spm2&#39;</span><span class="p">,</span> <span class="s">&#39;spm5&#39;</span><span class="p">},</span> <span class="s">&#39;fir_files&#39;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.distance">
<code class="descclassname">Misc_utilities.</code><code class="descname">distance</code><span class="sig-paren">(</span><em>u</em>, <em>v</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.distance" title="Permalink to this definition">¶</a></dt>
<dd><p>Euclidean distance between u and v</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">d</span> <span class="o">=</span> <span class="n">distance</span><span class="p">(</span><span class="n">u</span><span class="p">,</span> <span class="n">v</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>u</strong> and <strong>v:</strong></dt>
<dd><p class="first">should be column vectors or matrices in k-dim space, where k is
the number of columns of u and v.</p>
<p class="last">if u is a single row, replicates u to size(v,1)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.distance_euclid">
<code class="descclassname">Misc_utilities.</code><code class="descname">distance_euclid</code><span class="sig-paren">(</span><em>u</em>, <em>v</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.distance_euclid" title="Permalink to this definition">¶</a></dt>
<dd><p>Euclidean distance between u and v</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">d</span> <span class="o">=</span> <span class="n">distance_euclid</span><span class="p">(</span><span class="n">u</span><span class="p">,</span> <span class="n">v</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>u</strong> and <strong>v:</strong></dt>
<dd><p class="first">should be column vectors or matrices in k-dim space, where k is
the number of columns of u and v.</p>
<p class="last">if u is a single row, replicates u to size(v,1)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.erase_and_display">
<code class="descclassname">Misc_utilities.</code><code class="descname">erase_and_display</code><span class="sig-paren">(</span><em>old_str</em>, <em>new_str</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.erase_and_display" title="Permalink to this definition">¶</a></dt>
<dd><p>Backspaces the number of chars of the length of old_str, then prints new_str</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.erase_string">
<code class="descclassname">Misc_utilities.</code><code class="descname">erase_string</code><span class="sig-paren">(</span><em>str</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.erase_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Backspaces the number of chars of the length of str</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.explode">
<code class="descclassname">Misc_utilities.</code><code class="descname">explode</code><span class="sig-paren">(</span><em>string</em>, <em>delimiters</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.explode" title="Permalink to this definition">¶</a></dt>
<dd><p>EXPLODE    Splits string into pieces.</p>
<p>EXPLODE(STRING,DELIMITERS) returns a cell array with the pieces
of STRING found between any of the characters in DELIMITERS.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">SPLIT</span><span class="p">,</span><span class="n">NUMPIECES</span><span class="p">]</span> <span class="o">=</span> <span class="n">EXPLODE</span><span class="p">(</span><span class="n">STRING</span><span class="p">,</span><span class="n">DELIMITERS</span><span class="p">)</span>
</pre></div>
</div>
<p>also returns the number of pieces found in STRING.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>STRING:</strong></dt>
<dd><p class="first last">the string to split (string)</p>
</dd>
<dt><strong>DELIMITERS:</strong></dt>
<dd><p class="first last">the delimiter characters (string)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>SPLIT:</strong></dt>
<dd><p class="first last">the split string (cell array), each cell is a piece</p>
</dd>
<dt><strong>NUMPIECES:</strong></dt>
<dd><p class="first last">the number of pieces found (integer)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">STRING</span> <span class="o">=</span> <span class="s">&#39;ab_c,d,e fgh&#39;</span>
<span class="n">DELIMITERS</span> <span class="o">=</span> <span class="s">&#39;_,&#39;</span>
<span class="p">[</span><span class="n">SPLIT</span><span class="p">,</span><span class="n">NUMPIECES</span><span class="p">]</span> <span class="o">=</span> <span class="n">EXPLODE</span><span class="p">(</span><span class="n">STRING</span><span class="p">,</span><span class="n">DELIMITERS</span><span class="p">)</span>
<span class="n">SPLIT</span> <span class="o">=</span> <span class="s">&#39;ab&#39;</span>    <span class="s">&#39;c&#39;</span>    <span class="s">&#39;d&#39;</span>    <span class="s">&#39;e fgh&#39;</span>
<span class="n">NUMPIECES</span> <span class="o">=</span> <span class="mi">4</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>IMPLODE, STRTOK</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.fast_conv_fft">
<code class="descclassname">Misc_utilities.</code><code class="descname">fast_conv_fft</code><span class="sig-paren">(</span><em>hrf</em>, <em>x</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.fast_conv_fft" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">y</span> <span class="o">=</span> <span class="n">fast_conv_fft</span><span class="p">(</span><span class="n">hrf</span><span class="p">,</span><span class="n">x</span><span class="p">)</span>
</pre></div>
</div>
<p>Much faster than conv or using matrix multiplication.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>hrf:</strong></dt>
<dd><p class="first last">should be length of x</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">y</span> <span class="o">=</span> <span class="n">fast_conv_fft</span><span class="p">(</span><span class="n">hrf</span><span class="p">,</span><span class="n">x</span><span class="p">);</span>
<span class="n">y</span> <span class="o">=</span> <span class="n">fast_conv_fft</span><span class="p">(</span><span class="n">hrf</span><span class="p">,</span><span class="n">x3</span><span class="p">,</span><span class="s">&#39;deconv&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.getRandom">
<code class="descclassname">Misc_utilities.</code><code class="descname">getRandom</code><span class="sig-paren">(</span><em>stimList</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.getRandom" title="Permalink to this definition">¶</a></dt>
<dd><p>Randomizes rows of a matrix, preserving dependencies between columns</p>
<blockquote>
<div><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stimList</span> <span class="o">=</span> <span class="n">getRandom</span><span class="p">(</span><span class="n">stimList</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><p class="first">a col. vector or matrix of stimulus conditions,</p>
<p class="last">e.g. [1 1 1 1 2 2 2 2 3 3 4 4]&#8217;</p>
</td>
</tr>
</tbody>
</table>
</div></blockquote>
<dl class="docutils">
<dt>output: a randomized permutation of this vector or matrix</dt>
<dd>all columns are resorted with the same order</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.get_first_help_lines">
<code class="descclassname">Misc_utilities.</code><code class="descname">get_first_help_lines</code><span class="sig-paren">(</span><em>functionname</em>, <em>maxlines</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.get_first_help_lines" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns the first n lines of the help for a function in a cell array</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">helptext</span> <span class="o">=</span> <span class="n">get_first_help_lines</span><span class="p">(</span><span class="n">functionname</span><span class="p">,</span> <span class="p">[</span><span class="n">maxlines</span><span class="p">])</span>

<span class="n">helptext</span> <span class="o">=</span> <span class="n">get_first_help_lines</span><span class="p">(</span><span class="n">functionname</span><span class="p">,</span> <span class="n">maxlines</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>helptext = get_first_help_lines(&#39;fmri_data.apply_mask&#39;, 5);
char(helptext{:})
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.implode">
<code class="descclassname">Misc_utilities.</code><code class="descname">implode</code><span class="sig-paren">(</span><em>pieces</em>, <em>delimiter</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.implode" title="Permalink to this definition">¶</a></dt>
<dd><p>IMPLODE    Joins strings with delimiter in between.</p>
<p>IMPLODE(PIECES,DELIMITER) returns a string containing all the
strings in PIECES joined with the DELIMITER string in between.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>PIECES:</strong></dt>
<dd><p class="first last">the pieces of string to join (cell array), each cell is a piece</p>
</dd>
<dt><strong>DELIMITER:</strong></dt>
<dd><p class="first last">the delimiter string to put between the pieces (string)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>STRING:</strong></dt>
<dd><p class="first last">all the pieces joined with the delimiter in between (string)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>PIECES = {&#39;ab&#39;,&#39;c&#39;,&#39;d&#39;,&#39;e fgh&#39;}
DELIMITER = &#39;-&gt;&#39;
STRING = IMPLODE(PIECES,DELIMITER)
STRING = ab-&gt;c-&gt;d-&gt;e fgh
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>EXPLODE, STRCAT</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.naninsert">
<code class="descclassname">Misc_utilities.</code><code class="descname">naninsert</code><span class="sig-paren">(</span><em>nanvec</em>, <em>y</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.naninsert" title="Permalink to this definition">¶</a></dt>
<dd><p>Insert NaNs back into data series from which they were removed</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">yout</span> <span class="o">=</span> <span class="n">naninsert</span><span class="p">(</span><span class="n">nanvec</span><span class="p">,</span> <span class="n">y</span><span class="p">)</span>
</pre></div>
</div>
<p>nanvec is indicator for NaNs, of size size(yout).  y is data.</p>
<p>If y is a matrix, length(nanvec) should equal number of rows in y.
NaNs are inserted for true values in nanvec in all columns of y.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>nanremove.m</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.nanremove">
<code class="descclassname">Misc_utilities.</code><code class="descname">nanremove</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.nanremove" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>varargout = nanremove(varargin)
[nanvec, x1, x2, etc.] = nanremove(x1,x2,etc...)
</pre></div>
</div>
<p>removes cases that are NaN in any column of any variable x1...xn</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.oneinsert">
<code class="descclassname">Misc_utilities.</code><code class="descname">oneinsert</code><span class="sig-paren">(</span><em>removed_voxels</em>, <em>data</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.oneinsert" title="Permalink to this definition">¶</a></dt>
<dd><p>Insert ones (for p values) back into data series (i.e., obj.p)
from which they were removed.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">yout</span> <span class="o">=</span> <span class="n">oneinsert</span><span class="p">(</span><span class="n">removed_voxels</span><span class="p">,</span> <span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
<p>removed_voxels is indicator for the locations of removed voxels</p>
<p>This function is a modified version of naninsert.m, but different.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>naninsert.m, nanremove.m, zeroinsert.m</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.orthviews_multiple_objs">
<code class="descclassname">Misc_utilities.</code><code class="descname">orthviews_multiple_objs</code><span class="sig-paren">(</span><em>imgs</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.orthviews_multiple_objs" title="Permalink to this definition">¶</a></dt>
<dd><p>plot multiple image objects on one orthviews</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body">cell array of image_vectors or statistic_images</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.pad">
<code class="descclassname">Misc_utilities.</code><code class="descname">pad</code><span class="sig-paren">(</span><em>x</em>, <em>L</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.pad" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">x</span> <span class="o">=</span> <span class="n">pad</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">L</span><span class="p">)</span>
</pre></div>
</div>
<p>pads x with zeros of length L</p>
<p>or, if L is a vector, to longest of two (NOT DONE)</p>
<p>COLUMN VECTORS</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.padwithnan">
<code class="descclassname">Misc_utilities.</code><code class="descname">padwithnan</code><span class="sig-paren">(</span><em>A</em>, <em>B</em>, <em>dim</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.padwithnan" title="Permalink to this definition">¶</a></dt>
<dd><p>returns the two input arrays, with the smaller padded to the size of the
larger in the particular dimension(s) with NaNs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[Anew Bnew] = padwithnan(A, B, dim)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.parse_char_to_cell">
<code class="descclassname">Misc_utilities.</code><code class="descname">parse_char_to_cell</code><span class="sig-paren">(</span><em>invar</em>, <em>sepval</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.parse_char_to_cell" title="Permalink to this definition">¶</a></dt>
<dd><p>take a row of characters and separate into cells, breaking at either
spaces or tabs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
<tr class="field-even field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">outcell</span> <span class="o">=</span> <span class="n">parse_char_to_cell</span><span class="p">(</span><span class="n">invar</span><span class="p">,</span><span class="n">sepval</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% copy from Excel as row, then parse:
disorder = [&#39;SAD    PTSD    PTSD    PTSD    PTSD    PTSD    SP      SAD     PTSD    PTSD    PTSD    PTSD    PTSD    SAD     SAD     PTSD    PTSD    SP      SP      SP      PTSD    PTSD    PTSD    PTSD    SAD     SAD     SAD     SP      SAD     SAD     SAD     SP      SP      SP      SAD     SP      PTSD    SP      PTSD    PTSD&#39;];
disorder = parse_char_to_cell(disorder, &#39;tab&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.parse_edat_txt">
<code class="descclassname">Misc_utilities.</code><code class="descname">parse_edat_txt</code><span class="sig-paren">(</span><em>fname</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.parse_edat_txt" title="Permalink to this definition">¶</a></dt>
<dd><p>Reads EPrime .txt output (equivalent to EDAT) directly into matlab cell arrays/structures</p>
<p>Inputs:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>fname:</strong></dt>
<dd>name of file to parse</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>edat_struct:</strong></dt>
<dd><p class="first">Structure containing the following fields:</p>
<p>header: 1-element structure whose fields are header items from EDAT</p>
<p>run: 1-element structure whose fields are run-specific items from EDAT</p>
<dl class="last docutils">
<dt>trials: n-element structure whose fields are trial-specific items from EDAT</dt>
<dd><p class="first last">where n is the number of trials</p>
</dd>
</dl>
</dd>
<dt><strong>edat_cells:</strong></dt>
<dd><p class="first">Structure containing the following fields:</p>
<p>header_cols: 1-row cell array whose fields are header column names from EDAT</p>
<p>run_cols: 1-row cell array whose cells are run-specific column names from EDAT</p>
<p>trials_cols: 1-row cell array whose cells are trial-specific column names from EDAT</p>
<p>header: 1-row cell array whose cells are header items from EDAT</p>
<p>run: 1-row cell array whose cells are run-specific items from EDAT</p>
<dl class="last docutils">
<dt>trials: n-row cell array whose cells are trial-specific items from EDAT</dt>
<dd><p class="first last">where n is the number of trials</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>Two passes are necessary:</dt>
<dd><ol class="first last arabic simple">
<li>find all fields used in the file</li>
<li>actually read the data</li>
</ol>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.print_matrix">
<code class="descclassname">Misc_utilities.</code><code class="descname">print_matrix</code><span class="sig-paren">(</span><em>x</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.print_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>prints matrix values as tab delimited, 2 decimal places</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>print_matrix(x,[col names cell array], [row names cell], [format string])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>t = [1 2; 3 4; 5 6];
print_matrix(t,{&#39;col1&#39; &#39;col2&#39;},{&#39;row1&#39; &#39;row2&#39; &#39;row3&#39;);

print_matrix(rand(5), [], [], &#39;%3.2f&#39;);
print_matrix(rand(5), {&#39;A&#39; &#39;B&#39; &#39;C&#39; &#39;D&#39; &#39;E&#39;}, {&#39;A&#39; &#39;B&#39; &#39;C&#39; &#39;D&#39; &#39;E&#39;}, &#39;%3.2f&#39;);
print_matrix(rand(5), {&#39;A&#39; &#39;B&#39; &#39;C&#39; &#39;D&#39; &#39;E&#39;}, {&#39;A&#39; &#39;B&#39; &#39;C&#39; &#39;D&#39; &#39;E&#39;}, &#39;%d&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.progressbar">
<code class="descclassname">Misc_utilities.</code><code class="descname">progressbar</code><span class="sig-paren">(</span><em>meth</em>, <em>val</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.progressbar" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a progress bar window with tag &#8216;progressbar&#8217;
that can be updated</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">progressbar</span><span class="p">(</span><span class="n">meth</span><span class="p">,</span><span class="n">val</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>meth:</strong></dt>
<dd><p class="first last">can be &#8216;init&#8217; or &#8216;update&#8217;</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>x-axis limits are 0 - 100, so val should be % complete for best results</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body">the f and ax handles are not really needed
as &#8216;update&#8217; finds the axis with &#8216;progressbar&#8217; tag.</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>progressbar(&#39;update&#39;,100*i./nvars);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.read_edat_output_2008">
<code class="descclassname">Misc_utilities.</code><code class="descname">read_edat_output_2008</code><span class="sig-paren">(</span><em>fname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.read_edat_output_2008" title="Permalink to this definition">¶</a></dt>
<dd><p>Function that creates a structure DATA containing columns
of the edat file output (saved in text tab delimited &#8220;excel&#8221; format)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">DATA</span> <span class="o">=</span> <span class="n">read_edat_output_2008</span><span class="p">(</span><span class="n">fname</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>For this code to work on a Mac, you must: 1) export .edat2 file as an Excel file, 
then 2) open this file in Excel on a Mac and save as a .csv, 3) read that
.csv file</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fname</span> <span class="o">=</span> <span class="s">&#39;myfile.txt&#39;</span><span class="p">;</span> 
<span class="n">DATA</span> <span class="o">=</span> <span class="n">read_edat_output_2008</span><span class="p">(</span><span class="n">fname</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Defaults:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>These are the default formats this function expects:
tab delimited, 1 header row, then row of column names, then data</p>
<p>You can override some of them by using the following &#8211;
E.g., for zero header rows and comma delimited data:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">DATA</span> <span class="o">=</span> <span class="n">read_edat_output_2008</span><span class="p">(</span><span class="n">fname</span><span class="p">,</span> <span class="s">&#39;nheaderrows&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="s">&#39;mydelimiter&#39;</span><span class="p">,</span> <span class="s">&#39;,&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>You can force the number of columns to be a certain value by doing the
following:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">DATA</span> <span class="o">=</span> <span class="n">read_edat_output_2008</span><span class="p">(</span><span class="n">fname</span><span class="p">,</span> <span class="s">&#39;nheaderrows&#39;</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="s">&#39;numc&#39;</span><span class="p">,</span> <span class="mi">103</span><span class="p">);</span>
</pre></div>
</div>
<p>This could be useful if your last row contains empty cells at the end,
which will mess up the automatic calculation of number of columns.</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.robustcsvread">
<code class="descclassname">Misc_utilities.</code><code class="descname">robustcsvread</code><span class="sig-paren">(</span><em>filename</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.robustcsvread" title="Permalink to this definition">¶</a></dt>
<dd><p>ROBUSTCSVREAD reads in CSV files with different number of columns
on different lines</p>
<p>This returns a struct, with one field per column of the csv file.
Each field is a cell array whose length = rows in the csv file.  Column
names are assumed to be in the first row.
If column names are invalid struct field names, edits them by replacing
funky characters with an underscore, or if first char is a number, I
prepend <a href="#id67"><span class="problematic" id="id68">aa_</span></a> to the field name.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>varargin:</strong></dt>
<dd><p class="first">cols:  how many cols to read in, by defaults reads them all</p>
<p>rows_to_skip:  how many rows to skip</p>
<p>delim:  cell delimiter</p>
<dl class="last docutils">
<dt>missing:  followed by cell array, first cell is val for missing,</dt>
<dd><p class="first last">second cell is what to replace with</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.scn_get_datetime">
<code class="descclassname">Misc_utilities.</code><code class="descname">scn_get_datetime</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.scn_get_datetime" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="nb">str</span> <span class="o">=</span> <span class="n">scn_get_datetime</span>
</pre></div>
</div>
<p>pass in &#8216;ymd&#8217; to get the string in yyyy_mm_dd-HH_MM format, so that
alphanumeric order will correspond to chronological order</p>
<p>Returns a string with the date and time
Useful for annotating data and output</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.scn_mat_conform">
<code class="descclassname">Misc_utilities.</code><code class="descname">scn_mat_conform</code><span class="sig-paren">(</span><em>in</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.scn_mat_conform" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function in = scn_mat_conform(in)
</pre></div>
</div>
<p>sets flipping to 0 (no flip) in SPM2 and adjusts mat file accordingly
input in spm-style mat file or struct with .mat or .M fields</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.search_struct_fields">
<code class="descclassname">Misc_utilities.</code><code class="descname">search_struct_fields</code><span class="sig-paren">(</span><em>search_struct</em>, <em>fieldname</em>, <em>fieldpath</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.search_struct_fields" title="Permalink to this definition">¶</a></dt>
<dd><p>Returns a list of all paths inside structure search_struct that match
the fieldname (or start with it)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">found_paths</span> <span class="o">=</span> <span class="n">search_struct_fields</span><span class="p">(</span><span class="n">search_struct</span><span class="p">,</span> <span class="n">fieldname</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>foo = [];
foo.foo = [];
foo.foo.foo = [];
search_struct_fields(foo, &#39;foo&#39;)
ans = 
   &#39;foo.foo&#39;
   &#39;foo.foo.foo&#39;

% or

search_struct_fields(SPM, &#39;x&#39;)
ans = 
   &#39;SPM.xX&#39;
   &#39;SPM.xM&#39;
   &#39;SPM.xsDes&#39;
   &#39;SPM.xX.xVi&#39;
   &#39;SPM.xX.xKXs&#39;
   &#39;SPM.xM.xs&#39;
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.strip_path_dirs">
<code class="descclassname">Misc_utilities.</code><code class="descname">strip_path_dirs</code><span class="sig-paren">(</span><em>patterns</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.strip_path_dirs" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes directories from the Matlab path, based on the rgex patterns passed in.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function strip_path_dirs(regexes)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.strip_svn_dirs">
<code class="descclassname">Misc_utilities.</code><code class="descname">strip_svn_dirs</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.strip_svn_dirs" title="Permalink to this definition">¶</a></dt>
<dd><p>Removes the .svn dirs from the path</p>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.strrep_recurse">
<code class="descclassname">Misc_utilities.</code><code class="descname">strrep_recurse</code><span class="sig-paren">(</span><em>old_var</em>, <em>old_string</em>, <em>new_string</em>, <em>depth</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.strrep_recurse" title="Permalink to this definition">¶</a></dt>
<dd><p>Recursively traverses depth-first through an entire variable, replacing
old_string with new_string everywhere it goes</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">new_var</span> <span class="o">=</span> <span class="n">strrep_recurse</span><span class="p">(</span><span class="n">old_var</span><span class="p">,</span> <span class="n">old_string</span><span class="p">,</span> <span class="n">new_string</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.struct_strrep">
<code class="descclassname">Misc_utilities.</code><code class="descname">struct_strrep</code><span class="sig-paren">(</span><em>old_struct</em>, <em>old_string</em>, <em>new_string</em>, <em>depth</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.struct_strrep" title="Permalink to this definition">¶</a></dt>
<dd><p>Traverses depth-first through an entire structure, replacing 
old_string with new_string everywhere it goes</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">new_struct</span> <span class="o">=</span> <span class="n">struct_strrep</span><span class="p">(</span><span class="n">old_struct</span><span class="p">,</span> <span class="n">old_string</span><span class="p">,</span> <span class="n">new_string</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.tor_ga">
<code class="descclassname">Misc_utilities.</code><code class="descname">tor_ga</code><span class="sig-paren">(</span><em>gensize</em>, <em>numgen</em>, <em>inputs</em>, <em>ofun</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.tor_ga" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[best_params,fit,beff,in,isconverged] = tor_ga(gensize,numgen,inputs,ofun,[optional in any order: genfun,fixed inputs,cmd strings])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body">a cell array describing the inputs to the optimization function
(parameters to be optimized).</td>
</tr>
</tbody>
</table>
<p>Each cell of inputs is a p x q matrix of parameters.
p and q are arbitrary, as each organism is described by a p x q
matrix...but the objective function must be able to handle inputs in
the format you provide.
Internally, a set of &#8216;organisms&#8217; is created that is
params x params x organisms (3-D).
This matrix is subject to crossover across orgs. separately for each
cell.
If inputs is a p x q matrix, it will be placed in a single cell.</p>
<p>By default, it is only necessary to enter a single set of params
for an example organism.  The range of those input values is used to
generate random starting values for each organism.</p>
<p>There can be more than one set of
parameters that are combined in some way by ofun to produce a fitness
value.  if there is more than one set of input parameters,
inputs should be entered as a cell array, one cell per input.
inputs should be in ORDER of inputs entered to ofun!</p>
<p><strong>RECOMMENDED</strong></p>
<p>If you enter each cell of inputs as a 3-D array so that inputs(:,:,1)
is the min acceptable value for each param and inputs(:,:,2) is the
max acceptable value, then the ga will create a series of organisms
at start that evenly span the range of the multivariate parameter
space, with the spacing between values determined by the gensize.
This can provide a huge advantage in efficiency for the GA.
This is the idea behind the &#8220;Sobol sequence,&#8221; which chooses values
that evenly span a multivariate space.
With this option, if gensize is sufficiently large and the param
space is sufficiently small, then the ga may find the correct
solution on the first iteration.
However, it is not likely to work well if the num. params &gt;&gt; gensize</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>start = [-15 -15];
start(:,:,2) = [15 15];
[best_params,fit,beff,in] = tor_ga(324,30,{start},objfun_ga,&#39;genconverge&#39;,5);
</pre></div>
</div>
<dl class="docutils">
<dt><strong>ofun</strong></dt>
<dd><ul class="first last simple">
<li>the objective function that combines the inputs.</li>
</ul>
</dd>
<dt>There are two options for passing this in:</dt>
<dd><ol class="first last arabic">
<li><p class="first">enter the name of the function as a string.  the program creates a handle for the
function, and evaluates it using inputs specified in the inputs variable.
In this case, pass in fixed inputs after ofun, in the varargin fields
fixed inputs  optional, fixed inputs that do not change!  same structure
as inputs.</p>
</li>
<li><p class="first">You can also enter ofun as a function handle durectly, with fixed inputs already
embedded before running the program.
The function should take as input a param list, and return fitness.
e.g.,</p>
<div class="highlight-python"><div class="highlight"><pre>objhan = @(params) my_function_name(params,fixed_inputs1,fixed_inputs1,fixed_inputs1);
objhan = @(wh) prospect_organism(ceil(wh),pop,truep,iter);
</pre></div>
</div>
</li>
</ol>
</dd>
</dl>
<p>Pass in objhan as the &#8216;ofun&#8217; input argument</p>
<dl class="docutils">
<dt><strong>genfun</strong></dt>
<dd><ul class="first last simple">
<li>[optional] input param generation function</li>
</ul>
</dd>
</dl>
<p>A function handle that generates a parameter set for each organism</p>
<dl class="docutils">
<dt><strong>command strings</strong></dt>
<dd><ul class="first last simple">
<li>&#8216;noverbose&#8217;   turn off verbose reporting and plots</li>
<li>&#8216;genconverge&#8217; followed by integer x: converge if no change in last x generations</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Example for fitting indscal model:</p>
<div class="highlight-python"><div class="highlight"><pre>inputs{1} = X; fixin{1} = sp; fixin{2} = B1; fixin{3} = B2;
tor_ga(30,10,inputs,&#39;indscalf&#39;,fixin);

W = rand(size(W)); W(1,:) = [10 10];, W(2,:) = [-10 -10];
inputs{2} = W;
</pre></div>
</div>
<p>Example: Optimize gambles for prospect theory model
See prospect_optimize_design.m for definition of population of
gambles from which to draw (pop), truep, iter (all fixed inputs)</p>
<div class="highlight-python"><div class="highlight"><pre>objhan = @(wh) prospect_organism(ceil(wh),pop,truep,iter);
genfun = @() randsample(gindx,ntrials,&#39;true&#39;)&#39;;
[best_params,fit,beff,in] = tor_ga(5,3,wh,objhan,genfun);
</pre></div>
</div>
<p>Using string inputs to control behavior:</p>
<div class="highlight-python"><div class="highlight"><pre>[best_params,fit,beff,in] = tor_ga(300,30,{[15; -15]},objfun_ga,&#39;genconverge&#39;,5,&#39;noverbose&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Misc_utilities.zeroinsert">
<code class="descclassname">Misc_utilities.</code><code class="descname">zeroinsert</code><span class="sig-paren">(</span><em>wasbad</em>, <em>y</em><span class="sig-paren">)</span><a class="headerlink" href="#Misc_utilities.zeroinsert" title="Permalink to this definition">¶</a></dt>
<dd><p>Re-insert removed CASES (rows) and fill with zeros</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">yout</span> <span class="o">=</span> <span class="n">zeroinsert</span><span class="p">(</span><span class="n">wasbad</span><span class="p">,</span> <span class="n">y</span><span class="p">)</span>
</pre></div>
</div>
<p>wasbad is indicator for removed cases, of size size(yout).  y is data.</p>
<p>if you enter y&#8217;, inserts VARIABLES (cols).  here, pass in v x n matrix, y&#8217;
to fill empty/removed vars</p>
<p>See nanremove.m and naninsert.m</p>
</dd></dl>

</div>
<div class="section" id="model-building-tools">
<h2>Model_building_tools<a class="headerlink" href="#model-building-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Model_building_tools"></span><dl class="function">
<dt id="Model_building_tools.fmri_spline_basis">
<code class="descclassname">Model_building_tools.</code><code class="descname">fmri_spline_basis</code><span class="sig-paren">(</span><em>TR</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.fmri_spline_basis" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>xBF = spline_hrf_basis(TR, optional args)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>imTR:</strong></dt>
<dd><p class="first last">repetition time; sampling resolution of data</p>
</dd>
<dt><strong>&#8216;plot&#8217;:</strong></dt>
<dd><p class="first last">optional: plot basis set</p>
</dd>
<dt><strong>&#8216;nbasis&#8217;:</strong></dt>
<dd><p class="first last">optional: number of knot points</p>
</dd>
<dt><strong>&#8216;order&#8217;:</strong></dt>
<dd><p class="first last">optional: order of spline model (# matched derivatives)</p>
</dd>
<dt><strong>&#8216;length&#8217;:</strong></dt>
<dd><p class="first last">optional: length of window to model, in seconds</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Outputs:
:Inputs:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>xBF.dt:</strong></dt>
<dd>time bin length {seconds}</dd>
<dt><strong>xBF.name:</strong></dt>
<dd>description of basis functions specified</dd>
<dt><strong>xBF.length:</strong></dt>
<dd>window length (seconds)</dd>
<dt><strong>xBF.order:</strong></dt>
<dd>order</dd>
<dt><strong>xBF.bf:</strong></dt>
<dd><p class="first">Matrix of basis functions</p>
<p class="last">32 second long spline basis set for fmri model</p>
</dd>
<dt><strong>xBF_hires:</strong></dt>
<dd>Sampled at high resolution, TR * 16</dd>
<dt><strong>xBF:</strong></dt>
<dd>Sampled at TR</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">xBF_hires</span><span class="p">,</span> <span class="n">xBF</span><span class="p">]</span> <span class="o">=</span> <span class="n">fmri_spline_basis</span><span class="p">(</span><span class="n">TR</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
<span class="p">[</span><span class="n">xBF_hires</span><span class="p">,</span> <span class="n">xBF</span><span class="p">]</span> <span class="o">=</span> <span class="n">fmri_spline_basis</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span> <span class="s">&#39;length&#39;</span><span class="p">,</span> <span class="mi">12</span><span class="p">,</span> <span class="s">&#39;nbasis&#39;</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="s">&#39;order&#39;</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="s">&#39;plot&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.getPredictors">
<code class="descclassname">Model_building_tools.</code><code class="descname">getPredictors</code><span class="sig-paren">(</span><em>stimList</em>, <em>HRF</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.getPredictors" title="Permalink to this definition">¶</a></dt>
<dd><p>Build predictors and delta functions, given a condition function or delta
function and either a convolution matrix or vector.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">model</span><span class="p">,</span><span class="n">delta</span><span class="p">]</span> <span class="o">=</span> <span class="n">getPredictors</span><span class="p">(</span><span class="n">stimList</span><span class="p">,</span> <span class="n">HRF</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>IMPORTANT: YOU MUST ADD THE INTERCEPT YOURSELF!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>stimList:</strong></dt>
<dd><p class="first last">condition function OR delta function (1/0 indicator)</p>
</dd>
<dt><strong>HRF:</strong></dt>
<dd><ol class="first last arabic">
<li><p class="first">hemodynamic response function</p>
</li>
<li><p class="first">Basis set (columns)</p>
</li>
<li><p class="first">or convolution matrix (columns
are HRF), defined as:
HRF = tril(toeplitz(hrf));</p>
<p>multiple column vectors for HRF are treated as basis functions!</p>
</li>
</ol>
</dd>
<dt><strong>varargin</strong> for downsampleing:</dt>
<dd><p class="first">&#8216;dsrate&#8217;: takes every nth element of the design matrix</p>
<p class="last">&#8216;dslen&#8217;: the target number (length) you want to downsample to</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Other Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;force_delta&#8217;:</strong></dt>
<dd><p class="first last">getPredictors tries to determine if the input stimList
is a condition function with integers or a delta function with
indicators, but this can fail in some cases.  Use this to force
it to treat as a delta function.</p>
</dd>
</dl>
<ol class="arabic">
<li><p class="first">a col. vector of stimulus conditions OR a delta function matrix</p>
</li>
<li><dl class="first docutils">
<dt>an HRF vector sampled at the frequency of the stimulus vector, OR</dt>
<dd><p class="first last">a convolution matrix H (empty for default)</p>
</dd>
</dl>
</li>
<li><p class="first">Optional: downsampling factor for final design (i.e., TR)</p>
</li>
<li><p class="first">Optional: parametric modulator keyword and modulator values</p>
<dl class="docutils">
<dt>&#8216;parametric_singleregressor&#8217; <span class="classifier-delimiter">:</span> <span class="classifier">Parametrically modulate onsets by</span></dt>
<dd><blockquote class="first">
<div><p>modulator values, using single regressor with modulated amplitude
Enter a cell array with modulator values for each event
type, with a column vector (empty cell for no modulation)</p>
</div></blockquote>
<dl class="last docutils">
<dt>&#8216;parametric_standard&#8217; <span class="classifier-delimiter">:</span> <span class="classifier">Parametrically modulate onsets by</span></dt>
<dd><p class="first last">modulator values, using two regressors per event type - One
to model the average response, and one for the
mean-centered modulator values</p>
</dd>
</dl>
</dd>
</dl>
</li>
</ol>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><ol class="first arabic simple">
<li>a n x 2 matrix of regressors (cols) for each condition</li>
<li>a n x k delta matrix with onsets</li>
</ol>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"><p class="first last">TR = 2, 16 samples per second in hi-res delta dhr</p>
</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">X</span> <span class="o">=</span> <span class="n">getPredictors</span><span class="p">(</span><span class="n">dhr</span><span class="p">,</span> <span class="n">hrf</span><span class="p">,</span> <span class="s">&#39;dsrate&#39;</span><span class="p">,</span> <span class="n">res</span><span class="o">*</span><span class="n">TR</span><span class="p">);</span> 
<span class="n">X</span> <span class="o">=</span> <span class="n">getPredictors</span><span class="p">(</span><span class="n">dhr</span><span class="p">,</span> <span class="n">hrf</span><span class="p">,</span> <span class="s">&#39;dslen&#39;</span><span class="p">,</span> <span class="nb">len</span><span class="o">/</span><span class="p">(</span><span class="n">res</span><span class="o">*</span><span class="n">TR</span><span class="p">));</span> 
</pre></div>
</div>
<p>stimList can be condition function e.g., [1 3 2 4 3 2 1]&#8217; or
delta matrix (n x k), n samples and k conditions, e.g., [1 0 0 0 1 0 1]&#8217;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Resampling:</th><td class="field-body">the default N in matlab resample has built-in antialiasing,</td>
</tr>
</tbody>
</table>
<p>but may not be good for fmri designs!  The appropriate downsampling
is expected to be res*TR (res is units of samples/s), but we use 0
because the model will depend on the analysis method used, and this is
the most veridical approach.  With N = 0, every ith sample is used, where
i is the downsampling factor you input.  Popular choices are 16*TR (for
onsets2delta.m), using the SPM default res of 16.
Delta is NOT resampled.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body">TR = 2, 16 samples per second in hi-res delta dhr</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">tmp</span><span class="p">,</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">downsample_delta</span><span class="p">(</span><span class="n">dhr</span><span class="p">,</span><span class="mi">16</span><span class="o">*</span><span class="mi">2</span><span class="p">);</span> <span class="n">X</span><span class="o">=</span><span class="n">getPredictors</span><span class="p">(</span><span class="n">d</span><span class="p">,</span><span class="n">hrf</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.ideal_deconv6">
<code class="descclassname">Model_building_tools.</code><code class="descname">ideal_deconv6</code><span class="sig-paren">(</span><em>conditions</em>, <em>mspec</em>, <em>ttype</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.ideal_deconv6" title="Permalink to this definition">¶</a></dt>
<dd><p>Tests deconvolution matrix directly against idealized data
you put in the exact temporal sequence to be deconvolved,
in the form of the DX matrix.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">rmsd</span><span class="p">,</span><span class="n">msdstd</span><span class="p">,</span><span class="n">msdb</span><span class="p">,</span><span class="n">biasmean</span><span class="p">,</span><span class="n">meanest</span><span class="p">,</span><span class="n">min95est</span><span class="p">,</span><span class="n">max95est</span><span class="p">,</span><span class="n">ALLINFO</span><span class="p">,</span><span class="n">hrf</span><span class="p">,</span><span class="n">snr</span><span class="p">,</span><span class="n">TR</span><span class="p">]</span> <span class="o">=</span> <span class="n">ideal_deconv6</span><span class="p">(</span><span class="n">conditions</span><span class="p">,</span><span class="n">mspec</span><span class="p">,</span><span class="n">ttype</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>DX:</strong></dt>
<dd><p class="first last">deconvolution matrix</p>
</dd>
<dt><strong>tp:</strong></dt>
<dd><p class="first last">time points estimated for each condition in DX</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">repetition time of scan</p>
</dd>
<dt><strong>ttype:</strong></dt>
<dd><p class="first last">trial types to test (out of 1:n different conditions in DX)
recommended for time saving to use ttype = a single number only</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This function is like ideal_deconv5, but tests variability across designs as well.</p>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.intercept_model">
<code class="descclassname">Model_building_tools.</code><code class="descname">intercept_model</code><span class="sig-paren">(</span><em>nvols_per_run</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.intercept_model" title="Permalink to this definition">¶</a></dt>
<dd><p>Build design matrix X for intercepts
given vector of session lengths [s1 s2 s3] in images</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>x = intercept_model(nvols_per_run, [indx of dummy scans in each session])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>nvols_per_run = [166 166 144 137];
x = intercept_model(nvols_per_run);

x = intercept_model(repmat(166, 1, 5));

Xi = intercept_model(EXPT.FIR.nruns, 1:2);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.modifiedconv">
<code class="descclassname">Model_building_tools.</code><code class="descname">modifiedconv</code><span class="sig-paren">(</span><em>tr</em>, <em>condf</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.modifiedconv" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"><div class="first highlight-python"><div class="highlight"><pre>model = modifiedconv(tr,condf,heighteq [all opt],delayeq,ttopeakeq,uonseteq)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>tr:</strong></dt>
<dd><p class="first last">repetition time (sampling rate) of scanning, in seconds</p>
</dd>
<dt><strong>condf:</strong></dt>
<dd><dl class="first last docutils">
<dt>condition function</dt>
<dd><p class="first last">an indicator vector of zeros and ones, where ones indicate event
onsets</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>USES nonlinear saturation in height only
with a guess as to what the decrease in saturation is as a function of
the time since previous stimulation (exponential model, alpha version)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>condf = [1 1 0 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0]&#39;;
X = modifiedconv(2,condf);
% X is convolved predictor
plot(X)
X2 = conv(condf,spm_hrf(2)./max(spm_hrf(2)));
hold on;
plot(X2,&#39;r&#39;);
legend({&#39;Modified&#39; &#39;Linear&#39;})
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Please see:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p class="last">Wager, T. D., Hernandez, L., Vasquez, A., Nichols, T., and Noll, D.
C. (in press). Accounting for nonlinear BOLD effects in fMRI: Parameter 
estimates and model for accurate prediction in variable-duration blocked 
and rapid event-related studies.  Neuroimage.</p>
</td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>heighteq = []; delayeq = []; peakeq = []; uonseteq = [];</dt>
<dd><ul class="first last simple">
<li>defaults</li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.onsets2delta">
<code class="descclassname">Model_building_tools.</code><code class="descname">onsets2delta</code><span class="sig-paren">(</span><em>ons</em>, <em>len</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.onsets2delta" title="Permalink to this definition">¶</a></dt>
<dd><p>Builds high-res delta function, given cell array of onset times</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>delta = onsets2delta(ons, [length: num rows in original units])
</pre></div>
</div>
<p>Tor Wager, 2 / 24 / 04, update 10 / 12 / 10</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>ons:</strong></dt>
<dd><p class="first">onsets for each of a series of conditions</p>
<p>One cell per condition per session, e.g., ons{1} = [24 27 29 44]&#8217;;</p>
<p>Units are arbitrary (e.g., TRs or seconds)</p>
<p>Onsets are assumed to start at time 0 (0 is start of run/session)</p>
<p>e.g., from an SPM.mat fmri design structure, for one session:</p>
<div class="last highlight-python"><div class="highlight"><pre>ons = cell(1, nconds);
[ons{:}] = deal(reportmod_model.Sess(1).U(:).ons)
</pre></div>
</div>
</dd>
<dt><strong>len:</strong></dt>
<dd><blockquote class="first">
<div><p>optional: number of rows in original units.  Useful for making</p>
</div></blockquote>
<p class="last">a design matrix with the right number of rows after convolution and downsampling</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>delta:</strong></dt>
<dd><p class="first">indicator matrix of vectors for each condition with 1/0 for each onset</p>
<p>Type is logical; you may want to do double(delta) before operating</p>
<p>Resolution of high-res delta functions = original units (secs or TRs) * 16</p>
<p class="last">The number of rows is the max onset + 1, times 16</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>For what to do with output, see:
getPredictors : for design-matrix building
downsample_canlab : for downsampling to TR/secs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body">ONSETS2FMRIDESIGN and object-oriented fmri_model object</td>
</tr>
</tbody>
</table>
<p>(methods: build, etc.)</p>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.onsets2dx">
<code class="descclassname">Model_building_tools.</code><code class="descname">onsets2dx</code><span class="sig-paren">(</span><em>onsets</em>, <em>TR</em>, <em>scansperrun</em>, <em>numseconds</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.onsets2dx" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">DX</span><span class="p">,</span><span class="n">delta</span><span class="p">]</span> <span class="o">=</span> <span class="n">onsets2dx</span><span class="p">(</span><span class="n">onsets</span><span class="p">,</span> <span class="n">TR</span><span class="p">,</span> <span class="n">scansperrun</span><span class="p">,</span> <span class="n">numseconds</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>onsets:</strong></dt>
<dd><p class="first last">cell array whose length is num runs * num conditions,
e.g., {run 1 ant onsets, run 1 stim onsets, run 2 ant onsets,
run 2 stim onsets}</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">TR in seconds</p>
</dd>
<dt><strong>scansperrun:</strong></dt>
<dd><p class="first last">number of volumes in each run</p>
</dd>
<dt><strong>numseconds:</strong></dt>
<dd><p class="first last">number of seconds after event onsets to generate regressors for [default: 32]</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>TR = 2;
scansperrun = [192 196 196 184 190 192];
numseconds = 30;
[DX,delta] = onsets2dx(onsets, TR, scansperrun, numseconds)
EXPT.FIR.model{subjectnumber} = DX;
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.onsets2fmridesign">
<code class="descclassname">Model_building_tools.</code><code class="descname">onsets2fmridesign</code><span class="sig-paren">(</span><em>ons</em>, <em>TR</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.onsets2fmridesign" title="Permalink to this definition">¶</a></dt>
<dd><p>Builds design matrix X and delta function, given cell array of onset times in s
One cell per condition per session, e.g., ons{1} = [24 27 29 44]&#8217;;</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[X, delta, delta_hires, hrf] = onsets2fmridesign(onsets, TR, [len], [custom hrf or basis set name],[other optional args, i.e., norm])
</pre></div>
</div>
<dl class="docutils">
<dt>Summary:</dt>
<dd><ul class="first last simple">
<li>handles multiple conditions</li>
<li>handles custom HRFs and multiple basis functions</li>
<li>handles input event durations</li>
<li>handles two kinds of parametric modulators</li>
<li>handles variable-duration onsets</li>
<li>handles nonlinear saturation (see hrf_saturation.m)</li>
<li>Can build single-trial model</li>
<li>not yet: variable-duration parametric
modulators</li>
<li>See the code comments for a discussion of absolute scaling of
regressors and efficiency.</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>onsets:</strong></dt>
<dd><ul class="first last simple">
<li>1st column is onsets for events,</li>
<li>2nd column is optional durations for each event</li>
<li>Enter single condition or cell vector with cells for each condition (each event type).</li>
</ul>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">RT in seconds</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">First two are fixed, then keywords:</p>
<dl class="last docutils">
<dt><strong>len:</strong></dt>
<dd><p class="first last">optional length in s for model, or [] to skip if using additional. 
&#8220;len&#8221; is usually the number of images multiplied by TR.</p>
</dd>
<dt><strong>HRF name:</strong></dt>
<dd><dl class="first last docutils">
<dt>string or actual values</dt>
<dd><ol class="first last arabic simple">
<li>a string used by spm_get_bf.m</li>
<li>a custom HRF, sampled in seconds</li>
<li>or [] to skip</li>
</ol>
</dd>
</dl>
</dd>
<dt><strong>&#8216;norm&#8217;:</strong></dt>
<dd><p class="first last">mean-center, orthogonalize, and L2-norm basis set</p>
</dd>
<dt><strong>&#8216;parametric_singleregressor&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>Parametrically modulate onsets by</dt>
<dd><p class="first last">modulator values, using single regressor with modulated amplitude
Enter a cell array with modulator values for each event
type, with a column vector (empty cell for no modulation)</p>
</dd>
</dl>
</dd>
<dt><strong>&#8216;parametric_standard&#8217;:</strong></dt>
<dd><p class="first last">Parametrically modulate onsets by
modulator values, using two regressors per event type - One
to model the average response, and one for the
mean-centered modulator values
of modulator values in each cell</p>
</dd>
<dt><strong>&#8216;noampscale&#8217;:</strong></dt>
<dd><p class="first last">Do not scale HRF to max amplitude = 1; default with SPM
basis sets is to scale.  Custom HRF entries are not scaled.</p>
</dd>
<dt><strong>&#8216;noundershoot&#8217;:</strong></dt>
<dd><p class="first last">Do not model undershoot - ONLY when using the canonical HRF</p>
</dd>
<dt><strong>&#8216;customneural&#8217;:</strong></dt>
<dd><p class="first last">Followed by a vector of custom neural values (instead of
standard event/epochs), sampled in sec.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Limitation: can only handle two events max within the same TR</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first last">model, sampled at TR</p>
</dd>
<dt><strong>delta:</strong></dt>
<dd><p class="first last">indicator mtx, sampled at TR</p>
</dd>
<dt><strong>delta_hires:</strong></dt>
<dd><p class="first last">indicator sampled at 16 Hz</p>
</dd>
<dt><strong>hrf:</strong></dt>
<dd><p class="first last">hemodynamic response function, sampled at 16 Hz</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>X = onsets2fmridesign(ons, TR, size(imgs, 1) .* TR, &#39;hrf (with time derivative)&#39;);

X = onsets2fmridesign({[0 30 60]&#39; [15 45 90&#39;]}, 1.5, 180, spm_hrf(1), &#39;parametric_standard&#39;, {[2 .5 1]&#39; [1 2 2.5]&#39;}); figure; plot(X)
X = onsets2fmridesign({[0 30 60]&#39; [15 45 90&#39;]}, 1, 180, spm_hrf(1), &#39;parametric_standard&#39;, {[2 .5 1]&#39; [1 2 2.5]&#39;}); figure; plot(X)
X = onsets2fmridesign({[0 30 60]&#39; [15 45 90&#39;]}, 2, 180, spm_hrf(1), &#39;parametric_standard&#39;, {[2 .5 1]&#39; [1 2 2.5]&#39;}); figure; plot(X)
X = onsets2fmridesign({[0 30 60]&#39; [15 45 90&#39;]}, 2, 180, spm_hrf(1), &#39;parametric_singleregressor&#39;, {[2 .5 1]&#39; [1 2 2.5]&#39;}); figure; plot(X)
X = onsets2fmridesign({[0 30 60]&#39; [15 45 90&#39;]}, 1, 180, spm_hrf(1), &#39;parametric_singleregressor&#39;, {[2 .5 1]&#39; [1 2 2.5]&#39;}); figure; plot(X)

% Here, spm_hrf(1) is for the canonical HRF in spm.

X = onsets2fmridesign(ons, 2, size(dat.dat, 2) .* 2, [], [], &#39;noampscale&#39;);

X2 = onsets2fmridesign(onsp, 2, length(dat.removed_images) .* 2, [], [], &#39;customneural&#39;, report_predictor);
plotDesign(ons,[], TR, &#39;yoffset&#39;, 1);
</pre></div>
</div>
<p>A note on absolute scaling and efficiency:
Scaling of the response influences efficiency. It does not affect model
fits or power when the scaling is equated (constant across events in a
design), but it does affect efficiency simulations.</p>
<dl class="docutils">
<dt>Event duration:</dt>
<dd>assumes that the max neural sampling rate is about 1 Hz,
which produces responses that do not exceed about 5x the
unit, single-event response.  This is a reasonable
assumption, and affects only the absolute scaling across
ISIs/block lengths.</dd>
</dl>
<p>An event duration of about 1 sec produces a response of unit amplitude.
The sampling resolution is 0.1 sec, so that is the lowest you can go - and it will produce lower response amplitude/less efficient designs, as less neural activity is being delivered.
If event durations are not entered, then events will have unit amplitude by default.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">tor_make_deconv_mtx3, [or 2], plotDesign</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.onsets2parametric_mod_X">
<code class="descclassname">Model_building_tools.</code><code class="descname">onsets2parametric_mod_X</code><span class="sig-paren">(</span><em>ons</em>, <em>pm_vals</em>, <em>nscan</em>, <em>basisset</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.onsets2parametric_mod_X" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>ons = ons{1}; %obj.Sess(1).U(1).ons ./ TR;
pm_vals = obj.Sess(1).U(1).P.P;
nscan = obj.nscan(s);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.plot_ideal_deconv5">
<code class="descclassname">Model_building_tools.</code><code class="descname">plot_ideal_deconv5</code><span class="sig-paren">(</span><em>rmsd</em>, <em>msdstd</em>, <em>msdb</em>, <em>biasmean</em>, <em>meanest</em>, <em>min95est</em>, <em>max95est</em>, <em>INFO</em>, <em>hrf</em>, <em>snr</em>, <em>TR</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.plot_ideal_deconv5" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function plot_ideal_deconv5(rmsd,msdstd,msdb,biasmean,meanest,min95est,max95est,INFO,hrf,snr,TR,[truer])
</pre></div>
</div>
<p>delta should be matrix of column vectors</p>
<p>optional: truer, vector of &#8220;true&#8221; responses for each trial type</p>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.spm_mat2batchinput">
<code class="descclassname">Model_building_tools.</code><code class="descname">spm_mat2batchinput</code><span class="sig-paren">(</span><em>SPM</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.spm_mat2batchinput" title="Permalink to this definition">¶</a></dt>
<dd><p>Extract info from SPM structure and format in batch input mode</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">imgs</span><span class="p">,</span> <span class="n">TR</span><span class="p">,</span> <span class="n">scanspersess</span><span class="p">,</span> <span class="n">names</span><span class="p">,</span> <span class="n">onsets</span><span class="p">,</span> <span class="n">durations</span><span class="p">]</span> <span class="o">=</span> <span class="n">spm_mat2batchinput</span><span class="p">(</span><span class="n">SPM</span><span class="p">)</span>
</pre></div>
</div>
<p>:Input into SPM8:</p>
<div class="highlight-python"><div class="highlight"><pre>tmod = time modulator (number per cell, order, linear = 1)
</pre></div>
</div>
<p>For parametric modulation include a structure array, which is up to 1 x n in size, called pmod. n must be less
than  or  equal  to  the  number  of  cells in the names/onsets/durations cell arrays. The structure array pmod
must  have the fields: name, param and poly.  Each of these fields is in turn a cell array to allow the inclusion
R for regressors of no interest</p>
<p>NOTES:
This is a preliminary version of this function and does not extract
covariates, parametric modulators, etc.</p>
<p>Returns output in the format you would use to enter into the SPM GUI to
set up a model. Same format is used by canlab_spm_fmri_model_job.m</p>
</dd></dl>

<dl class="function">
<dt id="Model_building_tools.tor_make_deconv_mtx3">
<code class="descclassname">Model_building_tools.</code><code class="descname">tor_make_deconv_mtx3</code><span class="sig-paren">(</span><em>sf</em>, <em>tp</em>, <em>eres</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Model_building_tools.tor_make_deconv_mtx3" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [DX,sf] = tor_make_deconv_mtx(sf,tp,eres,[opt] TRs before stim onset,[num. sessions],[docenter],[scanspersess])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>sf:</strong></dt>
<dd><p class="first">cell array of stick functions, one per condition
all sf cells should be of the same length</p>
<p class="last">Or matrix of stick functions, 1 column per condition</p>
</dd>
<dt><strong>tp:</strong></dt>
<dd><p class="first last">number of timepoints to estimate in hrf deconvolution matrix</p>
</dd>
<dt><strong>eres:</strong></dt>
<dd><p class="first last">timebins in sf array for each TR</p>
</dd>
<dt><strong>DX:</strong></dt>
<dd><p class="first last">deconvolution matrix
estimates O.tp time points for each condition
Time resolution is in TRs</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Optional:</th><td class="field-body"><ol class="first arabic">
<li><p class="first">TRs before: 0 or number of time-points to shift LEFT</p>
</li>
<li><p class="first">number of sessions; if &gt; 1, adds session-specific intercepts</p>
</li>
<li><p class="first">docenter, 1/0 for do/do not center columns, default 0</p>
</li>
<li><dl class="first docutils">
<dt>scanspersess: how many scans per session?  prevents regressors from</dt>
<dd><p class="first last">running over into the next session (recursive).</p>
</dd>
</dl>
</li>
</ol>
<p>No parametric modulation of sf&#8217;s allowed.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>sf:</strong></dt>
<dd><p class="first last">stick function resampled at TR</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
<div class="section" id="parcellation-tools">
<h2>Parcellation_tools<a class="headerlink" href="#parcellation-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Parcellation_tools"></span><dl class="function">
<dt id="Parcellation_tools.cluster_princomp">
<code class="descclassname">Parcellation_tools.</code><code class="descname">cluster_princomp</code><span class="sig-paren">(</span><em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Parcellation_tools.cluster_princomp" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [clusters,subclusters] = cluster_princomp(clusters,[behavioral score vector],[corr flag],[plotflag],[locflag])
</pre></div>
</div>
<p>ALSO TRY: subcluster_montage(subclusters{1}) % to plot the output</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>clusters:</strong></dt>
<dd><p class="first last">is structure of clusters from tor_extract_rois.m</p>
</dd>
</dl>
<p>behavioral vector is row vector of behavioral or other scores to correlate</p>
<p>corr flag:  <a href="#id37"><span class="problematic" id="id38">*</span></a>1 = work on correlations among voxels, 2 = work on covariance</p>
<p>plotflag:   <a href="#id39"><span class="problematic" id="id40">*</span></a>1 = yes, 0 = no.  plots.</p>
<dl class="last docutils">
<dt>locflag:    1 yes, <a href="#id41"><span class="problematic" id="id42">*</span></a>0 no; add XYZ voxel locations (scaled) to data submitted to clustering</dt>
<dd><p class="first last">pushes voxels closer in space to be classified in the same cluster</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>try this to test the program on random data:</p>
<div class="highlight-python"><div class="highlight"><pre>cl(1).all_data = randn(23,30);cl(1).numVox = 30;cl = cluster_princomp(cl,EXPT.behavior,1,1);
cl(1).all_data(:,1:10) = cl(1).all_data(:,1:10) + 10; cl = cluster_princomp(cl,EXPT.behavior,1,1);
cl(1).all_data(:,25:30) = cl(1).all_data(:,25:30) + repmat((EXPT.behavior .* 3)&#39;,1,6);
cl(1).all_data(:,21:24) = cl(1).all_data(:,21:24) + repmat((1:23)&#39;,1,4);
cl = cluster_princomp(cl,EXPT.behavior,1,1);
</pre></div>
</div>
<p>mean-center everything now:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span><span class="o">.</span><span class="n">PCA</span> <span class="o">=</span> <span class="p">[];</span>
<span class="n">cl</span><span class="o">.</span><span class="n">all_data</span> <span class="o">-</span> <span class="n">cl</span><span class="o">.</span><span class="n">all_data</span> <span class="o">-</span> <span class="n">repmat</span><span class="p">(</span><span class="n">mean</span><span class="p">(</span><span class="n">cl</span><span class="o">.</span><span class="n">all_data</span><span class="p">),</span><span class="n">size</span><span class="p">(</span><span class="n">cl</span><span class="o">.</span><span class="n">all_data</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span><span class="mi">1</span><span class="p">);</span>
<span class="n">cl</span> <span class="o">=</span> <span class="n">cluster_princomp</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">EXPT</span><span class="o">.</span><span class="n">behavior</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">);</span>
</pre></div>
</div>
<p>add another correlated group:</p>
<div class="highlight-python"><div class="highlight"><pre>cl.all_data(:,1:5) = cl.all_data(:,1:5) + repmat(rand(23,1)*5,1,5);
cl = cluster_princomp(cl,EXPT.behavior,1,1);
</pre></div>
</div>
<p>if component scores are used and correlated with behavior, this means that the subjects
tend to show the behavioral effect who also show the pattern associated with comp. x.  
this may mean high on a number of voxels, or high on some and low on others.  
the weights may be used to interpret what the components mean, and this can be done
graphically.</p>
<p>t-tests on component scores have ambiguous interpretations, because a high t-score
may indicate negative values or close-to-zero values on some voxels.
a component could have the interpretation, &#8220;high on this component means high on V1
and low on V2.&#8221;</p>
<p>classifying voxels is done using cluster analysis (hierarchical, centroid linkage)
on the voxels (observations) using the PCA weights (eigenvectors) as variables.
This lets the clustering algorithm work in the reduced variable space with dimensionality
equal to the number of components.  
The max number of clusters is restricted based on the gradient of the eigenvalues in the PCA
maxclusters = 1 + the number of eigenvalues with gradient at least 20% of the initial drop 
from 1 to 2 eigenvalues.</p>
<p>Requires clustering library in Matlab.
Robust option also uses the robust PCA algorithm RAPCA,
created by:
Hubert, M., Rousseeuw, P.J., Verboven, S. (2002),</p>
<blockquote>
<div>&#8220;A fast method for robust principal components with applications to chemometrics&#8221;, by Mia Hubert, Peter J. Rousseeuw, 
Chemometrics and Intelligent Laboratory Systems, 60, 101-111.</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Parcellation_tools.inconsistent">
<code class="descclassname">Parcellation_tools.</code><code class="descname">inconsistent</code><span class="sig-paren">(</span><em>Z</em>, <em>depth</em><span class="sig-paren">)</span><a class="headerlink" href="#Parcellation_tools.inconsistent" title="Permalink to this definition">¶</a></dt>
<dd><p>Inconsistent values of a cluster tree.</p>
<p>Y = INCONSISTENT(Z) computes the inconsistent value of each non-leaf
node in the hierarchical cluster tree Z.  Z is a (M-1)-by-3 matrix
generated by the function LINKAGE.  Each inconsistent value is a
measure of separation between the two clusters whose merge is
represented by that node, compared to the separation between
subclusters merged within those clusters.</p>
<p>Y = INCONSISTENT(Z, DEPTH) computes inconsistent values by looking
to a depth DEPTH below each node.</p>
<p>Y is a (M-1)-by-4 matrix, with rows corresponding to each of the
non-leaf nodes represented in Z.  INCONSISTENT computes the
inconsistent value for node (M+i) using S_i, the set of nodes less than
DEPTH branches below node (M+i), excluding any leaf nodes.  Then</p>
<blockquote>
<div><ul class="simple">
<li>Y(i,1) = mean(Z(S_i,3)), the mean height of nodes in S_i</li>
<li>Y(i,2) = std(Z(S_i,3)), the standard deviation of node heights in S_i</li>
<li>Y(i,3) = length(S_i), the number of nodes in S_i</li>
<li>Y(i,4) = (Z(i,3) - Y(i,1))/Y(i,2), the inconsistent value</li>
</ul>
</div></blockquote>
<p>The default value for DEPTH is 2.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">PDIST, LINKAGE, COPHENET, DENDROGRAM, CLUSTER, CLUSTERDATA.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Parcellation_tools.mask_princomp">
<code class="descclassname">Parcellation_tools.</code><code class="descname">mask_princomp</code><span class="sig-paren">(</span><em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Parcellation_tools.mask_princomp" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[clusters] = mask_princomp(clusters,[behavioral score vector],[corr flag],[plotflag],[saveflag])
</pre></div>
</div>
<p>This function is just like cluster_princomp, except that it works on the SET
of activations in all clusters, rather than within each cluster.</p>
<p>clusters is structure of clusters from tor_extract_rois.m</p>
<p>behavioral vector is row vector of behavioral or other scores to correlate</p>
<p>corr flag:  1 = work on correlations among voxels, 2 = work on covariance</p>
<p>plotflag:   1 = yes, 0 = no.  plots.</p>
<p>try this to test the program on random data:</p>
<div class="highlight-python"><div class="highlight"><pre>cl(1).all_data = randn(23,30);cl(1).numVox = 30;cl = cluster_princomp(cl,EXPT.behavior,1,1);
cl(1).all_data(:,1:10) = cl(1).all_data(:,1:10) + 10; cl = cluster_princomp(cl,EXPT.behavior,1,1);
cl(1).all_data(:,25:30) = cl(1).all_data(:,25:30) + repmat((EXPT.behavior .* 3)&#39;,1,6);
cl(1).all_data(:,21:24) = cl(1).all_data(:,21:24) + repmat((1:23)&#39;,1,4);
cl = cluster_princomp(cl,EXPT.behavior,1,1);
</pre></div>
</div>
<p>mean-center everything now:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span><span class="o">.</span><span class="n">PCA</span> <span class="o">=</span> <span class="p">[];</span> <span class="n">cl</span><span class="o">.</span><span class="n">all_data</span> <span class="o">-</span> <span class="n">cl</span><span class="o">.</span><span class="n">all_data</span> <span class="o">-</span> <span class="n">repmat</span><span class="p">(</span><span class="n">mean</span><span class="p">(</span><span class="n">cl</span><span class="o">.</span><span class="n">all_data</span><span class="p">),</span><span class="n">size</span><span class="p">(</span><span class="n">cl</span><span class="o">.</span><span class="n">all_data</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span><span class="mi">1</span><span class="p">);</span>
<span class="n">cl</span> <span class="o">=</span> <span class="n">cluster_princomp</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">EXPT</span><span class="o">.</span><span class="n">behavior</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="mi">1</span><span class="p">);</span>
</pre></div>
</div>
<p>add another correlated group:</p>
<div class="highlight-python"><div class="highlight"><pre>cl.all_data(:,1:5) = cl.all_data(:,1:5) + repmat(rand(23,1)*5,1,5);
cl = cluster_princomp(cl,EXPT.behavior,1,1);
</pre></div>
</div>
<p>if component scores are used and correlated with behavior, this means that the subjects
tend to show the behavioral effect who also show the pattern associated with comp. x.  
this may mean high on a number of voxels, or high on some and low on others.  
the weights may be used to interpret what the components mean, and this can be done
graphically.</p>
<p>t-tests on component scores have ambiguous interpretations, because a high t-score
may indicate negative values or close-to-zero values on some voxels.
a component could have the interpretation, &#8220;high on this component means high on V1
and low on V2.&#8221;</p>
<p>classifying voxels is done using cluster analysis (hierarchical, centroid linkage)
on the voxels (observations) using the PCA weights (eigenvectors) as variables.
This lets the clustering algorithm work in the reduced variable space with dimensionality
equal to the number of components.  
The max number of clusters is restricted based on the gradient of the eigenvalues in the PCA
maxclusters = 1 + the number of eigenvalues with gradient at least 20% of the initial drop 
from 1 to 2 eigenvalues.</p>
<p>Requires clustering library in Matlab.
Robust option also uses the robust PCA algorithm RAPCA,
created by:
Hubert, M., Rousseeuw, P.J., Verboven, S. (2002),</p>
<blockquote>
<div>&#8220;A fast method for robust principal components with applications to chemometrics&#8221;, by Mia Hubert, Peter J. Rousseeuw, 
Chemometrics and Intelligent Laboratory Systems, 60, 101-111.</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Parcellation_tools.parcel_cl_nmds">
<code class="descclassname">Parcellation_tools.</code><code class="descname">parcel_cl_nmds</code><span class="sig-paren">(</span><em>parcel_cl_avgs</em><span class="sig-paren">)</span><a class="headerlink" href="#Parcellation_tools.parcel_cl_nmds" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">parcel_cl_avgs</span><span class="p">,</span> <span class="n">NMDS</span><span class="p">,</span> <span class="n">class_clusters</span><span class="p">]</span> <span class="o">=</span> <span class="n">parcel_cl_nmds</span><span class="p">(</span><span class="n">parcel_cl_avgs</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>load Parcellation_info/parcellation.mat
[parcel_cl_avgs, NMDS, class_clusters] = parcel_cl_nmds(parcel_cl_avgs)

parcel_cl_nmds_plots(parcel_cl_avgs, NMDS, &#39;save&#39;)
parcel_cl_nmds_plots(parcel_cl_avgs, NMDS, &#39;save&#39;, &#39;savedir&#39;, &#39;Parcellation_info&#39;)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Complete methods example:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<ol class="arabic simple">
<li>Dimension reduction</li>
</ol>
<p>Clustering of multivariate data is most stable when the data is not sparse, i.e., the dimensionality is low relative to the number of observations. To limit the dimensionality of the data, a spatio-temporal dimension-reduction step is first performed on the [n x v x N] data matrix of  AUC data for n trials x v voxels x N participants (here, n = 48 trials [usually], v = 17,112 voxels , and N = 26 participants).  A temporal data reduction is first performed to identify components with correlated AUC trial time series within each participant, followed by a spatial reduction to identify components with correlated spatial patterns across subjects.  First, the [n x v] matrix of AUC data for each participant was subjected to PCA, using the [v x v] correlation matrices.  Based on the scree plots across subjects, we saved the first 7 eigenvectors (spatial maps).  These eigenvectors explained 74 +- 2.5% (st. dev. across subjects) of the variance in the full dataset. These eigenvectors were scaled by their variances (eigenvalues) and concatenated across subjects to form an [v x N*7] matrix of eigenvectors.   This matrix was subjected to second (across participant) PCA step to identify components with similar spatial maps across participants.  We retained 12 eigenvectors (maps) based on the scree plot, which explained 65% of the variance across individuals. Component scores in this space were used for clustering.  This is a data reduction step, and the results are not expected to depend strongly on the number of eigenvectors retained at either step, as long as most of the variance in the data is explained.</p>
<ol class="arabic simple" start="2">
<li>Parcellation</li>
</ol>
<p>Hierarchical agglomerative clustering with average linkage was used to group voxels into parcels&#8211;sets of contiguous voxels with similar profiles&#8211;in the [v x 12] matrix of component maps.  The goal of parcellation was to reduce the space from voxels to parcels (regions) for non-metric multidimensional scaling (NMDS)-based clustering of regions, so that NMDS is computationally tractable. Voxels whose trial AUC time series did not correlate with that of other voxels in the same parcel at p &lt; .001 (in a random effects analysis across participants) were pruned from each parcel. Out of 140 parcels total, 127 parcels with more than 3 voxels after pruning were retained for subsequent analysis.</p>
<ol class="arabic simple" start="3">
<li>NMDS and clustering</li>
</ol>
<p>Parcels were now treated as the unit of analysis, and the AUC trial time series data were extracted from each parcel for each subject.  Values in the  [n trials x parcels x N subjects]  data matrix were z-scored within participant to remove inter-subject differences in scaling, then concatenated into an [n*N x parcels] data matrix.  Correlations among parcels were converted into a [parcels x parcels] matrix of distances using the formula distance = (1 - r) / 2. The NMDS stress plot (which operates on ranked distances and is therefore more robust to outliers and does not require a strictly Euclidean distance space) was examined and 7 dimensions (which explained 79% of the variance in distances) were retained for the final cluster analysis.  
Hierarchical agglomerative clustering with average linkage was used to cluster the parcels in this space into interconnected networks with correlated AUC trial time series. To choose the number of clusters in the final solution (k), for every possible choice of clusters between  k = 2 and 20, we compared the cluster solution to the average and standard deviation of 1,000 clustering iterations with permuted parcel time series.  This yielded a Z-score ([actual solution - mean permuted solution] /  standard deviation of permuted solution for each value of k. The best solution was k = 13, with a value of Z = 5.57 compared to the null-hypothesis single-cluster solution (p &lt; .0001).</p>
</dd></dl>

<dl class="function">
<dt id="Parcellation_tools.parcel_cl_nmds_plots">
<code class="descclassname">Parcellation_tools.</code><code class="descname">parcel_cl_nmds_plots</code><span class="sig-paren">(</span><em>parcel_cl_avgs</em>, <em>NMDS</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Parcellation_tools.parcel_cl_nmds_plots" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot: data panel</p>
<p>case &#8216;save&#8217;, dosave = 1;
case {&#8216;savedir&#8217;, &#8216;mysavedir&#8217;}, mysavedir = varargin{i+1};
case {&#8216;figs&#8217;, &#8216;dotfigs&#8217;}, savedotfigs = 1;</p>
<p>Doc not complete yet.  Please update me!</p>
<p>See parcel_clusters.m
See parcel_cl_nmds.m</p>
<p>load Parcellation_info/parcellation.mat</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">parcel_cl_nmds_plots</span><span class="p">(</span><span class="n">parcel_cl_avgs</span><span class="p">,</span> <span class="n">NMDS</span><span class="p">,</span> <span class="s">&#39;save&#39;</span><span class="p">)</span>
<span class="n">parcel_cl_nmds_plots</span><span class="p">(</span><span class="n">parcel_cl_avgs</span><span class="p">,</span> <span class="n">NMDS</span><span class="p">,</span> <span class="s">&#39;save&#39;</span><span class="p">,</span> <span class="s">&#39;savedir&#39;</span><span class="p">,</span> <span class="s">&#39;Parcellation_info&#39;</span><span class="p">)</span>
<span class="n">parcel_cl_nmds_plots</span><span class="p">(</span><span class="n">parcel_cl_avgs</span><span class="p">,</span> <span class="n">NMDS</span><span class="p">,</span> <span class="s">&#39;save&#39;</span><span class="p">,</span> <span class="s">&#39;savedir&#39;</span><span class="p">,</span> <span class="s">&#39;Parcellation_info_tor_mask_try1&#39;</span><span class="p">,</span> <span class="s">&#39;savedotfig&#39;</span><span class="p">)</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Parcellation_tools.parcel_clusters">
<code class="descclassname">Parcellation_tools.</code><code class="descname">parcel_clusters</code><span class="sig-paren">(</span><em>clpos_data</em>, <em>clneg_data</em><span class="sig-paren">)</span><a class="headerlink" href="#Parcellation_tools.parcel_clusters" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">parcel_clusters</span><span class="p">(</span><span class="n">clpos_data</span><span class="p">,</span> <span class="n">clneg_data</span><span class="p">)</span>
</pre></div>
</div>
<p>No outputs. Saves all output in separate directory.</p>
<p>First, get eigenvectors for each subject.</p>
<p>We&#8217;re interested obtaining PARCELS of voxels that tend to co-activate, 
or have the same activation profile.</p>
<p>We can find these by using clustering algorithms to group voxels with
similar profiles.</p>
<p>Because we have a many voxel x many voxel covariance matrix for each
subject (lots of data!), it&#8217;s important to reduce the dimensionality of
the problem and peform clustering on a REDUCED_DIMENSIONAL space.</p>
<p>We use PCA to do this.  Instead of clustering activation profiles (e.g.,
time-courses) directly, we cluster eigenvector loadings for each voxel on
a reduced set of components that explains most of the variance in the
data.</p>
<p>Similar voxels will have similar loadings across the set of 
eigenvectors.  e.g., two voxels may load high on components [1 3 and 5],
and low on components [10 and 13].  If they have the same pattern of
loadings, they should be considered part of the same CLASS.  Groups
of voxels that are contiguous in space and are members of the same CLASS 
are called parcels.</p>
<p>Images and outputs are saved in their own subdirectory called
Parcellation_info</p>
<p>The main outputs are: 
parcel_cl             % parcels, one cell per subject, one parcel per</p>
<blockquote>
<div>element within cells. same format as clpos_data</div></blockquote>
<dl class="docutils">
<dt>parcel_cl_avgs        % parcels, one parcel per element within cells. </dt>
<dd>same format as clpos_data2</dd>
</dl>
<p>parcel_cl_avgs(x).timeseries contains one cell per subject, with data
averaged across voxels within that parcel for that subject
This kind of output is useful, because you can input it directly into
other mediation analyses.
[paths, stats2] = mediation(SETUP.data.X, SETUP.data.Y, parcel_cl_avgs(1).timeseries, &#8216;plots&#8217;, &#8216;verbose&#8217;, &#8216;names&#8217;, {&#8216;Hi-Low Cue&#8217; &#8216;Pain Report&#8217; &#8216;Parcel&#8217;}, &#8216;boot&#8217;);
cluster_orthviews(parcel_cl_avgs(1), {[0 1 0]}, &#8216;add&#8217;);</p>
<div class="highlight-python"><div class="highlight"><pre>cd(&#39;/Volumes/SCNAlpha/Data_and_Tools/SpeechTask/analysis/wb_multisubject_correl_HR_corrected/mediation_Xprepvsb_Mbrain_Yhr&#39;)
load cl_b_fdr05_002_01_k3_1_1_prune
</pre></div>
</div>
<p>then run</p>
<dl class="docutils">
<dt>There are 2 dimension-reduction steps:</dt>
<dd><ol class="first last arabic simple">
<li>within-subjects</li>
<li>is on eigenvectors concatenated across subjects</li>
</ol>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Parcellation_tools.parcel_complete_sets">
<code class="descclassname">Parcellation_tools.</code><code class="descname">parcel_complete_sets</code><span class="sig-paren">(</span><em>s</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Parcellation_tools.parcel_complete_sets" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">mysets</span><span class="p">,</span><span class="n">n_in_set</span><span class="p">,</span><span class="n">sets_by_vars</span><span class="p">,</span><span class="n">classes</span><span class="p">]</span> <span class="o">=</span> <span class="n">parcel_complete_sets</span><span class="p">(</span><span class="n">s</span><span class="p">,[</span><span class="s">&#39;dounique&#39;</span><span class="p">,</span><span class="s">&#39;nofuzzy&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>Find sets of elements in logical n x n matrix s in which all pairs in set are &#8216;true&#8217; (1 in
matrix s)</p>
<p>Optional: input a distance/correlation/etc. matrix and threshold</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;dounique&#8217;:</strong></dt>
<dd><p class="first last">provides single-variable sets as well</p>
</dd>
<dt><strong>&#8216;nofuzzy&#8217;:</strong></dt>
<dd><p class="first last">chooses closest set for each var so that each var</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>can belong to only one set</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;threshold&#8217;:</strong></dt>
<dd><p class="first">followed by threshold thr for matrix s.</p>
<p>if thr is a string, thr will be evaluated on s: e.g., &#8216;s &lt; 10&#8217;</p>
<p>in this case, thr should be a logical expr. involving s</p>
<p>if thr is a number, s &gt;= thr will be evaluated</p>
<p class="last">the resulting logical matrix will be used to determine sets</p>
</dd>
<dt><strong>&#8216;min&#8217; or &#8216;max&#8217;:</strong></dt>
<dd><p class="first">works only with &#8216;nofuzzy&#8217; option.</p>
<p>if max, uses max to find most similar set; good if s is a
covariance matrix</p>
<p>if min, uses min to find closest set; good if s is a
distance matrix</p>
<p class="last">default is max</p>
</dd>
</dl>
</div></blockquote>
<p>runs on Matlab 7.2 or higher %%%</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Find sets of coordinates within 10 mm of one another
xyz = cat(1,cl{1}.mm_center);
d = pdist(xyz); d = squareform(d);
[mysets,n_in_set,sets_by_vars,classes] = parcel_complete_sets(d,&#39;dounique&#39;,&#39;nofuzzy&#39;,&#39;threshold&#39;,&#39;s&lt;10&#39;,&#39;min&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Parcellation_tools.parcel_images">
<code class="descclassname">Parcellation_tools.</code><code class="descname">parcel_images</code><span class="sig-paren">(</span><em>image_names</em>, <em>extract_mask</em>, <em>nuisance_covs</em><span class="sig-paren">)</span><a class="headerlink" href="#Parcellation_tools.parcel_images" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">parcel_images</span><span class="p">(</span><span class="n">image_names</span><span class="p">,</span> <span class="n">extract_mask</span><span class="p">,</span> <span class="n">nuisance_covs</span><span class="p">)</span>
</pre></div>
</div>
<p>This function performs the following steps, in this order</p>
<blockquote>
<div><ul class="simple">
<li>map mask to functional space</li>
<li>extract data from in-mask voxels</li>
<li>remove nuisance covariates (before assessing connectivity)</li>
<li>data reduction (pca)</li>
<li>plot cases (detect outliers)</li>
<li>separate data into a priori anatomical regions (LBPA40 hard-coded
right now; downloadable from web; see wiki)
(save label image mapped to functional space)</li>
<li>cluster voxels in each region to get parcels</li>
<li>save parcels and images</li>
<li>NMDS on the parcels to group them into &#8220;networks&#8221; (default = use rank data)</li>
<li>Visualization of the networks</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Creates and goes to a new directory: parcel_images_output</p>
<dl class="docutils">
<dt>Outputs saved to disk include</dt>
<dd><ol class="first last arabic simple">
<li>An image with unique numerical codes for each parcel</li>
<li>A &#8216;clusters&#8217; structure containing the parcels, with image data extracted and
averaged over voxels within each parcel</li>
</ol>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>image_names:</strong></dt>
<dd><p class="first last">names of images to extract data from, and to use for
functional parcellation. SEE ALTERNATE FORMAT BELOW FOR DIRECT DATA INPUT</p>
</dd>
<dt><strong>extract_mask:</strong></dt>
<dd><p class="first last">a mask image</p>
</dd>
<dt><strong>nuisance_covs:</strong></dt>
<dd><p class="first">columns of a matrix to remove from the data, so that this
subspace is not used to determine connectivity</p>
<p class="last">e.g., nuisance_covs = SPM.xX.X(:, 1:3); if these are nuisance
covariates...</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Alternate input formats for image_names:</p>
<p>If you have data already extracted, image_names can be a structure with
these fields:</p>
<p>image_names.V, spm_vol-style volume info for the mask volume and image space
image_names.data, extracted data from all valid in-mask (non-zero,
non-nan) voxels, one column per voxel, in standard matlab (:) order.
The mask and the data must match!</p>
</dd></dl>

</div>
<div class="section" id="peak-coordinates">
<h2>peak_coordinates<a class="headerlink" href="#peak-coordinates" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-peak_coordinates"></span><dl class="function">
<dt id="peak_coordinates.cluster_manova">
<code class="descclassname">peak_coordinates.</code><code class="descname">cluster_manova</code><span class="sig-paren">(</span><em>clusters</em>, <em>fnames</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#peak_coordinates.cluster_manova" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cluster_manova</span><span class="p">(</span><span class="n">clusters</span><span class="p">,</span><span class="n">fnames</span><span class="p">,</span><span class="n">verbose</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>clusters:</strong></dt>
<dd><p class="first last">is output of clusters2database, with all fields
from database</p>
</dd>
<dt><strong>fnames:</strong></dt>
<dd><p class="first last">is cell array of strings with names to test
e.g., {&#8216;Rule&#8217; &#8216;Task&#8217;}</p>
</dd>
</dl>
<p>uses stats toolbox</p>
<dl class="last docutils">
<dt><strong>verbose:</strong></dt>
<dd><p class="first last">optional, produces more output and tests</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="peak_coordinates.extract_ind_peak">
<code class="descclassname">peak_coordinates.</code><code class="descname">extract_ind_peak</code><span class="sig-paren">(</span><em>imnames</em>, <em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#peak_coordinates.extract_ind_peak" title="Permalink to this definition">¶</a></dt>
<dd><p>This function gets individual spatial peaks from clusters.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[clusters] = extract_ind_peak(imnames [can be empty],clusters,[vols])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>imnames::</strong></dt>
<dd><p class="first">a matrix of image names, in spm_list_files output format</p>
<p class="last">if imnames is empty, enter full data (vols) as 3rd argument
or else this program will use cl.all_data to get data</p>
</dd>
<dt><strong>cl:</strong></dt>
<dd><p class="first last">clusters, see tor_extract_rois</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>NOTE (WARNING): WORKS ON XYZ VOXEL COORDINATES - NO TRANSFORMATION TO DIFFERENT SPACES
see transform_coordinates.m for this.</p>
<dl class="docutils">
<dt>Functions called</dt>
<dd><p class="first">C:matlabR12toolboxmatlabdatatypessqueeze.m</p>
<p class="last">c:tor_scriptsvoistatutilitynanmean.m
(calls other spm functions)</p>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">cluster_manova</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="peak_coordinates.image2coordinates">
<code class="descclassname">peak_coordinates.</code><code class="descname">image2coordinates</code><span class="sig-paren">(</span><em>img</em><span class="sig-paren">)</span><a class="headerlink" href="#peak_coordinates.image2coordinates" title="Permalink to this definition">¶</a></dt>
<dd><p>Threshold a statistic image and turn it into a list of x, y, z
coordinates (in mm &#8220;world space&#8221;) with SPM</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">XYZmm</span> <span class="o">=</span> <span class="n">image2coordinates</span><span class="p">(</span><span class="n">img</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>img = &#39;h25_aerger.img&#39;;
XYZmm = image2coordinates(img)
figure;
plot3(XYZmm{1}(1, :)&#39;, XYZmm{1}(2, :)&#39;, XYZmm{1}(3,:)&#39;, &#39;ko&#39;);
addbrain
axis image
</pre></div>
</div>
<p>Example with multiple images:</p>
<div class="highlight-python"><div class="highlight"><pre>% list all images in dir
imgs = filenames(fullfile(pwd, &#39;*img&#39;), &#39;char&#39;, &#39;absolute&#39;)
XYZmm = image2coordinates(imgs); 
% save images names and coordinates
% imgs2 has cell array of image names, without full paths
imgs2 = filenames(fullfile(pwd, &#39;*img&#39;)); 
save silke_senders_xyz_coordinates imgs imgs2 XYZmm
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="peak_coordinates.spatial_contrast">
<code class="descclassname">peak_coordinates.</code><code class="descname">spatial_contrast</code><span class="sig-paren">(</span><em>XYZ1</em>, <em>XYZ2</em><span class="sig-paren">)</span><a class="headerlink" href="#peak_coordinates.spatial_contrast" title="Permalink to this definition">¶</a></dt>
<dd><p>This function tests relative locations of individual 
spatial peaks from clusters.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="nb">str</span><span class="p">,</span><span class="n">dcon</span><span class="p">]</span> <span class="o">=</span> <span class="n">spatial_contrast</span><span class="p">(</span><span class="n">XYZ1</span><span class="p">,</span><span class="n">XYZ2</span><span class="p">)</span>
</pre></div>
</div>
<p>plots position of XYZ2 relative to XYZ1
thus, an &#8216;anterior&#8217; group position means that XYZ2
peaks are anterior to XYZ1 peaks</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>XYZ1,2:</strong></dt>
<dd><p class="first last">n x 3 coordinates (in mm)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>con:</strong></dt>
<dd><p class="first last">contrast vector, e.g., [1 -1]</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>See help conf_region for details of the test.</p>
</dd></dl>

</div>
<div class="section" id="roi-drawing-tools">
<h2>ROI_drawing_tools<a class="headerlink" href="#roi-drawing-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-ROI_drawing_tools"></span><dl class="function">
<dt id="ROI_drawing_tools.add2mask">
<code class="descclassname">ROI_drawing_tools.</code><code class="descname">add2mask</code><span class="sig-paren">(</span><em>mask</em>, <em>x</em>, <em>r</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#ROI_drawing_tools.add2mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Adds or subtracts spheres around x coordinates to/from existing mask</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">add2mask</span><span class="p">(</span><span class="n">mask</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">r</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>mask:</strong></dt>
<dd><p class="first last">is a string filename</p>
</dd>
<dt><strong>x:</strong></dt>
<dd><p class="first last">is n x 3 list of coordinates</p>
</dd>
<dt><strong>r:</strong></dt>
<dd><p class="first last">is radius</p>
</dd>
</dl>
<p>a 4th argument causes us to SUBTRACT!</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>mask = &#39;insula_from_part1.img&#39;
x = [-29.6 28.6 5.3; 37.0 27.5 3.2; -34.9 8.5 -14.8;38.1 10.6 -14.8];
add2mask(mask,x,8);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="ROI_drawing_tools.clusters2roimask">
<code class="descclassname">ROI_drawing_tools.</code><code class="descname">clusters2roimask</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#ROI_drawing_tools.clusters2roimask" title="Permalink to this definition">¶</a></dt>
<dd><p>The purpose of this function is to facilitate making masks with ROIs for
future studies, give a clusters structure.  ROIs are constrained to be
within activation blobs specified by input clusters, and are masked by
selected ICBM regions. Clusters may be smoothed before or after masking.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">clout</span><span class="p">]</span> <span class="o">=</span> <span class="n">clusters2roimask</span><span class="p">(</span><span class="n">cl</span><span class="p">)</span>
</pre></div>
</div>
<dl class="docutils">
<dt>Option to do 3 things, in order:</dt>
<dd><ol class="first last arabic simple">
<li>Enlarge selected clusters</li>
<li>Mask clusters with anatomical regions from ICBM atlas</li>
<li>Subdivide clusters using hierarchical clustering of voxel coordinates</li>
</ol>
</dd>
</dl>
<p>Output is a clusters file and a mask file in a 2 x 2 x 2 standard brain
space.</p>
</dd></dl>

<dl class="function">
<dt id="ROI_drawing_tools.draw_anatomical_roi_2008">
<code class="descclassname">ROI_drawing_tools.</code><code class="descname">draw_anatomical_roi_2008</code><span class="sig-paren">(</span><em>meth</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#ROI_drawing_tools.draw_anatomical_roi_2008" title="Permalink to this definition">¶</a></dt>
<dd><p>Quick start guide:
type draw_anatomical_roi_2008</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">draw_anatomical_roi_2008</span><span class="p">(</span><span class="s">&#39;init&#39;</span><span class="p">);</span>
<span class="n">draw_anatomical_roi_2008</span><span class="p">(</span><span class="s">&#39;load&#39;</span><span class="p">,</span> <span class="s">&#39;ROI_midbrain.img&#39;</span><span class="p">);</span>
<span class="n">draw_anatomical_roi_2008</span><span class="p">(</span><span class="s">&#39;init&#39;</span><span class="p">,</span> <span class="s">&#39;overlay&#39;</span><span class="p">,</span>  <span class="s">&#39;remi_mean_T2.img&#39;</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first">no arguments: init</p>
<dl class="docutils">
<dt><strong>&#8216;init&#8217;:</strong></dt>
<dd><p class="first last">initialize gui and orthviews and remove previous ROI</p>
</dd>
<dt><strong>&#8216;load&#8217;:</strong></dt>
<dd><p class="first last">load an ROI from a mask</p>
</dd>
<dt><strong>&#8216;free&#8217;:</strong></dt>
<dd><p class="first last">draw an ROI freehand (click on one of the slices in the Slices window 1st)</p>
</dd>
<dt><strong>&#8216;poly&#8217;:</strong></dt>
<dd><p class="first last">don&#8217;t run this yet</p>
</dd>
<dt><strong>&#8216;add&#8217;:</strong></dt>
<dd><p class="first last">add a region you&#8217;ve drawn on a slice to your ROI</p>
</dd>
<dt><strong>&#8216;remove&#8217;:</strong></dt>
<dd><p class="first last">remove a region you&#8217;ve drawn from your ROI</p>
</dd>
<dt><strong>&#8216;smooth&#8217;:</strong></dt>
<dd><p class="first last">3-d smoothing of ROI</p>
</dd>
<dt><strong>&#8216;write&#8217;:</strong></dt>
<dd><p class="first last">write mask image of ROI and return clusters to workspace</p>
</dd>
<dt><strong>&#8216;exit&#8217;:</strong></dt>
<dd><p class="first last">exit. ROI data is stored in the Slices figure, so you can
continue to edit, etc. after exiting.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Note:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>You can use cluster_orthviews to image multiple blobs, and then
draw relative to those.
this function saves it&#8217;s data in the Slices figure, so you can draw,
re-initialize the orthviews, and keep drawing before you save.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cluster_orthviews(red, {[1 0 0]}, &#39;overlay&#39;, &#39;remi_mean_T2.img&#39;);
cluster_orthviews(stn, {[0 1 0]}, &#39;add&#39;);
set(findobj(&#39;Tag&#39;,&#39;Graphics&#39;), &#39;WindowButtonUpFcn&#39;, &#39;draw_anatomical_roi_2008(&#39;&#39;moveslice&#39;&#39;);&#39;);
% Use the spm_orthviews menu to ZOOM IN...and keep drawing!
</pre></div>
</div>
<p>Example of brainstem ROI drawing:</p>
<div class="highlight-python"><div class="highlight"><pre>draw_anatomical_roi_2008(&#39;init&#39;, &#39;overlay&#39;,  &#39;remi_mean_T2.img&#39;);
set(findobj(&#39;Tag&#39;,&#39;Graphics&#39;), &#39;WindowButtonUpFcn&#39;, &#39;&#39;);
cluster_orthviews(red, {[1 0 0]}, &#39;overlay&#39;, &#39;remi_mean_T2.img&#39;);
stn = mask2clusters(&#39;ROI_STN.img&#39;);
cluster_orthviews(stn, {[0 1 1]}, &#39;add&#39;);
% Now zoom in to the midbrain in SPM orthviews and draw new ROIs
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="ROI_drawing_tools.sphere_mask">
<code class="descclassname">ROI_drawing_tools.</code><code class="descname">sphere_mask</code><span class="sig-paren">(</span><em>P</em>, <em>XYZmm</em>, <em>r</em>, <em>outname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#ROI_drawing_tools.sphere_mask" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates mask images and clusters for a set of spheres defined around
coordinates you specify.  Used for creating regions of interest (ROIs).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">clusters</span><span class="p">,</span><span class="n">maskCLU</span><span class="p">]</span> <span class="o">=</span> <span class="n">sphere_mask</span><span class="p">(</span><span class="n">fname</span><span class="p">,</span><span class="n">XYZmm</span><span class="p">,</span><span class="n">r</span><span class="p">,</span><span class="n">outname</span><span class="p">,[</span><span class="n">maskname</span><span class="p">],[</span><span class="n">overlay</span><span class="p">])</span>
</pre></div>
</div>
<p>Spheres may be masked with an anatomical mask file.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>P:</strong></dt>
<dd><p class="first last">is input image name with correct dimensions and vox sizes for your
study</p>
</dd>
<dt><strong>XYZmm:</strong></dt>
<dd><p class="first last">is mm coordinates (row vector) for sphere center</p>
</dd>
<dt><strong>r:</strong></dt>
<dd><p class="first last">is radius in mm</p>
</dd>
<dt><strong>outname:</strong></dt>
<dd><p class="first last">is string for output mask name, e.g., &#8216;sphere_mask.img&#8217;</p>
</dd>
<dt><strong>[maskname]:</strong></dt>
<dd><p class="first last">is optional mask .img containing additional constraints
(e.g., gray matter mask, etc.), can be in different dimensions</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">See Also:</th><td class="field-body"><p class="first">mask2clusters, montage_clusters</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>tmp = sphere_mask(EXPT.SNPM.P{1}(1,:),[25.6 -45.5 77.0],10,&#39;test.img&#39;,&#39;ICBM_area74.img&#39;);
icbm_localize(tmp)

M =which(&#39;ICBM_brainonly_1mm_seg1.img&#39;)
cl = sphere_mask(d(1).name,[35 -57 54; -23 -59 56;13 -63 62; -7 -77 50; 25 -79 30; -23 -81 18;41 -7 46;-37 -7 46; 35 -1 30; 29 -3 60; -29 -13 46; -5 -1 56;-37 29 30; 53 21 32;51 11 -4;43 -69 14;-45 -71 12;29 -83 4;11 -89 -4;-11 -79 2; -1 -95 -14; 17 -99 -8;31 -77 -20;-23 -81 -20;41 -71 -20;-43 -79 -12],8,&#39;tmp.img&#39;,M);
</pre></div>
</div>
<p>Matlab 6.5/OSX bug gives seg fault or something if mask is too big.</p>
</dd></dl>

</div>
<div class="section" id="statistics-tools">
<h2>Statistics_tools<a class="headerlink" href="#statistics-tools" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Statistics_tools"></span><dl class="function">
<dt id="Statistics_tools.Bspline">
<code class="descclassname">Statistics_tools.</code><code class="descname">Bspline</code><span class="sig-paren">(</span><em>t</em>, <em>k</em>, <em>u</em>, <em>v</em>, <em>ForceSup</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.Bspline" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>Bspline(t,k,u[,v])
</pre></div>
</div>
<p>Create B-Spline basis of order k, with knots u, evaluated at t.
If control verticies v are specified then then B is the spline
function instead of the basis.</p>
<p>u must be at least length(k)+1</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.ContinuousAccuracy">
<code class="descclassname">Statistics_tools.</code><code class="descname">ContinuousAccuracy</code><span class="sig-paren">(</span><em>obj</em>, <em>pattern</em>, <em>predrange</em>, <em>unit</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.ContinuousAccuracy" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate forced choice accuracy for unit increases in continuous
predictions.  Requires units to be ranked ordered from 1:end.  Will work
for Gianaros or Pain datasets.  Accuracies are not penalized for missing
cases.  Probably best to run this on single subjects and then aggregate
accuracies across subjects.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">acc</span> <span class="o">=</span> <span class="n">ContinuousAccuracy</span><span class="p">(</span><span class="n">obj</span><span class="p">,</span> <span class="n">pattern</span><span class="p">,</span> <span class="n">unit</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>obj:</strong></dt>
<dd><p class="first last">fmri_data() object with data stacked by
increasing levels of prediction.  Make sure 
obj.Y includes the training labels</p>
</dd>
<dt><strong>pattern:</strong></dt>
<dd><p class="first last">fmri_data() object with weight pattern</p>
</dd>
<dt><strong>predrange:</strong></dt>
<dd><p class="first last">specify the range of predictions (e.g., 1:5)</p>
</dd>
<dt><strong>unit:</strong></dt>
<dd><p class="first last">specify the unit increase in prediction (e.g., 1 or 2)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>acc:</strong></dt>
<dd><p class="first last">accuracy of prediction for specified units</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>acc = ContinuousAccuracy(dat, pine, 1:5, 1)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.F_test_full_vs_red">
<code class="descclassname">Statistics_tools.</code><code class="descname">F_test_full_vs_red</code><span class="sig-paren">(</span><em>y</em>, <em>X</em>, <em>Xred</em>, <em>px</em>, <em>pxred</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.F_test_full_vs_red" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">F</span><span class="p">,</span> <span class="n">p</span><span class="p">,</span> <span class="n">resid</span><span class="p">,</span> <span class="n">df_model</span><span class="p">,</span> <span class="n">df_error</span><span class="p">]</span> <span class="o">=</span> <span class="n">F_test_full_vs_red</span><span class="p">(</span><span class="n">y</span><span class="p">,</span> <span class="n">X</span><span class="p">,</span> <span class="n">Xred</span><span class="p">,</span> <span class="n">px</span><span class="p">,</span> <span class="n">pxred</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>X = randn(100, 3); Xred = X(:,1); y = X(:,2) + randn(100, 1);
px = pinv(X); pxred = pinv(Xred);
[F, p, resid] = F_test_full_vs_red(y, X, Xred, px, pxred);


% Test full-model F-value against regress.m
Xred = X(:,end); % intercept only
px = pinv(X);
pxred = pinv(Xred);
[F, p, resid, dfm, dfe] = F_test_full_vs_red(y, X, Xred, px, pxred); % full model F-test
[b, bint, r, rint, stats] = regress(y, X);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.F_test_no_intercept">
<code class="descclassname">Statistics_tools.</code><code class="descname">F_test_no_intercept</code><span class="sig-paren">(</span><em>X</em>, <em>y</em>, <em>s</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.F_test_no_intercept" title="Permalink to this definition">¶</a></dt>
<dd><p>Test of the null hypothesis that the ENTIRE regression model X explains no
variance in the data (y)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">Fobs</span><span class="p">,</span> <span class="n">p</span><span class="p">,</span> <span class="n">dfb</span><span class="p">,</span> <span class="n">dfe</span><span class="p">]</span> <span class="o">=</span> <span class="n">F_test_no_intercept</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="n">s</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Assess false positive rate with robust regression:</p>
<div class="highlight-python"><div class="highlight"><pre>iter = 5000; 
warning off
fvals = zeros(1,iter); pvals = fvals;
for i = 1:iter
  X = randn(10,1); y = randn(10,1);
  [bb,stats]=robustfit(X,y);
  % [bb,dev,stats] =glmfit(X,y);
  [Fobs, p, dfb, dfe] = F_test_no_intercept(X,y,stats.s);
  fvals(i) = Fobs; pvals(i) = p;
end
fpr = sum(pvals&lt;.05) ./ iter
warning on
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Based on:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Johnson &amp; Wichern, 5th ed., p. 371, Result 7.6</p>
<p>for a regression model with k predictors, and q regression parameters 
in reduced model, test the addition of q+1...k by extra sums of squares
[n(s2red - s2full) / (k-q)]  / [n*s2full / (n-k-1)] ~ F(k-q),(n-k-1)</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.ICC">
<code class="descclassname">Statistics_tools.</code><code class="descname">ICC</code><span class="sig-paren">(</span><em>cse</em>, <em>typ</em>, <em>dat</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.ICC" title="Permalink to this definition">¶</a></dt>
<dd><dl class="docutils">
<dt>Function to work out ICCs according to shrout &amp; fleiss&#8217; schema (Shrout PE,</dt>
<dd><p class="first">Fleiss JL. Intraclass correlations: uses in assessing rater reliability.
Psychol Bull. 1979;86:420-428).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>iccvalue = ICC([1 to 6],[&#39;single&#39; or &#39;k&#39;], data matrix)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;dat&#8217;:</strong></dt>
<dd><p class="first last">is data whose <em>columns</em> represent k different raters (judges) &amp; whose
<em>rows</em> represent n different cases or targets being measured. Each target
is assumed to be a random sample from a population of targets.</p>
</dd>
<dt><strong>&#8216;cse&#8217;:</strong></dt>
<dd><p class="first last">is either 1,2,3. &#8216;cse&#8217; is: 1 if each target is measured by a
different set of raters from a population of raters, 2 if each target is
measured by the same raters, but that these raters are sampled from a
population of raters, 3 if each target is measured by the same raters and
these raters are the only raters of interest.</p>
</dd>
<dt><strong>&#8216;typ&#8217;:</strong></dt>
<dd><p class="first last">is either &#8216;single&#8217; or &#8216;k&#8217; &amp; denotes whether the ICC is based on a
single measurement or on an average of k measurements, where k = the
number of ratings/raters.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This has been tested using the example data in the paper by shrout &amp; fleiss.</p>
<p>Example: out = ICC(3,&#8217;k&#8217;,S_Fdata)
returns ICC(3,k) of data &#8216;S_Fdata&#8217; to double &#8216;out&#8217;.</p>
<p>Kevin Brownhill, Imaging Sciences, KCL, London <a class="reference external" href="mailto:kevin&#46;brownhill&#37;&#52;&#48;kcl&#46;ac&#46;uk">kevin<span>&#46;</span>brownhill<span>&#64;</span>kcl<span>&#46;</span>ac<span>&#46;</span>uk</a></p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Additional documentation:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>iccvalue = ICC([1 to 6],[&#8216;single&#8217; or &#8216;k&#8217;], data matrix)</p>
<p>Here, columns are &#8216;judges&#8217;, or more generally, &#8216;measures&#8217; that are
usually ideally intercorrelated.  Rows are items being assessed.
The ICC assesses the proportion of variance attributed to the items,</p>
<blockquote>
<div>shared across measures.</div></blockquote>
<p>As the correlation between the measures grows, the icc grows.
Another way of saying this is that if the rows are consistently different
across measures, the icc will be high.</p>
<p>Think of rows as criminals, and columns as judges. The data values are &#8216;guilt scores&#8217;,
where higher is more guilty.  If all the judges agree, the most guilty
cases will be rated as most guilty by all judges, and the icc will be
high. This is actually consistent with Case 2 or 3 in Shrout and Fleiss.</p>
<p>In Case 1, the columns don&#8217;t have any real meaning, as there are
different &#8216;judges&#8217; for each row, and variance components due to judge
cannot be separated from error and the judge x target interaction.
In Case 2 and 3, they are crossed. Case 2 treats judge as a random
effect, whereas Case 3 treats judge as a fixed effect.</p>
<p>If the data were an individual differences study of cognitive performance, 
then the rows would be subjects, and the columns would be tests.
A high icc would indicate a high correlation across the tests, which
indicates that subjects are reliably different from one another, i.e.,
that a large proportion of the total variance is related to subject.
In such a case, as tests are fixed entities, then Case 3 might be
appropriate.</p>
<p>Cronbach&#8217;s alpha is equal to ICC(3, k) - case 3, k
This assumes no target x rater interaction</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>dat = mvnrnd([1 1 1], [1 .5 .5; .5 1 .5; .5 .5 1], 50); whos dat
corrcoef(dat)
ri = ICC(2, &#39;k&#39;, dat)
dat = mvnrnd([1 1 1], [1 .9 .9; .9 1 .9; .9 .9 1], 50); whos dat
corrcoef(dat)
ri = ICC(2, &#39;k&#39;, dat)
</pre></div>
</div>
<p>In the example below, judges (measures) have systematically different
means, and the ICC values are different.  ICC(1, 1) is low because judge
is not considered as a source of variance.  ICC(2, 1) is higher, but
intermediate, because judge is considered as a random effect and modeled,
but we want to generalize to new judges.  ICC(3, 1) is highest, because
judge is modeled</p>
<blockquote class="last">
<div>dat = mvnrnd([1 2 3], [1 .5 .5; .5 1 .5; .5 .5 1], 50); whos dat
ri = ICC(1, &#8216;single&#8217;, dat)
ri = ICC(2, &#8216;single&#8217;, dat)
ri = ICC(3, &#8216;single&#8217;, dat)</div></blockquote>
</dd>
</dl>
<p>number of raters/ratings</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.ancova">
<code class="descclassname">Statistics_tools.</code><code class="descname">ancova</code><span class="sig-paren">(</span><em>groups</em>, <em>x</em>, <em>y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.ancova" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>b,t,p,pthandles] = ancova(groups,x,y,[plot],[covs of no interest])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Elements of b, t, p:</strong></dt>
<dd><p class="first last">1st = intercept, 2nd = group effect, 3rd = slope, 4th = grp x slope
interaction</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>recursive &#8211; call ancova repeately if y is a matrix
get pairwise standardized slopes (corrs)
and group diffs and slope interaction for all 
pairs of y vectors</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.barplot_get_within_ste">
<code class="descclassname">Statistics_tools.</code><code class="descname">barplot_get_within_ste</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.barplot_get_within_ste" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">se_within</span><span class="p">,</span> <span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">barplot_get_within_ste</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
</pre></div>
</div>
<p>Compute within-subjects standard errors</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body">The old version of this used average s.e.&#8217;s for contrasts</td>
</tr>
</tbody>
</table>
<p>of interest, which depend on contrast scaling and are less appropriate
for use as error bars.</p>
<p>The new version as of 12/10 uses the Loftus &amp; Masson 1994 method and has
been checked against the data in that paper.</p>
<p>See help below for the L &amp; M data and additional
code for getting the ANOVA decomposition.</p>
<p>Useful for barplots of conditions when calculating within-subject SEs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>%% data from Loftus &amp; Masson, Table 2

mtx = [1 10 13 13 12.00
2 6 8 8 7.33
3 11 14 14 13.00
4 22 23 25 23.33
5 16 18 20 18.00
6 15 17 17 16.33
7 1 1 4 2.00
8 12 15 17 14.67
9 9 12 12 11.00
10 8 9 12 9.67];

dat = mtx(:, 2:4);  % data from Loftus &amp; Masson, Table 2

[se_within, stats] = barplot_get_within_ste(dat)
fprintf(&#39;Within ste: %3.2f, 95%% CI: mean +/- %3.2f\n&#39;, se_within, stats.ci);
</pre></div>
</div>
<p>Extra stuff from ANOVA table</p>
<p>Mean square for condition: Variance of condition means * sample
size...average squared variance accounted for by condition means</p>
<div class="highlight-python"><div class="highlight"><pre>[n, k] = size(dat); 
MS_cond = var( mean(dat) - mean(dat(:)) ) * n;

MS_subject = var( mean(dat&#39;) - mean(dat(:)) ) * k;

datv = dat(:);
MS_total = scale(datv, 1)&#39; * scale(datv, 1);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.bayes_get_probabilities">
<code class="descclassname">Statistics_tools.</code><code class="descname">bayes_get_probabilities</code><span class="sig-paren">(</span><em>Y</em>, <em>Xi</em>, <em>k</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.bayes_get_probabilities" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">priors</span><span class="p">,</span> <span class="n">pa1_given_t</span><span class="p">,</span> <span class="n">pa0_given_t</span><span class="p">,</span> <span class="n">pt_given_act1</span><span class="p">,</span> <span class="n">pt_given_act0</span><span class="p">,</span> <span class="n">pa1_given_not_t</span><span class="p">]</span> <span class="o">=</span> <span class="n">bayes_get_probabilities</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">Xi</span><span class="p">,</span> <span class="n">k</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>k:</strong></dt>
<dd><p class="first last">is regularization param, biases towards 0.5</p>
</dd>
<dt><strong>Y:</strong></dt>
<dd><p class="first last">is data matrix, obs x features, 1/0 (active/not)</p>
</dd>
<dt><strong>Xi:</strong></dt>
<dd><p class="first last">is task indicator matrix</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This is a sub-function of classify_naive_bayes.m
For complete help, see classify_naive_bayes.m</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.bayes_get_probabilities_2010">
<code class="descclassname">Statistics_tools.</code><code class="descname">bayes_get_probabilities_2010</code><span class="sig-paren">(</span><em>Y</em>, <em>Xi</em>, <em>k</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.bayes_get_probabilities_2010" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">priors_Pt</span><span class="p">,</span> <span class="n">pa1_given_t</span><span class="p">,</span> <span class="n">pa0_given_t</span><span class="p">,</span> <span class="n">pt_given_act1</span><span class="p">,</span> <span class="n">pt_given_act0</span><span class="p">,</span> <span class="n">pa1_given_not_t</span><span class="p">]</span> <span class="o">=</span> <span class="n">bayes_get_probabilities_2010</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">Xi</span><span class="p">,</span> <span class="n">k</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>k:</strong></dt>
<dd><p class="first last">is regularization param, biases P(activity | task class) towards 0.5</p>
</dd>
<dt><strong>kY:</strong></dt>
<dd><p class="first last">is data matrix, obs x features, 1/0 (active/not)</p>
</dd>
<dt><strong>kXi:</strong></dt>
<dd><p class="first last">is task indicator matrix</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This is a sub-function of classify_naive_bayes.m
For complete help, see classify_naive_bayes.m</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.bayes_meta_feature_abstract">
<code class="descclassname">Statistics_tools.</code><code class="descname">bayes_meta_feature_abstract</code><span class="sig-paren">(</span><em>Y</em>, <em>thresh</em>, <em>shrink</em>, <em>bayes_model</em>, <em>volInfo</em>, <em>doplot</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.bayes_meta_feature_abstract" title="Permalink to this definition">¶</a></dt>
<dd><p>Purpose: Instead of original matrix of activations in studies x voxels,
we may want to work with a data matrix of activations in contiguous
regions x voxels</p>
<p>This function calculates Y = 0,1 for Y = 1 in any voxel in each
contiguous cluster</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">reducedY</span> <span class="o">=</span> <span class="n">bayes_meta_feature_abstract</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="o">.</span><span class="mi">1</span><span class="p">,</span> <span class="n">bayes_model</span><span class="p">,</span> <span class="n">volInfo</span><span class="p">);</span>
<span class="n">xval</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;xval&#39;</span><span class="p">,</span> <span class="n">reducedY</span><span class="p">,</span> <span class="n">Xi</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="n">bestk</span><span class="p">,</span> <span class="n">bestg</span> <span class="p">);</span>
<span class="n">xval</span><span class="o">.</span><span class="n">prop_correct_by_class</span>
</pre></div>
</div>
<p>needs bayes_model.pa1_given_t, nclasses, params.k</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.binotest">
<code class="descclassname">Statistics_tools.</code><code class="descname">binotest</code><span class="sig-paren">(</span><em>X</em>, <em>p</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.binotest" title="Permalink to this definition">¶</a></dt>
<dd><p>Test the number of &#8220;hits&#8221; in each column of X against a null-hypothesis
proportion p, using a binomial test.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">RES</span> <span class="o">=</span> <span class="n">binotest</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">p</span><span class="p">)</span>
</pre></div>
</div>
<p>Assumes elements of each column of X are independent Bernoulli trials.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first last">is a matrix of &#8220;hits&#8221; and &#8220;misses&#8221;, coded as 1s and 0s.</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">is the null hypothesis proportion of &#8220;hits&#8221;, e.g., often p = 0.5</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.binotest_dependent">
<code class="descclassname">Statistics_tools.</code><code class="descname">binotest_dependent</code><span class="sig-paren">(</span><em>X</em>, <em>Po</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.binotest_dependent" title="Permalink to this definition">¶</a></dt>
<dd><p>This function runs several different types of tests on dependent binomial data.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">varargout</span><span class="p">]</span> <span class="o">=</span> <span class="n">binotest_dependent</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">Po</span><span class="p">)</span>
</pre></div>
</div>
<p>Overall, it tests the number of &#8220;hits&#8221; for each subject (row in X) against a null-hypothesis
proportion p, across all subjects using a Z-test (two-tailed).
The second level null hypothesis should be approximated by a normal
distribution with a mean of p.  This approach assumes that each subject
has an equal number of independent Bernoulli trials (columns in X) and
that the number of subjects exceeds n=20 the test will be more accurate as n -&gt; infinity.
Also, calculates tests for each separate trial (e.g., subject columns),
and the difference between proportions (two proportion z-test).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first last">X is a matrix of &#8220;hits&#8221; and &#8220;misses&#8221;, coded as 1s and 0s.
where rows = subjects and columns = observations within
subject</p>
</dd>
<dt><strong>Po:</strong></dt>
<dd><p class="first last">Po is the null hypothesis proportion of &#8220;hits&#8221;, e.g., often p = 0.5</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>RES [1:5]:</strong></dt>
<dd><p class="first last">a structure containing the output of the stats for the
z-test, includes the number of subject (N), number of overall 
hits (hits), the overall proportion of hits (prop), the standard 
deviation (SE), z-statistic (Z), and the two tailed p-value (pval) 
trial across all subjects.  Assumes independence</p>
</dd>
<dt><strong>RES1:</strong></dt>
<dd><p class="first last">Independent Single Interval Test for Column 1 (Column 1 only against Po)</p>
</dd>
<dt><strong>RES2:</strong></dt>
<dd><p class="first last">Independent Single Interval Test for Column 2 (Column 2 only against Po)</p>
</dd>
<dt><strong>RES3:</strong></dt>
<dd><p class="first last">Two proportion dependent difference z-test (Column 1 minus Column 2 against 0)</p>
</dd>
<dt><strong>RES4:</strong></dt>
<dd><p class="first last">Dependent single-interval test (Mean of Column 1 and Column 2 against Po)</p>
</dd>
<dt><strong>RES5:</strong></dt>
<dd><dl class="first last docutils">
<dt>Two proportion dependent addition z-test (Column 1 plus Column 2 against 2 * Po) </dt>
<dd><p class="first last">(Similar to mean, not sure what this will be used for)</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[RES1, RES2, RES3, RES4, RES5] = binotest_dependent([1,1,1,1,0; 1,0,1,0,1]&#39;,.5)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.cancor">
<code class="descclassname">Statistics_tools.</code><code class="descname">cancor</code><span class="sig-paren">(</span><em>X</em>, <em>w</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.cancor" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[cc,stats] = cancor(X,w,[permutations],[MCD robust outlier removal])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first last">data matrix, columns are variables, rows observations</p>
</dd>
<dt><strong>w:</strong></dt>
<dd><p class="first last">first w columns are Set 1, rest are Set 2</p>
</dd>
<dt><strong>cc:</strong></dt>
<dd><p class="first last">canonical correlations</p>
</dd>
<dt><strong>UV:</strong></dt>
<dd><p class="first last">canonical variates</p>
</dd>
<dt><strong>UVa:</strong></dt>
<dd><p class="first last">caonical variates for set a</p>
</dd>
<dt><strong>UVb:</strong></dt>
<dd><p class="first last">canonical variates for set b</p>
</dd>
<dt><strong>ccor:</strong></dt>
<dd><p class="first last">correlations between canonical variates and X variables</p>
</dd>
<dt><strong>ab:</strong></dt>
<dd><p class="first last">weights of canon. variates in row vectors, e.g., U = Xa&#8217;</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Johnson &amp; Wichern, Applied Multivariate Statistical Analysis, 5th ed. 
tested on examples from the book.  cc and ab are right, not positive about ccor</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Compute canonical correlations between first 2 and last 2 columns
of X, testing against permuted column data with 1000 iterations.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">cc</span><span class="p">,</span><span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">cancor</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1000</span><span class="p">,</span><span class="mi">1</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.canlab_connectivity_predict">
<code class="descclassname">Statistics_tools.</code><code class="descname">canlab_connectivity_predict</code><span class="sig-paren">(</span><em>dat</em>, <em>subject_grouping</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.canlab_connectivity_predict" title="Permalink to this definition">¶</a></dt>
<dd><p>Connectivity and multivariate pattern-based prediction for multi-subject timeseries data
Currently runs predictive algorithm(s) on pairwise correlations among regions</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">OUT</span> <span class="o">=</span> <span class="n">canlab_connectivity_predict</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">subject_grouping</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;outcome&#39;</span><span class="p">,</span> <span class="n">outcome_dat</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Features:</th><td class="field-body"><ul class="first last simple">
<li>Within-subject correlation matrices and &#8216;random effects&#8217; statistics</li>
<li>[optional] Prediction with LASSO-PCR/SVR/SVM of outcomes from pairwise connectivity</li>
<li>Time-lagged cross-correlation options</li>
<li>Graph theoretic measures</li>
<li>[optional] Prediction with LASSO-PCR/SVR/SVM of outcomes from graph measures</li>
<li>Can easily be extended to handle partial regression/correlation coefficients</li>
</ul>
</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><blockquote class="first">
<div><dl class="docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">concatenated data matrix of time points (t) within subjects x variables (e.g., ROIs)
[subj x time] x [variables]</p>
</dd>
</dl>
</div></blockquote>
<dl class="docutils">
<dt><strong>subject_grouping:</strong></dt>
<dd><p class="first last">[subj x time]-length integer vector of which
observations belong to which subjects, e.g., [1 1 1 ... 2 2 2 ... 3 3 3 ...]&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;outcome&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by outcome data for multivariate prediction.  connectivity</dt>
<dd><p class="first last">values and graph metrics are used to predict outcome data.</p>
</dd>
</dl>
</dd>
<dt><strong>&#8216;shift_by&#8217;:</strong></dt>
<dd><p class="first last">Followed by integer value for max number of time points to shift</p>
</dd>
<dt><strong>&#8216;partialr&#8217;:</strong></dt>
<dd><p class="first last">Use partial correlation instead of raw correlation</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>OUT:</strong></dt>
<dd><p class="first last">A structure containing subject correlation matrices, the mean
matrix, and raw and FDR-thresholded group matrix
Also contains matrices with [subjects x variables] pairwise
correlation elements and graph metrics</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Use partial correlations:
OUT = canlab_connectivity_predict(dat, subject_grouping, &#39;partialr&#39;);

% Omit graph met
OUT = canlab_connectivity_predict(dat, subject_grouping, &#39;outcome&#39;, y, &#39;nograph&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>parcel_cl, parcel_cl_nmds_plots, canlab_force_directed_graph,
canlab_connectivity_preproc</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.classify_bayes">
<code class="descclassname">Statistics_tools.</code><code class="descname">classify_bayes</code><span class="sig-paren">(</span><em>meth</em>, <em>y</em>, <em>Xi</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.classify_bayes" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[corrclass, taskclass, realclass, likeratio, m, misclass,ptask,indx,cl] = classify_bayes(meth,y,Xi,[var args])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>y:</strong></dt>
<dd><p class="first">observations (e.g., studies) x variables (e.g., brain voxels)</p>
<p class="last">y = SOMResults.dat&#8217;;</p>
</dd>
<dt><strong>Xi:</strong></dt>
<dd><p class="first last">Xi = MC_Setup.Xi(:,1:2);</p>
</dd>
<dt><strong>meth:</strong></dt>
<dd><p class="first">can be: {&#8216;linear&#8217;,&#8217;diagLinear&#8217;,&#8217;quadratic&#8217;,&#8217;diagQuadratic&#8217;,&#8217;mahalanobis&#8217;}</p>
<p class="last">discriminant analysis
&#8216;bayes&#8217; : simple Bayes posterior prob classifier</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Feature selection parameters:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>selectivity_cutoff:</strong></dt>
<dd><p class="first">max probability of task given a response in a variable,
divided by number of tasks.</p>
<p>1 = variable must exceed .5 for 2 tasks, .2 for 5 tasks, etc.</p>
<p>1.5 = .3 for 5 tasks, .75 for 2 tasks, etc.</p>
<p class="last">0 = no selectivity</p>
</dd>
<dt><strong>activation_cutoff:</strong></dt>
<dd><p class="first">max proportion of studies of some type that produced a
response in a variable (e.g., voxel)</p>
<p>.1 is default</p>
<p class="last">0 is no selectivity</p>
</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[corrclass, taskclass, realclass, likeratio, m, misclass] = ...
classify_bayes(&#39;bayes&#39;,MC_Setup.unweighted_study_data&#39;,Xi,&#39;selectivity_cutoff&#39;,1); corrclass

[corrclass, taskclass, realclass, likeratio, m, misclass] =
classify_bayes(&#39;bayes&#39;,MC_Setup.unweighted_study_data&#39;,Xi,&#39;selectivity_cutoff&#39;,1,&#39;activation_cutoff&#39;,.08); corrclass
</pre></div>
</div>
<p>Batch modes:
Permutation test: input &#8216;permtest&#8217; followed by number of permutations
Do not request more than one output variable
[corrclass_nullhyp] = classify_bayes(&#8216;bayes&#8217;,dat,Xi,&#8217;selectivity_cutoff&#8217;,1.8,&#8217;activation_cutoff&#8217;,.02,&#8217;permtest&#8217;,5);</p>
<p>To get mask index of which areas meet feature selection:
whsave = sum(indx &gt; 0, 2) &gt; 0;</p>
<p>select features first, instead of in x-validation (makes x-val
invalid; don&#8217;t do it)</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.classify_choose_most_likely">
<code class="descclassname">Statistics_tools.</code><code class="descname">classify_choose_most_likely</code><span class="sig-paren">(</span><em>ptask</em>, <em>testvec</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.classify_choose_most_likely" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">taskclass</span><span class="p">,</span><span class="n">maxlike</span><span class="p">,</span><span class="n">likeratio</span><span class="p">,</span> <span class="n">taskprob</span><span class="p">]</span> <span class="o">=</span> <span class="n">classify_choose_most_likely</span><span class="p">(</span><span class="n">ptask</span><span class="p">,</span><span class="n">testvec</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>ptask:</strong></dt>
<dd><p class="first last">likelihood of each task given activation, p(task | activation)
classses x variables (features, brain voxels)</p>
</dd>
<dt><strong>testvec:</strong></dt>
<dd><p class="first last">activation values across features variables x 1</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>taskclass:</strong></dt>
<dd><p class="first last">integer for which is max likelihood class</p>
</dd>
<dt><strong>maxlike :</strong></dt>
<dd><dl class="first last docutils">
<dt>log likelihood of chosen class given data (if testvec is 1/0</dt>
<dd><p class="first last">indicator)</p>
</dd>
</dl>
</dd>
<dt><strong>likeratio :</strong></dt>
<dd><p class="first last">likelihood ratio for most likely vs. least likely class</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% voxels in original image space that were in dataset
whsave = sum(indx &gt; 0, 2) &gt; 0;

[tc, ml, lr] = classify_choose_most_likely(ptask, MC_Setup.unweighted_study_data(whsave,1))
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.classify_naive_bayes">
<code class="descclassname">Statistics_tools.</code><code class="descname">classify_naive_bayes</code><span class="sig-paren">(</span><em>meth</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.classify_naive_bayes" title="Permalink to this definition">¶</a></dt>
<dd><p>Naive Bayes classifier</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% set up model structure
bayes_model = classify_naive_bayes(&#39;setup&#39;, Y, Xi, [activation_cutoff, selectivity_cutoff]);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">is full (not sparse); empty features will be eliminated</p>
</dd>
<dt><strong>Xi:</strong></dt>
<dd><p class="first last">is obs x classes, an indicator matrix of 1&#8217;s and 0&#8217;s</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">TEST:</th><td class="field-body"><p class="first last">test classifier; make a prediction about classes from data</p>
</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[class_est log_joint best_log map p_obs_act_given_class] = classify_naive_bayes(&#39;test&#39;, Y, bayes_model);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">EVAL:</th><td class="field-body">evaluate classification accuracy</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">prop_correct</span><span class="p">,</span> <span class="n">confusion_mtx</span><span class="p">,</span> <span class="n">misclass</span><span class="p">,</span> <span class="n">prop_correct_by_class</span><span class="p">,</span> <span class="n">chance</span><span class="p">,</span> <span class="n">chance_95_ci</span><span class="p">]</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;eval&#39;</span><span class="p">,</span> <span class="n">true_class</span><span class="p">,</span> <span class="n">class_est</span><span class="p">,</span> <span class="n">wh_obs</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">APPARENT:</th><td class="field-body">apparent classification; with full dataset</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">bayes_model</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;apparent&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">bayes_model</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">XVAL:</th><td class="field-body">crossvalidate</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">xval</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;xval&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">Xi</span><span class="p">);</span>

<span class="n">bayes_model</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;write&#39;</span><span class="p">,</span> <span class="n">bayes_model</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">volInfo</span><span class="p">,</span> <span class="n">conditionnames</span><span class="p">)</span>
<span class="n">bayes_model</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;write&#39;</span><span class="p">,</span> <span class="n">bayes_model</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">MC_Setup</span><span class="o">.</span><span class="n">volInfo</span><span class="p">,</span> <span class="n">MC_Setup</span><span class="o">.</span><span class="n">Xinms</span><span class="p">);</span>
</pre></div>
</div>
<p>To add feature abstraction step within xval:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">xval</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;xval&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">Xi</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="o">.</span><span class="mi">9</span><span class="p">,</span> <span class="o">.</span><span class="mo">05</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">volInfo</span> <span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">PLOT:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>classify_naive_bayes(&#39;plot&#39;, bayes_model, [&#39;pa|t&#39;, &#39;lr&#39;, &#39;lr surface&#39;,  &#39;map plot&#39;, or &#39;class plot&#39;]);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body">(any order)</td>
</tr>
</tbody>
</table>
<p>Threshold (abs. value), and colors in cell array</p>
<div class="highlight-python"><div class="highlight"><pre>classify_naive_bayes(&#39;plot&#39;, bayes_model, &#39;lr&#39;, .10, {[1 .7 0] [0 0 1]});
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[bayes_model, Y] = classify_naive_bayes(&#39;setup&#39;, Y, Xi);

% Test obs. 2
tic, [class_est log_joint best_log map] = classify_naive_bayes(&#39;test&#39;,Y(2,:), bayes_model); toc

% Get apparent classification rate and look at confusion matrix
tic, bayes_model = classify_naive_bayes(&#39;apparent&#39;, Y, bayes_model); toc
bayes_model.apparent.confusion_mtx

% Cross-validate
bayes_model.xval = classify_naive_bayes(&#39;xval&#39;, Y, Xi);
bayes_model.xval.confusion_mtx
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example 2:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre> Select features, and do apparent and cross-validated classification
Y = MC_Setup.unweighted_study_data&#39;;
wh = sum(Y) &gt; 5;
Y = Y(:, wh);
whos Y
[bayes_model, Y] = classify_naive_bayes(&#39;setup&#39;, Y, Xi);
bayes_model = classify_naive_bayes(&#39;apparent&#39;, Y, bayes_model);
bayes_model.apparent.prop_correct, bayes_model.apparent.confusion_mtx
bayes_model.xval = classify_naive_bayes(&#39;xval&#39;, Y, Xi);
bayes_model.xval.prop_correct, bayes_model.xval.confusion_mtx
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example 3:</th><td class="field-body">create_figure(&#8216;hist&#8217;); hist(bayes_model.pa1_given_t, 100);
xval = classify_naive_bayes(&#8216;xval&#8217;, Y, Xi, .05, .3);</td>
</tr>
</tbody>
</table>
<p>Get results from key regions and run classifier only on those regions:</p>
<div class="highlight-python"><div class="highlight"><pre> cl = classify_naive_bayes(&#39;plot&#39;, bayes_model, &#39;lr&#39;, .10, {[1 .7 0] [0 0 1]});
[studybyroi,studybyset] = Meta_cluster_tools(&#39;getdata&#39;,cl{1},dat&#39;,volInfo);
bayes_model_regions = classify_naive_bayes(&#39;setup&#39;, studybyroi, Xi);
bayes_model_regions = classify_naive_bayes(&#39;apparent&#39;, studybyroi,bayes_model_regions);
disp(&#39;Apparent confusion&#39;)
disp(bayes_model_regions.apparent.confusion_mtx)

bayes_model_regions.xval = classify_naive_bayes(&#39;xval&#39;, studybyroi, Xi);
disp(&#39;Cross-validated confusion&#39;)
disp(bayes_model_regions.xval.confusion_mtx)

fprintf(&#39;Proportion correct: Apparent: %3.0f%%  Xval: %3.0f%%\n&#39;, 100*bayes_model_regions.apparent.prop_correct, 100*bayes_model_regions.xval.prop_correct);
fprintf(&#39;Proportion correct by class: \t&#39;); fprintf(&#39;%3.0f%%\t&#39;, 100*bayes_model_regions.xval.prop_correct_by_class);
fprintf(&#39;\n&#39;);
sz = cat(1,cl{1}(:).numVox);
cl{1}(sz &lt; 10) = [];

subcl = subclusters_from_local_max(cl{1}, 10);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.classify_naive_bayes_2010">
<code class="descclassname">Statistics_tools.</code><code class="descname">classify_naive_bayes_2010</code><span class="sig-paren">(</span><em>meth</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.classify_naive_bayes_2010" title="Permalink to this definition">¶</a></dt>
<dd><p>Naive Bayes classifier</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% set up model structure
bayes_model = classify_naive_bayes(&#39;setup&#39;, Y, Xi, [activation_cutoff, selectivity_cutoff]);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">is full (not sparse); empty features will be eliminated</p>
</dd>
<dt><strong>Xi:</strong></dt>
<dd><p class="first last">is obs x classes, an indicator matrix of 1&#8217;s and 0&#8217;s</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">TEST:</th><td class="field-body"><p class="first last">test classifier; make a prediction about classes from data</p>
</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[class_est log_joint best_log map p_obs_act_given_class] = classify_naive_bayes(&#39;test&#39;, Y, bayes_model);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">EVAL:</th><td class="field-body">evaluate classification accuracy</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">prop_correct</span><span class="p">,</span> <span class="n">confusion_mtx</span><span class="p">,</span> <span class="n">misclass</span><span class="p">,</span> <span class="n">prop_correct_by_class</span><span class="p">,</span> <span class="n">chance</span><span class="p">,</span> <span class="n">chance_95_ci</span><span class="p">]</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;eval&#39;</span><span class="p">,</span> <span class="n">true_class</span><span class="p">,</span> <span class="n">class_est</span><span class="p">,</span> <span class="n">wh_obs</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">APPARENT:</th><td class="field-body">apparent classification; with full dataset</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">bayes_model</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;apparent&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">bayes_model</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">XVAL:</th><td class="field-body">crossvalidate</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">xval</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;xval&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">Xi</span><span class="p">);</span>

<span class="n">bayes_model</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;write&#39;</span><span class="p">,</span> <span class="n">bayes_model</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">volInfo</span><span class="p">,</span> <span class="n">conditionnames</span><span class="p">)</span>
<span class="n">bayes_model</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;write&#39;</span><span class="p">,</span> <span class="n">bayes_model</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">MC_Setup</span><span class="o">.</span><span class="n">volInfo</span><span class="p">,</span> <span class="n">MC_Setup</span><span class="o">.</span><span class="n">Xinms</span><span class="p">);</span>
</pre></div>
</div>
<p>To add feature abstraction step within xval:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">xval</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;xval&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">Xi</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="o">.</span><span class="mi">9</span><span class="p">,</span> <span class="o">.</span><span class="mo">05</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="n">volInfo</span> <span class="p">);</span>
</pre></div>
</div>
<p>:PLOT</p>
<div class="highlight-python"><div class="highlight"><pre>classify_naive_bayes(&#39;plot&#39;, bayes_model, [&#39;pa|t&#39;, &#39;lr&#39;, &#39;lr surface&#39;,  &#39;map plot&#39;, or &#39;class plot&#39;]);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body">(any order)</td>
</tr>
</tbody>
</table>
<p>Threshold (abs. value), and colors in cell array</p>
<div class="highlight-python"><div class="highlight"><pre>classify_naive_bayes(&#39;plot&#39;, bayes_model, &#39;lr&#39;, .10, {[1 .7 0] [0 0 1]});
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[bayes_model, Y] = classify_naive_bayes(&#39;setup&#39;, Y, Xi);

% Test obs. 2
tic, [class_est log_joint best_log map] = classify_naive_bayes(&#39;test&#39;,Y(2,:), bayes_model); toc

% Get apparent classification rate and look at confusion matrix
tic, bayes_model = classify_naive_bayes(&#39;apparent&#39;, Y, bayes_model); toc
bayes_model.apparent.confusion_mtx

% Cross-validate
bayes_model.xval = classify_naive_bayes(&#39;xval&#39;, Y, Xi);
bayes_model.xval.confusion_mtx
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example 2:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Select features, and do apparent and cross-validated classification</p>
<div class="highlight-python"><div class="highlight"><pre>Y = MC_Setup.unweighted_study_data&#39;;
wh = sum(Y) &gt; 5;
Y = Y(:, wh);
whos Y
[bayes_model, Y] = classify_naive_bayes(&#39;setup&#39;, Y, Xi);
bayes_model = classify_naive_bayes(&#39;apparent&#39;, Y, bayes_model);
bayes_model.apparent.prop_correct, bayes_model.apparent.confusion_mtx
bayes_model.xval = classify_naive_bayes(&#39;xval&#39;, Y, Xi);
bayes_model.xval.prop_correct, bayes_model.xval.confusion_mtx
</pre></div>
</div>
<p>Example 3:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">create_figure</span><span class="p">(</span><span class="s">&#39;hist&#39;</span><span class="p">);</span> <span class="n">hist</span><span class="p">(</span><span class="n">bayes_model</span><span class="o">.</span><span class="n">pa1_given_t</span><span class="p">,</span> <span class="mi">100</span><span class="p">);</span>
<span class="n">xval</span> <span class="o">=</span> <span class="n">classify_naive_bayes</span><span class="p">(</span><span class="s">&#39;xval&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">Xi</span><span class="p">,</span> <span class="o">.</span><span class="mo">05</span><span class="p">,</span> <span class="o">.</span><span class="mi">3</span><span class="p">);</span>
</pre></div>
</div>
<p>Get results from key regions and run classifier only on those regions:</p>
<div class="highlight-python"><div class="highlight"><pre>cl = classify_naive_bayes(&#39;plot&#39;, bayes_model, &#39;lr&#39;, .10, {[1 .7 0] [0 0 1]});
[studybyroi,studybyset] = Meta_cluster_tools(&#39;getdata&#39;,cl{1},dat&#39;,volInfo);
bayes_model_regions = classify_naive_bayes(&#39;setup&#39;, studybyroi, Xi);
bayes_model_regions = classify_naive_bayes(&#39;apparent&#39;, studybyroi,bayes_model_regions);
disp(&#39;Apparent confusion&#39;)
disp(bayes_model_regions.apparent.confusion_mtx)

bayes_model_regions.xval = classify_naive_bayes(&#39;xval&#39;, studybyroi, Xi);
disp(&#39;Cross-validated confusion&#39;)
disp(bayes_model_regions.xval.confusion_mtx)

fprintf(&#39;Proportion correct: Apparent: %3.0f%%  Xval: %3.0f%%\n&#39;, 100*bayes_model_regions.apparent.prop_correct, 100*bayes_model_regions.xval.prop_correct);
fprintf(&#39;Proportion correct by class: \t&#39;); fprintf(&#39;%3.0f%%\t&#39;, 100*bayes_model_regions.xval.prop_correct_by_class);
fprintf(&#39;\n&#39;);

sz = cat(1,cl{1}(:).numVox);
cl{1}(sz &lt; 10) = [];

subcl = subclusters_from_local_max(cl{1}, 10);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.classify_naive_bayes_objfun">
<code class="descclassname">Statistics_tools.</code><code class="descname">classify_naive_bayes_objfun</code><span class="sig-paren">(</span><em>dat</em>, <em>Xi</em>, <em>volInfo</em>, <em>a</em>, <em>s</em>, <em>g</em>, <em>k</em>, <em>t</em>, <em>h</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.classify_naive_bayes_objfun" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">goodness</span> <span class="o">=</span> <span class="n">classify_naive_bayes_objfun</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">Xi</span><span class="p">,</span> <span class="n">volInfo</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="o">.</span><span class="mi">9</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="o">.</span><span class="mo">05</span><span class="p">,</span> <span class="o">.</span><span class="mi">1</span><span class="p">,</span> <span class="o">.</span><span class="mi">1</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>objfun = @(t, h) classify_naive_bayes_objfun(dat, Xi, volInfo, 0, .9, 1, .05, t, h);
goodness = objfun(.1, .1)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.classify_viz_regions">
<code class="descclassname">Statistics_tools.</code><code class="descname">classify_viz_regions</code><span class="sig-paren">(</span><em>indx</em>, <em>colors</em>, <em>ptaskcutoff</em>, <em>sizecutoff</em>, <em>maskimage</em>, <em>names</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.classify_viz_regions" title="Permalink to this definition">¶</a></dt>
<dd><p>Visualize output of classify_bayes.m, or comparable output</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">classify_viz_regions</span><span class="p">(</span><span class="n">indx</span><span class="p">,</span><span class="n">colors</span><span class="p">,</span><span class="n">ptaskcutoff</span><span class="p">,</span><span class="n">sizecutoff</span><span class="p">,</span><span class="n">maskimage</span><span class="p">,</span><span class="n">names</span><span class="p">)</span>
</pre></div>
</div>
<p>indx should be voxels x images, with prob task (ptask) in non-zero
elements</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">classify_viz_regions</span><span class="p">(</span><span class="n">indx</span><span class="p">,[],</span><span class="o">.</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">);</span>
<span class="n">cl</span> <span class="o">=</span> <span class="n">classify_viz_regions</span><span class="p">(</span><span class="n">indx</span><span class="p">,[],</span><span class="o">.</span><span class="mi">6</span><span class="p">,</span><span class="mi">5</span><span class="p">,[],</span><span class="n">names</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.contrast_code">
<code class="descclassname">Statistics_tools.</code><code class="descname">contrast_code</code><span class="sig-paren">(</span><em>vec</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.contrast_code" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">vec</span><span class="p">,</span><span class="n">outnames</span><span class="p">]</span> <span class="o">=</span> <span class="n">contrast_code</span><span class="p">(</span><span class="n">vec</span><span class="p">)</span>
</pre></div>
</div>
<p>Changes values to 1, -1, or 0 for contrast coding</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.correl_compare_dep">
<code class="descclassname">Statistics_tools.</code><code class="descname">correl_compare_dep</code><span class="sig-paren">(</span><em>y1</em>, <em>y2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.correl_compare_dep" title="Permalink to this definition">¶</a></dt>
<dd><p>Compare dependent correlations between pairs of vectors in y1 and y2.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">out</span> <span class="o">=</span> <span class="n">correl_compare_dep</span><span class="p">(</span><span class="n">y1</span><span class="p">,</span><span class="n">y2</span><span class="p">,[</span><span class="s">&#39;alpha&#39;</span><span class="p">,</span><span class="n">myalpha</span><span class="p">],[</span><span class="s">&#39;rank&#39;</span><span class="p">],[</span><span class="s">&#39;table&#39;</span><span class="p">])</span>
</pre></div>
</div>
<p>Each of y1 and y2 would contain at least 2 columns, which would be
correlated and saved in r1 and r2 matrices in output.
Then, the r1 and r2 matrices are subtracted, and P-values are
returned for the differences.</p>
<blockquote>
<div><ul class="simple">
<li>In the simplest case, y1 would contain vectors [a b] and y2 would
contain vectors [a c].  tests are provided on the a-b vs. a-c
difference in correlations.</li>
</ul>
</div></blockquote>
<p>Repeats dep. correl. analysis for each pair of columns</p>
<p>Returns results in correlation matrix form, where number of rows and
cols. are the number of pairs [y1(:,i) y2(:,i)]</p>
<p>myalpha is 2-tailed alpha value; p-values are 2-tailed
FDR correction is at .01, 2-tailed</p>
<p>Based on Steiger, 1980, tests for comparing dependent correlations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>for i = 1:length(cl), y1(:,i) = cl.CONTRAST.data(:,2); y2(:,i) = cl.CONTRAST.data(:,1); end
for i = 1:length(cl), y1(:,i) = cl(i).CONTRAST.data(:,2); y2(:,i) = cl(i).CONTRAST.data(:,1); end

% y1 is matrix of obs x data vectors for condition 1
% y2 is matrix of obs x data vectors for condition 2
out = correl_compare_dep(y1,y2)

figure(&#39;Color&#39;,&#39;w&#39;);nmdsfig(c.GroupSpace,c.ClusterSolution.classes, ...
c.names,out.sig,1,{&#39;Pos&#39; &#39;Neg&#39;});
nmdsfig_legend(c.ClusterSolution.X,c.r)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>c_compare = correl_compare_dep(y1avg,y2avg,&#39;alpha&#39;,.05,&#39;rank&#39;,&#39;table&#39;,&#39;names&#39;,c.APPLY_CLUSTER.names);

out = correl_compare_dep([ypred pain],[ypred temp], &#39;alpha&#39;, .06, &#39;table&#39;, &#39;names&#39;, {&#39;biomarker resp&#39; &#39;pain or temp&#39;});
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.correl_compare_dep_permtest">
<code class="descclassname">Statistics_tools.</code><code class="descname">correl_compare_dep_permtest</code><span class="sig-paren">(</span><em>y1</em>, <em>y2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.correl_compare_dep_permtest" title="Permalink to this definition">¶</a></dt>
<dd><p>Compare dependent correlations between pairs of vectors in y1 and y2</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">out</span> <span class="o">=</span> <span class="n">correl_compare_dep_permtest</span><span class="p">(</span><span class="n">y1</span><span class="p">,</span><span class="n">y2</span><span class="p">,[</span><span class="s">&#39;alpha&#39;</span><span class="p">,</span><span class="n">myalpha</span><span class="p">],[</span><span class="s">&#39;rank&#39;</span><span class="p">],[</span><span class="s">&#39;table&#39;</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">PERMUTATION TEST:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body">for correl_compare_dep</td>
</tr>
</tbody>
</table>
<p>Repeats dep. correl. analysis for each pair of columns
Returns results in correlation matrix form, where number of rows and
cols. are the number of pairs [y1(:,i) y2(:,i)]</p>
<p>myalpha is 2-tailed alpha value; p-values are 2-tailed
FDR correction is at .01, 2-tailed</p>
<p>Based on Steiger, 1980, tests for comparing dependent correlations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>for i = 1:length(cl), y1(:,i) = cl.CONTRAST.data(:,2); y2(:,i) = cl.CONTRAST.data(:,1); end
for i = 1:length(cl), y1(:,i) = cl(i).CONTRAST.data(:,2); y2(:,i) = cl(i).CONTRAST.data(:,1); end
y1 is matrix of obs x data vectors for condition 1
y2 is matrix of obs x data vectors for condition 2
out = correl_compare_dep(y1,y2)

figure(&#39;Color&#39;,&#39;w&#39;);nmdsfig(c.GroupSpace,c.ClusterSolution.classes, ...
c.names,out.sig,1,{&#39;Pos&#39; &#39;Neg&#39;});
nmdsfig_legend(c.ClusterSolution.X,c.r)

% compare correlations on cluster averages
c_compare = correl_compare_dep(y1avg,y2avg,&#39;alpha&#39;,.05,&#39;rank&#39;,&#39;table&#39;,&#39;names&#39;,c.APPLY_CLUSTER.names);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.correl_compare_dep_search">
<code class="descclassname">Statistics_tools.</code><code class="descname">correl_compare_dep_search</code><span class="sig-paren">(</span><em>seed1</em>, <em>seed2</em>, <em>y1</em>, <em>y2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.correl_compare_dep_search" title="Permalink to this definition">¶</a></dt>
<dd><p>Compare dependent correlations between seed1&lt;-&gt;y1 and seed2&lt;-&gt;y2</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">correl_compare_dep_search</span><span class="p">(</span><span class="n">seed1</span><span class="p">,</span><span class="n">seed2</span><span class="p">,</span><span class="n">y1</span><span class="p">,</span><span class="n">y2</span><span class="p">,[</span><span class="s">&#39;alpha&#39;</span><span class="p">,</span><span class="n">myalpha</span><span class="p">],[</span><span class="s">&#39;rank&#39;</span><span class="p">],[</span><span class="s">&#39;mask&#39;</span><span class="p">,</span><span class="n">maskimage</span><span class="p">])</span>
</pre></div>
</div>
<p>Repeats dep. correl. analysis for each pair of columns
Returns results in correlation matrix form, where number of rows and
cols. are the number of pairs [y1(:,i) y2(:,i)]</p>
<p>myalpha is 2-tailed alpha value; p-values are 2-tailed
FDR correction is at .01, 2-tailed</p>
<p>Based on Steiger, 1980, tests for comparing dependent correlations.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>for i = 1:length(cl), y1(:,i) = cl.CONTRAST.data(:,2); y2(:,i) = cl.CONTRAST.data(:,1); end
for i = 1:length(cl), y1(:,i) = cl(i).CONTRAST.data(:,2); y2(:,i) = cl(i).CONTRAST.data(:,1); end
y1 is matrix of obs x data vectors for condition 1
y2 is matrix of obs x data vectors for condition 2
out = correl_compare_dep(y1,y2)

figure(&#39;Color&#39;,&#39;w&#39;);nmdsfig(c.GroupSpace,c.ClusterSolution.classes, ...
c.names,out.sig,1,{&#39;Pos&#39; &#39;Neg&#39;});
nmdsfig_legend(c.ClusterSolution.X,c.r)

% compare correlations on cluster averages
c_compare = correl_compare_dep(y1avg,y2avg,&#39;alpha&#39;,.05,&#39;rank&#39;,&#39;table&#39;,&#39;names&#39;,c.APPLY_CLUSTER.names);

% get image names
EXPT.subjects = {&#39;ambar_carvalho&#39; &#39;andreas_nguyen&#39; &#39;angela_valle&#39; &#39;brad_wilson&#39; &#39;dominic_ricci&#39;};
EXPT = getfunctnames2(EXPT,&#39;con_0004.img&#39;,&#39;tmp&#39;,&#39;SPM_analysis/physical_pain/full_model_gv_p_v_np&#39;)
self = str2mat(EXPT.tmp{:})

EXPT = getfunctnames2(EXPT,&#39;con_0003.img&#39;,&#39;tmp&#39;,&#39;SPM_analysis/Videos/event_pain_only&#39;)
other = str2mat(EXPT.tmp{:})

maskimage = which(&#39;scalped_avg152T1_graymatter_smoothed.img&#39;)

% get seeds
cd Jamil/GROUP_ANALYSES/Overlaps/Overlap_29_Aug/
cl = mask2clusters(&#39;con_0002.img&#39;);
cl = cl(4)
cl = extract_contrast_data([{self} {other}],cl);
seedself = cl(1).CONTRAST.data(:,1);
seedother = cl(1).CONTRAST.data(:,2);

correl_compare_dep_search(seedself,seedother,self,other,&#39;alpha&#39;,.005,&#39;mask&#39;,mask);

% RESULTS:
cl = mask2clusters(&#39;Correl_seed1_sig.img&#39;); cluster_orthviews(cl,&#39;bivalent&#39;);

% Try the whole thing on ranks:
correl_compare_dep_search(seedself,seedother,self,other,&#39;alpha&#39;,.005,&#39;mask&#39;,mask,&#39;rank&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.correl_compare_indep">
<code class="descclassname">Statistics_tools.</code><code class="descname">correl_compare_indep</code><span class="sig-paren">(</span><em>y_g1</em>, <em>y_g2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.correl_compare_indep" title="Permalink to this definition">¶</a></dt>
<dd><p>Compute statistics on the difference between correlation coefficients
for indepndent samples g1 and g2.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">r1</span><span class="p">,</span> <span class="n">r2</span><span class="p">,</span> <span class="n">rdiff</span><span class="p">,</span> <span class="n">rdiff_p</span><span class="p">,</span> <span class="n">rdiff_zscore</span><span class="p">]</span> <span class="o">=</span> <span class="n">correl_compare_indep</span><span class="p">(</span><span class="n">y_g1</span><span class="p">,</span> <span class="n">y_g2</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Correlations are computed over y (n observations x k variables) for each
group.</p>
<p>Source, Hubert Blalock, Social Statistics, NY: McGraw-Hill, 1972:  406-407.
From notes on Garson stats website.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>y_g1:</strong></dt>
<dd><p class="first last">Matrix of variables for group 1 (n observations x k variables)</p>
</dd>
<dt><strong>y_g2:</strong></dt>
<dd><p class="first last">Matrix of variables for group 2 (n observations x k variables)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>varargin:</strong></dt>
<dd><p class="first last">String &#8216;noverbose&#8217; can be used to prevent output from being
printed to command line</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>r1:</strong></dt>
<dd><p class="first last">Matrix of correlation coefficients for group 1</p>
</dd>
<dt><strong>r2:</strong></dt>
<dd><p class="first last">Matrix of correlation coefficients for group 2</p>
</dd>
<dt><strong>rdiff:</strong></dt>
<dd><p class="first last">Difference in correlation coefficient matrices (r1 - r2)</p>
</dd>
<dt><strong>rdiff_p:</strong></dt>
<dd><p class="first last">p values testing against no difference between r1 and r2. 
Hubert Blalock, Social Statistics, NY: McGraw-Hill, 1972:  406-407.</p>
</dd>
<dt><strong>rdiff_zscore:</strong></dt>
<dd><p class="first last">z scores for the difference between r1 and r2.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>%generate two random input matrices (20 subjects x 5 variables)
y_g1 = rand(20,5);
y_g2 = rand(20,5);
[r1, r2, rdiff, rdiff_p, rdiff_zscore] = correl_compare_indep(y_g1,y_g2);

%supress output
[r1, r2, rdiff, rdiff_p, rdiff_zscore] = correl_compare_indep(y_g1,y_g2,&#39;noverbose&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"><ul class="first last simple">
<li>correl_compare_indep_inputr, correl_compare_permute*</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.correl_compare_indep_inputr">
<code class="descclassname">Statistics_tools.</code><code class="descname">correl_compare_indep_inputr</code><span class="sig-paren">(</span><em>r1</em>, <em>r2</em>, <em>N1</em>, <em>N2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.correl_compare_indep_inputr" title="Permalink to this definition">¶</a></dt>
<dd><p>Compare two Pearson&#8217;s correlation values collected from independent samples
:Based on: www.stat-help.com/
:Usage:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">rdiff</span><span class="p">,</span> <span class="n">Z</span><span class="p">,</span> <span class="n">pval</span><span class="p">,</span> <span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">correl_compare_indep_inputr</span><span class="p">(</span><span class="n">r1</span><span class="p">,</span> <span class="n">r2</span><span class="p">,</span> <span class="n">N1</span><span class="p">,</span> <span class="n">N2</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>r1:</strong></dt>
<dd><p class="first last">Correlation coefficient for group 1</p>
</dd>
<dt><strong>r2:</strong></dt>
<dd><p class="first last">Correlation coefficient for group 2</p>
</dd>
<dt><strong>N1:</strong></dt>
<dd><p class="first last">Sample size for group 1</p>
</dd>
<dt><strong>N2:</strong></dt>
<dd><p class="first last">Sample size for group 2</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>rdiff:</strong></dt>
<dd><p class="first last">Difference in correlation coefficient matrices (r1 - r2)</p>
</dd>
<dt><strong>Z:</strong></dt>
<dd><p class="first last">Z score for difference in correlation coefficients.</p>
</dd>
<dt><strong>pval:</strong></dt>
<dd><p class="first last">p value for two-tailed z-test</p>
</dd>
<dt><strong>stats:</strong></dt>
<dd><p class="first last">Structure of variables including r1, r2, N1, N2, rdiff, zdiff, Z,
pval, myalpha, sig</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre> %generate two random input matrices (20 subjects x 5 variables)
 r1 = .77;
 r2 = .33;
 N1 = 100;
 N2 = 100;

[rdiff, Z, pval, stats] = correl_compare_indep_inputr(r1, r2, N1, N2);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"><ul class="first last simple">
<li>correl_compare_indep_inputr, correl_compare_permute*</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.correl_compare_permute">
<code class="descclassname">Statistics_tools.</code><code class="descname">correl_compare_permute</code><span class="sig-paren">(</span><em>meth</em>, <em>dat</em>, <em>nperms</em>, <em>condition</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.correl_compare_permute" title="Permalink to this definition">¶</a></dt>
<dd><p>General function for comparing correlation matrices on two different
sets of observations</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>::</dt>
<dd><p class="first">correl_compare_permute(meth,dat,nperms,condition)</p>
<p class="last">dat is obs x variables, and an n x n association matrix will be computed
on the n(n - 1) pairs of columns using one of the methods in
correlation.m and specified by meth</p>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>meth:</strong></dt>
<dd><p class="first last">String indicating the method for computing the correlation
coefficient (e.g., &#8216;taub&#8217;) see correlation.m for details</p>
</dd>
<dt><strong>dat:</strong></dt>
<dd><p class="first last">Matrix of observations (n instances by p variables)</p>
</dd>
<dt><strong>nperms:</strong></dt>
<dd><p class="first last">Number of permutations for inferential tests</p>
</dd>
<dt><strong>condition:</strong></dt>
<dd><p class="first last">Vector indicating condition membership for each instance
(currently only works for two conditions)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>OUT:</strong></dt>
<dd><p class="first last">Output stats structure</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"><ul class="first last simple">
<li>correl_compare_indep_inputr, correl_compare_indep, correlation*</li>
</ul>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.correlation">
<code class="descclassname">Statistics_tools.</code><code class="descname">correlation</code><span class="sig-paren">(</span><em>meth</em>, <em>x</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.correlation" title="Permalink to this definition">¶</a></dt>
<dd><p>Multiple types of correlations, including Spearman&#8217;s rho
(nonparametric) and phi (dichotomous)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[corr,t,p,fdrp, fdrthresh] = correlation(method,x,[y],[&#39;matrix&#39;])

IN PROGRESS : Warning : Use at your own risk.
Some methods are not adequately tested yet.
Spearman&#39;s rho does not correct for ties
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Methods:</strong></dt>
<dd><p class="first last">String indicating the method for computing the correlation
coefficient
- Pearson&#8217;s r.          Enter: {&#8216;r&#8217;,&#8217;pearson&#8217;,[]}
- IRLS                  Enter: {&#8216;irls&#8217;,&#8217;robust&#8217;}
- Phi                   Enter: {&#8216;phi&#8217;}
- Spearman&#8217;s rho        Enter: {&#8216;rho&#8217;,&#8217;spearman&#8217;}
- Kendall&#8217;s Tau (a)     Enter: {&#8216;taua&#8217;,&#8217;kendalla&#8217;}
- Tau (b)               Enter: {&#8216;tau&#8217;,&#8217;kendall&#8217;,&#8217;taub&#8217;,&#8217;kendallb&#8217;}
- Gamma                 Enter: {&#8216;gamma&#8217;,&#8217;kruskal&#8217;}</p>
</dd>
<dt><strong>x:</strong></dt>
<dd><p class="first last">Matrix of observations (n instances by p varianbles)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>varargin:</strong></dt>
<dd><p class="first last">To be documented</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>OUT:</strong></dt>
<dd><p class="first last">Output stats structure</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Corelation between two variables, Pearson&#39;s
x = rand(10,1); y = rand(10,1);
[corr,t,p] = correlation(&#39;r&#39;,x,y);

% Correlation matrix of 10 variables, phi correlation:
studybyroi = magic(10);
[corr,t,p] = correlation(&#39;phi&#39;,studybyroi);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">correlation_fast_series.m</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.correlation_fast_series">
<code class="descclassname">Statistics_tools.</code><code class="descname">correlation_fast_series</code><span class="sig-paren">(</span><em>Xi</em>, <em>Yi</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.correlation_fast_series" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">r</span><span class="p">,</span> <span class="n">p</span><span class="p">,</span> <span class="n">t</span><span class="p">]</span> <span class="o">=</span> <span class="n">correlation_fast_series</span><span class="p">(</span><span class="n">Xi</span><span class="p">,</span> <span class="n">Yi</span><span class="p">)</span>  
</pre></div>
</div>
<p>Fast, memory-efficient way to get correlation between each column of Xi and Yi</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Xi:</strong></dt>
<dd><p class="first last">Matrix of observations (n instances by p variables)</p>
</dd>
<dt><strong>Yi:</strong></dt>
<dd><p class="first last">Matrix of observations (n instances by p variables)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>r:</strong></dt>
<dd><p class="first last">Pearson correlation coefficients</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">Corresponding p-value for correlation coeffieints</p>
</dd>
<dt><strong>Tstat:</strong></dt>
<dd><p class="first last">T statistic for comparing correlation coefficients against 0</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>r = ((Xi&#39; * Yi) ./ N) ./ sqrt(diag(Xi&#39; * Xi) ./ N * (Yi&#39; * Yi) ./ N);
[r2, p, Tstat] = correlation_fast_series(Xi, Yi);
[r3, p2] = corrcoef([Yi, Xi]); r3 = r3(1, 2:end); p2 = p2(1, 2:end);
create_figure(&#39;test&#39;, 1, 3); plot(r3, r, &#39;kx&#39;); axis equal; grid on; subplot(1, 3, 2); plot(r3, r2, &#39;kx&#39;);  axis equal; grid on; subplot(1, 3, 3); plot(p2, p, &#39;kx&#39;); axis equal; grid on;
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.create_orthogonal_contrast_set">
<code class="descclassname">Statistics_tools.</code><code class="descname">create_orthogonal_contrast_set</code><span class="sig-paren">(</span><em>nconditions</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.create_orthogonal_contrast_set" title="Permalink to this definition">¶</a></dt>
<dd><p>Create an orthogonal contrast set across nconditions conditions</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">create_orthogonal_contrast_set</span><span class="p">(</span><span class="n">nconditions</span><span class="p">)</span>
</pre></div>
</div>
<p>Contrasts sum to zero, are orthogonal, and positive values and negative
values each sum to 1.</p>
<p>This is useful in evaluating designs, and for F-tests across all pairwise
differences (e.g., one-way ANOVA contrast).</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.dice_coeff_image">
<code class="descclassname">Statistics_tools.</code><code class="descname">dice_coeff_image</code><span class="sig-paren">(</span><em>imagelist</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.dice_coeff_image" title="Permalink to this definition">¶</a></dt>
<dd><p>dice coefficient for the overlap between each pair of a set of images</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>dice_coeff = dice_coeff_image([imagelist or fmri_data object])
</pre></div>
</div>
<p>tor wager, June 2010
..</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.doquality">
<code class="descclassname">Statistics_tools.</code><code class="descname">doquality</code><span class="sig-paren">(</span><em>Xcx</em>, <em>X</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.doquality" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">equality</span><span class="p">]</span> <span class="o">=</span> <span class="n">doquality</span><span class="p">(</span><span class="n">Xcx</span><span class="p">,</span><span class="n">X</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Xcx:</strong></dt>
<dd><p class="first last">binary indicator matrix of cluster assignments,</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first">stimulus coordinates in group space</p>
<p class="last">also: takes group spaces with zeros;</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.fisherz">
<code class="descclassname">Statistics_tools.</code><code class="descname">fisherz</code><span class="sig-paren">(</span><em>r</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.fisherz" title="Permalink to this definition">¶</a></dt>
<dd><p>Fisher&#8217;s r to z&#8217; transform, and the inverse</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><blockquote class="first">
<div><dl class="docutils">
<dt><strong>z:</strong></dt>
<dd><p class="first last">z = z&#8217;, treating input r as correlation</p>
</dd>
</dl>
</div></blockquote>
<dl class="last docutils">
<dt><strong>r:</strong></dt>
<dd><p class="first last">treating input r as a z&#8217; score</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.fit_gls">
<code class="descclassname">Statistics_tools.</code><code class="descname">fit_gls</code><span class="sig-paren">(</span><em>y</em>, <em>X</em>, <em>c</em>, <em>p</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.fit_gls" title="Permalink to this definition">¶</a></dt>
<dd><p>Fit a linear model using generalized least squares and an AR(p) model</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[beta, t, pvals, convals, con_t, con_pvals, sigma, Phi, df, stebeta, conste, F] = fit_gls(y,X,c,p,[PX, equal to pinv(X), for speed], [Weights])
</pre></div>
</div>
<p>This program uses the Cochrane-Orcutt algorithm to iteratively find the
GLS solution and estimate the noise parameters.</p>
<p>Step 1: Find the OLS solution.</p>
<p>Step 2: Use residuals from the previous fit to estimate the parameters in
the AR(p) noise model.</p>
<p>Step 3: Find the GLS solution using the covariance matrix corresponding
to an AR(p) model with parameters estimated in Step 2 inserted.</p>
<p>Step 4: Repeat steps 2-3 until convergence.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>y:</strong></dt>
<dd><p class="first last">fMRI time course (T x 1 vector)</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">Design matrix (T x param matrix)</p>
</dd>
<dt><strong>c:</strong></dt>
<dd><p class="first last">contrast vector(s) (param x # contrasts matrix)</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">order of AR model.</p>
</dd>
<dt><strong>PX:</strong></dt>
<dd><p class="first">pinv(X), for speeded, repeated calculations with different y vectors</p>
<dl class="last docutils">
<dt>Note: if using weights, px = inv(X&#8217; * W * X) * X&#8217; * W; </dt>
<dd><p class="first last">where W = diag(Weights);</p>
</dd>
</dl>
</dd>
<dt><strong>Weights:</strong></dt>
<dd><p class="first">Optional, vector of weights for each observation</p>
<p class="last">Empty or missing: Unweighted analysis.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Note that setting p=0 implies a white noise model.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>t:</strong></dt>
<dd><p class="first last">t-value for the contrast c&#8217;beta</p>
</dd>
<dt><strong>df:</strong></dt>
<dd><p class="first last">degrees of freedom using Satterthwaite approximation</p>
</dd>
<dt><strong>beta:</strong></dt>
<dd><p class="first last">beta vector</p>
</dd>
<dt><strong>Phi:</strong></dt>
<dd><p class="first last">vector of coefficients in AR(p) model</p>
</dd>
<dt><strong>sigma:</strong></dt>
<dd><p class="first last">standard deviation</p>
</dd>
<dt><strong>stebeta:</strong></dt>
<dd><p class="first last">standard error of betas</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.fit_gls_brain">
<code class="descclassname">Statistics_tools.</code><code class="descname">fit_gls_brain</code><span class="sig-paren">(</span><em>imgs</em>, <em>X</em>, <em>arorder</em>, <em>conditionnames</em>, <em>maskimg</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.fit_gls_brain" title="Permalink to this definition">¶</a></dt>
<dd><p>Run Generalized Least Squares model with AR-model autoregression (optional)
at each voxel in a set of images.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fit_gls_brain</span><span class="p">(</span><span class="n">imgs</span><span class="p">,</span> <span class="n">X</span><span class="p">,</span> <span class="n">arorder</span><span class="p">,</span> <span class="n">conditionnames</span><span class="p">,</span> <span class="n">maskimg</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;contrasts&#39;</span><span class="p">,</span> <span class="n">contrast_mtx</span><span class="p">,</span> <span class="s">&#39;contrastnames&#39;</span><span class="p">,</span> <span class="n">contrastnames</span><span class="p">],</span> <span class="p">[</span><span class="s">&#39;weights&#39;</span><span class="p">,</span> <span class="n">weights</span><span class="p">])</span>
</pre></div>
</div>
<p>Single trial model to get trial amplitude, width, AUC, etc.
Take one of those (e.g., AUC) and get a list of trial images
Pass that into this function.</p>
<p>fit_gls_brain(trial_amp_imgs, eventdesign{1}, contrasts</p>
<p>You must add the intercept to X yourself if you want one!</p>
<p>contrasts can be empty, and if it is, this function will write images
for statistics on individual predictors.  If you enter contrast values,
then it will write images for contrasts instead.</p>
<p>conditionnames = column or contrast image names, in cell array</p>
<p>e.g. eventnames{1} = {&#8216;high&#8217;    &#8216;medium&#8217;    &#8216;low&#8217;    &#8216;warm&#8217;}</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>This one looks &#64; significance for betas in X model:</p>
<div class="highlight-python"><div class="highlight"><pre>load ../Multilev_mediation-try4(resliced)_10k/mediation_SETUP.mat
imgs = SETUP.data.M{1};
X = eventdesign{1};
arorder = 1;
contrasts = [];
conditionnames = eventnames{1};
maskimg = spm_select(1); % try gray matter mask...
fit_gls_brain(imgs, X, arorder, contrasts, conditionnames, maskimg)
</pre></div>
</div>
<p>Now define contrasts and re-run on contrast values:</p>
<div class="highlight-python"><div class="highlight"><pre>contrasts = [3 1 -1 -3; .25 .25 .25 .25; 1 0 0 -1]&#39;
conditionnames = {&#39;Linearpain&#39; &#39;Average_resp&#39; &#39;High-Low&#39;};
fit_gls_brain(imgs, X, arorder, contrasts, conditionnames, maskimg)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.getmeanquality">
<code class="descclassname">Statistics_tools.</code><code class="descname">getmeanquality</code><span class="sig-paren">(</span><em>X</em>, <em>c</em>, <em>linkagetype</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.getmeanquality" title="Permalink to this definition">¶</a></dt>
<dd><p>point estimate cluster solution</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.glmfit_general">
<code class="descclassname">Statistics_tools.</code><code class="descname">glmfit_general</code><span class="sig-paren">(</span><em>Y</em>, <em>X</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.glmfit_general" title="Permalink to this definition">¶</a></dt>
<dd><p>This function was designed to run a second-level GLS analysis on
coefficients and variance estimates from a 1st-level model, but it may be
used generally to implement weighted GLS with options to bootstrap or run
a sign permutation test to get p-values.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first">data for n subjects on k variables</p>
<p>run tests on each column of y independently</p>
<p class="last">if this is a 2nd-level analysis in a two-level model, Y is coefficients from 1st level (mu-hat)</p>
</dd>
<dt><strong>s2:</strong></dt>
<dd><p class="first last">estimates of variance (sigma-squared??) for k variables</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">design matrix; include intercept (first) + predictors</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Estimation method:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><ul class="first last simple">
<li>&#8216;unweighted&#8217; / &#8216;weighted&#8217;</li>
<li>&#8216;robust&#8217; / &#8216;nonrobust&#8217;</li>
<li>&#8216;s2&#8217;</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>NEW: followed by n-length cell array of var/cov mtx
(xtxi*sigma2) estimates from previous level</p>
<p>OLD: followed by n x k matrix of first-level variance estimates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Inference method:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><ul class="first simple">
<li>&#8216;t-test&#8217;</li>
<li>&#8216;bootstrap&#8217;</li>
<li>&#8216;signperm&#8217;</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;names&#8217;:</strong></dt>
<dd><p class="first last">names of each col. of y:  string followed by input</p>
</dd>
</dl>
<p><a href="#id43"><span class="problematic" id="id44">**</span></a>&#8216;name&#8217;  : text string tag for analysis</p>
<dl class="docutils">
<dt><strong>&#8216;verbose&#8217;:</strong></dt>
<dd><p class="first last">print verbose output and tables</p>
</dd>
<dt><strong>&#8216;nresample&#8217;:</strong></dt>
<dd><p class="first last">number of boot/sign perm samples to run; string followed by input</p>
</dd>
<dt><strong>&#8216;noint&#8217;:</strong></dt>
<dd><p class="first last">do not require intercept to be first column of X.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Notes:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Same as glmfit.m (2007a) without weights.&#8217;</p>
<p>With weights, we estimate w and use Satterthwaite, whereas glmfit uses
n - k for dfe (assumes correct weights are known.)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>stats.b_star:</strong></dt>
<dd><p class="first last">Weighted fixed effects estimate (if using &#8216;weighted&#8217; option)</p>
</dd>
<dt><strong>stats.Y_star:</strong></dt>
<dd><p class="first">Y_star&#8217;;  % Empirical Bayes estimates; don&#8217;t use for group
inference, but save;</p>
<p class="last">if this is run in a context of a multi-level model, these are the individual subjects&#8217;
(first-level) Empirical Bayes slope estimates</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Output to screen:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Coeff is &#8220;beta&#8221;, the effect magnitude
STE is the standard error, a function of variance and df</p>
<blockquote>
<div><ul class="simple">
<li>T = beta / STE</li>
<li>Z = beta / the STE for infinite df</li>
<li>p</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>Y = randn(100, 4); X = Y(:,1) + randn(100,1); X = [ones(size(X)) X];
stats = glmfit_general(Y, X, &#39;verbose&#39;);
stats
first_lev_var = rand(100, 4);

stats = glmfit_general(Y, X, &#39;weighted&#39;, first_lev_var, &#39;dfwithin&#39;, 20, ...
&#39;verbose&#39;, &#39;names&#39;, {&#39;DMPFC&#39; &#39;ACC&#39; &#39;VMPFC&#39; &#39;SII&#39;}, ...
&#39;beta_names&#39;, {&#39;Mean activity&#39; &#39;Rating covariate&#39;}, ...
&#39;analysisname&#39;,&#39;My Sample ROI analysis&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.glmfit_multilevel">
<code class="descclassname">Statistics_tools.</code><code class="descname">glmfit_multilevel</code><span class="sig-paren">(</span><em>Y</em>, <em>X1</em>, <em>X2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.glmfit_multilevel" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stats</span> <span class="o">=</span> <span class="n">glmfit_multilevel</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">X1</span><span class="p">,</span> <span class="n">X2</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">is data in cell array, one cell per subject.
Column vector of subject outcome data in each cell.</p>
</dd>
<dt><strong>X1 and X2:</strong></dt>
<dd><dl class="first last docutils">
<dt>are first and 2nd level design matrices</dt>
<dd><blockquote class="first">
<div><ul class="simple">
<li>X1 in cell array, one cell per subject
design matrix for each subject in each cell.  
<em>columns must code for the same variable for all subjects</em></li>
</ul>
</div></blockquote>
<ul class="last simple">
<li>X2 in rect. matrix</li>
<li>can be empty (intercept only)</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>E.g., with one 2nd-level predictor:
stats = glmfit_multilevel(Y, X1, X2, ...
&#8216;names&#8217;, {&#8216;Int&#8217; &#8216;Temp&#8217;}, &#8216;beta_names&#8217;, {&#8216;2nd-level Intercept (overall group effect)&#8217; &#8216;2nd-lev predictor: Group membership&#8217;});</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>stats:</strong></dt>
<dd><p class="first last">is structure with results.
Intercept is always added as first column!
(do not add intercept to input predictors)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>See glmfit_general.m for varargin variable input options.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>len = 200; sub = 20;
x = zeros(len,sub);
x(11:20,:) = 2;                   % create signal
x(111:120,:) = 2;
c = normrnd(0.5,0.1,sub,1);       % slope between-subjects variations
d = normrnd(3,0.2,sub,1);         % intercept between-subjects variations
% Create y: Add between-subjects error (random effects) and measurement noise
% (within-subjects error)
for i=1:sub, y(:,i) = d(i) + c(i).*x(:,i) + normrnd(0,0.5,len,1);
end;

for i = 1:size(y, 2), YY{i} = y(:, i); end
for i = 1:size(y, 2), XX{i} = x(:, i); end

% one-sample t-test, weighted by inv of btwn + within vars
stats = glmfit_multilevel(YY, XX, [], &#39;verbose&#39;, &#39;weighted&#39;);

statsg = glmfit_multilevel(y, x, covti, &#39;names&#39;, {&#39;L1 Intercept&#39; &#39;L1 Slope&#39;},...
&#39;beta_names&#39;, {&#39;Group Average&#39;, &#39;L2_Covt&#39;});
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input Options:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>General Defaults</dt>
<dd><ul class="first last simple">
<li>case &#8216;names&#8217;, Names of first-level predictors, starting
with &#8216;Intercept&#8217;, in cell array</li>
<li>case &#8216;analysisname&#8217;, analysisname = varargin{i+1}; varargin{i+1} = [];</li>
<li>case &#8216;beta_names&#8217;, beta_names = Names of 2nd-level predictors, starting
with &#8216;Intercept&#8217;, in cell array</li>
</ul>
</dd>
<dt>Estimation Defaults</dt>
<dd><ul class="first last simple">
<li>case &#8216;robust&#8217;, robust_option = &#8216;yes&#8217;;</li>
<li>case {&#8216;weight&#8217;, &#8216;weighted&#8217;, &#8216;var&#8217;, &#8216;s2&#8217;}, weight_option = &#8216;weighted&#8217;;</li>
</ul>
</dd>
<dt>Inference defaults</dt>
<dd><ul class="first last simple">
<li>case {&#8216;boot1&#8217;, &#8216;boot&#8217;, &#8216;bootstrap&#8217;}, inference_option = &#8216;bootstrap&#8217;;</li>
<li>case {&#8216;sign perm&#8217;, &#8216;signperm&#8217;, &#8216;sign&#8217;}, inference_option = &#8216;signperm&#8217;;</li>
<li>case {&#8216;t-test&#8217;, &#8216;ttest&#8217;}, inference_option = &#8216;t-test&#8217;;</li>
</ul>
</dd>
<dt>Display control defaults</dt>
<dd><ul class="first last simple">
<li>case &#8216;plots&#8217;, doplots = 1; plotstr = &#8216;plots&#8217;;</li>
<li>case &#8216;noplots&#8217;, doplots = 0; plotstr = &#8216;noplots&#8217;;</li>
<li>case {&#8216;dosave&#8217;, &#8216;save&#8217;, &#8216;saveplots&#8217;}, dosave = 1; savestr = &#8216;save&#8217;;</li>
<li>case &#8216;verbose&#8217;, verbose = 1; verbstr = &#8216;verbose&#8217;;</li>
<li>case &#8216;noverbose&#8217;, verbose = 0; verbstr = &#8216;noverbose&#8217;;</li>
<li>case {&#8216;savefile&#8217;, &#8216;savefilename&#8217;}, savefilename = varargin{i + 1}; varargin{i+1} = [];</li>
</ul>
</dd>
<dt>Bootstrap defaults</dt>
<dd><ul class="first last simple">
<li>case &#8216;nresample&#8217;, nresample = varargin{i+1};</li>
<li>case {&#8216;pvals&#8217;, &#8216;whpvals_for_boot&#8217;}, whpvals_for_boot = varargin{i+1};</li>
</ul>
</dd>
<dt>Sign perm defaults</dt>
<dd><ul class="first last simple">
<li>case {&#8216;permsign&#8217;}, permsign = varargin{i+1};</li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.glmfit_multilevel_varexplained">
<code class="descclassname">Statistics_tools.</code><code class="descname">glmfit_multilevel_varexplained</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em>, <em>b</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.glmfit_multilevel_varexplained" title="Permalink to this definition">¶</a></dt>
<dd><p>Variance explained estimates, table, and pie chart for multi-level linear model</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">vardecomp</span> <span class="o">=</span>  <span class="n">glmfit_multilevel_varexplained</span><span class="p">(</span><span class="n">XX</span><span class="p">,</span><span class="n">YY</span><span class="p">,</span> <span class="n">b</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;noplots&#39;</span> <span class="s">&#39;notable&#39;</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>b:</strong></dt>
<dd><p class="first last">is a column vector of beta weights for fixed effects (i.e., returned by glmfit_multilevel: stats.beta&#8217;)</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">is a cell vector containing the design matrix for each subject, each subject in a cell. 
<em>columns must code for the same variable for all subjects</em></p>
</dd>
<dt><strong>Y:</strong></dt>
<dd><p class="first last">is the data vector for each subject in each cell</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;noplots&#8217;, &#8216;notable&#8217;:</strong></dt>
<dd><p class="first last">suppress output</p>
</dd>
<dt><strong>&#8216;colors&#8217;:</strong></dt>
<dd><p class="first last">followed by n x 3 vector of rgb colors
other inputs to wani_pie OK too.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>vardecomp:</strong></dt>
<dd><p class="first last">structure of variance explained and description</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">See also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>glmfit_multilevel, igls_multicond, glmfit_general, wani_pie</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.intercept">
<code class="descclassname">Statistics_tools.</code><code class="descname">intercept</code><span class="sig-paren">(</span><em>X</em>, <em>meth</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.intercept" title="Permalink to this definition">¶</a></dt>
<dd><p>Intercept-related functions for working with design matrices, etc.</p>
<p>Return which columns are intercept(s), if any</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">wh</span> <span class="o">=</span> <span class="n">intercept</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="s">&#39;which&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Remove an intercept, if there is one</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">X</span> <span class="o">=</span> <span class="n">intercept</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="s">&#39;remove&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>Add an intercept to the end</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">X</span> <span class="o">=</span> <span class="n">intercept</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="s">&#39;add&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>Ensure that the intercept is at the end, moving or adding it as necessary</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">X</span> <span class="o">=</span> <span class="n">intercept</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="s">&#39;end&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.loess_multilevel">
<code class="descclassname">Statistics_tools.</code><code class="descname">loess_multilevel</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.loess_multilevel" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stats</span> <span class="o">=</span> <span class="n">loess_multilevel</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<dl class="docutils">
<dt>Options:</dt>
<dd><ul class="first last simple">
<li>case &#8216;fit_and_average&#8217;, meth = &#8216;fit_and_average&#8217;;</li>
<li>case &#8216;regularization&#8217;, regularization = varargin{i+1};</li>
<li>case &#8216;order&#8217;, loessorder = varargin{i+1};</li>
<li>case {&#8216;robust&#8217;, &#8216;dorobust&#8217;}, dorobust = varargin{i+1};</li>
<li>case {&#8216;nboot&#8217;, &#8216;bootsamples&#8217;}, nboot = varargin{i+1};</li>
<li>case &#8216;plot&#8217;, doplot = 1;</li>
<li>case {&#8216;plotall&#8217;}, plotsummary = 0; doplot = 1;</li>
<li>case &#8216;color&#8217;, color = varargin{i+1};</li>
<li>case {&#8216;existingfig&#8217;, &#8216;samefig&#8217;}, newfig = 0;</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>stats = loess_multilevel(models{i}.X, models{i}.Y, &#39;regularization&#39;, .8, &#39;order&#39;, 1, &#39;color&#39;, [.8 .3 0], &#39;plot&#39;, &#39;samefig&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">scn_stats_helper_functions</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.matrix_direct_effects">
<code class="descclassname">Statistics_tools.</code><code class="descname">matrix_direct_effects</code><span class="sig-paren">(</span><em>sig</em>, <em>data</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.matrix_direct_effects" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">direct_mtx</span><span class="p">,</span> <span class="n">mediated_mtx</span><span class="p">,</span> <span class="n">mediators</span><span class="p">]</span> <span class="o">=</span> <span class="n">matrix_direct_effects</span><span class="p">(</span><span class="n">sig</span><span class="p">,</span> <span class="n">data</span><span class="p">);</span>
</pre></div>
</div>
<p>Take an n x n matrix of significant correlations and the data for each
variable. Test whether each link is completely mediated by another
variable. Prune the correlation significance matrix, also noting which
variables are mediators.</p>
<p>Example:
Take output from cluster_nmds (c structure) and prune to return direct
links only:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">c</span><span class="o">.</span><span class="n">direct_mtx</span><span class="p">,</span> <span class="n">c</span><span class="o">.</span><span class="n">mediated_mtx</span><span class="p">,</span> <span class="n">c</span><span class="o">.</span><span class="n">mediators</span><span class="p">]</span> <span class="o">=</span> <span class="n">matrix_direct_effects</span><span class="p">(</span><span class="n">c</span><span class="o">.</span><span class="n">STATS</span><span class="o">.</span><span class="n">sigmat</span><span class="p">,</span> <span class="n">c</span><span class="o">.</span><span class="n">dat</span><span class="p">);</span>
<span class="n">nmdsfig_tools</span><span class="p">(</span><span class="s">&#39;removelines&#39;</span><span class="p">);</span>
<span class="n">nmdsfig_tools</span><span class="p">(</span><span class="s">&#39;drawlines&#39;</span><span class="p">,</span><span class="n">c</span><span class="o">.</span><span class="n">GroupSpace</span><span class="p">,</span> <span class="n">c</span><span class="o">.</span><span class="n">direct_mtx</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.matrix_direct_effects_ridge">
<code class="descclassname">Statistics_tools.</code><code class="descname">matrix_direct_effects_ridge</code><span class="sig-paren">(</span><em>data</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.matrix_direct_effects_ridge" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[out] = matrix_direct_effects_ridge(data, [optional args]);
</pre></div>
</div>
<p>Take an n x v matrix of data for each of N subjects.
data are entered in a square matrix within each cell of data,
data{1}, {2}, etc.</p>
<p>Uses ridge regression to assess linear slopes for each variable on
each other one.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;k&#8217;:</strong></dt>
<dd><p class="first last">followed by ridge parameter value.  Default = 0.</p>
</dd>
<dt><strong>&#8216;scale&#8217; or &#8216;zscore&#8217;:</strong></dt>
<dd><p class="first last">which z-scores within-subjects</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Future: Consider weighting based on multicolinearity</p>
<p>Note: Matrix is asymmetrical! Rows predicting columns...</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Take mediation brain results clusters cell clpos_data{...}
Concatenate data matrix and run to get connections.</p>
<div class="highlight-python"><div class="highlight"><pre>for i = 1:length(clpos_data), data{i} = cat(2, clpos_data{i}.timeseries); end
out = matrix_direct_effects_ridge(data);
[out.GroupSpace,out.obs_dist,out.implied_dissim] = shepardplot(out.dissim,[]);
create_figure(&#39;nmdsfig&#39;);
nmdsfig(out.GroupSpace,&#39;classes&#39;,ones(out.k, 1),&#39;names&#39;,[],&#39;sig&#39;,out.fdrsig, &#39;legend&#39;,{&#39;Pos&#39; &#39;Neg&#39;},&#39;sizescale&#39;,[4 16]);
</pre></div>
</div>
<p>Example 2:
Include mediation predictor and outcome</p>
<div class="highlight-python"><div class="highlight"><pre>for i = 1:length(clpos_data), data{i} = [SETUP.data.X{i} SETUP.data.Y{i} data{i}]; end
out = matrix_direct_effects_ridge(data);
</pre></div>
</div>
<p>Example of MDS and plotting direct relationships:</p>
<div class="highlight-python"><div class="highlight"><pre>OUT.ridge = matrix_direct_effects_ridge(data);
D = OUT.ridge.mean; D(find(eye(size(D)))) = 1;
D = (D&#39; + D) ./ 2;
OUT.ridge.D = (1 - D) ./ 2;
[OUT.stats_mds.GroupSpace,OUT.stats_mds.obs,OUT.stats_mds.implied_dissim] = shepardplot(OUT.ridge.D,[]);
OUT.stats_mds = nmdsfig_tools(&#39;cluster_solution&#39;,OUT.stats_mds, OUT.stats_mds.GroupSpace, 2:10, 1000, []);
OUT.stats_mds.colors = {&#39;ro&#39; &#39;go&#39; &#39;bo&#39; &#39;yo&#39; &#39;co&#39; &#39;mo&#39; &#39;ko&#39; &#39;r^&#39; &#39;g^&#39; &#39;b^&#39; &#39;y^&#39; &#39;c^&#39; &#39;m^&#39; &#39;k^&#39;};
create_figure(&#39;nmdsfig&#39;);
nmdsfig(OUT.stats_mds.GroupSpace,&#39;classes&#39;,OUT.stats_mds.ClusterSolution.classes,&#39;names&#39;,OUT.stats_mds.names,&#39;sig&#39;,OUT.ridge.fdrsig);
hh = nmdsfig_fill(OUT.stats_mds);
</pre></div>
</div>
<p>Prediction using multiple regions:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stats</span> <span class="o">=</span> <span class="n">rsquare_multiple_regions_multilevel</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">X</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Visualization of networks:</p>
<div class="highlight-python"><div class="highlight"><pre>classes = OUT.stats_mds.ClusterSolution.classes;
% create_figure(&#39;Surfaces&#39;);
% shan = addbrain;
shan = mediation_brain_surface_figs({}, {});
scolors = {[1 0 0] [0 1 0] [0 0 1] [1 1 0] [0 1 1] [1 0 1] [0 0 0] [1 0 0] [0 1 0] [0 0 1] [1 1 0] [0 1 1] [1 0 1] [0 0 0] };
for j = 1:max(classes)
wh = (classes == j); sum(wh)
montage_clusters([], cl(wh), scolors(j))
% create_figure(&#39;Surfaces&#39;, 1, 1, 1);
cluster_surf(cl(wh), 5, scolors(j), shan);
end
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">xcorr_multisubject.m</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.matrix_eval_function">
<code class="descclassname">Statistics_tools.</code><code class="descname">matrix_eval_function</code><span class="sig-paren">(</span><em>Y</em>, <em>fhandle</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.matrix_eval_function" title="Permalink to this definition">¶</a></dt>
<dd><p>Evaluate any arbitrary function fhandle on each column of an input matrix Y</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">varargout</span> <span class="o">=</span> <span class="n">matrix_eval_function</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span><span class="n">fhandle</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>evaluate fhandle on paired columns of X and Y</p>
<p>fhandle is a function handle:</p>
<div class="highlight-python"><div class="highlight"><pre>fhandle = @(variable inputs) fit_gls(variable_input,fixed_inputs);
fhandle = @(y) fit_gls(y,X,c,p,PX);
</pre></div>
</div>
<p>Function is fhandle should return (multiple possible) outputs, each
in a row vector.  Matrix outputs of this function are Y-cols x output
values for each output argument.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body">Generalized least squares fitting on 100 Y-variables, same X</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>y = rand(100,100); X = y + rand(100,1); X(:,end+1) = 1; c = [1 0]&#39;; p = 2; PX = pinv(X);
fhandle = @(y) fit_gls(y,X,c,p,PX);
[t, df, beta, Phi, sigma,stebeta, F] = fhandle(y);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.monotonic_regression">
<code class="descclassname">Statistics_tools.</code><code class="descname">monotonic_regression</code><span class="sig-paren">(</span><em>x</em>, <em>y</em>, <em>doplot</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.monotonic_regression" title="Permalink to this definition">¶</a></dt>
<dd><p>See lsqisotonic.m</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.moving_average">
<code class="descclassname">Statistics_tools.</code><code class="descname">moving_average</code><span class="sig-paren">(</span><em>meth</em>, <em>data</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.moving_average" title="Permalink to this definition">¶</a></dt>
<dd><p>Symmetrical moving average filters
(matlab&#8217;s internal moving avg filter functions are asymmetric)</p>
<p>Works on each column of the data input matrix</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>y = moving_average(&#39;gaussian&#39;,data,fwhm)

% fwhm is not actually fhwm now...it&#39;s related to width though (temporary)
y = moving_average(&#39;gaussian&#39;,data,20);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.noise_arp">
<code class="descclassname">Statistics_tools.</code><code class="descname">noise_arp</code><span class="sig-paren">(</span><em>n</em>, <em>phi</em>, <em>sigma</em>, <em>e</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.noise_arp" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">w</span> <span class="o">=</span> <span class="n">noise_arp</span><span class="p">(</span><span class="n">n</span><span class="p">,</span> <span class="n">phi</span><span class="p">)</span>
</pre></div>
</div>
<p>Generate n-length AR(p) noise vector w
given phi (default = [.5 .1])</p>
<p>Based on Gaussian noise process e with standard deviation sigma (default = 1)</p>
<p>w will have a variance greater than that of the underlyling gaussian
process e</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.nonlin_fit">
<code class="descclassname">Statistics_tools.</code><code class="descname">nonlin_fit</code><span class="sig-paren">(</span><em>y</em>, <em>x</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.nonlin_fit" title="Permalink to this definition">¶</a></dt>
<dd><p>Multi-purpose nonlinear fitting to data</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[p,errval,fit] = nonlin_fit(y,x,[&#39;link&#39;,linktype],[&#39;err&#39;,objtype],[&#39;start&#39;,startp], ...
                           [&#39;plot&#39;],[&#39;verbose&#39;],[noverbose&#39;],[&#39;quickstart&#39;],[&#39;smartstart&#39;])
</pre></div>
</div>
<p>(Optional arguments are in [ ])</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>y:</strong></dt>
<dd><p class="first last">data vector</p>
</dd>
<dt><strong>x:</strong></dt>
<dd><p class="first last">vector of x-values</p>
</dd>
<dt><strong>linktype:</strong></dt>
<dd><p class="first">functional form to fit to data.  options are:</p>
<p class="last">&#8216;sigmoid&#8217; &#8216;exp0&#8217; &#8216;exp2&#8217; &#8216;exp3&#8217; &#8216;exps&#8217;</p>
</dd>
<dt><strong>objtype:</strong></dt>
<dd><p class="first">error measures to be minimized.  options are:</p>
<p>&#8216;sse&#8217; : sums of squared errors</p>
<p class="last">&#8216;mad&#8217; : median absolute deviation (more robust to outliers)</p>
</dd>
<dt><strong>&#8216;plot&#8217;:</strong></dt>
<dd><p class="first last">Create plot of data and fit</p>
</dd>
<dt><strong>startp:</strong></dt>
<dd><p class="first last">Vector of starting parameters for minimization</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Functions:</th><td class="field-body"><p class="first"><strong>&#8216;sigmoid&#8217;:</strong></p>
<dl class="docutils">
<dt><strong>&#8216;exp1&#8217;, &#8216;exp2&#8217;, &#8216;exp3&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>Exponential function</dt>
<dd><ul class="first last simple">
<li>&#8216;exp1&#8217;        y = e^(ax)          with free parameter a</li>
<li>&#8216;exp2&#8217;        y = b*e^(ax)        with free a, b</li>
<li>&#8216;exp3&#8217;        y = c + b*e^(ax)    with free a, b, c</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;pow0&#8217; &#8216;pow2&#8217; &#8216;pow3&#8217; &#8216;pows1&#8217; &#8216;pows2&#8217; &#8216;pows&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>Power functions: </dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>&#8216;pows&#8217;        a + -(bx)^(-c)     with params a, b, c</dt>
<dd><p class="first last">simplifies to 1 - (1/x) with all p = 1
range: (for + b and c) -Inf at x = 0, to asymptote = a at x = Inf.</p>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Default behavior: sigmoid fit, SSE objective:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">p</span><span class="p">,</span><span class="n">sse</span><span class="p">,</span><span class="n">fit</span><span class="p">]</span> <span class="o">=</span> <span class="n">nonlin_fit</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="n">x</span><span class="p">,</span><span class="s">&#39;plot&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>Specify sigmoid function and SSE objective, starting at params [2 1 1]:</p>
<div class="highlight-python"><div class="highlight"><pre>[p,sse,fit] = nonlin_fit(y,x,&#39;start&#39;,[2 1 1],&#39;link&#39;,&#39;sigmoid&#39;,&#39;err&#39;,&#39;sse&#39;);
</pre></div>
</div>
<p>Specify median absolute deviation error minimization (more robust):</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">p</span><span class="p">,</span><span class="n">sse</span><span class="p">,</span><span class="n">fit</span><span class="p">]</span> <span class="o">=</span> <span class="n">nonlin_fit</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="n">x</span><span class="p">,</span><span class="s">&#39;err&#39;</span><span class="p">,</span><span class="s">&#39;mad&#39;</span><span class="p">,</span><span class="s">&#39;plot&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>Fit and add a fit line on a graph</p>
<div class="highlight-python"><div class="highlight"><pre>[p,errval,fit,linkfun,fhan] = nonlin_fit(y,x,&#39;linktype&#39;,&#39;exps&#39;,&#39;start&#39;,[1.2 .5 .8]);
fitx = 1:.1:5; fitline = fhan(p,fitx);
hold on; plot(fitx,fitline,&#39;r&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% sigmoid fit to y
% Generate fake data:
x = -5:.01:5;
sigmoid = inline(&#39;p(1) .* ( 1 ./ (1 + p(2)*exp(-p(3)*x)) )&#39;,&#39;p&#39;,&#39;x&#39;);
y = sigmoid([2 1 1.5],x);
y = y + .3 .* randn(1,size(y,2)) ;

% Fit, starting at param values [1 1 1]:
[p,sse,fit] = nonlin_fit(y,x,&#39;plot&#39;,&#39;start&#39;,[1 1 1]);

% Bit more complicated data
x2 = -5:.01:5;
y = sigmoid([2 1 .5],x2);
y = [y 2*ones(1,100)];
y = y + .3 .* randn(1,size(y,2)) ;
x = 1:length(y);

% Exponential function (see also exp1, exp2, exp3 keywords)
expfun = inline(&#39;p(1)*exp(p(2)*x)&#39;,&#39;p&#39;,&#39;x&#39;);

% simulated sample data
x = -1:.01:5;
y = expfun([.5 .7], x);                               
y = y + 1 .* randn(1,size(y,2));

% function handle to pass in
funhandle = @(p, x) expfun(p, x);
[p,sse,fit] = nonlin_fit(y,x,&#39;linktype&#39;, funhandle, &#39;plot&#39;,&#39;start&#39;,[1 1]);

% Then, try to fit one-parameter exponential model to the same data:
[p,sse,fit] = nonlin_fit(y,x,&#39;linktype&#39;, &#39;exp1&#39;, &#39;plot&#39;,&#39;start&#39;, 1);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.nonlin_param_mod_brain">
<code class="descclassname">Statistics_tools.</code><code class="descname">nonlin_param_mod_brain</code><span class="sig-paren">(</span><em>ons</em>, <em>modulator</em>, <em>image_names</em>, <em>SETUP</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.nonlin_param_mod_brain" title="Permalink to this definition">¶</a></dt>
<dd><p>Nonlinear fits with a parametric modulator on a set of brain images</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>nonlin_param_mod_brain(X, image_names, SETUP, [SETUPional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>ons:</strong></dt>
<dd><p class="first last">onsets for each condition; one cell per
condition, one col. vector per series of onsets</p>
</dd>
<dt><strong>modulator:</strong></dt>
<dd><p class="first last">modulator values for each condition; same
format as above</p>
</dd>
<dt><strong>image_names:</strong></dt>
<dd><p class="first last">outcome variable; Images (volume names) for each subject, in
string matrix (list of image names); 3-D for now!</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">SETUP.(fields):</th><td class="field-body"><dl class="first last docutils">
<dt><strong>.mask:</strong></dt>
<dd><p class="first last">name of mask image</p>
</dd>
<dt><strong>.preprocX:</strong></dt>
<dd><p class="first last">flag for whether to HP filter X data</p>
</dd>
<dt><strong>.preprocY:</strong></dt>
<dd><p class="first last">flag for whether to HP filter Y data</p>
</dd>
<dt><strong>&#8216;nopreproc&#8217;:</strong></dt>
<dd><p class="first last">Turn off preproc</p>
</dd>
<dt><strong>.TR:</strong></dt>
<dd><p class="first last">repetition time of volume (image) acquisition</p>
</dd>
<dt><strong>.HPlength:</strong></dt>
<dd><p class="first last">high-pass filter length, in s</p>
</dd>
<dt><strong>.scans_per_session:</strong></dt>
<dd><p class="first last">vector of # volumes in each run, e.g., [128 128 128 128 128]</p>
</dd>
<dt><strong>.dummyscans:</strong></dt>
<dd><p class="first last">indices of images in each run that will be modeled
with separate dummy variables</p>
</dd>
<dt><strong>.startslice:</strong></dt>
<dd><p class="first last">starting slice number (to resume analysis)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>SETUPional inputs:
Any of the SETUPions in mediation.m</p>
<p>Also: &#8216;nopreproc&#8217; to skip preprocessing (i.e., for trial-level inputs)</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.nonlin_param_modulator">
<code class="descclassname">Statistics_tools.</code><code class="descname">nonlin_param_modulator</code><span class="sig-paren">(</span><em>y</em>, <em>ons</em>, <em>modulator_centered</em>, <em>tr</em>, <em>xvals</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.nonlin_param_modulator" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[h_mean h_by_mod d_mean d_by_mod intcpt auc_mean_trial auc_by_modulator errval] =  nonlin_param_modulator(y, ons, modulator_centered, tr, xvals)
</pre></div>
</div>
<p>USE THIS with nonlin_param_mod_brain.m</p>
<p>This function takes data (y) and other things (onsets, etc.) and
produces parameter estimates for amplitude, duration, amp*modulator,
dur*modulator</p>
<p>See rt_fit_brain.m for more info, and for examples creating fit
plots, etc.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>  y = rand(360, 1);
  [h_mean h_by_mod d_mean d_by_mod intcpt auc_mean_trial auc_by_modulator errval] =  nonlin_param_modulator(y, ons, scale(RTs, 1), 2);
  nonlin_parammod_predfun(y, ons, scale(RTs), [h_mean h_by_mod d_mean d_by_mod intcpt], &#39;plot&#39;);
  hold on; plot(y, &#39;k&#39;)
  % THIS stuff is the same for each voxel
 RTcenter = RTs - mean(RTs);
 xvals = (1:length(y))&#39;;

% fitting function: times is a dummy var to get this to work with nonlin_fit
 fhan = @(p, times) nonlin_parammod_predfun(y,ons,RTcenter,p);

 fitting_fun = @(y) nonlin_fit(y, xvals, &#39;link&#39;, fhan, &#39;start&#39;,[1 1 1 1 mean(y)]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.nonlin_parammod_predfun">
<code class="descclassname">Statistics_tools.</code><code class="descname">nonlin_parammod_predfun</code><span class="sig-paren">(</span><em>y</em>, <em>ons</em>, <em>pm_vals</em>, <em>p</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.nonlin_parammod_predfun" title="Permalink to this definition">¶</a></dt>
<dd><p>Predict data given onsets, parametric modulator values, and parameter
estimates for effects of events and PMs on the magnitude and delay of an
estimated neural &#8216;boxcar.&#8217;</p>
<p>A standard HRF model is assumed, and the parameters are estimates of how
trials affect the &#8216;neural&#8217; stimulus function.</p>
<p>This function replaces nonlin_parammod_predfun.m, which is obsolete</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">yhat</span><span class="p">,</span><span class="n">yhi</span><span class="p">,</span><span class="n">loTime</span><span class="p">,</span><span class="n">highTime</span><span class="p">,</span><span class="n">x</span><span class="p">]</span> <span class="o">=</span> <span class="n">nonlin_parammod_predfun</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="n">ons</span><span class="p">,</span><span class="n">pm_vals</span><span class="p">,</span><span class="n">p</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>p:</strong></dt>
<dd><dl class="first last docutils">
<dt>params</dt>
<dd><ol class="first last arabic simple">
<li>mag. scale (amplitude)</li>
<li>mag. X RT slope (linear effect of RT)</li>
<li>duration intercept (neural epoch duration)</li>
<li>duration x RT slope (linear effect of RT on duration)</li>
<li>overall fitted response intercept</li>
</ol>
</dd>
</dl>
</dd>
<dt><strong>y:</strong></dt>
<dd><p class="first last">data. Can be dummy data, needs only be the correct size
This function does not actually fit the data, so y is only used
to get the correct vector size</p>
</dd>
<dt><strong>ons:</strong></dt>
<dd><p class="first last">onsets (in samples)</p>
</dd>
<dt><strong>pm_vals:</strong></dt>
<dd><p class="first last">modulator values, preferably centered</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">parameter values; see below</p>
</dd>
<dt><strong>varargin:</strong></dt>
<dd><p class="first last">optional inputs, see below</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Special cases of parameter sets:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>p = [1 0 1 0 0];  % &#8220;Impulse model&#8221; : fixed boxcar of 1 s</p>
<p>p = [1 0 3 0 0];  % &#8220;Epoch model&#8221; : fixed boxcar of 3 s</p>
<p>p = [1 1 1 0 0];  % &#8220;Parametric modulator&#8221; : Impulse height modulated by
RT (with centered pm_vals)</p>
<p>p = [1 0 0 1 0];  % &#8220;Variable epoch&#8221;: convolve RT with duration 
(with non-centered, raw pm_vals).  Parametric modulation of duration</p>
<p>p = [1 1 1 1 0];  % combo of Parametric and Duration modulators of impulses</p>
<dl class="docutils">
<dt>p = [1 1.5 3 1.5 0];  % combo of Parametric and Duration modulators of</dt>
<dd>a 3-s epoch model</dd>
</dl>
<p>event signal magnitude = p1 * p2*RT           % was previously: RT^p2
event signal  duration = p3 + pm_vals * p4</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><ul class="first simple">
<li>case &#8216;random&#8217;, dorandom = 1;</li>
<li>case &#8216;plot&#8217;, doplot = 1;</li>
<li>case &#8216;hrf&#8217;, hrf = varargin{i + 1}; % HIGH_RES HRF</li>
<li>case &#8216;tr&#8217;, followed by TR</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Examples of generating predicted BOLD timeseries:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>y = rand(330,1);
ons = [1:20:320]&#39;; pm_vals = rand(size(ons));
[yhat,yhi] = nonlin_parammod_predfun(y,ons,pm_vals,[1 1 1 0 0],&#39;plot&#39;);  % linear RT modulation of height
[yhat,yhi] = nonlin_parammod_predfun(y,ons,pm_vals,[1 0 1 10 0],&#39;plot&#39;);  % linear RT modulation of duration

[yhat,yhi] = nonlin_parammod_predfun(y,ons,pm_vals,[.5 1 1 0 0],&#39;plot&#39;);  % other linear RT modulation of height
[yhat,yhi] = nonlin_parammod_predfun(y,ons,pm_vals,[.5 .7 1 5 0],&#39;plot&#39;);  % saturated RT modulation of height; linear width
</pre></div>
</div>
<p>Example of fitting observed timeseries and estimating parameters:</p>
<ul class="simple">
<li>SEE ALSO nonlin_param_modulator</li>
</ul>
<div class="highlight-python"><div class="highlight"><pre>RTcenter = pm_vals - mean(pm_vals);
true_p = [1 1.5 3 1.5 0];          % true parameters
xvals = (1:length(y))&#39;;
</pre></div>
</div>
<p>Fitting function: times is a dummy var to get this to work with nonlin_fit</p>
<div class="highlight-python"><div class="highlight"><pre>fhan = @(p, times) nonlin_parammod_predfun(y,ons,RTcenter,p);

y = fhan(true_p);                  % generate simulated &quot;true&quot; signal

fitting_fun = @(y) nonlin_fit(y, xvals, &#39;link&#39;, fhan, &#39;start&#39;,[1 1 1 1 .5]);
[p, errval, fit] = fitting_fun(y)     % get parameters for a timeseries of interest

[p,errval,fit] = nonlin_fit(y,(1:length(y))&#39;,&#39;link&#39;,fhan,&#39;start&#39;,[1 1 1 1 .5]);
hold on; plot(fit,&#39;r&#39;);
</pre></div>
</div>
<p>A second example using the genetic algorithm:</p>
<div class="highlight-python"><div class="highlight"><pre>objfun = @(p) sum(abs(y - fhan(p))); % objective function: absolute error
start = [-10 -10 0 -10 0]; start(:,:,2) = [10 10 10 10 2000];
[best_params,fit,beff,in] = tor_ga(200,50,{start},objfun,&#39;genconverge&#39;,5,&#39;noverbose&#39;);
%***note: does not work now, needs debugging***
</pre></div>
</div>
<p>Simulation: Test cov of param estimates</p>
<p>Run 1000 times to see cov. of param estimates
Right now: <a href="#id45"><span class="problematic" id="id46">**</span></a><a href="#id47"><span class="problematic" id="id48">**</span></a>development: p(1,2) are highly + corr, p(3:4) are high - corr
should probably choose one or the other for each.</p>
<div class="highlight-python"><div class="highlight"><pre>for i = 1:1000
   yi = y + randn(length(y),1) * 10; % new noise
   p(i,:) = fitting_fun(yi);
   if mod(i,10) == 0, fprintf(1,&#39; %3.0f&#39;,i); end
end
</pre></div>
</div>
<p>Note: Could create linear basis set of plausible forms</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.pairwise_diffs">
<code class="descclassname">Statistics_tools.</code><code class="descname">pairwise_diffs</code><span class="sig-paren">(</span><em>x</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.pairwise_diffs" title="Permalink to this definition">¶</a></dt>
<dd><p>Pairwise operations on columns of a matrix x, for each row</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>d = pairwise_diffs(x, [function handle])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>x:</strong></dt>
<dd><p class="first last">an n x k matrix</p>
</dd>
</dl>
<p>Optional: a function handle for the operation to perform on pairwise
elements of x(i, :)</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">if x is n x k, d is n x (k*(k-1)/2)</p>
<p>columns of x are arranged this way, e.g., with k = 5:</p>
<p>[x(1, 1) - x(1, 4)  x(1, 1) - x(1, 5)  x(1, 2) - x(1, 3) ...]</p>
<p>squareform(d(1, :)) is a matrix of pairwise diffs for d</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>x = magic(5)                  % generate data
d = pairwise_diffs(x);        % d = pairwise differences
row1 = squareform(d(1, :));   % square matrix of pairwise diffs for row 1

d = pairwise_diffs(x, @(a, b) a + b);  % return pairwise sum instead
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.partition_variables_indevel">
<code class="descclassname">Statistics_tools.</code><code class="descname">partition_variables_indevel</code><span class="sig-paren">(</span><em>x</em>, <em>k</em>, <em>obsk</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.partition_variables_indevel" title="Permalink to this definition">¶</a></dt>
<dd><p>partition n x v matrix into k classes</p>
<p>maximize within-class condition number or minimize cov. determinant</p>
<p>minimize between-class condition number or maximize cov. determinant...</p>
<p>Simulate two-class data</p>
<div class="highlight-python"><div class="highlight"><pre>nvars = 100; nsubj = 20; corval = .5;
S = eye(nvars./2) + corval*(1 - eye(nvars./2));
S = blkdiag(S,S); S(S==0) = corval;
x = mvnrnd(zeros(1,nvars), S, nsubj); det((corrcoef(x)))
figure; imagesc(corrcoef(x)); colorbar
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.permute_setupperms">
<code class="descclassname">Statistics_tools.</code><code class="descname">permute_setupperms</code><span class="sig-paren">(</span><em>n</em>, <em>nperms</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.permute_setupperms" title="Permalink to this definition">¶</a></dt>
<dd><p>Set up permutations x observations matrix of observation indices for
permutation test</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% approximate test with nperms obs.
permindx = permute_setupperms(n,nperms)
permindx = permute_setupperms(n,[])

% exact test (all permutations) of n observations
permindx = permute_setupperms(n)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.permute_signtest">
<code class="descclassname">Statistics_tools.</code><code class="descname">permute_signtest</code><span class="sig-paren">(</span><em>data</em>, <em>nperms</em>, <em>w</em>, <em>permsign</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.permute_signtest" title="Permalink to this definition">¶</a></dt>
<dd><p>One-sample t-test against zero on each column of data
Using weighted sign permutation test</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">p</span><span class="p">,</span> <span class="n">Z</span><span class="p">,</span> <span class="n">xbar</span><span class="p">,</span> <span class="n">permsign</span><span class="p">,</span> <span class="n">pmean</span><span class="p">]</span> <span class="o">=</span> <span class="n">permute_signtest</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">nperms</span><span class="p">,</span> <span class="p">[</span><span class="n">w</span><span class="p">],</span> <span class="p">[</span><span class="n">permsign</span><span class="p">])</span>
</pre></div>
</div>
<p>p-values are 2-tailed</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>data:</strong></dt>
<dd><p class="first last">n x k matrix of k vectors to test against zero</p>
</dd>
<dt><strong>nperms:</strong></dt>
<dd><p class="first last">number of permutations</p>
</dd>
<dt><strong>w:</strong></dt>
<dd><p class="first">n x k matrix of weights</p>
<p>weights for each column should sum to 1</p>
<p class="last">default (for empty or missing input) is equal weights, i.e., 1/n</p>
</dd>
<dt><strong>permindx:</strong></dt>
<dd><p class="first last">nperms x n matrix of exact permutation indices
default (for empty or missing input) generates nperms permutations</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"><p class="first last">Generate 5-vector dataset and test, first generating perms, then
using already-generated ones</p>
</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">tic</span><span class="p">,</span> <span class="p">[</span><span class="n">p</span><span class="p">,</span> <span class="n">z</span><span class="p">,</span> <span class="n">xbar</span><span class="p">,</span> <span class="n">permsign</span><span class="p">]</span> <span class="o">=</span> <span class="n">permute_signtest</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="mi">1000</span><span class="p">);</span> <span class="n">toc</span>
<span class="n">tic</span><span class="p">,</span> <span class="p">[</span><span class="n">p2</span><span class="p">,</span> <span class="n">z2</span><span class="p">,</span> <span class="n">xbar</span><span class="p">,</span> <span class="n">permsign</span><span class="p">]</span> <span class="o">=</span> <span class="n">permute_signtest</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="p">[],</span> <span class="p">[],</span> <span class="n">permsign</span><span class="p">);</span> <span class="n">toc</span>
</pre></div>
</div>
<p>Example: Generate fake data and simulate false positive rate</p>
<div class="highlight-python"><div class="highlight"><pre>permsign = [];  % initialize to empty for first iteration
nperms = 2000; nreps = 5000;
tic, for i = 1:nreps
x = randn(20,1); stat = mean(x);
[p(i), z(i), xbar, permsign] = permute_signtest(x, nperms, [], permsign);
if mod(i,10) == 0,fprintf(1,&#39;%03d &#39;,i); end
end
fprintf(1,&#39;\n&#39;);
</pre></div>
</div>
<p>Example: Generate fake data and simulate false positive rate
This uses the column-wise capabilities of this function and is much
faster
%</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">nperms</span> <span class="o">=</span> <span class="mi">2000</span><span class="p">;</span> <span class="n">nreps</span> <span class="o">=</span> <span class="mi">5000</span><span class="p">;</span> <span class="n">nsubj</span> <span class="o">=</span> <span class="mi">15</span><span class="p">;</span>
<span class="n">x</span> <span class="o">=</span> <span class="n">randn</span><span class="p">(</span><span class="n">nsubj</span><span class="p">,</span> <span class="n">nreps</span><span class="p">);</span>
<span class="p">[</span><span class="n">p</span><span class="p">,</span> <span class="n">z</span><span class="p">,</span> <span class="n">xbar</span><span class="p">,</span> <span class="n">permsign</span><span class="p">]</span> <span class="o">=</span> <span class="n">permute_signtest</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">nperms</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.plssquash">
<code class="descclassname">Statistics_tools.</code><code class="descname">plssquash</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.plssquash" title="Permalink to this definition">¶</a></dt>
<dd><p>Decomposes data X into K components that are ordered in their 
covariance with Y, designed to predict orthogonal parts of Y</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">V</span><span class="p">,</span> <span class="n">S</span><span class="p">,</span> <span class="n">varexp</span><span class="p">,</span> <span class="n">w</span><span class="p">,</span> <span class="n">Yhat</span><span class="p">]</span> <span class="o">=</span> <span class="n">plssquash</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><ul class="first simple">
<li>case {&#8216;noplot&#8217;}, turn off plotting</li>
<li>case &#8216;ndims&#8217;, save only first ndims (K) vectors</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>V:</strong></dt>
<dd><p class="first last">&#8216;eigenvetors&#8217;, or weights, on data (columns)</p>
</dd>
<dt><strong>S:</strong></dt>
<dd><p class="first last">score matrix, N x K</p>
</dd>
<dt><strong>varexp:</strong></dt>
<dd><p class="first last">sqrt(r-square) with first k components predicting Y</p>
</dd>
<dt><strong>w:</strong></dt>
<dd><p class="first last">V*b, integrated weights.  for predicting new data, pred = X*w</p>
</dd>
<dt><strong>Yhat:</strong></dt>
<dd><p class="first last">X*V*b, or S*b</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.princomp_largedata">
<code class="descclassname">Statistics_tools.</code><code class="descname">princomp_largedata</code><span class="sig-paren">(</span><em>x</em>, <em>econFlag</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.princomp_largedata" title="Permalink to this definition">¶</a></dt>
<dd><p>Principal Components Analysis</p>
<p>This is a version created for large data sets by Matthew Davidson</p>
<p>The default Matlab PRINCOMP is naive to large data sets. Out-of-memory
errors are easily obtained on imaging data. The solution is to replace
concise but inefficient calls to repmat with loops and to eliminate
large, unused variables.</p>
<p>Tested on random data sets and produces identical output as original
PRINCOMP. Speed penalty is drastic for small data. 50% slower on
50x1000 element data set, but on a 50x10000, only 2% slower.</p>
<p>COEFF = PRINCOMP(X) performs principal components analysis on the N-by-P
data matrix X, and returns the principal component coefficients, also
known as loadings.  Rows of X correspond to observations, columns to
variables.  COEFF is a P-by-P matrix, each column containing coefficients
for one principal component.  The columns are in order of decreasing
component variance.</p>
<p>PRINCOMP centers X by subtracting off column means, but does not
rescale the columns of X.  To perform PCA with standardized variables,
i.e., based on correlations, use PRINCOMP(ZSCORE(X)).  To perform PCA
directly on a covariance or correlation matrix, use PCACOV.</p>
<p>[COEFF, SCORE] = PRINCOMP(X) returns the principal component scores,
i.e., the representation of X in the principal component space.  Rows
of SCORE correspond to observations, columns to components.</p>
<p>[COEFF, SCORE, LATENT] = PRINCOMP(X) returns the principal component
variances, i.e., the eigenvalues of the covariance matrix of X, in
LATENT.</p>
<p>[COEFF, SCORE, LATENT, TSQUARED] = PRINCOMP(X) returns Hotelling&#8217;s
T-squared statistic for each observation in X.</p>
<p>When N &lt;= P, SCORE(:,N:P) and LATENT(N:P) are necessarily zero, and the
columns of COEFF(:,N:P) define directions that are orthogonal to X.</p>
<p>[...] = PRINCOMP(X,&#8217;econ&#8217;) returns only the elements of LATENT that are
not necessarily zero, i.e., when N &lt;= P, only the first N-1, and the
corresponding columns of COEFF and SCORE.  This can be significantly
faster when P &gt;&gt; N.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body"><p class="first">BARTTEST, BIPLOT, CANONCORR, FACTORAN, PCACOV, PCARES, ROTATEFACTORS.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">References:</th><td class="field-body"><ol class="first last arabic simple">
<li>Jackson, J.E., A User&#8217;s Guide to Principal Components,
Wiley, 1988.</li>
<li>Jolliffe, I.T. Principal Component Analysis, 2nd ed.,
Springer, 2002.</li>
<li>Krzanowski, W.J., Principles of Multivariate Analysis,
Oxford University Press, 1988.</li>
<li>Seber, G.A.F., Multivariate Observations, Wiley, 1984.</li>
</ol>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.prplot_multilevel">
<code class="descclassname">Statistics_tools.</code><code class="descname">prplot_multilevel</code><span class="sig-paren">(</span><em>Y</em>, <em>X</em>, <em>wh_col</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.prplot_multilevel" title="Permalink to this definition">¶</a></dt>
<dd><p>Partial correlation plot for multi-level analysis</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">X_resid</span><span class="p">,</span> <span class="n">Y_resid</span><span class="p">,</span> <span class="n">handles</span><span class="p">]</span> <span class="o">=</span> <span class="n">prplot_multilevel</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">X</span><span class="p">,</span> <span class="n">wh_col</span><span class="p">)</span>
</pre></div>
</div>
<p>Uses unweighted estimates.</p>
<p>do not enter intercept in X</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.r2z">
<code class="descclassname">Statistics_tools.</code><code class="descname">r2z</code><span class="sig-paren">(</span><em>r</em>, <em>n</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.r2z" title="Permalink to this definition">¶</a></dt>
<dd><p>Fisher&#8217;s r to Z transformation for providing CIs for a correlation</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">rci</span><span class="p">,</span><span class="n">sig</span><span class="p">,</span><span class="n">Z</span><span class="p">,</span><span class="n">p</span><span class="p">,</span><span class="n">rcrit</span><span class="p">]</span> <span class="o">=</span> <span class="n">r2z</span><span class="p">(</span><span class="n">r</span><span class="p">,</span><span class="n">n</span><span class="p">,[</span><span class="n">alph</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>n:</strong></dt>
<dd><p class="first">n, # of observations going into correlation
(count each row/subject once)</p>
<p class="last">df = n - 3</p>
</dd>
<dt><strong>alph:</strong></dt>
<dd><p class="first last">two-tailed p-value cutoff, 
default is p &lt; .05</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>rci:</strong></dt>
<dd><p class="first last">confidence interval in correlation values</p>
</dd>
<dt><strong>sig:</strong></dt>
<dd><p class="first last">significant at alpha value?</p>
</dd>
<dt><strong>Z:</strong></dt>
<dd><p class="first last">z-scores of correlations</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">p-values</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>can take a vector of r values</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[rci,sig,z] = r2z(.1:.05:.9,5,.05); figure(&#39;Color&#39;,&#39;w&#39;);hold on; plot(.1:.05:.9,rci(:,1),&#39;g&#39;,&#39;LineWidth&#39;,2)
[rci,sig,z] = r2z(.1:.05:.9,10,.05); hold on; plot(.1:.05:.9,rci(:,1),&#39;r&#39;,&#39;LineWidth&#39;,2)
[rci,sig,z] = r2z(.1:.05:.9,20,.05); hold on; plot(.1:.05:.9,rci(:,1),&#39;b&#39;,&#39;LineWidth&#39;,2)
[rci,sig,z] = r2z(.1:.05:.9,40,.05); hold on; plot(.1:.05:.9,rci(:,1),&#39;m&#39;,&#39;LineWidth&#39;,2)
[rci,sig,z] = r2z(.1:.05:.9,80,.05); hold on; plot(.1:.05:.9,rci(:,1),&#39;k&#39;,&#39;LineWidth&#39;,2)
set(gca,&#39;FontSize&#39;,18)
legend({&#39;n = 5&#39; &#39;n = 10&#39; &#39;n = 20&#39; &#39;n = 40&#39; &#39;n = 80&#39;})
title(&#39;.05 Confidence interval lower bound on Pearson&#39;&#39;s r&#39;)
xlabel(&#39;Correlation (r)&#39;)
ylabel(&#39;CI Lower Bound (r)&#39;)
c=.2:.01:.5;,[rci,sig,z,p,rcrit]=r2z(c,39,.05);[c&#39; sig p]

ind=1;for i=1:10:5000,[rci,s,z,p,rc(ind)]=r2z(.5,39,.05/i);,ind=ind+1;,end
figure;plot(1:10:5000,rc);title(&#39;Critical r with Bonf correction&#39;),xlabel(&#39;Comparisons&#39;)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.regress_best_subsets_ga">
<code class="descclassname">Statistics_tools.</code><code class="descname">regress_best_subsets_ga</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.regress_best_subsets_ga" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>GA-based best subsets regression</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">wh_predictors</span><span class="p">,</span> <span class="n">betas</span><span class="p">,</span> <span class="n">b_subset</span><span class="p">,</span> <span class="n">stat_subset</span><span class="p">]</span> <span class="o">=</span> <span class="n">regress_best_subsets_ga</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">Y</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">is outcome data</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">is predictor matrix</p>
</dd>
<dt><strong>wh_predictors:</strong></dt>
<dd><p class="first last">is the primary outcome &#8211; it is vector of which predictors to include in the model</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>the objective criterion is AIC</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.repeated_ancova">
<code class="descclassname">Statistics_tools.</code><code class="descname">repeated_ancova</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em>, <em>wicons</em>, <em>btwnnames</em>, <em>winames</em>, <em>ynames</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.repeated_ancova" title="Permalink to this definition">¶</a></dt>
<dd><p>Repeated measures ANCOVA with table and plot
uses Robust IRLS</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">b</span><span class="p">,</span><span class="n">stats</span><span class="p">,</span><span class="n">yadj</span><span class="p">]</span> <span class="o">=</span> <span class="n">repeated_ancova</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">Y</span><span class="p">,</span><span class="n">wicons</span><span class="p">,</span><span class="n">btwnnames</span><span class="p">,</span><span class="n">winames</span><span class="p">,</span><span class="n">ynames</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>Y = rand(15,2);
X = Y + rand(15,2);
cons = [-1 1 -1 1; 1 1 -1 -1];  % placebo vs control, hot vs. warm

X = R.X(:,1:2);
Y = cl(2).CONTRAST.data;
cons = [-1 1 0 0; 0 0 -1 1];  % placebo vs control for heat then warm
repeated_ancova(X,Y,cons,{&#39;Reported Placebo (C - P)&#39; &#39;Order&#39;},{&#39;P-C Heat&#39; &#39;P-C Warm&#39;},{&#39;CH&#39; &#39;PH&#39; &#39;CW&#39; &#39;PW&#39;});
</pre></div>
</div>
<p>DOES NOT WORK WITH FIXED BTWN-SUBJECTS COVARIATES</p>
<p>X must be a random variable that is observed multiple times for each
subject, as does Y</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.rmanova2">
<code class="descclassname">Statistics_tools.</code><code class="descname">rmanova2</code><span class="sig-paren">(</span><em>data</em>, <em>alpha</em>, <em>doplot</em>, <em>ttst</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.rmanova2" title="Permalink to this definition">¶</a></dt>
<dd><p>Repeated-measures two-way ANOVA</p>
<blockquote>
<div><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stats</span> <span class="o">=</span> <span class="n">rmanova2</span><span class="p">(</span><span class="n">data</span><span class="p">,[</span><span class="n">alpha</span><span class="p">],[</span><span class="n">doplot</span><span class="p">],[</span><span class="n">ttst</span><span class="p">]);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>data:</strong></dt>
<dd><p class="first">can be one of two formats:
1. Cell array - each row represents a level of factor 1, and</p>
<blockquote>
<div><p>each column represents a level of factor 2. Each cell contains a
vector of values of the dependent variable for each subject.</p>
</div></blockquote>
<ol class="last arabic" start="2">
<li><p class="first">Matrix - each row represents a trial, with the following
columns:</p>
<blockquote>
<div><ul class="simple">
<li>column1 - dependent variable</li>
<li>column2 - grouping variable for subject</li>
<li>column3 - grouping variable for factor 1</li>
<li>column4 - grouping variable for factor 2</li>
</ul>
</div></blockquote>
</li>
</ol>
</dd>
<dt><strong>alpha:</strong></dt>
<dd><p class="first last">(optional) p-value threshold (default: 0.05)</p>
</dd>
<dt><strong>doplot:</strong></dt>
<dd><dl class="first last docutils">
<dt>(optional) if 1, will produce a line plot.  </dt>
<dd><p class="first last">Works only for cell input data (default: 1)</p>
</dd>
</dl>
</dd>
<dt><strong>ttst:</strong></dt>
<dd><p class="first last">(optional) if 1, will perform pairwise t-tests (default: 0)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.robust_reg_pooled">
<code class="descclassname">Statistics_tools.</code><code class="descname">robust_reg_pooled</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.robust_reg_pooled" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">betas</span><span class="p">,</span><span class="n">w</span><span class="p">]</span> <span class="o">=</span> <span class="n">robust_reg_pooled</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">Y</span><span class="p">)</span>
<span class="p">[</span><span class="n">betas</span><span class="p">,</span><span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">weighted_reg</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">Y</span><span class="p">,</span><span class="s">&#39;w&#39;</span><span class="p">,</span><span class="n">w</span><span class="p">,</span><span class="s">&#39;uni&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.roc_boot">
<code class="descclassname">Statistics_tools.</code><code class="descname">roc_boot</code><span class="sig-paren">(</span><em>input_vals</em>, <em>binary_outcome</em>, <em>thr</em>, <em>verbose</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.roc_boot" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[ci, names] = roc_boot(input_vals, binary_outcome, thr, [verbose flag])
</pre></div>
</div>
<p>Returns bootstrapped 95% confidence intervals for sensitivity,
specificity, and PPV at a given threshold.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">thr</span> <span class="o">=</span> <span class="n">ROC</span><span class="o">.</span><span class="n">class_threshold</span>
<span class="n">input_vals</span> <span class="o">=</span> <span class="nb">input</span><span class="p">(</span><span class="n">ind</span><span class="p">);</span>
<span class="n">binary_outcome</span> <span class="o">=</span> <span class="n">outcome</span><span class="p">(</span><span class="n">ind</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.roc_calc">
<code class="descclassname">Statistics_tools.</code><code class="descname">roc_calc</code><span class="sig-paren">(</span><em>input_vals</em>, <em>binary_outcome</em>, <em>xvals</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.roc_calc" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate Receiver Operating Characteristic plot (ROC) given P-values</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[xvals, tpr, fpr, auc, c_bias] = roc_calc(input_vals or input values, binary_outcome, [xvals : threshold vals to assess])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>input_vals:</strong></dt>
<dd><p class="first last">continuous-valued observations to classify (e.g., fMRI activity)</p>
</dd>
<dt><strong>binary_outcome:</strong></dt>
<dd><p class="first last">1 / 0 vector of which input observations are &#8220;hits&#8221;</p>
</dd>
<dt><strong>xvals:</strong></dt>
<dd><p class="first last">Criterion values you put in or every 10th percentile of the input
data distribution by default</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>tpr:</strong></dt>
<dd><p class="first last">True positive rate for every step of ROC curve (sensitivity)</p>
</dd>
<dt><strong>fpr:</strong></dt>
<dd><p class="first last">False positive rate (1 - specificity)</p>
</dd>
<dt><strong>auc:</strong></dt>
<dd><p class="first last">Empirical estimate of area under the ROC curve</p>
</dd>
<dt><strong>c_bias:</strong></dt>
<dd><p class="first last">c measure of response bias at each step; MacMillan and Creelman 2005</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% May not work for p-values? may need to convert to t or something.
pvals = STATS.WTS.p;
isnull = DATA.true_weights == 0;
[xvals, tpr, fpr] = roc_calc(pvals, isnull);
figure; plot(fpr, tpr, &#39;ko-&#39;,&#39;Color&#39;, &#39;k&#39;, &#39;LineWidth&#39;, 2);

figure; plot(xvals, fpr, &#39;bo-&#39;)
hold on; plot([0 1], [0 1], &#39;k&#39;, &#39;LineWidth&#39;, 2);
set(gca, &#39;XLim&#39;, [0 .2], &#39;YLim&#39;, [0 .2])
xlabel(&#39;Nominal false positive rate&#39;);
ylabel(&#39;Actual false positive rate&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">roc_plot.m</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.roc_plot">
<code class="descclassname">Statistics_tools.</code><code class="descname">roc_plot</code><span class="sig-paren">(</span><em>input_values</em>, <em>binary_outcome</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.roc_plot" title="Permalink to this definition">¶</a></dt>
<dd><p>This function makes a specific kind of ROC curve plot, based on input
values along a continuous distribution and a binary outcome variable
(logical)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">input_values</span><span class="p">,</span> <span class="n">binary_outcome</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;include&#39;</span><span class="p">,</span> <span class="n">include</span><span class="p">])</span>
</pre></div>
</div>
<p>Include is an optional logical variable of cases to include</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;include&#8217;:</strong></dt>
<dd><p class="first last">followed by logical vector of cases to include</p>
</dd>
<dt><strong>&#8216;threshold&#8217;:</strong></dt>
<dd><p class="first last">followed by a priori threshold cutoff for determining misclassification</p>
</dd>
<dt><strong>&#8216;threshold_type&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by thresh type: choices below:</dt>
<dd><ul class="first last simple">
<li>&#8216;Optimal balanced error rate&#8217;</li>
<li>&#8216;Optimal overall accuracy&#8217; [default]</li>
<li>&#8216;Minimum SDT bias&#8217;</li>
<li>[Enter threshold OR threshold_type]</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd><p class="first last">followed by color, e.g., &#8216;r&#8217; or [1 .5 0]</p>
</dd>
<dt><strong>&#8216;plotmethod&#8217;:</strong></dt>
<dd><p class="first last">followed by &#8216;deciles&#8217; [default] or &#8216;observed&#8217;</p>
</dd>
<dt><strong>&#8216;nonormfit&#8217;:</strong></dt>
<dd><p class="first last">suppress normal curve fitting to ROC</p>
</dd>
<dt><strong>&#8216;plothistograms&#8217;:</strong></dt>
<dd><p class="first last">plot histograms of the signal present/absent distributions</p>
</dd>
<dt><strong>&#8216;writerscoreplus&#8217;:</strong></dt>
<dd><p class="first last">Write text file for input into RScorePlus by Lew Harvey</p>
</dd>
<dt><strong>&#8216;boot&#8217;:</strong></dt>
<dd><p class="first last">[default] Bootstrap 95% confidence intervals for sens, spec, PPV at threshold</p>
</dd>
<dt><strong>&#8216;noboot&#8217;:</strong></dt>
<dd><p class="first last">Skip bootstrap</p>
</dd>
<dt><strong>&#8216;balanced&#8217;:</strong></dt>
<dd><p class="first last">Balanced accuracy for single interval classification</p>
</dd>
<dt><strong>&#8216;dependent&#8217;:</strong></dt>
<dd><p class="first">followed by vector of subject IDs, e.g., (&#8216;dependent&#8217;,[1,1,2,2,3,3].</p>
<p class="last">This will perform multilevel version of binomial test for single interval classifation.</p>
</dd>
<dt><strong>&#8216;noplot&#8217;:</strong></dt>
<dd><p class="first last">Skip generating plots</p>
</dd>
<dt><strong>&#8216;nooutput&#8217;:</strong></dt>
<dd><p class="first last">Suppress text output</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">A structure containing the true and false pos rates (tpr, fpr) along the curve
and the criterion threshold values of the input variable (thr) corresponding to these rates.</p>
<p>Uses the function roc_calc.m</p>
<p>Also returns some information about misclassified observations
and line handle for ROC line plot and other statistics:</p>
<blockquote>
<div><ul class="simple">
<li>area under ROC curve</li>
<li>accuracy statistics based on binomial test</li>
<li>PPV</li>
</ul>
</div></blockquote>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pattern_exp_values</span><span class="p">,</span> <span class="n">ishot</span><span class="p">);</span>
<span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pattern_exp_values</span><span class="p">,</span> <span class="n">ishot</span><span class="p">,</span> <span class="s">&#39;threshold&#39;</span><span class="p">,</span> <span class="mf">2.5</span><span class="p">);</span>
<span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pattern_exp_values</span><span class="p">,</span> <span class="n">ishot</span><span class="p">,</span> <span class="s">&#39;color&#39;</span><span class="p">,</span> <span class="s">&#39;r&#39;</span><span class="p">,</span> <span class="s">&#39;twochoice&#39;</span><span class="p">);</span>
<span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pattern_exp_values</span><span class="p">,</span> <span class="n">ishot</span><span class="p">,</span> <span class="s">&#39;color&#39;</span><span class="p">,</span> <span class="s">&#39;r&#39;</span><span class="p">,</span> <span class="s">&#39;twochoice&#39;</span><span class="p">,</span> <span class="s">&#39;nonormfit&#39;</span><span class="p">);</span>
<span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pexp</span><span class="p">,</span> <span class="n">logical</span><span class="p">(</span><span class="n">outcome</span><span class="p">),</span> <span class="s">&#39;color&#39;</span><span class="p">,</span> <span class="s">&#39;g&#39;</span><span class="p">,</span> <span class="s">&#39;plothistograms&#39;</span><span class="p">,</span> <span class="s">&#39;threshold&#39;</span><span class="p">,</span> <span class="mf">0.3188</span><span class="p">);</span>
<span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pexp</span><span class="p">,</span> <span class="n">logical</span><span class="p">(</span><span class="n">outcome</span><span class="p">),</span> <span class="s">&#39;twochoice&#39;</span><span class="p">,</span> <span class="s">&#39;color&#39;</span><span class="p">,</span> <span class="s">&#39;b&#39;</span><span class="p">,</span> <span class="s">&#39;plothistograms&#39;</span><span class="p">);</span>
<span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pexp</span><span class="p">,</span> <span class="n">logical</span><span class="p">(</span><span class="n">outcome</span><span class="p">),</span> <span class="s">&#39;writerscoreplus&#39;</span><span class="p">);</span>
<span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pexp</span><span class="p">,</span> <span class="n">logical</span><span class="p">(</span><span class="n">outcome</span><span class="p">),</span> <span class="s">&#39;color&#39;</span><span class="p">,</span> <span class="s">&#39;r&#39;</span><span class="p">,</span> <span class="s">&#39;plotmethod&#39;</span><span class="p">,</span> <span class="s">&#39;observed&#39;</span><span class="p">,</span> <span class="s">&#39;plothistograms&#39;</span><span class="p">);</span>
<span class="n">ROC</span> <span class="o">=</span> <span class="n">roc_plot</span><span class="p">(</span><span class="n">pexp</span><span class="p">,</span> <span class="n">logical</span><span class="p">(</span><span class="n">outcome</span><span class="p">),</span> <span class="s">&#39;color&#39;</span><span class="p">,</span> <span class="s">&#39;m&#39;</span><span class="p">,</span> <span class="s">&#39;plotmethod&#39;</span><span class="p">,</span> <span class="s">&#39;observed&#39;</span><span class="p">,</span> <span class="s">&#39;plothistograms&#39;</span><span class="p">,</span> <span class="s">&#39;Optimal overall accuracy&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>For a whole image with p-values, this may be helpful.</p>
<p>Pre-specifies p-values you want to evaluate at.</p>
<div class="highlight-python"><div class="highlight"><pre>rocout = roc_plot(1-t.p, truevals, &#39;plotmethod&#39;, &#39;observed&#39;, &#39;valuestoevaluate&#39;, 1 - [.5:-.1:.1 .05 .01 .005 .001 .0001 .00001 .000001 .0000001 .00000001]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.rsquare_calc">
<code class="descclassname">Statistics_tools.</code><code class="descname">rsquare_calc</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.rsquare_calc" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">rsquare</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">y</span><span class="p">)</span>
</pre></div>
</div>
<p>Returns variance in each col. of Y explained by X</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.rsquare_multiple_regions_multilevel">
<code class="descclassname">Statistics_tools.</code><code class="descname">rsquare_multiple_regions_multilevel</code><span class="sig-paren">(</span><em>Y</em>, <em>X</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.rsquare_multiple_regions_multilevel" title="Permalink to this definition">¶</a></dt>
<dd><p>Predict outcome var (y) using data from multiple vars (X, e.g., brain regions)
Test R-square against permuted data</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stats</span> <span class="o">=</span> <span class="n">rsquare_multiple_regions_multilevel</span><span class="p">(</span><span class="n">Y</span><span class="p">,</span> <span class="n">X</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first">Variable args</p>
<dl class="docutils">
<dt><strong>&#8216;colors&#8217;:</strong></dt>
<dd><p class="first last">followed by colors cell ({&#8216;r&#8217; &#8216;g&#8217; &#8216;b&#8217;}) for each col. of X</p>
</dd>
<dt><strong>&#8216;nperms:</strong></dt>
<dd><p class="first last">then num perms</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% SETUP data
cl = [clpos_data clneg_data];

cl(cat(1, cl.numVox) &lt; min_cluster_size) = [];

% get brain data cell
for i = 1:size(cl(1).all_data, 2) 
    for c = 1:length(cl)
        data{i}(:,c) = cl(c).all_data(:, i); 
    end
end

% NMDS ANALYSIS
OUT = [];

OUT.ridge = matrix_direct_effects_ridge(data);
D = OUT.ridge.mean; D(find(eye(size(D)))) = 1;
D = (D&#39; + D) ./ 2;
OUT.ridge.D = (1 - D) ./ 2;
[OUT.stats_mds.GroupSpace,OUT.stats_mds.obs,OUT.stats_mds.implied_dissim] = shepardplot(OUT.ridge.D,[]);

OUT.stats_mds = nmdsfig_tools(&#39;cluster_solution&#39;,OUT.stats_mds, OUT.stats_mds.GroupSpace, 2:max_networks, nperms, []);
OUT.stats_mds.colors = {&#39;ro&#39; &#39;go&#39; &#39;bo&#39; &#39;yo&#39; &#39;co&#39; &#39;mo&#39; &#39;ko&#39; &#39;r^&#39; &#39;g^&#39; &#39;b^&#39; &#39;y^&#39; &#39;c^&#39; &#39;m^&#39; &#39;k^&#39;};
create_figure(&#39;nmdsfig&#39;);

OUT.stats_mds.names = [];
nmdsfig(OUT.stats_mds.GroupSpace,&#39;classes&#39;,OUT.stats_mds.ClusterSolution.classes,&#39;names&#39;,OUT.stats_mds.names,&#39;sig&#39;,OUT.ridge.fdrsig);
hh = nmdsfig_fill(OUT.stats_mds);
axis image; axis equal


% Multiple regions predict behavior

% Design matrix with cluster averages
classes = OUT.stats_mds.ClusterSolution.classes;
clear X
for i = 1:length(data)
    for j = 1:max(classes)
        X{i}(:, j) = nanmean(data{i}(:, classes == j), 2); 
    end
end

OUT.stats_regression = rsquare_multiple_regions_multilevel(acc_x_dist, X, &#39;colors&#39;, OUT.stats_mds.colors, &#39;nperms&#39;, 100);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.scn_stats_helper_functions">
<code class="descclassname">Statistics_tools.</code><code class="descname">scn_stats_helper_functions</code><span class="sig-paren">(</span><em>meth</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.scn_stats_helper_functions" title="Permalink to this definition">¶</a></dt>
<dd><p>Helper functions for stats routines, plotting, and printing</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Available Methods:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;print&#8217;:</strong></dt>
<dd><p class="first last">print outcome table from stats structure</p>
</dd>
<dt><strong>&#8216;gls&#8217;:</strong></dt>
<dd><p class="first last">GLS function, weighted or unweighted; with AR model if
specified as last input</p>
</dd>
<dt><strong>&#8216;boot&#8217;:</strong></dt>
<dd><p class="first last">Boostrapping of GLS</p>
</dd>
<dt><strong>&#8216;signperm&#8217;:</strong></dt>
<dd><p class="first last">Sign permutation test for intercept of GLS</p>
</dd>
</dl>
<p><strong>&#8216;plot&#8217;:</strong></p>
<dl class="docutils">
<dt><strong>&#8216;loess_xy&#8217;:</strong></dt>
<dd><p class="first">loess plots of multilevel X and Y data</p>
<p class="last">X is cell array with X data for each subject, Y is cell array, same format</p>
</dd>
<dt><strong>&#8216;xycatplot&#8217;:</strong></dt>
<dd><p class="first last">LOESS (or no loess) plots of N subjects, categorical X vs Y</p>
</dd>
</dl>
<p><strong>&#8216;loess_partial&#8217;:</strong></p>
<dl class="docutils">
<dt><strong>&#8216;xyplot&#8217;:</strong></dt>
<dd><p class="first last">multilevel line plot of x vs y within subjects, grouping trials by
optional G variable. X can be categorical or continuous.</p>
</dd>
<dt><strong>&#8216;xytspanelplot&#8217;:</strong></dt>
<dd><p class="first last">X and Y are timeseries data; separate panel plot for each cell</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Format Strings:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">scn_stats_helper_functions</span><span class="p">(</span><span class="s">&#39;print&#39;</span><span class="p">,</span> <span class="n">stats</span><span class="p">,</span> <span class="n">stats_within</span><span class="p">)</span>

<span class="p">[</span><span class="n">b</span><span class="p">,</span> <span class="n">s2between_ols</span><span class="p">,</span> <span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">scn_stats_helper_functions</span><span class="p">(</span><span class="s">&#39;gls&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">W</span><span class="p">,</span> <span class="n">X</span><span class="p">)</span>
<span class="p">[</span><span class="n">b</span><span class="p">,</span> <span class="n">s2between_ols</span><span class="p">]</span> <span class="o">=</span> <span class="n">scn_stats_helper_functions</span><span class="p">(</span><span class="s">&#39;gls&#39;</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">W</span><span class="p">,</span> <span class="n">X</span><span class="p">,</span> <span class="n">arorder</span><span class="p">);</span>
</pre></div>
</div>
<p>See glmfit_general for context and usage</p>
<div class="highlight-python"><div class="highlight"><pre>stats = scn_stats_helper_functions(&#39;boot&#39;, Y, W, X, bootsamples, stats, whpvals_for_boot, targetu, verbose )
stats = scn_stats_helper_functions(&#39;boot&#39;, Y, stats.W, X, 1000, stats, 1:size(Y,2), .005, 1 )

stats = scn_stats_helper_functions(&#39;signperm&#39;, Y, W, X, nperms, stats, permsign, verbose )
stats = scn_stats_helper_functions(&#39;signperm&#39;, Y, W, X, 5000, stats, [], 1 );

stats = scn_stats_helper_functions(&#39;xycatplot&#39;, X, Y);
stats_plot = scn_stats_helper_functions(&#39;xycatplot&#39;, stats.inputOptions.X, stats.inputOptions.Y, &#39;weighted&#39;, &#39;samefig&#39;);

scn_stats_helper_functions(&#39;xyplot&#39;, data(1:19), SETUP.data.Y, &#39;weighted&#39;, &#39;groupby&#39;, G, &#39;colors&#39;, {&#39;y&#39; [.2 .2 .2]});&#39;y&#39;
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body">using sign permutation test</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>Y = randn(20, 4); X = Y(:,1) + randn(20,1); X = [ones(size(X)) X];
first_lev_var = rand(20, 4);
stats = glmfit_general(Y, X, &#39;weighted&#39;, &#39;s2&#39;, first_lev_var, &#39;dfwithin&#39;, 20, &#39;verbose&#39;);
W = stats.W;
stats = scn_stats_helper_functions(&#39;signperm&#39;, Y, W, X, 5000, stats, [], 1 );
% Re-run using already set-up permsign:
stats = scn_stats_helper_functions(&#39;signperm&#39;, Y, W, X, 5000, stats, stats.permsign, 1 );


scn_stats_helper_functions(&#39;loess_xy&#39;, stats.inputOptions.X, stats.inputOptions.Y)


% Loading mediation clusters and making line plot
G = SETUP.data.X;
whcl = 30; cluster_orthviews(clneg_data{1}(whcl));
clear data, for i = 1:length(clneg_data), data{i} = clneg_data{i}(whcl).timeseries; end
scn_stats_helper_functions(&#39;xyplot&#39;, data, SETUP.data.Y, &#39;weighted&#39;, &#39;groupby&#39;, G, &#39;colors&#39;, {&#39;y&#39; [.2 .2 .2]});
scn_stats_helper_functions(&#39;xyplot&#39;, SETUP.data.Y, data, &#39;weighted&#39;, &#39;groupby&#39;, G, &#39;colors&#39;, {&#39;y&#39; [.4 .4 .4]}); xlabel(&#39;Report&#39;); ylabel(&#39;Brain&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Note:</th><td class="field-body"><p class="first">Out of memory errors for large n!</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Called by:</th><td class="field-body"><ul class="first last simple">
<li>mediation.m</li>
<li>glmfit_general.m</li>
</ul>
</td>
</tr>
</tbody>
</table>
<p>Tor Wager, Sept 2007</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.sdt_A">
<code class="descclassname">Statistics_tools.</code><code class="descname">sdt_A</code><span class="sig-paren">(</span><em>H</em>, <em>F</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.sdt_A" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate signal detection theory A statistic: 
a non-parametric estimate of sensitivity in ROC analysis
per Zhang and Mueller (2005)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">A</span> <span class="o">=</span> <span class="n">sdt_A</span><span class="p">(</span><span class="n">H</span><span class="p">,</span> <span class="n">F</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>H:</strong></dt>
<dd><p class="first last">hit rate</p>
</dd>
<dt><strong>F:</strong></dt>
<dd><p class="first last">false alarm rate</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Meaning of values where F &gt; H (&#8220;reverse skill&#8221;):</p>
<p>The meaningful measure of sensitivity in this case is A(F,H) 
as opposed to A(H,F).  Subject is assumed to still be sensitive, 
but to have inverted their interpretation (or &#8220;reverse skill&#8221;).</p>
<p>To distinguish from the normal case, such results are returned
as (1 - sdt_A(F,H)), so the full range from 0 to 1 is used.
For comparisons of sensitivity use abs(sdt_A(H,F) - 0.5).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>sdt_A(0.6,0.8) = 0.325. This indicates the same sensitivity as 
sdt_A(0.8,0.6) = 0.675, but with reversed skill.</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.searchlight_applymask">
<code class="descclassname">Statistics_tools.</code><code class="descname">searchlight_applymask</code><span class="sig-paren">(</span><em>dat1</em>, <em>dat2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.searchlight_applymask" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimate local pattern weight on train data using SVR and searchlight and 
apply weights to the test dataset</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">dat</span><span class="p">,</span> <span class="n">sl_size</span><span class="p">]</span> <span class="o">=</span> <span class="n">searchlight_applymask</span><span class="p">(</span><span class="n">train</span><span class="p">,</span> <span class="n">test</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat1:</strong></dt>
<dd><p class="first last">fmri_data object with train.Y == size(train.dat,2)</p>
</dd>
<dt><strong>dat2:</strong></dt>
<dd><p class="first last">Data to apply local weight map.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;r&#8217;:</strong></dt>
<dd><p class="first last">searchlight sphere radius (in voxel) (default: r = 3 voxels)</p>
</dd>
<dt><strong>&#8216;parallel&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>run subset of voxels to distribute on a cluster.  flag must</dt>
<dd><p class="first last">be followed by array specifing id and total number of jobs
(e.g., &#8216;parallel&#8217;,[1,10]);</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">This contains an fmri_data object that contain
correlation pattern expression values</p>
</dd>
<dt><strong>sl_size:</strong></dt>
<dd><p class="first last">The number of voxels within each searchlight. Based on this
number, you can detect searchlights on the edge (searchlights
with low sl_size should be on the edge of the brain.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">r</span><span class="p">,</span> <span class="n">dat</span><span class="p">]</span> <span class="o">=</span> <span class="n">searchlight_applymask</span><span class="p">(</span><span class="n">train</span><span class="p">,</span> <span class="n">test</span><span class="p">,</span> <span class="s">&#39;r&#39;</span><span class="p">,</span> <span class="mi">5</span><span class="p">);</span>

<span class="p">[</span><span class="n">r</span><span class="p">,</span> <span class="n">dat</span><span class="p">]</span> <span class="o">=</span> <span class="n">searchlight_applymask</span><span class="p">(</span><span class="n">train</span><span class="p">,</span> <span class="n">test</span><span class="p">,</span> <span class="s">&#39;r&#39;</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span><span class="s">&#39;parallel&#39;</span><span class="p">,[</span><span class="mi">1</span><span class="p">,</span><span class="mi">10</span><span class="p">]);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.searchlight_applymask_collate">
<code class="descclassname">Statistics_tools.</code><code class="descname">searchlight_applymask_collate</code><span class="sig-paren">(</span><em>dat2</em>, <em>file_list</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.searchlight_applymask_collate" title="Permalink to this definition">¶</a></dt>
<dd><p>Estimate local pattern weight on train data using SVR and searchlight and
apply weights to the test dataset</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">dat_comb</span> <span class="o">=</span> <span class="n">searchlight_applymask</span><span class="p">(</span><span class="n">dat2</span><span class="p">,</span> <span class="n">file_list</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat2:</strong></dt>
<dd><p class="first last">fmri_data test object</p>
</dd>
<dt><strong>file_list:</strong></dt>
<dd><p class="first last">Cellarray of list of file distributed in parallel.  Make sure
it is sorted correctly (e.g., sort_nat())</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dat_comb:</strong></dt>
<dd><p class="first last">An fmri_data object with searchlight weights applied to
test dataset.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">dat_comb</span> <span class="o">=</span> <span class="n">searchlight_applymask_collate</span><span class="p">(</span><span class="n">dat2</span><span class="p">,</span> <span class="n">file_list</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.searchlight_correlation">
<code class="descclassname">Statistics_tools.</code><code class="descname">searchlight_correlation</code><span class="sig-paren">(</span><em>mask1</em>, <em>mask2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.searchlight_correlation" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate the local pattern similarity between two pattern maps using
the searchlight approach.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">r_corr</span><span class="p">,</span> <span class="n">dat</span><span class="p">,</span> <span class="n">sl_size</span><span class="p">]</span> <span class="o">=</span> <span class="n">searchlight_correlation</span><span class="p">(</span><span class="n">mask1</span><span class="p">,</span> <span class="n">mask2</span><span class="p">,</span> <span class="p">[</span><span class="n">additional_inputs</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>mask1:</strong></dt>
<dd><p class="first last">pattern or activation maps 1</p>
</dd>
<dt><strong>mask2:</strong></dt>
<dd><p class="first last">pattern or activation mediaps 2</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;r&#8217;:</strong></dt>
<dd><p class="first last">searchlight sphere radius (in voxel) (default: r = 3 voxels)</p>
</dd>
<dt><strong>&#8216;type&#8217;:</strong></dt>
<dd><p class="first">This calls corr.m, and can take &#8216;type&#8217; option.</p>
<p class="last">&#8216;Pearson&#8217; (default), &#8216;Kendall&#8217;, &#8216;Spearman&#8217;.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>r_corr:</strong></dt>
<dd><p class="first last">Correlation between weights of two pattern masks</p>
</dd>
<dt><strong>dat:</strong></dt>
<dd><p class="first last">This contains a statistic_image object that contain 
correlation values between weights of two pattern masks 
(=r_corr; in .dat) and p values for the correlation values 
(in .p).</p>
</dd>
<dt><strong>sl_size:</strong></dt>
<dd><p class="first last">The number of voxels within each searchlight. Based on this 
number, you can detect searchlights on the edge (searchlights 
with low sl_size should be on the edge of the brain.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">mask1</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="s">&#39;weights_NSF_grouppred_cvpcr.img&#39;</span><span class="p">);</span>
<span class="n">mask2</span> <span class="o">=</span> <span class="n">which</span><span class="p">(</span><span class="s">&#39;nonnoc_v6_109subjmap_mean.nii&#39;</span><span class="p">);</span>

<span class="p">[</span><span class="n">r</span><span class="p">,</span> <span class="n">dat</span><span class="p">]</span> <span class="o">=</span> <span class="n">searchlight_correlation</span><span class="p">(</span><span class="n">mask1</span><span class="p">,</span> <span class="n">mask2</span><span class="p">,</span> <span class="s">&#39;r&#39;</span><span class="p">,</span> <span class="mi">5</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.searchlight_disti">
<code class="descclassname">Statistics_tools.</code><code class="descname">searchlight_disti</code><span class="sig-paren">(</span><em>dat</em>, <em>mask</em>, <em>dist_i</em>, <em>additional_inputs</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.searchlight_disti" title="Permalink to this definition">¶</a></dt>
<dd><p>Run the actual searchlight analysis on each brain chunck 
searchlight_dream.m will generate codes to run this funtion.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">out</span> <span class="o">=</span> <span class="n">searchlight_disti</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">mask</span><span class="p">,</span> <span class="n">dist_i</span><span class="p">,</span> <span class="p">[</span><span class="n">additional_inputs</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>searchlight_dream.m, xval_cross_classify.m, fmri_data.predict.m,</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.searchlight_dream">
<code class="descclassname">Statistics_tools.</code><code class="descname">searchlight_dream</code><span class="sig-paren">(</span><em>dat</em>, <em>dist_n</em>, <em>mask</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.searchlight_dream" title="Permalink to this definition">¶</a></dt>
<dd><p>This function generates codes for submitting multiple distributed jobs to 
clusters to run a searchlight analysis on multiple chunks of the brain.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>searchlight_dream(dat, dist_n, mask, &#39;algorithm_name&#39;, &#39;cv_svm&#39; (or &#39;cv_lassopcr&#39;), &#39;cv_assign&#39;, whfolds, [optional input])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Features:</th><td class="field-body"><ul class="first last simple">
<li>generates dist_n scripts in modeldir (or current directory)</li>
<li>can run a searchlight analysis on one dataset, or two datasets
(cross-classification)</li>
<li>can apply different radius</li>
<li>can obtain cross-validated results with &#8216;cv_assign&#8217; option</li>
<li>can use SVM (linear svm is a default) and LASSO-PCR. You need to have
a spider toolbox and lasso rocha toolbox in your path</li>
<li>you can save predictive weights for each searchlight or discard them</li>
</ul>
</td>
</tr>
</tbody>
</table>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">image_vector or fmri_data object with data</p>
</dd>
<dt><strong>dat1.Y(:,1):</strong></dt>
<dd><p class="first last">for svm: true(1) or false(-1) for each observation (image) in Y(:,1)
for lassopcr: continuous value for Y(:,1)</p>
</dd>
<dt><strong>dat1.Y(:,[2:n]):</strong></dt>
<dd><p class="first last">Test sets: could be binary: and true(1), false(-1),
ignore(0) or continuous values</p>
</dd>
<dt><strong>dist_n:</strong></dt>
<dd><p class="first last">The number of jobs (brain chunks) you want to create</p>
</dd>
<dt><strong>mask:</strong></dt>
<dd><p class="first last">This will be run on voxels within the mask
e.g., which(&#8216;scalped_avg152T1_graymatter.img&#8217;)</p>
</dd>
<dt><strong>&#8216;algorithm_name&#8217;:</strong></dt>
<dd><p class="first last">should be followed by &#8216;cv_svm&#8217; or &#8216;cv_lassopcr&#8217;</p>
</dd>
<dt><strong>&#8216;cv_assign&#8217;:</strong></dt>
<dd><p class="first last">a vector of integers for membership in custom holdout set 
of each fold</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;dat2&#8217;:</strong></dt>
<dd><p class="first last">cross-classification; should be followed by dat2 and dat2.Y
dat2.Y(:,1) - for training/testing, dat2(:,[2:n]) - for testing</p>
</dd>
<dt><strong>&#8216;r&#8217;:</strong></dt>
<dd><p class="first last">searchlight sphere radius (in voxel) (default: r = 3 voxels)</p>
</dd>
<dt><strong>&#8216;modeldir&#8217;:</strong></dt>
<dd><p class="first last">the directory where all the variables and results will be 
saved; should be followed by a directory location 
(default: the current directory)</p>
</dd>
<dt><strong>&#8216;scale&#8217;:</strong></dt>
<dd><p class="first last">z-scored input data in image_vector or fmri_data object
(default = false)</p>
</dd>
<dt><strong>&#8216;balanced&#8217;:</strong></dt>
<dd><p class="first last">use the balanced ridge option - balanced ridge value should
be followed. (default = false)</p>
</dd>
<dt><strong>&#8216;outcome_method&#8217;:</strong></dt>
<dd><blockquote class="first">
<div><dl class="docutils">
<dt>followed by the following options</dt>
<dd><ul class="first last simple">
<li>&#8216;correlation&#8217; - &#8220;default&#8221; for for continuous measures</li>
</ul>
</dd>
</dl>
</div></blockquote>
<p class="last">&#8216;twochoice&#8217;- &#8220;default&#8221; for binary outcome
&#8216;singleinterval&#8217;  - for binary outcome</p>
</dd>
<dt><strong>&#8216;save_weights&#8217;:</strong></dt>
<dd><p class="first last">save weights for each searchlight (default = false)</p>
</dd>
<dt><strong>&#8216;email&#8217;:</strong></dt>
<dd><p class="first last">should be followed by an email adress (default = false)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>This function will generate codes that call &#8220;searchlight_disti.m&#8221;, which
will save the following output variables.</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>out.test_results:</strong></dt>
<dd>accuracy, p, and se for binary classification, and
correlation (pearson), p for prediction of continuous values
For the cross-classification, test_results will save
four values for each searchlight. The order of the test 
results are [dat1-on-dat1, dat1-on-dat2, dat2-on-dat1, 
dat2-on-dat2]. All results are cross-validated results.</dd>
<dt><strong>out.stats1 &amp; stats2:</strong></dt>
<dd>stats1 and stats2 are similar to the outputs of predict function.</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Examples for lassopcr:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% data preparation
dat = fmri_data(which(&#39;brainmask.nii&#39;));
dat.dat = randn(dat.volInfo.n_inmask, 30);

% setting up training values
dat.Y = dat.dat(111111, :)&#39; + .3 * randn(30, 1);

% setting up testing values
dat.Y(:,2) = [ones(10,1); zeros(10,1); -ones(10,1)];
dat.Y(:,3) = dat.dat(111111, :)&#39; + .3 * randn(30, 1);
mask = which(&#39;scalped_avg152T1_graymatter.img&#39;);
dist_n = 50;

% data for cross classification
dat2 = fmri_data(which(&#39;brainmask.nii&#39;));
dat2.dat = randn(dat.volInfo.n_inmask, 30);
dat2.Y = dat2.dat(111, :)&#39; + .3 * randn(30, 1);
dat2.Y(:,2) = dat.Y(:,2);
dat2.Y(:,3) = dat2.dat(111111, :)&#39; + .3 * randn(30, 1);

% setting up other variables
r = 6;
modeldir = &#39;/dreamio/home/chwo9116/Documents/searchlight_dream_test&#39;;
holdout_set = ones(6, 1); for i = 2:5, holdout_set = [holdout_set; i*ones(6, 1)]; end

% generate scripts in modeldir
searchlight_dream(dat, dist_n, mask, &#39;dat2&#39;, dat2, &#39;algorithm_name&#39;, ...
&#39;cv_lassopcr&#39;, &#39;r&#39;, 6, &#39;modeldir&#39;, modeldir, &#39;cv_assign&#39;, holdout_set, ...
&#39;save_weights&#39;, &#39;outcome_method&#39;, &#39;singleinterval&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>searchlight_disti.m, xval_cross_classify.m, fmri_data.predict.m,</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.searchlight_saveresults">
<code class="descclassname">Statistics_tools.</code><code class="descname">searchlight_saveresults</code><span class="sig-paren">(</span><em>modeldir</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.searchlight_saveresults" title="Permalink to this definition">¶</a></dt>
<dd><p>This function combines and saves searchlight analysis results.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">searchlight_saveresults</span><span class="p">(</span><span class="n">modeldir</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>modeldir:</strong></dt>
<dd><p class="first last">the directory where the searchlight result mat files (e.g.,
searchlight_results_*_01Aug2014.mat)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">This function saves the brain maps where each voxel contains a summary 
stat value (e.g., accuracy or outcome correlation) in the modeldir. The
naming convention of the result maps are as follow:</p>
<dl class="docutils">
<dt>results_searchlight_(a)_(b)_dat(c).nii</dt>
<dd><ol class="first last arabic">
<li><p class="first">will be a number - the number of test results</p>
</li>
<li><p class="first">acc: accuracy, r: outcome correlation, p: p-values, se: standard error, 
and thr: threshold for the single-interval test</p>
</li>
<li><p class="first">If the test was on one dataset, (c) will be empty, but if the
test was cross-prediction, (c) will be 11, 12, 21, or 22.</p>
<blockquote>
<div><ul class="simple">
<li>11: trained on the first dataset, and tested on the first dataset</li>
<li>12: trained on the first dataset, and tested on the second dataset</li>
<li>21: trained on the second dataset, and tested on the first dataset</li>
<li>22: trained on the second dataset, and tested on the second dataset</li>
</ul>
</div></blockquote>
</li>
</ol>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Output Examples:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>results_searchlight_1_r_dat11.nii (outcome correlation)
results_searchlight_1_p_dat11.nii (p value for the correlation values)

results_searchlight_2_acc_dat11.nii (accuracy)
results_searchlight_2_p_dat11.nii (p value for the accuracy)
results_searchlight_2_se_dat11.nii (standard error for the accuracy)
results_searchlight_2_thr_dat11.nii (threshold for the accuracy test)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.shift_correl">
<code class="descclassname">Statistics_tools.</code><code class="descname">shift_correl</code><span class="sig-paren">(</span><em>a</em>, <em>b</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.shift_correl" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[shiftvals, corrvals, bestshift, bestcorr, aout, bout] = shift_correl(a, b, [&#39;max_shift&#39;, max_shift], [&#39;shift_step&#39;, shift_step], [&#39;use_rob_stats&#39;, 0|1], [&#39;return_betas&#39;, 0|1], [&#39;display_plots&#39;, 0|1],[&#39;priors&#39;,[mu sig]])
</pre></div>
</div>
<p>Shifts a backwards and forwards by max_shift elements (default 12)
in each direction, computing the correlation
between a and b at each shift value</p>
<p>Thus, negative values mean a happens AFTER b
Positive values means b happens later</p>
<p>truncates the tails of a and b where necessary
to ensure they&#8217;re the same length.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>a, b:</strong></dt>
<dd><p class="first last">two vectors to correlate</p>
</dd>
<dt><strong>max_shift:</strong></dt>
<dd><p class="first last">max number of elements to shift (default 12)</p>
</dd>
<dt><strong>shift_step:</strong></dt>
<dd><p class="first last">size of incremental shifts (default .1)</p>
</dd>
<dt><strong>use_rob_stats:</strong></dt>
<dd><p class="first last">robust regression, IRLS (default 0)</p>
</dd>
<dt><strong>return_betas:</strong></dt>
<dd><p class="first last">return betas rather than corr coeffs (default 0)</p>
</dd>
<dt><strong>display_plots:</strong></dt>
<dd><p class="first">display plot of shift vals and correlations/betas (default 0)</p>
<p class="last">NOTE: robust r value of chosen solution may not be max,
as weighted_corrcoef now does not account for variance
inflation with low weights for some observations</p>
</dd>
<dt><strong>priors:</strong></dt>
<dd><p class="first last">Incorporate gaussian priors with mean priors(1) and std
priors(2)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% first extract data for a subject into clusters:
clusters = roi_probe(spm_get(Inf), &#39;snpm0002_varsm_cov_clusters.mat&#39;);

% OR

clusters = mask2clusters(&#39;SnPMt_filtered.img&#39;, spm_get(Inf));

% Then do the shifting:

for i=1:length(clusters)
    [shiftvals, corrvals] = shift_correl(xX.X(:, 1), clusters(i).timeseries);
    title([&#39;Cluster&#39; num2str(i)]);
    disp([&#39;Cluster &#39; num2str(i) &#39; estimate shift by &#39; num2str(1.5 * shiftvals(corrvals == max(corrvals)))]);
end


% Simulation
trueval = 1;
n = 100;
noisevar = 0;
a = randn(n,1);
b = a + randn(n,1).*noisevar;
a=smooth_timeseries(a,n./5);
b=smooth_timeseries(b,n./5);
a = shift_signal(a,trueval);
b = b(1:length(a));
figure;plot(a);
hold on;
plot(b,&#39;r&#39;)
[shiftvals, corrvals, bestshift, bestcorr, aout, bout] = ...
shift_correl(a, b, &#39;max_shift&#39;, 4, &#39;shift_step&#39;, .2, &#39;use_rob_stats&#39;, 1, &#39;return_betas&#39;, 0, &#39;display_plots&#39;, 1,&#39;optimize&#39;); bestshift

r = .7; shiftstep = 1;
optstr = &#39;optimize&#39;; %    or &#39;noopt&#39;
tic
for i = 1:10
    ab = mvnrnd([0 0],[1 r; r 1],n);
    a = ab(:,1);
    b = ab(:,2);
    a=smooth_timeseries(a,n./5);
    b=smooth_timeseries(b,n./5);
    a = shift_signal(a,trueval);
    b = b(1:length(a));
    [shiftvals, corrvals, bestshift(i), bestcorr(i), aout, bout] = shift_correl(a, b, &#39;max_shift&#39;, 4, &#39;shift_step&#39;, shiftstep, &#39;use_rob_stats&#39;, 1,optstr);
end
toc
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.shift_signal">
<code class="descclassname">Statistics_tools.</code><code class="descname">shift_signal</code><span class="sig-paren">(</span><em>y</em>, <em>shiftby</em>, <em>interp_method</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.shift_signal" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[shiftedy] = shift_signal(y, shiftby, [interpolation method])
</pre></div>
</div>
<p>Shifts signal contained in y back &#8216;shiftby&#8217; steps</p>
<p>Note the variable shiftby should be positive.</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.signtest_matrix">
<code class="descclassname">Statistics_tools.</code><code class="descname">signtest_matrix</code><span class="sig-paren">(</span><em>dat</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.signtest_matrix" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">stats</span> <span class="o">=</span> <span class="n">signtest_matrix</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
</pre></div>
</div>
<p>This is a matric-ized version of Matlab 2010&#8217;s signtest.m
it returns identical P-values to Matlab&#8217;s function</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.ste">
<code class="descclassname">Statistics_tools.</code><code class="descname">ste</code><span class="sig-paren">(</span><em>dat</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.ste" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">my_ste</span><span class="p">,</span><span class="n">t</span><span class="p">,</span><span class="n">n_in_column</span><span class="p">,</span><span class="n">p</span><span class="p">,</span><span class="n">mean</span><span class="p">]</span> <span class="o">=</span> <span class="n">ste</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
</pre></div>
</div>
<p>standard error of the mean
and t-values, columnwise
and p-values, if asked for</p>
<p>omits NaN values row-wise within each column</p>
<p>does NOT use n - 1
matches matlab t-test function</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.stepwise_tor">
<code class="descclassname">Statistics_tools.</code><code class="descname">stepwise_tor</code><span class="sig-paren">(</span><em>dat</em>, <em>y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.stepwise_tor" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">STEPWISE</span> <span class="o">=</span> <span class="n">stepwise_tor</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="p">[</span><span class="n">pred</span><span class="o">.</span> <span class="n">names</span><span class="p">],</span> <span class="p">[</span><span class="n">alpha</span><span class="p">])</span>
</pre></div>
</div>
<p>Stepwise regression using Matlab with a couple of extras:</p>
<p>Print omnibus F-values for stepwise regression</p>
<p>get adjusted R-squared</p>
<p>save output structure</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.stouffer">
<code class="descclassname">Statistics_tools.</code><code class="descname">stouffer</code><span class="sig-paren">(</span><em>p</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.stouffer" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">z</span><span class="p">,</span><span class="n">p</span><span class="p">,</span><span class="n">sig</span><span class="p">]</span> <span class="o">=</span> <span class="n">stouffer</span><span class="p">(</span><span class="n">p</span><span class="p">,[</span><span class="n">alph</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>p:</strong></dt>
<dd><p class="first">p values in 4-D array</p>
<p>1st 3 dims are within images, dim4 = image</p>
<p class="last">optional: alpha value for thresholding</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>z:</strong></dt>
<dd><p class="first last">stouffer&#8217;s combined test statistic, compare to normal</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">p-values for combined test</p>
</dd>
<dt><strong>sig:</strong></dt>
<dd><p class="first last">signficance 1 / 0 binary mask, if alpha is specified</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Described in:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Lazar, N. A., Luna, B., Sweeney, J. A., &amp; Eddy, W. F. (2002). 
Combining brains: a survey of methods for statistical pooling 
of information. Neuroimage, 16(2), 538-550.</p>
<p>Stouffer, S. A., Suchman, E. A., DeVinney, L. C., Star, S. A., and
Williams, R. M. 1949. The American Soldier: Vol. I. Adjustment
During Army Life. Princeton University Press, Princeton.</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.subset_indicator_matrix">
<code class="descclassname">Statistics_tools.</code><code class="descname">subset_indicator_matrix</code><span class="sig-paren">(</span><em>n</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.subset_indicator_matrix" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a matrix whose rows contain indicators (1/0 values) for all
possible subsets of n variables</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">subsets</span> <span class="o">=</span> <span class="n">subset_indicator_matrix</span><span class="p">(</span><span class="n">n</span><span class="p">)</span>
</pre></div>
</div>
<p>Use this, for example, to create a matrix that tells you all possible
combinations of regressors to include in a regression.</p>
<blockquote>
<div>Tor Wager, March 07</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.t_test2">
<code class="descclassname">Statistics_tools.</code><code class="descname">t_test2</code><span class="sig-paren">(</span><em>x</em>, <em>m</em>, <em>alpha</em>, <em>tail</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.t_test2" title="Permalink to this definition">¶</a></dt>
<dd><p>TTEST Hypothesis test: Compares the sample average to a constant.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">H</span><span class="p">,</span><span class="n">SIG</span><span class="p">]</span> <span class="o">=</span> <span class="n">TTEST</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">M</span><span class="p">,</span><span class="n">ALPHA</span><span class="p">,</span><span class="n">TAIL</span><span class="p">)</span>
</pre></div>
</div>
<p>performs a T-test to determine if a sample from a normal distribution
(in X) could have mean M.</p>
<p>M = 0, ALPHA = 0.05 and TAIL = 0 by default.</p>
<dl class="docutils">
<dt>The Null hypothesis is: &#8220;mean is equal to M&#8221;.</dt>
<dd><ul class="first last simple">
<li>For TAIL=0,  alternative: &#8220;mean is not M&#8221;.</li>
<li>For TAIL=1,  alternative: &#8220;mean is greater than M&#8221;</li>
<li>For TAIL=-1, alternative: &#8220;mean is less than M&#8221;</li>
<li>TAIL = 0 by default.</li>
</ul>
</dd>
</dl>
<p>ALPHA is desired significance level.</p>
<p>SIG is the probability of observing the given result by chance
given that the null hypothesis is true. Small values of SIG cast
doubt on the validity of the null hypothesis.</p>
<blockquote>
<div><ul class="simple">
<li>H=0 =&gt; &#8220;Do not reject null hypothesis at significance level of alpha.&#8221;</li>
<li>H=1 =&gt; &#8220;Reject null hypothesis at significance level of alpha.&#8221;</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">References:</th><td class="field-body">[1] E. Kreyszig, &#8220;Introductory Mathematical Statistics&#8221;,
John Wiley, 1970, page 206.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.testclustnew">
<code class="descclassname">Statistics_tools.</code><code class="descname">testclustnew</code><span class="sig-paren">(</span><em>X</em>, <em>clust</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.testclustnew" title="Permalink to this definition">¶</a></dt>
<dd><p>This function tests for the significance of the cluster solution</p>
<p>Usage</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">bestpval</span><span class="p">,</span><span class="n">bestmyclass</span><span class="p">,</span><span class="n">bestnames</span><span class="p">,</span><span class="n">bestX</span><span class="p">,</span><span class="n">where</span><span class="p">,</span><span class="n">clustnames</span><span class="p">]</span><span class="o">=</span><span class="n">testclustnew</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">clust</span><span class="p">,[</span><span class="n">r</span><span class="p">],[</span><span class="n">nperm</span><span class="p">],[</span><span class="n">names</span><span class="p">],[</span><span class="n">remove</span><span class="p">],[</span><span class="n">linkagetype</span><span class="p">]);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first last">is the Group Space</p>
</dd>
<dt><strong>clust:</strong></dt>
<dd><p class="first last">is the cluster solutions (between how many and how many solutions is reasonable?)
the default for this option is 2:r/2 where r is the number of</p>
</dd>
<dt><strong>REGIONS:</strong></dt>
<dd><p class="first last">r is the number of DIMENSIONS in the solution (from choose_ndims)</p>
</dd>
<dt><strong>nperm:</strong></dt>
<dd><dl class="first last docutils">
<dt>is the number of permutations for nonparametric testing</dt>
<dd><p class="first last">(default 1000)</p>
</dd>
</dl>
</dd>
<dt><strong>names:</strong></dt>
<dd><p class="first last">specifies the names of each region</p>
</dd>
<dt><strong>remove:</strong></dt>
<dd><p class="first">specifies what to do about elements which fit the cluster
solution poorly.  There are 3 possibilities:</p>
<p>&#8216;keep&#8217; - keeps all elements regardless of quality.  If you choose
this option, you are assuming that every region you enter
contributes something to your solution
&#8216;thresh&#8217; - this removes elements which fail to reach 95%
confidence as determined by random permutation testing</p>
<p class="last">&#8216;iter&#8217;  - this option removes elements which fall in the bottom
5% of the permuted distribution, and remcomputes the solution
without these elements. This iterative recomputation means that
you are pruning your solution <em>until</em> you get a good one.  The
p-values which accompany each solution are thus difficult to
intepret if you use this option</p>
</dd>
<dt><strong>linkagetype:</strong></dt>
<dd><p class="first">is the input to linkage (&#8216;single&#8217;,&#8217;average&#8217;,etc)</p>
<p class="last">see help linkage_t</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>bestpval:</strong></dt>
<dd><p class="first last">is the overall significance of the derived solution.
significance is calculated by permuting the group space in all
dimensions to derive a completely new configuration of elements in
n-dimensional space.  These elements are then assigned to clusters
exactly as for the point estimate solution, and the mean quality
(silhouette value) across the entire plot is used to form a
distribution against which to test the sihouette value of the true
solution</p>
</dd>
<dt><strong>bestmyclass:</strong></dt>
<dd><p class="first last">is the assignment of REGIONS to CLUSTERS for the best
soluton</p>
</dd>
<dt><strong>bestnames:</strong></dt>
<dd><p class="first last">contains the names of the REGIONS included in the
best solution (if remove==keep, this will be the same as names);</p>
</dd>
<dt><strong>bestX:</strong></dt>
<dd><p class="first last">is the best group space</p>
</dd>
<dt><strong>where:</strong></dt>
<dd><p class="first last">is a vector indexing which of the original regions (in
order) are included in the new group space.
clustnames is a cell containing the names of the elements in each
cluster</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Calls:</th><td class="field-body"><p class="first last">getmeanquality,clustquality,pdist1,linkage_t,cluster_t,makebinary</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.time_varying_estimate">
<code class="descclassname">Statistics_tools.</code><code class="descname">time_varying_estimate</code><span class="sig-paren">(</span><em>meth</em>, <em>data</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.time_varying_estimate" title="Permalink to this definition">¶</a></dt>
<dd><p>Performs correlation operation (default)
or any function you pass in (as a function handle)
on symmetric moving average of data</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>y = time_varying_estimate(meth,data,[window width],[function handle])
</pre></div>
</div>
<p>Works on all columns of the data input matrix together 
so function handles can be multivariate</p>
<dl class="docutils">
<dt>Window options (meth argument):</dt>
<dd><ul class="first last simple">
<li>&#8216;gaussian&#8217;</li>
<li>&#8216;tukey&#8217;</li>
</ul>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Optional Input:</th><td class="field-body"><p class="first">stepby</p>
<p>Sometimes input data is very high resolution, and it would take too much
time to work element by element across the inputs.  You can enter an
option here to compute estimates at every n-th lag.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>y =  time_varying_estimate(&#39;gaussian&#39;,data,20);

% Generate sample data:
x = mvnrnd([0 0], [1 .6; .6 1], 100);

% Correlation between columns of x:
r = time_varying_estimate(&#39;tukey&#39;, x, 20);

% St. deviation of first column
mystd = time_varying_estimate(&#39;tukey&#39;, x(:, 1), 20, @(y) std(y));
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.tscv">
<code class="descclassname">Statistics_tools.</code><code class="descname">tscv</code><span class="sig-paren">(</span><em>vectorlen</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.tscv" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a crossvalidation test and train index for time series data.
Larger h will ensure less dependence.  Larger v creates larger test sets
Number of folds = vectorlen - (2*h + 2*v)</p>
<p>-See <a class="reference external" href="http://robjhyndman.com/hyndsight/tscvexample/">http://robjhyndman.com/hyndsight/tscvexample/</a> for more info about rolling cv
-See Racine, J. (2000). Consistent cross-validatory model-selection for dependent data: hv-block cross-validation. Journal of Econometrics, 99(1), 39-61.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">trIdx</span><span class="p">,</span> <span class="n">teIdx</span><span class="p">]</span> <span class="o">=</span> <span class="n">tscv</span><span class="p">(</span><span class="n">vectorlen</span><span class="p">,</span> <span class="n">stepsize</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>vectorlen:</strong></dt>
<dd><p class="first last">length of vector to create holdout cross-validation set</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs with their default values:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;hvblock&#8217; = [h,v]:</strong></dt>
<dd><p class="first last">use hvblock cross-validation with a block
size of &#8216;h&#8217; (0 reduces to v-fold xval)and
number of test observations &#8216;v&#8217; (0 reduces
to h-block xval)</p>
</dd>
<dt><strong>&#8216;rolling&#8217; = [h,v,g]:</strong></dt>
<dd><p class="first last">use hvblock cross-validation with g training steps
surrounding hv block.  Akin to Rolling
crossval. Same properties as hvblock.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>trIdx:</strong></dt>
<dd><p class="first last">structure with training label index</p>
</dd>
<dt><strong>teIdx:</strong></dt>
<dd><p class="first last">structure with test label index</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[trIdx, teIdx] = tscv(100, &#39;hvblock&#39;,[5,2]); % use hvblock with h=5 and v=2
[trIdx, teIdx] = tscv(100, &#39;rolling&#39;,[5,2,10]); % use hvblock with h=5, v=2 and g=10
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.tsquaretest">
<code class="descclassname">Statistics_tools.</code><code class="descname">tsquaretest</code><span class="sig-paren">(</span><em>X</em>, <em>pthresh</em>, <em>u0</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.tsquaretest" title="Permalink to this definition">¶</a></dt>
<dd><p>Hotelling&#8217;s t-square test that the multivariate sample mean of X is
different from u0.</p>
<p>X is an observations x variables data matrix
If pthresh is entered, the critical t2 value is returned
If u0 is not entered, the default null hypothesis is mean zero (the
origin)</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.ttest2_printout">
<code class="descclassname">Statistics_tools.</code><code class="descname">ttest2_printout</code><span class="sig-paren">(</span><em>sc1</em>, <em>sc2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.ttest2_printout" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">H</span><span class="p">,</span><span class="n">p</span><span class="p">,</span><span class="n">ci</span><span class="p">,</span><span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">ttest2_printout</span><span class="p">(</span><span class="n">sc1</span><span class="p">,</span><span class="n">sc2</span><span class="p">,[</span><span class="n">doplot</span><span class="p">],[</span><span class="n">covts</span><span class="p">])</span>
</pre></div>
</div>
<p>one or two sample t-test printout and plot
covariates are not done yet!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>sc1:</strong></dt>
<dd><p class="first last">data from first group</p>
</dd>
<dt><strong>sc2:</strong></dt>
<dd><p class="first last">data from second group (if missing or empty, performs one-sample
t-test)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.ttest3d">
<code class="descclassname">Statistics_tools.</code><code class="descname">ttest3d</code><span class="sig-paren">(</span><em>xc</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.ttest3d" title="Permalink to this definition">¶</a></dt>
<dd><p>calculate t-statistic values for a k x k x n 3-D matrix of
values across n subjects.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">mean</span><span class="p">,</span><span class="n">t</span><span class="p">,</span><span class="n">sig</span><span class="p">,</span><span class="n">out</span><span class="p">]</span> <span class="o">=</span> <span class="n">ttest3d</span><span class="p">(</span><span class="n">xc</span><span class="p">)</span>
</pre></div>
</div>
<p>stats on each element, across 3rd dim</p>
<p>to print output, see also:
correlation_to_text(mxc,sig);</p>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.var_prctile">
<code class="descclassname">Statistics_tools.</code><code class="descname">var_prctile</code><span class="sig-paren">(</span><em>p</em>, <em>n_in_sample</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.var_prctile" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">var_prc</span><span class="p">,</span><span class="n">pci</span><span class="p">,</span><span class="n">Nneeded</span><span class="p">,</span><span class="n">pcitarget</span><span class="p">]</span> <span class="o">=</span> <span class="n">var_prctile</span><span class="p">(</span><span class="n">p</span><span class="p">,</span><span class="n">n_in_sample</span><span class="p">,[</span><span class="s">&#39;nboot&#39;</span><span class="p">,</span><span class="n">nboot</span><span class="p">],[</span><span class="s">&#39;x&#39;</span><span class="p">,</span><span class="n">data</span><span class="p">])</span>

<span class="p">[</span><span class="n">var_prc</span><span class="p">,</span><span class="n">pci</span><span class="p">,</span><span class="n">Nneeded</span><span class="p">,</span><span class="n">pcitarget</span><span class="p">]</span> <span class="o">=</span> <span class="n">var_prctile</span><span class="p">(</span><span class="n">p</span><span class="p">,</span><span class="mi">50</span><span class="p">,</span><span class="s">&#39;nboot&#39;</span><span class="p">,</span><span class="mi">5000</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>p:</strong></dt>
<dd><p class="first last">is desired prctile of data (threshold)</p>
</dd>
<dt><strong>nboot:</strong></dt>
<dd><p class="first last">is number of bootstrap samples (if bootstrapping)</p>
</dd>
<dt><strong>n_in_sample:</strong></dt>
<dd><p class="first last">is number of obs. in original sample</p>
</dd>
<dt><strong>x:</strong></dt>
<dd><p class="first">is data sample of distribution of interest (empirical pdf based on
this)</p>
<p class="last">Note: using empirical PDF/CDF depends a great deal on choice of h (see
code).</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>Nneeded = [];
for p = [.05:-.001:.001]
    [var_prc,pci,Nneeded(end+1),pcitarget] = var_prctile(x,p);
end
figure;
plot([.05:-.001:.001],Nneeded)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.weighted_glmfit">
<code class="descclassname">Statistics_tools.</code><code class="descname">weighted_glmfit</code><span class="sig-paren">(</span><em>Y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.weighted_glmfit" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate weighted average using weighted linear least squares
See examples below for usage</p>
<p>Model:
Y_i = 1*Ypop + noise</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">data matrix (nsub x T)</p>
</dd>
<dt><strong>w:</strong></dt>
<dd><p class="first last">weights</p>
</dd>
<dt><strong>varY:</strong></dt>
<dd><p class="first last">variance of data at each time point (nsub x T) + var between</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Ymean:</strong></dt>
<dd><p class="first last">weighted mean of each column of Y</p>
</dd>
<dt><strong>dfe:</strong></dt>
<dd><p class="first last">error degrees of freedom, adjusted for inequality of variance
(Sattherwaite) and pooled across data columns</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Extended output in stats structure:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>stats.t:</strong></dt>
<dd><p class="first last">t-values for weighted t-test</p>
</dd>
<dt><strong>stats.p:</strong></dt>
<dd><p class="first last">2-tailed p-values for weighted t-test</p>
</dd>
<dt><strong>r:</strong></dt>
<dd><p class="first last">weighted correlation coeff across columns of Y</p>
</dd>
<dt><strong>xy:</strong></dt>
<dd><p class="first last">weighted covariance matrix</p>
</dd>
<dt><strong>v:</strong></dt>
<dd><dl class="first last docutils">
<dt>weighted variance estimates for each column of Y</dt>
<dd><ul class="first last simple">
<li>sqrt(v) is the standard error of the mean (or grp difference)</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>stats.fits:</strong></dt>
<dd><p class="first last">fits  for each group (Ymean by group), low contrast weight group then high</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Fastest if no stats are asked for.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Computation time:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt>For FULL stats report</dt>
<dd><ul class="first last simple">
<li>Triples from 500 -&gt; 1000 columns of Y, continues to increase</li>
</ul>
</dd>
</dl>
<p>For mean/dfe only, fast for full dataset (many columns of Y)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Basic multivariate stats for 1000 columns of dat, no weighting
Multivariate covariances are meaningful if cols of Y are organized, e.g.,
timeseries</p>
<div class="highlight-python"><div class="highlight"><pre>[means,stats] = weighted_glmfit(dat(:,1:1000));
</pre></div>
</div>
<p>The same, but return univariate stats only (good for large Y)</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">means</span><span class="p">,</span><span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">weighted_glmfit</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span><span class="s">&#39;uni&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>A weighted version, where we put in the weights, and with a design matrix too:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">means</span><span class="p">,</span><span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">weighted_glmfit</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">dat</span><span class="p">,</span><span class="s">&#39;uni&#39;</span><span class="p">,</span><span class="s">&#39;w&#39;</span><span class="p">,</span><span class="n">weights</span><span class="p">);</span>
</pre></div>
</div>
<p>A weighted version, where weights are determined from w/i subject variances:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">means</span><span class="p">,</span><span class="n">stats</span><span class="p">]</span> <span class="o">=</span> <span class="n">weighted_glmfit</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">dat</span><span class="p">,</span><span class="s">&#39;uni&#39;</span><span class="p">,</span><span class="s">&#39;vary&#39;</span><span class="p">,</span><span class="n">variances</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.xcorr_multisubject">
<code class="descclassname">Statistics_tools.</code><code class="descname">xcorr_multisubject</code><span class="sig-paren">(</span><em>data</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.xcorr_multisubject" title="Permalink to this definition">¶</a></dt>
<dd><p>Cross-correlation and partial correlation matrices for 3-D data, i.e., a cell array of subject data matrices</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>OUT = xcorr_multisubject(data, [optional inputs])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>data:</strong></dt>
<dd><p class="first last">A cell array, one cell per subject/replicate, of n x k
data to be inter-correlated.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;partialr&#8217;:</strong></dt>
<dd><p class="first last">Use partial correlations obtained via ridge regression (k = 1 fixed)</p>
</dd>
<dt><strong>&#8216;shift_by&#8217;:</strong></dt>
<dd><p class="first last">Followed by integer value for max number of time points to shift</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>OUT:</strong></dt>
<dd><p class="first last">A structure containing subject correlation matrices, the mean
matrix, and raw and FDR-thresholded group matrix</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% cl cell structure:
for i = 1:length(clpos_data), data{i} = cat(2, clpos_data{i}.timeseries); end

% cl structure:
for i = 1:size(cl(1).all_data, 2), for c = 1:length(cl), data{i}(:,c) = cl(c).all_data(:, i); end, end

% parcel_cl_avgs or clpos_data2 structure:
for i = 1:length(parcel_cl_avgs), for j = 1:N, data{j}(:,i) = parcel_cl_avgs(i).timeseries{j}; end, end

create_figure(&#39;Xcorr&#39;, 1, 3);
imagesc(OUT.stats.mean);
subplot(1, 3, 2);
imagesc(OUT.stats.sig);
subplot(1, 3, 3);
imagesc(OUT.stats.fdrsig);
colormap gray
</pre></div>
</div>
<p>Example of MDS and plotting total (not decomposed) relationships:</p>
<div class="highlight-python"><div class="highlight"><pre>OUT.stats.D = (1 - OUT.stats.mean) ./ 2;
[OUT.stats_mds.GroupSpace,OUT.stats_mds.obs,OUT.stats_mds.implied_dissim] = shepardplot(OUT.stats.D,[]);
OUT.stats_mds = nmdsfig_tools(&#39;cluster_solution&#39;,OUT.stats_mds, OUT.stats_mds.GroupSpace, 2:5, 1000, []);
nmdsfig(OUT.stats_mds.GroupSpace,&#39;classes&#39;,OUT.stats_mds.ClusterSolution.classes,&#39;names&#39;,OUT.stats_mds.names,&#39;sig&#39;,OUT.stats.fdrsig);
</pre></div>
</div>
<p>Example of MDS and plotting direct relationships:</p>
<div class="highlight-python"><div class="highlight"><pre>OUT.ridge = matrix_direct_effects_ridge(data);
D = OUT.ridge.mean; D(find(eye(size(D)))) = 1;
D = (D&#39; + D) ./ 2;
OUT.ridge.D = (1 - D) ./ 2;
[OUT.stats_mds.GroupSpace,OUT.stats_mds.obs,OUT.stats_mds.implied_dissim] = shepardplot(OUT.ridge.D,[]);
OUT.stats_mds = nmdsfig_tools(&#39;cluster_solution&#39;,OUT.stats_mds, OUT.stats_mds.GroupSpace, 2:10, 1000, []);
nmdsfig(OUT.stats_mds.GroupSpace,&#39;classes&#39;,OUT.stats_mds.ClusterSolution.classes,&#39;names&#39;,OUT.stats_mds.names,&#39;sig&#39;,OUT.ridge.fdrsig);
hh = nmdsfig_fill(OUT.stats_mds);
axis image, axis equal

OUT.stats_mds = nmdsfig_tools(&#39;cluster_solution&#39;,OUT.stats_mds, OUT.stats_mds.GroupSpace, 2:5, 1000, []);

% data is cell, one cell per subject
[OUT.stats_mds.GroupSpace,OUT.stats_mds.obs,OUT.stats_mds.implied_dissim] = shepardplot(OUT.stats.D,[]);
OUT.stats_mds = nmdsfig_tools(&#39;cluster_solution&#39;,OUT.stats_mds, OUT.stats_mds.GroupSpace, 2:5, 1000, []);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Statistics_tools.xcorr_xy_multisubject">
<code class="descclassname">Statistics_tools.</code><code class="descname">xcorr_xy_multisubject</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em><span class="sig-paren">)</span><a class="headerlink" href="#Statistics_tools.xcorr_xy_multisubject" title="Permalink to this definition">¶</a></dt>
<dd><p>This function will cross-correlate two variables, X and Y for each of N
subjects.  Correlation and latency values will be saved.</p>
<p>Second-level tests are done across the N subjects on each of the
correlation and latency values.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first">must be an observations x N matrix</p>
<p>NOTE: observations are assumed to be timeseries values!</p>
<p class="last">this DOES matter because an AR(2) model is used...see below</p>
</dd>
<dt><strong>Y:</strong></dt>
<dd><p class="first">must be the same.</p>
<p class="last">cross-correlations will be computed for pairs of columns, separately for 
each successive column of X / Y</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This function ues shift_correl.m, with a two-pass procedure.</p>
<p>The first pass provides initial estimates, including an estimate of the latency standard
deviation, which is used as an Empirical Bayes prior.  The second pass is used to
apply the EBayes priors and estimate latencies and cross-correlation values.
An ar(2) model is used to estimate the DF.</p>
<p>The estimates will be biased towards zero in the second pass, but they
will likely be lower variance.</p>
<p>stats.latency and stats.association return group-level stats on these
computed using a sign pemutation test (to avoid normality assumption).
the &#8220;association&#8221; test is a test of the relationship, but is different
than a standard multilevel model because it uses correlation values rather
than slope values, and the sign permutation test.</p>
</dd></dl>

</div>
<div class="section" id="visualization-functions">
<h2>Visualization_functions<a class="headerlink" href="#visualization-functions" title="Permalink to this headline">¶</a></h2>
<span class="target" id="module-Visualization_functions"></span><dl class="function">
<dt id="Visualization_functions.addbrain">
<code class="descclassname">Visualization_functions.</code><code class="descname">addbrain</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.addbrain" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>handle = addbrain([method],enter 2nd arg to suppress lighting changes)
</pre></div>
</div>
<p>quick function to add transparent brain surface to figure</p>
<div class="highlight-python"><div class="highlight"><pre>han = addbrain;   % lateral surface

han = addbrain(&#39;brainstem&#39;);
</pre></div>
</div>
<p>NOTE: this version uses structures in SPM2 space (Colin atlas)
Available keywords:</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">CORTICAL SURFACES:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;transparent_surface&#8217;:</strong></dt>
<dd><p class="first last">the default.  2 mm res SPM2 brain surface</p>
</dd>
<dt><strong>&#8216;hires&#8217;:</strong></dt>
<dd><p class="first last">a high-resolution surface (from Caret segmentation)</p>
</dd>
<dt><strong>&#8216;hires left&#8217;:</strong></dt>
<dd><p class="first last">hi-resolution left medial with cerebellum (Caret seg)</p>
</dd>
<dt><strong>&#8216;hires right&#8217;:</strong></dt>
<dd><p class="first last">same, right hem</p>
</dd>
<dt><strong>&#8216;left&#8217;:</strong></dt>
<dd><p class="first last">2 mm resolution left hem, no cerebellum</p>
</dd>
</dl>
<p><strong>&#8216;right&#8217;:</strong></p>
<p><strong>&#8216;vmpfc&#8217;:</strong></p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">CUTAWAY SURFACES:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first"><strong>&#8216;brainbottom&#8217;:</strong></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">COMPOSITES:</th><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;limbic&#8217;:</strong></dt>
<dd><p class="first last">A collection of subcortical nuclei with left surface</p>
</dd>
<dt><strong>&#8216;foursurfaces&#8217;:</strong></dt>
<dd><p class="first last">Lateral and medial views, with brainstem</p>
</dd>
<dt><strong>&#8216;BG&#8217;:</strong></dt>
<dd><p class="first last">Basal ganglia</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">SUBCORTICAL SURFACES:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><ul class="first simple">
<li>&#8216;brainstem&#8217;</li>
<li>&#8216;amygdala&#8217;</li>
<li>&#8216;thalamus&#8217;</li>
<li>&#8216;hippocampus&#8217;</li>
<li>&#8216;midbrain&#8217;</li>
<li>&#8216;caudate&#8217;</li>
<li>&#8216;globus pallidus&#8217;</li>
<li>&#8216;putamen&#8217;</li>
<li>&#8216;nucleus accumbens&#8217;</li>
<li>&#8216;hypothalamus&#8217;</li>
<li>&#8216;cerebellum&#8217;</li>
<li>{&#8216;md&#8217;,&#8217;mediodorsal&#8217;}</li>
<li>{&#8216;cm&#8217;,&#8217;centromedian&#8217;}</li>
<li>&#8216;pbn&#8217;</li>
<li>&#8216;rvm&#8217;</li>
<li>&#8216;nts&#8217;</li>
<li>&#8216;lc&#8217;</li>
<li>{&#8216;sn&#8217;, &#8216;substantia nigra&#8217;}</li>
<li>{&#8216;stn&#8217;, &#8216;subthalamic nucleus&#8217;}</li>
<li>{&#8216;rn&#8217;, &#8216;red nucleus&#8217;}</li>
<li>{&#8216;olive&#8217;, &#8216;inferior olive&#8217;}</li>
<li>{&#8216;nrm&#8217;, &#8216;raphe magnus&#8217;}</li>
</ul>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">SPECIAL COMMANDS:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>han = addbrain(&#8216;colorchange&#8217;,my_rgb_color,han);</p>
<dl class="docutils">
<dt>Change gray background to some other color, excluding blobs already rendered</dt>
<dd><ul class="first last simple">
<li>han: Input handles with patch object</li>
<li>my_rgb_color: [x x x] color triplet</li>
<li>Only works for changing from gray background right now.</li>
</ul>
</dd>
</dl>
<p>han = addbrain(&#8216;eraseblobs&#8217;,han);</p>
<dl class="docutils">
<dt>Set all rendered blob colors back to gray; useful for re-rendering on existing surfaces.</dt>
<dd><ul class="first last simple">
<li>han: Input handles with patch object</li>
<li>Only works for changing to gray background right now.</li>
</ul>
</dd>
</dl>
<p>See also: cluster_surf, img2surf.m, surface() methods for objects, cluster_cutaways</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.addbrainleft">
<code class="descclassname">Visualization_functions.</code><code class="descname">addbrainleft</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.addbrainleft" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>handle = addbrainleft(enter arg to suppress lighting changes)
</pre></div>
</div>
<p>quick function to add transparent brain surface to figure</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.addbrainright">
<code class="descclassname">Visualization_functions.</code><code class="descname">addbrainright</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.addbrainright" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>handle = addbrain(enter arg to suppress lighting changes)
</pre></div>
</div>
<p>quick function to add transparent brain surface to figure</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.applycolormap">
<code class="descclassname">Visualization_functions.</code><code class="descname">applycolormap</code><span class="sig-paren">(</span><em>h</em>, <em>mapname</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.applycolormap" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="Visualization_functions.arrow">
<code class="descclassname">Visualization_functions.</code><code class="descname">arrow</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.arrow" title="Permalink to this definition">¶</a></dt>
<dd><p>ARROW  Draw a line with an arrowhead.</p>
<blockquote>
<div><dl class="docutils">
<dt>ARROW(Start,Stop) draws a line with an arrow from Start to Stop (points</dt>
<dd>should be vectors of length 2 or 3, or matrices with 2 or 3
columns), and returns the graphics handle of the arrow(s).</dd>
</dl>
<p>ARROW uses the mouse (click-drag) to create an arrow.</p>
<p>ARROW DEMO &amp; ARROW DEMO2 show 3-D &amp; 2-D demos of the capabilities of ARROW.</p>
<dl class="docutils">
<dt>ARROW may be called with a normal argument list or a property-based list.</dt>
<dd>ARROW(Start,Stop,Length,BaseAngle,TipAngle,Width,Page,CrossDir) is
the full normal argument list, where all but the Start and Stop
points are optional.  If you need to specify a later argument (e.g.,
Page) but want default values of earlier ones (e.g., TipAngle),
pass an empty matrix for the earlier ones (e.g., TipAngle=[]).</dd>
<dt>ARROW(&#8216;Property1&#8217;,PropVal1,&#8217;Property2&#8217;,PropVal2,...) creates arrows with the</dt>
<dd><p class="first">given properties, using default values for any unspecified or given as
&#8216;default&#8217; or NaN.  Some properties used for line and patch objects are
used in a modified fashion, others are passed directly to LINE, PATCH,
or SET.  For a detailed properties explanation, call ARROW PROPERTIES.</p>
<p>Start         The starting points.                     B
Stop          The end points.                         /<a href="#id59"><span class="problematic" id="id60">|\           ^
Length        Length of the arrowhead in pixels.     /|||</span></a>         |
BaseAngle     Base angle in degrees (ADE).          //|||\        L|
TipAngle      Tip angle in degrees (ABC).          ///|||\      e|
Width         Width of the base in pixels.        ////|||\\      n|
Page          Use hardcopy proportions.          /////<a href="#id61"><span class="problematic" id="id62">|D|</span></a>\\    g|
CrossDir      Vector || to arrowhead plane.     ////  |||  \\    t|
NormalDir     Vector out of arrowhead plane.   ///    |||    \  h|
Ends          Which end has an arrowhead.     //&lt;&#8212;&#8211;&gt;||      \   |
ObjectHandles Vector of handles to update.   /   base |||         V</p>
<blockquote class="last">
<div>E    angle||&lt;&#8212;&#8212;&#8211;&gt;C</div></blockquote>
</dd>
<dt>ARROW(H,&#8217;Prop1&#8217;,PropVal1,...), where H is a                 |||tipangle</dt>
<dd>vector of handles to previously-created arrows        |||
and/or line objects, will update the previously-      |||
created arrows according to the current view       &#8211;&gt;|A|&lt;&#8211; width
and any specified properties, and will convert
two-point line objects to corresponding arrows.  ARROW(H) will update
the arrows if the current view has changed.  Root, figure, or axes
handles included in H are replaced by all descendant Arrow objects.</dd>
</dl>
<p>A property list can follow any specified normal argument list, e.g.,
ARROW([1 2 3],[0 0 0],36,&#8217;BaseAngle&#8217;,60) creates an arrow from (1,2,3) to
the origin, with an arrowhead of length 36 pixels and 60-degree base angle.</p>
<p>The basic arguments or properties can generally be vectorized to create
multiple arrows with the same call.  This is done by passing a property
with one row per arrow, or, if all arrows are to have the same property
value, just one row may be specified.</p>
<p>You may want to execute AXIS(AXIS) before calling ARROW so it doesn&#8217;t change
the axes on you; ARROW determines the sizes of arrow components BEFORE the
arrow is plotted, so if ARROW changes axis limits, arrows may be malformed.</p>
<p>This version of ARROW uses features of MATLAB 5 and is incompatible with
earlier MATLAB versions (ARROW for MATLAB 4.2c is available separately);
some problems with perspective plots still exist.</p>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.bar_wani">
<code class="descclassname">Visualization_functions.</code><code class="descname">bar_wani</code><span class="sig-paren">(</span><em>y</em>, <em>e</em>, <em>bar_width</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.bar_wani" title="Permalink to this definition">¶</a></dt>
<dd><p>Draw a bar plot with error bars with some additional useful features 
(work with up to the 2014a matlab).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">bar_wani</span><span class="p">(</span><span class="n">y</span><span class="p">,</span> <span class="n">e</span><span class="p">,</span> <span class="n">bar_width</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>y:</strong></dt>
<dd><p class="first last">y values for bars (row: bar grouping, column: bar values 
within a bar group) (e.g., if there are m bar groups and 
n bars for each group, y should be a m x n matrix)</p>
</dd>
<dt><strong>e:</strong></dt>
<dd><p class="first last">error bars (m x n matrix)</p>
</dd>
<dt><strong>bar_width:</strong></dt>
<dd><p class="first last">value for bar width between 0 and 1. This will determine
the intervals between bars.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">Enter keyword followed by variable with values</p>
<dl class="docutils">
<dt><strong>&#8216;ylim&#8217;:</strong></dt>
<dd><p class="first last">y axis range, (e.g., &#8216;ylim&#8217;, [-1 1])</p>
</dd>
<dt><strong>&#8216;ytick&#8217;:</strong></dt>
<dd><p class="first last">y tick values (e.g., &#8216;ytick&#8217;, -.002:.001:.002)</p>
</dd>
<dt><strong>&#8216;errbar_width&#8217;:</strong></dt>
<dd><p class="first last">the horizontal width of error bars (e.g., &#8216;errbar_width&#8217;, [-.01 .01])</p>
</dd>
<dt><strong>&#8216;colors&#8217;:</strong></dt>
<dd><p class="first last">bar colors: each row determines the color for each bar in order (n x 3 matrix)</p>
</dd>
<dt><strong>&#8216;ast&#8217;:</strong></dt>
<dd><p class="first last">put asterisks according to p values, which should be
given. (e.g., &#8216;ast&#8217;, p [m x n]) <a href="#id49"><span class="problematic" id="id50">*</span></a>p&lt;.05, <a href="#id51"><span class="problematic" id="id52">**</span></a>p&lt;.01, <a href="#id53"><span class="problematic" id="id54">**</span></a><a href="#id55"><span class="problematic" id="id56">*</span></a>p&lt;.001</p>
</dd>
<dt><strong>&#8216;btwlines&#8217;:</strong></dt>
<dd><p class="first last">this option puts lines between bar groups. This is
followed by the line style (e.g., &#8216;btwlines&#8217;, &#8216;&#8211;&#8217;);</p>
</dd>
<dt><strong>&#8216;dosave&#8217;:</strong></dt>
<dd><p class="first last">followed by savename (e.g., &#8216;dosave&#8217;, savename)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Some advanced options:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;scatter&#8217;:</strong></dt>
<dd><p class="first last">show individual data points, which should be in cell array</p>
</dd>
<dt><strong>&#8216;text&#8217;:</strong></dt>
<dd><p class="first last">this will put a number for each bar (e.g., &#8216;text&#8217;, round(y) [m x n])</p>
</dd>
<dt><strong>&#8216;ast_adj_y_pos&#8217;:</strong></dt>
<dd><p class="first last">When the asterisk locations (on y axis) for bars with positive
values are off, you can adjust it using this option</p>
</dd>
<dt><strong>&#8216;ast_adj_y_neg&#8217;:</strong></dt>
<dd><p class="first last">When the asterisk locations (on y axis) for bars with negative
values are off, you can adjust it using this option</p>
</dd>
<dt><strong>&#8216;ast_adj_x&#8217;:</strong></dt>
<dd><p class="first last">When the asterisk locations (on x axis) are off, you 
can adjust it using this option</p>
</dd>
<dt><strong>&#8216;bar_edgecol&#8217;:</strong></dt>
<dd><p class="first last">You can use different colors for bar edges (col [n x 3 matrix])</p>
</dd>
<dt><strong>&#8216;bar_edgewidth&#8217;:</strong></dt>
<dd><p class="first last">You can change linewidths for bar edges</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>h:</strong></dt>
<dd><p class="first last">graphic handles for a bar plot</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"><p class="first last">you can see the output in</p>
</td>
</tr>
</tbody>
</table>
<p><a class="reference external" href="http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery">http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery</a>
:Examples:</p>
<div class="highlight-python"><div class="highlight"><pre>% data
y = [-0.6518   -0.6934   -0.5417   -0.6496   -0.5946   -0.3839
    1.1511    0.9090    1.1681    1.2892    0.9346    1.1383];
e = [0.3226    0.2936    0.3080    0.3203    0.3368    0.3167
    0.4026    0.4088    0.4012    0.5586    0.3734    0.4257];
p = [0.0433    0.0182    0.0785    0.0426    0.0775    0.2255
    0.0042    0.0262    0.0036    0.0210    0.0123    0.0075];

col =  [0    0.1157    0.2686
        0.1157    0.2765    0.4725
        0.4843    0.1157    0.1078
        0.3667    0.4765    0.1353
        0.2765    0.1902    0.3824
        0.0922    0.4216    0.5118
        0.7941    0.3235   0];

% draw
bar_wani(y, e, .8, &#39;colors&#39;, col, &#39;errbar_width&#39;, [0 0], &#39;ast&#39;, p, &#39;ylim&#39;, [-2.5 2.5], &#39;ytick&#39;, -2:2, &#39;ast_adj_x&#39;, 0, &#39;ast_adj_y_neg&#39;, .15);
set(gca, &#39;ytickLabel&#39;, num2str(get(gca, &#39;ytick&#39;)&#39;));
set(gcf, &#39;position&#39;, [1   531   399   169]);

savename = &#39;example_barwani.pdf&#39;;

try
    pagesetup(gcf);
    saveas(gcf, savename);
catch
    pagesetup(gcf);
    saveas(gcf, savename);   
end
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.barplot_colored">
<code class="descclassname">Visualization_functions.</code><code class="descname">barplot_colored</code><span class="sig-paren">(</span><em>data</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.barplot_colored" title="Permalink to this definition">¶</a></dt>
<dd><p>Make a barplot of data with error bars, with colors specified by colormap
or color string.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[h, axh]=barplot_colored(data, [optional arguments])
</pre></div>
</div>
<p>this is a good function though
within-subject error bars now added; use &#8216;within&#8217;</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">bar_handles</span><span class="p">,</span> <span class="n">axis_handle</span><span class="p">]</span><span class="o">=</span><span class="n">barplot_colored</span><span class="p">(</span><span class="n">data</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Input arguments:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">Optional</p>
<dl class="docutils">
<dt><strong>&#8216;within&#8217;:</strong></dt>
<dd><p class="first last">Do within-subject STE bars, average obs x variable interaction
Loftus and Masson 1994 style.</p>
</dd>
</dl>
<p>Strings, followed by values for each:</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">COLOR CONTROL:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<dl class="docutils">
<dt><strong>&#8216;colormap&#8217;:</strong></dt>
<dd><p class="first last">followed by colormap name to use</p>
</dd>
<dt><strong>&#8216;colors&#8217;:</strong></dt>
<dd><p class="first last">followed by cell array of colors per bar; supercedes colormap</p>
</dd>
</dl>
<table class="last docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Display items:</th><td class="field-body"><ul class="first simple">
<li>&#8216;fontsize&#8217;</li>
<li>&#8216;title&#8217;</li>
<li>&#8216;XTickLabels&#8217;</li>
<li>&#8216;ylabel&#8217;</li>
<li>&#8216;xlabel&#8217;</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Bar Locations:</th><td class="field-body"><ul class="first last simple">
<li>&#8216;x&#8217; : followed by x values for bars (locations)</li>
</ul>
</td>
</tr>
</tbody>
</table>
</td>
</tr>
</tbody>
</table>
<p>NOTE: For this function, keywords must be even-numbered argument entries,
e.g., arg 2, 4, 6.  Odd argument entries are values.</p>
<p>For example: This works, and you need the extra empty arg after &#8216;within&#8217;</p>
<div class="highlight-python"><div class="highlight"><pre>[h1, s1] = barplot_colored(pexp1, &#39;within&#39;, &#39; &#39;, &#39;title&#39;, &#39;Pattern expression&#39;, &#39;XTickLabels&#39;, dat.Y_names, &#39;x&#39;, 1:nterms);
</pre></div>
</div>
<p>You can assign arbitrary colors to bars by setting the colormap:</p>
<div class="highlight-python"><div class="highlight"><pre>[h, s] = barplot_colored([corr_temp corr_rep]);
cm = [1 .5 0; .5 0 1];
colormap(cm)
</pre></div>
</div>
<p>Example: A grouped barplot</p>
<div class="highlight-python"><div class="highlight"><pre>dat = rand(20, 4);
create_figure(&#39;bars&#39;);
[h1, s1] = barplot_colored(dat, &#39;x&#39;, [1 2 4 5]);
% set(h2, &#39;BarWidth&#39;, .9)
</pre></div>
</div>
<p>Change colormap:</p>
<div class="highlight-python"><div class="highlight"><pre>[h1, s1] = barplot_colored(dat, &#39;x&#39;, [1 2 4 5], &#39;colormap&#39;, &#39;summer&#39;);
</pre></div>
</div>
<p>Enter values:</p>
<div class="highlight-python"><div class="highlight"><pre>colors = {[.8 .25 .25] [.8 .5 .25] [.4 .5 .8] [.25 .25 .9]};
[h1, s1] = barplot_colored(dat, &#39;x&#39;, [1 2 4 5], &#39;colors&#39;, colors);
</pre></div>
</div>
<p>Set X Tick Label:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">h1</span><span class="p">,</span> <span class="n">s1</span><span class="p">]</span> <span class="o">=</span> <span class="n">barplot_colored</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;XTicklabels&#39;</span><span class="p">,</span> <span class="p">{</span><span class="s">&#39;A&#39;</span> <span class="s">&#39;B&#39;</span> <span class="s">&#39;C&#39;</span> <span class="s">&#39;D&#39;</span><span class="p">});</span>
</pre></div>
</div>
<p>See also: barplot_columns, lineplot_columns</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.barplot_columns">
<code class="descclassname">Visualization_functions.</code><code class="descname">barplot_columns</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.barplot_columns" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[axishandle,adjusted data,x-data, barhandle] = barplot_columns(dat, [other optional arguments])
</pre></div>
</div>
<p>This function makes a barplot of columns of data, with standard error
bars.  Optional arguments include removing continuous covariates before plotting,
robust (IRLS) estimation of means and correlations with covariates, and
within-subject error bars based on the subject x condition interaction
(overall), which is not quite the standard error contrasts of interest,
but is the standard error for a 1-way repeated measures ANOVA.</p>
<p>plots circles around points at z &gt;= 1.96</p>
<p>plots individual points, unless you enter 4th argument</p>
<p>if dat is a cell array, each entry becomes one &#8220;bar&#8221;.  Useful if n
observations is different for each column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">Just plot means and SE</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">barplot_columns</span><span class="p">(</span><span class="n">tmp</span><span class="p">,</span><span class="s">&#39;Cluster 1&#39;</span><span class="p">,[],</span><span class="mi">1</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional arguments:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><ol class="first arabic">
<li><p class="first">Title for figure</p>
</li>
<li><p class="first">covariates</p>
</li>
<li><dl class="first docutils">
<dt>String Arguments</dt>
<dd><ul class="first last">
<li><p class="first">&#8216;nofig&#8217; : do not make figure</p>
</li>
<li><p class="first">&#8216;noind&#8217; : do not plot individual scores</p>
</li>
<li><p class="first">&#8216;plotout&#8217;: circle potential outliers at z&gt;1.96 in red</p>
</li>
<li><p class="first">&#8216;dorob&#8217; : do robust IRLS means and correlations</p>
</li>
<li><p class="first">&#8216;dolines&#8217; : plot lines showing individual effects</p>
</li>
<li><dl class="first docutils">
<dt>&#8216;within&#8217; <span class="classifier-delimiter">:</span> <span class="classifier">within-subjects standard errors, followed by contrast</span></dt>
<dd><p class="first last">matrix</p>
</dd>
</dl>
</li>
<li><p class="first">&#8216;95CI&#8217;   : error bars are 95% CI instead of SE</p>
</li>
<li><p class="first">&#8216;line&#8217; : Make line plot instead of bar plot</p>
</li>
<li><p class="first">&#8216;number&#8217; : plot case numbers instead of points</p>
</li>
<li><p class="first">&#8216;x&#8217; : followed by x-axis values for bars</p>
</li>
<li><dl class="first docutils">
<dt>&#8216;color&#8217; <span class="classifier-delimiter">:</span> <span class="classifier">followed by color for bars (text: &#8216;r&#8217; or [r g b]) OR</span></dt>
<dd><p class="first last">cell array with names of colors cell for each line/bar</p>
</dd>
</dl>
</li>
<li><p class="first">&#8216;violin&#8217;: add violin plot to each bar, with data points</p>
</li>
</ul>
</dd>
</dl>
</li>
</ol>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">barplot_columns</span><span class="p">(</span><span class="n">ctmp</span><span class="p">,</span><span class="s">&#39;RT effects by Switch Type&#39;</span><span class="p">,</span><span class="n">overall_sw</span><span class="p">,</span><span class="s">&#39;nofig&#39;</span><span class="p">,</span><span class="s">&#39;dorob&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Standard Errors ARE NOT Adjusted for covariate, right now.</p>
<p>Example: within-subjects std. errors</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">barplot_columns</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;Means&#39;</span><span class="p">,</span> <span class="p">[],</span> <span class="s">&#39;nofig&#39;</span><span class="p">,</span> <span class="s">&#39;within&#39;</span><span class="p">,</span> <span class="n">c</span><span class="p">);</span>
</pre></div>
</div>
<p>The example below uses color, width, and xposition arguments to make a grouped</p>
<div class="highlight-python"><div class="highlight"><pre>barplot showing effects for two groups:
exp_dat = EXPT.error_rates(EXPT.group==1,:);
control_dat = EXPT.error_rates(EXPT.group==-1,:);
barplot_columns(exp_dat, &#39;Error rates&#39;, [], &#39;nofig&#39;, &#39;noind&#39;, &#39;color&#39;, &#39;r&#39;,&#39;width&#39;, .4);
barplot_columns(control_dat, &#39;Error rates&#39;, [], &#39;nofig&#39;, &#39;noind&#39;, &#39;color&#39;, &#39;b&#39;,&#39;width&#39;, .4, &#39;x&#39;, (1:9)+.5);
set(gca, &#39;XLim&#39;, [0 10], &#39;XTick&#39;, 1:9)

barplot_columns(nps_by_study, &#39;NPS by study&#39;, [], &#39;doind&#39;, &#39;colors&#39;, mycolors, &#39;nofig&#39;);

create_figure(&#39;example_plots&#39;, 1, 4);

Y{:,1} = rand(20,1);
Y{:,2} = rand(100,1);

[h, L, MX, MED, bw, F, U] = violinplot(Y,&#39;facecolor&#39;,[1 .5 0; 0 .5 1],&#39;edgecolor&#39;,[1 .5 0; 0 .5 1].*.75,&#39;mc&#39;, [1 .5 0].*.5, &#39;x&#39;, [1 3], &#39;medc&#39;, []);
title(&#39;Violinplot.m&#39;, &#39;FontSize&#39;, 16)

subplot(1, 4, 2)
barplot_columns(Y, &#39;nofig&#39;)
title(&#39;barplot\_columns.m default&#39;, &#39;FontSize&#39;, 16)

subplot(1, 4, 3)
barplot_columns(Y, &#39;nofig&#39;, &#39;violin&#39;, &#39;colors&#39;, {[1 .5 0] [0 .5 1]})
title(&#39;barplot\_columns.m colored&#39;, &#39;FontSize&#39;, 16)

subplot(1, 4, 4)

Y{:,1} = randn(50,1) + 5;
Y{:,2} = Y{1} + .3 * randn(50,1) + 3; 

barplot_columns(Y, &#39;nofig&#39;, &#39;noviolin&#39;, &#39;colors&#39;, {[1 .5 0] [0 .5 1]}, &#39;dolines&#39;)
title(&#39;barplot\_columns.m parallel coords&#39;, &#39;FontSize&#39;, 16)
</pre></div>
</div>
<p>See also: lineplot_columns, barplot_colored, line_plot_multisubject, violinplot
..</p>
<blockquote>
<div>Defaults</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.barplot_columns2">
<code class="descclassname">Visualization_functions.</code><code class="descname">barplot_columns2</code><span class="sig-paren">(</span><em>dat</em>, <em>plottitle</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.barplot_columns2" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[axishandle,adjusted data,x-data,barhandle] = barplot_columns(dat,title,[options])
</pre></div>
</div>
<p>Makes a barplot of columns of data, with standard error bars.
Optional arguments include removing continuous covariates before plotting,
robust (IRLS) estimation of means and correlations with covariates, and
within-subject error bars based on the subject x condition interaction
(overall), which is not quite the standard error contrasts of interest,
but is the standard error for a 1-way repeated measures ANOVA.</p>
<p>plots circles around points at z &gt;= 1.96
plots individual points, unless you enter 4th argument</p>
<p>if dat is a cell array, each entry becomes one &#8220;bar&#8221;.  Useful if n
observations is different for each column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">Just plot means and SE</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">barplot_columns</span><span class="p">(</span><span class="n">tmp</span><span class="p">,</span><span class="s">&#39;Cluster 1&#39;</span><span class="p">,[],</span><span class="mi">1</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;cov&#8217;:</strong></dt>
<dd><p class="first last">followed by matrix of covariates</p>
</dd>
<dt><strong>&#8216;labels&#8217;:</strong></dt>
<dd><p class="first last">followed by cellstring of bar labels</p>
</dd>
<dt><strong>&#8216;xlabelslant&#8217;:</strong></dt>
<dd><p class="first last">followed by number of degrees to slant bar labels
(DEFAULT: 45)</p>
</dd>
<dt><strong>&#8216;nofig&#8217;:</strong></dt>
<dd><p class="first last">do not make figure</p>
</dd>
<dt><strong>&#8216;ind&#8217;:</strong></dt>
<dd><p class="first last">plot individual scores on top of bars</p>
</dd>
<dt><strong>&#8216;plotout&#8217;:</strong></dt>
<dd><p class="first last">circle potential outliers at z&gt;1.96 in red</p>
</dd>
<dt><strong>&#8216;robust&#8217;:</strong></dt>
<dd><p class="first last">do robust IRLS means and correlations</p>
</dd>
<dt><strong>&#8216;indlines&#8217;:</strong></dt>
<dd><p class="first last">plot lines showing individual effects</p>
</dd>
<dt><strong>&#8216;within&#8217;:</strong></dt>
<dd><p class="first last">within-subjects standard errors, followed by contrast
matrix</p>
</dd>
<dt><strong>&#8216;line&#8217;:</strong></dt>
<dd><p class="first last">Make line plot instead of bar plot</p>
</dd>
<dt><strong>&#8216;number&#8217;:</strong></dt>
<dd><p class="first last">plot case numbers instead of points</p>
</dd>
<dt><strong>&#8216;x&#8217;:</strong></dt>
<dd><p class="first last">followed by x-axis values for bars</p>
</dd>
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd><p class="first last">followed by color for bars (text: &#8216;r&#8217; or [r g b]) OR
cell array with names of colors cell for each line/bar</p>
</dd>
<dt><strong>&#8216;denan&#8217;:</strong></dt>
<dd><p class="first last">remove rows that have NaNs</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">barplot_columns</span><span class="p">(</span><span class="n">ctmp</span><span class="p">,</span><span class="s">&#39;RT effects by Switch Type&#39;</span><span class="p">,</span><span class="n">overall_sw</span><span class="p">,</span><span class="s">&#39;nofig&#39;</span><span class="p">,</span><span class="s">&#39;robust&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Standard Errors ARE NOT Adjusted for covariate, right now.</p>
<p>Example: within-subjects std. errors</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">barplot_columns</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;Means&#39;</span><span class="p">,</span> <span class="s">&#39;nofig&#39;</span><span class="p">,</span> <span class="s">&#39;within&#39;</span><span class="p">,</span> <span class="s">&#39;ind&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>The example below uses color, width, and xposition arguments to make a grouped</p>
<div class="highlight-python"><div class="highlight"><pre>barplot showing effects for two groups:
exp_dat = EXPT.error_rates(EXPT.group==1,:);
control_dat = EXPT.error_rates(EXPT.group==-1,:);
barplot_columns(exp_dat, &#39;Error rates&#39;, &#39;nofig&#39;, &#39;color&#39;, &#39;r&#39;, &#39;width&#39;, .4);
barplot_columns(control_dat, &#39;Error rates&#39;, &#39;nofig&#39;, &#39;color&#39;, &#39;b&#39;, &#39;width&#39;, .4, &#39;x&#39;, (1:9)+.5);
set(gca, &#39;XLim&#39;, [0 10], &#39;XTick&#39;, 1:9)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.barplot_columns3">
<code class="descclassname">Visualization_functions.</code><code class="descname">barplot_columns3</code><span class="sig-paren">(</span><em>dat</em>, <em>plottitle</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.barplot_columns3" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[axishandle,adjusted data,x-data,barhandle] = barplot_columns(dat,title,[options])
</pre></div>
</div>
<p>Makes a barplot of columns of data, with standard error bars.
Optional arguments include removing continuous covariates before plotting,
robust (IRLS) estimation of means and correlations with covariates, and
within-subject error bars based on the subject x condition interaction
(overall), which is not quite the standard error contrasts of interest,
but is the standard error for a 1-way repeated measures ANOVA.</p>
<p>plots circles around points at z &gt;= 1.96</p>
<p>plots individual points, unless you enter 4th argument</p>
<p>if dat is a cell array, each entry becomes one &#8220;bar&#8221;.  Useful if n
observations is different for each column.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">Just plot means and SE</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">barplot_columns</span><span class="p">(</span><span class="n">tmp</span><span class="p">,</span><span class="s">&#39;Cluster 1&#39;</span><span class="p">,[],</span><span class="mi">1</span><span class="p">);</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;cov&#8217;:</strong></dt>
<dd><p class="first last">followed by matrix of covariates</p>
</dd>
<dt><strong>&#8216;labels&#8217;:</strong></dt>
<dd><p class="first last">followed by cellstring of bar labels</p>
</dd>
<dt><strong>&#8216;nofig&#8217;:</strong></dt>
<dd><p class="first last">do not make figure</p>
</dd>
<dt><strong>&#8216;ind&#8217;:</strong></dt>
<dd><p class="first last">plot individual scores on top of bars</p>
</dd>
<dt><strong>&#8216;plotout&#8217;:</strong></dt>
<dd><p class="first last">circle potential outliers at z&gt;1.96 in red</p>
</dd>
<dt><strong>&#8216;robust&#8217;:</strong></dt>
<dd><p class="first last">do robust IRLS means and correlations</p>
</dd>
<dt><strong>&#8216;indlines&#8217;:</strong></dt>
<dd><p class="first last">plot lines showing individual effects</p>
</dd>
<dt><strong>&#8216;within&#8217;:</strong></dt>
<dd><p class="first last">within-subjects standard errors, followed by contrast
matrix</p>
</dd>
<dt><strong>&#8216;line&#8217;:</strong></dt>
<dd><p class="first last">Make line plot instead of bar plot</p>
</dd>
<dt><strong>&#8216;number&#8217;:</strong></dt>
<dd><p class="first last">plot case numbers instead of points</p>
</dd>
<dt><strong>&#8216;x&#8217;:</strong></dt>
<dd><p class="first last">followed by x-axis values for bars</p>
</dd>
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd><p class="first last">followed by color for bars (text: &#8216;r&#8217; or [r g b]) OR
cell array with names of colors cell for each line/bar</p>
</dd>
<dt><strong>&#8216;cons&#8217;:</strong></dt>
<dd><p class="first last">followed by contrastXweights matrix</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>To convert from long form to wide form</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;subcol&#8217;:</strong></dt>
<dd>column numbers with subject numbers</dd>
<dt><strong>&#8216;propcols&#8217;:</strong></dt>
<dd>vector of column numbers with properties</dd>
<dt><strong>&#8216;propnames&#8217;:</strong></dt>
<dd>cell array of names of properties</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">barplot_columns</span><span class="p">(</span><span class="n">ctmp</span><span class="p">,</span><span class="s">&#39;RT effects by Switch Type&#39;</span><span class="p">,</span><span class="n">overall_sw</span><span class="p">,</span><span class="s">&#39;nofig&#39;</span><span class="p">,</span><span class="s">&#39;robust&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Standard Errors ARE NOT Adjusted for covariate, right now.</p>
<p>Example: within-subjects std. errors</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">barplot_columns</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="s">&#39;Means&#39;</span><span class="p">,</span> <span class="s">&#39;nofig&#39;</span><span class="p">,</span> <span class="s">&#39;within&#39;</span><span class="p">,</span> <span class="s">&#39;ind&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>The example below uses color, width, and xposition arguments to make a grouped</p>
<div class="highlight-python"><div class="highlight"><pre>barplot showing effects for two groups:
exp_dat = EXPT.error_rates(EXPT.group==1,:);
control_dat = EXPT.error_rates(EXPT.group==-1,:);
barplot_columns(exp_dat, &#39;Error rates&#39;, &#39;nofig&#39;, &#39;color&#39;, &#39;r&#39;, &#39;width&#39;, .4);
barplot_columns(control_dat, &#39;Error rates&#39;, &#39;nofig&#39;, &#39;color&#39;, &#39;b&#39;, &#39;width&#39;, .4, &#39;x&#39;, (1:9)+.5);
set(gca, &#39;XLim&#39;, [0 10], &#39;XTick&#39;, 1:9)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.barplot_grouped">
<code class="descclassname">Visualization_functions.</code><code class="descname">barplot_grouped</code><span class="sig-paren">(</span><em>dat</em>, <em>X</em>, <em>xnames</em>, <em>seriesnames</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.barplot_grouped" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>han = barplot_grouped(dat,X,xnames,seriesnames, [optional args]);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first last">n x 4 data matrix (2 x 2 grouping only for now, but easy to
expand later)</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">covariates, no intercept; will be centered by this function</p>
</dd>
<dt><strong>xnames:</strong></dt>
<dd><p class="first last">x-axis labels</p>
</dd>
<dt><strong>seriesnames:</strong></dt>
<dd><p class="first last">series labels</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>First two bars are group, and last two bars are group</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">(keywords)</p>
<dl class="last docutils">
<dt><strong>&#8216;within&#8217;:</strong></dt>
<dd><p class="first last">within-error flag.  1 = errors based on subject x
condition interaction</p>
</dd>
<dt><strong>&#8216;stars&#8217;:</strong></dt>
<dd><p class="first last">put stars for significance on graph (default)</p>
</dd>
<dt><strong>&#8216;nostars&#8217;:</strong></dt>
<dd><p class="first last">do not plot stars</p>
</dd>
<dt><strong>&#8216;bars&#8217;:</strong></dt>
<dd><p class="first last">followed by number of bars in group</p>
</dd>
<dt><strong>&#8216;pvals&#8217;:</strong></dt>
<dd><p class="first last">followed by matrix of p-values (for stars; will calculate if missing)</p>
</dd>
<dt><strong>&#8216;inputmeans&#8217;:</strong></dt>
<dd><p class="first last">input means and errors in first 2 inputs rather than data
and X</p>
</dd>
<dt><strong>&#8216;colors&#8217;:</strong></dt>
<dd><p class="first last">followed by colors, e.g., mycol = {[1 0 0] [0 1 0] [1 0 1] [1 1 0] [0 0 1]}</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.barplotter">
<code class="descclassname">Visualization_functions.</code><code class="descname">barplotter</code><span class="sig-paren">(</span><em>data</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.barplotter" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">barplotter</span><span class="p">(</span><span class="n">data</span><span class="p">)</span>
</pre></div>
</div>
<p>Creates a barplot out of the columns or elements (if data is a cell
array containing vectors) of data.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;groups&#39;</span><span class="p">,</span> <span class="n">grouping</span><span class="p">)</span>
</pre></div>
</div>
<p>grouping must have the same number of columns or elements (if data is a
cell array) as data, and must consist of integers beginning with 1 and
ending with the total number of groups. Data belonging to the same group
will be &#8216;clustered&#8217; together in the plot. Ommitting groups (eg, [1 1 3 3])
will create extra spacing between groups.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;std&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>overrides the default behavior and plots standard deviation bars instead
of standard error bars</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;CI&#39;</span><span class="p">,</span> <span class="n">alpha</span><span class="p">)</span>
</pre></div>
</div>
<p>this will override the default behavior (as well as &#8216;std&#8217;) and plot
1-alpha confidence intervals.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;labels&#39;</span><span class="p">,</span> <span class="n">labels</span><span class="p">)</span>
</pre></div>
</div>
<p>labels must be a cell array of strings with the same number of elements
as data has columns or elements. The x-axis will be labeled with these.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;label_groups&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Applies labels to groups instead to individual bars</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;legend&#39;</span><span class="p">,</span> <span class="n">names</span><span class="p">)</span>
</pre></div>
</div>
<p>Plots a legend in the figure, with labels corresponding to the elements
of the cell vector of strings, names.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;plegend&#39;</span><span class="p">,</span> <span class="n">names</span><span class="p">,</span> <span class="n">p</span><span class="p">)</span>
</pre></div>
</div>
<p>As &#8216;legend&#8217;, above, but p is a vector of the barplots to include in the
legend (e.g. if you plotted 6 bars and p = [1 3], only the first and third
bar would be included in the legend.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;PlotLineHor&#39;</span><span class="p">,</span> <span class="n">value</span><span class="p">)</span>
</pre></div>
</div>
<p>Plots a horizontal line at the Y value indicated.</p>
<p>h = barplotter(..., &#8216;LinePlot&#8217;, colors, styles, markers)</p>
<p>Changes the graph from a bar graph to a line graph. Setting groups will
cause one line to be drawn for each group.</p>
<p>h = barplotter(..., &#8216;ErrorWidth&#8217;, errorwidth)</p>
<p>Sets the line weight (in points) for error bars</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span><span class="n">barplotter</span><span class="p">(</span><span class="o">...</span><span class="p">,</span> <span class="s">&#39;PropertyName&#39;</span><span class="p">,</span> <span class="n">PropertyValue</span><span class="p">)</span>
</pre></div>
</div>
<p>Properties correspond to various Matlab figure properties, as
appropriate. Currently supported properties (more to be added) are:</p>
<blockquote>
<div><ul>
<li><p class="first">&#8216;Title&#8217;</p>
</li>
<li><p class="first">&#8216;XLabel&#8217;</p>
</li>
<li><p class="first">&#8216;YLabel&#8217;</p>
</li>
<li><p class="first">&#8216;YLim&#8217;</p>
</li>
<li><p class="first">&#8216;XLim&#8217;</p>
</li>
<li><p class="first">&#8216;YTick&#8217;</p>
</li>
<li><p class="first">&#8216;YTickLabel&#8217;</p>
</li>
<li><p class="first">&#8216;YMinorTick&#8217;</p>
</li>
<li><p class="first">&#8216;FontSize&#8217;</p>
</li>
<li><p class="first">&#8216;xFontSize&#8217; %for xlabel</p>
</li>
<li><p class="first">&#8216;yFontSize&#8217; %for ylabel</p>
</li>
<li><p class="first">&#8216;tFontSize&#8217; %for title</p>
</li>
<li><dl class="first docutils">
<dt>&#8216;FaceColor&#8217; %note that you may specify a matrix of 3-element RGB vectors,</dt>
<dd><p class="first last">rather than a single vector. Barplotter will then cycle
the rows of the matrix until all bars have been drawn.</p>
</dd>
</dl>
</li>
<li><p class="first">&#8216;Colormap&#8217;</p>
</li>
<li><p class="first">&#8216;GridLineStyle&#8217; % - | - -| {:} | -. | none</p>
</li>
<li><p class="first">&#8216;TickDir&#8217; % in or out</p>
</li>
<li><p class="first">&#8216;MarkerSize&#8217;</p>
</li>
<li><p class="first">&#8216;LineWidth&#8217;</p>
</li>
</ul>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.canlab_force_directed_graph">
<code class="descclassname">Visualization_functions.</code><code class="descname">canlab_force_directed_graph</code><span class="sig-paren">(</span><em>activationdata</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.canlab_force_directed_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Creates a force-directed graph from a set of variables, and plots
clusters on 3-D brain as well if entered. Requires matlab BGL toolbox.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>canlab_force_directed_graph(activationdata OR connection matrix, [&#39;cl&#39;, cl])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>activationdata:</strong></dt>
<dd><p class="first">observations x variables matrix of data to be
inter-correlated</p>
<p>OR</p>
<p class="last">signed, thresholded connection matrix to be used (e.g.,
thresholded t-values from multi-subject group analysis</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">Enter keyword followed by variable with values</p>
<dl class="docutils">
<dt><strong>&#8216;cl&#8217;:</strong></dt>
<dd><p class="first last">followed by clusters or region structure with brain clusters</p>
</dd>
<dt><strong>&#8216;threshtype&#8217;:</strong></dt>
<dd><p class="first last">followed by threshold type; &#8216;bonf&#8217; is option now</p>
</dd>
</dl>
<p><strong>&#8216;connectmetric&#8217;:</strong></p>
<dl class="docutils">
<dt><strong>&#8216;sizescale&#8217;:</strong></dt>
<dd><p class="first last">Followed by values to use in sizing of nodes on graph</p>
</dd>
<dt><strong>&#8216;setcolors&#8217;:</strong></dt>
<dd><p class="first last">Cell array of colors for each group, [1 x g]</p>
</dd>
<dt><strong>&#8216;rset&#8217;:</strong></dt>
<dd><p class="first last">Cell of vectors, with indices (integers) of member
elements in each group, [1 x g] cell
rset can ALSO be a vector of integers, i.e., output
from clusterdata</p>
</dd>
</dl>
<p><strong>&#8216;names&#8217;:</strong></p>
<p><strong>&#8216;namesfield&#8217;:</strong></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>stats:</strong></dt>
<dd><p class="first last">structure with descriptive statistics, including
betweenness-centrality, degree of each node</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">stats</span><span class="p">,</span> <span class="n">handles</span><span class="p">]</span> <span class="o">=</span> <span class="n">canlab_force_directed_graph</span><span class="p">(</span><span class="n">activationdata</span><span class="p">,</span> <span class="s">&#39;cl&#39;</span><span class="p">,</span> <span class="n">cl</span><span class="p">,</span> <span class="s">&#39;namesfield&#39;</span><span class="p">,</span> <span class="s">&#39;shorttitle&#39;</span><span class="p">);</span>
<span class="p">[</span><span class="n">stats</span><span class="p">,</span> <span class="n">handles</span><span class="p">]</span> <span class="o">=</span> <span class="n">canlab_force_directed_graph</span><span class="p">(</span><span class="n">activationdata</span><span class="p">,</span> <span class="s">&#39;cl&#39;</span><span class="p">,</span> <span class="n">cl</span><span class="p">,</span> <span class="s">&#39;namesfield&#39;</span><span class="p">,</span> <span class="s">&#39;shorttitle&#39;</span><span class="p">,</span> <span class="s">&#39;degree&#39;</span><span class="p">);</span>
<span class="p">[</span><span class="n">stats</span><span class="p">,</span> <span class="n">handles</span><span class="p">]</span> <span class="o">=</span> <span class="n">canlab_force_directed_graph</span><span class="p">(</span><span class="n">activationdata</span><span class="p">,</span> <span class="s">&#39;cl&#39;</span><span class="p">,</span> <span class="n">cl</span><span class="p">,</span> <span class="s">&#39;namesfield&#39;</span><span class="p">,</span> <span class="s">&#39;shorttitle&#39;</span><span class="p">,</span> <span class="s">&#39;degree&#39;</span><span class="p">,</span> <span class="s">&#39;rset&#39;</span><span class="p">,</span> <span class="n">rset</span><span class="p">,</span> <span class="s">&#39;setcolors&#39;</span><span class="p">,</span> <span class="n">setcolors</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.canlab_results_fmridisplay">
<code class="descclassname">Visualization_functions.</code><code class="descname">canlab_results_fmridisplay</code><span class="sig-paren">(</span><em>input_activation</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.canlab_results_fmridisplay" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>canlab_results_fmridisplay(input_activation, [optional inputs])
</pre></div>
</div>
<p>purpose:  This function display fmri results.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>input_activation:</strong></dt>
<dd><p class="first">nii, img,</p>
<p>This image has the blobs you want to
display. You can also enter a cl &#8220;clusters&#8221; structure or
&#8220;region&#8221; object.</p>
<p class="last">you can also get a thresholded image like the examples used here
from a number of places - by thresholding your results in SPM
and using &#8220;write filtered&#8221; to save the image, by creating masks
from meta-analysis or anatomical atlases, or by using
mediation_brain_results, robust_results_threshold,
robust_results_batch_script, threshold_imgs, or object
oriented tools including fmri_data and statistic_image objects.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;noblobs&#8217;:</strong></dt>
<dd><p class="first last">do not display blobs</p>
</dd>
<dt><strong>&#8216;nooutline&#8217;:</strong></dt>
<dd><p class="first last">do not display blob outlines</p>
</dd>
<dt><strong>&#8216;addmontages&#8217;:</strong></dt>
<dd><p class="first last">when entering existing fmridisplay obj, add new montages</p>
</dd>
<dt><strong>&#8216;noremove&#8217;:</strong></dt>
<dd><p class="first last">do not remove current blobs when adding new ones</p>
</dd>
<dt><strong>&#8216;outlinecolor:</strong></dt>
<dd><p class="first last">followed by new outline color</p>
</dd>
<dt><strong>&#8216;splitcolor&#8217;:</strong></dt>
<dd><p class="first last">followed by 4-cell new split colormap colors (help fmridisplay or edit code for defaults as example)</p>
</dd>
<dt><strong>&#8216;montagetype&#8217;:</strong></dt>
<dd><p class="first">&#8216;full&#8217; for full montages of axial and sagg slices.</p>
<p>&#8216;compact&#8217; [default] for single-figure parasagittal and axials slices.</p>
<p class="last">&#8216;compact2&#8217;: like &#8216;compact&#8217;, but fewer axial slices.</p>
</dd>
<dt><strong>&#8216;noverbose&#8217;:</strong></dt>
<dd><p class="first last">suppress verbose output, good for scripts/publish to html, etc.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Other inputs to addblobs (fmridisplay method) are allowed, e.g., &#8216;cmaprange&#8217;, [-2 2], &#8216;trans&#8217;</p>
<p>See help fmridisplay
e.g., &#8216;color&#8217;, [1 0 0]</p>
<p>You can also input an existing fmridisplay object, and it will use the
one you have created rather than setting up the canonical slices.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example Script:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>input_activation = &#39;Pick_Atlas_PAL_large.nii&#39;;

% set up the anatomical underlay and display blobs
% (see the code of this function and help fmridisplay for more examples)

o2 = canlab_results_fmridisplay(input_activation);

%% ========== remove those blobs and change the color ==========

cl = mask2clusters(input_activation);
removeblobs(o2);
o2 = addblobs(o2, cl, &#39;color&#39;, [0 0 1]);

%% ========== OR

r = region(input_activation);
o2 = removeblobs(o2);
o2 = addblobs(o2, r, &#39;color&#39;, [1 0 0]);

%% ========== Create empty fmridisplay object on which to add blobs:
o2 = canlab_results_fmridisplay
o2 = canlab_results_fmridisplay([], &#39;compact2&#39;, &#39;noverbose&#39;);

%% ========== If you want to start over with a new fmridisplay object,
% make sure to clear o2, because it uses lots of memory

% This image should be on your path in the &quot;canlab_canonical_brains&quot; subfolder:

input_activation = &#39;pain-emotion_2s_z_val_FDR_05.img&#39;;
clear o2
close all
o2 = canlab_results_fmridisplay(input_activation);

%% ========== save PNGs of your images to insert into powerpoint, etc.
% for your paper/presentation

scn_export_papersetup(400);
saveas(gcf, &#39;results_images/pain_meta_fmridisplay_example_sagittal.png&#39;);

scn_export_papersetup(350);
saveas(gcf, &#39;results_images/pain_meta_fmridisplay_example_sagittal.png&#39;);

Change colors, removing old blobs and replacing with new ones:
o2 = canlab_results_fmridisplay(d, o2, &#39;cmaprange&#39;, [.3 .45], &#39;splitcolor&#39;, {[0 0 1] [.3 0 .8] [.9 0 .5] [1 1 0]}, &#39;outlinecolor&#39;, [.5 0 .5]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cl_line_plots">
<code class="descclassname">Visualization_functions.</code><code class="descname">cl_line_plots</code><span class="sig-paren">(</span><em>clusters</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cl_line_plots" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl_line_plots</span><span class="p">(</span><span class="n">clusters</span><span class="p">)</span>
</pre></div>
</div>
<p>purpose:  ???</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><p class="first"><strong>clusters</strong></p>
<blockquote>
<div><p>cluster object</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><p class="first"><strong>lineH</strong></p>
<blockquote class="last">
<div><p>handle</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cl_overlap">
<code class="descclassname">Visualization_functions.</code><code class="descname">cl_overlap</code><span class="sig-paren">(</span><em>in1</em>, <em>in2</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cl_overlap" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl_overlap</span><span class="p">(</span><span class="n">in1</span><span class="p">,</span><span class="n">in2</span><span class="p">)</span>
</pre></div>
</div>
<p>purpose:  compute overlap of two clusters</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><p class="first"><strong>in1</strong></p>
<blockquote>
<div><p>cluster object</p>
</div></blockquote>
<p><strong>in1</strong></p>
<blockquote>
<div><p>cluster object</p>
</div></blockquote>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><p class="first"><strong>ov_mat</strong></p>
<blockquote class="last">
<div><p>matrix of overlaps</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.close_non_spm_graphics_figures">
<code class="descclassname">Visualization_functions.</code><code class="descname">close_non_spm_graphics_figures</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.close_non_spm_graphics_figures" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">close_non_spm_graphics_figures</span><span class="p">()</span>
</pre></div>
</div>
<p>purpose:  closes all figure windows that are not the SPM orthviews window</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"></td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_barplot">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_barplot</code><span class="sig-paren">(</span><em>P</em>, <em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_barplot" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">clusters</span><span class="p">,</span> <span class="n">subcl</span><span class="p">]</span> <span class="o">=</span> <span class="n">cluster_barplot</span><span class="p">(</span><span class="n">P</span><span class="p">,</span> <span class="n">clusters</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>this function not only plots, but separates subclusters using pca / cluster_princomp.m
based on pattern across all conditions, covariance (not correlation),</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>P:</strong></dt>
<dd><p class="first last">cell array of strings containing image file names (data extracted from these)</p>
</dd>
</dl>
<p><strong>clusters:</strong></p>
<dl class="docutils">
<dt><strong>[opt] - &#8216;subclusters&#8217;:</strong></dt>
<dd><p class="first last">to get sub-clustering based on pca and clustering of voxels</p>
</dd>
<dt><strong>[opt] - cell array:</strong></dt>
<dd><p class="first last">of strings with condition names</p>
</dd>
<dt><strong>[opt] - &#8216;split&#8217;:</strong></dt>
<dd><p class="first">value is 1: to split into 2 plots (first half and last half of P)</p>
<p class="last">value is 2: to plot individual subjects over bars</p>
</dd>
<dt><strong>[opt] - &#8216;center&#8217;:</strong></dt>
<dd><p class="first last">center parameter values in plot (subtract row means)
this gives closer to correct &#8220;within subjects&#8221; error bars
and may be used when the overall parameter values have no meaning</p>
</dd>
<dt><strong>[opt] - &#8216;covs&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by between-subject covariates (e.g., behavioral regressors)</dt>
<dd><p class="first last">plots remove these before plotting means and std. errors</p>
</dd>
</dl>
</dd>
<dt><strong>[opt] - &#8216;max&#8217;:</strong></dt>
<dd><p class="first last">to make plots based on max z-values within region for each dataset P
not compatible with &#8216;split&#8217; and &#8216;center&#8217; (ignores these commands)
right now, special for inhib - see also inhib2_cluster_barplot (good function)</p>
</dd>
<dt><strong>[opt] - &#8216;indiv&#8217;:</strong></dt>
<dd><p class="first">to threshold based on individual t-statistics</p>
<p>FOLLOW with cell array of t-images &#8211; usually, there will be one cell, with images
for each subject in rows, to define voxels for each ss.</p>
<p class="last">BUT Tnames can be the same length as
contrast images, one t-img per subject per contrast, if
desired.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>clusters:</strong></dt>
<dd><p class="first last">clusters struture, with BARPLOT substructure added</p>
</dd>
<dt><strong>subcl:</strong></dt>
<dd><p class="first last">substructure contains data extracted and image file names</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>This program uses XYZmm millimeter coordinates in clusters to find voxels
So clusters and data files may have different dimensions.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cluster_barplot(EXPT.SNPM.P(4:6), clusters(2:3))
cluster_barplot(EXPT.SNPM.P(7:12), clusters(2:3), {&#39;ObjE&#39; &#39;AttE&#39; &#39;InteractE&#39; &#39;ObjI&#39; &#39;AttI&#39; &#39;InteractI&#39;}, &#39;split&#39;)
[clusters, subclusters] = cluster_barplot(EXPT.SNPM.P(17:24), clusters, &#39;subclusters&#39;, &#39;split&#39;)
RS2_8vs2_placeboCP = cluster_barplot(EXPT.SNPM.P([8 10 12 14
16]), RS2meta, &#39;indiv&#39;, T);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">mask2clusters.m, a simpler version that just extracts clusters from a mask file.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_cutaways">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_cutaways</code><span class="sig-paren">(</span><em>clusters</em>, <em>textprefix</em>, <em>mycolor</em>, <em>whichcuts</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_cutaways" title="Permalink to this definition">¶</a></dt>
<dd><p>groups clusters to image on brains by the first letter in whichcuts</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>O = cluster_cutaways(clusters,textprefix,mycolor,whichcuts,[coords for center of cuts],[revx])
</pre></div>
</div>
<p>groups clusters to image on brains by the first letter in whichcuts
therefore, if you enter &#8216;y&#8217; for whichcuts, the program will select
clusters with similar y values (w/i 15 mm) and image them on the same
brain. you need files called brain_render_T1.img/hdr on the path
these should be scalped anatomicals for the brain image.</p>
<p>you would also need &#8216;brain_render_T1.mat&#8217; for the
transparent brain surface, if you changed this script to get
the transparent brain surface.</p>
<p>if it cuts from the wrong side, try &#8216;revx&#8217; as input into tor_3d.m
or enter anything as 2nd var arg.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>clusters:</strong></dt>
<dd><p class="first last">cluster object to display on brain cuts</p>
</dd>
<dt><strong>textprefix:</strong></dt>
<dd><blockquote class="first">
<div><p>prefix for saving images to files</p>
</div></blockquote>
<dl class="last docutils">
<dt><strong>mycolor:</strong></dt>
<dd><blockquote class="first">
<div><p>cluster colors - 3 el. vector, single letter, or string of letters</p>
</div></blockquote>
<p class="last">enter letter or letter string in single quotes. e.g., O.clcol = &#8216;yrgb&#8217;;</p>
</dd>
<dt><strong>whichcuts:</strong></dt>
<dd><p class="first last">letter string (no quotes) - which axes to cut along - xyz are choices</p>
</dd>
</dl>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first"><strong>bestc</strong>
best cut, 3-element vector of coordinates, default [0 0 0]</p>
<p><strong>revx</strong>
text string to reverse x cut direction, enter 1 to do it.</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>O:</strong></dt>
<dd><p class="first last">output returned from renderCluster_ui.m</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>O = cluster_cutaways(clusters,&#39;myoutname&#39;,&#39;y&#39;,&#39;yzx&#39;,[0 0 0],[revx])
</pre></div>
</div>
<p>Uses renderCluster_ui.m</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_image_shape">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_image_shape</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_image_shape" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[hpatch, cl] = cluster_image_sphere(cl or [k x 3 list of mm coords], varargin)
</pre></div>
</div>
<p>Images spheres at cluster centers
Combine with addbrain and cluster_tools(&#8216;connect3d&#8217;) 
or cluster_nmdsfig_glassbrain
to get 3-D plots of connected blobs</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">{&#8216;color&#8217;, &#8216;colors&#8217;}, mycolor = varargin{i+1};</p>
<p>&#8216;radius&#8217;, myradius = varargin{i+1};</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>hpatch:</strong></dt>
<dd><p class="first last">patch handles</p>
</dd>
<dt><strong>cl:</strong></dt>
<dd><p class="first last">new cl with sphere coordiates in XYZmm and XYZ</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [hpatch, cl] = cluster_image_sphere(cl)

% With optional arguments:
[hpatch, cl] = cluster_image_sphere(cl, &#39;color&#39;, &#39;b&#39;, &#39;radius&#39;, 10)
[hpatch, cl] = cluster_image_sphere(cl, &#39;color&#39;, {&#39;r&#39; &#39;g&#39; &#39;b&#39; etc}, &#39;radius&#39;, 10)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body">Given an MNI coordinate, plot a sphere on a brain surface</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>my_mm_coord = [40, 46, 22]&#39;;
create_figure(&#39;surface&#39;)
cl = [];
cl.XYZmm = my_mm_coord;
cl.mm_center = my_mm_coord&#39;;
V = spm_vol(which(&#39;brainmask.nii&#39;));
cl.M = V.mat;
[hpatch, cl] = cluster_image_sphere(cl, &#39;color&#39;, &#39;g&#39;, &#39;radius&#39;, 10)
p = addbrain;
set(p, &#39;FaceAlpha&#39;, 1);
axis image
view(135, 30);
lighting gouraud;
lightRestoreSingle;
material dull;

% Turn xyz mm coordinates into clusters and image them

my_mm_coord = [40 46 22; 50 26 40; 45 36 50; 60 12 0]
[hpatch, cl] = cluster_image_sphere(my_mm_coord, &#39;color&#39;, &#39;b&#39;, &#39;radius&#39;, 4);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_image_sphere">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_image_sphere</code><span class="sig-paren">(</span><em>cl</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_image_sphere" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[hpatch, cl] = cluster_image_sphere(cl or [k x 3 list of mm coords], varargin)
</pre></div>
</div>
<p>Images spheres at cluster centers
Combine with addbrain and cluster_tools(&#8216;connect3d&#8217;) 
or cluster_nmdsfig_glassbrain to get 3-D plots of connected blobs
or surface_cutaway</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">{&#8216;color&#8217;, &#8216;colors&#8217;}, mycolor = varargin{i+1};</p>
<p>&#8216;radius&#8217;, myradius = varargin{i+1};</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><p class="first">patch handles, and new cl with sphere coordiates in XYZmm and XYZ</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"><p class="first last">Usage</p>
</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>function [hpatch, cl] = cluster_image_sphere(cl)

% With optional arguments:
[hpatch, cl] = cluster_image_sphere(cl, &#39;color&#39;, &#39;b&#39;, &#39;radius&#39;, 10)
[hpatch, cl] = cluster_image_sphere(cl, &#39;color&#39;, {&#39;r&#39; &#39;g&#39; &#39;b&#39; etc}, &#39;radius&#39;, 10)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">Given an MNI coordinate, plot a sphere on a brain surface</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>my_mm_coord = [40, 46, 22]&#39;;
create_figure(&#39;surface&#39;)
cl = [];
cl.XYZmm = my_mm_coord;
cl.mm_center = my_mm_coord&#39;;
V = spm_vol(which(&#39;brainmask.nii&#39;));
cl.M = V.mat;
[hpatch, cl] = cluster_image_sphere(cl, &#39;color&#39;, &#39;g&#39;, &#39;radius&#39;, 10)
p = addbrain;
set(p, &#39;FaceAlpha&#39;, 1);
axis image
view(135, 30); lighting gouraud; lightRestoreSingle; material dull;
</pre></div>
</div>
<p>Example: Turn xyz mm coordinates into clusters and image them</p>
<div class="highlight-python"><div class="highlight"><pre>my_mm_coord = [40 46 22; 50 26 40; 45 36 50; 60 12 0]
[hpatch, cl] = cluster_image_sphere(my_mm_coord, &#39;color&#39;, &#39;b&#39;, &#39;radius&#39;, 4);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_orthviews">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_orthviews</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_orthviews" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cluster_orthviews(inputs in any order)
</pre></div>
</div>
<p>This function uses spm_orthviews to display activation blobs from a
clusters structure on a canonical structural brain image. Multiple
clusters may be plotted in multiple colors, and blobs may be added to an
existing orthviews figure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first">clusters structures</p>
<p>colors cell array, e.g., {[0 0 1] [1 0 0] [0 1 0]}
if no colors cell array, uses Z- or t-scores to color map</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;add&#8217;:</strong></dt>
<dd><p class="first last">to suppress making new orthviews</p>
</dd>
<dt><strong>&#8216;copy&#8217;:</strong></dt>
<dd><p class="first last">to copy to smaller subfigure with empty axis beside it</p>
</dd>
<dt><strong>&#8216;unique&#8217;:</strong></dt>
<dd><p class="first last">to display in unique colors for each cluster</p>
</dd>
<dt><strong>&#8216;overlay&#8217;:</strong></dt>
<dd><p class="first last">followed by name of image, to use a custom anatomical overlay</p>
</dd>
<dt><strong>&#8216;bivalent&#8217;:</strong></dt>
<dd><p class="first">to plot increases in solid colors specified by colors cell
{1} and {2}</p>
<p class="last">OR, if no colors entered, use hot/cool map in 
spm_orthviews_hotcool_colormap</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Options if you specify colors (addColouredBlobs):</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;trans&#8217;:</strong></dt>
<dd><p class="first last">to make blobs transparent</p>
</dd>
<dt><strong>&#8216;solid&#8217;:</strong></dt>
<dd><p class="first last">to make them solid</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Options for using spm&#8217;s color map (addBlobs):</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>&#8216;blue&#8217;:</strong></dt>
<dd><p class="first last">display in blue split color map instead of default red-yellow</p>
</dd>
<dt><strong>&#8216;handle&#8217;:</strong></dt>
<dd><p class="first last">followed by integer for which orthviews window to use (default = 1)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_orthviews_classes">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_orthviews_classes</code><span class="sig-paren">(</span><em>cl</em>, <em>classes</em>, <em>overlay</em>, <em>myview</em>, <em>domontage</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_orthviews_classes" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>colors2 = cluster_orthviews_classes(cl,classes,overlay,myview,domontage, [orthview axis], [colors cell])
</pre></div>
</div>
<p>Makes montage and cluster orthviews of classes of clusters in different
colors (hard coded colors right now).</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>cl:</strong></dt>
<dd><p class="first last">clusters structure with regions</p>
</dd>
<dt><strong>classes:</strong></dt>
<dd><p class="first last">integers with classes (i.e., networks) to be color-coded on plot</p>
</dd>
<dt><strong>overlay:</strong></dt>
<dd><p class="first last">anatomical underlay image</p>
</dd>
<dt><strong>myview:</strong></dt>
<dd><p class="first last">if entered, shows centers on plot</p>
</dd>
<dt><strong>domontage:</strong></dt>
<dd><p class="first last">if non-zero, make montages of networks</p>
</dd>
<dt><strong>[orthview axis]:</strong></dt>
<dd><p class="first last">Optional; integer for which orthviews axis to use, 1:kk</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><p class="first">Cell of colors, in order, for use in other functions</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">classes</span> <span class="o">=</span> <span class="n">c</span><span class="o">.</span><span class="n">ClusterSolution</span><span class="o">.</span><span class="n">classes</span><span class="p">;</span>
<span class="n">overlay</span> <span class="o">=</span> <span class="n">EXPT</span><span class="o">.</span><span class="n">overlay</span><span class="p">;</span>
<span class="n">cluster_orthviews_classes</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">c</span><span class="o">.</span><span class="n">ClusterSolution</span><span class="o">.</span><span class="n">classes</span><span class="p">,</span> <span class="n">EXPT</span><span class="o">.</span><span class="n">overlay</span><span class="p">,</span> <span class="s">&#39;sagittal&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">);</span>
<span class="n">cluster_orthviews_classes</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">c</span><span class="o">.</span><span class="n">ClusterSolution</span><span class="o">.</span><span class="n">classes</span><span class="p">,</span> <span class="n">EXPT</span><span class="o">.</span><span class="n">overlay</span><span class="p">,</span> <span class="s">&#39;axial&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">);</span>
<span class="n">cluster_orthviews_classes</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">c</span><span class="o">.</span><span class="n">ClusterSolution</span><span class="o">.</span><span class="n">classes</span><span class="p">,</span> <span class="n">EXPT</span><span class="o">.</span><span class="n">overlay</span><span class="p">,</span> <span class="s">&#39;coronal&#39;</span><span class="p">,</span> <span class="mi">0</span><span class="p">);</span>
<span class="n">cluster_orthviews_classes</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="n">c</span><span class="o">.</span><span class="n">ClusterSolution</span><span class="o">.</span><span class="n">classes</span><span class="p">,</span> <span class="n">EXPT</span><span class="o">.</span><span class="n">overlay</span><span class="p">,</span> <span class="p">[],</span> <span class="mi">0</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_orthviews_montage">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_orthviews_montage</code><span class="sig-paren">(</span><em>spacing</em>, <em>myview</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_orthviews_montage" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[slices_fig_h, slice_mm_coords, slice_vox_coords, axis_handles] = cluster_orthviews_montage(spacing, myview, [overlay], [other optional args])
</pre></div>
</div>
<p>Runs on top of spm_orthviews, creates montages from current orthviews
display, whatever it is</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cluster_orthviews_montage(6, &#39;coronal&#39;);   % 6 mm spacing
cluster_orthviews_montage(10, &#39;sagittal&#39;, &#39;range&#39;, [-10 10]); % 10 mm spacing sag view with only parasagittal slices
cluster_orthviews_montage(12, &#39;axial&#39;);    % 12 mm spacing, axial view
</pre></div>
</div>
<dl class="docutils">
<dt>additional options: enter AFTER overlay:</dt>
<dd><ul class="first last simple">
<li>&#8216;whichorth&#8217;, whichorth = varargin{i+1}; varargin{i:i+1} = [];</li>
<li>&#8216;onerow&#8217;, doonerow = 1; varargin{i} = [];</li>
<li>&#8216;range&#8217;, followed by [min max] mm coords for slices</li>
<li>&#8216;xhairs&#8217;, xhairs = 1; turn on cross-hairs on slice plot</li>
</ul>
</dd>
</dl>
<p>used in cluster_orthviews_classes</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_orthviews_overlap">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_orthviews_overlap</code><span class="sig-paren">(</span><em>mask1</em>, <em>mask2</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_orthviews_overlap" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot two clusters on the orthviews, and their intersections for
activations and deactivations in intermediate colors</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cluster_orthviews_overlap(mask1, mask2, [colors cell])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first"><strong>mask1</strong>
path to nifit image</p>
<p><strong>mask2</strong>
path to nifit image</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first last"><strong>colors</strong>
cell array of RGB colors like: {[1 0 0] [0 0 1] [1 1 0] [0 1 1]}</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_orthviews_overlap2">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_orthviews_overlap2</code><span class="sig-paren">(</span><em>masks</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_orthviews_overlap2" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot blobs on the orthviews, and their intersections for
activations and deactivations in intermediate colors</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cluster_orthviews_overlap2(masks, [&#39;colors&#39;, colors cell], [&#39;surface&#39;], [&#39;negative&#39;] )
</pre></div>
</div>
<p>Positive effects only!</p>
<p>:Inputs</p>
<blockquote>
<div><strong>masks</strong>
image_vector  or fmri_data object</div></blockquote>
<p>:Optional Inputs</p>
<blockquote>
<div><blockquote>
<div><strong>colors</strong></div></blockquote>
<p>&#8216;colors&#8217;,{[RGB],[RGB],..} cell array of RGB triplets</p>
<p><strong>surface</strong>
&#8216;surface&#8217;,[0/1]
default =1</p>
<p><strong>negative</strong>
&#8216;negative&#8217; string to only use negative clusters, otherwise only positive clusters</p>
<p><strong>nomontage</strong>
&#8216;nomontage&#8217;,[0/1]
don&#8217;t do montage, default do a montage</p>
</div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_orthviews_overlap_3colors">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_orthviews_overlap_3colors</code><span class="sig-paren">(</span><em>mask1</em>, <em>mask2</em>, <em>mask3</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_orthviews_overlap_3colors" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot three clusters on the orthviews, and their intersections for
activations and deactivations in intermediate colors</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cluster_orthviews_overlap_3colors(mask1, mask2, mask3, [&#39;colors&#39;, colors cell], [&#39;surface&#39;] )
</pre></div>
</div>
<p>Positive effects only!</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body">mask1 mask2 mask3 are either image names (preferred!) or clusters
structures.  for clusters structures, need to add .dim field
and you need the 2010 object-oriented code in the canlab repository.</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_orthviews_showcenters">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_orthviews_showcenters</code><span class="sig-paren">(</span><em>cl</em>, <em>myview</em>, <em>overlay</em>, <em>xhairs</em>, <em>sliceorder</em>, <em>bgcolor</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_orthviews_showcenters" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>slices_fig_h = cluster_orthviews_showcenters(cl, myview, [overlay], [xhairs], [order slices flag], [background color], [var args])

cluster_orthviews_showcenters(cl, &#39;coronal&#39;);
cluster_orthviews_showcenters(cl, &#39;sagittal&#39;);
cluster_orthviews_showcenters(cl, &#39;axial&#39;);
</pre></div>
</div>
<p>used in cluster_orthviews_classes</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_surf">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_surf</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_surf" title="Permalink to this definition">¶</a></dt>
<dd><p>Surface plot of clusters on a standard brain</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">suface_handle</span><span class="p">,</span><span class="n">colorchangestring</span><span class="p">]</span> <span class="o">=</span> <span class="n">cluster_surf</span><span class="p">(</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>CLUSTERS:</strong></dt>
<dd><p class="first last">clusters structures, as created in tor_extract_rois.m</p>
</dd>
<dt><strong>COLORS:</strong></dt>
<dd><blockquote class="first">
<div><p>cell array of colors for each cluster: {[1 0 0] [0 1 0] [0 0 1]}</p>
</div></blockquote>
<p class="last">if number of colors specified is greater than number of clusters
structures entered, n+1 and n+2 colors are overlap of 2 and overlap
of all clusters, respectively.</p>
</dd>
<dt><strong>SURFACE MAT FILE:</strong></dt>
<dd><blockquote class="first">
<div><p>file name of mat file containing brain surface vertices and faces</p>
</div></blockquote>
<p class="last">as created with isosurface.</p>
</dd>
<dt><strong>SPECIAL SURFACE KEYWORDS:</strong></dt>
<dd><p class="first">special string: &#8216;bg&#8217; &#8216;hipp&#8217; (hcmp,thal,amy)
number of mm to plot from surface (mmdeep)</p>
<blockquote class="last">
<div><ul>
<li><p class="first">Special keywords for sets of surfaces are:
left, right, bg, limbic, cerebellum, brainstem</p>
</li>
<li><p class="first">Other keywords: &#8216;left&#8217; &#8216;right&#8217; &#8216;amygdala&#8217; &#8216;thalamus&#8217; &#8216;hippocampus&#8217; &#8216;
&#8216;midbrain&#8217;  &#8216;caudate&#8217;   &#8216;globus pallidus&#8217;  &#8216;putamen&#8217;  &#8216;nucleus accumbens&#8217;</p>
<blockquote>
<div><p>&#8216;hypothalamus&#8217; &#8216;cerebellum&#8217;</p>
</div></blockquote>
</li>
</ul>
</div></blockquote>
</dd>
<dt><strong>EXISTING SURFACE HANDLE(S):</strong></dt>
<dd><p class="first last">handles for surface patches, created,
e.g., with addbrain.m.  This lets you be very flexible in the
surfaces you image onto.</p>
</dd>
<dt><strong>&#8216;colorscale&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>This scales colors by Z-scores of voxels if used</dt>
<dd><ul class="first last simple">
<li>Uses input color, unless also used with &#8216;heatmap&#8217;</li>
<li>Z scores should be in ROW vector</li>
<li>use with &#8216;normalize&#8217; to scale Z-scores between -1 and 1</li>
<li>will also create transparent effects, mixing
blob color with existing surface color in linear
proportion to Z-scores</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;heatmap&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>Map Z-scores to surface colors</dt>
<dd><ul class="first last simple">
<li>Used WITH or instead of &#8216;colorscale&#8217;</li>
<li>Blobs can have a range of colors</li>
<li>Use with REFERENCE RANGE option below to control scale</li>
<li>solid colors entered as input will be ignored</li>
<li>use with &#8216;colormaps&#8217; option below to be flexible
in which color maps you apply.</li>
<li>if &#8216;colorscale&#8217; is also used, will produce transparent blobs.</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>REFERENCE RANGE:</strong></dt>
<dd><p class="first">reference Z-scores range, [zmin_act zmax_act
zmax_negact zmin_negact], e.g., [0 5 -5 0], use only
with &#8216;heatmap&#8217; option
to get refZ from clusters, try:</p>
<div class="last highlight-python"><div class="highlight"><pre>clZ = cat(2,clusters.Z);
refZ = [min(clZ(clZ &gt; 0)) max(clZ) min(clZ(clZ &lt; 0)) min(clZ)];
</pre></div>
</div>
</dd>
<dt><strong>&#8216;colormaps&#8217;:</strong></dt>
<dd><ul class="first simple">
<li>followed by custom [colors x 3] matrices for positive colors
and negative colors.</li>
<li>matlab can create some: e.g., colormap summer, jet, etc.
others can be created with colormap_tor.m</li>
</ul>
<p class="last">color [0 1 1] (cyan) is reserved for the overlap color btwn cluster sets.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>P = &#39;C:\tor_scripts\3DheadUtility\canonical_brains\surf_single_subj_T1_gray.mat&#39;;
cluster_surf(tcl,acl,P,10,{[0 1 0] [1 0 0]},&#39;colorscale&#39;,&#39;heatmap&#39;)

or P = h (surface handle) to use current surface in figure, and refZ
cluster_surf(tcl,acl,h,[3 5 -5 -3],10,{[0 1 0] [1 0 0]},&#39;colorscale&#39;,&#39;heatmap&#39;)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">More examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cluster_surf(cl,2,&#39;heatmap&#39;);     % brain surface.  vertices colored @2 mm
cluster_surf(cl,2,&#39;bg&#39;,&#39;heatmap&#39;);    % heatmap on basal ganglia
cluster_surf(cl,5,&#39;left&#39;,&#39;heatmap&#39;);  % heatmap on left surface @5 mm
cluster_surf(cl,2,&#39;right&#39;,&#39;heatmap&#39;);

% A multi-color, multi-threshold display on the cerebellum
colors = {[1 1 0] [1 .5 0] [1 .3 .3]};
tor_fig;
sh = cluster_surf(cl{3},colors(3),5,&#39;cerebellum&#39;);
cluster_surf(cl{2},colors(2),5,sh);
cluster_surf(cl{1},colors(1),5,sh);

% Custom colormaps:
create_figure(&#39;Brain Surface&#39;); cluster_surf(cl, 2, &#39;heatmap&#39;,&#39;left&#39;);

poscm = colormap_tor([.2 .2 .4], [1 1 0], [.9 .6 .1]);  %slate to orange to yellow
negcm = colormap_tor([0 0 1], [0 .3 1]);  % light blue to dark blue
create_figure(&#39;Brain Surface&#39;); cluster_surf(cl, 2, &#39;heatmap&#39;, &#39;colormaps&#39;, poscm, negcm, &#39;left&#39;);

% Single-color transparent map (green):
cluster_surf(cl, 2, {[0 1 0]}, &#39;colorscale&#39;, p3(2), &#39;normalize&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">addbrain, img2surf.m, surface() methods for objects, cluster_cutaways</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_surf_batch">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_surf_batch</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_surf_batch" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">surf_handles</span> <span class="o">=</span> <span class="n">cluster_surf_batch</span><span class="p">(</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Single-map visualization
P2 = threshold_imgs(&#39;rob_tmap_0001.img&#39;,tinv(1-.005,12),15,&#39;pos&#39;);
cluster_surf_batch(P2);
surf_handles = cluster_surf_batch(cl,{[1 0 0]},cl2);

% Two maps with overlap
surf_handles = cluster_surf_batch(cl,{[1 0 0] [0 1 0] [1 1 0]},cl2);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.cluster_surf_batch2">
<code class="descclassname">Visualization_functions.</code><code class="descname">cluster_surf_batch2</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.cluster_surf_batch2" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">surf_handles</span> <span class="o">=</span> <span class="n">cluster_surf_batch</span><span class="p">(</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Single-map visualization
P2 = threshold_imgs(&#39;rob_tmap_0001.img&#39;, tinv(1-.005, 12), 15, &#39;pos&#39;);
cluster_surf_batch(P2);
surf_handles = cluster_surf_batch({cl cl2}, {[1 0 0]});

% Two maps with overlap
surf_handles = cluster_surf_batch(cl, {[1 0 0] [0 1 0] [1 1 0]}, cl2);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.colormap_tor">
<code class="descclassname">Visualization_functions.</code><code class="descname">colormap_tor</code><span class="sig-paren">(</span><em>lowcolor</em>, <em>hicolor</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.colormap_tor" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>newcolormap = colormap_tor(lowcolor, hicolor, [midcolor], [midcolor2], etc.)
</pre></div>
</div>
<p>Create a new colormap of your choosing.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>colormap_tor([.2 .2 .6], [1 1 0]);  % slate to yellow
colormap_tor([.9 .5 .2], [1 1 0]);  % orange to yellow
colormap_tor([.8 .1 .1], [1 1 0], [.9 .6 .1]);  %red to orange to yellow
colormap_tor([.2 .2 .4], [1 1 0], [.9 .6 .1]);  %slate to orange to yellow
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.compare_filtered_t">
<code class="descclassname">Visualization_functions.</code><code class="descname">compare_filtered_t</code><span class="sig-paren">(</span><em>anatP</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.compare_filtered_t" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">compare_filtered_t</span><span class="p">(</span><span class="n">anatP</span><span class="p">,</span><span class="n">P1</span><span class="p">,</span><span class="n">P2</span><span class="p">,</span> <span class="n">etc</span><span class="o">...</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>compare_filtered_t([],&#39;rob_tmap_filtered_0001.img&#39;,&#39;rob_tmap_filtered_0002.img&#39;)

Threshold spm T images and display them together in SPM orthviews
threshold_spm_t(.005,22,0,&#39;pos&#39;)
compare_filtered_t([],&#39;rfx0009/spmT_filtered_0002.img&#39;,&#39;rfx0011/spmT_filtered_0002.img&#39;, ...
&#39;rfx0013/spmT_filtered_0002.img&#39;,&#39;rfx0015/spmT_filtered_0002.img&#39;,&#39;rfx0017/spmT_filtered_0002.img&#39;)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.compare_slice">
<code class="descclassname">Visualization_functions.</code><code class="descname">compare_slice</code><span class="sig-paren">(</span><em>ovlP</em>, <em>sxyz</em>, <em>sz</em>, <em>VOL</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.compare_slice" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">compare_slice</span><span class="p">(</span><span class="n">ovlP</span><span class="p">,</span><span class="n">sxyz</span><span class="p">,</span><span class="n">sz</span><span class="p">,</span><span class="n">VOL</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>ovlP:</strong></dt>
<dd><p class="first last">is name of overlay image file (anatomical)</p>
</dd>
<dt><strong>sxzy:</strong></dt>
<dd><p class="first last">is a cell array, where each cell is a list of</p>
</dd>
</dl>
<p>VOXEL coordinates in 3 x n matrix</p>
<dl class="last docutils">
<dt><strong>sz:</strong></dt>
<dd><p class="first last">is a cell array, where each cell contains the z values
corresponding to sxyz.  This is used to create 
pseudocolor on the plots.</p>
</dd>
<dt><strong>VOL:</strong></dt>
<dd><p class="first">is a structure containing the field M,
where M is the SPM99-style mapping matrix from voxel to mm 
space. This is used to map sxyz values onto the overlay image.</p>
<p class="last">VOL must also contain field dim, which has the voxel dims
of the results image for all sxyz.
All sxyz must have the same dimensions!</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.conf_region">
<code class="descclassname">Visualization_functions.</code><code class="descname">conf_region</code><span class="sig-paren">(</span><em>X</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.conf_region" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">ax</span><span class="p">,</span><span class="n">ci</span><span class="p">,</span><span class="n">cen</span><span class="p">,</span><span class="n">ub</span><span class="p">,</span><span class="n">lb</span><span class="p">,</span><span class="n">S</span><span class="p">,</span><span class="n">e</span><span class="p">,</span><span class="n">lam</span><span class="p">,</span><span class="n">Fm</span><span class="p">,</span><span class="n">Fc</span><span class="p">,</span><span class="n">F</span><span class="p">,</span><span class="n">pval</span><span class="p">,</span> <span class="n">msb</span><span class="p">]</span> <span class="o">=</span> <span class="n">conf_region</span><span class="p">(</span><span class="n">X</span><span class="p">,[</span><span class="n">doplot</span><span class="p">])</span>
</pre></div>
</div>
<p>alternative conf region multivariate based on Johnson &amp; Wichern, 4th ed., p. 236
2D (3D)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>ax:</strong></dt>
<dd><p class="first last">axes of confidence region, scaled to be half-length of confidence hyperellipsoid</p>
</dd>
<dt><strong>ci:</strong></dt>
<dd><p class="first last">length of axes of conf region (diagonal matrix), axes in reverse order of importance
as in ouput of eig</p>
</dd>
<dt><strong>cen:</strong></dt>
<dd><p class="first last">center of region (means of variables)</p>
</dd>
<dt><strong>ub:</strong></dt>
<dd><p class="first last">upper boundary coordinates for region, rows are dims (vars), cols index coordinates
last coordinate is on axis of greatest variation (&#8220;reverse order&#8221;), from output of eig</p>
</dd>
<dt><strong>lb:</strong></dt>
<dd><p class="first last">lower boundary coordinates
to plot, try: plot(ub(1,:),ub(2,:),&#8217;bo&#8217;); hold on; plot(lb(1,:),lb(2,:),&#8217;ro&#8217;)</p>
</dd>
<dt><strong>S:</strong></dt>
<dd><p class="first last">covariance matrix</p>
</dd>
<dt><strong>e:</strong></dt>
<dd><p class="first last">eigenvectors</p>
</dd>
<dt><strong>lam:</strong></dt>
<dd><p class="first last">eigenvalues</p>
</dd>
<dt><strong>Fm:</strong></dt>
<dd><p class="first last">degrees of freedom multiplier, p(n-1) / n(n-p)</p>
</dd>
<dt><strong>Fc:</strong></dt>
<dd><p class="first last">critical F value based on alpha level</p>
</dd>
<dt><strong>F:</strong></dt>
<dd><p class="first last">F value for test - probably not quite right</p>
</dd>
<dt><strong>pval:</strong></dt>
<dd><p class="first last">p value for test - probably not quite right</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">To plot:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Either enter 1 or color as a 2nd argument, or do it yourself:</p>
<div class="highlight-python"><div class="highlight"><pre>[ax,ci,cen,ub,lb,S,e,lam] = conf_region(X);
b = pinv([X(:,1) ones(size(X,1),1)] ) * X(:,2);
theta = atan(b(1)) - pi/2;
[h,h2] = plot_ellipse(cen(1),cen(2),theta,ci(1,1),ci(2,2));
</pre></div>
</div>
<p>There is also an example for rendering individual subject conf regions</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>N = 250;
X = mvnrnd([1 2], [1 .6; .6 1], N);
[ax,ci,cen,ub,lb,S,e,lam] = conf_region(X);
b = pinv([X(:,1) ones(size(X,1),1)] ) * X(:,2);
theta = atan(b(1)) - pi/2;
%create_figure(&#39;test&#39;);
[h,h2] = plot_ellipse(cen(1),cen(2),theta,ci(1,1),ci(2,2));

% plot standard deviation - for bootstrapping, or for individual cases
[h,h2] = plot_ellipse(cen(1),cen(2),theta,ci(1,1)*sqrt(N),ci(2,2)*sqrt(N));
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body">Render individual subject conf regions</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>X = x(wh, [3 1]);
dfe = size(X, 1) - size(X, 2);   % df
alph = .1;  % .1 for 90%, .05 for 95%
tcrit = tinv(1 - alph, dfe); 
ci = ((lam) .^ .5) .* tcrit;
b = pinv([X(:,1) ones(size(X,1),1)] ) * X(:,2);
theta = atan(b(1)) - pi/2;
[h,h2] = plot_ellipse(cen(1),cen(2),theta,ci(1,1),ci(2,2));

set(h, &#39;Color&#39;, &#39;k&#39;, &#39;LineWidth&#39;, 1);
set(h2, &#39;FaceColor&#39;, [.5 1 0]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.create_figure">
<code class="descclassname">Visualization_functions.</code><code class="descname">create_figure</code><span class="sig-paren">(</span><em>tagname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.create_figure" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>f1 = create_figure([&#39;tagname&#39;], [subplotrows], [subplotcols], [do not clear flag])
</pre></div>
</div>
<p>checks for old figure with tag of tagname,
clears it if it exists, or creates new one if it doesn&#8217;t</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.errorbar_horizontal">
<code class="descclassname">Visualization_functions.</code><code class="descname">errorbar_horizontal</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.errorbar_horizontal" title="Permalink to this definition">¶</a></dt>
<dd><p>Error bar plot</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">errorbar_horizontal</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">Y</span><span class="p">,</span><span class="n">L</span><span class="p">,</span><span class="n">U</span><span class="p">)</span>
</pre></div>
</div>
<p>plots the graph of vector X vs. vector Y with
error bars specified by the vectors L and U.  L and U contain the
lower and upper error ranges for each point in Y.  Each error bar
is L(i) + U(i) long and is drawn a distance of U(i) above and L(i)
below the points in (X,Y).  The vectors X,Y,L and U must all be
the same length.  If X,Y,L and U are matrices then each column
produces a separate line.</p>
<div class="highlight-python"><div class="highlight"><pre>errorbar_horizontal(X,Y,E)
% or
errorbar_horizontal(Y,E)
</pre></div>
</div>
<p>plots Y with error bars [Y-E Y+E].</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ERRORBAR</span><span class="p">(</span><span class="o">...</span><span class="p">,</span><span class="s">&#39;LineSpec&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>uses the color and linestyle specified by
the string &#8216;LineSpec&#8217;.  The color is applied to the data line and
error bars while the linestyle and marker are applied to the data 
line only.  See PLOT for possibilities.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">errorbar_horizontal</span><span class="p">(</span><span class="n">AX</span><span class="p">,</span><span class="o">...</span><span class="p">)</span>
</pre></div>
</div>
<p>plots into AX instead of GCA.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">H</span> <span class="o">=</span> <span class="n">errorbar_horizontal</span><span class="p">(</span><span class="o">...</span><span class="p">)</span>
</pre></div>
</div>
<p>returns a vector of errorbarseries handles in H.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">To draws symmetric error bars of unit standard deviation</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>x = 1:10;
y = sin(x);
e = std(y)*ones(size(x));
errorbar(x,y,e)
</pre></div>
</div>
<p>If means(:, 1) is x-values, means(:, 2) is y-values, stds(:, 1) is error
on x, and stds(:, 2) is error on y, then:</p>
<div class="highlight-python"><div class="highlight"><pre>lineh = errorbar_horizontal(means(:, 1), means(:, 2), stds(:, 1));
linehx = errorbar(means(:, 1), means(:, 2), stds(:, 2));
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.errorbar_width">
<code class="descclassname">Visualization_functions.</code><code class="descname">errorbar_width</code><span class="sig-paren">(</span><em>h</em>, <em>x</em>, <em>interval</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.errorbar_width" title="Permalink to this definition">¶</a></dt>
<dd><p>Work with errorbar.m: Adjust the width of errorbar</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">errorbar_width</span><span class="p">(</span><span class="n">h</span><span class="p">,</span> <span class="n">x</span><span class="p">,</span> <span class="n">interval</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>h:</strong></dt>
<dd><p class="first last">errorbar graphic handle</p>
</dd>
<dt><strong>x:</strong></dt>
<dd><p class="first last">vector x, which is used in errorbar</p>
</dd>
<dt><strong>interval:</strong></dt>
<dd><p class="first last">e.g., [-.1 .1] or [0 0]</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"><p class="first last">you can see this output in</p>
</td>
</tr>
</tbody>
</table>
<p><a class="reference external" href="http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery">http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery</a></p>
<div class="highlight-python"><div class="highlight"><pre>x = 1:5; % x values
y = [32 40 55 84 130]; % mean
e = [6 6 6 6 6]; % standard error of the mean

create_figure(y_axis);
set(gcf, &#39;Position&#39;, [1   512   268   194]);
col = [0.3333    0.6588    1.0000];
markercol = col-.2;

h = errorbar(x, y, e, &#39;o&#39;, &#39;color&#39;, &#39;k&#39;, &#39;linewidth&#39;, 1.5, &#39;markersize&#39;, 7, &#39;markerfacecolor&#39;, col);
hold on;
sepplot(x, y, .75, &#39;color&#39;, col, &#39;linewidth&#39;, 2);
errorbar_width(h, x, [0 0]); % here

set(gca, &#39;xlim&#39;, [.5 5.5], &#39;linewidth&#39;, 1.5);

try
   pagesetup(gcf);
   saveas(gcf, &#39;example.pdf&#39;);
catch
   pagesetup(gcf);
   saveas(gcf, &#39;example.pdf&#39;);
end
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.fill_area_around_points">
<code class="descclassname">Visualization_functions.</code><code class="descname">fill_area_around_points</code><span class="sig-paren">(</span><em>x</em>, <em>y</em>, <em>borderscale</em>, <em>color</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.fill_area_around_points" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">fill_area_around_points</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">borderscale</span><span class="p">,</span> <span class="n">color</span><span class="p">)</span>
</pre></div>
</div>
<p>fills area around list of coordinates in a color
using spline interpolation and other stuff.
designed for cluster imaging in nmdsfig figures.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">x</span> <span class="o">=</span> <span class="n">randn</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span><span class="mi">1</span><span class="p">);</span>
<span class="n">y</span> <span class="o">=</span> <span class="n">randn</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span><span class="mi">1</span><span class="p">);</span>
<span class="n">figure</span><span class="p">;</span> <span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="s">&#39;k.&#39;</span><span class="p">);</span>
<span class="n">h</span> <span class="o">=</span> <span class="n">fill_area_around_points</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="o">.</span><span class="mi">2</span><span class="p">,</span> <span class="s">&#39;r&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.getVertexColors">
<code class="descclassname">Visualization_functions.</code><code class="descname">getVertexColors</code><span class="sig-paren">(</span><em>xyz</em>, <em>v</em>, <em>actcolor</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.getVertexColors" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">c</span><span class="p">,</span> <span class="n">alld</span><span class="p">]</span> <span class="o">=</span> <span class="n">getVertexColors</span><span class="p">(</span><span class="n">xyz</span><span class="p">,</span> <span class="n">v</span><span class="p">,</span> <span class="n">actcolor</span><span class="p">,</span> <span class="p">[</span><span class="n">basecolor</span><span class="p">],</span> <span class="p">[</span><span class="n">mind</span><span class="p">],</span> <span class="s">&#39;vert&#39;</span><span class="p">,</span> <span class="p">[</span><span class="n">xyz2</span><span class="p">],</span> <span class="p">[</span><span class="n">actcolor2</span><span class="p">],</span> <span class="s">&#39;vert&#39;</span><span class="p">,</span> <span class="p">[</span><span class="n">xyz3</span><span class="p">],</span> <span class="p">[</span><span class="n">actcolor3</span><span class="p">])</span>
</pre></div>
</div>
<p>given a point list of XYZ mm coordinates (3 columns)
and a list of vertices in an isosurface, 
returns FaceVertexCData color values for brain near points and brain not near points.
c is vertex color specification, 3 columns indicating RGB values</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>xyz:</strong></dt>
<dd><p class="first last">a 3-vol list of vertices to color</p>
</dd>
<dt><strong>v:</strong></dt>
<dd><p class="first last">can be a matrix of vertices
or a handle to a patch object containing vertices
if it&#8217;s a handle, this function sets the color to interp
and the FaceVertexCData to the color matrix c</p>
</dd>
<dt><strong>actcolor:</strong></dt>
<dd><p class="first last">[r g b] activation color</p>
</dd>
<dt><strong>basecolor:</strong></dt>
<dd><p class="first last">[r g b] baseline color - optional.</p>
</dd>
<dt><strong>mind:</strong></dt>
<dd><p class="first last">optional - min distance to color vertex
Vertices within mind of an xyz coordinate will be colored</p>
</dd>
<dt><a href="#id57"><span class="problematic" id="id58">**</span></a>colorscale</dt>
<dd><p class="first">optional.  followed by vector of values by which to multiply input
color
these are scaled to be between .3 and one.</p>
<p>if entered, this will make the colors vary by, for example, Z score
so Z-scores are an acceptable input.</p>
<p>cscale should be in the same coordinate order as xyz</p>
<p>for ADDITIONAL clusters, repeat the &#8216;colorscale&#8217;, Z argument pair in the function call</p>
<p class="last">YOU CAN ALSO pass true RGB values for each xyz coordinate in: &#8216;colorscale&#8217;, rgblist, 
IF cscale is a 3-vector, it specifies the ACTUAL colors, and is not scaled to .3 - 1</p>
</dd>
</dl>
<p>Following basecolor and mind:</p>
<p>additional xyz coordinate lists, with syntax:
vert&#8217;, xyz2 [your xyz input], [r g b] color for xyz plot</p>
<p>also, you can enter &#8216;ovlcolor&#8217; followed by [r g b] for overlaps between xyz sets
colors will ONLY appear in the overlap color if they share actual coordinates in common, 
not necessarily if surface vertices are within the specified distance from both sets of coords.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% to get a good brain surface, try this:
figure
p = patch(&#39;Faces&#39;, faces, &#39;Vertices&#39;, vertices, &#39;FaceColor&#39;, [.5 .5 .5], ...
        å  &#39;EdgeColor&#39;, &#39;none&#39;, &#39;SpecularStrength&#39;, .2, &#39;FaceAlpha&#39;, 1, &#39;SpecularExponent&#39;, 200);
lighting gouraud;
camlight right
axis image;
myLight = camlight(0, 0);
set(myLight, &#39;Tag&#39;, &#39;myLight&#39;);
set(gcf, &#39;WindowButtonUpFcn&#39;, &#39;lightFollowView&#39;);
lightfollowview
drawnow
</pre></div>
</div>
<p>by Tor Wager  August 25, 2002
..</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.get_cluster_volume">
<code class="descclassname">Visualization_functions.</code><code class="descname">get_cluster_volume</code><span class="sig-paren">(</span><em>clIn</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.get_cluster_volume" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">clOut</span> <span class="o">=</span> <span class="n">get_cluster_volume</span><span class="p">(</span><span class="n">clIn</span><span class="p">)</span>
</pre></div>
</div>
<p>Adapted from:</p>
<p>FORMAT clusters = ihb_getClusters</p>
<p>Get cluster information (use [SPM,VOL,xX,xCon,xSDM] = spm_getSPM;)</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.glassbrain_avi">
<code class="descclassname">Visualization_functions.</code><code class="descname">glassbrain_avi</code><span class="sig-paren">(</span><em>fps</em>, <em>len</em>, <em>clusters</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.glassbrain_avi" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">M</span> <span class="o">=</span> <span class="n">glassbrain_avi</span><span class="p">(</span><span class="n">fps</span><span class="p">,</span><span class="nb">len</span><span class="p">,</span><span class="n">clusters</span><span class="p">)</span>
</pre></div>
</div>
<p>works with cluster_surf.m
assumes a gray brain where no activation, or creates one</p>
<p>SEE ALSO cluster_surf_movie.m, which works well too.</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.imageCluster">
<code class="descclassname">Visualization_functions.</code><code class="descname">imageCluster</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.imageCluster" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[out,cl] = imageCluster(arguments as specified below)
</pre></div>
</div>
<p>Images a cluster isosurface on an existing 3D head or brain plot</p>
<p>works with tsu (Talairach Space Utility) main window as current figure
old: clusters = getappdata(gcf, &#8216;clusters&#8217;);</p>
<p>Inputs (in any order): keyword followed by input argument</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;cluster&#8217;:</strong></dt>
<dd>followed by cluster to image, from SPM or TSU.</dd>
<dt><strong>&#8216;getclusters&#8217;:</strong></dt>
<dd>no other args necessary - starts gui for cluster selection
function returns all clusters.  select with clusters(i)</dd>
<dt><strong>&#8216;getfigclusters&#8217;:</strong></dt>
<dd>get clusters from TSU main figure.  must be current figure.</dd>
<dt><strong>&#8216;figure&#8217;:</strong></dt>
<dd>create a new figure to image color on</dd>
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd>followed by color value - either text or vector</dd>
<dt><strong>&#8216;alpha&#8217;:</strong></dt>
<dd>followed by transparency value for cluster surface, 0-1
1 is opaque, 0 is completely transparent</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Control of smoothing:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;heightthresh&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by cutoff threshold post-smooth, in percentage of min Z value in cl</dt>
<dd><ul class="first last simple">
<li>enter a number between 0 and 1</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;fwhm&#8217;:</strong></dt>
<dd><p class="first last">followed by smoothing kernel FWHM (Gaussian)</p>
</dd>
<dt><strong>&#8216;kernelsize&#8217;:</strong></dt>
<dd><p class="first last">followed by box size for kernel support (5 5 5 is  default)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>out:</strong></dt>
<dd><p class="first last">Patch handle</p>
</dd>
<dt><strong>cl:</strong></dt>
<dd><p class="first last">cluster struct</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Uses:</th><td class="field-body"><p class="first last">cl.XYZmm and cl.voxSize</p>
</td>
</tr>
</tbody>
</table>
<p>Works in Matlab 5.3, but with no transparency.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>p(i) = imageCluster(&#39;cluster&#39;,region2struct(r(i)),&#39;color&#39;,colors{i},&#39;alpha&#39;,1, &#39;fwhm&#39;, 1.2, &#39;heightthresh&#39;, .3); 
view(135, 30); lighting gouraud; lightRestoreSingle; axis image; camlight right;
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.imageCluster_block">
<code class="descclassname">Visualization_functions.</code><code class="descname">imageCluster_block</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.imageCluster_block" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[out,cl] = imageCluster_block(input arguments)
</pre></div>
</div>
<p>Images a cluster isosurface on an existing 3D head or brain plot</p>
<p>works with tsu (Talairach Space Utility) main window as current figure
old: clusters = getappdata(gcf, &#8216;clusters&#8217;);</p>
<p>Inputs (in any order): keyword followed by input argument</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;cluster&#8217;:</strong></dt>
<dd>followed by cluster to image, from SPM or TSU.</dd>
<dt><strong>&#8216;getclusters&#8217;:</strong></dt>
<dd>no other args necessary - starts gui for cluster selection
function returns all clusters.  select with clusters(i)</dd>
<dt><strong>&#8216;getfigclusters&#8217;:</strong></dt>
<dd>get clusters from TSU main figure.  must be current figure.</dd>
<dt><strong>&#8216;figure&#8217;:</strong></dt>
<dd>create a new figure to image color on</dd>
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd>followed by color value - either text or vector</dd>
<dt><strong>&#8216;alpha&#8217;:</strong></dt>
<dd>followed by transparency value for cluster surface, 0-1
1 is opaque, 0 is completely transparent</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>out:</strong></dt>
<dd><p class="first last">Patch handle</p>
</dd>
<dt><strong>cl:</strong></dt>
<dd><p class="first last">cluster struct</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.image_histogram">
<code class="descclassname">Visualization_functions.</code><code class="descname">image_histogram</code><span class="sig-paren">(</span><em>P</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.image_histogram" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">image_histogram</span><span class="p">(</span><span class="n">P</span><span class="p">,[</span><span class="n">method</span><span class="p">(</span><span class="n">string</span><span class="p">)],[</span><span class="nb">range</span><span class="p">])</span>
</pre></div>
</div>
<p>eliminates 0, NaN voxels from either image
red then blue</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>P:</strong></dt>
<dd><p class="first last">is one or two images in string array</p>
</dd>
<dt><strong>Methods:</strong></dt>
<dd><p class="first last">&#8216;def&#8217;</p>
</dd>
<dt><strong>range:</strong></dt>
<dd><p class="first last">optional 3rd input, range of values to include in histogram</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>h = image_histogram(&#39;p_Omnibus.img&#39;,&#39;def&#39;,[0 1-eps]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.line_plot_multisubject">
<code class="descclassname">Visualization_functions.</code><code class="descname">line_plot_multisubject</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.line_plot_multisubject" title="Permalink to this definition">¶</a></dt>
<dd><p>Plots a scatterplot with multi-subject data, with one line per subject
in a unique color.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">han</span><span class="p">,</span> <span class="n">X</span><span class="p">,</span> <span class="n">Y</span><span class="p">]</span> <span class="o">=</span> <span class="n">line_plot_multisubject</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">Y</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>X and Y:</strong></dt>
<dd><p class="first last">are cell arrays, one cell per upper level unit (subject)</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>varargin:</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>&#8216;n_bins:</strong></dt>
<dd>pass in the number of point &#8220;bins&#8221;.  Will divide each subj&#8217;s trials
into bins, get the avg X and Y per bin, and plot those points.</dd>
<dt><strong>&#8216;noind&#8217;:</strong></dt>
<dd>suppress points</dd>
<dt><strong>&#8216;subjid&#8217;:</strong></dt>
<dd>followed by integer vector of subject ID numbers. Use when
passing in vectors (with subjects concatenated) rather than
cell arrays in X and Y</dd>
<dt><strong>&#8216;center&#8217;:</strong></dt>
<dd>subtract means of each subject before plotting</dd>
<dt><strong>&#8216;colors&#8217;:</strong></dt>
<dd>followed by array size N of desired colors.  if not passed
in, will use scn_standard_colors</dd>
<dt><strong>&#8216;MarkerTypes&#8217;:</strong></dt>
<dd>followed by char string.  if not passed in, uses
&#8216;osvd^&lt;&gt;ph&#8217; by default</dd>
<dt><strong>&#8216;group_avg_ref_line&#8217;:</strong></dt>
<dd>will make a reference line for the group avg</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>han:</strong></dt>
<dd><p class="first last">handles to points and lines</p>
</dd>
<dt><strong>X, Y:</strong></dt>
<dd><p class="first last">new variables (binned if bins requested)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.lineplot_columns">
<code class="descclassname">Visualization_functions.</code><code class="descname">lineplot_columns</code><span class="sig-paren">(</span><em>dat_matrix</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.lineplot_columns" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">out</span> <span class="o">=</span> <span class="n">lineplot_columns</span><span class="p">(</span><span class="n">dat_matrix</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Default values:</th><td class="field-body"><p class="first">(see below for how to change)
- w = 3;                          % width
- color = &#8216;k&#8217;;                    % color
- wh = true(size(dat_matrix));    % which observations
- x = 1:size(dat_matrix, 2);      % x values
- marker = &#8216;o&#8217;;                   % marker style
- linestyle = &#8216;-&#8216;;                % line
- markersize = 8;                 % markersize
- markerfacecolor = [.5 .5 .5];   % face color
- dowithinste = 0;                % enter &#8216;within&#8217; to get within-ss ste
- atleast = 1;                    % &#8216;atleast&#8217; followed by n for at least n valid obs to plot 
- doshading = 0;                  % shaded vs. error-bar plots
- CIs95 = 0;                      % 95% CI&#8217;s.  Does not work with within subj error</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat_matrix:</strong></dt>
<dd><p class="first">is usually a rectangular matrix with rows = observations,
columns = variables.</p>
<p class="last">It can also be a cell array with column vectors in each cell, for unequal
numbers of observations, but then the rows will not be the same
observations across variables.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">followed by values:</p>
<p>{&#8216;w&#8217;, &#8216;color&#8217;, &#8216;x&#8217;, &#8216;marker&#8217;, &#8216;linestyle&#8217;, &#8216;markersize&#8217;, &#8216;markerfacecolor&#8217;, &#8216;wh&#8217;}</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Keywords:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>{&#8216;within&#8217;, &#8216;dowithinste&#8217;}, &#8216;shade&#8217;, &#8216;atleast&#8217;, &#8216;CIs&#8217; &lt;- for 95% CI&#8217;s</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>out = lineplot_columns(dat_matrix, &#39;w&#39;, 3, &#39;color&#39;, &#39;r&#39;, &#39;markerfacecolor&#39;, [1 .5 0], &#39;wh&#39;, ispain);
out = lineplot_columns(dat_matrix, &#39;w&#39;, 3, &#39;color&#39;, [0 1 0], &#39;markerfacecolor&#39;, [1 .5 0], &#39;within&#39;);
out = lineplot_columns(dat_matrix, &#39;w&#39;, 3, &#39;color&#39;, &#39;b&#39;, &#39;markerfacecolor&#39;, [0 .5 1], &#39;within&#39;);

% shaded error regions
out = lineplot_columns(hotopen, &#39;color&#39;, &#39;r&#39;, &#39;marker&#39;, &#39;none&#39;, &#39;w&#39;, 1, &#39;shade&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.make3Davi">
<code class="descclassname">Visualization_functions.</code><code class="descname">make3Davi</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.make3Davi" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre>mov = make3Davi([opt] Options_Structure)
</pre></div>
</div>
<p>Makes an avi movie file called head3d[x].avi or whatever you specify.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>O.name:</strong></dt>
<dd><p class="first last">&#8216;output_name.avi&#8217;;</p>
</dd>
<dt><strong>O.fps:</strong></dt>
<dd><p class="first last">frames per second</p>
</dd>
<dt><strong>O.length:</strong></dt>
<dd><p class="first last">length of movie in s</p>
</dd>
<dt><strong>O.H:</strong></dt>
<dd><p class="first last">cluster handles in vector   (for adjusting transparency with time)</p>
</dd>
<dt><strong>O.timecourse{i}:</strong></dt>
<dd><p class="first last">cell array of time courses for each cluster</p>
</dd>
<dt><strong>O.timeres:</strong></dt>
<dd><p class="first last">resolution, in s, of timecourse data</p>
</dd>
<dt><strong>O.azOffset:</strong></dt>
<dd><p class="first last">azimuth value to offset, positive = move clockwise</p>
</dd>
<dt><strong>O.elOffset:</strong></dt>
<dd><p class="first last">elevation value to move through, positive = inf to superior</p>
</dd>
<dt><strong>O.zoom:</strong></dt>
<dd><p class="first last">zoom value to end up with</p>
</dd>
<dt><strong>O.add2movie:</strong></dt>
<dd><p class="first last">add to existing movie - enter mov structure in this field</p>
</dd>
<dt><strong>O.closemovie:</strong></dt>
<dd><p class="first last">1 or 0, close the movie afterward or not.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Blank fields for az, el, zoom, timecourse indicate that these functions should not be performed
This script spirals up, right, and in 36 degrees</p>
<dl class="docutils">
<dt>Notes: my indeo5 one wouldn&#8217;t work in media player</dt>
<dd>also only seems to work if you add clusters before head isosurfaces
for Mac OS X / UNIX, choose &#8220;no compression&#8221;</dd>
</dl>
<p>start with the image in the location you want to zoom in on
but with no zoom.</p>
<p>The movie FINISHES at the current axis locations</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body">&#8220;Surface tour&#8221;</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">O</span> <span class="o">=</span> <span class="n">struct</span><span class="p">(</span><span class="s">&#39;name&#39;</span><span class="p">,</span><span class="s">&#39;all4_union_bg_surf.avi&#39;</span><span class="p">,</span><span class="s">&#39;fps&#39;</span><span class="p">,</span><span class="mi">10</span><span class="p">,</span><span class="s">&#39;length&#39;</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="s">&#39;azOffset&#39;</span><span class="p">,</span><span class="mi">180</span><span class="p">,</span><span class="s">&#39;zoom&#39;</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="s">&#39;elOffset&#39;</span><span class="p">,</span><span class="mi">90</span><span class="p">,</span><span class="s">&#39;closemovie&#39;</span><span class="p">,</span><span class="mi">0</span><span class="p">);</span> <span class="n">mov</span> <span class="o">=</span> <span class="n">make3davi</span><span class="p">(</span><span class="n">O</span><span class="p">);</span>
<span class="n">view</span><span class="p">(</span><span class="mi">90</span><span class="p">,</span><span class="mi">0</span><span class="p">);</span><span class="n">lightfollowview</span><span class="p">;</span><span class="n">O</span> <span class="o">=</span> <span class="n">struct</span><span class="p">(</span><span class="s">&#39;name&#39;</span><span class="p">,</span><span class="s">&#39;all4_union_bg_surf.avi&#39;</span><span class="p">,</span><span class="s">&#39;add2movie&#39;</span><span class="p">,</span><span class="n">mov</span><span class="p">,</span><span class="s">&#39;fps&#39;</span><span class="p">,</span><span class="mi">10</span><span class="p">,</span><span class="s">&#39;length&#39;</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="s">&#39;azOffset&#39;</span><span class="p">,</span><span class="mi">90</span><span class="p">,</span><span class="s">&#39;zoom&#39;</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="s">&#39;elOffset&#39;</span><span class="p">,</span><span class="o">-</span><span class="mi">90</span><span class="p">,</span><span class="s">&#39;closemovie&#39;</span><span class="p">,</span><span class="mi">0</span><span class="p">);</span> <span class="n">mov</span> <span class="o">=</span> <span class="n">make3davi</span><span class="p">(</span><span class="n">O</span><span class="p">);</span>
<span class="n">view</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="mi">90</span><span class="p">);</span><span class="n">lightfollowview</span><span class="p">;</span><span class="n">O</span> <span class="o">=</span> <span class="n">struct</span><span class="p">(</span><span class="s">&#39;name&#39;</span><span class="p">,</span><span class="s">&#39;all4_union_bg_surf.avi&#39;</span><span class="p">,</span><span class="s">&#39;add2movie&#39;</span><span class="p">,</span><span class="n">mov</span><span class="p">,</span><span class="s">&#39;fps&#39;</span><span class="p">,</span><span class="mi">10</span><span class="p">,</span><span class="s">&#39;length&#39;</span><span class="p">,</span><span class="mi">6</span><span class="p">,</span><span class="s">&#39;azOffset&#39;</span><span class="p">,</span><span class="o">-</span><span class="mi">90</span><span class="p">,</span><span class="s">&#39;zoom&#39;</span><span class="p">,</span><span class="mi">1</span><span class="p">,</span><span class="s">&#39;elOffset&#39;</span><span class="p">,</span><span class="mi">90</span><span class="p">,</span><span class="s">&#39;closemovie&#39;</span><span class="p">,</span><span class="mi">0</span><span class="p">);</span> <span class="n">mov</span> <span class="o">=</span> <span class="n">make3davi</span><span class="p">(</span><span class="n">O</span><span class="p">);</span>
<span class="n">mov</span> <span class="o">=</span> <span class="n">close</span><span class="p">(</span><span class="n">mov</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.make3Davi_uncompressed">
<code class="descclassname">Visualization_functions.</code><code class="descname">make3Davi_uncompressed</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.make3Davi_uncompressed" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>mov = make3davi([opt] Options_Structure)
</pre></div>
</div>
<p>Makes an avi movie file called head3d[x].avi or whatever you specify.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>O.name:</strong></dt>
<dd><p class="first last">&#8216;output_name.avi&#8217;;</p>
</dd>
<dt><strong>O.fps:</strong></dt>
<dd><p class="first last">frames per second</p>
</dd>
<dt><strong>O.length:</strong></dt>
<dd><p class="first last">length of movie in s</p>
</dd>
<dt><strong>O.H:</strong></dt>
<dd><p class="first last">cluster handles in vector   (for adjusting transparency with time)</p>
</dd>
<dt><strong>O.timecourse{i}:</strong></dt>
<dd><p class="first last">cell array of time courses for each cluster</p>
</dd>
<dt><strong>O.timeres:</strong></dt>
<dd><p class="first last">resolution, in s, of timecourse data</p>
</dd>
<dt><strong>O.azOffset:</strong></dt>
<dd><p class="first last">azimuth value to offset, positive = move clockwise</p>
</dd>
<dt><strong>O.elOffset:</strong></dt>
<dd><p class="first last">elevation value to move through, positive = inf to superior</p>
</dd>
<dt><strong>O.zoom:</strong></dt>
<dd><p class="first last">zoom value to end up with</p>
</dd>
<dt><strong>O.add2movie:</strong></dt>
<dd><p class="first last">add to existing movie - enter mov structure in this field</p>
</dd>
<dt><strong>O.closemovie:</strong></dt>
<dd><p class="first last">1 or 0, close the movie afterward or not.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Blank fields for az, el, zoom, timecourse indicate that these functions should not be performed
This script spirals up, right, and in 36 degrees</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Notes:</th><td class="field-body">my indeo5 one wouldn&#8217;t work in media player
also only seems to work if you add clusters before head isosurfaces</td>
</tr>
</tbody>
</table>
<p>Start with the image in the location you want to zoom in on
but with no zoom.</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.make_figure_into_orthviews">
<code class="descclassname">Visualization_functions.</code><code class="descname">make_figure_into_orthviews</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.make_figure_into_orthviews" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">hh1</span><span class="p">,</span><span class="n">hh2</span><span class="p">,</span><span class="n">hh3</span><span class="p">,</span><span class="n">hl</span><span class="p">,</span><span class="n">a1</span><span class="p">,</span><span class="n">a2</span><span class="p">,</span><span class="n">a3</span><span class="p">]</span> <span class="o">=</span> <span class="n">make_figure_into_orthviews</span>
</pre></div>
</div>
<p>Copies a surface rendering or glass brain into three separate view
panels, one sagittal, one axial, and one coronal</p>
<p>returns handles to objects in each view and hl light handles</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">hh1</span><span class="p">,</span><span class="n">hh2</span><span class="p">,</span><span class="n">hh3</span><span class="p">,</span><span class="n">hl</span><span class="p">,</span><span class="n">a1</span><span class="p">,</span><span class="n">a2</span><span class="p">,</span><span class="n">a3</span><span class="p">]</span> <span class="o">=</span> <span class="n">make_figure_into_orthviews</span><span class="p">;</span>
<span class="n">axes</span><span class="p">(</span><span class="n">a1</span><span class="p">)</span>
<span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mi">55</span><span class="p">,</span><span class="mi">60</span><span class="p">,</span><span class="mi">70</span><span class="p">,</span><span class="s">&#39;L&#39;</span><span class="p">,</span><span class="s">&#39;FontSize&#39;</span><span class="p">,</span><span class="mi">24</span><span class="p">);</span><span class="n">text</span><span class="p">(</span><span class="mi">55</span><span class="p">,</span><span class="mi">60</span><span class="p">,</span><span class="mi">70</span><span class="p">,</span><span class="s">&#39;R&#39;</span><span class="p">,</span><span class="s">&#39;FontSize&#39;</span><span class="p">,</span><span class="mi">24</span><span class="p">);</span>
<span class="n">axes</span><span class="p">(</span><span class="n">a3</span><span class="p">)</span>
<span class="n">text</span><span class="p">(</span><span class="o">-</span><span class="mi">55</span><span class="p">,</span><span class="mi">60</span><span class="p">,</span><span class="mi">70</span><span class="p">,</span><span class="s">&#39;L&#39;</span><span class="p">,</span><span class="s">&#39;FontSize&#39;</span><span class="p">,</span><span class="mi">24</span><span class="p">);</span><span class="n">text</span><span class="p">(</span><span class="mi">55</span><span class="p">,</span><span class="mi">60</span><span class="p">,</span><span class="mi">70</span><span class="p">,</span><span class="s">&#39;R&#39;</span><span class="p">,</span><span class="s">&#39;FontSize&#39;</span><span class="p">,</span><span class="mi">24</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.makelegend">
<code class="descclassname">Visualization_functions.</code><code class="descname">makelegend</code><span class="sig-paren">(</span><em>names</em>, <em>colors</em>, <em>makefig</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.makelegend" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre>makelegend(names,colors,[decimal places if numeric entries for names])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>names:</strong></dt>
<dd><p class="first last">must be cell array of names OR a vector of numbers (i.e., thresholds) that will be converted to
text</p>
</dd>
<dt><strong>colors:</strong></dt>
<dd><p class="first last">can be cell array of text or rgb values, or matrix of [r g b]
values</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>han = makelegend({&#39;red&#39; &#39;green&#39; &#39;blue&#39;}, {&#39;r&#39; &#39;g&#39; &#39;b&#39;});

han = makelegend([.001 .005 .01], {[1 1 0] [1 .5 0] [.7 .3 .3]});
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.map_data_to_colormap">
<code class="descclassname">Visualization_functions.</code><code class="descname">map_data_to_colormap</code><span class="sig-paren">(</span><em>datavaluesets</em>, <em>poscm</em>, <em>negcm</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.map_data_to_colormap" title="Permalink to this definition">¶</a></dt>
<dd><p>Usage</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">actcolor</span> <span class="o">=</span> <span class="n">map_data_to_colormap</span><span class="p">(</span><span class="n">datavaluesets</span><span class="p">,</span> <span class="n">poscm</span><span class="p">,</span> <span class="n">negcm</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Given sets of data values (each cell is a row vector of data values,
e.g., z-scores) and color maps for positive and negative values,
returns mapped colors for each data value in order.
These colors can be used for direct plotting.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt>input 1:</dt>
<dd><p class="first last">data values (e.g., z-scores). k data value sets, in cells.  Each cell contains row vector of data values</p>
</dd>
<dt>input 2/3:</dt>
<dd><p class="first last">color maps [n x 3] for positive and negative values</p>
</dd>
<dt>input 4:</dt>
<dd><p class="first last">optional: fixed range of data defining max and min colors</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>poscm = colormap_tor([0 0 0], [1 1 0]);
negcm = colormap_tor([0 0 1], [0 0 0]);
Z = randn(40, 1)&#39;;
actcolors = map_data_to_colormap({Z}, poscm, negcm)
[Z&#39; actcolors{1}]
</pre></div>
</div>
<p>Tor Wager, Sept. 2007
..</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.mask2surface">
<code class="descclassname">Visualization_functions.</code><code class="descname">mask2surface</code><span class="sig-paren">(</span><em>P</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.mask2surface" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">hPatch</span><span class="p">,</span><span class="n">outP</span><span class="p">,</span><span class="n">FV</span><span class="p">,</span> <span class="n">cl</span><span class="p">,</span> <span class="n">myLight</span><span class="p">]</span> <span class="o">=</span> <span class="n">mask2surface</span><span class="p">(</span><span class="n">P</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first docutils">
<dt><strong>P:</strong></dt>
<dd><p class="first last">a file name for a mask file, OR a clusters structure</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>hPatch:</strong></dt>
<dd><p class="first last">handle to surface patch</p>
</dd>
<dt><strong>outP:</strong></dt>
<dd><p class="first last">file name, with path, of mat file containing faces and vertices</p>
</dd>
<dt><strong>FV:</strong></dt>
<dd><p class="first last">isosurface</p>
</dd>
<dt><strong>cl:</strong></dt>
<dd><p class="first last">clusters structure with coords and meshgrid info</p>
</dd>
<dt><strong>varargin:</strong></dt>
<dd><p class="first last">suppress lighting (0) or do lighting (1)</p>
</dd>
<dt><strong>varargin{2}:</strong></dt>
<dd><p class="first last">color for patch objects</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>uses get_cluster_volume, adapted from Sergey Pakhomov</p>
<p>use with cluster_surf.m to map activations onto surfaces:
cluster_surf(clusters1,clusters2,outP,10,{[0 1 0] [1 0 0]})</p>
<p>or use getvertexcolors.m to map colors onto surface using hPatch.</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.mdsfig">
<code class="descclassname">Visualization_functions.</code><code class="descname">mdsfig</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.mdsfig" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">f1</span><span class="p">,</span><span class="n">hh</span><span class="p">]</span> <span class="o">=</span> <span class="n">mdsfig</span><span class="p">(</span><span class="n">pc</span><span class="p">,</span><span class="n">names</span><span class="p">,</span><span class="n">classes</span><span class="p">,</span><span class="n">linemat</span><span class="p">,</span><span class="n">cordir</span><span class="p">)</span>
</pre></div>
</div>
<p>Create the plot with stimulus coordinates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>pc:</strong></dt>
<dd><p class="first last">is objects x dimensions</p>
</dd>
<dt><strong>clus:</strong></dt>
<dd><p class="first last">is a vector of object classes</p>
</dd>
<dt><strong>names:</strong></dt>
<dd><p class="first last">is cell array of names for rows of pc (objects), or empty ([])</p>
</dd>
<dt><strong>sigm:</strong></dt>
<dd><p class="first last">is optional matrix of 1, -1, and 0 entries
signifies which pairs to connect with lines
positive elements are solid lines, negative elements are dashed</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>mdsfig(pc,clus,names,sigmat);

x = randn(5,2);
y = [1 1 1 2 2]&#39;;
c = eye(5); c(1,2) = 1; c(1,4) = -1; c(2,3) = 1;
mdsfig(x,y,[],c)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example2:</th><td class="field-body">using data output from mvroi.m</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>DAT.coords = CLUSTER.Gs(:,1:2); DAT.names = CLUSTER.names; DAT.classes =
CLUSTER.classes; DAT.lines = DATA.CORRELS.AVGSTATS.sigmat; 
figure;
mdsfig(DAT.coords,DAT.names,DAT.classes,DAT.lines);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.mdsfig_3d">
<code class="descclassname">Visualization_functions.</code><code class="descname">mdsfig_3d</code><span class="sig-paren">(</span><em>X</em>, <em>names</em>, <em>clus</em>, <em>linemat</em>, <em>cordir</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.mdsfig_3d" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre><span class="n">mdsfig_3d</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">names</span><span class="p">,</span><span class="n">clus</span><span class="p">,</span><span class="n">linemat</span><span class="p">,</span><span class="n">cordir</span><span class="p">)</span>
</pre></div>
</div>
<p>3-D MDS figure</p>
<p>Subfunction of mdsfig used if 3+ dims are available</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.mea_visualise">
<code class="descclassname">Visualization_functions.</code><code class="descname">mea_visualise</code><span class="sig-paren">(</span><em>plotmat</em>, <em>xaxis</em>, <em>yaxis</em>, <em>caxis</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.mea_visualise" title="Permalink to this definition">¶</a></dt>
<dd><p>This program allows the visualisation of 3D images in separate subplots</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.montage_clusters">
<code class="descclassname">Visualization_functions.</code><code class="descname">montage_clusters</code><span class="sig-paren">(</span><em>ovl</em>, <em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.montage_clusters" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">fig_handle</span> <span class="o">=</span> <span class="n">montage_clusters</span><span class="p">(</span><span class="n">ovl</span><span class="p">,</span> <span class="n">clusters</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Varargin:</th><td class="field-body"><p class="first">(in any order) =
a) additional clusters structures
b) cell array of colors (text format), must be ROW vector {&#8216;r&#8217; &#8216;g&#8217;} etc...</p>
<blockquote>
<div><p>if length of color string is longer than number of clusters inputs,
additional colors are interpreted as &#8216;2-intersection&#8217; and &#8216;all-intersection&#8217;
colors, in that order.  This overrides single string argument (c, below) for
color input</p>
</div></blockquote>
<ol class="loweralpha" start="3">
<li><dl class="first docutils">
<dt>single string color argument for overlaps (intersections)</dt>
<dd><p class="first last">plots intersections of ANY TWO clusters right now.
also color for plotting points, if entered.
use &#8216;nooverlap&#8217; as an input argument to suppress this.</p>
</dd>
</dl>
</li>
<li><p class="first">[n x 3] matrix of points to plot on map</p>
</li>
<li><p class="first">text labels for points, must be cell array in COLUMN vector</p>
</li>
<li><dl class="first docutils">
<dt>single number, 1/0 for whether to plot overlapping coordinates in overlap colors</dt>
<dd><p class="first last">default is 1.</p>
</dd>
</dl>
</li>
</ol>
<p>g) color limit vector [min max] indicates color mapping for blobs
rather than</p>
<blockquote class="last">
<div><p>solid colors.  This will do hot/cool mapping</p>
</div></blockquote>
</td>
</tr>
</tbody>
</table>
<p>Intersections of 2 colors are magenta, and ALL colors are yellow
unless otherwise specified
try:</p>
<div class="highlight-python"><div class="highlight"><pre>CLU = clusters2clu(clusters);
spm_orthviews(&#39;AddColouredBlobs&#39;, 1, CLU.XYZ, CLU.Z, CLU.M, [1 0 0])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.montage_clusters_maxslice">
<code class="descclassname">Visualization_functions.</code><code class="descname">montage_clusters_maxslice</code><span class="sig-paren">(</span><em>ovl</em>, <em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.montage_clusters_maxslice" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Varargin:</th><td class="field-body"><p class="first">(in any order) = 
a) additional clusters structures 
b) cell array of colors (text format), must be ROW vector {&#8216;r&#8217; &#8216;g&#8217;} etc...</p>
<blockquote>
<div><p>if length of color string is longer than number of clusters inputs,
additional colors are interpreted as &#8216;2-intersection&#8217; and &#8216;all-intersection&#8217; 
colors, in that order.  This overrides single string argument (c, below) for
color input</p>
</div></blockquote>
<ol class="last loweralpha" start="3">
<li><dl class="first docutils">
<dt>single string color argument for overlaps (intersections)</dt>
<dd><p class="first last">plots intersections of ANY TWO clusters right now.
also color for plotting points, if entered.
use &#8216;nooverlap&#8217; as an input argument to suppress this.</p>
</dd>
</dl>
</li>
<li><p class="first">[n x 3] matrix of points to plot on map</p>
</li>
<li><p class="first">text labels for points, must be cell array in COLUMN vector</p>
</li>
<li><dl class="first docutils">
<dt>single number, 1/0 for whether to plot overlapping coordinates in overlap colors</dt>
<dd><p class="first last">default is 1.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>color limit vector [min max] indicates color mapping for blobs rather than</dt>
<dd><p class="first last">solid colors.  This will do hot/cool mapping</p>
</dd>
</dl>
</li>
</ol>
</td>
</tr>
</tbody>
</table>
<p>Intersections of 2 colors are magenta, and ALL colors are yellow
unless otherwise specified</p>
<p>This maxslice version does 1) Not create a new figure, and 2) finds max
slice through clusters to create single slice image.</p>
<p>Useful for displaying next to timecourse plots (for example)
try:</p>
<div class="highlight-python"><div class="highlight"><pre>CLU = clusters2clu(clusters);
spm_orthviews(&#39;AddColouredBlobs&#39;,1,CLU.XYZ,CLU.Z,CLU.M,[1 0 0])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.montage_clusters_medial">
<code class="descclassname">Visualization_functions.</code><code class="descname">montage_clusters_medial</code><span class="sig-paren">(</span><em>ovl</em>, <em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.montage_clusters_medial" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Varargin:</th><td class="field-body"><p class="first">(in any order) =
a) additional clusters structures
b) cell array of colors (text format), must be ROW vector {&#8216;r&#8217; &#8216;g&#8217;} etc...</p>
<blockquote>
<div><p>if length of color string is longer than number of clusters inputs,
additional colors are interpreted as &#8216;2-intersection&#8217; and &#8216;all-intersection&#8217;
colors, in that order.  This overrides single string argument (c, below) for
color input</p>
</div></blockquote>
<ol class="last loweralpha" start="3">
<li><dl class="first docutils">
<dt>single string color argument for overlaps (intersections)</dt>
<dd><p class="first last">plots intersections of ANY TWO clusters right now.
also color for plotting points, if entered.
use &#8216;nooverlap&#8217; as an input argument to suppress this.</p>
</dd>
</dl>
</li>
<li><p class="first">[n x 3] matrix of points to plot on map</p>
</li>
<li><p class="first">text labels for points, must be cell array in COLUMN vector</p>
</li>
<li><dl class="first docutils">
<dt>single number, 1/0 for whether to plot overlapping coordinates in overlap colors</dt>
<dd><p class="first last">default is 1.</p>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>color limit vector [min max] indicates color mapping for blobs rather than</dt>
<dd><p class="first last">solid colors.  This will do hot/cool mapping</p>
</dd>
</dl>
</li>
</ol>
</td>
</tr>
</tbody>
</table>
<p>Intersections of 2 colors are magenta, and ALL colors are yellow
unless otherwise specified</p>
<p>try:</p>
<div class="highlight-python"><div class="highlight"><pre>CLU = clusters2clu(clusters);
spm_orthviews(&#39;AddColouredBlobs&#39;,1,CLU.XYZ,CLU.Z,CLU.M,[1 0 0])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.montage_clusters_points">
<code class="descclassname">Visualization_functions.</code><code class="descname">montage_clusters_points</code><span class="sig-paren">(</span><em>ovl</em>, <em>clusters</em>, <em>XYZpts</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.montage_clusters_points" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre><span class="n">ph</span> <span class="o">=</span> <span class="n">montage_clusters_points</span><span class="p">(</span><span class="n">ovl</span><span class="p">,</span><span class="n">clusters</span><span class="p">,</span><span class="n">XYZpts</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>varargin:</strong></dt>
<dd><p class="first last">additional clusters structures</p>
</dd>
<dt><strong>XYZpts:</strong></dt>
<dd><p class="first last">XYZ mm coordinates of points to plot</p>
</dd>
<dt><strong>ph:</strong></dt>
<dd><p class="first last">point handles</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.montage_clusters_text">
<code class="descclassname">Visualization_functions.</code><code class="descname">montage_clusters_text</code><span class="sig-paren">(</span><em>ovl</em>, <em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.montage_clusters_text" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>varargin:</strong></dt>
<dd><p class="first last">additional clusters structures
this function puts text cluster numbers on cluster centers</p>
</dd>
<dt><strong>color:</strong></dt>
<dd><p class="first last">cell array of text strings indicating colors {&#8216;r&#8217; &#8216;g&#8217;} etc...</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.montage_clusters_text2">
<code class="descclassname">Visualization_functions.</code><code class="descname">montage_clusters_text2</code><span class="sig-paren">(</span><em>cl</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.montage_clusters_text2" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="Visualization_functions.montage_image_Worsley">
<code class="descclassname">Visualization_functions.</code><code class="descname">montage_image_Worsley</code><span class="sig-paren">(</span><em>image_name</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.montage_image_Worsley" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre>data = montage_image_Worsley(3D or 4D image name)
</pre></div>
</div>
<p>Make a compact montage of some images
Designed by Keith Worsley for pca_image.m
Adapted by Tor Wager, Feb 2008</p>
<p>Limits and Colormap are designed for component loadings between [-1 1]</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>create_figure(&#39;Montage&#39;);
montage_image_Worsley(&#39;test_run1_pca.img&#39;);

data = montage_image_Worsley(imgname, &#39;pcacov&#39;)  % changes scaling and colormap
data = montage_image_Worsley(imgname, &#39;pcacov&#39;, [1 3 5])  % show only
                                                       volumes 1, 3, 5 in image(s)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.movie_of_slice_timeseries">
<code class="descclassname">Visualization_functions.</code><code class="descname">movie_of_slice_timeseries</code><span class="sig-paren">(</span><em>imgs</em>, <em>slicenumber</em>, <em>moviename</em>, <em>orientation</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.movie_of_slice_timeseries" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>imgs:</strong></dt>
<dd><p class="first last">Image names</p>
</dd>
<dt><strong>slicenumber:</strong></dt>
<dd><p class="first last">which slice to view (in volume)</p>
</dd>
<dt><strong>moviename:</strong></dt>
<dd><p class="first last">something like: &#8216;slice10_timeseries.avi&#8217;</p>
</dd>
<dt><strong>orientation:</strong></dt>
<dd><p class="first last">&#8216;axial&#8217; or &#8216;sagittal&#8217;</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">slicenumber</span> <span class="o">=</span> <span class="mi">10</span><span class="p">;</span>
<span class="n">moviename</span> <span class="o">=</span> <span class="s">&#39;slice10_timeseries.avi&#39;</span><span class="p">;</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.movie_stillframes">
<code class="descclassname">Visualization_functions.</code><code class="descname">movie_stillframes</code><span class="sig-paren">(</span><em>numframes</em>, <em>mov</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.movie_stillframes" title="Permalink to this definition">¶</a></dt>
<dd><p>Add still frames to a movie</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.multi_threshold">
<code class="descclassname">Visualization_functions.</code><code class="descname">multi_threshold</code><span class="sig-paren">(</span><em>P</em>, <em>type</em>, <em>df</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.multi_threshold" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre>cl = multi_threshold(P,type,df,[overlay image name])
</pre></div>
</div>
<p>F contrast:  df = xSPM.df;</p>
<p>type = &#8216;F&#8217; or &#8216;T&#8217;or &#8216;none&#8217;</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.mvroi_mdsfig_plot2">
<code class="descclassname">Visualization_functions.</code><code class="descname">mvroi_mdsfig_plot2</code><span class="sig-paren">(</span><em>CLUSTER</em>, <em>SPEC</em>, <em>sigmatavg</em>, <em>sigmatdif</em>, <em>titlestr</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.mvroi_mdsfig_plot2" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre><span class="n">mvroi_mdsfig_plot2</span><span class="p">(</span><span class="n">CLUSTER</span><span class="p">,</span><span class="n">SPEC</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">CORRELS</span><span class="o">.</span><span class="n">AVGSTATS</span><span class="o">.</span><span class="n">sigmat_uncorrected</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">CORRELS</span><span class="o">.</span><span class="n">DIFSTATS</span><span class="o">.</span><span class="n">sigmat_uncorrected</span><span class="p">,</span><span class="n">titlestr</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">mvroi_mdsfig_plot2</span><span class="p">(</span><span class="n">DATA</span><span class="o">.</span><span class="n">CLUSTER</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">SPEC</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">CORRELS</span><span class="o">.</span><span class="n">AVGSTATS</span><span class="o">.</span><span class="n">sigmat_uncorrected</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">CORRELS</span><span class="o">.</span><span class="n">DIFSTATS</span><span class="o">.</span><span class="n">sigmat_uncorrected</span><span class="p">,</span><span class="s">&#39;Uncorrected&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.mvroi_mdsfig_plot_sepstates">
<code class="descclassname">Visualization_functions.</code><code class="descname">mvroi_mdsfig_plot_sepstates</code><span class="sig-paren">(</span><em>CLUSTER</em>, <em>CORRELS</em>, <em>SPEC</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.mvroi_mdsfig_plot_sepstates" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre><span class="n">mvroi_mdsfig_plot_sepstates</span><span class="p">(</span><span class="n">CLUSTER</span><span class="p">,</span><span class="n">CORRELS</span><span class="p">,</span><span class="n">SPEC</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">CORRELS</span><span class="o">.</span><span class="n">AVGSTATS</span><span class="o">.</span><span class="n">sigmat_uncorrected</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">CORRELS</span><span class="o">.</span><span class="n">DIFSTATS</span><span class="o">.</span><span class="n">sigmat_uncorrected</span><span class="p">,</span><span class="n">titlestr</span><span class="p">)</span>
</pre></div>
</div>
<p>Plots SEPARATE STATES</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Example:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">mvroi_mdsfig_plot2</span><span class="p">(</span><span class="n">DATA</span><span class="o">.</span><span class="n">CLUSTER</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">SPEC</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">CORRELS</span><span class="o">.</span><span class="n">AVGSTATS</span><span class="o">.</span><span class="n">sigmat_uncorrected</span><span class="p">,</span><span class="n">DATA</span><span class="o">.</span><span class="n">CORRELS</span><span class="o">.</span><span class="n">DIFSTATS</span><span class="o">.</span><span class="n">sigmat_uncorrected</span><span class="p">,</span><span class="s">&#39;Uncorrected&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.mvroi_mdsfig_plugin2">
<code class="descclassname">Visualization_functions.</code><code class="descname">mvroi_mdsfig_plugin2</code><span class="sig-paren">(</span><em>CLUSTER</em>, <em>SPEC</em>, <em>sigmatavg</em>, <em>sigmatdif</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.mvroi_mdsfig_plugin2" title="Permalink to this definition">¶</a></dt>
<dd><p>mvroi_mdsfig_plugint2(CLUSTER,SPEC,DATA.CORRELS.AVGSTATS.sigmat_uncorrected,DATA.CORRELS.DIFSTATS.sigmat_uncorrected)</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.mvroi_plot_firs">
<code class="descclassname">Visualization_functions.</code><code class="descname">mvroi_plot_firs</code><span class="sig-paren">(</span><em>DATA</em>, <em>r</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.mvroi_plot_firs" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">h</span><span class="p">,</span><span class="n">t</span><span class="p">,</span><span class="n">w</span><span class="p">]</span> <span class="o">=</span> <span class="n">mvroi_plot_firs</span><span class="p">(</span><span class="n">DATA</span><span class="p">,</span><span class="n">region</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>DATA.SPEC.firnames = {&#39;Antic (C)&#39;  &#39;Pain (C)&#39; &#39;Response (C)&#39; &#39;Antic (P)&#39; &#39;Pain (P)&#39; &#39;Response (P)&#39;};
DATA.SPEC.firconditions = [1 4]
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.nmdsfig">
<code class="descclassname">Visualization_functions.</code><code class="descname">nmdsfig</code><span class="sig-paren">(</span><em>pc</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.nmdsfig" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre>f1 = nmdsfig(pc,[opt. inputs in any order])
</pre></div>
</div>
<p>Create a 1-D or 2-D plot with stimulus coordinates</p>
<dl class="docutils">
<dt>reserved keywords, each followed by appropriate input:</dt>
<dd><ul class="first last">
<li><p class="first">case &#8216;classes&#8217;, clus = varargin{i+1};</p>
</li>
<li><p class="first">case &#8216;names&#8217;, names = varargin{i+1};</p>
</li>
<li><p class="first">case &#8216;sig&#8217;, sigmat = varargin{i+1};</p>
</li>
<li><p class="first">case &#8216;thr&#8217;, thr = varargin{i+1};</p>
</li>
<li><p class="first">case &#8216;legend&#8217;, legmat = varargin{i+1};</p>
</li>
<li><p class="first">case &#8216;sig2&#8217;, sigmat2 = varargin{i+1};a
sigmat2 can be thresholded at multiple values in thr</p>
</li>
<li><p class="first">case &#8216;colors&#8217;, colors = varargin{i+1};</p>
</li>
<li><p class="first">case &#8216;sizes&#8217;, sizes = varargin{i+1};</p>
</li>
<li><p class="first">case &#8216;sigonly&#8217; plot regions with significant connections only</p>
</li>
<li><p class="first">case &#8216;nolines&#8217;, do not plot lines (lines plotted by default, but only if
sigmat is entered)</p>
<p>&#8216;linethickness&#8217;, followed by matrix of line thickness values</p>
<p>NOTE: can enter sig matrix with non-zero values equal to line
thickness and use for both sig and linethickness inputs
% but thickness values should be scaled to integers for line thickness</p>
</li>
</ul>
</dd>
</dl>
<p>Creates a figure only if f1 output is requested</p>
<blockquote>
<div><p>&#8216;fill&#8217;, fill in areas around groups</p>
<blockquote>
<div><dl class="docutils">
<dt><strong>pc:</strong></dt>
<dd>is objects x dimensions</dd>
<dt><strong>clus:</strong></dt>
<dd>is a vector of object classes</dd>
<dt><strong>names:</strong></dt>
<dd>is cell array of names for rows of pc (objects), or empty ([])</dd>
<dt><strong>sig:</strong></dt>
<dd><p class="first">is optional matrix of 1, -1, and 0 entries
signifies which pairs to connect with lines
positive elements are solid lines, negative elements are dashed</p>
<blockquote class="last">
<div><ul class="simple">
<li>Can be a series of t-maps in 3-D array</li>
</ul>
</div></blockquote>
</dd>
</dl>
<p>[opt] threshold vector of critical t-values, e.g., [2.2 5.4]
If used, enter t-maps in sigmat</p>
<p>[opt] a 2nd sigmat, if entered, will plot dashed lines instead of solid
ones. This is used by cluster_nmdsfig to plot interactions between
covariance and behavioral scores</p>
</div></blockquote>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Figure creation:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>If existing fig with tag &#8216;nmdsfig&#8217;, activates
Otherwise, if fig handle requested as output, creates
or if not, uses current figure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">c is output of cluster_nmdsfig</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>sizes = sum(c.STATS.sigmat);
f1 = nmdsfig(c.GroupSpace,&#39;classes&#39;,c.ClusterSolution.classes,&#39;names&#39;,c.names,&#39;sig&#39;,p_vs_c_heat.sig,&#39;legend&#39;,{&#39;Pos&#39; &#39;Neg&#39;},&#39;sizes&#39;,sizes,&#39;sizescale&#39;,[4 12]);

% add length legend
f1 =
nmdsfig(c.GroupSpace,&#39;classes&#39;,c.ClusterSolution.classes,&#39;names&#39;,c.names,&#39;sig&#39;,p_vs_c_heat.sig,&#39;legend&#39;,{&#39;Pos&#39; &#39;Neg&#39;}, ...
&#39;sizes&#39;,sizes,&#39;sizescale&#39;,[4 12],&#39;lengthlegend&#39;,c.r);

% Auto size scaling based on number of connections:
f1 =
nmdsfig(c.GroupSpace,&#39;classes&#39;,c.ClusterSolution.classes,&#39;names&#39;,c.names,&#39;sig&#39;,c.STATS.sigmat,&#39;legend&#39;,{&#39;Pos&#39; &#39;Neg&#39;},&#39;sizescale&#39;,[4 12],&#39;lengthlegend&#39;,c.r);

f1 =
nmdsfig(c.GroupSpace,&#39;classes&#39;,c.ClusterSolution.classes,&#39;names&#39;,c.names,&#39;sig&#39;,p_vs_c_heat.sig,&#39;legend&#39;,{&#39;Pos&#39; &#39;Neg&#39;},&#39;sizescale&#39;,[4 16],&#39;sigonly&#39;);
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">SEE ALSO:</th><td class="field-body">cluster_nmdsfig</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.nmdsfig1D">
<code class="descclassname">Visualization_functions.</code><code class="descname">nmdsfig1D</code><span class="sig-paren">(</span><em>pc</em>, <em>clus</em>, <em>names</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.nmdsfig1D" title="Permalink to this definition">¶</a></dt>
<dd><div class="highlight-python"><div class="highlight"><pre><span class="n">f1</span> <span class="o">=</span> <span class="n">nmdsfig</span><span class="p">(</span><span class="n">pc</span><span class="p">,</span><span class="n">clus</span><span class="p">,</span><span class="n">names</span><span class="p">)</span>
</pre></div>
</div>
<p>Create the plot with stimulus coordinates</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>pc:</strong></dt>
<dd><p class="first last">is objects x dimensions</p>
</dd>
<dt><strong>clus:</strong></dt>
<dd><p class="first last">is a vector of object classes</p>
</dd>
<dt><strong>names:</strong></dt>
<dd><p class="first last">is cell array of names for rows of pc (objects), or empty ([])</p>
</dd>
<dt><strong>sigm:</strong></dt>
<dd><p class="first last">is optional matrix of 1, -1, and 0 entries
signifies which pairs to connect with lines
positive elements are solid lines, negative elements are dashed</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>nmdsfig(pc,clus,names,sigmat);

x = randn(5,2);
y = [1 1 1 2 2]&#39;;
c = eye(5); c(1,2) = 1; c(1,4) = -1; c(2,3) = 1;
nmdsfig(x,y,[],c)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.nmdsfig_fill">
<code class="descclassname">Visualization_functions.</code><code class="descname">nmdsfig_fill</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.nmdsfig_fill" title="Permalink to this definition">¶</a></dt>
<dd><p>Purpose: to take information about objects in multidimensional space
and draw colored contours around them.
Used within nmdsfig.m</p>
<p>Usage:</p>
<p>1) If c is a structure from cluster_nmdsfig, which is compatible with
nmdsfig_tools, then:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">hh</span> <span class="o">=</span> <span class="n">nmdsfig_fill</span><span class="p">(</span><span class="n">c</span><span class="p">)</span>
</pre></div>
</div>
<p>Fields used are classes, groupSpace, and colors (see code for
details)</p>
<p>2) Pass in arguments directly:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">hh</span> <span class="o">=</span> <span class="n">nmdsfig_fill</span><span class="p">(</span><span class="n">classes</span><span class="p">,</span> <span class="n">positions</span><span class="p">,</span> <span class="n">colors</span><span class="p">)</span>
</pre></div>
</div>
<p>classes is n x 1 vector of group assignments (or all ones for one
group)</p>
<p>positions is n x 2 matrix of x, y coordinates
colors is a cell containing as many colors as classes, {&#8216;r&#8217; &#8216;g&#8217; &#8216;b&#8217; ...}
or {[1 0 0] [0 1 0] [0 0 1] ...}</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>load nmdsfig_output
hh = nmdsfig_fill(c)
set(findobj(&#39;Type&#39;,&#39;Line&#39;), &#39;Color&#39;, &#39;k&#39;)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.nmdsfig_legend">
<code class="descclassname">Visualization_functions.</code><code class="descname">nmdsfig_legend</code><span class="sig-paren">(</span><em>X</em>, <em>r</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.nmdsfig_legend" title="Permalink to this definition">¶</a></dt>
<dd><p>X is stim coords, r is correlation coefficient matrix
nmds figure should be current fig.</p>
<p>nmdsfig_legend(c.ClusterSolution.X,c.r)
get x limit and dims of current nmds fig.</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot3d">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot3d</code><span class="sig-paren">(</span><em>X</em>, <em>names</em>, <em>clus</em>, <em>linemat</em>, <em>cordir</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot3d" title="Permalink to this definition">¶</a></dt>
<dd><p>define scaling %%%%%%%</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plotDesign">
<code class="descclassname">Visualization_functions.</code><code class="descname">plotDesign</code><span class="sig-paren">(</span><em>ons</em>, <em>rt</em>, <em>TR</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plotDesign" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">X</span><span class="p">,</span><span class="n">d</span><span class="p">,</span><span class="n">out</span><span class="p">,</span><span class="n">handles</span><span class="p">]</span> <span class="o">=</span> <span class="n">plotDesign</span><span class="p">(</span><span class="n">ons</span><span class="p">,</span><span class="n">rt</span><span class="p">,</span><span class="n">TR</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>simple function to plot a design
plots regressors and color-coded onset times as little sticks, with RT represented as height of the stick</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>ons:</strong></dt>
<dd><p class="first">a cell array of onset times in s</p>
<p>OR a delta indicator function matrix
Event durations (durs) will ONLY work with cell array inputs</p>
<ul class="simple">
<li>optional *</li>
</ul>
<p class="last">The second column of each cell of ons can be a series of event durations for
each event.</p>
</dd>
<dt><strong>rt:</strong></dt>
<dd><p class="first last">is a cell array of rts or other parametric modulator for each onset event (or empty if no values)</p>
</dd>
<dt><strong>TR:</strong></dt>
<dd><p class="first last">repetition time for sampling, in s</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>returns the model matrix (X) and the delta function d</p>
<blockquote>
<div><dl class="docutils">
<dt>optional arguments</dt>
<dd><ol class="first last arabic simple">
<li>y offset for plotting rts, default = 2</li>
<li>vector of epoch durations in sec for each trial type, default is events</li>
</ol>
</dd>
<dt><strong>&#8216;yoffset&#8217;:</strong></dt>
<dd>followed by yoffset for plotting rts; default is auto scale</dd>
<dt><strong>&#8216;durs&#8217;:</strong></dt>
<dd><p class="first">followed by durations in sec, either:</p>
<p>Constant duration</p>
<p class="last">Vector of one duration for each event type
Cell array of one duration per trial
* Note: You can also add duration to ons input
instead; see ons above for more info *</p>
</dd>
<dt><strong>{&#8216;color&#8217;, &#8216;colors&#8217;}:</strong></dt>
<dd>followed by cell array of colors</dd>
<dt><strong>&#8216;samefig&#8217;:</strong></dt>
<dd>keep on same figure</dd>
<dt><strong>&#8216;basisset&#8217;:</strong></dt>
<dd>followed by name of basis set</dd>
<dt><strong>&#8216;overlapping&#8217;:</strong></dt>
<dd>Default is to plot separate lines in separate
vertical positions. To plot overlapping in same location, enter this.</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">plot epochs of different lengths stored in conditions(*).stimlength</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">X3</span><span class="p">,</span><span class="n">d</span><span class="p">]</span> <span class="o">=</span> <span class="n">plotDesign</span><span class="p">(</span><span class="n">evtonsets</span><span class="p">,[],</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="n">cat</span><span class="p">(</span><span class="mi">2</span><span class="p">,</span><span class="n">conditions</span><span class="o">.</span><span class="n">stimlength</span><span class="p">));</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">See Also:</th><td class="field-body">onsets2fmridesign</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_correlation">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_correlation</code><span class="sig-paren">(</span><em>X</em>, <em>Y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_correlation" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">handles</span> <span class="o">=</span> <span class="n">plot_correlation</span><span class="p">(</span><span class="n">X</span><span class="p">,</span><span class="n">Y</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>plots robust or OLS simple or partial correlations
replaces prplot and plot_correlation_samefig</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first last">is matrix of columns of interest plus nuisance
default is to plot partial effect of 1st column</p>
</dd>
<dt><strong>Y:</strong></dt>
<dd><p class="first last">is one or more columns of data</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;robust&#8217;:</strong></dt>
<dd><p class="first last">robust IRLS plot</p>
</dd>
<dt><strong>&#8216;noprint&#8217;:</strong></dt>
<dd><p class="first last">suppress text output</p>
</dd>
<dt><strong>&#8216;doquad&#8217;:</strong></dt>
<dd><p class="first last">quadratic term; not tested, may not work</p>
</dd>
<dt><strong>&#8216;col&#8217;:</strong></dt>
<dd><p class="first last">followed by column of interest</p>
</dd>
<dt><strong>&#8216;labels&#8217;:</strong></dt>
<dd><p class="first last">followed by cell array of text labels for each obs.</p>
</dd>
<dt><strong>&#8216;colors&#8217;:</strong></dt>
<dd><p class="first last">followed by cell array of colors for each column of Y</p>
</dd>
<dt><strong>&#8216;ylabel&#8217;:</strong></dt>
<dd><p class="first last">followed by y-axis label string</p>
</dd>
<dt><strong>&#8216;xlabel&#8217;:</strong></dt>
<dd><p class="first last">followed by x-axis label string</p>
</dd>
<dt><strong>&#8216;weights&#8217;:</strong></dt>
<dd><p class="first last">followed by weights that override any computed ones</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"><p class="first last">Plot robust partial corr. 2 of X against col. 17 of Y,</p>
</td>
</tr>
</tbody>
</table>
<p>controlling for other X</p>
<div class="highlight-python"><div class="highlight"><pre>figure;
h = plot_correlation(X,Y(:,17),&#39;col&#39;,2,&#39;robust&#39;,&#39;ylabel&#39;,&#39;Brain
                     data&#39;,&#39;xlabel&#39;,&#39;Order effect&#39;);
</pre></div>
</div>
<p>Plot Col. 1 of X vs. Y in red squares</p>
<div class="highlight-python"><div class="highlight"><pre>figure;
h = plot_correlation(X,Y(:,17),&#39;robust&#39;,&#39;colors&#39;,{&#39;rs&#39;});

tor_fig;
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_correlation_samefig">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_correlation_samefig</code><span class="sig-paren">(</span><em>xvec</em>, <em>yvec</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_correlation_samefig" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">r</span><span class="p">,</span><span class="n">infostring</span><span class="p">,</span><span class="n">sig</span><span class="p">,</span><span class="n">h</span><span class="p">]</span> <span class="o">=</span> <span class="n">plot_correlation_samefig</span><span class="p">(</span><span class="n">xvec</span><span class="p">,</span><span class="n">yvec</span><span class="p">,[</span><span class="n">textlabs</span><span class="p">],[</span><span class="n">color</span><span class="p">],[</span><span class="n">doquad</span><span class="p">],[</span><span class="n">dorobust</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>xvec:</strong></dt>
<dd><p class="first last">x vector</p>
</dd>
<dt><strong>yvec:</strong></dt>
<dd><p class="first last">y vector</p>
</dd>
<dt><strong>varargin</strong></dt>
<dd><p class="first last">string of text labels</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>doquad:</strong></dt>
<dd><p class="first last">flag for quadratic correlations as well!</p>
</dd>
<dt><strong>dorobust:</strong></dt>
<dd><p class="first last">remove n outliers from data, using Min Cov Determinant (MCD) 
Rousseeuw, P.J. (1984), &#8220;Least Median of Squares Regression,&#8221; 
Journal of the American Statistical Association, Vol. 79, pp. 871-88
outliers calculated using IRLS (robustfit.m) do not work well.
you enter n</p>
</dd>
</dl>
<p>empty variable arguments are OK, defaults will be used</p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>figure; [r, infos] = plot_correlation_samefig(x, y, [], &#39;ko&#39;, 0, 1);

% for text labels only, try:
plot_correlation(beh1,mri1,highlow,&#39;w.&#39;);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_dx_hrfs">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_dx_hrfs</code><span class="sig-paren">(</span><em>EXPT</em>, <em>clusters</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_dx_hrfs" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">EXPT</span> <span class="o">=</span> <span class="n">plot_dx_hrfs</span><span class="p">(</span><span class="n">EXPT</span><span class="p">,</span><span class="n">clusters</span><span class="p">,[</span><span class="n">dolegend</span><span class="p">],[</span><span class="n">dosave</span><span class="p">],[</span><span class="n">dosmooth</span><span class="p">],[</span><span class="n">doindiv</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first">uses EXPT.FIR and clusters</p>
<p>If not found, creates:</p>
<p>EXPT.FIR.regsofinterest = trial types
EXPT.FIR.mcol = colors</p>
<p>Seems to be a problem with showing the brain slice when it makes the
legend as well!  Weird bug.  Optional argument turns legend off.</p>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">(all defaults are zero)</p>
<dl class="docutils">
<dt><strong>dolegend:</strong></dt>
<dd><p class="first last">1 on, 0 off</p>
</dd>
<dt><strong>dosave:</strong></dt>
<dd><p class="first last">1 saves tiff files in timecourse_plots subdir, 0 does not</p>
</dd>
<dt><strong>dosmooth:</strong></dt>
<dd><p class="first last">n smooths hrfs and re-calculates st. errors, 0 plots saved
values stored in clusters.HRF.HRF and .STE</p>
</dd>
<dt><strong>doindiff:</strong></dt>
<dd><p class="first last">plot low vs. high groups of individuals (indiv diffs)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">See also:</th><td class="field-body"><ul class="first simple">
<li>extract_dxbeta_data.m, plot_dx_hrfs_indiffs.m</li>
</ul>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% has some smoothing (0 weight @ 3 time pts)
plot_dx_hrfs(EXPT,cl(1),0,1,3);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_ellipse">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_ellipse</code><span class="sig-paren">(</span><em>x</em>, <em>y</em>, <em>theta</em>, <em>a</em>, <em>b</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_ellipse" title="Permalink to this definition">¶</a></dt>
<dd><p>PLOT_ELLIPSE</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">h</span><span class="p">,</span><span class="n">h2</span><span class="p">]</span><span class="o">=</span><span class="n">plot_ellipse</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="p">,</span><span class="n">theta</span><span class="p">,</span><span class="n">a</span><span class="p">,</span><span class="n">b</span><span class="p">)</span>
</pre></div>
</div>
<p>This routine plots an ellipse with centre (x,y), axis lengths a,b
with major axis at an angle of theta radians from the horizontal.</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_error">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_error</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_error" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot a matrix with shaded error area around it</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[line_handle patch_handle] = plot_error(varargin)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% plots matrix Y against x-values in X, where Y is a matrix with 
% each row representing a signal. The shaded area represents the 
% standard error across the columns of Y.
PLOT_ERROR(X, Y) 

% plots the data matrix Y versus its index.
PLOT_ERROR(Y) 

% uses external data for the error areas. In this case, Y is assumed 
% to be a mean timeseries already. Y and errorData must be vectors 
% of the same length.
PLOT_ERROR(..., &#39;errorData&#39;, errorData)

% The following example indicates whether to handle NaNs in the data. 
% If set, plot_error will use nanmean, nanstd, etc. Off by default.
PLOT_ERROR(..., &#39;allowNaNs&#39;, [0|1]) 

% The following example plots the line according to the designated 
% ColorSpec string, and shades the error area by the color of the line
PLOT_ERROR(..., colorSpecString) 

% The following example plots into the axes designated by the AX 
% axes handle.
PLOT_ERROR(AX, ...) 

% The following example returns the handle of the main line object
[line_handle patch_handle] = PLOT_ERROR 
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_horizontal_line">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_horizontal_line</code><span class="sig-paren">(</span><em>y</em>, <em>color</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_horizontal_line" title="Permalink to this definition">¶</a></dt>
<dd><p>function han = plot_horizontal_line(y,color)</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_hrf_model_fit">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_hrf_model_fit</code><span class="sig-paren">(</span><em>m</em>, <em>TR</em>, <em>bf</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_hrf_model_fit" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>plot_hrf_model_fit(m,TR,bf,[stim input function for epoch, in 1 s resolution])
</pre></div>
</div>
<p>if m is an hrf curve sampled at 1 s,</p>
<p>plot_model_fit(m,1,&#8217;fir&#8217;);</p>
<p>plot_model_fit(m,1,&#8217;hrf&#8217;);</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_joint_hist_contour">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_joint_hist_contour</code><span class="sig-paren">(</span><em>z</em>, <em>xbins</em>, <em>ybins</em>, <em>color</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_joint_hist_contour" title="Permalink to this definition">¶</a></dt>
<dd><p>plot a 95% 2-D density region for a 2-D histogram</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">plot_joint_hist_contour</span><span class="p">(</span><span class="n">z</span><span class="p">,</span> <span class="n">xbins</span><span class="p">,</span> <span class="n">ybins</span><span class="p">,</span> <span class="n">color</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;confval&#39;</span><span class="p">,</span> <span class="n">confval</span><span class="p">],</span> <span class="p">[</span><span class="s">&#39;maxalpha&#39;</span><span class="p">,</span> <span class="n">maxalpha</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>z:</strong></dt>
<dd><p class="first last">2-D histogram values, counts in bins.  See joint_hist.m</p>
</dd>
<dt><strong>confval:</strong></dt>
<dd><p class="first last">optional input; [0 - 1], retain this proportion of values in confidence region</p>
</dd>
<dt><strong>maxalpha:</strong></dt>
<dd><p class="first last">optional; [0 - 1], maximum transparency</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>h:</strong></dt>
<dd><p class="first last">handle to graphical contour object</p>
</dd>
<dt><strong>z:</strong></dt>
<dd><p class="first last">thresholded z matrix of counts</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>z = joint_hist(nnmfscores{i}{j}(:, 1),nnmfscores{i}{j}(:, 2), 50, &#39;noplot&#39;);
h = plot_joint_hist_contour(z, [0 0 1]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_matrix_cols">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_matrix_cols</code><span class="sig-paren">(</span><em>X</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_matrix_cols" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>han = plot_matrix_cols(X, [method], [x-values], [colors cell], [linewidth], [axis limits])
</pre></div>
</div>
<p>Plot line plots showing each column of a matrix as a vertical or
horizontal line</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">han</span> <span class="o">=</span> <span class="n">plot_matrix_cols</span><span class="p">(</span><span class="n">X</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first last docutils">
<dt><strong>method:</strong></dt>
<dd><p class="first">plot_matrix_cols(X, &#8216;horiz&#8217;)</p>
<p class="last">plot_matrix_cols(X, &#8216;vertical&#8217;)</p>
</dd>
<dt><strong>x-values:</strong></dt>
<dd><p class="first last">plot_matrix_cols(X, &#8216;horiz&#8217;, x_in_secs)</p>
</dd>
<dt><strong>colors:</strong></dt>
<dd><p class="first last">plot_matrix_cols(X, &#8216;horiz&#8217;, [], {&#8216;r&#8217; &#8216;g&#8217; &#8216;b&#8217; &#8216;y&#8217; &#8216;m&#8217;})</p>
</dd>
<dt><strong>Linewidth:</strong></dt>
<dd><p class="first last">plot_matrix_cols(X, &#8216;horiz&#8217;, [], {&#8216;r&#8217; &#8216;g&#8217; &#8216;b&#8217; &#8216;y&#8217; &#8216;m&#8217;}, 2)</p>
</dd>
<dt><strong>axis limits:</strong></dt>
<dd><p class="first last">plot_matrix_cols(X, &#8216;horiz&#8217;, [], [], [], [0 10])</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>legacy &#8216;method&#8217; string:
within denoising, plot_matrix_cols(X, &#8216;denoising&#8217;) to make red plots</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.plot_vertical_line">
<code class="descclassname">Visualization_functions.</code><code class="descname">plot_vertical_line</code><span class="sig-paren">(</span><em>x</em>, <em>color</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.plot_vertical_line" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="function">
<dt id="Visualization_functions.prplot">
<code class="descclassname">Visualization_functions.</code><code class="descname">prplot</code><span class="sig-paren">(</span><em>yy</em>, <em>X</em>, <em>k</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.prplot" title="Permalink to this definition">¶</a></dt>
<dd><p>Partial residual plot of y ~ X for column k
Partial residual plot of one column of X against y. Uses IRLS estimation 
to downweight outliers if you enter a 4th argument.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">r</span><span class="p">,</span><span class="nb">str</span><span class="p">,</span><span class="n">sig</span><span class="p">,</span><span class="n">ry</span><span class="p">,</span><span class="n">rx</span><span class="p">,</span><span class="n">h</span><span class="p">,</span><span class="n">rr</span><span class="p">]</span> <span class="o">=</span> <span class="n">prplot</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="n">X</span><span class="p">,</span><span class="n">col</span><span class="p">,[</span><span class="n">dorobust</span><span class="p">],[</span><span class="n">colors</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>y</strong> </dt>
<dd><p class="first last">if y contains multiple columns, different colors and symbols will 
be used, with a separate regression for each.</p>
</dd>
<dt><strong>colors</strong></dt>
<dd><p class="first last">e.g., {&#8216;ro&#8217; &#8216;bs&#8217; &#8216;gd&#8217; &#8216;y^&#8217; &#8216;cv&#8217; &#8216;mx&#8217;}</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.renderCluster_ui">
<code class="descclassname">Visualization_functions.</code><code class="descname">renderCluster_ui</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.renderCluster_ui" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>O = renderCluster_ui([opt] O)
</pre></div>
</div>
<p>This ui is set to render on the default single_subj_T1, in default colors
More flexibility is available if you use the functions.</p>
<dl class="docutils">
<dt>main functions used:</dt>
<dd><ul class="first simple">
<li>tor_3d.m - images head with cutaway views</li>
<li>imageCluster.m - images a cluster isosurface</li>
<li>mni_TSU.m and tor_ihb_TalSpace.m - to get clusters
these and related functions are part of Talairach Space Utility
written by Sergey Pakhomov, 2001
modified very slightly by Tor Wager to not convert to Talairach Space
and use MNI coordinates instead.</li>
</ul>
<blockquote class="last">
<div>Use of the TSU functions for getting clusters require SPM99.</div></blockquote>
</dd>
</dl>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Output to workspace:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">Isosurface handles are in p, for head isosurfaces, and cH, for the cluster isosurface</p>
<p class="last">D = image data, Ds = smoothed data, hdr = img header</p>
</td>
</tr>
</tbody>
</table>
<p>Add more clusters by using:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cH</span><span class="p">(</span><span class="mi">2</span><span class="p">)</span> <span class="o">=</span> <span class="n">imageCluster</span><span class="p">(</span><span class="s">&#39;cluster&#39;</span><span class="p">,</span><span class="n">clusters</span><span class="p">(</span><span class="n">i</span><span class="p">));</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">O:</th><td class="field-body"><p class="first">= option structure with fields</p>
<dl class="last docutils">
<dt><strong>dohead:</strong></dt>
<dd><p class="first last">y/n add head surface</p>
</dd>
<dt><strong>head:</strong></dt>
<dd><p class="first last">filename or &#8216;default&#8217; for default canonical brain</p>
</dd>
<dt><strong>dobrain:</strong></dt>
<dd><p class="first last">y/n/filename add (transparent) brain surface, y for default brain or enter filename</p>
</dd>
<dt><strong>get_from:</strong></dt>
<dd><p class="first last">workspace/file/TSU/TSUfigure/none get clusters from here
if workspace, enter clusters in O.clusters</p>
</dd>
<dt><strong>which_cl:</strong></dt>
<dd><p class="first last">vector of clusters to image (from list)</p>
</dd>
<dt><strong>clcol:</strong></dt>
<dd><p class="first last">cluster colors - 3 el. vector, single letter, or string of letters
enter letter or letter string in single quotes. e.g., O.clcol = &#8216;yrgb&#8217;;</p>
</dd>
<dt><strong>whichc:</strong></dt>
<dd><p class="first last">letter string (no quotes) - which axes to cut along - xyzw are choices</p>
</dd>
<dt><strong>addtext:</strong></dt>
<dd><p class="first last">y/n add text to clusters</p>
</dd>
<dt><strong>textfield:</strong></dt>
<dd><p class="first last">field in cluster structure containing text</p>
</dd>
<dt><strong>textcol:</strong></dt>
<dd><p class="first last">character code (r, b, g, etc.) for color of text</p>
</dd>
<dt><strong>bestCoords:</strong></dt>
<dd><p class="first last">coordinates in mm to define x,y, and z cuts</p>
</dd>
<dt><strong>revx:</strong></dt>
<dd><p class="first last">text string to reverse x cut direction, enter 1 to do it.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>for solid brain rendering without the scalp, where dohead gives you a
brain image, use brain_render_T1.img</p>
<p>see cluster_cutaways.m for an easy-to-use version.</p>
<p>for transparent brain rendering of a set of clusters, try:</p>
<div class="highlight-python"><div class="highlight"><pre>figure(&#39;Color&#39;,&#39;w&#39;);
O = struct(&#39;dohead&#39;,&#39;n&#39;,&#39;dobrain&#39;,&#39;y&#39;,&#39;get_from&#39;,&#39;workspace&#39;, ...
            &#39;clusters&#39;,clusters, ...
            &#39;which_cl&#39;,1:length(clusters),&#39;whichc&#39;,&#39;y&#39;,&#39;bestCoords&#39;,[0 0 0],&#39;clcol&#39;,&#39;y&#39;,&#39;addtext&#39;,&#39;n&#39;, ...
            &#39;head&#39;,&#39;single_subj_T1&#39;);
renderCluster_ui(O)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.renderCluster_ui4">
<code class="descclassname">Visualization_functions.</code><code class="descname">renderCluster_ui4</code><span class="sig-paren">(</span><em>O</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.renderCluster_ui4" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">O</span> <span class="o">=</span> <span class="n">renderCluster_ui4</span><span class="p">(</span><span class="n">O</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first"><strong>O</strong></p>
<dl class="docutils">
<dt><strong>O.head:</strong></dt>
<dd><p class="first last">name of head to use, no .img extension</p>
</dd>
<dt><strong>O.surf:</strong></dt>
<dd><p class="first last">&#8216;y&#8217; or &#8216;n&#8217;: image surface</p>
</dd>
<dt><strong>O.sets:</strong></dt>
<dd><p class="first last">number of sets</p>
</dd>
<dt><strong>O.getfrom:</strong></dt>
<dd><p class="first last">cell array of where to get clusters from: &#8216;workspace&#8217; &#8216;file&#8217; etc.</p>
</dd>
<dt><strong>O.varname:</strong></dt>
<dd><p class="first last">cell array of variable names</p>
</dd>
<dt><strong>O.color:</strong></dt>
<dd><p class="first last">cell array of colors - strings or vectors</p>
</dd>
<dt><strong>O.numbers:</strong></dt>
<dd><p class="first last">&#8216;y&#8217; or &#8216;n&#8217;: add numbers to plot</p>
</dd>
</dl>
<p class="last">also need individual fields named contents of varname, which contain clusters</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.roi_contour_map">
<code class="descclassname">Visualization_functions.</code><code class="descname">roi_contour_map</code><span class="sig-paren">(</span><em>dat</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.roi_contour_map" title="Permalink to this definition">¶</a></dt>
<dd><p>Draw a pattern map of one slice (either sagittal, axial, or coronal) that
shows the most voxels, or the slice that you specify (e.g., x = #). 
You can also draw outlines for the significant voxels from a statistical test.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">info</span> <span class="o">=</span> <span class="n">roi_contour_map</span><span class="p">(</span><span class="n">dat</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>dat:</strong></dt>
<dd><p class="first">dat can be fmri_data, statistic_image (to mark significant 
voxels), and region objects. If your dat is the &#8220;region&#8221; 
object, please add &#8216;cluster&#8217; as an optional input. 
You can display two pattern maps for the purpose of
comparison by putting additional columns of data in cell array.</p>
<p>An example of displaying two pattern maps:</p>
<div class="last highlight-python"><div class="highlight"><pre>dat{1} = region(&#39;img1.nii&#39;); 
dat{2} = region(&#39;img2.nii&#39;); )
</pre></div>
</div>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;cluster&#8217;:</strong></dt>
<dd><p class="first last">When the data is a region object, you need this option.</p>
</dd>
<dt><strong>&#8216;sig&#8217;:</strong></dt>
<dd><p class="first last">This option outlines significant voxels. To use this
option, data in a format of statistic_image with a &#8220;sig&#8221; 
field should be given.</p>
</dd>
<dt><strong>&#8216;colorbar&#8217;:</strong></dt>
<dd><p class="first last">display colorbar under the plot.</p>
</dd>
<dt><strong>&#8216;use_same_range&#8217;:</strong></dt>
<dd><p class="first last">When you display two pattern maps, this option uses the
same color range for the two maps.</p>
</dd>
<dt><strong>&#8216;surf&#8217;:</strong></dt>
<dd><p class="first last">surface plot rather than voxel-by-voxel mapping.</p>
</dd>
<dt><strong>&#8216;xyz&#8217;:</strong></dt>
<dd><p class="first last">When you want a specific view and slice, you can use this
option with &#8216;coord&#8217;. (1:x - sagittal view, 2:y - coronal 
view, 3:z - axial view)</p>
</dd>
<dt><strong>&#8216;coord&#8217;:</strong></dt>
<dd><p class="first last">With &#8216;xyz&#8217; option, this specifies the slice displayed.</p>
</dd>
<dt><strong>&#8216;notfill&#8217;:</strong></dt>
<dd><p class="first last">Default is to fill in the blank voxels using a black
color. With this option, you can color the blank voxels
with the white color.</p>
</dd>
<dt><strong>&#8216;whole&#8217;:</strong></dt>
<dd><p class="first last">Default is dividing the data into contiguous regions and
show only one region that has the most voxels. This option
akes this function not to divide into contiguous regions.</p>
</dd>
<dt><strong>&#8216;colors&#8217; or &#8216;color&#8217;:</strong></dt>
<dd><p class="first last">you can specify your own colormap.</p>
</dd>
<dt><strong>&#8216;contour&#8217;:</strong></dt>
<dd><p class="first last">Not fully implemented yet.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>info:</strong></dt>
<dd><p class="first last">information about the display with the following fields.</p>
</dd>
<dt>info.dat:</dt>
<dd><ul class="first last simple">
<li>[2x30 double] (xyz mesh)</li>
<li>Z: [1x30 double] (z values)</li>
<li>xyz: 3 (1:x-sagittal, 2:y-coronal, 3:z-axial)</li>
<li>xyz_coord: -2 (slice coordinate; in this case, z = 3)</li>
<li>region_idx: 1</li>
</ul>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"><p class="first last">you can also see the same example and output in</p>
</td>
</tr>
</tbody>
</table>
<p><a class="reference external" href="http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery">http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery</a></p>
<div class="highlight-python"><div class="highlight"><pre>mask{1} = &#39;dACC_hw_pattern_sl6mm.nii&#39;;
mask{2} = &#39;dACC_rf_pattern_sl6mm.nii&#39;;
for i = 1:2, cl{i} = region(mask{i}); end
info = roi_contour_map([cl{1} cl{2}], &#39;cluster&#39;, &#39;use_same_range&#39;, &#39;colorbar&#39;); 
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.scn_export_papersetup">
<code class="descclassname">Visualization_functions.</code><code class="descname">scn_export_papersetup</code><span class="sig-paren">(</span><em>minsize</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.scn_export_papersetup" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>scn_export_papersetup([opt: min size in pixels, default = 400])
</pre></div>
</div>
<p>set paper size for current figure so that print to png or tiff looks as
it should (as it does on-screen)</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.scn_standard_colors">
<code class="descclassname">Visualization_functions.</code><code class="descname">scn_standard_colors</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.scn_standard_colors" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a set of unique colors in a standardized order.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">colors</span> <span class="o">=</span> <span class="n">scn_standard_colors</span><span class="p">(</span><span class="mi">100</span><span class="p">)</span>
</pre></div>
</div>
<p>Optional input: minimum number of colors to generate</p>
<p>Repeats after 36 colors
unique colors for each blob</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.selective_average_interactive_view_init">
<code class="descclassname">Visualization_functions.</code><code class="descname">selective_average_interactive_view_init</code><span class="sig-paren">(</span><em>imgs</em>, <em>onsets</em>, <em>scans_per_sess</em>, <em>TR</em>, <em>hp_length</em>, <em>t</em>, <em>basepts</em>, <em>plotstes</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.selective_average_interactive_view_init" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>selective_average_interactive_view_init(imgs, onsets, 20, 1:2, 1);
</pre></div>
</div>
<p>Initialize point-and-click data extraction and plotting of selective
averages</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>imgs:</strong></dt>
<dd><p class="first">cell array of image files, one cell per subject</p>
<p class="last">imgs is cell array of time series image names for each subject</p>
</dd>
<dt><strong>onsets:</strong></dt>
<dd><p class="first">cell array of onsets</p>
<p class="last">Each cell contains its own cell array, with one cell per
condition.</p>
</dd>
<dt><strong>t:</strong></dt>
<dd><p class="first last">time points in average to estimate</p>
</dd>
<dt><strong>basepts:</strong></dt>
<dd><p class="first last">indices of baseline points to subtract from individual averages</p>
</dd>
<dt><strong>plotstes:</strong></dt>
<dd><p class="first last">plot standard errors: 1 or 0</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.sepplot">
<code class="descclassname">Visualization_functions.</code><code class="descname">sepplot</code><span class="sig-paren">(</span><em>x</em>, <em>y</em>, <em>prop</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.sepplot" title="Permalink to this definition">¶</a></dt>
<dd><p>Draw a shorter line plots between points. To see a figure example, please visit
<a class="reference external" href="http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery">http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery</a>.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">sepplot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="n">prop</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>x, y:</strong></dt>
<dd><p class="first last">The function plots vector Y against vector X</p>
</dd>
<dt><strong>prop:</strong></dt>
<dd><p class="first last">The proportion of the lines: prop can be between 0 and 1</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">Enter keyword followed by variable with values</p>
<dl class="docutils">
<dt><strong>&#8216;color&#8217;:</strong></dt>
<dd><p class="first last">followed by color (e.g., &#8216;color&#8217;, [.5 .5 .5]) (default = black)</p>
</dd>
<dt><strong>&#8216;linewidth&#8217;:</strong></dt>
<dd><p class="first last">followed by a number for linewidth (e.g., &#8216;linewidth&#8217;, 2) (default = .5)</p>
</dd>
<dt><strong>&#8216;linestyle&#8217;:</strong></dt>
<dd><p class="first last">linestyle, e.g., followed by &#8216;-&#8216;, &#8216;&#8211;&#8217;, &#8216;:&#8217; (default = &#8216;-&#8216;)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>h:</strong></dt>
<dd><p class="first last">graphic handles for lines</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"><p class="first last">you can see the output in</p>
</td>
</tr>
</tbody>
</table>
<p><a class="reference external" href="http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery">http://wagerlab.colorado.edu/wiki/doku.php/help/core/figure_gallery</a></p>
<div class="highlight-python"><div class="highlight"><pre>x = 1:5; % x values
y = [32 40 55 84 130]; % mean
e = [6 6 6 6 6]; % standard error of the mean

create_figure(y_axis);
set(gcf, &#39;Position&#39;, [1   512   268   194]);
col = [0.3333    0.6588    1.0000];
markercol = col-.2;

h = errorbar(x, y, e, &#39;o&#39;, &#39;color&#39;, &#39;k&#39;, &#39;linewidth&#39;, 1.5, &#39;markersize&#39;, 7, &#39;markerfacecolor&#39;, col);
hold on;
sepplot(x, y, .75, &#39;color&#39;, col, &#39;linewidth&#39;, 2);
errorbar_width(h, x, [0 0]);

set(gca, &#39;xlim&#39;, [.5 5.5], &#39;linewidth&#39;, 1.5);

try
    pagesetup(gcf);
    saveas(gcf, &#39;example.pdf&#39;);
catch
    pagesetup(gcf);
    saveas(gcf, &#39;example.pdf&#39;);
end
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.shepardplot">
<code class="descclassname">Visualization_functions.</code><code class="descname">shepardplot</code><span class="sig-paren">(</span><em>D</em>, <em>ntest</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.shepardplot" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[Y,obs,imp,stress] = shepardplot(D, [ntest], [k dims to save])
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">is stimulus locations (cols are dims, rows are stimuli)
= eigenvectors of scalar product matrix from cmdscale</p>
</dd>
<dt><strong>obs:</strong></dt>
<dd><p class="first last">is vector of observed distances</p>
</dd>
<dt><strong>imp:</strong></dt>
<dd><p class="first last">is vector of implied distances</p>
</dd>
</dl>
<p class="last">squareform(obs) or (imp) = full matrix of distances</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.sphere_roi_tool">
<code class="descclassname">Visualization_functions.</code><code class="descname">sphere_roi_tool</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.sphere_roi_tool" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">clusters</span> <span class="o">=</span> <span class="n">sphere_roi_tool</span><span class="p">(</span><span class="s">&#39;mask&#39;</span><span class="p">,</span><span class="s">&#39;mask.img&#39;</span><span class="p">,</span><span class="s">&#39;bilat&#39;</span><span class="p">,</span><span class="mi">1</span><span class="p">)</span>
<span class="n">cl</span> <span class="o">=</span> <span class="n">sphere_roi_tool</span><span class="p">(</span><span class="s">&#39;mask&#39;</span><span class="p">,</span><span class="n">EXPT</span><span class="o">.</span><span class="n">mask</span><span class="p">,</span><span class="s">&#39;bilat&#39;</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="s">&#39;overlay&#39;</span><span class="p">,</span><span class="n">EXPT</span><span class="o">.</span><span class="n">overlay</span><span class="p">);</span>
</pre></div>
</div>
<p>A tool for building a number of spherical ROIs masked with some other
mask (e.g., gray matter).  High-level end-user function.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><p class="first">arguments are in string - value pairs, in any order</p>
<dl class="last docutils">
<dt><strong>mask:</strong></dt>
<dd><p class="first last">mask image for spheres, default:
which(&#8216;scalped_avg152T1_graymatter.img&#8217;);</p>
</dd>
<dt><strong>bilat:</strong></dt>
<dd><p class="first">make all rois bilateral, default is 1</p>
<p class="last">radius of sphere is defined on a region-by-region basis</p>
</dd>
<dt><strong>write:</strong></dt>
<dd><p class="first last">followed by name of image to write out</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>afterwards, try eliminating few-voxel regions:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span><span class="p">(</span><span class="n">find</span><span class="p">(</span><span class="n">cat</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="n">cl</span><span class="o">.</span><span class="n">numVox</span><span class="p">)</span><span class="o">&lt;</span><span class="mi">10</span><span class="p">))</span> <span class="o">=</span> <span class="p">[];</span>
</pre></div>
</div>
<p>and naming the clusters:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">clusters</span> <span class="o">=</span> <span class="n">cluster_names</span><span class="p">(</span><span class="n">clusters</span><span class="p">);</span>
</pre></div>
</div>
<p>draw on existing regions:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">sphere_roi_tool</span><span class="p">(</span><span class="s">&#39;mask&#39;</span><span class="p">,</span><span class="n">EXPT</span><span class="o">.</span><span class="n">mask</span><span class="p">,</span><span class="s">&#39;bilat&#39;</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="s">&#39;overlay&#39;</span><span class="p">,</span><span class="n">EXPT</span><span class="o">.</span><span class="n">overlay</span><span class="p">,</span><span class="s">&#39;add&#39;</span><span class="p">);</span>
</pre></div>
</div>
<p>use existing clusters</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cl</span> <span class="o">=</span> <span class="n">sphere_roi_tool</span><span class="p">(</span><span class="n">cl</span><span class="p">,</span><span class="s">&#39;mask&#39;</span><span class="p">,</span><span class="n">mask</span><span class="p">,</span><span class="s">&#39;bilat&#39;</span><span class="p">,</span><span class="mi">0</span><span class="p">,</span><span class="s">&#39;overlay&#39;</span><span class="p">,</span><span class="n">ovl</span><span class="p">,</span><span class="s">&#39;add&#39;</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.spm_orthviews_change_colormap">
<code class="descclassname">Visualization_functions.</code><code class="descname">spm_orthviews_change_colormap</code><span class="sig-paren">(</span><em>lowcolor</em>, <em>hicolor</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.spm_orthviews_change_colormap" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cm = spm_orthviews_change_colormap(lowcolor, hicolor, midcolor1, midcolor2, etc.)
</pre></div>
</div>
<p>Create a new split colormap of your choosing and apply it to the
spm_orthviews figure.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>spm_orthviews_change_colormap([.2 .2 .6], [1 1 0]);  % slate to yellow
spm_orthviews_change_colormap([.9 .5 .2], [1 1 0]);  % orange to yellow
spm_orthviews_change_colormap([.8 .1 .1], [1 1 0], [.9 .6 .1]);  %red to orange to yellow
spm_orthviews_change_colormap([.2 .2 .4], [1 1 0], [.9 .6 .1]);  %slate to orange to yellow
cm = spm_orthviews_change_colormap([0 0 1], [1 1 0], [0 .5 1], [0 .5 .5], ...
[0 1 .5], [0 1 0], [.5 1 0]);
cm = spm_orthviews_change_colormap([0 0 1], [1 1 0], [.5 0 1], [.5 .5 1], ... 
[1 .5 1], [1 .5 .5], [1 .5 0]);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.spm_orthviews_hotcool_colormap">
<code class="descclassname">Visualization_functions.</code><code class="descname">spm_orthviews_hotcool_colormap</code><span class="sig-paren">(</span><em>t</em>, <em>thr</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.spm_orthviews_hotcool_colormap" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">cm</span> <span class="o">=</span> <span class="n">spm_orthviews_hotcool_colormap</span><span class="p">(</span><span class="n">t</span><span class="p">,</span> <span class="n">thr</span><span class="p">)</span>
</pre></div>
</div>
<p>Create split-level colormap with hot colors for positive values and
cool colors for negative ones</p>
<p>Apply this colormap to the spm Graphics window (or any figure with the
tag &#8216;Graphics&#8217;)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>t:</strong></dt>
<dd><p class="first last">range of input statistic values (doesn&#8217;t have to be t-values)</p>
</dd>
<dt><strong>tthr:</strong></dt>
<dd><blockquote class="first">
<div><p>threshold more extreme than which (+/-) colors should be used</p>
</div></blockquote>
<p class="last">enter a positive value</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>Designed to work on blobs added using spm_orthviews &#8216;addblobs&#8217; feature</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">for using &#8216;clusters&#8217; in torlab format</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Threshold an image to get clusters:
[dat, volInfo, cl] = iimg_threshold(&#39;test_statistic.img&#39;, &#39;thr&#39;, 3.1440, &#39;k&#39;, 20);

% Display the clusters using spm_orthviews and apply the new color map:
cluster_orthviews(cl); cm = spm_orthviews_hotcool_colormap(cat(2,cl.Z), 3);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.spm_orthviews_name_axis">
<code class="descclassname">Visualization_functions.</code><code class="descname">spm_orthviews_name_axis</code><span class="sig-paren">(</span><em>name</em>, <em>axisnum</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.spm_orthviews_name_axis" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>spm_orthviews_name_axis(name, axis#)
</pre></div>
</div>
<p>put names on spm_orthviews axes</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.spm_orthviews_showposition">
<code class="descclassname">Visualization_functions.</code><code class="descname">spm_orthviews_showposition</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.spm_orthviews_showposition" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">handles</span> <span class="o">=</span> <span class="n">spm_orthviews_showposition</span><span class="p">;</span>
</pre></div>
</div>
<p>plots x, y, and z coordinates on SPM orthviews figure</p>
<p>bring context structure variable into this script.
contains handles, etc.</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.spm_orthviews_white_background">
<code class="descclassname">Visualization_functions.</code><code class="descname">spm_orthviews_white_background</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.spm_orthviews_white_background" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">spm_orthviews_white_background</span>
</pre></div>
</div>
<p>Set up orthviews to draw with a white background and softened grayscale
brain</p>
<p>No input arguments.  Updates spm&#8217;s global variable st</p>
<p>Uses SPM_OV_BLACK2WHITE.M</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.spm_ov_black2white">
<code class="descclassname">Visualization_functions.</code><code class="descname">spm_ov_black2white</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.spm_ov_black2white" title="Permalink to this definition">¶</a></dt>
<dd><p>Plugin for spm_orthviews to change the black background and dark edges to
a white background with softer gray edges, for pub. quality figures</p>
<p>if st (a global variable) st.plugins has &#8216;black2white&#8217; added, and the
st.vols{i} struct has a field called &#8216;black2white&#8217;, then this will be
called.</p>
<p>To initialize, add this code to the calling function:</p>
<div class="highlight-python"><div class="highlight"><pre>st.vols{1}.black2white = 1;
bwexist = strfind(st.plugins, &#39;black2white&#39;)
bwexist = any(cat(2, bwexist{:}))
if ~bwexist
    st.plugins{end+1} = &#39;black2white&#39;;
end
</pre></div>
</div>
<p>soften edges; 0 is no softening, a range is more softening</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.standardMRIlighting">
<code class="descclassname">Visualization_functions.</code><code class="descname">standardMRIlighting</code><span class="sig-paren">(</span><em>option</em>, <em>handles</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.standardMRIlighting" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">myLight</span> <span class="o">=</span> <span class="n">standardMRIlighting</span><span class="p">(</span><span class="n">option</span><span class="p">,</span><span class="n">handles</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>option:</strong></dt>
<dd><p class="first">&#8216;full&#8217; - all lighting adjustments</p>
<p class="last">&#8216;reflectance&#8217; - ambient strength and reflectance only</p>
</dd>
<dt><strong>handles:</strong></dt>
<dd><p class="first last">[isosurfaceHandle isocapsHandle]</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.surf_plot_tor">
<code class="descclassname">Visualization_functions.</code><code class="descname">surf_plot_tor</code><span class="sig-paren">(</span><em>data_matrix1</em>, <em>xvals</em>, <em>yvals</em>, <em>xname</em>, <em>yname</em>, <em>zname</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.surf_plot_tor" title="Permalink to this definition">¶</a></dt>
<dd><p>Stylized surface plot of one or two surfaces</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">data_matrix12</span><span class="p">,</span> <span class="n">xvals2</span><span class="p">,</span> <span class="n">yvals2</span><span class="p">,</span> <span class="n">data_matrix22</span><span class="p">]</span> <span class="o">=</span> <span class="n">surf_plot_tor</span><span class="p">(</span><span class="n">data_matrix1</span><span class="p">,</span> <span class="n">xvals</span><span class="p">,</span> <span class="n">yvals</span><span class="p">,</span> <span class="n">xname</span><span class="p">,</span> <span class="n">yname</span><span class="p">,</span> <span class="n">zname</span><span class="p">,</span> <span class="p">[</span><span class="n">data_matrix2</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body">See classify_search_script3...in meta-analysis classification</td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">surf_plot_tor</span><span class="p">(</span><span class="n">corrc_mean</span><span class="p">,</span> <span class="n">mya</span><span class="p">,</span> <span class="n">mys</span><span class="p">,</span> <span class="s">&#39;Activation feature cutoff&#39;</span><span class="p">,</span> <span class="s">&#39;Sensitivity feature cutoff&#39;</span><span class="p">,</span> <span class="s">&#39;Classification accuracy&#39;</span><span class="p">,</span> <span class="n">worstcat</span><span class="p">)</span>
</pre></div>
</div>
<p>xvals are columns, yvals are rows!</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.surface_cutaway">
<code class="descclassname">Visualization_functions.</code><code class="descname">surface_cutaway</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.surface_cutaway" title="Permalink to this definition">¶</a></dt>
<dd><p>Make a specialized cutaway surface and add blobs if entered</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">surface_handles</span> <span class="o">=</span> <span class="n">surface_cutaway</span><span class="p">(</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><p class="first">With no inputs, this function creates a vector of surface handles and
returns them in surface_handles</p>
<dl class="docutils">
<dt><strong>&#8216;cl&#8217;:</strong></dt>
<dd><p class="first last">followed by clusters structure or region object with blob info</p>
</dd>
<dt><strong>&#8216;surface_handles&#8217;:</strong></dt>
<dd><p class="first last">followed by vector of existing surface handles to plot blobs on</p>
</dd>
<dt><strong>&#8216;ycut_mm&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by y-cutoff in mm for coronal section.</dt>
<dd><ul class="first last simple">
<li>if absent, shows entire medial surface</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;existingfig&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>Use existing figure axis; do not create new one</dt>
<dd><ul class="first last simple">
<li>if &#8216;handles&#8217; are passed in, this is irrelevant</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;pos_colormap&#8217;:</strong></dt>
<dd><dl class="first last docutils">
<dt>followed by colormap for positive-going values</dt>
<dd><ul class="first last simple">
<li>n x 3 vector of rgb colors</li>
<li>see colormap_tor or matlab&#8217;s colormap</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;neg_colormap&#8217;:</strong></dt>
<dd><p class="first last">followed by colormap for negative-going values</p>
</dd>
<dt><strong>&#8216;color_upperboundpercentile&#8217;:</strong></dt>
<dd><p class="first last">followed by 1-100 percentile threshold; see Color values below</p>
</dd>
<dt><strong>&#8216;color_lowerboundpercentile&#8217;:</strong></dt>
<dd><p class="first last">followed by 1-100 percentile threshold; see Color values below</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>surface_handles:</strong></dt>
<dd><p class="first last">vector of surface handles for all surface objects</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Color values:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>Creates color scale using color-mapped colors, thresholded based on
percentile of the distribution of scores in .Z field.</p>
<blockquote>
<div><ul class="simple">
<li>To change the upper bound for which percentile of Z-scores is mapped to
the highest color value, enter &#8216;color_upperboundpercentile&#8217;, [new percentile from 1-100]</li>
<li>To change the lower bound value mapped to the lowest color value, 
enter &#8216;color_lowerboundpercentile&#8217; followed by [new percentile from 1-100]</li>
</ul>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% Create new surfaces:
% r is a region object made from a thresholded image (see region.m)
surface_handles = surface_cutaway(&#39;cl&#39;, r, &#39;ycut_mm&#39;, -30);

% Image blobs on existing handles:
surface_handles = surface_cutaway(&#39;cl&#39;, r, &#39;handles&#39;, surface_handles);

% Do it step by step:
p = surface_cutaway();
p = surface_cutaway(&#39;cl&#39;, r, &#39;surface_handles&#39;, p);

% Use custom colormaps (you can define &#39;pos_colormap&#39; and &#39;neg_colormap&#39;):
poscm = colormap_tor([.5 0 .5], [1 0 0]);  % purple to red
p = surface_cutaway(&#39;cl&#39;, r, &#39;surface_handles&#39;, p, &#39;pos_colormap&#39;, poscm);
p = surface_cutaway(&#39;cl&#39;, r, &#39;ycut_mm&#39;, -30, &#39;pos_colormap&#39;, poscm);

% use mediation_brain_surface_figs and re-make colors
all_surf_handles = mediation_brain_surface_figs([]);
surface(t2, &#39;cutaway&#39;, &#39;surface_handles&#39;, all_surf_handles, &#39;color_upperboundpercentile&#39;, 95, &#39;color_lowerboundpercentile&#39;, 5, &#39;neg_colormap&#39;, colormap_tor([0 0 1], [.2 0 .5]));
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.talairach_clusters">
<code class="descclassname">Visualization_functions.</code><code class="descname">talairach_clusters</code><span class="sig-paren">(</span><em>xyz</em>, <em>L</em>, <em>str</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.talairach_clusters" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>cl = talairach_clusters(xyz,L,str,[color, e.g., &#39;r&#39;])
</pre></div>
</div>
<p>Saves and/or visualizes contiguous clusters given coordinates from the Carmack Talairach atlas and a 
textlabel (e.g., &#8220;Amygdala&#8221;) for coordinates to retrieve</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>load talairach_info L5 x y z; xyz = [x y z]; 
cl = talairach_clusters(xyz,L5,&#39;Amygdala&#39;,&#39;y&#39;); save cl_amy cl

% OR

load 
cl = talairach_clusters(cl,[],[],&#39;y&#39;);
</pre></div>
</div>
<p>get clusters structure given an XYZ mm list, a list of labels L in a cell array, and a
string to match labels to</p>
<p>Complete example of making a limbic figure (some parts shown)</p>
<div class="highlight-python"><div class="highlight"><pre>load talairach_info L5 L3 x y z; xyz = [x y z];
cl = talairach_clusters(xyz,L5,&#39;Amygdala&#39;,&#39;y&#39;);
cl = talairach_clusters(xyz,L3,&#39;Caudate&#39;,&#39;b&#39;);
cl = talairach_clusters(xyz,L5,&#39;Putamen&#39;,&#39;g&#39;); save cl_putamen cl
cl = talairach_clusters(xyz,L5,&#39;Lateral Globus Pallidus&#39;,&#39;c&#39;); save cl_glo1 cl
cl = talairach_clusters(xyz,L5,&#39;Medial Globus Pallidus&#39;,&#39;c&#39;); save cl_glo2 cl
axis off
set(gcf,&#39;Color&#39;,&#39;w&#39;)
h = lightangle(0,30);
material dull
h2 = lightangle(0,-30);
h3 = lightangle(90,-30);
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.timeseries_prplot">
<code class="descclassname">Visualization_functions.</code><code class="descname">timeseries_prplot</code><span class="sig-paren">(</span><em>y</em>, <em>X</em>, <em>cols</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.timeseries_prplot" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">r</span><span class="p">,</span><span class="n">bb</span><span class="p">,</span><span class="n">f</span><span class="p">]</span> <span class="o">=</span> <span class="n">timeseries_prplot</span><span class="p">(</span><span class="n">y</span><span class="p">,</span><span class="n">X</span><span class="p">,</span><span class="n">cols</span><span class="p">,</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>Plots timeseries data (y&#8217;) against fitted response (X)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>y:</strong></dt>
<dd><p class="first">y is n x 1 data points, X is n x k model matrix of predictors</p>
<p class="last">y is adjusted to remove all effects OTHER THAN those columns of X
specified in cols, creating a partial residual vector y&#8217;.</p>
</dd>
<dt><strong>X:</strong></dt>
<dd><p class="first last">The fitted response to X(:,cols) is plotted against y&#8217; to graphically assess the
effect of particular columns of X on y.</p>
</dd>
<dt>varargin includes two optional arguments:</dt>
<dd><ol class="first last arabic simple">
<li>a vector of trial onsets (to shade in plot)</li>
<li>length of elements to shade after trial onset.</li>
</ol>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>r:</strong></dt>
<dd><p class="first last">partial residuals</p>
</dd>
<dt><strong>bb:</strong></dt>
<dd><p class="first last">betas</p>
</dd>
<dt><strong>f:</strong></dt>
<dd><p class="first last">partial fitted response</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>timeseries_prplot(Yfla,X,[2 4],x2,18);
</pre></div>
</div>
<p>to average across sessions 1 and 2:</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.tor_3d">
<code class="descclassname">Visualization_functions.</code><code class="descname">tor_3d</code><span class="sig-paren">(</span><em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.tor_3d" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="p">[</span><span class="n">D</span><span class="p">,</span><span class="n">Ds</span><span class="p">,</span><span class="n">hdr</span><span class="p">,</span><span class="n">p</span><span class="p">,</span><span class="n">coords</span><span class="p">,</span><span class="n">X</span><span class="p">,</span><span class="n">Y</span><span class="p">,</span><span class="n">Z</span><span class="p">]</span> <span class="o">=</span> <span class="n">tor_3d</span><span class="p">(</span><span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<p>made to use single_subj_T1.img from SPM99</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Options:</th><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;data&#8217;:</strong></dt>
<dd><p class="first last">followed by image data, must also use &#8216;hdr&#8217;; data is full image volume</p>
</dd>
<dt><strong>&#8216;hdr&#8217;:</strong></dt>
<dd><p class="first last">followed by hdr structure (use read_hdr)</p>
</dd>
<dt><strong>&#8216;coords&#8217;:</strong></dt>
<dd><p class="first last">3 element row vector of x,y,z coordinates at which to cut away</p>
</dd>
<dt><strong>&#8216;figure&#8217;:</strong></dt>
<dd><p class="first last">create a new figure to plot on</p>
</dd>
<dt><strong>&#8216;whichcuts&#8217;:</strong></dt>
<dd><p class="first">followed by incisions; choices are x y z w (whole)</p>
<p>example:&#8217;whichcuts&#8217;,&#8217;xyz&#8217; (xyz is default)
order of output handles (p(i)) is &#8216;wyzx&#8217;</p>
<dl class="last docutils">
<dt>New: Special methods:</dt>
<dd><ul class="first last simple">
<li>&#8216;coronal slice right&#8217;</li>
<li>&#8216;coronal slice left&#8217;</li>
<li>&#8216;coronal slice&#8217;</li>
</ul>
</dd>
</dl>
</dd>
<dt><strong>&#8216;filename&#8217;:</strong></dt>
<dd><p class="first">followed by filename of image data to load, in single quotes</p>
<p>should be analyze img format, without the .img extension</p>
<p class="last">cluster imaging assumes neurological orientation, but should work anyway.</p>
</dd>
<dt><strong>&#8216;revx&#8217;:</strong></dt>
<dd><p class="first last">reverse x cut direction, so cut in from left instead of right</p>
</dd>
<dt><strong>&#8216;topmm&#8217;:</strong></dt>
<dd><p class="first last">topmm = varargin{i+1};</p>
</dd>
<dt><strong>&#8216;intensity_threshold&#8217;:</strong></dt>
<dd><p class="first last">percentile of data above which is considered in-object
higher = more sparse object</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>D:</strong></dt>
<dd><p class="first last">img data</p>
</dd>
<dt><strong>Ds:</strong></dt>
<dd><p class="first last">smoothed data</p>
</dd>
<dt><strong>hdr:</strong></dt>
<dd><p class="first last">header</p>
</dd>
<dt><strong>p:</strong></dt>
<dd><p class="first last">image handles</p>
</dd>
<dt><strong>coords:</strong></dt>
<dd><p class="first last">coordinates</p>
</dd>
<dt><strong>X, Y, Z:</strong></dt>
<dd><p class="first last">are the millimeter, origin centered reference frame for the head isosurface</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[D,Ds,hdr,p,coords] = tor_3d(&#39;figure&#39;,&#39;data&#39;,D,&#39;hdr&#39;,hdr,&#39;whichcuts&#39;,&#39;yzx&#39;);
[D,Ds,hdr,p,coords] = tor_3d(&#39;figure&#39;);
[D,Ds,hdr,headhandle,coords] = tor_3d(&#39;whichcuts&#39;,&#39;z&#39;, &#39;coords&#39;, [-Inf
-Inf Inf], &#39;filename&#39;, &#39;T1_face_exemplar&#39;, &#39;intensity_threshold&#39;, 80); set(gcf, &#39;Color&#39;, &#39;w&#39;); axis on

% Special slice example:
[D,Ds,hdr,handle,coords]  = tor_3d(&#39;whichcuts&#39;, &#39;coronal slice right&#39;, &#39;coords&#39;, [0 12 0], &#39;topmm&#39;, 100);
lightRestoreSingle
set(handle(1), &#39;FaceColor&#39;, [.5 .5 .5])
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.tor_fill_steplot">
<code class="descclassname">Visualization_functions.</code><code class="descname">tor_fill_steplot</code><span class="sig-paren">(</span><em>dat</em>, <em>color</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.tor_fill_steplot" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>[h,t] = tor_fill_steplot(dat,color,[robust flag],[p-thresh],[x vector],[covs no interest])
</pre></div>
</div>
<p>Plots a mean vector (mean of each column of dat)
surrounded by a fill with standard err bars</p>
<p>If dat has 3 dimensions, then
the diff between dat(:,:,1) and dat(:,:,2) is
used as the difference for computing standard err
(as in repeated measures)</p>
<p>if behavior is entered as optional argument, removes it before plotting
lines.  Also returns adjusted output in d, dat</p>
<p>Optional: robust flag (1/0), robust IRLS</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">tor_fig</span><span class="p">;</span>
<span class="n">tor_fill_steplot</span><span class="p">(</span><span class="n">dat</span><span class="p">,{</span><span class="s">&#39;b&#39;</span> <span class="s">&#39;r&#39;</span><span class="p">},</span><span class="mi">0</span><span class="p">,</span><span class="o">.</span><span class="mo">05</span><span class="p">,</span><span class="n">secs</span><span class="p">);</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.tor_ihb_GetClusterSet">
<code class="descclassname">Visualization_functions.</code><code class="descname">tor_ihb_GetClusterSet</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.tor_ihb_GetClusterSet" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">bIsEmpty</span> <span class="o">=</span> <span class="n">ihb_GetClusterSet</span>
</pre></div>
</div>
<p>Function to select set of cluster (SPM.mat) and select one cluster
from this set</p>
<dl class="docutils">
<dt>Return value == 1 if cluster set is nonempty and one cluster selected</dt>
<dd>0 otherwise</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.tor_ihb_GetClusters">
<code class="descclassname">Visualization_functions.</code><code class="descname">tor_ihb_GetClusters</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.tor_ihb_GetClusters" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">clusters</span> <span class="o">=</span> <span class="n">ihb_getClusters</span>
</pre></div>
</div>
<p>Get cluster information (use [SPM,VOL,xX,xCon,xSDM] = spm_getSPM;)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>clusters:</strong></dt>
<dd><p class="first last">array of structs with fields:</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Common to all clusters in the set:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>isSpmCluster:</strong></dt>
<dd><dl class="first last docutils">
<dt>1 if got from SPM data; 0 - if constructed by </dt>
<dd><p class="first last">symmetrical or intersection</p>
</dd>
</dl>
</dd>
<dt><strong>title:</strong></dt>
<dd><p class="first last">title for comparison (string) (SPM.title)</p>
</dd>
<dt><strong>hThreshold:</strong></dt>
<dd><p class="first last">height threshold (SPM.u)</p>
</dd>
<dt><strong>voxSize:</strong></dt>
<dd><p class="first last">voxel dimensions {mm} - column vector (VOL.VOX)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name" colspan="2">Specific for every cluster:</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>name:</strong></dt>
<dd><p class="first last">name of the cluster</p>
</dd>
<dt><strong>numVox:</strong></dt>
<dd><p class="first last">number of voxels in cluster</p>
</dd>
<dt><strong>Z:</strong></dt>
<dd><p class="first last">minimum of n Statistics {filtered on u and k} (1 x num_vox)</p>
</dd>
<dt><strong>XYZmm:</strong></dt>
<dd><p class="first last">location of voxels {mm} (3 x num_vox)</p>
</dd>
<dt><strong>pVoxelLev:</strong></dt>
<dd><p class="first last">corrected p for max value in cluster</p>
</dd>
<dt><strong>pClustLev:</strong></dt>
<dd><p class="first last">corrected p for given cluster (cluster level)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Talariach volume:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>xTal:</strong></dt>
<dd><p class="first last">x Talariach coordinates ready for contourslice &amp; isosurface</p>
</dd>
<dt><strong>yTal:</strong></dt>
<dd><p class="first last">y Talariach coordinates ready for contourslice &amp; isosurface</p>
</dd>
<dt><strong>zTal:</strong></dt>
<dd><p class="first last">z Talariach coordinates ready for contourslice &amp; isosurface</p>
</dd>
<dt><strong>vTal:</strong></dt>
<dd><p class="first last">Talariach volume values ready for contourslice &amp; isosurface</p>
</dd>
</dl>
<p class="last">xMin, yMin, zMin, xMax, yMax, zMax - bounding box in mm for Talariach</p>
</td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.tor_ihb_TalSpace">
<code class="descclassname">Visualization_functions.</code><code class="descname">tor_ihb_TalSpace</code><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.tor_ihb_TalSpace" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"><div class="first highlight-python"><div class="highlight"><pre><span class="n">tor_ihb_TalSpace</span>
</pre></div>
</div>
<p class="last">Main function to view SPM99 cluster&#8217;s (contours &amp; 3D) in Talariach space</p>
</td>
</tr>
</tbody>
</table>
<blockquote>
<div>Graphic objects and their &#8216;Tags&#8217;:</div></blockquote>
<blockquote>
<div>type        handle          Tag                     Description</div></blockquote>
<blockquote>
<div><p>figure      hFigMain        ihb_TalSpaceMain_fig    Main figure    
figure      hFig3D          ihb_TalSpace3D_fig      3D view figure
figure      hFigLeg         ihb_TalSpaceLeg_fig     Legend figure</p>
<p>axes        hAxis3D         ihb_axesTalSpace3D      Axes for 3D view
light       hLight          ihb_lightTalSpace3D     Light object for 3D view</p>
<p>patch                       ihb_cntAxial            contours patch handles
patch                       ihb_cntFrontal              in 3D view window
patch                       ihb_cntSaggitalL
patch                       ihb_cntSaggitalR</p>
<p>patch                       ihb_surfaceCluster      isosurface patch</p>
<p>uicontrol                   ihb_CursorCoord         coordinates under cursor
uicontrol                   ihb_ClusterLevelProb    cluster probability
uicontrol                   ihb_VoxelLevelProb      extr. probability
uicontrol                   ihb_VolumeInVox         cluster volume
uicontrol                   ihb_ClusterPopUp        select cluster combo box</p>
<p>uicontrol                   ihb_clr_Orig            pushbuttons for
uicontrol                   ihb_clr_Symm            color change in
uicontrol                   ihb_clr_Surface         legend window
uicontrol                   ihb_clr_Axial
uicontrol                   ihb_clr_Frontal
uicontrol                   ihb_clr_Saggital</p>
<p>line                        ihb_LineOriginal        lines for original clusters
line                        ihb_LineSymmetrical     lines for symmetrical clusters</p>
<p>uimenu                      ihb_VoxSizeMenu111
uimenu                      ihb_VoxSizeMenu222
uimenu                      ihb_VoxSizeMenu444</p>
<p>uimenu                      ihb_RendererMenuZbuffer
uimenu                      ihb_RendererMenuOpenGL</p>
<p>uimenu                      ihb_TransparencyMenu
uimenu                      ihb_TransparencyMenuOpaque
uimenu                      ihb_TransparencyMenuLow
uimenu                      ihb_TransparencyMenuMedium
uimenu                      ihb_TransparencyMenuHigh</p>
<p>uimenu                      ihb_Axes3DMenuClusterOnly
uimenu                      ihb_Axes3DMenuWholeBrain</p>
<p>uimenu                      ihb_LineWidthMenuThin
uimenu                      ihb_LineWidthMenuNormal
uimenu                      ihb_LineWidthMenuThick</p>
</div></blockquote>
<blockquote>
<div>Application data (name used with setappdata the same as handle or variable name</div></blockquote>
<blockquote>
<div><p>for hFigMain:</p>
<blockquote>
<div>axes            hAxAxial            axial               
axes            hAxFront            frontal      
axes            hAxSagL             left sagittal       
axes            hAxSagR             right sagittal       
strucure array  clusters            array of selected clusters
integer         indClusterToView    index of currently viewed cluster in clusters array
bool            drawSymLS           == 1 if draw symmetrical on left sagittal
bool            drawSymRS           == 1 if draw symmetrical on right sagittal</div></blockquote>
</div></blockquote>
<blockquote>
<div><dl class="docutils">
<dt>for hAxis (where hAxis is one of hAxAxial, hAxFront, hAxSagL, hAxSagR):</dt>
<dd><p class="first">(see ihb_LoadSlice for detail)</p>
<p class="last">strucure array  sliceInfo           array of Talarich slice information (ihbdfl_ax_info etc.)  
integer array   rngArray            array of local indexes in sliceInfo
integer         sliceIndex          index of slice in the local range array rngArray
double          sliceDist           slice distance from origin
string          axType              axis type: &#8216;ax&#8217; &#8216;fr&#8217; &#8216;sl&#8217; &#8216;sr&#8217; 
1x2 vector      xLim                limit values for X direction
1x2 vector      yLim                limit values for Y direction</p>
</dd>
</dl>
</div></blockquote>
<blockquote>
<div>Default information in ihb_TalSpaceDfl.mat file (see ihb_ResetDefaults)</div></blockquote>
<blockquote>
<div><p>To get default variable with name var use:</p>
<p>load(&#8216;ihb_TalSpaceDfl.mat&#8217;, &#8216;var&#8217;);</p>
<p>To save:</p>
<p>save(&#8216;ihb_TalSpaceDfl.mat&#8217;, &#8216;var&#8217;, &#8216;-append&#8217;);</p>
</div></blockquote>
<blockquote>
<div>Figures tags for windows created by SPM during call spm_getSPM</div></blockquote>
<blockquote>
<div>ihbdfl_spm_fig_Interactive
ihbdfl_spm_fig_SelFileWin
ihbdfl_spm_fig_ConMan</div></blockquote>
<blockquote>
<div>Minimal size of clusters to draw</div></blockquote>
<blockquote>
<div>ihbdfl_min_size_to_draw</div></blockquote>
<blockquote>
<div>Sizes of voxels used in Talariach space</div></blockquote>
<blockquote>
<div>ihbdfl_tal_x_vox
ihbdfl_tal_y_vox
ihbdfl_tal_z_vox</div></blockquote>
<blockquote>
<div>Sizes of main window and axis for ihb_TalSpace</div></blockquote>
<blockquote>
<div>ihbdfl_main_width
ihbdfl_main_height
ihbdfl_main_gap
ihbdfl_main_x
ihbdfl_main_y
ihbdfl_main_z
ihbdfl_main_info_h</div></blockquote>
<blockquote>
<div>Talariach slices information</div></blockquote>
<blockquote>
<div>ihbdfl_ax_info
ihbdfl_fr_info
ihbdfl_sl_info
ihbdfl_sr_info</div></blockquote>
<blockquote>
<div>Axis limits to draw slice</div></blockquote>
<blockquote>
<div>ihbdfl_xLimD
ihbdfl_yLimD
ihbdfl_zLimD
ihbdfl_xLimI
ihbdfl_yLimI
ihbdfl_zLimI</div></blockquote>
<blockquote>
<div>3D view Renderer (&#8216;zbuffer&#8217; or &#8216;OpenGL&#8217;)
3D view transparency (available only for OpenGL</div></blockquote>
<blockquote>
<div>ihbdfl_renderer
ihbdfl_transparency</div></blockquote>
<blockquote>
<div><dl class="docutils">
<dt>Axes for 3D view type. If ihbdfl_bAxesIsClusterOnly == 1 - axes have cluster range</dt>
<dd>0 - axes have brain range</dd>
</dl>
</div></blockquote>
<blockquote>
<div>ihbdfl_bAxesIsClusterOnly</div></blockquote>
<blockquote>
<div>Colors defaults</div></blockquote>
<blockquote>
<div>ihbdfl_color_clOrig
ihbdfl_color_clSymm
ihbdfl_color_surface
ihbdfl_color_cntAx
ihbdfl_color_cntFr
ihbdfl_color_cntSag</div></blockquote>
<blockquote>
<div>Line width</div></blockquote>
<blockquote>
<div>ihbdfl_line_width</div></blockquote>
<blockquote>
<div></div></blockquote>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.tor_ihb_UpdateClusterTalVoxSize">
<code class="descclassname">Visualization_functions.</code><code class="descname">tor_ihb_UpdateClusterTalVoxSize</code><span class="sig-paren">(</span><em>clIn</em>, <em>nTotalVox</em>, <em>nTotalProcessed</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.tor_ihb_UpdateClusterTalVoxSize" title="Permalink to this definition">¶</a></dt>
<dd><table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">clOut</span> <span class="o">=</span> <span class="n">ihb_ChangeTalVoxSize</span><span class="p">(</span><span class="n">clIn</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>clIn:</strong></dt>
<dd><p class="first last">initial input cluster (structure see ihb_GetClusters)</p>
</dd>
<dt><strong>nTotalVox:</strong></dt>
<dd><p class="first last">total nmb of voxel in all clusters (used for waitbar)</p>
</dd>
<dt><strong>nTotalProcessed:</strong></dt>
<dd><p class="first last">total nmb of voxel in already processed clusters (used for waitbar)</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Output:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>clOut:</strong></dt>
<dd><p class="first last">output cluster with new Talariach related fields</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p>It is assumed that waitbar already exists</p>
<p>Load current voxel size
..</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.tor_polar_plot">
<code class="descclassname">Visualization_functions.</code><code class="descname">tor_polar_plot</code><span class="sig-paren">(</span><em>vals</em>, <em>colors</em>, <em>names</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.tor_polar_plot" title="Permalink to this definition">¶</a></dt>
<dd><p>Make polar line plot(s)</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">hh</span> <span class="o">=</span> <span class="n">tor_polar_plot</span><span class="p">(</span><span class="n">vals</span><span class="p">,</span> <span class="n">colors</span><span class="p">,</span> <span class="n">names</span><span class="p">,</span> <span class="p">[</span><span class="s">&#39;nofigure&#39;</span><span class="p">])</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>vals:</strong></dt>
<dd><p class="first">cell array, one cell per plot</p>
<p class="last">in each cell, matrix of observations x variables
plots one line for each variable.</p>
</dd>
<dt><strong>names:</strong></dt>
<dd><p class="first">is cell array, one cell per plot</p>
<p class="last">contains cell array of names for each condition</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><dl class="first docutils">
<dt><strong>&#8216;nofigure&#8217;:</strong></dt>
<dd><p class="first last">suppress figure</p>
</dd>
<dt><strong>&#8216;nonneg&#8217;:</strong></dt>
<dd><p class="first last">make all values non-negative by subtracting min value from all values in series (plot)</p>
</dd>
<dt><strong>&#8216;nofill&#8217;:</strong></dt>
<dd><p class="first last">Do not fill in polygons</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>hh:</strong></dt>
<dd><p class="first last">Handles to line objects</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">tor_polar_plot</span><span class="p">({</span><span class="n">w</span><span class="o">+</span><span class="mi">1</span><span class="p">},</span> <span class="p">{</span><span class="s">&#39;r&#39;</span> <span class="s">&#39;b&#39;</span><span class="p">},</span> <span class="n">setnames</span><span class="p">(</span><span class="mi">1</span><span class="p">))</span>
</pre></div>
</div>
<p>Note: Dark grey inner line is zero point if nonneg option is used.
Otherwise, zero is the origin.</p>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.violinplot">
<code class="descclassname">Visualization_functions.</code><code class="descname">violinplot</code><span class="sig-paren">(</span><em>Y</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.violinplot" title="Permalink to this definition">¶</a></dt>
<dd><p>Simple violin plot using matlab default kernel density estimation</p>
<p>This function creates violin plots based on kernel density estimation
using ksdensity with default settings. Please be careful when comparing pdfs
estimated with different bandwidth!</p>
<p>Differently to other boxplot functions, you may specify the x-position.
This is particularly usefule when overlaying with other data / plots.</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Input:</th><td class="field-body"><dl class="first last docutils">
<dt><strong>Y:</strong></dt>
<dd><p class="first last">Data to be plotted, being either
n x m matrix. A &#8216;violin&#8217; is plotted for each column m, OR
1 x m Cellarry with elements being numerical colums of nx1 length.</p>
</dd>
</dl>
</td>
</tr>
</tbody>
</table>
<p><strong>varargin:</strong></p>
<blockquote>
<div><dl class="docutils">
<dt><strong>xlabel:</strong></dt>
<dd>xlabel. Set either [] or in the form {&#8216;txt1&#8217;,&#8217;txt2&#8217;,&#8217;txt3&#8217;,...}</dd>
<dt><strong>facecolor=[1 0.5 0]:</strong></dt>
<dd>FaceColor: Specify abbrev. or m x 3 matrix (e.g. [1 0 0])</dd>
<dt><strong>edgecolor=&#8217;k&#8217;:</strong></dt>
<dd>LineColor: Specify abbrev. (e.g. &#8216;k&#8217; for black); set either [],&#8217;&#8217; or &#8216;none&#8217; if the mean should not be plotted</dd>
<dt><strong>linewidth=2:</strong></dt>
<dd>Linewidth for boundary of violin plot</dd>
<dt><strong>facealpha=0.5:</strong></dt>
<dd>Alpha value (transparency)</dd>
<dt><strong>mc=&#8217;k&#8217;:</strong></dt>
<dd>Color of the bars indicating the mean; set either [],&#8217;&#8217; or &#8216;none&#8217; if the mean should not be plotted</dd>
<dt><strong>medc=&#8217;r&#8217;:</strong></dt>
<dd>Color of the bars indicating the median; set either [],&#8217;&#8217; or &#8216;none&#8217; if the mean should not be plotted</dd>
<dt><strong>bw=[];:</strong></dt>
<dd><dl class="first last docutils">
<dt>Kernel bandwidth, prescribe if wanted.</dt>
<dd>%If b is a single number, b will be applied to all estimates
%If b is an array of 1xm or mx1, b(i) will be applied to
column (i).</dd>
</dl>
</dd>
<dt><strong>&#8216;x&#8217;:</strong></dt>
<dd>followed by x position for center(s) of plots</dd>
<dt><strong>&#8216;nopoints&#8217;:</strong></dt>
<dd>don&#8217;t display dots</dd>
<dt><strong>&#8216;pointsize&#8217;:</strong></dt>
<dd>you can define point size.</dd>
<dt><strong>&#8216;pointcolor&#8217;:</strong></dt>
<dd>if you want to use different color for points, you can use this
option.</dd>
<dt><strong>&#8216;weights&#8217;:</strong></dt>
<dd>You can add weights in the same format of the data. It will
use weighted version of kdensity function, show weighted mean, and
weighted point sizes. Be careful: if some data points have NaNs,
this will remove those data points from the plot.</dd>
</dl>
</div></blockquote>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Outputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>h:</strong></dt>
<dd><p class="first last">figure handle</p>
</dd>
<dt><strong>L:</strong></dt>
<dd><p class="first last">Legend handle</p>
</dd>
<dt><strong>MX:</strong></dt>
<dd><p class="first last">Means of groups</p>
</dd>
<dt><strong>MED:</strong></dt>
<dd><p class="first last">Medians of groups</p>
</dd>
<dt><strong>bw:</strong></dt>
<dd><p class="first last">bandwidth of kernel</p>
</dd>
<dt><strong>pointloc:</strong></dt>
<dd><p class="first last">point locations on violin plot
This has x, y, idx as subfields. 
idx indicates the order of the data in the original input.</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Example1 (default):</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">disp</span><span class="p">(</span><span class="s">&#39;this example uses the statistical toolbox&#39;</span><span class="p">)</span>
<span class="n">Y</span><span class="o">=</span><span class="p">[</span><span class="n">rand</span><span class="p">(</span><span class="mi">1000</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span><span class="n">gamrnd</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1000</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span><span class="n">normrnd</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">1000</span><span class="p">,</span><span class="mi">1</span><span class="p">),</span><span class="n">gamrnd</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span><span class="mf">0.1</span><span class="p">,</span><span class="mi">1000</span><span class="p">,</span><span class="mi">1</span><span class="p">)];</span>
<span class="p">[</span><span class="n">h</span><span class="p">,</span><span class="n">L</span><span class="p">,</span><span class="n">MX</span><span class="p">,</span><span class="n">MED</span><span class="p">]</span><span class="o">=</span><span class="n">violinplot</span><span class="p">(</span><span class="n">Y</span><span class="p">);</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&#39;\Delta [yesno^{-2}]&#39;</span><span class="p">,</span><span class="s">&#39;FontSize&#39;</span><span class="p">,</span><span class="mi">14</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Example2 (specify facecolor, edgecolor, xlabel):</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>disp(&#39;this example uses the statistical toolbox&#39;)
Y=[rand(1000,1),gamrnd(1,2,1000,1),normrnd(10,2,1000,1),gamrnd(10,0.1,1000,1)];
violinplot(Y,&#39;xlabel&#39;,{&#39;a&#39;,&#39;b&#39;,&#39;c&#39;,&#39;d&#39;},&#39;facecolor&#39;,[1 1 0;0 1 0;.3 .3 .3;0 0.3 0.1],&#39;edgecolor&#39;,&#39;b&#39;,...
           &#39;bw&#39;,0.3,...
           &#39;mc&#39;,&#39;k&#39;,...
           &#39;medc&#39;,&#39;r--&#39;)
ylabel(&#39;\Delta [yesno^{-2}]&#39;,&#39;FontSize&#39;,14)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Example3 (specify x axis location):</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>disp(&#39;this example uses the statistical toolbox&#39;)
Y=[rand(1000,1),gamrnd(1,2,1000,1),normrnd(10,2,1000,1),gamrnd(10,0.1,1000,1)];
violinplot(Y,&#39;x&#39;,[-1 .7 3.4 8.8],&#39;facecolor&#39;,[1 1 0;0 1 0;.3 .3 .3;0 0.3 0.1],&#39;edgecolor&#39;,&#39;none&#39;,...
           &#39;bw&#39;,0.3,&#39;mc&#39;,&#39;k&#39;,&#39;medc&#39;,&#39;r-.&#39;)
axis([-2 10 -0.5 20])
ylabel(&#39;\Delta [yesno^{-2}]&#39;,&#39;FontSize&#39;,14)
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name" colspan="2">Example4 (Give data as cells with different n):</th></tr>
<tr class="field-odd field"><td>&nbsp;</td><td class="field-body"></td>
</tr>
</tbody>
</table>
<p>disp(&#8216;this example uses the statistical toolbox&#8217;)</p>
<div class="highlight-python"><div class="highlight"><pre>Y{:,1}=rand(10,1);
Y{:,2}=rand(1000,1);
violinplot(Y,&#39;facecolor&#39;,[1 1 0;0 1 0;.3 .3 .3;0 0.3 0.1],&#39;edgecolor&#39;,&#39;none&#39;,&#39;bw&#39;,0.1,&#39;mc&#39;,&#39;k&#39;,&#39;medc&#39;,&#39;r-.&#39;)
ylabel(&#39;\Delta [yesno^{-2}]&#39;,&#39;FontSize&#39;,14)
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.wani_pie">
<code class="descclassname">Visualization_functions.</code><code class="descname">wani_pie</code><span class="sig-paren">(</span><em>X</em>, <em>varargin</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.wani_pie" title="Permalink to this definition">¶</a></dt>
<dd><p>Draw a little better pie chart</p>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Usage:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre><span class="n">h</span> <span class="o">=</span> <span class="n">wani_pie</span><span class="p">(</span><span class="n">X</span><span class="p">,</span> <span class="n">varargin</span><span class="p">)</span>
</pre></div>
</div>
<table class="docutils field-list" frame="void" rules="none">
<col class="field-name" />
<col class="field-body" />
<tbody valign="top">
<tr class="field-odd field"><th class="field-name">Inputs:</th><td class="field-body"><dl class="first docutils">
<dt><strong>X:</strong></dt>
<dd><p class="first last">a vector</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name" colspan="2">Optional Inputs:</th></tr>
<tr class="field-even field"><td>&nbsp;</td><td class="field-body"><p class="first">Enter keyword followed by variable with values</p>
<dl class="docutils">
<dt><strong>&#8216;cols&#8217; or &#8216;colors&#8217;:</strong></dt>
<dd><p class="first last">colors N x 3 {default: using colors from microsoft office}
OR cell array of 3-element colors (no text input for colors!)</p>
</dd>
<dt><strong>&#8216;notext&#8217;:</strong></dt>
<dd><p class="first last">no text for percentage {default: false}</p>
</dd>
<dt><strong>&#8216;fontsize&#8217;:</strong></dt>
<dd><p class="first last">font size for percentage {default: 15}</p>
</dd>
<dt><strong>&#8216;hole&#8217;:</strong></dt>
<dd><p class="first last">add a hole in the middle of the pie chart {default: no hole}</p>
</dd>
<dt><strong>&#8216;hole_size&#8217;:</strong></dt>
<dd><p class="first last">specify the size of the middle hole {default: 5000}</p>
</dd>
</dl>
<p><strong>&#8216;outline&#8217;</strong></p>
<p><strong>&#8216;outlinecol&#8217;</strong></p>
<p><strong>&#8216;outlinewidth&#8217;</strong></p>
</td>
</tr>
<tr class="field-odd field"><th class="field-name">Output:</th><td class="field-body"><dl class="first docutils">
<dt><strong>h:</strong></dt>
<dd><p class="first last">graphic handles</p>
</dd>
</dl>
</td>
</tr>
<tr class="field-even field"><th class="field-name">Examples:</th><td class="field-body"></td>
</tr>
</tbody>
</table>
<div class="highlight-python"><div class="highlight"><pre>% data
X = rand(10,1);
h = wani_pie(X, &#39;notext&#39;, &#39;hole&#39;)

savename = &#39;example_pie.pdf&#39;;

try
    pagesetup(gcf);
    saveas(gcf, savename);
catch
    pagesetup(gcf);
    saveas(gcf, savename);
end
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Visualization_functions.xval_lasso_brain_permutation_histogram">
<code class="descclassname">Visualization_functions.</code><code class="descname">xval_lasso_brain_permutation_histogram</code><span class="sig-paren">(</span><em>stats</em><span class="sig-paren">)</span><a class="headerlink" href="#Visualization_functions.xval_lasso_brain_permutation_histogram" title="Permalink to this definition">¶</a></dt>
<dd><p>Plot histograms and get permutation test-based p-value for xval_lasso_brain output structure</p>
<p>xval_lasso_brain_permutation_histogram(stats)</p>
</dd></dl>

</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table Of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Object Oriented Tools</a><ul>
<li><a class="reference internal" href="#module-&#64;canlab_dataset">canlab_dataset</a></li>
<li><a class="reference internal" href="#module-&#64;fmri_data">fmri_data</a></li>
<li><a class="reference internal" href="#module-&#64;fmri_mask_image">fmri_mask_image</a></li>
<li><a class="reference internal" href="#module-&#64;fmri_model">fmri_model</a></li>
<li><a class="reference internal" href="#module-&#64;fmridisplay">fmridisplay</a></li>
<li><a class="reference internal" href="#fmridisplay-helper-functions">fmridisplay_helper_functions</a></li>
<li><a class="reference internal" href="#module-&#64;image_vector">image_vector</a></li>
<li><a class="reference internal" href="#module-&#64;region">region</a></li>
<li><a class="reference internal" href="#module-&#64;statistic_image">statistic_image</a></li>
</ul>
</li>
<li><a class="reference internal" href="#toolboxes">Toolboxes</a><ul>
<li><a class="reference internal" href="#cluster-contig-region-tools">Cluster_contig_region_tools</a></li>
<li><a class="reference internal" href="#module-diagnostics">diagnostics</a></li>
<li><a class="reference internal" href="#module-GLM_Batch_tools">GLM_Batch_tools</a></li>
<li><a class="reference internal" href="#hrf-est-toolbox2">HRF_Est_Toolbox2</a></li>
<li><a class="reference internal" href="#optimizedesign11">OptimizeDesign11</a></li>
</ul>
</li>
<li><a class="reference internal" href="#miscellaneous-tools">Miscellaneous Tools</a><ul>
<li><a class="reference internal" href="#module-Data_extraction">Data_extraction</a></li>
<li><a class="reference internal" href="#data-processing-tools">Data_processing_tools</a></li>
<li><a class="reference internal" href="#module-Filename_tools">Filename_tools</a></li>
<li><a class="reference internal" href="#module-hewma_utility">hewma_utility</a></li>
<li><a class="reference internal" href="#image-computation-tools">Image_computation_tools</a></li>
<li><a class="reference internal" href="#image-space-tools">Image_space_tools</a></li>
<li><a class="reference internal" href="#image-thresholding">Image_thresholding</a></li>
<li><a class="reference internal" href="#index-image-manip-tools">Index_image_manip_tools</a></li>
<li><a class="reference internal" href="#module-Misc_utilities">Misc_utilities</a></li>
<li><a class="reference internal" href="#model-building-tools">Model_building_tools</a></li>
<li><a class="reference internal" href="#parcellation-tools">Parcellation_tools</a></li>
<li><a class="reference internal" href="#peak-coordinates">peak_coordinates</a></li>
<li><a class="reference internal" href="#roi-drawing-tools">ROI_drawing_tools</a></li>
<li><a class="reference internal" href="#statistics-tools">Statistics_tools</a></li>
<li><a class="reference internal" href="#visualization-functions">Visualization_functions</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">CanlabCore documentation home</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/moduleslist.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="mat-modindex.html" title="MATLAB Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="index.html" title="CanlabCore documentation home"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">CanlabCore 1.0 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, Tor Wager.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3.1.
    </div>
  </body>
</html>